<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.43">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jason Geller">
<meta name="author" content="Robert Kubinec">
<meta name="author" content="Chelsea M. Parlett Pelleriti">
<meta name="author" content="Matti Vuorre">
<meta name="keywords" content="Beta regression, Beta distribution, R, Tutorial, Psychology, Learning and memory">
<meta name="description" content="BETA REGRESSION TUTORIAL">

<title>A Beta Way: A Tutorial For Using Beta Regression in Psychological Research</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="ms_files/libs/clipboard/clipboard.min.js"></script>
<script src="ms_files/libs/quarto-html/quarto.js"></script>
<script src="ms_files/libs/quarto-html/popper.min.js"></script>
<script src="ms_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="ms_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="ms_files/libs/quarto-html/quarto-syntax-highlighting-d4d76bf8491c20bad77d141916dc28e1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="ms_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="ms_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="ms_files/libs/bootstrap/bootstrap-9ac97cd188ec939c181fdab479d986e2.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>

  <script>window.backupDefine = window.define; window.define = undefined;</script><script src="https://cdn.jsdelivr.net/npm/katex@latest/dist/katex.min.js"></script>
  <script>document.addEventListener("DOMContentLoaded", function () {
 var mathElements = document.getElementsByClassName("math");
 var macros = [];
 for (var i = 0; i < mathElements.length; i++) {
  var texText = mathElements[i].firstChild;
  if (mathElements[i].tagName == "SPAN") {
   katex.render(texText.data, mathElements[i], {
    displayMode: mathElements[i].classList.contains('display'),
    throwOnError: false,
    macros: macros,
    fleqn: false
   });
}}});
  </script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@latest/dist/katex.min.css">

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="_extensions/apaquarto/apa.css">
</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Manuscript Outline</h2>
   
  <ul class="collapse">
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
  <li><a href="#a-beta-way-is-possible" id="toc-a-beta-way-is-possible" class="nav-link" data-scroll-target="#a-beta-way-is-possible">A Beta Way Is Possible</a></li>
  <li><a href="#beta-distribution" id="toc-beta-distribution" class="nav-link" data-scroll-target="#beta-distribution">Beta Distribution</a></li>
  <li><a href="#i-cant-believe-its-not-beta" id="toc-i-cant-believe-its-not-beta" class="nav-link" data-scroll-target="#i-cant-believe-its-not-beta">I Can’t Believe It’s Not Beta</a></li>
  <li><a href="#bayesian-approach-to-beta-regression" id="toc-bayesian-approach-to-beta-regression" class="nav-link" data-scroll-target="#bayesian-approach-to-beta-regression">Bayesian Approach to Beta Regression</a></li>
  </ul></li>
  <li><a href="#beta-regression-tutorial" id="toc-beta-regression-tutorial" class="nav-link" data-scroll-target="#beta-regression-tutorial">Beta Regression Tutorial</a>
  <ul class="collapse">
  <li><a href="#example-data" id="toc-example-data" class="nav-link" data-scroll-target="#example-data">Example Data</a></li>
  <li><a href="#reanalysis-of-wilford-et-al.-experiment-1a" id="toc-reanalysis-of-wilford-et-al.-experiment-1a" class="nav-link" data-scroll-target="#reanalysis-of-wilford-et-al.-experiment-1a">Reanalysis of Wilford et al.&nbsp;Experiment 1A</a></li>
  <li><a href="#beta-regression" id="toc-beta-regression" class="nav-link" data-scroll-target="#beta-regression">Beta Regression</a></li>
  <li><a href="#zero-inflated-beta-zib-regression" id="toc-zero-inflated-beta-zib-regression" class="nav-link" data-scroll-target="#zero-inflated-beta-zib-regression">Zero-Inflated Beta (ZIB) Regression</a></li>
  <li><a href="#zero-one-inflated-beta-zoib" id="toc-zero-one-inflated-beta-zoib" class="nav-link" data-scroll-target="#zero-one-inflated-beta-zoib">Zero-One-Inflated Beta (ZOIB)</a></li>
  <li><a href="#ordered-beta-regression" id="toc-ordered-beta-regression" class="nav-link" data-scroll-target="#ordered-beta-regression">Ordered Beta Regression</a></li>
  <li><a href="#marginal-effects-1" id="toc-marginal-effects-1" class="nav-link" data-scroll-target="#marginal-effects-1">Marginal Effects</a></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion">Discussion</a>
  <ul class="collapse">
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion">Conclusion</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="ms.pdf"><i class="bi bi-file-pdf"></i>PDF (apaquarto)</a></li><li><a href="ms.docx"><i class="bi bi-file-word"></i>MS Word (apaquarto)</a></li></ul></div></nav>
</div>
<main class="content" id="quarto-document-content"><div class="quarto-title-block"><div class="quarto-title-tools-only"><h1></h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>




<br>

<br>

<section id="title" class="level1 title unnumbered unlisted">
<h1 class="title unnumbered unlisted">A Beta Way: A Tutorial For Using Beta Regression in Psychological Research</h1>
<div class="Author">
<br>

<p>Jason Geller<sup>1</sup>, Robert Kubinec<sup>2</sup>, Chelsea M. Parlett Pelleriti<sup>3</sup>, and Matti Vuorre<sup>4</sup></p>
<p><sup>1</sup>Department of Psychology and Neuroscience, Boston College</p>
<p><sup>2</sup>University of South Carolina</p>
<p><sup>3</sup>Canva</p>
<p><sup>4</sup>Tilburg University</p>
</div>
<br>

<br>

</section>
<section id="author-note" class="level1 unnumbered unlisted AuthorNote">
<h1 class="unnumbered unlisted AuthorNote">Author Note</h1>
<p>Jason Geller <img src="_extensions/apaquarto/ORCID-iD_icon-vector.svg" id="orchid" class="img-fluid" style="width:4.23mm" alt="Orcid ID Logo: A green circle with white letters ID"> <a href="https://orcid.org/0000-0002-7459-4505">https://orcid.org/0000-0002-7459-4505</a></p>
<p>Robert Kubinec <img src="_extensions/apaquarto/ORCID-iD_icon-vector.svg" id="orchid" class="img-fluid" style="width:4.23mm" alt="Orcid ID Logo: A green circle with white letters ID"> <a href="https://orcid.org/0000-0001-6655-4119">https://orcid.org/0000-0001-6655-4119</a></p>
<p>Chelsea M. Parlett Pelleriti <img src="_extensions/apaquarto/ORCID-iD_icon-vector.svg" id="orchid" class="img-fluid" style="width:4.23mm" alt="Orcid ID Logo: A green circle with white letters ID"> <a href="https://orcid.org/0000-0001-9301-1398">https://orcid.org/0000-0001-9301-1398</a></p>
<p>Matti Vuorre <img src="_extensions/apaquarto/ORCID-iD_icon-vector.svg" id="orchid" class="img-fluid" style="width:4.23mm" alt="Orcid ID Logo: A green circle with white letters ID"> <a href="https://orcid.org/0000-0001-5052-066X">https://orcid.org/0000-0001-5052-066X</a></p>
<p>No preregistration. Data, code, and materials for this manuscript can be found at https://github.com/jgeller112/Beta_regression_tutorial The authors have no conflicts of interest to disclose.</p>
<p>Author roles were classified using the Contributor Role Taxonomy (CRediT; https://credit.niso.org/) as follows: <em>Jason Geller</em><strong>: </strong>conceptualization, data curation, formal analysis, project administration, resources, visualization, and writing – original draft. <em>Robert Kubinec</em><strong>: </strong>formal analysis, validation, and Writing - review &amp; editing. <em>Chelsea M. Parlett Pelleriti</em><strong>: </strong>formal analysis and Writing - review &amp; editing. <em>Matti Vuorre</em><strong>: </strong>formal analysis, resources, supervision, validation, and Writing - review &amp; editing</p>
<p>Correspondence concerning this article should be addressed to Jason Geller, Department of Psychology and Neuroscience, Boston College, McGuinn 300z, Chestnut Hill, MA 2467, USA, Email: <a href="mailto:drjasongeller@gmail.com">drjasongeller@gmail.com</a></p>
<br>
</section>
<section id="abstract" class="level1 unnumbered unlisted AuthorNote">
<h1 class="unnumbered unlisted AuthorNote">Abstract</h1>
<div class="AbstractFirstParagraph">
<p>Rates, percentages, and proportions are common outcomes in psychology and the social sciences. These outcomes are often analyzed using models that assume normality, but this practice overlooks important features of the data, such as their natural bounds at 0 and 1. As a result, estimates can become distorted. In contrast, treating such outcomes as Beta-distributed respects these limits and can yield more accurate estimates. Despite these advantages, the use of Beta models in applied research remains limited. Our goal is to provide researchers with practical guidance for adopting Beta regression models, illustrated with an example drawn from the psychological literature. We begin by introducing the Beta distribution and Beta regression, emphasizing key components and assumptions. Next, using data from a learning and memory study, we demonstrate how to fit a Beta regression model in R with the Bayesian package <code>brms</code> and how to interpret results on the response scale. We also discuss model extensions, including zero-inflated, zero- and one-inflated, and ordered Beta models. Basic familarity with regression modeling and R is assumed. To promote wider adoption of these methods, we provide detailed code and materials at https://github.com/jgeller112/Beta_regression_tutorial.</p>
</div>
<p><em>Keywords</em>: Beta regression, Beta distribution, R, Tutorial, Psychology, Learning and memory</p>
<p><em>Word Count</em>: 24579</p>
<br>
</section>
<section id="firstheader" class="level1 title unnumbered unlisted">
<h1 class="title unnumbered unlisted">A Beta Way: A Tutorial For Using Beta Regression in Psychological Research</h1>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Preprint
</div>
</div>
<div class="callout-body-container callout-body">
<p>This manuscript is currently <strong>under review</strong> and has not been peer-reviewed. Content is <strong>subject to change</strong>. Please feel free to provide feedback!</p>
</div>
</div>
</section>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>Many key outcomes in psychological research are naturally expressed as proportions or percentages. Examples include the proportion of correct responses on a test <span class="citation" data-cites="kornell2008">(e.g., Kornell &amp; Bjork, 2008)</span>, the proportion of time a participant fixates on a particular stimulus in an eye-tracking task <span class="citation" data-cites="james2025">(e.g., James et al., 2025)</span>, or the proportion of respondents agreeing with a given statement or belief <span class="citation" data-cites="Costello2024">(e.g., Costello et al., 2024)</span>. Consider, for instance, a memory experiment in which participants read a short passage, complete a brief distractor task, and then take a final memory test consisting of 10 short‑answer questions. If each question carries a different point value (e.g., question 1 is worth 4 points, question 2 is worth 1 point), a meaningful outcome measure could be the proportion of points earned for each question relative to its maximum possible value.</p>
<p>A key question arises from this example: how should proportional outcomes be analyzed? Researchers frequently default to linear models that assume Gaussian (normal) distributions, such as t-tests, ANOVAs, and linear regression. However, these models make strong assumptions: (1) residuals are normally distributed, (2) the outcome is unbounded (from <span class="math inline">-\infty</span> to <span class="math inline">\infty</span>), and (3) variance is constant across the range of the data. These assumptions are rarely satisfied in practice <span class="citation" data-cites="sladekova2024unicorns">(Sladekova &amp; Field, 2024)</span>, and they are especially ill-suited for proportional outcomes, which are bounded between 0 and 1 and often exhibit heteroscedasticity—non-constant variance, particularly near the boundaries <span class="citation" data-cites="smithson2006 ferrari2004 paolino2001">(Ferrari &amp; Cribari-Neto, 2004; Paolino, 2001; Smithson &amp; Verkuilen, 2006)</span>. Violating these assumptions can lead to biased estimates and spurious inferences, especially when many observations cluster near 0 or 1.</p>
<p>In some cases, it is possible to use a generalized version of the linear model (GLM) that relaxes the assumption of normality. For instance, binomial and Bernoulli models—often referred to as logistic regression when paired with a logit link—are well-suited for binary outcomes (e.g., 0 or 1) or success counts out of a fixed number of trials. However, these models require discretized data and may perform poorly when the data are continuous proportions or exhibit excess variability (i.e., overdispersion), especially near the scale’s boundaries.</p>
<p>The challenges of analyzing proportional data are not new <span class="citation" data-cites="bartlett1936">(see Bartlett, 1936)</span>. Fortunately, several existing approaches address the limitations of commonly used models. One such approach is Beta regression, an extension of the generalized linear model that employs the Beta distribution <span class="citation" data-cites="ferrari2004 paolino2001">(Ferrari &amp; Cribari-Neto, 2004; Paolino, 2001)</span>. Beta regression offers a flexible and robust solution for modeling proportional data directly by accounting for boundary effects and over-dispersion, making it a valuable alternative to traditional binomial models. This approach is particularly well-suited for psychological research because it can handle both the bounded nature of proportional data and the non-constant variance often encountered in these datasets <span class="citation" data-cites="sladekova2024unicorns">(Sladekova &amp; Field, 2024)</span>. In addition, the direct modeling of proportions allows comparability across tasks and scales, and can be particularly valuable when only the proportional data is available, as is often the case with secondary data that lack item-level structure or point values.</p>
<p>While in this paper we will focus on proportional-responses that lie between 0 and 1–it is important to note that our analysis applies to any bounded continuous scale. Any bounded scale can be mapped to lie within 0 and 1 without resulting in a loss of information as the transformation is linear.<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a> Consequently, a scale that has natural end points of -1,234 and +8,451–or any other end points on the real number line short of infinity—can be modeled using the approaches we describe in this paper.</p>
<section id="a-beta-way-is-possible" class="level2">
<h2 data-anchor-id="a-beta-way-is-possible">A Beta Way Is Possible</h2>
<p>With the combination of open-source programming languages like R <span class="citation" data-cites="R">(R Core Team, 2024)</span> and their user-developed extensions, analyses such as Beta regression have become increasingly accessible. Yet, adoption of these methods–particularly in psychology–remains limited. One reason may be the lack of informative examples that directly apply to psychological research. Although recent years have seen a surge of interest in Beta regression <span class="citation" data-cites="smithson2006 heiss2021 vuorre2019 bendixen2023 coretta2025bayesian">(Bendixen &amp; Purzycki, 2023; Coretta &amp; Bürkner, 2025; Heiss, 2021; Smithson &amp; Verkuilen, 2006; Vuorre, 2019)</span>, its adoption in psychology remains limited.</p>
<p>While previous tutorials have discussed Beta regression, most have been limited in scope–focusing either on the basic model or offering only brief mentions of more applicable alternatives. This tutorial aims to fill that gap by offering a comprehensive and practical tutorial of Beta regression and its extensions. In addition to covering the standard Beta model, we walk through its extensions such as zero-inflated, zero-one-inflated, and ordered Beta regression. These models are important for researchers dealing with boundary values (e.g., exact 0s or 1s) or ordinal response structures.</p>
<p>Beyond model specification, we place strong emphasis on interpreting results on the response scale–that is, in terms of probabilities and proportions–rather than relying on often difficult to interpret parameters. This focus makes the models more accessible and meaningful for psychological applications, where effects are often easier to communicate when framed on the original scale of the outcome (e.g., changes in recall accuracy or task performance). Throughout, we provide reproducible code and annotated examples to help readers implement and interpret these models in their own work.</p>
<p>We begin the tutorial with a non-technical overview of the Beta distribution and its core parameters. We then walk through the process of estimating Beta regression models using the R package <code>brms</code> <span class="citation" data-cites="brms">(Bürkner, 2017)</span>, illustrating each step with applied examples. To guide interpretation, we emphasize coefficients, predicted probabilities, and marginal effects calculated using the <code>marginaleffects</code> package <span class="citation" data-cites="margeffects2024">(Arel-Bundock et al., 2024)</span>. We also introduce several useful extensions–zero-inflated (ZIB), zero-one-inflated (ZOIB), and ordered Beta regression–that enable researchers to model outcomes that include boundary values. Finally, all code and materials used in this tutorial are fully reproducible and available via our GitHub repository: <a href="https://github.com/jgeller112/Beta_regression_tutorial" class="uri">https://github.com/jgeller112/Beta_regression_tutorial</a><a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>.</p>
</section>
<section id="beta-distribution" class="level2">
<h2 data-anchor-id="beta-distribution">Beta Distribution</h2>
<p>Proportional data pose some challenges for standard modeling approaches: The data are bounded between 0 and 1 and often exhibit non-constant variance (heteroscedasticity) <span class="citation" data-cites="ferrari2004 paolino2001">(Ferrari &amp; Cribari-Neto, 2004; Paolino, 2001)</span>. Common distributions used within the generalized linear model frameworks often fail to capture these properties adequately, which can necessitate alternative modeling strategies.</p>
<p>While we do not have time to delve fully into its derivation, the Beta distribution is a preferred distribution for this type of response because of certain unique properties. The Beta distribution is defined as a distribution of the uncertainty of probabilities, which must lie within 0 and 1. As a consequence, the Beta distribution is the maximum entropy distribution for any bounded continuous random variable, which means that the Beta distribution can represent the full range of possibilities of such a scale.<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a> As a consequence, if we have a continuous scale with upper and lower bounds–and no other special conditions–the Beta distribution will in principle provide a very good approximation of the uncertainty of the scale.</p>
<p>Typically, the expected value (or mean) of the response variable is the central estimand scholars want to estimate. A model should specify how this expected value depends on explanatory variables through two main components: a linear predictor, which combines the explanatory variables in a linear form (<span class="math inline">a+b_1x_1+b_2x_2</span>, etc.), and a link function, which connects the expected value of the response variable to the linear predictor (e.g, <span class="math inline">E[Y] = g(a + b_1x_1 + b_2x_2)</span>). In addition, a random component specifies the distribution of the response variable around its expected value (such as Poisson or binomial distributions, which belong to the exponential family) <span class="citation" data-cites="nelder1972">(Nelder &amp; Wedderburn, 1972)</span>. Together, these components provide a flexible framework for modeling data with different distributional properties.</p>
<p>The Beta distribution is continuous and restricted to values between 0 and 1 (exclusive). Its two parameters–commonly called shape1 (<span class="math inline">\alpha</span>) and shape2 (<span class="math inline">\beta</span>)–govern the distribution’s location, skewness, and spread. By adjusting these parameters, the distribution can take many functional forms (e.g., it can be symmetric, skewed, U-shaped, or even approximately uniform; see <a href="#fig-curves" class="quarto-xref" aria-expanded="false">Figure&nbsp;1</a>).</p>
<p>To illustrate, consider a test question worth seven points. Suppose a participant scores five out of seven. The number of points received (5) can be treated as <span class="math inline">\alpha</span>, and the number of points missed (2) as <span class="math inline">\beta</span>. The resulting Beta distribution would be skewed toward higher values, reflecting a high performance (yellow line in <a href="#fig-curves" class="quarto-xref" aria-expanded="false">Figure&nbsp;1</a>; “Beta(5, 2)”). Reversing these values would produce a distribution skewed toward lower values, representing poorer performance (green line in <a href="#fig-curves" class="quarto-xref" aria-expanded="false">Figure&nbsp;1</a>; “Beta(2, 5)”).</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-curves" class="quarto-float quarto-figure quarto-figure-center" prefix="" data-fignum="1">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-curves-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;1</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Beta distributions with different shape1 and shape2 parameters.</p>
</div>
</figcaption>
<div aria-describedby="fig-curves-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-curves-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
</section>
<section id="i-cant-believe-its-not-beta" class="level2">
<h2 data-anchor-id="i-cant-believe-its-not-beta">I Can’t Believe It’s Not Beta</h2>
<p>While the standard parameterization of the Beta distribution uses <span class="math inline">\alpha</span> and <span class="math inline">\beta</span>, a reparameterization to a mean (<span class="math inline">\mu</span>) and precision (<span class="math inline">\phi</span>) is more useful for regression models. The mean represents the expected value of the distribution, while the dispersion, which is inversely related to variance, reflects how concentrated the distribution is around the mean, with higher values indicating a narrower distribution and lower values indicating a wider one. The connections between the Beta distribution’s parameters are shown in <a href="#eq-beta" class="quarto-xref">Equation&nbsp;1</a>. Importantly, the variance depends on the average value of the response because uncertainty intervals need to adjust for how close the value of the response is to the boundary.</p>
<p><span id="eq-beta"><span class="math display">
\begin{aligned}[t]
\text{Shape 1:} &amp;&amp; a &amp;= \mu \phi \\
\text{Shape 2:} &amp;&amp; b &amp;= (1 - \mu) \phi
\end{aligned}
\qquad\qquad\qquad
\begin{aligned}[t]
\text{Mean:} &amp;&amp; \mu &amp;= \frac{a}{a + b} \\
\text{Precision:} &amp;&amp; \phi &amp;= a + b \\
\text{Variance:} &amp;&amp; var &amp;= \frac{\mu \cdot (1 - \mu)}{1 + \phi}
\end{aligned}
\tag{1}</span></span></p>
<p>Thus, Beta regression allows modeling both the mean and precision of the outcome distribution. To ensure that <span class="math inline">\mu</span> stays between 0 and 1, we apply a link function, which allows linear modeling of the mean on an unbounded scale. A common link-function choice is the logit, but other functions such as the probit or complementary log-log are possible.</p>
<p>The logit function, <span class="math inline">\text{logit}(\mu) = \log \left( \frac{\mu}{1 - \mu} \right)</span> links the mean to log-odds which are unbounded, making linear modeling possible. The inverse of the logit, called the logistic function, maps the linear predictor <span class="math inline">\eta</span> back to the original scale of the data <span class="math inline">\left(\mu = \frac{1}{1 + e^{-\eta}}\right)</span>. Similarly, the strictly positive dispersion parameter is usually modeled through a log link function, ensuring it remains positive.</p>
<p>By accounting for the observations’ natural limits and non-constant variance across different values, the Beta distribution is useful in psychology where outcomes like performance rates or response scales frequently exhibit these features.</p>
</section>
<section id="bayesian-approach-to-beta-regression" class="level2">
<h2 data-anchor-id="bayesian-approach-to-beta-regression">Bayesian Approach to Beta Regression</h2>
<p>Beta regression models can be estimated with both frequentist and Bayesian methods. We adopt a Bayesian framework because it makes estimating and interpreting more complex models easier <span class="citation" data-cites="gelman2013bayesian mcelreath2020statistical bayes_rules">(Gelman et al., 2013; Johnson et al., 2022; McElreath, 2020)</span>. Generally speaking, most Bayesian analyses can also be implemented with frequentist methods like maximum likelihood, but more complex techniques may require adjustments like bootstrapping. The main limitation of Bayesian modeling is that it is slower than frequentist approaches, but we note that modern Bayesian computation engines are reasonably fast and that explanatory modeling necessarily emphasizes deriving appropriate estimands over computational convenience. We use the R package <code>brms</code> <span class="citation" data-cites="brms">(Bürkner, 2017)</span>, a high-level interface to the probabilistic programming language Stan <span class="citation" data-cites="stan2023">(Team, 2023)</span>, because it uses standard R regression formula syntax but extends its scope while remaining accessible for non-expert users. The package also implements parallel processing that can dramatically shorten computational times for larger datasets.</p>
<p>There are several important differences between our Bayesian analysis and the frequentist methods readers may be more familiar with—most notably, the absence of <em>t</em>- and <em>p</em>-values. To estimate models, the <code>brms</code> package uses Stan’s computational algorithms to draw random samples from the posterior distribution, which represents uncertainty about the model parameters. This posterior is conceptually analogous to a frequentist sampling distribution.</p>
<p>By default, Bayesian models run 4 chains with 2,000 iterations each. The first 1,000 iterations per chain are warmup and are discarded. The remaining 1,000 iterations per chain are retained as posterior draws, yielding 4,000 total post-warmup draws across all chains. From these draws, we can compute the posterior mean (analogous to a frequentist point estimate) and the 95% credible interval (Cr.I.), which is often compared to a confidence interval. In addition, we report the <em>probability of direction</em> (pd), which reflects the probability that a parameter is strictly positive or negative. When a uniform prior is used (all values equally likely in the prior), a pd of 95%, 97.5%, 99.5%, and 99.95% corresponds approximately to two-sided p-values of .10, .05, .01, and .001 (i.e., pd ≈ 1 − p/2 for symmetric posteriors with weak/flat priors) (see <a href="#fig-p-value-pd-comp" class="quarto-xref" aria-expanded="false">Figure&nbsp;2</a> for an illustrative comparison). For directional hypotheses, the pd can be interpreted as roughly equivalent to one minus the <em>p</em>-value <span class="citation" data-cites="marsmanThreeInsightsBayesian2016">(Marsman &amp; Wagenmakers, 2016)</span>.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-p-value-pd-comp" class="quarto-float quarto-figure quarto-figure-center" prefix="" data-fignum="2">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-p-value-pd-comp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;2</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>A Bayesian posterior distribution (assuming a uniform prior) centered at a point estimate chosen so that the probability of direction (pd) equals 0.95, and a frequentist sampling distribution (under the null; centered at 0). In the Bayesian posterior distribution, the blue area represents the pd, and the red area represents the remaining 1 − pd of the distribution. In the frequentist sampling distribution, the red tail areas represent the rejection region at α = 0.10. In this example, the posterior mean lies exactly at the <span class="math inline">1 - \frac{\alpha}{2}</span> quantile of the null sampling distribution. For symmetric posteriors with flat priors, the pd is numerically equivalent to the one-sided <span class="math inline">p</span>-value.</p>
</div>
</figcaption>
<div aria-describedby="fig-p-value-pd-comp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-p-value-pd-comp-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>For reasons of space, we refer readers unfamiliar to Bayesian data analysis to several existing books on the topic <span class="citation" data-cites="gelman2013bayesian mcelreath2020statistical kruschke2015doing">(Gelman et al., 2013; Kruschke, 2015; McElreath, 2020)</span>. In addition, we assume readers are familiar with R, but those in need of a refresher should find <span class="citation" data-cites="wickhamDataScienceImport2023">Wickham et al. (2023)</span> useful.</p>
</div>
</section>
</section>
<section id="beta-regression-tutorial" class="level1">
<h1>Beta Regression Tutorial</h1>
<section id="example-data" class="level2">
<h2 data-anchor-id="example-data">Example Data</h2>
<p>Throughout this tutorial, we analyze data from a memory experiment examining whether the fluency of an instructor’s delivery affects recall performance <span class="citation" data-cites="wilford2020">(Wilford et al., 2020, Experiment 1A)</span>. Instructor fluency—marked by expressive gestures, dynamic vocal tone, and confident pacing—has been shown to influence students’ perceptions of learning, often leading learners to rate fluent instructors more favorably <span class="citation" data-cites="carpenter2013">(Carpenter et al., 2013)</span>. However, previous research suggests that these impressions do not reliably translate into improved memory performance <span class="citation" data-cites="carpenter2013 toftness2017 witherby2022">(e.g., Carpenter et al., 2013; Toftness et al., 2017; Witherby &amp; Carpenter, 2022)</span>. In contrast, <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span> found that participants actually recalled more information after watching a fluent instructor compared to a disfluent one. This surprising finding makes the dataset a compelling case study for analyzing proportion data, as recall was scored out of 10 possible idea units per video.</p>
<p>In Experiment 1A, participants watched two short instructional videos, each delivered either fluently or disfluently. Fluent videos featured instructors with smooth delivery and natural pacing, while disfluent videos included hesitations, monotone speech, and awkward pauses. After a distractor task, participants completed a free recall test, writing down as much content as they could remember from each video within a three-minute window. Their recall was then scored for the number of idea units correctly remembered.</p>
<div class="cell">
<div id="lst-data" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-data-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;1: Data needed to run examples
</figcaption>
<div aria-describedby="lst-data-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-data"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-data-1"><a href="#lst-data-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get data here from github</span></span>
<span id="lst-data-2"><a href="#lst-data-2" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> <span class="fu">str_glue</span>(</span>
<span id="lst-data-3"><a href="#lst-data-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"https://raw.githubusercontent.com/jgeller112/"</span>,</span>
<span id="lst-data-4"><a href="#lst-data-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Beta_regression_tutorial/refs/heads/main/"</span>,</span>
<span id="lst-data-5"><a href="#lst-data-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"manuscript/data/fluency_data.csv"</span></span>
<span id="lst-data-6"><a href="#lst-data-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="lst-data-7"><a href="#lst-data-7" aria-hidden="true" tabindex="-1"></a>fluency_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(url)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<p>Our primary outcome variable is the proportion of idea units recalled on the final test, calculated by dividing the number of correct units by 10. We show a sample of these data in <a href="#tbl-dataset" class="quarto-xref" aria-expanded="false">Table&nbsp;1</a>. The dataset can be downloaded from Github (<a href="#lst-data" class="quarto-xref">Listing&nbsp;1</a>). Because this is a bounded continuous variable (i.e., it ranges from 0 to 1), it violates the assumptions of typical linear regression models that treat outcomes as normally distributed. Despite this, it remains common in psychological research to analyze proportion data using models that assume normality. In what follows, we reproduce <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span>’s analysis and then re-analyze the data using Beta regression and highlight how it can improve our inferences.</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="1" data-custom-style="FigureWithoutNote">
<div id="tbl-dataset" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="1">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-dataset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;1</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Four observations from <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span>. Accuracy refers to the proportion of correctly recalled idea units.</p>
</div>
</figcaption>
<div aria-describedby="tbl-dataset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_vlt0rvpjyj85iahvzzbr(i, j, css_id) {
          var table = document.getElementById("tinytable_vlt0rvpjyj85iahvzzbr");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_vlt0rvpjyj85iahvzzbr');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_vlt0rvpjyj85iahvzzbr(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_vlt0rvpjyj85iahvzzbr");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 4, j: 0 }, { i: 4, j: 1 }, { i: 4, j: 2 },  ], css_id: 'tinytable_css_oh59sux2c8ud147akbj2',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 },  ], css_id: 'tinytable_css_c2ij2gyxaqhn45q40otz',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_vlt0rvpjyj85iahvzzbr(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_oh59sux2c8ud147akbj2, .table th.tinytable_css_oh59sux2c8ud147akbj2 { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_c2ij2gyxaqhn45q40otz, .table th.tinytable_css_c2ij2gyxaqhn45q40otz { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_vlt0rvpjyj85iahvzzbr" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Participant</th>
                <th scope="col">Fluency</th>
                <th scope="col">Accuracy</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>64</td>
                  <td>Disfluent</td>
                  <td>0.10</td>
                </tr>
                <tr>
                  <td>30</td>
                  <td>Fluent</td>
                  <td>0.60</td>
                </tr>
                <tr>
                  <td>12</td>
                  <td>Fluent</td>
                  <td>0.10</td>
                </tr>
                <tr>
                  <td>37</td>
                  <td>Fluent</td>
                  <td>0.35</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
</section>
<section id="reanalysis-of-wilford-et-al.-experiment-1a" class="level2">
<h2 data-anchor-id="reanalysis-of-wilford-et-al.-experiment-1a">Reanalysis of Wilford et al.&nbsp;Experiment 1A</h2>
<p>In their original analysis of Experiment 1A, <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span> compared memory performance between fluent and disfluent instructor conditions using a traditional independent-samples t-test. They found that participants who watched the fluent instructor recalled significantly more idea units than those who viewed the disfluent version (see <a href="#fig-flu1" class="quarto-xref" aria-expanded="false">Figure&nbsp;3</a>).</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-flu1" class="quarto-float quarto-figure quarto-figure-center" prefix="" data-fignum="3">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-flu1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;3</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Raincloud plot depicting accuracy distributions for the Fluent and Disfluent conditions. Each condition shows individual data points, a density plot, and summary statistics to illustrate variability and central tendency.</p>
</div>
</figcaption>
<div aria-describedby="fig-flu1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-flu1-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>We first replicate this analysis in a regression framework using brms. We model final test accuracy–the proportion of correctly recalled idea units across the videos–as the dependent variable. Our predictor is instructor fluency, with two levels: Fluent and Disfluent. We use treatment (dummy) coding, which is the default in R. This coding scheme sets the first level of a factor (in alphabetical order) as the reference level. In this case, Disfluent is the reference, and the coefficient for Fluent reflects the contrast between fluent and disfluent instructor conditions.</p>
</div>
<div class="cell FigureWithNote" data-tbl-align="center" data-after-caption-space="0pt" data-apa-note="Link functions: b_mean = logit; b_phi = log; b_zoi (zero-one inflation) = logit; b_coi (conditional one-inflation) = logit. Asterisks reflect approximate two-sided p-values derived from the posterior pd. pd ≥ 0.975 (p ≤ .05) = `*`; pd ≥ 0.990 (p ≤ .01) = `**`; pd ≥ 0.998 (p ≤ .001) = `***`." prefix="" data-tblnum="2" data-custom-style="FigureWithNote">
<div id="tbl-all" class="cell quarto-float quarto-figure quarto-figure-center" data-after-caption-space="0pt" data-tbl-align="center" prefix="" data-tblnum="2" data-apa-note="Link functions: b_mean = logit; b_phi = log; b_zoi (zero-one inflation) = logit; b_coi (conditional one-inflation) = logit. Asterisks reflect approximate two-sided p-values derived from the posterior pd. pd ≥ 0.975 (p ≤ .05) = `*`; pd ≥ 0.990 (p ≤ .01) = `**`; pd ≥ 0.998 (p ≤ .001) = `***`.">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-all-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;2</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Bayesian regression summaries for each model</p>
</div>
</figcaption>
<div aria-describedby="tbl-all-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_vl6zj24ltgc6o77oii04(i, j, css_id) {
          var table = document.getElementById("tinytable_vl6zj24ltgc6o77oii04");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_vl6zj24ltgc6o77oii04');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_vl6zj24ltgc6o77oii04(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_vl6zj24ltgc6o77oii04");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 30, j: 3 }, { i: 30, j: 0 }, { i: 30, j: 4 }, { i: 30, j: 2 }, { i: 30, j: 1 }, { i: 30, j: 5 }, { i: 30, j: 6 },  ], css_id: 'tinytable_css_481wwtqouisypus0t6ra',}, 
          { positions: [ { i: 21, j: 0 }, { i: 24, j: 0 }, { i: 6, j: 0 }, { i: 15, j: 0 }, { i: 3, j: 1 }, { i: 18, j: 0 }, { i: 8, j: 2 }, { i: 27, j: 1 }, { i: 3, j: 0 }, { i: 2, j: 1 }, { i: 9, j: 0 }, { i: 18, j: 1 }, { i: 6, j: 1 }, { i: 12, j: 0 }, { i: 21, j: 1 }, { i: 9, j: 1 }, { i: 27, j: 0 }, { i: 24, j: 1 }, { i: 12, j: 1 }, { i: 21, j: 2 }, { i: 9, j: 2 }, { i: 15, j: 1 }, { i: 24, j: 2 }, { i: 8, j: 3 }, { i: 17, j: 4 }, { i: 27, j: 2 }, { i: 2, j: 2 }, { i: 3, j: 2 }, { i: 8, j: 4 }, { i: 18, j: 2 }, { i: 6, j: 2 }, { i: 2, j: 0 }, { i: 3, j: 3 }, { i: 17, j: 3 }, { i: 5, j: 3 }, { i: 6, j: 3 }, { i: 12, j: 2 }, { i: 21, j: 3 }, { i: 9, j: 3 }, { i: 15, j: 2 }, { i: 11, j: 3 }, { i: 12, j: 3 }, { i: 8, j: 1 }, { i: 5, j: 2 }, { i: 2, j: 3 }, { i: 11, j: 1 }, { i: 8, j: 5 }, { i: 18, j: 3 }, { i: 5, j: 0 }, { i: 11, j: 2 }, { i: 3, j: 4 }, { i: 8, j: 0 }, { i: 5, j: 1 }, { i: 24, j: 3 }, { i: 11, j: 0 }, { i: 17, j: 5 }, { i: 27, j: 3 }, { i: 15, j: 3 }, { i: 11, j: 4 }, { i: 12, j: 4 }, { i: 17, j: 0 }, { i: 14, j: 4 }, { i: 2, j: 4 }, { i: 20, j: 0 }, { i: 26, j: 5 }, { i: 5, j: 4 }, { i: 6, j: 4 }, { i: 20, j: 4 }, { i: 21, j: 4 }, { i: 9, j: 4 }, { i: 14, j: 3 }, { i: 24, j: 4 }, { i: 29, j: 0 }, { i: 26, j: 4 }, { i: 27, j: 4 }, { i: 15, j: 4 }, { i: 20, j: 3 }, { i: 8, j: 6 }, { i: 18, j: 4 }, { i: 14, j: 2 }, { i: 2, j: 5 }, { i: 3, j: 5 }, { i: 17, j: 2 }, { i: 5, j: 5 }, { i: 6, j: 5 }, { i: 20, j: 2 }, { i: 17, j: 6 }, { i: 9, j: 5 }, { i: 14, j: 1 }, { i: 11, j: 5 }, { i: 12, j: 5 }, { i: 17, j: 1 }, { i: 14, j: 5 }, { i: 15, j: 5 }, { i: 20, j: 1 }, { i: 26, j: 6 }, { i: 18, j: 5 }, { i: 14, j: 0 }, { i: 20, j: 5 }, { i: 21, j: 5 }, { i: 26, j: 1 }, { i: 23, j: 5 }, { i: 24, j: 5 }, { i: 29, j: 1 }, { i: 21, j: 6 }, { i: 27, j: 5 }, { i: 23, j: 0 }, { i: 29, j: 5 }, { i: 12, j: 6 }, { i: 26, j: 0 }, { i: 23, j: 4 }, { i: 2, j: 6 }, { i: 3, j: 6 }, { i: 29, j: 2 }, { i: 5, j: 6 }, { i: 6, j: 6 }, { i: 29, j: 4 }, { i: 29, j: 3 }, { i: 9, j: 6 }, { i: 23, j: 3 }, { i: 11, j: 6 }, { i: 20, j: 6 }, { i: 26, j: 3 }, { i: 14, j: 6 }, { i: 15, j: 6 }, { i: 24, j: 6 }, { i: 29, j: 6 }, { i: 18, j: 6 }, { i: 23, j: 2 }, { i: 23, j: 1 }, { i: 26, j: 2 }, { i: 23, j: 6 }, { i: 27, j: 6 },  ], css_id: 'tinytable_css_65siqxuil3dvvsvr5m6h',}, 
          { positions: [ { i: 16, j: 0 }, { i: 7, j: 0 }, { i: 4, j: 4 }, { i: 7, j: 5 }, { i: 7, j: 4 }, { i: 25, j: 0 }, { i: 1, j: 0 }, { i: 19, j: 0 }, { i: 28, j: 0 }, { i: 7, j: 2 }, { i: 4, j: 3 }, { i: 10, j: 0 }, { i: 1, j: 1 }, { i: 7, j: 3 }, { i: 16, j: 1 }, { i: 1, j: 3 }, { i: 1, j: 2 }, { i: 19, j: 1 }, { i: 7, j: 1 }, { i: 1, j: 4 }, { i: 1, j: 6 }, { i: 10, j: 1 }, { i: 16, j: 6 }, { i: 4, j: 6 }, { i: 4, j: 0 }, { i: 10, j: 5 }, { i: 7, j: 6 }, { i: 16, j: 4 }, { i: 13, j: 5 }, { i: 1, j: 5 }, { i: 10, j: 3 }, { i: 16, j: 5 }, { i: 4, j: 5 }, { i: 13, j: 0 }, { i: 10, j: 4 }, { i: 19, j: 2 }, { i: 25, j: 1 }, { i: 13, j: 4 }, { i: 13, j: 1 }, { i: 28, j: 1 }, { i: 19, j: 3 }, { i: 25, j: 2 }, { i: 22, j: 0 }, { i: 10, j: 6 }, { i: 28, j: 2 }, { i: 16, j: 2 }, { i: 4, j: 2 }, { i: 22, j: 1 }, { i: 19, j: 5 }, { i: 16, j: 3 }, { i: 25, j: 4 }, { i: 22, j: 5 }, { i: 10, j: 2 }, { i: 19, j: 4 }, { i: 25, j: 5 }, { i: 4, j: 1 }, { i: 13, j: 2 }, { i: 22, j: 4 }, { i: 25, j: 6 }, { i: 13, j: 6 }, { i: 13, j: 3 }, { i: 19, j: 6 }, { i: 28, j: 3 }, { i: 28, j: 4 }, { i: 22, j: 6 }, { i: 22, j: 3 }, { i: 28, j: 5 }, { i: 25, j: 3 }, { i: 22, j: 2 }, { i: 28, j: 6 },  ], css_id: 'tinytable_css_saw90wi5hnuy1zsfeolo',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 5 }, { i: 0, j: 4 }, { i: 0, j: 2 }, { i: 0, j: 6 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_opttvuwyjysmg8ds6pau',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_vl6zj24ltgc6o77oii04(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_481wwtqouisypus0t6ra, .table th.tinytable_css_481wwtqouisypus0t6ra { font-size: 0.9em; text-align: center; font-style: italic; font-size: 0.6em; border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_65siqxuil3dvvsvr5m6h, .table th.tinytable_css_65siqxuil3dvvsvr5m6h { font-size: 0.9em; text-align: center; font-style: italic; font-size: 0.6em; }
      .table td.tinytable_css_saw90wi5hnuy1zsfeolo, .table th.tinytable_css_saw90wi5hnuy1zsfeolo { font-size: 0.9em; text-align: center; }
      .table td.tinytable_css_opttvuwyjysmg8ds6pau, .table th.tinytable_css_opttvuwyjysmg8ds6pau { font-size: 0.9em; text-align: center; border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_vl6zj24ltgc6o77oii04" style="table-layout: fixed; width: 95% !important; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Parameter</th>
                <th scope="col">Stat</th>
                <th scope="col">Bayesian LM</th>
                <th scope="col">Beta Regression</th>
                <th scope="col">ZIB</th>
                <th scope="col">ZOIB</th>
                <th scope="col">Ordered Beta</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>b_Intercept</td>
                  <td>Mean</td>
                  <td>0.257</td>
                  <td>-0.830</td>
                  <td>-0.832</td>
                  <td>-0.831</td>
                  <td>-0.865</td>
                </tr>
                <tr>
                  <td></td>
                  <td>Cr.I</td>
                  <td>[0.199, 0.315]</td>
                  <td>[-1.087, -0.55]</td>
                  <td>[-1.094, -0.552]</td>
                  <td>[-1.098, -0.559]</td>
                  <td>[-1.119, -0.596]</td>
                </tr>
                <tr>
                  <td></td>
                  <td>pd</td>
                  <td>1.000***</td>
                  <td>1.000***</td>
                  <td>1.000***</td>
                  <td>1.000***</td>
                  <td>1.000***</td>
                </tr>
                <tr>
                  <td>b_Fluency</td>
                  <td>Mean</td>
                  <td>0.085</td>
                  <td>0.204</td>
                  <td>0.204</td>
                  <td>0.203</td>
                  <td>0.262</td>
                </tr>
                <tr>
                  <td></td>
                  <td>Cr.I</td>
                  <td>[0.002, 0.166]</td>
                  <td>[-0.155, 0.539]</td>
                  <td>[-0.139, 0.545]</td>
                  <td>[-0.147, 0.541]</td>
                  <td>[-0.07, 0.598]</td>
                </tr>
                <tr>
                  <td></td>
                  <td>pd</td>
                  <td>0.977*</td>
                  <td>0.875</td>
                  <td>0.872</td>
                  <td>0.880</td>
                  <td>0.936</td>
                </tr>
                <tr>
                  <td>b_phi_Intercept</td>
                  <td>Mean</td>
                  <td>-</td>
                  <td>1.609</td>
                  <td>1.601</td>
                  <td>1.604</td>
                  <td>1.609</td>
                </tr>
                <tr>
                  <td></td>
                  <td>Cr.I</td>
                  <td>-</td>
                  <td>[1.193, 2]</td>
                  <td>[1.187, 1.988]</td>
                  <td>[1.183, 1.989]</td>
                  <td>[1.179, 1.993]</td>
                </tr>
                <tr>
                  <td></td>
                  <td>pd</td>
                  <td>-</td>
                  <td>1.000***</td>
                  <td>1.000***</td>
                  <td>1.000***</td>
                  <td>1.000***</td>
                </tr>
                <tr>
                  <td>b_phi_Fluency</td>
                  <td>Mean</td>
                  <td>-</td>
                  <td>0.420</td>
                  <td>0.425</td>
                  <td>0.426</td>
                  <td>0.408</td>
                </tr>
                <tr>
                  <td></td>
                  <td>Cr.I</td>
                  <td>-</td>
                  <td>[-0.143, 0.993]</td>
                  <td>[-0.158, 0.994]</td>
                  <td>[-0.126, 0.994]</td>
                  <td>[-0.156, 0.983]</td>
                </tr>
                <tr>
                  <td></td>
                  <td>pd</td>
                  <td>-</td>
                  <td>0.931</td>
                  <td>0.926</td>
                  <td>0.930</td>
                  <td>0.918</td>
                </tr>
                <tr>
                  <td>b_zi_Intercept</td>
                  <td>Mean</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-1.673</td>
                  <td>-</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td></td>
                  <td>Cr.I</td>
                  <td>-</td>
                  <td>-</td>
                  <td>[-2.46, -0.978]</td>
                  <td>-</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td></td>
                  <td>pd</td>
                  <td>-</td>
                  <td>-</td>
                  <td>1.000***</td>
                  <td>-</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td>b_zi_Fluency</td>
                  <td>Mean</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-2.137</td>
                  <td>-</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td></td>
                  <td>Cr.I</td>
                  <td>-</td>
                  <td>-</td>
                  <td>[-4.618, -0.34]</td>
                  <td>-</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td></td>
                  <td>pd</td>
                  <td>-</td>
                  <td>-</td>
                  <td>0.992**</td>
                  <td>-</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td>b_zoi_Intercept</td>
                  <td>Mean</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-1.549</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td></td>
                  <td>Cr.I</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-</td>
                  <td>[-2.339, -0.859]</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td></td>
                  <td>pd</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-</td>
                  <td>1.000***</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td>b_zoi_Fluency</td>
                  <td>Mean</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-2.201</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td></td>
                  <td>Cr.I</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-</td>
                  <td>[-4.449, -0.465]</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td></td>
                  <td>pd</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-</td>
                  <td>0.996***</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td>b_coi_Intercept</td>
                  <td>Mean</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-2.022</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td></td>
                  <td>Cr.I</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-</td>
                  <td>[-4.408, -0.287]</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td></td>
                  <td>pd</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-</td>
                  <td>0.991**</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td>b_coi_Fluency</td>
                  <td>Mean</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-</td>
                  <td>0.245</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td></td>
                  <td>Cr.I</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-</td>
                  <td>[-6.946, 5.716]</td>
                  <td>-</td>
                </tr>
                <tr>
                  <td></td>
                  <td>pd</td>
                  <td>-</td>
                  <td>-</td>
                  <td>-</td>
                  <td>0.571</td>
                  <td>-</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
<div>
<div class="FigureNote NoIndent" data-custom-style="FigureNote">
<em>Note</em>. Link functions: b_mean = logit; b_phi = log; b_zoi (zero-one inflation) = logit; b_coi (conditional one-inflation) = logit. Asterisks reflect approximate two-sided p-values derived from the posterior pd. pd ≥ 0.975 (p ≤ .05) = <code>*</code>; pd ≥ 0.990 (p ≤ .01) = <code>**</code>; pd ≥ 0.998 (p ≤ .001) = <code>***</code>.
</div>
</div>
</div>
<section id="regression-model" class="level3">
<h3 data-anchor-id="regression-model">Regression Model</h3>
<p>We first start by loading the <code>brms</code> <span class="citation" data-cites="brms">(Bürkner, 2017)</span> and <code>cmdstanr</code> <span class="citation" data-cites="cmdstanr">(Gabry et al., 2024)</span> packages (<a href="#lst-loadbrms" class="quarto-xref">Listing&nbsp;2</a>). We use the <code>cmdstanr</code> backend for Stan <span class="citation" data-cites="stan2023">(Team, 2023)</span> because it’s faster than the default used to run models (i.e., <code>rstan</code>),<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a> though all of these models can also be fit with <code>brms</code> defaults.</p>
<div class="cell">
<div id="lst-loadbrms" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-loadbrms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;2: Load the <code>brms</code> and <code>cmdstanr</code> packages
</figcaption>
<div aria-describedby="lst-loadbrms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-loadbrms"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-loadbrms-1"><a href="#lst-loadbrms-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="lst-loadbrms-2"><a href="#lst-loadbrms-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cmdstanr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<div class="cell" data-hide="false">
<div id="lst-gauss" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-gauss-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;3: Fitting a gaussian model with brm().
</figcaption>
<div aria-describedby="lst-gauss-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-gauss"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-gauss-1"><a href="#lst-gauss-1" aria-hidden="true" tabindex="-1"></a>bayes_reg_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="lst-gauss-2"><a href="#lst-gauss-2" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency,</span>
<span id="lst-gauss-3"><a href="#lst-gauss-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="lst-gauss-4"><a href="#lst-gauss-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">gaussian</span>(),</span>
<span id="lst-gauss-5"><a href="#lst-gauss-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"model_reg_bayes"</span></span>
<span id="lst-gauss-6"><a href="#lst-gauss-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<p>We fit the model using the <code>brm()</code> function from the <code>brms</code> package (<a href="#lst-gauss" class="quarto-xref">Listing&nbsp;3</a>). Although not shown here, we ran the models using four chains (the default), executed in parallel across four cores. When the model is run in <a href="#lst-gauss" class="quarto-xref">Listing&nbsp;3</a>, the model summary output will appear in the R console. The output from <code>bayes_reg_model</code> shows each parameter’s posterior summary: The posterior distribution’s mean and standard deviation (analogous to the frequentist standard error) and its 95% credible interval, which indicate the 95% of the most credible parameter values. In <code>brms</code>, the reported Cr.I is an equal-tailed interval, meaning that the probability mass excluded from the interval is split equally between the lower and upper tails. Additionally, the output indicates numerical estimates of the sampling algorithm’s performance: Rhat should be close to one, and the ESS (effective sample size) metrics should be as large as possible given the number of iterations specified (default is 4000). Generally, ESS &gt;= 1000 is recommended <span class="citation" data-cites="brms">(Bürkner, 2017)</span>. For the models we present in this paper, convergence is trivial with standard linear models, though we note that these metrics are still important to pay attention to in case of model mis-fit.</p>
<p>Our main question of interest is: Does instructor fluency have an effect on final test performance? In order to answer this question, we will have to look at the output summary produced by <a href="#lst-gauss" class="quarto-xref">Listing&nbsp;3</a>. the <code>Intercept</code> refers to the posterior mean accuracy in the disfluent condition, M = 0.257 , as fluency was dummy-coded. The fluency coefficient (<code>FluencyFluent</code>) reflects the mean posterior difference in recall accuracy between the fluent and disfluent conditions: <em>b</em> = 0.083. The 95% credible interval for this estimate spans from -0.002 to 0.17. These values are shown in the “95% Cr.I” columns of the output. These results closely mirror the findings reported by <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span> (Experiment 1A).</p>
<div class="cell">
<div class="cell-output cell-output-stdout">
<pre><code> Family: gaussian 
  Links: mu = identity; sigma = identity 
Formula: Accuracy ~ Fluency 
   Data: fluency_data (Number of observations: 96) 

Regression Coefficients:
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
Intercept         0.26      0.03     0.20     0.31 1.00     3774     2926
FluencyFluent     0.08      0.04    -0.00     0.17 1.00     3685     3067

Further Distributional Parameters:
      Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sigma     0.21      0.02     0.18     0.24 1.00     3468     2614</code></pre>
</div>
</div>
<p>The output also includes the effective sample size (ESS) and R̂ (R-hat) values, both of which fall within acceptable ranges, indicating good model convergence. Throughout the tutorial, we focus on posterior mean estimates and their 95% credible intervals. In addition, we include the <em>pd</em> measure in the main summary table (<a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a>), which is provided by the <code>bayestestR</code> package <span class="citation" data-cites="makowski2019a makowski2019">(Makowski, Ben-Shachar, Chen, et al., 2019; Makowski, Ben-Shachar, &amp; Lüdecke, 2019)</span>. This measure offers an intuitive parallel to p-values that many readers may be familiar with.</p>
<div class="cell">
<div id="lst-hypo" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-hypo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;4: Getting BF using <code>hypothesis()</code> function from <code>brms</code>
</figcaption>
<div aria-describedby="lst-hypo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-hypo"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-hypo-1"><a href="#lst-hypo-1" aria-hidden="true" tabindex="-1"></a><span class="co"># test non-zero effect</span></span>
<span id="lst-hypo-2"><a href="#lst-hypo-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="lst-hypo-3"><a href="#lst-hypo-3" aria-hidden="true" tabindex="-1"></a>fluency_effect <span class="ot">&lt;-</span> <span class="fu">hypothesis</span>(bayes_reg_model, <span class="st">"FluencyFluent &gt; 0"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="3" data-custom-style="FigureWithoutNote">
<div id="tbl-hypo" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="3">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-hypo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;3</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Evidence ratio for the difference between fluent and disfluent conditions</p>
</div>
</figcaption>
<div aria-describedby="tbl-hypo-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_srqz09x11sc6yyp7ngnf(i, j, css_id) {
          var table = document.getElementById("tinytable_srqz09x11sc6yyp7ngnf");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_srqz09x11sc6yyp7ngnf');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_srqz09x11sc6yyp7ngnf(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_srqz09x11sc6yyp7ngnf");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 }, { i: 1, j: 2 }, { i: 1, j: 3 },  ], css_id: 'tinytable_css_irpkh3i93m9ymqhes0nz',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_ewztz188k8qlkl0rhgvx',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_srqz09x11sc6yyp7ngnf(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_irpkh3i93m9ymqhes0nz, .table th.tinytable_css_irpkh3i93m9ymqhes0nz { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_ewztz188k8qlkl0rhgvx, .table th.tinytable_css_ewztz188k8qlkl0rhgvx { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_srqz09x11sc6yyp7ngnf" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Hypothesis</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
                <th scope="col">ER</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>(FluencyFluent) &gt; 0</td>
                  <td>0.083</td>
                  <td>[0.012, 0.155]</td>
                  <td>37.095</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>For example, the fluency effect has a <em>pd</em> of .977, which indicates a high probability that the effect is positive rather than negative (akin to p &lt; .05). However, <em>pd</em> does not tell us the probability that the effect is <em>non-zero</em>. To assess this, we can compare a model where the effect is greater than 0 to an alternative model where the effect is less than 0. The <code>hypothesis()</code> function in <code>brms</code> (see <a href="#lst-hypo" class="quarto-xref">Listing&nbsp;4</a>) allows us to perform this comparison. Running the code returns a column called <code>ER</code> (<code>Evid.Ratio</code>), which contains a numerical value that represents the Bayes factor for a directional hypothesis (<a href="#tbl-hypo" class="quarto-xref" aria-expanded="false">Table&nbsp;3</a>). The evidence ratio (ER) for the fluency effect is 37.0952381, providing very strong evidence that the effect is both non-zero and positive <span class="citation" data-cites="jeffries1961 kass1995">(Jeffries, 1961; Kass &amp; Raftery, 1995)</span>.</p>
</div>
</section>
</section>
<section id="beta-regression" class="level2">
<h2 data-anchor-id="beta-regression">Beta Regression</h2>
<p><span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span> observed that instructor fluency impacts actual learning, using a t-test. But recall this approach assumes normality of residuals and homoscedacity. These assumptions are unrealistic when the response values approach the scale boundaries <span class="citation" data-cites="sladekova2024unicorns">(Sladekova &amp; Field, 2024)</span>. Does the data we have meet those assumptions? We can use the function <code>check_model()</code> from <code>easystats</code> <span class="citation" data-cites="easystats">(Lüdecke et al., 2022)</span> to check our assumptions easily. The code in <a href="#lst-check1" class="quarto-xref">Listing&nbsp;5</a> automatically produces <a href="#fig-reg-assump" class="quarto-xref" aria-expanded="false">Figure&nbsp;4</a>. We can see some issues with our data. Specifically, there appears to be violations of constant variance across the values of the scale (homoskedasticity). In plain terms, this type of model mis-specfication means that a standard OLS model can predict non-sensical values outside the bounds of the scale.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div id="lst-check1" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-check1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;5: Checking assumptions with the <code>check_model()</code> from <code>easystats</code> package .
</figcaption>
<div aria-describedby="lst-check1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-check1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-check1-1"><a href="#lst-check1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">check_model</span>(bayes_reg_model, <span class="at">check =</span> <span class="fu">c</span>(<span class="st">"homogeneity"</span>, <span class="st">"normality"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
<div class="cell-output-display">
<div id="fig-reg-assump" class="quarto-float quarto-figure quarto-figure-center" prefix="" data-fignum="4">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-reg-assump-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;4</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Two assumption checks for our OLS model: Normality (left) and Homoskedasticity (right)</p>
</div>
</figcaption>
<div aria-describedby="fig-reg-assump-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-reg-assump-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>We can also examine how well the data fits the model by performing a posterior predictive check using the <code>pp_check()</code> function from <code>brms</code>. A posterior predictive check involves looking at multiple draws or repetitions from the posterior distribution and plotting it against the observed data. Ideally, the predictive draws (the light blue lines) should show reasonable resemblance with the observed data (dark blue line). In our example (see <a href="#fig-ppcheckall" class="quarto-xref" aria-expanded="false">Figure&nbsp;5</a> (A)) the model-predicted density is slightly too peaked and narrow compared to the data. In addition, some of the draws extend into negative accuracy values.</p>
</div>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-ppcheckall" class="quarto-float quarto-figure quarto-figure-center" prefix="" data-fignum="5">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-ppcheckall-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;5</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>The plots show 100 posterior predicted distributions with the label yrep (light blue), the distribution of accuracy in dark blue for regular regression (A) , Beta regression (B), and ZIB (C) models</p>
</div>
</figcaption>
<div aria-describedby="fig-ppcheckall-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-ppcheckall-1.png" class="img-fluid figure-img" width="2400">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>Given the outcome variable is proportional, one solution would be to run a Beta regression model. Again, we can create the Beta regression model in <code>brms</code>. In <code>brms</code>, we model each parameter independently. Recall from the introduction that in a Beta model we model two parameters–<span class="math inline">\mu</span> and <span class="math inline">\phi</span>. We can easily do this by using the <code>bf()</code> function from <code>brms</code> (<a href="#lst-Beta01" class="quarto-xref">Listing&nbsp;6</a>). <code>bf()</code> facilitates the specification of several sub-models within the same formula call. We fit two formulas, one for <span class="math inline">\mu</span> and one for <span class="math inline">\phi</span> and store it in the <code>model_Beta_bayes</code> object below. In the below <code>bf()</code> call, we are modeling Fluency as a function of Accuracy only for the <span class="math inline">\mu</span> parameter. For the <span class="math inline">\phi</span> parameter, we are only modeling the intercept value. This is saying dispersion does not change as a function of fluency.</p>
</div>
<p>To run our Beta regression model, we need to exclude 0s and 1s in our data set. If we try to run a model with our data <code>data_fluency</code> we get an error: <code>Error: Family 'Beta' requires response greater than 0</code>. This is because the Beta distribution only supports observations in the 0 to 1 interval <em>excluding exact 0s and 1s</em>. We need make sure there are no 0s and 1s in our dataset.</p>
<p>The dataset contains nine 0s and one 1. One approach is to nudge our 0s towards .01 and our 1s to .99, or apply a special formula <span class="citation" data-cites="smithson2006">(Smithson &amp; Verkuilen, 2006)</span> so values fall within the [0, 1] interval. We implore readers not to engage in this practice. <span class="citation" data-cites="kubinec2022">Kubinec (2022)</span> showed that this practice can result in serious distortion of the outcome as the sample size 𝑁 grows larger, resulting in ever smaller values that are “nudged”. Because the Beta distribution is a non-linear model of the outcome, values that are very close to the boundary, such as 0.00001 or 0.99999, will be highly influential outliers.To run this Beta model we will remove the 0s and 1s, and later in this article we will show how to jointly model these scale end points with the rest of the data. The model from <a href="#lst-Beta01" class="quarto-xref">Listing&nbsp;6</a> uses a transformed <code>data_fluency</code> object (called <code>data_Beta</code>) where 0s and 1s are removed. When we run this code we should not get an error.</p>
<div class="cell">
<div id="lst-Beta01" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-Beta01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;6: Fitting a Beta model without 0s and 1s in brm().
</figcaption>
<div aria-describedby="lst-Beta01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-Beta01"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-Beta01-1"><a href="#lst-Beta01-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set up model formual</span></span>
<span id="lst-Beta01-2"><a href="#lst-Beta01-2" aria-hidden="true" tabindex="-1"></a>model_Beta_bayes <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="lst-Beta01-3"><a href="#lst-Beta01-3" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># fit mu model</span></span>
<span id="lst-Beta01-4"><a href="#lst-Beta01-4" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> <span class="dv">1</span> <span class="co"># fit phi model</span></span>
<span id="lst-Beta01-5"><a href="#lst-Beta01-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="lst-Beta01-6"><a href="#lst-Beta01-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="lst-Beta01-7"><a href="#lst-Beta01-7" aria-hidden="true" tabindex="-1"></a><span class="co"># transform 0 to 0.1 and 1 to .99</span></span>
<span id="lst-Beta01-8"><a href="#lst-Beta01-8" aria-hidden="true" tabindex="-1"></a>data_Beta <span class="ot">&lt;-</span> fluency_data <span class="sc">|&gt;</span></span>
<span id="lst-Beta01-9"><a href="#lst-Beta01-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="lst-Beta01-10"><a href="#lst-Beta01-10" aria-hidden="true" tabindex="-1"></a>    Accuracy <span class="sc">!=</span> <span class="dv">0</span>,</span>
<span id="lst-Beta01-11"><a href="#lst-Beta01-11" aria-hidden="true" tabindex="-1"></a>    Accuracy <span class="sc">!=</span> <span class="dv">1</span></span>
<span id="lst-Beta01-12"><a href="#lst-Beta01-12" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="lst-Beta01-13"><a href="#lst-Beta01-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="lst-Beta01-14"><a href="#lst-Beta01-14" aria-hidden="true" tabindex="-1"></a>Beta_brms <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="lst-Beta01-15"><a href="#lst-Beta01-15" aria-hidden="true" tabindex="-1"></a>  model_Beta_bayes,</span>
<span id="lst-Beta01-16"><a href="#lst-Beta01-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_Beta,</span>
<span id="lst-Beta01-17"><a href="#lst-Beta01-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">Beta</span>(),</span>
<span id="lst-Beta01-18"><a href="#lst-Beta01-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"model_Beta_bayes_reg_01"</span></span>
<span id="lst-Beta01-19"><a href="#lst-Beta01-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<section id="model-parameters" class="level3">
<h3 data-anchor-id="model-parameters">Model Parameters</h3>
<p>In <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a> under the Beta Regression column, the first set of coefficients represent how factors influence the <span class="math inline">\mu</span> parameter estimates (which is the mean of the Beta distribution), which are labeled with an underscore <code>b_</code>. These coefficients are interpreted on the scale of the logit, meaning they represent linear changes on a nonlinear space. The intercept term <code>(b_Intercept)</code> represents the log odds of the mean on accuracy for the fluent instructor. Log odds that are negative indicate that it is more likely a “success” (like getting the correct answer) will NOT happen than that it will happen. Similarly, regression coefficients in log odds forms that are negative indicate that an increase in that predictor leads to a decrease in the predicted probability of a “success”.</p>
<p>The other component we need to pay attention to is the dispersion or precision parameter coefficients labeled as <code>b_phi</code> in <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a>. The dispersion (<span class="math inline">\phi</span>) parameter tells us how precise our estimate is. Specifically, <span class="math inline">\phi</span> in Beta regression tells us about the variability of the response variable around its mean. Specifically, a higher dispersion parameter indicates a narrower distribution, reflecting less variability. Conversely, a lower dispersion parameter suggests a wider distribution, reflecting greater variability. The main difference between a dispersion parameter and the variance is that the dispersion has a different interpretation depending on the value of the outcome, as we show below. The best way to understand dispersion is to examine visual changes in the distribution as the dispersion increases or decreases.</p>
<p>Understanding the dispersion parameter helps us gauge the precision of our predictions and the consistency of the response variable. In <code>Beta_brms</code> we only modeled the dispersion of the intercept. When <span class="math inline">\phi</span> is not specified, the intercept is modeled by default (see <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a>). The intercept under the precision heading is not that interesting. It represents the overall dispersion in the outcome across all conditions. Instead, we can model different dispersions across levels of the <code>Fluency</code> factor. To do so, we add <code>Fluency</code> to the phi model in <code>bf()</code>. We model the precision (<code>phi</code>) of the <code>Fluency</code> factor by using a <code>~</code> and adding factors of interest to the right of it (<a href="#lst-Betadis" class="quarto-xref">Listing&nbsp;7</a>).</p>
<div class="cell">
<div id="lst-Betadis" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-Betadis-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;7: Fitting Beta model with dispersion in <code>brm()</code>.
</figcaption>
<div aria-describedby="lst-Betadis-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-Betadis"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-Betadis-1"><a href="#lst-Betadis-1" aria-hidden="true" tabindex="-1"></a>model_Beta_bayes_disp <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="lst-Betadis-2"><a href="#lst-Betadis-2" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># Model of the mean</span></span>
<span id="lst-Betadis-3"><a href="#lst-Betadis-3" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency <span class="co"># Model of the precision</span></span>
<span id="lst-Betadis-4"><a href="#lst-Betadis-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="lst-Betadis-5"><a href="#lst-Betadis-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="lst-Betadis-6"><a href="#lst-Betadis-6" aria-hidden="true" tabindex="-1"></a>Beta_brms_dis <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="lst-Betadis-7"><a href="#lst-Betadis-7" aria-hidden="true" tabindex="-1"></a>  model_Beta_bayes_disp,</span>
<span id="lst-Betadis-8"><a href="#lst-Betadis-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_Beta,</span>
<span id="lst-Betadis-9"><a href="#lst-Betadis-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">Beta</span>(),</span>
<span id="lst-Betadis-10"><a href="#lst-Betadis-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"model_Beta_bayes_dis_run01"</span></span>
<span id="lst-Betadis-11"><a href="#lst-Betadis-11" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<p><a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a> displays the model summary with the precision parameter added to our model as a function of fluency. It is important to note that the estimates are logged and not on the original scale (this is only the case when additional parameters are modeled). To interpret them on the original scale, we can exponentiate the log-transformed value–this transformation gets us back to our original scale. In the below model call, we set <code>exponentiate</code> = TRUE.</p>
<div class="cell" data-after-caption-space="0pt">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>Beta_model_dis_exp <span class="ot">&lt;-</span> Beta_brms_dis <span class="sc">|&gt;</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">model_parameters</span>(<span class="at">exponentiate =</span> <span class="cn">TRUE</span>, <span class="at">centrality =</span> <span class="st">"mean"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="4" data-custom-style="FigureWithoutNote">
<div id="tbl-phi-Beta-exp" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="4">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-phi-Beta-exp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;4</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Beta regression model summary for fluency factor with <span class="math inline">\phi</span> parameter exponentiated</p>
</div>
</figcaption>
<div aria-describedby="tbl-phi-Beta-exp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_p4tcefplw4hhka54c44n(i, j, css_id) {
          var table = document.getElementById("tinytable_p4tcefplw4hhka54c44n");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_p4tcefplw4hhka54c44n');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_p4tcefplw4hhka54c44n(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_p4tcefplw4hhka54c44n");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 2, j: 0 }, { i: 2, j: 1 }, { i: 2, j: 2 }, { i: 2, j: 3 },  ], css_id: 'tinytable_css_jq69zzgcf9s8y82kpsli',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_e6keeh8u6533hjydqz43',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_p4tcefplw4hhka54c44n(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_jq69zzgcf9s8y82kpsli, .table th.tinytable_css_jq69zzgcf9s8y82kpsli { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_e6keeh8u6533hjydqz43, .table th.tinytable_css_e6keeh8u6533hjydqz43 { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_p4tcefplw4hhka54c44n" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Parameter</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
                <th scope="col">pd</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>b_phi_Intercept</td>
                  <td>4.96</td>
                  <td>[3.238, 7.3]</td>
                  <td>1</td>
                </tr>
                <tr>
                  <td>b_phi_FluencyFluent</td>
                  <td>1.53</td>
                  <td>[0.844, 2.779]</td>
                  <td>0.923</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>The <span class="math inline">\phi</span> intercept represents the precision of the fluent condition. The <span class="math inline">\phi</span> coefficient for <code>FluencyFluent</code> represents the change in that precision for performance between the fluent vs.&nbsp;disfluent conditions. The credible interval does not include 0, meaning that zero is not among the 95% most credible parameter values.</p>
</div>
<p>It is important to note that these estimates are not the same as the marginal effects we discussed earlier. Changes in dispersion affect the spread or variability of the response distribution without necessarily altering its mean. This makes dispersion particularly relevant for research questions that focus on features of the distribution beyond the average—such as how concentrated responses are. For instance, high dispersion might indicate that individuals cluster at the extremes (e.g., very high or very low ratings), suggesting clustering in the outcome.</p>
<p>A critical assumption of the GLM is homoscedasticity, which means constant variance of the errors. WIth Beta regression model we can include a dispersion parameter for Fluency. Properly accounting for dispersion is crucial because it impacts the precision of our mean estimates and, consequently, the significance of our coefficients. The inclusion of dispersion in the our model increased the uncertainty of the <span class="math inline">\mu</span> coefficient (see <a href="#fig-plt-diff" class="quarto-xref" aria-expanded="false">Figure&nbsp;6</a>). This suggests that failing to account for the dispersion of the variables might lead to biased estimates. This highlights the potential utility of an approach like Beta regression over a traditional approach as Beta regression can explicitly model dispersion and address issues of heteroscedasticity.</p>
<p>It is only necessary to model the dispersion with covariates when there is reason to believe that this variation is substantively relevant to the research question. In case there is uncertainty about the best model, a relatively agnostic approach would be to compare models, for example with leave one out (loo) cross validation, to examine if a dispersion parameter should be considered in our model.<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a></p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-plt-diff" class="quarto-float quarto-figure quarto-figure-center" prefix="" data-fignum="6">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-plt-diff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;6</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Comparison of posterior distributions for the risk difference in fluency: Simple model (no dispersion for Fluency) vs.&nbsp;complex model with dispersion</p>
</div>
</figcaption>
<div aria-describedby="fig-plt-diff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-plt-diff-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
</section>
<section id="predicted-probabilities" class="level3">
<h3 data-anchor-id="predicted-probabilities">Predicted Probabilities</h3>
<p>Parameter estimates are usually difficult to interpret on their own. We argue that researchers should not spend too much time interpreting raw coefficients from non-linear models. We report them in this tutorial for completeness. Instead researchers should discuss the effects of the predictor on the actual outcome of interest (in this case the 0-1 scale). The logit link allows us to transform back and forth between the scale of a linear model and the nonlinear scale of the outcome, which is bounded by 0 and 1. By using the inverse of the logit, we can easily transform our linear coefficients to obtain average effects on the scale of the proportions or percentages, which is usually what is interesting to applied researchers. In a simple case, we can do this manually, but when there are many factors in your model this can be quite complex.</p>
<p>In our example, we can use the <code>plogis()</code> function in base R to convert estimates from the log-odds (logit) scale to the probability scale. The intercept of our model is -0.918, which reflects the log-odds of the mean accuracy in the disfluent condition. If the estimated difference between the fluent and disfluent conditions is 0.24 on the log-odds scale, we first add this value to the intercept value (-0.918) to get the log-odds for the fluent condition: <code>-0.83 + 0.20 = -0.63</code>. We then use <code>plogis()</code> to convert both log-odds values to probabilities (Fluent = 35%, Disfluent = 30%).</p>
<p>This is pretty easy to do manually, but when your model has many predictors, it can be quite cumbersome. To help us extract predictions from our model and visualize them we will use a package called <strong><code>marginaleffects</code></strong> <span class="citation" data-cites="margeffects2024">(Arel-Bundock et al., 2024)</span> (see <a href="#lst-me" class="quarto-xref">Listing&nbsp;8</a>). To get the proportions for each of our categorical predictors on the <span class="math inline">\mu</span> parameter we can use the function from the package called <code>predictions()</code>. These are displayed in <a href="#tbl-predict-prob" class="quarto-xref" aria-expanded="false">Table&nbsp;5</a>. These probabilities match what we calculated above.</p>
<div class="cell">
<div id="lst-me" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-me-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;8: Load the <code>marginaleffects</code> package.
</figcaption>
<div aria-describedby="lst-me-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-me"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-me-1"><a href="#lst-me-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(marginaleffects)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<div class="cell">
<div id="lst-pred" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-pred-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;9: Predictions from the Beta model for each level of Fluency.
</figcaption>
<div aria-describedby="lst-pred-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-pred"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-pred-1"><a href="#lst-pred-1" aria-hidden="true" tabindex="-1"></a><span class="fu">predictions</span>(</span>
<span id="lst-pred-2"><a href="#lst-pred-2" aria-hidden="true" tabindex="-1"></a>  Beta_brms,</span>
<span id="lst-pred-3"><a href="#lst-pred-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># need to specify the levels of the categorical predictor</span></span>
<span id="lst-pred-4"><a href="#lst-pred-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">newdata =</span> <span class="fu">datagrid</span>(<span class="at">Fluency =</span> <span class="fu">c</span>(<span class="st">"Disfluent"</span>, <span class="st">"Fluent"</span>))</span>
<span id="lst-pred-5"><a href="#lst-pred-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="5" data-tbl-cap-location="top" data-custom-style="FigureWithoutNote">
<div id="tbl-predict-prob" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="5" data-tbl-cap-location="top">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-predict-prob-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;5</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Predicted probabilitiesfor fluency factor.</p>
</div>
</figcaption>
<div aria-describedby="tbl-predict-prob-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_gv4x1m0e06smlxet3ys9(i, j, css_id) {
          var table = document.getElementById("tinytable_gv4x1m0e06smlxet3ys9");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_gv4x1m0e06smlxet3ys9');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_gv4x1m0e06smlxet3ys9(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_gv4x1m0e06smlxet3ys9");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 2, j: 0 }, { i: 2, j: 1 }, { i: 2, j: 2 },  ], css_id: 'tinytable_css_fe9ucg1v8zd1klhuxj5y',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 },  ], css_id: 'tinytable_css_m9t9gnhrm005nefdlxgr',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_gv4x1m0e06smlxet3ys9(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_fe9ucg1v8zd1klhuxj5y, .table th.tinytable_css_fe9ucg1v8zd1klhuxj5y { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_m9t9gnhrm005nefdlxgr, .table th.tinytable_css_m9t9gnhrm005nefdlxgr { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_gv4x1m0e06smlxet3ys9" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Fluency</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>Disfluent</td>
                  <td>0.297</td>
                  <td>[0.249, 0.346]</td>
                </tr>
                <tr>
                  <td>Fluent</td>
                  <td>0.354</td>
                  <td>[0.303, 0.409]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>For the <code>Fluency</code> factor, we can interpret <code>Mean</code> as proportions or percentages. That is, participants who watched the fluent instructor scored on average 35% on the final exam compared to 30% for those who watched the disfluent instructor. We can also visualize these from <code>marginaleffects</code> using the <code>plot_predictions()</code> function (see <a href="#lst-plotBeta" class="quarto-xref">Listing&nbsp;10</a>).</p>
</div>
<div class="cell">
<div id="lst-plotBeta" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-plotBeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;10: Plot predicted probabilities using <code>plot_predictions()</code> from <code>marginaleffects</code>
</figcaption>
<div aria-describedby="lst-plotBeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-plotBeta"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-plotBeta-1"><a href="#lst-plotBeta-1" aria-hidden="true" tabindex="-1"></a>Beta_plot <span class="ot">&lt;-</span> <span class="fu">plot_predictions</span>(Beta_brms, <span class="at">by =</span> <span class="st">"Fluency"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<p>The <code>plot_predictions()</code> function will only display the point estimate with the 95% credible intervals. However, Bayesian estimation methods generate distributions for each parameter. This approach allows visualizing full uncertainty estimates beyond points and intervals. Using the <code>marginaleffects</code> package, we can obtain samples from the posterior distribution with the <code>posterior_draws()</code> function (see <a href="#lst-posdraws" class="quarto-xref">Listing&nbsp;11</a>). We can then plot these results to illustrate the range of plausible values for our estimates at different levels of uncertainty (see <a href="#fig-draws-fluency" class="quarto-xref" aria-expanded="false">Figure&nbsp;7</a>).</p>
<div class="cell">
<div id="lst-posdraws" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-posdraws-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;11: Extracting posterior draws from the Beta regression model.
</figcaption>
<div aria-describedby="lst-posdraws-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-posdraws"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-posdraws-1"><a href="#lst-posdraws-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a model identifier to each dataset</span></span>
<span id="lst-posdraws-2"><a href="#lst-posdraws-2" aria-hidden="true" tabindex="-1"></a>pred_draws_Beta <span class="ot">&lt;-</span> <span class="fu">avg_predictions</span>(Beta_brms, <span class="at">variables =</span> <span class="st">"Fluency"</span>) <span class="sc">|&gt;</span></span>
<span id="lst-posdraws-3"><a href="#lst-posdraws-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">posterior_draws</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-draws-fluency" class="quarto-float quarto-figure quarto-figure-center" prefix="" data-fignum="7">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-draws-fluency-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;7</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Predicted probablity posterior distributions by fluency</p>
</div>
</figcaption>
<div aria-describedby="fig-draws-fluency-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-draws-fluency-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
</section>
<section id="marginal-effects" class="level3">
<h3 data-anchor-id="marginal-effects">Marginal Effects</h3>
<p>Marginal effects provide a way to understand how changes in a predictor influence an outcome, holding all other factors constant in a specific manner. Technically, marginal effects are calculated using partial derivatives for continuous variables or finite differences for categorical and continuous variables, depending on the nature of the data and the research question. Substantively, these procedures translate raw regression coefficients back into effects that represent changes in the bounded outcome, such as an <span class="math inline">x</span>% change in the value of a proportion.</p>
<p>There are various types of marginal effects, and their calculation can vary across software packages. For example, the popular <code>emmeans</code> package <span class="citation" data-cites="emmeans">(Lenth, 2025)</span> computes marginal effects by holding all predictors at their means. In this tutorial, we will use the <code>marginaleffects</code> package <span class="citation" data-cites="margeffects2024">(Arel-Bundock et al., 2024)</span>, which focuses on average marginal effects (AMEs) by default. AMEs summarize effects by generating predictions for each row of the original dataset and then averaging these predictions. This approach retains a strong connection to the original data while offering a straightforward summary of the effect of interest.</p>
<p>One practical application of AMEs is calculating the average difference between two groups or conditions (called the risk difference). Using the <code>avg_comparisons()</code> function in the <code>marginaleffects</code> package (<a href="#lst-avgcomp" class="quarto-xref">Listing&nbsp;12</a>), we can compute this metric directly. By default, the function calculates the discrete difference between groups. The function can also compute other effect size metrics, such as odds ratios and risk ratios, depending on the research question. This flexibility makes it a powerful tool for interpreting regression results in a meaningful way.</p>
<div class="cell">
<div id="lst-avgcomp" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-avgcomp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;12: Calculating the difference between probabilities with <code>avg_comparisons()</code>
</figcaption>
<div aria-describedby="lst-avgcomp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-avgcomp"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-avgcomp-1"><a href="#lst-avgcomp-1" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="lst-avgcomp-2"><a href="#lst-avgcomp-2" aria-hidden="true" tabindex="-1"></a><span class="co"># get risk difference by default</span></span>
<span id="lst-avgcomp-3"><a href="#lst-avgcomp-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="lst-avgcomp-4"><a href="#lst-avgcomp-4" aria-hidden="true" tabindex="-1"></a>Beta_avg_comp <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(Beta_brms, <span class="at">comparison =</span> <span class="st">"difference"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="6" data-custom-style="FigureWithoutNote">
<div id="tbl-ame1" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="6">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-ame1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;6</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Fluency difference</p>
</div>
</figcaption>
<div aria-describedby="tbl-ame1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_31xd7vtq68d2qzib4z1o(i, j, css_id) {
          var table = document.getElementById("tinytable_31xd7vtq68d2qzib4z1o");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_31xd7vtq68d2qzib4z1o');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_31xd7vtq68d2qzib4z1o(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_31xd7vtq68d2qzib4z1o");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 }, { i: 1, j: 2 }, { i: 1, j: 3 },  ], css_id: 'tinytable_css_rsjz3kss9vyxtdy5t4qw',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_aguvel1zam3ssn6kq2l9',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_31xd7vtq68d2qzib4z1o(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_rsjz3kss9vyxtdy5t4qw, .table th.tinytable_css_rsjz3kss9vyxtdy5t4qw { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_aguvel1zam3ssn6kq2l9, .table th.tinytable_css_aguvel1zam3ssn6kq2l9 { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_31xd7vtq68d2qzib4z1o" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Term</th>
                <th scope="col">Contrast</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>Fluency</td>
                  <td>Fluent - Disfluent</td>
                  <td>0.058</td>
                  <td>[-0.012, 0.13]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p><a href="#tbl-ame1" class="quarto-xref" aria-expanded="false">Table&nbsp;6</a> presents the estimated difference for the fluency factor (Mean column). The difference between the fluent and disfluent conditions is 0.06, indicating that participants who watched a fluent instructor scored, on average, 6% higher on the final recall test than those who watched a disfluent instructor. However, the 95% credible interval includes 0 among the most credible values, suggesting we cannot rule out the possibility of a null or weakly negative effect.</p>
</div>
<p>To formally assess evidence in favor of the null hypothesis (i.e., that the effect is exactly zero), we can use the <code>hypothesis()</code> function from the <code>brms</code> package to compute an evidence ratio (ER), which is a Bayes factor for directional or point hypotheses.<a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a>However, it is important to note that this procedure requires weakly informative or informative priors; the default priors in <code>brms</code> are too diffuse and therefore cannot be used to accurately compute the ER in favor of a point null hypothesis.</p>
<p>In psychology, it is common to report effect size measures like Cohen’s <em>d</em> <span class="citation" data-cites="cohen1977">(Cohen, 1977)</span>. When working with proportions we can calculate something similar called Cohen’s <em>h</em>. Taking our proportions, we can use the below equation (<a href="#eq-cohenh" class="quarto-xref">Equation&nbsp;2</a>) to calculate Cohen’s <em>h</em> along with the 95% Cr.I around it. Using this metric we see the effect size is small (0.107), 95% credible interval [-0.002, 0.361].</p>
<p><span id="eq-cohenh"><span class="math display">
h = 2 \cdot \left( \arcsin\left(\sqrt{p_1}\right) - \arcsin\left(\sqrt{p_2}\right) \right)
\tag{2}</span></span></p>
</section>
<section id="posterior-predictive-check" class="level3">
<h3 data-anchor-id="posterior-predictive-check">Posterior Predictive Check</h3>
<p><a href="#fig-ppcheckall" class="quarto-xref" aria-expanded="false">Figure&nbsp;5</a> (B) shows the predictive check for our Beta model. The model does a pretty good job at capturing the data (The draws are now between 0-1) and the predicted values from the model follow the observed data. However, it could be better.</p>
</section>
</section>
<section id="zero-inflated-beta-zib-regression" class="level2">
<h2 data-anchor-id="zero-inflated-beta-zib-regression">Zero-Inflated Beta (ZIB) Regression</h2>
<p>A limitation of the Beta regression model is that it can only accommodate values strictly between 0 and 1—it cannot handle values exactly equal to 0 or 1. In our dataset, we observed 9 rows where Accuracy equals zero. To fit a Beta regression model, we removed these values, but we have left out potentially valuable information from our model–especially if the end points of the scale are distinctive in some way. In our case, these 0s may be structural—that is, they represent real, systematic instances where participants failed to answer correctly (rather than random noise or measurement error). For example, the fluency of the instructor might be a key factor in predicting these zero responses. We will discuss two approaches for jointly modeling these end points with the continuous data. First, we can use a zero-inflated Beta (ZIB) model. This model still estimates the mean (<span class="math inline">\mu</span>) and precision (<span class="math inline">\phi</span>) of the Beta distribution for values between 0 and 1, but it also includes an additional parameter, <span class="math inline">\alpha</span>, which captures the probability of observing structural 0s.</p>
<p>The zero-inflated Beta models a mixture of the data-generating process. The <span class="math inline">\alpha</span> parameter uses a logistic regression to model whether the data is 0 or not. Substantively, this could be a useful model when we think that 0s come from a process that is relatively distinct from the data that is greater than 0. For example, if we had a dataset with proportion of looks or eye fixations to certain areas on marketing materials, we might want a separate model for those that do not look at certain areas on the screen because individuals who do not look might be substantively different than those that look.</p>
<p>We can fit a ZIB model using <code>brms</code> and use the <code>marginaleffects</code> package to make inferences about our parameters of interest. Before we run a zero-inflated Beta model, we will need to transform our data again and remove the one 1 value in our data–we can keep our 0s. Similar to our Beta regression model we fit in <code>brms</code>, we will use the <code>bf()</code> function to fit several models. We fit our <span class="math inline">\mu</span> and <span class="math inline">\phi</span> parameters as well as our zero-inflated parameter (<span class="math inline">\alpha</span>; here labeled as <code>zi</code>). In <code>brms</code> we can use the zero_inflated_Beta family (see <a href="#lst-zib" class="quarto-xref">Listing&nbsp;13</a>).</p>
<div class="cell">
<div id="lst-zib" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-zib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;13: Fitting zib model with <code>brm()</code>
</figcaption>
<div aria-describedby="lst-zib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-zib"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-zib-1"><a href="#lst-zib-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep 0 but remove 1</span></span>
<span id="lst-zib-2"><a href="#lst-zib-2" aria-hidden="true" tabindex="-1"></a>data_Beta_0 <span class="ot">&lt;-</span> fluency_data <span class="sc">|&gt;</span></span>
<span id="lst-zib-3"><a href="#lst-zib-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Accuracy <span class="sc">!=</span> <span class="dv">1</span>)</span>
<span id="lst-zib-4"><a href="#lst-zib-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="lst-zib-5"><a href="#lst-zib-5" aria-hidden="true" tabindex="-1"></a><span class="co"># set up model formual for zero-inflated Beta in brm</span></span>
<span id="lst-zib-6"><a href="#lst-zib-6" aria-hidden="true" tabindex="-1"></a>zib_model <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="lst-zib-7"><a href="#lst-zib-7" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># The mean of the 0-1 values, or mu</span></span>
<span id="lst-zib-8"><a href="#lst-zib-8" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency, <span class="co"># The precision of the 0-1 values, or phi</span></span>
<span id="lst-zib-9"><a href="#lst-zib-9" aria-hidden="true" tabindex="-1"></a>  zi <span class="sc">~</span> Fluency, <span class="co"># The zero-or-one-inflated part, or alpha</span></span>
<span id="lst-zib-10"><a href="#lst-zib-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">zero_inflated_beta</span>()</span>
<span id="lst-zib-11"><a href="#lst-zib-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="lst-zib-12"><a href="#lst-zib-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="lst-zib-13"><a href="#lst-zib-13" aria-hidden="true" tabindex="-1"></a><span class="co"># fit zib model with brm</span></span>
<span id="lst-zib-14"><a href="#lst-zib-14" aria-hidden="true" tabindex="-1"></a>fit_zi <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="lst-zib-15"><a href="#lst-zib-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> zib_model,</span>
<span id="lst-zib-16"><a href="#lst-zib-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_Beta_0,</span>
<span id="lst-zib-17"><a href="#lst-zib-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"bayes_zib_model0not1"</span></span>
<span id="lst-zib-18"><a href="#lst-zib-18" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<section id="posterior-predictive-check-1" class="level3">
<h3 data-anchor-id="posterior-predictive-check-1">Posterior Predictive Check</h3>
<p>The ZIB model does a bit better at capturing the structure of the data then the Beta regression model (see <a href="#fig-ppcheckall" class="quarto-xref" aria-expanded="false">Figure&nbsp;5</a>). Specifically, the ZIB model more accurately captures the increased density of values near the lower end of the scale (i.e., near zero), which the standard Beta model underestimates. The ZIB model’s predictive distributions also align more closely with the observed data across the entire range, particularly in the peak and tail regions. This improved fit likely reflects the ZIB model’s ability to explicitly model excess 0s (or near-zero values) via its inflation component, allowing it to better account for features in the data that a standard Beta distribution cannot accommodate.</p>
</section>
<section id="predicted-probabilities-and-marginal-effects" class="level3">
<h3 data-anchor-id="predicted-probabilities-and-marginal-effects">Predicted Probabilities and Marginal Effects</h3>
<p><a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a> under the zero-inflated Beta regression column provides a summary of the posterior distribution for each parameter. As stated before, it is preferable to back-transform our estimates to get probabilities. To get the predicted probabilities we can again use the <code>avg_predictions()</code> and <code>avg_comparisons()</code> functions from <code>marginaleffects</code> package <span class="citation" data-cites="marginaleffects">(Arel-Bundock, 2024)</span> to get predicted probabilities and the probability difference between the levels of each factor. We can model the parameters separately using the <code>dpar</code> argument setting to: <span class="math inline">\mu</span>, <span class="math inline">\phi</span>, <span class="math inline">\alpha</span>. Here we look at the risk difference for Fluency under each parameter.</p>
<section id="mean" class="level4">
<h4 data-anchor-id="mean">Mean.</h4>
<p>As shown in <a href="#tbl-marg-mu-brms" class="quarto-xref" aria-expanded="false">Table&nbsp;7</a>, there is little evidence for an effect of Fluency – the 95% Cr.I includes zero, suggesting substantial uncertainty about the direction and magnitude of the effect–that is, though most of the posterior density supports positive effects, nil and weakly negative effects cannot be ruled out.</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="7" data-custom-style="FigureWithoutNote">
<div id="tbl-marg-mu-brms" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="7">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-marg-mu-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;7</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Probablity fluency difference (<span class="math inline">\mu</span>)</p>
</div>
</figcaption>
<div aria-describedby="tbl-marg-mu-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_7rxoc65xhklme1bsonjm(i, j, css_id) {
          var table = document.getElementById("tinytable_7rxoc65xhklme1bsonjm");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_7rxoc65xhklme1bsonjm');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_7rxoc65xhklme1bsonjm(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_7rxoc65xhklme1bsonjm");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 }, { i: 1, j: 2 }, { i: 1, j: 3 },  ], css_id: 'tinytable_css_3rj7npvfjw83wohav8c5',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_nds5cx9jgiewmh1fs33d',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_7rxoc65xhklme1bsonjm(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_3rj7npvfjw83wohav8c5, .table th.tinytable_css_3rj7npvfjw83wohav8c5 { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_nds5cx9jgiewmh1fs33d, .table th.tinytable_css_nds5cx9jgiewmh1fs33d { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_7rxoc65xhklme1bsonjm" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Term</th>
                <th scope="col">Contrast</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>Fluency</td>
                  <td>Fluent - Disfluent</td>
                  <td>0.044</td>
                  <td>[-0.029, 0.116]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
</section>
<section id="dispersion" class="level4">
<h4 data-anchor-id="dispersion">Dispersion.</h4>
<p>As shown in <a href="#tbl-marg-phi-brms" class="quarto-xref" aria-expanded="false">Table&nbsp;8</a>, the posterior estimates suggest a credible effect of Fluency on dispersion (<span class="math inline">\phi</span>), with disfluent responses showing greater variability. The 95% Cr.I for the fluency contrast does not include zero, indicating a high probability in differences in precision.</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="8" data-custom-style="FigureWithoutNote">
<div id="tbl-marg-phi-brms" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="8">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-marg-phi-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;8</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Probablity fluency difference (<span class="math inline">\phi</span>)</p>
</div>
</figcaption>
<div aria-describedby="tbl-marg-phi-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_ybn7idfnw7lc9p1q1zop(i, j, css_id) {
          var table = document.getElementById("tinytable_ybn7idfnw7lc9p1q1zop");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_ybn7idfnw7lc9p1q1zop');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_ybn7idfnw7lc9p1q1zop(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_ybn7idfnw7lc9p1q1zop");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 }, { i: 1, j: 2 }, { i: 1, j: 3 },  ], css_id: 'tinytable_css_6t47xy9zaf9sk6yo4h3z',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_8veji2770wu6euj5clu1',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_ybn7idfnw7lc9p1q1zop(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_6t47xy9zaf9sk6yo4h3z, .table th.tinytable_css_6t47xy9zaf9sk6yo4h3z { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_8veji2770wu6euj5clu1, .table th.tinytable_css_8veji2770wu6euj5clu1 { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_ybn7idfnw7lc9p1q1zop" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Term</th>
                <th scope="col">Contrast</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>Fluency</td>
                  <td>Fluent - Disfluent</td>
                  <td>2.66</td>
                  <td>[-0.938, 6.536]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
</section>
</section>
<section id="zero-inflation" class="level3">
<h3 data-anchor-id="zero-inflation">Zero-Inflation</h3>
<p>We can harness the power of <code>marginaleffects</code> again and plot the posterior difference between the fluent and disfluent conditions (see <a href="#fig-zi-plot" class="quarto-xref" aria-expanded="false">Figure&nbsp;8</a>). In <a href="#fig-zi-plot" class="quarto-xref" aria-expanded="false">Figure&nbsp;8</a>, there is evidence that watching a lecture video with a fluent instructor reduces the probability of a zero response by approximately 13%. The 95% Cr.I for this effect does not include zero, suggesting a meaningful reduction in the likelihood of zero outcomes under fluent instruction. We can harness the power of <code>marginaleffects</code> again and plot the posterior probability of each level (see <a href="#fig-zi-plot" class="quarto-xref" aria-expanded="false">Figure&nbsp;8</a>).</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-zi-plot" class="quarto-float quarto-figure quarto-figure-center" prefix="" data-fignum="8">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-zi-plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;8</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Visualization of the predicted difference for zero-inflated part of model</p>
</div>
</figcaption>
<div aria-describedby="fig-zi-plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-zi-plot-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="zero-one-inflated-beta-zoib" class="level2">
<h2 data-anchor-id="zero-one-inflated-beta-zoib">Zero-One-Inflated Beta (ZOIB)</h2>
<p>The ZIB model works well if you have 0s in your data, but not 1s.<a href="#fn7" class="footnote-ref" id="fnref7" role="doc-noteref"><sup>7</sup></a> In our previous examples we either got rid of both 0s and 1s (Beta regression), or removed the 1s (ZIB). Sometimes it is theoretically useful to model both 0s and 1s as separate processes or to consider these values as essentially similar parts of the continuous response, as we show later in the ordered Beta regression model. For example, this is important in visual analog scale data where there might be a prevalence of responses at the bounds <span class="citation" data-cites="kong2016">(Kong &amp; Edwards, 2016)</span>, in JOL tasks <span class="citation" data-cites="wilford2020">(Wilford et al., 2020)</span>, or in a free-list task where individuals provide open responses to some question or topic which are then recoded to fall between 0-1 <span class="citation" data-cites="bendixen2023">(Bendixen &amp; Purzycki, 2023)</span>. Here 0s and 1s are meaningful; 0 means item was not listed and 1 means the item was listed first.</p>
<p>Similar to our Beta and zero-inflated models above, we can fit a ZOIB model in brms quite easily using the <code>zero_one_inflated_Beta</code> family. In this model, we simultaneously estimate the mean (<span class="math inline">\mu</span>) and precision (<span class="math inline">\phi</span>) of the Beta distribution, a zero-one inflation parameter (<span class="math inline">\alpha</span>) that represents the probability that an observation is either exactly 0 or 1 (i.e., 0 or 1 vs.&nbsp;not 0 or 1) and a conditional one-inflation parameter (<span class="math inline">\gamma</span>) that represents the probability that, given an observation is at one of the endpoints, it is 1 (i.e, 1 vs.&nbsp;not 1). This specification captures the entire range of possible values while remaining constrained between 0 and 1. To get a better sense of how <span class="math inline">\alpha</span> and <span class="math inline">\gamma</span> control the distribution of values, <a href="#fig-zoib" class="quarto-xref" aria-expanded="false">Figure&nbsp;9</a> presents simulated data across combinations of these parameters. As <span class="math inline">\alpha</span> increases, we see a greater proportion of responses at the endpoints. As <span class="math inline">\gamma</span> increases, the proportion of endpoint responses at 1 grows relative to 0, making the spikes at 1 more prominent as <span class="math inline">\gamma</span> approaches 1. This visualization illustrates how the ZOIB model flexibly accounts for both the continuous portion of the distribution and the occurrence of exact 0s and 1s.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-zoib" class="quarto-float quarto-figure quarto-figure-center" prefix="" data-fignum="9">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-zoib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;9</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Simulated data from a ZOIB model illustrating the effects of the zero-one inflation parameter (<span class="math inline">\alpha</span>) and the conditional one-inflation parameter (<span class="math inline">\gamma</span>).</p>
</div>
</figcaption>
<div aria-describedby="fig-zoib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-zoib-1.png" class="img-fluid figure-img" width="2400">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>To fit a ZOIB model we use the <code>bf()</code> function. We model each parameter as a function of Fluency. We then pass the <code>zoib_model</code> to our <code>brm()</code> function (see <a href="#lst-zoib" class="quarto-xref">Listing&nbsp;14</a>). The summary of the output is in <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a> (under <code>ZOIB</code>).</p>
</div>
<div class="cell">
<div id="lst-zoib" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-zoib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;14: Fitting a ZOIB model with <code>brm()</code>.
</figcaption>
<div aria-describedby="lst-zoib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-zoib"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-zoib-1"><a href="#lst-zoib-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fit the zoib model</span></span>
<span id="lst-zoib-2"><a href="#lst-zoib-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="lst-zoib-3"><a href="#lst-zoib-3" aria-hidden="true" tabindex="-1"></a>zoib_model <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="lst-zoib-4"><a href="#lst-zoib-4" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># The mean of the 0-1 values, or mu</span></span>
<span id="lst-zoib-5"><a href="#lst-zoib-5" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency, <span class="co"># The precision of the 0-1 values, or phi</span></span>
<span id="lst-zoib-6"><a href="#lst-zoib-6" aria-hidden="true" tabindex="-1"></a>  zoi <span class="sc">~</span> Fluency, <span class="co"># The zero-or-one-inflated part, or alpha</span></span>
<span id="lst-zoib-7"><a href="#lst-zoib-7" aria-hidden="true" tabindex="-1"></a>  coi <span class="sc">~</span> Fluency, <span class="co"># The one-inflated part, conditional on the 1s, or gamma</span></span>
<span id="lst-zoib-8"><a href="#lst-zoib-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">zero_one_inflated_beta</span>()</span>
<span id="lst-zoib-9"><a href="#lst-zoib-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="lst-zoib-10"><a href="#lst-zoib-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="lst-zoib-11"><a href="#lst-zoib-11" aria-hidden="true" tabindex="-1"></a>fit_zoib <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="lst-zoib-12"><a href="#lst-zoib-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> zoib_model,</span>
<span id="lst-zoib-13"><a href="#lst-zoib-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="lst-zoib-14"><a href="#lst-zoib-14" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"bayes_zoib_model"</span></span>
<span id="lst-zoib-15"><a href="#lst-zoib-15" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<section id="model-parameters-1" class="level3">
<h3 data-anchor-id="model-parameters-1">Model Parameters</h3>
<p>The output for the model is lengthy because we are estimating three distinct components, each with their own independent responses and sub-models. All the coefficients are on the logit scale, except <span class="math inline">\phi</span> , which is on the log scale. Thankfully drawing inferences for all these different parameters, plotting their distributions, and estimating their average marginal effects looks exactly the same–all the <code>brms</code> and <code>marginaleffects</code> functions we used work the same.</p>
</section>
<section id="predictions-and-marginal-effects" class="level3">
<h3 data-anchor-id="predictions-and-marginal-effects">Predictions and Marginal Effects</h3>
<p>With <code>marginaleffects</code> we can choose marginalize over all the sub-models, averaged across the 0s, continuous responses, and 1s in the data, or we can model the parameters separately using the <code>dpar</code> argument like we did above setting it to: <span class="math inline">\mu</span>, <span class="math inline">\phi</span>, <span class="math inline">\alpha</span>, <span class="math inline">\gamma</span> (see below). Using <code>avg_predictions()</code> and not setting <code>dpar</code> we can get the predicted probabilities across all the sub-models. We can also plot the overall difference between fluency and disfluency for the whole model with <code>plot_predictions()</code>.</p>
<p>In addition, we show below how one can extract the predicted probabilities and marginal effects for <span class="math inline">\gamma</span> (and a similar process for any other model component, <code>zoi</code>, etc.):</p>
<div class="cell">
<div id="lst-coi" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-coi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;15: Extracting predicted probabilities and marginal effects for conditional-one parameter
</figcaption>
<div aria-describedby="lst-coi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-coi"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-coi-1"><a href="#lst-coi-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get average predictions for coi param</span></span>
<span id="lst-coi-2"><a href="#lst-coi-2" aria-hidden="true" tabindex="-1"></a>coi_probs <span class="ot">&lt;-</span> <span class="fu">avg_predictions</span>(fit_zoib, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Fluency"</span>), <span class="at">dpar =</span> <span class="st">"coi"</span>)</span>
<span id="lst-coi-3"><a href="#lst-coi-3" aria-hidden="true" tabindex="-1"></a><span class="co"># get differene between the two conditions</span></span>
<span id="lst-coi-4"><a href="#lst-coi-4" aria-hidden="true" tabindex="-1"></a>coi_me <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(fit_zoib, <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Fluency"</span>), <span class="at">dpar =</span> <span class="st">"coi"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
</section>
</section>
<section id="ordered-beta-regression" class="level2">
<h2 data-anchor-id="ordered-beta-regression">Ordered Beta Regression</h2>
<p>Looking at the output from the ZOIB model (<a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a>), we can see how running a model like this can become fairly complex as it is fitting distinct sub-models for each component of the scale. The ability to consider 0s and 1s as distinct processes from continuous values comes at a price in terms of complexity and interpretability. A simplified version of the zero-one-inflated beta (ZOIB) model, known as ordered Beta regression <span class="citation" data-cites="kubinec2022 Makowski2025CHOCO">(Kubinec, 2022; see also Makowski et al., 2025 for a reparameterized version called the <em>Beta-Gate</em> model)</span>, has been recently proposed. The ordered Beta regression model exploits the fact that, for most analyses, the continuous values (between 0-1) and the discrete outcomes (e.g., 0 or 1) are <em>ordered</em>. For example, as a covariate <span class="math inline">x</span> increases or decreases, we should expect the bounded outcome <span class="math inline">y</span> to increase or decrease monotonically as well from 0 to <span class="math inline">(0,1)</span> to 1. The ZOIB model does not impose this restriction; a covariate could increase and the response <span class="math inline">y</span> could increase in its continuous values while <em>simultaneously</em> decreasing at <em>both</em> end points.<a href="#fn8" class="footnote-ref" id="fnref8" role="doc-noteref"><sup>8</sup></a> This complexity is not immediately obvious when fitting the ZOIB, nor is it a potential relationship that many scholars want to consider when examining how covariates influence a bounded scale.</p>
<p>To make the response ordered, the ordered Beta regression model estimates a weighted combination of a standard Beta regression model for continuous responses and a logit model for the discrete values of the response. By doing so, the amount of distinctiveness between the continuous responses and the discrete end points is a function of the data (and any informative priors) rather than strictly defined as fully distinct processes as in the ZOIB. For some datasets, the continuous and discrete responses will be fairly distinct, and in others less so. To give a slightly absured example, if a value of 0 meant that the subject was a fish, and a value of 1 meant that the subject was now a trombone, then the ordered Beta model would no longer be appropriate.</p>
<p>The weights that average together the two parts of the outcome (i.e., discrete and continuous) are determined by cutpoints that are estimated in conjunction with the data in a similar manner to what is known as an ordered logit model. An in-depth explanation of ordinal regression is beyond the scope of this tutorial <span class="citation" data-cites="Fullerton2023 bürkner2019">(Bürkner &amp; Vuorre, 2019; but see Fullerton &amp; Anderson, 2021)</span>. At a basic level, ordinal regression models are useful for outcome variables that are categorical in nature and have some inherent ordering (e.g., Likert scale items). To preserve this ordering, ordinal models rely on the cumulative probability distribution. Within an ordinal regression model it is assumed that there is a continuous but unobserved latent variable that determines which of <span class="math inline">k</span> ordinal responses will be selected. For example on a typical Likert scale from ‘Strongly Disagree’ to ‘Strongly Agree’, you could assume that there is a continuous, unobserved variable called ‘Agreement’.</p>
<p>While we cannot measure Agreement directly, the ordinal response gives us some indication about where participants are on the continuous Agreement scale. <span class="math inline">k - 1</span> cutoffs are then estimated to indicate the point on the continuous Agreement scale at which your Agreement level is high enough to push you into the next ordinal category (say Agree to Strongly Agree). Coefficients in the model estimate how much different predictors change the estimated <em>continuous</em> scale (here, Agreement). Since there’s only one underlying process, there’s only one set of coefficients to work with (proportional odds assumption). In an ordered Beta regression, three ordered categories are modeled: (1) exactly zero, (2) somewhere between zero and one, and (3) exactly one. In an ordered Beta regression, (1) and (2) are modeled with cumulative logits, where one cutpoint is the the boundary between Exactly 0 and Between 0 and 1 and the other cutpoint is the boundary between <em>Between 0 and 1</em> and <em>Exactly 1.</em> The continuous values in the middle, 0 to 1 (3), are modeled as a vanilla Beta regression with parameters reflecting the mean response on the logit scale as we have described previously. Ultimately, employing cutpoints allows for a smooth transition between the bounds and the continuous values, permitting both to be considered together rather than modeled separately as the ZOIB requires.</p>
<p>The ordered Beta regression model has shown to be more efficient and less biased than some of the methods discussed <span class="citation" data-cites="kubinec2022">(Kubinec, 2022)</span> herein and has seen increasing use across the biomedical and social sciences <span class="citation" data-cites="wilkesQuantifyingCoextinctionsEcosystem2024 martinVocalComplexitySocially2024 smithEcologicalMomentaryAssessment2024 shresthaBigFiveTraits2024 nouvianGlyphosateImpairsAversive2023">(Martin et al., 2024; Nouvian et al., 2023; Shrestha et al., 2024; Smith et al., 2024; Wilkes et al., 2024)</span> because it produces only a single set of coefficient estimates in a similar manner to a standard Beta regression or OLS.</p>
<section id="fitting-an-ordered-beta-regression" class="level3">
<h3 data-anchor-id="fitting-an-ordered-beta-regression">Fitting an Ordered Beta Regression</h3>
<p>To fit an ordered Beta regression in a Bayesian context we use the <code>ordbetareg</code> <span class="citation" data-cites="ordbetareg">(Kubinec, 2023)</span> package. <code>ordbetareg</code> is a front-end to the <code>brms</code> package that we described earlier; in addition to the functions available in the package, most <code>brms</code> functions and plots, including the diverse array of regression modeling options, will work with <code>ordbetareg</code> models. (We note that the <code>ordBeta</code> model is also available as a maximum-likelihood variant in the R package <code>glmmTMB</code>.) We first load the <code>ordbetareg</code> package (see <a href="#lst-ordBeta" class="quarto-xref">Listing&nbsp;16</a>).</p>
<div class="cell">
<div id="lst-ordBeta" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-ordBeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;16: Load <code>ordbetareg</code>
</figcaption>
<div aria-describedby="lst-ordBeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-ordBeta"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-ordBeta-1"><a href="#lst-ordBeta-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ordbetareg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<p>The <code>ordbetareg</code> package uses <code>brms</code> on the front-end so all the arguments we used previously apply here. Instead of the <code>brm()</code> function we use <code>ordbetareg()</code>. To fit a model where dispersion does not vary as a function of fluency we can use the below code (see <a href="#lst-fitordBeta" class="quarto-xref">Listing&nbsp;17</a>).</p>
<div class="cell">
<div id="lst-fitordBeta" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-fitordBeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;17: Fitting ordered Beta model with <code>ordbetareg()</code>
</figcaption>
<div aria-describedby="lst-fitordBeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-fitordBeta"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-fitordBeta-1"><a href="#lst-fitordBeta-1" aria-hidden="true" tabindex="-1"></a>ord_fit_brms <span class="ot">&lt;-</span> <span class="fu">ordbetareg</span>(</span>
<span id="lst-fitordBeta-2"><a href="#lst-fitordBeta-2" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency,</span>
<span id="lst-fitordBeta-3"><a href="#lst-fitordBeta-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="lst-fitordBeta-4"><a href="#lst-fitordBeta-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"bayes_ordbeta_model"</span></span>
<span id="lst-fitordBeta-5"><a href="#lst-fitordBeta-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
<p>However, if we want dispersion to vary as a function of fluency we can easily do that (see <a href="#lst-ordBetadisp" class="quarto-xref">Listing&nbsp;18</a>). Note the addition of the <code>phi_reg</code> argument in <code>m.phi</code>. This argument allows us to include a model that explicitly models the dispersion parameter. Because we are modeling <span class="math inline">\phi</span> as a function of fluency, we set the the argument to <code>both.</code></p>
<div class="cell">
<div id="lst-ordBetadisp" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-ordBetadisp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;18: Fitting ordered Beta model with dispersion using <code>ordbetareg()</code>
</figcaption>
<div aria-describedby="lst-ordBetadisp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="sourceCode cell-code" id="lst-ordBetadisp"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="lst-ordBetadisp-1"><a href="#lst-ordBetadisp-1" aria-hidden="true" tabindex="-1"></a>ord_beta_phi <span class="ot">&lt;-</span> <span class="fu">bf</span>(Accuracy <span class="sc">~</span> Fluency, phi <span class="sc">~</span> Fluency)</span>
<span id="lst-ordBetadisp-2"><a href="#lst-ordBetadisp-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="lst-ordBetadisp-3"><a href="#lst-ordBetadisp-3" aria-hidden="true" tabindex="-1"></a>m.phi <span class="ot">&lt;-</span> <span class="fu">ordbetareg</span>(</span>
<span id="lst-ordBetadisp-4"><a href="#lst-ordBetadisp-4" aria-hidden="true" tabindex="-1"></a>  ord_beta_phi,</span>
<span id="lst-ordBetadisp-5"><a href="#lst-ordBetadisp-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="lst-ordBetadisp-6"><a href="#lst-ordBetadisp-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">phi_reg =</span> <span class="st">'both'</span>,</span>
<span id="lst-ordBetadisp-7"><a href="#lst-ordBetadisp-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"bayes_ordbeta_phi_model"</span></span>
<span id="lst-ordBetadisp-8"><a href="#lst-ordBetadisp-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</figure>
</div>
</div>
</section>
</section>
<section id="marginal-effects-1" class="level2">
<h2 data-anchor-id="marginal-effects-1">Marginal Effects</h2>
<p><a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a> presents the overall model summary (under <code>Ordered Beta</code>). We can use <code>marginaleffects</code> to calculate differences on the response scale that average over (or marginalize over) all our parameters.</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="9" data-custom-style="FigureWithoutNote">
<div id="tbl-ordBeta" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="9">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-ordBeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;9</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Marginal effect for fluency in ordered Beta model</p>
</div>
</figcaption>
<div aria-describedby="tbl-ordBeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_mi3505jpo06se5cxmo9m(i, j, css_id) {
          var table = document.getElementById("tinytable_mi3505jpo06se5cxmo9m");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_mi3505jpo06se5cxmo9m');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_mi3505jpo06se5cxmo9m(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_mi3505jpo06se5cxmo9m");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 }, { i: 1, j: 2 }, { i: 1, j: 3 },  ], css_id: 'tinytable_css_54lzh099u00iq7xesomo',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_iqv7cdzdcahm6umr7nms',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_mi3505jpo06se5cxmo9m(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_54lzh099u00iq7xesomo, .table th.tinytable_css_54lzh099u00iq7xesomo { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_iqv7cdzdcahm6umr7nms, .table th.tinytable_css_iqv7cdzdcahm6umr7nms { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_mi3505jpo06se5cxmo9m" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Term</th>
                <th scope="col">Contrast</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>Fluency</td>
                  <td>Fluent - Disfluent</td>
                  <td>0.062</td>
                  <td>[-0.017, 0.137]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>In <a href="#tbl-ordBeta" class="quarto-xref" aria-expanded="false">Table&nbsp;9</a> the credible interval is close enough to zero relative to its uncertainty that we can conclude there likely aren’t substantial differences between the conditions after taking dispersion and the 0s and 1s in our data into account.</p>
</div>
<section id="cutpoints" class="level3">
<h3 data-anchor-id="cutpoints">Cutpoints</h3>
<p>The model cutpoints are not reported by default in the summary output, but we can access them with the R package <code>posterior</code> <span class="citation" data-cites="posterior2025a">(Bürkner et al., 2025)</span> and the functions <code>as_draws</code> and <code>summary_draws</code>.</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="10" data-custom-style="FigureWithoutNote">
<div id="tbl-cutpoints" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="10">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-cutpoints-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;10</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Cutzero and cutone parameter summary</p>
</div>
</figcaption>
<div aria-describedby="tbl-cutpoints-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_ckgl26kxd2s61a1p991g(i, j, css_id) {
          var table = document.getElementById("tinytable_ckgl26kxd2s61a1p991g");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_ckgl26kxd2s61a1p991g');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_ckgl26kxd2s61a1p991g(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_ckgl26kxd2s61a1p991g");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 2, j: 0 }, { i: 2, j: 1 }, { i: 2, j: 2 },  ], css_id: 'tinytable_css_54tdze3ihfozjc36y9kr',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 },  ], css_id: 'tinytable_css_to6yoiibtauu8nqibhx6',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_ckgl26kxd2s61a1p991g(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_54tdze3ihfozjc36y9kr, .table th.tinytable_css_54tdze3ihfozjc36y9kr { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_to6yoiibtauu8nqibhx6, .table th.tinytable_css_to6yoiibtauu8nqibhx6 { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_ckgl26kxd2s61a1p991g" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Parameter</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>cutzero</td>
                  <td>-2.98</td>
                  <td>[-3.56, -2.42]</td>
                </tr>
                <tr>
                  <td>cutone</td>
                  <td>1.85</td>
                  <td>[1.64, 2.06]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>In <a href="#tbl-cutpoints" class="quarto-xref" aria-expanded="false">Table&nbsp;10</a>, <code>cutzero</code> is the first cutpoint (the difference between 0 and continuous values) and <code>cutone</code> is the second cutpoint (the difference between the continuous values and 1). These cutpoints are on the logit scale and as such the numbers do not have a simple substantive meaning. In general, as the cutpoints increase in absolute value (away from zero), then the discrete/boundary observations are more distinct from the continuous values. This will happen if there is a clear gap or bunching in the outcome around the bounds. This type of empirical feature of the distribution may be useful to scholars if they want to study differences in how people perceive the ends of the scale versus the middle. It is possible, though beyond the scope of this article, to model the location of the cutpoints with hierarchical (non-linear) covariates in <code>brms</code>.</p>
</div>
</section>
<section id="model-fit" class="level3">
<h3 data-anchor-id="model-fit">Model Fit</h3>
<p>The best way to visualize model fit is to plot the full predictive distribution relative to the original outcome. Because ordered Beta regression is a mixed discrete/continuous model, a separate plotting function, <code>pp_check_ordBetareg</code>, is included in the <code>ordbetareg</code> package that accurately handles the unique features of this distribution. The default plot in <code>brms</code> will collapse these two features of the outcome together, which will make the fit look worse than it actually is. The <code>ordbetareg</code> function returns a list with two plots, <code>discrete</code> and <code>continuous</code>, which can either be printed and plotted or further modified as <code>ggplot2</code> objects (see <a href="#fig-ppcheckord" class="quarto-xref" aria-expanded="false">Figure&nbsp;10</a>).</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-ppcheckord" class="quarto-float quarto-figure quarto-figure-center" prefix="" data-fignum="10">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-ppcheckord-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;10</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Posterior predictive check for ordered Beta regression model. A. Discrete posterior check. B. Continuous posterior check.</p>
</div>
</figcaption>
<div aria-describedby="fig-ppcheckord-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-ppcheckord-1.png" class="img-fluid figure-img" width="3600">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>The discrete plot, which is a bar graph, shows that the posterior distribution accurately captures the number of different types of responses (discrete or continuous) in the data. For the continuous plot shown as a density plot with one line per posterior draw, the model does a very good job at capturing the distribution.</p>
</div>
<p>Overall, it is clear from the posterior distribution plot that the ordered Beta model fits the data well. To fully understand model fit, both of these plots need to be inspected as they are conceptually distinct.</p>
<section id="model-visualization" class="level4">
<h4 data-anchor-id="model-visualization">Model Visualization.</h4>
<p><code>ordbetareg</code> provides a neat visualization function called <code>plot_heiss()</code> <span class="citation" data-cites="yeEnforcingBoundariesChinas2023">(Ye &amp; Heiss, 2023)</span> that can represent dispersion in the entire outcome as a function of discrete covariates. This function produces a plot of predicted proportions across the range of our <code>Fluency</code> factor. In <a href="#fig-heiss" class="quarto-xref" aria-expanded="false">Figure&nbsp;11</a> we get predicted proportions for Fluency across the bounded scale. Looking at the figure we can see there is much overlap between instructors in the middle portion (<span class="math inline">\mu</span>) . However, we do see some small differences at the zero bounds.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-heiss" class="quarto-float quarto-figure quarto-figure-center" prefix="" data-fignum="11">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-heiss-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;11</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Heiss plot of predicted probabilities across the scale (0-100)</p>
</div>
</figcaption>
<div aria-describedby="fig-heiss-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-heiss-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="ordered-beta-scale" class="level3">
<h3 data-anchor-id="ordered-beta-scale">Ordered Beta Scale</h3>
<p>In the <code>ordbetareg</code> function there is a <code>true_bound</code> argument. In the case where you data in not bounded between 0-1, you can use the argument to specify the bounds of the argument to fit the ordered Beta regression. For example, you data might be bounded between 1 and 7. If so, you can model it as such and <code>ordbetareg</code> will convert the model predictions back to the true bounds after estimation.</p>
</section>
</section>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
<p>The use of Beta regression in psychology, and the social sciences in general, is rare. With this tutorial, we hope to turn the tides. Beta regression models are an attractive alternative to models that impose unrealistic assumptions like normality, linearity, homoscedasticity, and unbounded data. Beyond these models, there are a diverse array of different models that can be used depending on your outcome of interest.</p>
<p>Throughout this tutorial our main aim was to help guide researchers in running analyses with proportional or percentage outcomes using Beta regression and some of its alternatives. In the current example, we used real data from <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span> and discussed how to fit these models in R, interpret model parameters, extract predicted probabilities and marginal effects, and visualize the results.</p>
<p>Comparing our analysis with that of <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span>, we demonstrated that using traditional approaches (e.g., t-tests) to analyze accuracy data can lead to inaccurate inferences. Although we successfully reproduced one of their key findings, our use of Beta regression and its extensions revealed important nuances in the results. With a traditional Beta regression model–which accounts for both the mean and the precision (dispersion)–we observed similar effects of instructor fluency on performance. However, the standard Beta model does not accommodate boundary values (i.e., 0s and 1s).</p>
<p>When we applied a ZIB model, which explicitly accounts for structural 0s, we found no effect of fluency on the mean (<span class="math inline">\mu</span>) part of the model. Instead, the effect of fluency emerged in the structural zero (inflated zero; <span class="math inline">\alpha</span>) component. This pattern was consistent when using a zero-one-inflated Beta (ZOIB) model. Furthermore, we fit an ordered Beta regression model <span class="citation" data-cites="kubinec2022">(Kubinec, 2022)</span>, which appropriately models the full range of values, including 0s and 1s. Here, we did not observe a reliable effect of fluency on the mean once we accounted for dispersion.</p>
<p>These analyses emphasize the importance of fitting a model that aligns with the nature of the data. The simplest and recommended approach when dealing with data that contains 0s and/or 1s is to fit an ordered Beta model, assuming the process is truly continuous. However, if you believe the process is distinct in nature, a ZIB or ZOIB model might be a better choice. Ultimately, this decision should be guided by theory.</p>
<p>For instance, if we believe fluency influences the structural zero part of the model, we might want to model this process separately using a ZIB or ZOIB. With the current dataset, fluency might affect specific aspects of performance (such as the likelihood of complete failure) rather than general performance levels. This effect could be due to participant disengagement during the disfluent lecture. If students fail to pay attention because of features of disfluency, they may miss relevant information, leading to a floor effect at the test. If this is the case, we would want to model this appropriately. However, if we believe fluency effects general performance levels, a model that takes in to account the entire process accounting for the 0s and 1s might be appropriate.</p>
<p>In the discussion section of <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span>, they were unable to offer a tenable explanation for performance differences based on instructor fluency. A model that accounts for the excess 0s in the dataset provides one testable explanation: watching a disfluent lecture may lead to lapses in attention, resulting in poorer performance in that group. These lapses, in turn, contribute to the observed differences in the fluent condition. This modeling approach opens a promising avenue for future research–one that would have remained inaccessible otherwise.</p>
<p>Not everyone will be eager to implement the techniques discussed herein. In such cases, the key question becomes: What is the least problematic approach to handling proportional data? One reasonable option is to fit multiple models tailored to the specific characteristics of your data. For example, if your data contain 0s, you might fit two models: a traditional linear model excluding the 0s, and a logistic model to account for the zero versus non-zero distinction. If your data contain both 0s and 1s, you could fit separate models for the 0s and 1s in addition to the OLS model. There are many defensible strategies to choose from depending on the context. However, we do not recommend transforming the values of your data (e.g., 0s to .01 and 1s to .99) or ignoring the properties of your data simply to fit traditional statistical models.</p>
<p>In this tutorial, we demonstrated how to analyze these models from a Bayesian perspective. While we recognize that not everyone identifies as a Bayesian, implementing these models using a Bayesian framework is relatively straightforward–it requires only a single package, lowering the barrier to entry. For those who prefer frequentist analyses, several R packages are available. For standard Beta regression, the <code>betareg</code> package <span class="citation" data-cites="betareg">(Cribari-Neto &amp; Zeileis, 2010)</span> is a solid option, while more complex models such as zero-inflated and ordered Beta regressions can be implemented using <code>glmmTMB</code> <span class="citation" data-cites="glmmTMB">(Brooks et al., 2017)</span>. For fitting zero-one models, there are is a new implementation in the <code>betareg</code> package <span class="citation" data-cites="betareg">(Cribari-Neto &amp; Zeileis, 2010)</span> called the XBX regression model <span class="citation" data-cites="kosmidis2025">(Kosmidis &amp; Zeileis, 2025)</span> that allows you to model these types of data.</p>
<section id="conclusion" class="level2">
<h2 data-anchor-id="conclusion">Conclusion</h2>
<p>Overall, this tutorial emphasizes the importance of modeling the data you have. Although the example provided is relatively simple (a one-factor model with two levels), we hope it demonstrates that even with a basic dataset, there is much nuance in interpretation and inference. Properly modeling your data can lead to deeper insights, far beyond what traditional measures might offer. With the tools introduced in this tutorial, researchers now have the means to analyze their data effectively, uncover patterns, make accurate predictions, and support their findings with robust statistical evidence. By applying these modeling techniques, researchers can improve the validity and reliability of their studies, ultimately leading to more informed decisions and advancements in their respective fields.</p>
</section>
</section>
<section id="references" class="level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list">
<div id="ref-marginaleffects" class="csl-entry" role="listitem">
Arel-Bundock, V. (2024). <em>Marginaleffects: Predictions, comparisons, slopes, marginal means, and hypothesis tests</em>. <a href="https://CRAN.R-project.org/package=marginaleffects">https://CRAN.R-project.org/package=marginaleffects</a>
</div>
<div id="ref-margeffects2024" class="csl-entry" role="listitem">
Arel-Bundock, V., Greifer, N., &amp; Heiss, A. (2024). How to interpret statistical models using <span class="nocase">marginaleffects</span> for <span>R</span> and <span>Python</span>. <em>Journal of Statistical Software</em>, <em>111</em>(9), 1–32. <a href="https://doi.org/10.18637/jss.v111.i09">https://doi.org/10.18637/jss.v111.i09</a>
</div>
<div id="ref-bartlett1936" class="csl-entry" role="listitem">
Bartlett, M. S. (1936). The Square Root Transformation in Analysis of Variance. <em>Journal of the Royal Statistical Society Series B: Statistical Methodology</em>, <em>3</em>(1), 68–78. <a href="https://doi.org/10.2307/2983678">https://doi.org/10.2307/2983678</a>
</div>
<div id="ref-bendixen2023" class="csl-entry" role="listitem">
Bendixen, T., &amp; Purzycki, B. G. (2023). Cognitive and cultural models in psychological science: A tutorial on modeling free-list data as a dependent variable in Bayesian regression. <em>Psychological Methods</em>. <a href="https://doi.org/10.1037/met0000553">https://doi.org/10.1037/met0000553</a>
</div>
<div id="ref-glmmTMB" class="csl-entry" role="listitem">
Brooks, M. E., Kristensen, K., van, K. J., Magnusson, A., Berg, C. W., Nielsen, A., Skaug, H. J., Maechler, M., &amp; Bolker, B. M. (2017). <em><span></span>glmmTMB<span></span> balances speed and flexibility among packages for zero-inflated generalized linear mixed modeling</em>. <em>9</em>. <a href="https://doi.org/10.32614/RJ-2017-066">https://doi.org/10.32614/RJ-2017-066</a>
</div>
<div id="ref-brms" class="csl-entry" role="listitem">
Bürkner, P.-C. (2017). <em><span></span>Brms<span></span>: An <span></span>r<span></span> package for <span></span>bayesian<span></span> multilevel models using <span></span>stan<span></span></em>. <em>80</em>. <a href="https://doi.org/10.18637/jss.v080.i01">https://doi.org/10.18637/jss.v080.i01</a>
</div>
<div id="ref-posterior2025a" class="csl-entry" role="listitem">
Bürkner, P.-C., Gabry, J., Kay, M., &amp; Vehtari, A. (2025). <em><span class="nocase">posterior</span>: Tools for working with posterior distributions</em>. <a href="https://mc-stan.org/posterior/">https://mc-stan.org/posterior/</a>
</div>
<div id="ref-bürkner2019" class="csl-entry" role="listitem">
Bürkner, P.-C., &amp; Vuorre, M. (2019). Ordinal Regression Models in Psychology: A Tutorial. <em>Advances in Methods and Practices in Psychological Science</em>, <em>2</em>(1), 77–101. <a href="https://doi.org/10.1177/2515245918823199">https://doi.org/10.1177/2515245918823199</a>
</div>
<div id="ref-carpenter2013" class="csl-entry" role="listitem">
Carpenter, S. K., Wilford, M. M., Kornell, N., &amp; Mullaney, K. M. (2013). Appearances can be deceiving: instructor fluency increases perceptions of learning without increasing actual learning. <em>Psychonomic Bulletin &amp; Review</em>, <em>20</em>(6), 1350–1356. <a href="https://doi.org/10.3758/s13423-013-0442-z">https://doi.org/10.3758/s13423-013-0442-z</a>
</div>
<div id="ref-cohen1977" class="csl-entry" role="listitem">
Cohen, J. (1977). <em>Statistical power analysis for the behavioral sciences, rev. ed.</em> Lawrence Erlbaum Associates, Inc.
</div>
<div id="ref-coretta2025bayesian" class="csl-entry" role="listitem">
Coretta, S., &amp; Bürkner, P.-C. (2025). <em>Bayesian beta regressions with brms in r: A tutorial for phoneticians</em>. <a href="https://doi.org/10.31219/osf.io/f9rqg_v1" class="uri">https://doi.org/10.31219/osf.io/f9rqg_v1</a>.
</div>
<div id="ref-Costello2024" class="csl-entry" role="listitem">
Costello, T. H. et al. (2024). Durably reducing conspiracy beliefs through dialogues with AI. <em>Science</em>, <em>385</em>, eadq1814. <a href="https://doi.org/10.1126/science.adq1814">https://doi.org/10.1126/science.adq1814</a>
</div>
<div id="ref-betareg" class="csl-entry" role="listitem">
Cribari-Neto, F., &amp; Zeileis, A. (2010). <em>Beta regression in <span></span>r<span></span></em>. <em>34</em>. <a href="https://doi.org/10.18637/jss.v034.i02">https://doi.org/10.18637/jss.v034.i02</a>
</div>
<div id="ref-nix" class="csl-entry" role="listitem">
Dolstra, E., &amp; contributors, T. N. (2006). <em>Nix</em> [Computer software]. <a href="https://nixos.org/">https://nixos.org/</a>
</div>
<div id="ref-ferrari2004" class="csl-entry" role="listitem">
Ferrari, S., &amp; Cribari-Neto, F. (2004). Beta Regression for Modelling Rates and Proportions. <em>Journal of Applied Statistics</em>, <em>31</em>(7), 799–815. <a href="https://doi.org/10.1080/0266476042000214501">https://doi.org/10.1080/0266476042000214501</a>
</div>
<div id="ref-Fullerton2023" class="csl-entry" role="listitem">
Fullerton, A. S., &amp; Anderson, K. F. (2021). Ordered Regression Models: a Tutorial. <em>Prevention Science</em>, <em>24</em>(3), 431–443. <a href="https://doi.org/10.1007/s11121-021-01302-y">https://doi.org/10.1007/s11121-021-01302-y</a>
</div>
<div id="ref-cmdstanr" class="csl-entry" role="listitem">
Gabry, J., Češnovar, R., Johnson, A., &amp; Bronder, S. (2024). <em>Cmdstanr: R interface to ’CmdStan’</em>. <a href="https://mc-stan.org/cmdstanr/">https://mc-stan.org/cmdstanr/</a>
</div>
<div id="ref-gelman2013bayesian" class="csl-entry" role="listitem">
Gelman, A., Carlin, J. B., Stern, H. S., Dunson, D. B., Vehtari, A., &amp; Rubin, D. B. (2013). <em>Bayesian data analysis</em> (Third). CRC. <a href="https://stat.columbia.edu/~gelman/book/">https://stat.columbia.edu/~gelman/book/</a>
</div>
<div id="ref-heiss2021" class="csl-entry" role="listitem">
Heiss, A. (2021). <em>A guide to modeling proportions with bayesian beta and zero-inflated beta regression models</em>. <a href="http://dx.doi.org/10.59350/7p1a4-0tw75">http://dx.doi.org/10.59350/7p1a4-0tw75</a>
</div>
<div id="ref-james2025" class="csl-entry" role="listitem">
James, A. N., Ryskin, R., Hartshorne, J. K., Backs, H., Bala, N., Barcenas-Meade, L., Bhattarai, S., Charles, T., Copoulos, G., Coss, C., Eisert, A., Furuhashi, E., Ginell, K., Guttman-McCabe, A., Harrison, E. (Chaz)., Hoban, L., Hwang, W. A., Iannetta, C., Koenig, K. M., … Leeuw, J. R. de. (2025). What Paradigms Can Webcam Eye-Tracking Be Used For? Attempted Replications of Five Cognitive Science Experiments. <em>Collabra: Psychology</em>, <em>11</em>(1). <a href="https://doi.org/10.1525/collabra.140755">https://doi.org/10.1525/collabra.140755</a>
</div>
<div id="ref-jeffries1961" class="csl-entry" role="listitem">
Jeffries, H. (1961). <em>Theory of probability</em>. Clarendon Press, Oxford.
</div>
<div id="ref-bayes_rules" class="csl-entry" role="listitem">
Johnson, A., Ott, M., &amp; Dogucu, M. (2022). <em>Bayes rules!: An introduction to applied bayesian modeling</em>. Routledge &amp; CRC Press.
</div>
<div id="ref-kass1995" class="csl-entry" role="listitem">
Kass, R. E., &amp; Raftery, A. E. (1995). Bayes Factors. <em>Journal of the American Statistical Association</em>, <em>90</em>(430), 773–795. <a href="https://doi.org/10.1080/01621459.1995.10476572">https://doi.org/10.1080/01621459.1995.10476572</a>
</div>
<div id="ref-kong2016" class="csl-entry" role="listitem">
Kong, E. J., &amp; Edwards, J. (2016). Individual differences in categorical perception of speech: Cue weighting and executive function. <em>Journal of Phonetics</em>, <em>59</em>, 40–57. <a href="https://doi.org/10.1016/j.wocn.2016.08.006">https://doi.org/10.1016/j.wocn.2016.08.006</a>
</div>
<div id="ref-kornell2008" class="csl-entry" role="listitem">
Kornell, N., &amp; Bjork, R. A. (2008). Learning Concepts and Categories. <em>Psychological Science</em>, <em>19</em>(6), 585–592. <a href="https://doi.org/10.1111/j.1467-9280.2008.02127.x">https://doi.org/10.1111/j.1467-9280.2008.02127.x</a>
</div>
<div id="ref-kosmidis2025" class="csl-entry" role="listitem">
Kosmidis, I., &amp; Zeileis, A. (2025). Extended-support beta regression for [0, 1] responses. <em>Journal of the Royal Statistical Society Series C: Applied Statistics</em>. <a href="https://doi.org/10.1093/jrsssc/qlaf039">https://doi.org/10.1093/jrsssc/qlaf039</a>
</div>
<div id="ref-kruschke2015doing" class="csl-entry" role="listitem">
Kruschke, J. K. (2015). <em>Doing bayesian data analysis: A tutorial with r, JAGS, and stan</em> (2nd ed.). Academic Press.
</div>
<div id="ref-kubinec2022" class="csl-entry" role="listitem">
Kubinec, R. (2022). Ordered Beta Regression: A Parsimonious, Well-Fitting Model for Continuous Data with Lower and Upper Bounds. <em>Political Analysis</em>, <em>31</em>(4), 519–536. <a href="https://doi.org/10.1017/pan.2022.20">https://doi.org/10.1017/pan.2022.20</a>
</div>
<div id="ref-ordbetareg" class="csl-entry" role="listitem">
Kubinec, R. (2023). <em>Ordbetareg: Ordered beta regression models with ’brms’</em>. <a href="https://CRAN.R-project.org/package=ordbetareg">https://CRAN.R-project.org/package=ordbetareg</a>
</div>
<div id="ref-emmeans" class="csl-entry" role="listitem">
Lenth, R. V. (2025). <em>Emmeans: Estimated marginal means, aka least-squares means</em>. <a href="https://doi.org/10.32614/CRAN.package.emmeans">https://doi.org/10.32614/CRAN.package.emmeans</a>
</div>
<div id="ref-liu2015" class="csl-entry" role="listitem">
Liu, F., &amp; Kong, Y. (2015). zoib: An R Package for Bayesian Inference for Beta Regression and Zero/One Inflated Beta Regression. <em>The R Journal</em>, <em>7</em>(2), 34. <a href="https://doi.org/10.32614/rj-2015-019">https://doi.org/10.32614/rj-2015-019</a>
</div>
<div id="ref-easystats" class="csl-entry" role="listitem">
Lüdecke, D., Ben-Shachar, M. S., Patil, I., Wiernik, B. M., Bacher, E., Thériault, R., &amp; Makowski, D. (2022). <em>Easystats: Framework for easy statistical modeling, visualization, and reporting</em>. <a href="https://easystats.github.io/easystats/">https://easystats.github.io/easystats/</a>
</div>
<div id="ref-makowski2019a" class="csl-entry" role="listitem">
Makowski, D., Ben-Shachar, M. S., Chen, S. H. A., &amp; Lüdecke, D. (2019). Indices of effect existence and significance in the bayesian framework. <em>Frontiers in Psychology</em>, <em>10</em>. <a href="https://doi.org/10.3389/fpsyg.2019.02767">https://doi.org/10.3389/fpsyg.2019.02767</a>
</div>
<div id="ref-makowski2019" class="csl-entry" role="listitem">
Makowski, D., Ben-Shachar, M., &amp; Lüdecke, D. (2019). bayestestR: Describing effects and their uncertainty, existence and significance within the bayesian framework. <em>Journal of Open Source Software</em>, <em>4</em>(40), 1541. <a href="https://doi.org/10.21105/joss.01541">https://doi.org/10.21105/joss.01541</a>
</div>
<div id="ref-Makowski2025CHOCO" class="csl-entry" role="listitem">
Makowski, D., Neves, A., &amp; Field, A. P. (2025). <em>Introducing the choice-confidence (CHOCO) model for bimodal data from subjective ratings: Application to the effect of attractiveness on reality beliefs about AI-generated faces</em>. <a href="https://doi.org/10.31234/osf.io/z68v3_v1">https://doi.org/10.31234/osf.io/z68v3_v1</a>
</div>
<div id="ref-marsmanThreeInsightsBayesian2016" class="csl-entry" role="listitem">
Marsman, M., &amp; Wagenmakers, E.-J. (2016). Three <span>Insights</span> from a <span>Bayesian Interpretation</span> of the <span>One-Sided P Value</span>. <em>Educational and Psychological Measurement</em>. <a href="https://doi.org/10.1177/0013164416669201">https://doi.org/10.1177/0013164416669201</a>
</div>
<div id="ref-martinVocalComplexitySocially2024" class="csl-entry" role="listitem">
Martin, K., Cornero, F. M., Clayton, N. S., Adam, O., Obin, N., &amp; Dufour, V. (2024). Vocal complexity in a socially complex corvid: Gradation, diversity and lack of common call repertoire in male rooks. <em>Royal Society Open Science</em>, <em>11</em>(1), 231713. <a href="https://doi.org/10.1098/rsos.231713">https://doi.org/10.1098/rsos.231713</a>
</div>
<div id="ref-mcelreath2020statistical" class="csl-entry" role="listitem">
McElreath, R. (2020). <em>Statistical rethinking: A bayesian course with examples in r and STAN</em> (2nd ed.). Chapman; Hall/CRC. <a href="https://doi.org/10.1201/9780429029608">https://doi.org/10.1201/9780429029608</a>
</div>
<div id="ref-nelder1972" class="csl-entry" role="listitem">
Nelder, J. A., &amp; Wedderburn, R. W. M. (1972). Generalized linear models. <em>Journal of the Royal Statistical Society. Series A (General)</em>, <em>135</em>(3), 370–384. <a href="https://doi.org/10.2307/2344614">https://doi.org/10.2307/2344614</a>
</div>
<div id="ref-nouvianGlyphosateImpairsAversive2023" class="csl-entry" role="listitem">
Nouvian, M., Foster, J. J., &amp; Weidenmüller, A. (2023). Glyphosate impairs aversive learning in bumblebees. <em>Science of The Total Environment</em>, <em>898</em>, 165527. <a href="https://www.sciencedirect.com/science/article/pii/S0048969723041505">https://www.sciencedirect.com/science/article/pii/S0048969723041505</a>
</div>
<div id="ref-paolino2001" class="csl-entry" role="listitem">
Paolino, P. (2001). Maximum Likelihood Estimation of Models with Beta-Distributed Dependent Variables. <em>Political Analysis</em>, <em>9</em>(4), 325–346. <a href="https://doi.org/10.1093/oxfordjournals.pan.a004873">https://doi.org/10.1093/oxfordjournals.pan.a004873</a>
</div>
<div id="ref-R" class="csl-entry" role="listitem">
R Core Team. (2024). <em>R: A language and environment for statistical computing</em>. R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>
</div>
<div id="ref-rix" class="csl-entry" role="listitem">
Rodrigues, B., &amp; Baumann, P. (2025). <em>Rix: Reproducible data science environments with ’nix’</em>. <a href="https://docs.ropensci.org/rix/">https://docs.ropensci.org/rix/</a>
</div>
<div id="ref-shresthaBigFiveTraits2024" class="csl-entry" role="listitem">
Shrestha, S., Sigdel, K., Pokharel, M., &amp; Columbus, S. (2024). Big five traits predict between- and within-person variation in loneliness. <em>European Journal of Personality</em>, 08902070241239834. <a href="https://doi.org/10.1177/08902070241239834">https://doi.org/10.1177/08902070241239834</a>
</div>
<div id="ref-sladekova2024unicorns" class="csl-entry" role="listitem">
Sladekova, M., &amp; Field, A. P. (2024). <em>In search of unicorns: Assessing statistical assumptions in real psychology datasets</em>. <a href="https://doi.org/10.31234/osf.io/4rznt">https://doi.org/10.31234/osf.io/4rznt</a>
</div>
<div id="ref-smithEcologicalMomentaryAssessment2024" class="csl-entry" role="listitem">
Smith, K. E., Panlilio, L. V., Feldman, J. D., Grundmann, O., Dunn, K. E., McCurdy, C. R., Garcia-Romeu, A., &amp; Epstein, D. H. (2024). Ecological momentary assessment of self-reported kratom use, effects, and motivations among US adults. <em>JAMA Network Open</em>, <em>7</em>(1), e2353401. <a href="https://doi.org/10.1001/jamanetworkopen.2023.53401">https://doi.org/10.1001/jamanetworkopen.2023.53401</a>
</div>
<div id="ref-smithson2006" class="csl-entry" role="listitem">
Smithson, M., &amp; Verkuilen, J. (2006). A better lemon squeezer? Maximum-likelihood regression with beta-distributed dependent variables. <em>Psychological Methods</em>, <em>11</em>(1), 54–71. <a href="https://doi.org/10.1037/1082-989x.11.1.54">https://doi.org/10.1037/1082-989x.11.1.54</a>
</div>
<div id="ref-stan2023" class="csl-entry" role="listitem">
Team, S. D. (2023). <em>Stan: A probabilistic programming language</em>. <a href="https://mc-stan.org">https://mc-stan.org</a>
</div>
<div id="ref-toftness2017" class="csl-entry" role="listitem">
Toftness, A. R., Carpenter, S. K., Geller, J., Lauber, S., Johnson, M., &amp; Armstrong, P. I. (2017). Instructor fluency leads to higher confidence in learning, but not better learning. <em>Metacognition and Learning</em>, <em>13</em>(1), 1–14. <a href="https://doi.org/10.1007/s11409-017-9175-0">https://doi.org/10.1007/s11409-017-9175-0</a>
</div>
<div id="ref-vuorre2019" class="csl-entry" role="listitem">
Vuorre, M. (2019, February 18). <em>How to Analyze Visual Analog (Slider) Scale Data?</em> <a href="https://vuorre.com/posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-beta-models">https://vuorre.com/posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-beta-models</a>
</div>
<div id="ref-wickhamDataScienceImport2023" class="csl-entry" role="listitem">
Wickham, H., Çetinkaya-Rundel, M., &amp; Grolemund, G. (2023). <em>R for <span>Data Science</span>: <span>Import</span>, <span>Tidy</span>, <span>Transform</span>, <span>Visualize</span>, and <span>Model Data</span></em>. O’Reilly. <a href="https://r4ds.hadley.nz/">https://r4ds.hadley.nz/</a>
</div>
<div id="ref-wilford2020" class="csl-entry" role="listitem">
Wilford, M. M., Kurpad, N., Platt, M., &amp; Weinstein-Jones, Y. (2020). Lecturer fluency can impact students’ judgments of learning and actual learning performance. <em>Applied Cognitive Psychology</em>, <em>34</em>(6), 1444–1456. <a href="https://doi.org/10.1002/acp.3724">https://doi.org/10.1002/acp.3724</a>
</div>
<div id="ref-wilkesQuantifyingCoextinctionsEcosystem2024" class="csl-entry" role="listitem">
Wilkes, L. N., Barner, A. K., Keyes, A. A., Morton, D., Byrnes, J. E. K., &amp; Dee, L. E. (2024). Quantifying co‐extinctions and ecosystem service vulnerability in coastal ecosystems experiencing climate warming. <em>Global Change Biology</em>, <em>30</em>(7), e17422. <a href="https://doi.org/10.1111/gcb.17422">https://doi.org/10.1111/gcb.17422</a>
</div>
<div id="ref-witherby2022" class="csl-entry" role="listitem">
Witherby, A. E., &amp; Carpenter, S. K. (2022). The impact of lecture fluency and technology fluency on students<span>’</span> online learning and evaluations of instructors. <em>Journal of Applied Research in Memory and Cognition</em>, <em>11</em>(4), 500–509. <a href="https://doi.org/10.1037/mac0000003">https://doi.org/10.1037/mac0000003</a>
</div>
<div id="ref-yeEnforcingBoundariesChinas2023" class="csl-entry" role="listitem">
Ye, M., &amp; Heiss, A. (2023). Enforcing <span>Boundaries</span>: <span>China</span>’s <span>Overseas NGO Law</span> and <span>Operational Constraints</span> for <span>Global Civil Society</span>. <em>Working Paper</em>. <a href="https://stats.andrewheiss.com/compassionate-clam/manuscript/output/manuscript.html">https://stats.andrewheiss.com/compassionate-clam/manuscript/output/manuscript.html</a>
</div>
</div>
<!-- -->

</section>


<div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p>Specifically, for any continuous bounded variable <span class="math inline">x</span>, we can rescale this variable to lie within 0 and 1 by using the formula <span class="math inline">x' = \frac{x - \min(x)}{\max(x) - \min(x)}</span> where <span class="math inline">0 \leq x' \leq 1</span>.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>In this article, we try to limit code where possible; however, the online version has all the code needed to reproduce all analyses herein. Furthermore, to promote transparency and reproducibility, the tutorial was written in R version 4.5.1 (<span class="citation" data-cites="R">R Core Team (2024)</span>) using Quarto (v.1.5.54), an open-source publishing system that allows for dynamic and static documents. This allows figures, tables, and text to be programmatically included directly in the manuscript, ensuring that all results are seamlessly integrated into the document. In addition, we use the <code>rix</code> <span class="citation" data-cites="rix">(Rodrigues &amp; Baumann, 2025)</span> R package which harnesses the power of the <code>nix</code> <span class="citation" data-cites="nix">(Dolstra &amp; contributors, 2006)</span> ecosystem to to help with computational reproducibility. Not only does this give us a snapshot of the packages used to create the current manuscript, but it also takes a snapshot of system dependencies used at run-time. This way reproducers can easily re-use the exact same environment by installing the <code>nix</code> package manager and using the included default.nix file to set up the right environment. The README file in the GitHub repository contains detailed information on how to set this up to reproduce the contents of the current manuscript, including a video.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3"><p>Technically, this maximum entropy condition is satisfied because the Beta(1,1) distribution is uniform over its support.<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4"><p>In order to use the <code>cmdstanr</code> backend you will need to first install the package (see <code>https://mc-stan.org/cmdstanr/</code>) and also run <code>cmdstanr::install_cmdstan()</code> if you have not done so already.<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5"><p>The model fit statistic LOO-CV can be compared for any set of fitted <code>brms</code> models with the function <code>loo()</code>.<a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn6"><p><code>bayestestr</code> also has functions that will allow users to compute Bayes Factors.<a href="#fnref6" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn7"><p>In cases where your data include exact 1s but no 0s, you can fit a one-inflated Beta regression model in <code>brms</code> by setting the <code>coi</code> parameter to <code>1</code>. This tells the model that all point masses occur at 1, rather than being split between 0 and 1. In other words, <code>coi = 1</code> assumes that any inflation in the data is due entirely to values at 1. In our data, we have exactly one value equal to 1[^6]. While probably not significant to alter our findings, we can model 1s with a special type of model called the zero-one-inflated Beta (ZOIB) model <span class="citation" data-cites="liu2015">(Liu &amp; Kong, 2015)</span> if we believe that both 0s and 1s are distinct outcomes.<a href="#fnref7" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn8"><p>For a more complete description of this issue, we refer the reader to <span class="citation" data-cites="kubinec2022">Kubinec (2022)</span>.<a href="#fnref8" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb3" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "A Beta Way: A Tutorial For Using Beta Regression in Psychological Research"</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a><span class="an">shorttitle:</span><span class="co"> "Beta Regression Tutorial"</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Jason Geller</span></span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a><span class="co">    corresponding: true</span></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0002-7459-4505</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a><span class="co">    email: drjasongeller@gmail.com</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliations:</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a><span class="co">      - name: Boston College</span></span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a><span class="co">        department: Department of Psychology and Neuroscience</span></span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a><span class="co">        address: McGuinn 300z</span></span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a><span class="co">        city: Chestnut Hill</span></span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a><span class="co">        region: MA</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a><span class="co">        country: USA</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a><span class="co">        postal-code: 02467</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a><span class="co">    roles:</span></span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a><span class="co">      - Conceptualization</span></span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a><span class="co">      - Data curation</span></span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a><span class="co">      - Formal analysis</span></span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a><span class="co">      - Project administration</span></span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a><span class="co">      - Resources</span></span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a><span class="co">      - Visualization</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a><span class="co">      - Writing - original draft</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Robert Kubinec</span></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0001-6655-4119</span></span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliations:</span></span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a><span class="co">      - name: University of South Carolina</span></span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a><span class="co">    roles:</span></span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a><span class="co">     - Formal analysis</span></span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a><span class="co">     - Validation</span></span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a><span class="co">     - Writing - review &amp; editing</span></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Chelsea M. Parlett Pelleriti</span></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0001-9301-1398</span></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliations:</span></span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a><span class="co">      - name: Canva</span></span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a><span class="co">    roles:</span></span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a><span class="co">     - Formal analysis</span></span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a><span class="co">     - Writing - review &amp; editing</span></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Matti Vuorre</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0001-5052-066X</span></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliations:</span></span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a><span class="co">      - name: Tilburg University</span></span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a><span class="co">    roles:</span></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a><span class="co">     - Formal analysis</span></span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a><span class="co">     - Resources</span></span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a><span class="co">     - Supervision</span></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a><span class="co">     - Validation</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a><span class="co">     - Writing - review &amp; editing</span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a><span class="an">author-note:</span></span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a><span class="co">  status-changes:</span></span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-change: ~</span></span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a><span class="co">    deceased: ~</span></span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a><span class="co">  disclosures:</span></span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a><span class="co">    # Example: This study was registered at X (Identifier Y).</span></span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a><span class="co">    study-registration: "No preregistration."</span></span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a><span class="co">    # Acknowledge and cite data/materials to be shared.</span></span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a><span class="co">    data-sharing: Data, code, and materials for this manuscript can be found at https://github.com/jgeller112/Beta_regression_tutorial </span></span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a><span class="co">    related-report: ~</span></span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a><span class="co">    conflict-of-interest: The authors have no conflicts of interest to disclose.</span></span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a><span class="co">    financial-support: ~</span></span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a><span class="co">    authorship-agreements: ~</span></span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a><span class="an">abstract:</span><span class="co"> "Rates, percentages, and proportions are common outcomes in psychology and the social sciences. These outcomes are often analyzed using models that assume normality, but this practice overlooks important features of the data, such as their natural bounds at 0 and 1. As a result, estimates can become distorted. In contrast, treating such outcomes as Beta-distributed respects these limits and can yield more accurate estimates. Despite these advantages, the use of Beta models in applied research remains limited. Our goal is to provide researchers with practical guidance for adopting Beta regression models, illustrated with an example drawn from the psychological literature. We begin by introducing the Beta distribution and Beta regression, emphasizing key components and assumptions. Next, using data from a learning and memory study, we demonstrate how to fit a Beta regression model in R with the Bayesian package `brms` and how to interpret results on the response scale. We also discuss model extensions, including zero-inflated, zero- and one-inflated, and ordered Beta models. Basic familarity with regression modeling and R is assumed. To promote wider adoption of these methods, we provide detailed code and materials at https://github.com/jgeller112/Beta_regression_tutorial."</span></span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a><span class="an">keywords:</span><span class="co"> [Beta regression, Beta distribution, R, Tutorial, Psychology, Learning and memory]</span></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a><span class="an">floatsintext:</span><span class="co"> true</span></span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Numbered lines (.pdf and .docx only)</span></span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a><span class="an">numbered-lines:</span><span class="co"> true</span></span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a><span class="co"># File with references</span></span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a><span class="an">suppress-title-page:</span><span class="co"> false</span></span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a><span class="co"># Link citations to references</span></span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a><span class="an">link-citations:</span><span class="co"> false</span></span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a><span class="co"># Masks references that appear in the masked-citations list</span></span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a><span class="an">mask:</span><span class="co"> false</span></span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a><span class="an">word-count:</span><span class="co"> true</span></span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a><span class="co"># Language options. See https://quarto.org/docs/authoring/language.html</span></span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a><span class="an">lang:</span><span class="co"> en</span></span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a><span class="an">language:</span></span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a><span class="co">  citation-last-author-separator: "and"</span></span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a><span class="co">  citation-masked-author: "Masked Citation"</span></span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a><span class="co">  citation-masked-date: "n.d."</span></span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a><span class="co">  citation-masked-title: "Masked Title"</span></span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a><span class="co">  email: "Email"</span></span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a><span class="co">  title-block-author-note: "Author Note"</span></span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a><span class="co">  title-block-correspondence-note: "Correspondence concerning this article should be addressed to"</span></span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a><span class="co">  title-block-role-introduction: "Author roles were classified using the Contributor Role Taxonomy (CRediT; https://credit.niso.org/) as follows:"</span></span>
<span id="cb3-87"><a href="#cb3-87" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> bibliography.bib</span></span>
<span id="cb3-88"><a href="#cb3-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-89"><a href="#cb3-89" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb3-90"><a href="#cb3-90" aria-hidden="true" tabindex="-1"></a><span class="co">  echo: false</span></span>
<span id="cb3-91"><a href="#cb3-91" aria-hidden="true" tabindex="-1"></a><span class="co">  cache: false</span></span>
<span id="cb3-92"><a href="#cb3-92" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb3-93"><a href="#cb3-93" aria-hidden="true" tabindex="-1"></a><span class="co">  message: false</span></span>
<span id="cb3-94"><a href="#cb3-94" aria-hidden="true" tabindex="-1"></a><span class="co">  results: hide</span></span>
<span id="cb3-95"><a href="#cb3-95" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: auto</span></span>
<span id="cb3-96"><a href="#cb3-96" aria-hidden="true" tabindex="-1"></a><span class="co">  fig-align: center</span></span>
<span id="cb3-97"><a href="#cb3-97" aria-hidden="true" tabindex="-1"></a><span class="co">  tbl-align: center</span></span>
<span id="cb3-98"><a href="#cb3-98" aria-hidden="true" tabindex="-1"></a><span class="co">  code-line-numbers: true</span></span>
<span id="cb3-99"><a href="#cb3-99" aria-hidden="true" tabindex="-1"></a><span class="co">  keep-with-next: true</span></span>
<span id="cb3-100"><a href="#cb3-100" aria-hidden="true" tabindex="-1"></a><span class="co">  code-overflow: wrap</span></span>
<span id="cb3-101"><a href="#cb3-101" aria-hidden="true" tabindex="-1"></a><span class="co">  fig-width: 6</span></span>
<span id="cb3-102"><a href="#cb3-102" aria-hidden="true" tabindex="-1"></a><span class="co">  fig-asp: 0.618</span></span>
<span id="cb3-103"><a href="#cb3-103" aria-hidden="true" tabindex="-1"></a><span class="co">  fig-pos: "h"</span></span>
<span id="cb3-104"><a href="#cb3-104" aria-hidden="true" tabindex="-1"></a><span class="co">  tbl-pos: "h"</span></span>
<span id="cb3-105"><a href="#cb3-105" aria-hidden="true" tabindex="-1"></a><span class="co">  fig-dpi: 300</span></span>
<span id="cb3-106"><a href="#cb3-106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-107"><a href="#cb3-107" aria-hidden="true" tabindex="-1"></a><span class="an">knitr:</span></span>
<span id="cb3-108"><a href="#cb3-108" aria-hidden="true" tabindex="-1"></a><span class="co">  opts_chunk:</span></span>
<span id="cb3-109"><a href="#cb3-109" aria-hidden="true" tabindex="-1"></a><span class="co">    dev: "ragg_png"</span></span>
<span id="cb3-110"><a href="#cb3-110" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb3-111"><a href="#cb3-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-112"><a href="#cb3-112" aria-hidden="true" tabindex="-1"></a>::: callout-warning</span>
<span id="cb3-113"><a href="#cb3-113" aria-hidden="true" tabindex="-1"></a><span class="fu">## Preprint</span></span>
<span id="cb3-114"><a href="#cb3-114" aria-hidden="true" tabindex="-1"></a>This manuscript is currently **under review** and has not been peer-reviewed. </span>
<span id="cb3-115"><a href="#cb3-115" aria-hidden="true" tabindex="-1"></a>Content is **subject to change**. Please feel free to provide feedback!</span>
<span id="cb3-116"><a href="#cb3-116" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb3-117"><a href="#cb3-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-118"><a href="#cb3-118" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-121"><a href="#cb3-121" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-122"><a href="#cb3-122" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: packages_1</span></span>
<span id="cb3-123"><a href="#cb3-123" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-124"><a href="#cb3-124" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb3-125"><a href="#cb3-125" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb3-126"><a href="#cb3-126" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-127"><a href="#cb3-127" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(easystats)</span>
<span id="cb3-128"><a href="#cb3-128" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb3-129"><a href="#cb3-129" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tinytable)</span>
<span id="cb3-130"><a href="#cb3-130" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(marginaleffects)</span>
<span id="cb3-131"><a href="#cb3-131" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(extraDistr)</span>
<span id="cb3-132"><a href="#cb3-132" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="cb3-133"><a href="#cb3-133" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(posterior)</span>
<span id="cb3-134"><a href="#cb3-134" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bayesplot)</span>
<span id="cb3-135"><a href="#cb3-135" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggdist)</span>
<span id="cb3-136"><a href="#cb3-136" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb3-137"><a href="#cb3-137" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrain)</span>
<span id="cb3-138"><a href="#cb3-138" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb3-139"><a href="#cb3-139" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggokabeito)</span>
<span id="cb3-140"><a href="#cb3-140" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ordbetareg)</span>
<span id="cb3-141"><a href="#cb3-141" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb3-142"><a href="#cb3-142" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb3-143"><a href="#cb3-143" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cmdstanr)</span>
<span id="cb3-144"><a href="#cb3-144" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(webshot2)</span>
<span id="cb3-145"><a href="#cb3-145" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(performance)</span>
<span id="cb3-146"><a href="#cb3-146" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(parameters)</span>
<span id="cb3-147"><a href="#cb3-147" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb3-148"><a href="#cb3-148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-149"><a href="#cb3-149" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(</span>
<span id="cb3-150"><a href="#cb3-150" aria-hidden="true" tabindex="-1"></a>  <span class="at">scipen =</span> <span class="dv">999</span>,</span>
<span id="cb3-151"><a href="#cb3-151" aria-hidden="true" tabindex="-1"></a>  <span class="at">marginaleffects_posterior_center =</span> mean,</span>
<span id="cb3-152"><a href="#cb3-152" aria-hidden="true" tabindex="-1"></a>  <span class="at">brms.short_summary =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-153"><a href="#cb3-153" aria-hidden="true" tabindex="-1"></a>  <span class="at">rms.backend =</span> <span class="fu">Sys.getenv</span>(<span class="st">"BRMS_BACKEND"</span>, <span class="st">"cmdstanr"</span>),</span>
<span id="cb3-154"><a href="#cb3-154" aria-hidden="true" tabindex="-1"></a>  <span class="at">brms.threads =</span> <span class="fu">as.numeric</span>(<span class="fu">Sys.getenv</span>(<span class="st">"BRMS_THREADS"</span>, <span class="dv">2</span>)),</span>
<span id="cb3-155"><a href="#cb3-155" aria-hidden="true" tabindex="-1"></a>  <span class="at">mc.cores =</span> <span class="fu">as.numeric</span>(<span class="fu">Sys.getenv</span>(<span class="st">"MAX_CORES"</span>, <span class="dv">4</span>))</span>
<span id="cb3-156"><a href="#cb3-156" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-157"><a href="#cb3-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-158"><a href="#cb3-158" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-159"><a href="#cb3-159" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">666</span>)</span>
<span id="cb3-160"><a href="#cb3-160" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-161"><a href="#cb3-161" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-164"><a href="#cb3-164" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-165"><a href="#cb3-165" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ggplot-theme</span></span>
<span id="cb3-166"><a href="#cb3-166" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-167"><a href="#cb3-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-168"><a href="#cb3-168" aria-hidden="true" tabindex="-1"></a>IS_HTML <span class="ot">&lt;-</span> knitr<span class="sc">::</span><span class="fu">is_html_output</span>()</span>
<span id="cb3-169"><a href="#cb3-169" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-170"><a href="#cb3-170" aria-hidden="true" tabindex="-1"></a>theme_publication <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb3-171"><a href="#cb3-171" aria-hidden="true" tabindex="-1"></a>  <span class="at">base_size =</span> <span class="fu">ifelse</span>(IS_HTML, <span class="dv">12</span>, <span class="dv">14</span>),</span>
<span id="cb3-172"><a href="#cb3-172" aria-hidden="true" tabindex="-1"></a>  <span class="at">base_family =</span> <span class="st">"Times"</span></span>
<span id="cb3-173"><a href="#cb3-173" aria-hidden="true" tabindex="-1"></a>) {</span>
<span id="cb3-174"><a href="#cb3-174" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb3-175"><a href="#cb3-175" aria-hidden="true" tabindex="-1"></a>    <span class="at">text =</span> <span class="fu">element_text</span>(</span>
<span id="cb3-176"><a href="#cb3-176" aria-hidden="true" tabindex="-1"></a>      <span class="at">size =</span> base_size,</span>
<span id="cb3-177"><a href="#cb3-177" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> base_family,</span>
<span id="cb3-178"><a href="#cb3-178" aria-hidden="true" tabindex="-1"></a>      <span class="at">color =</span> <span class="st">"black"</span></span>
<span id="cb3-179"><a href="#cb3-179" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb3-180"><a href="#cb3-180" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> base_size <span class="sc">+</span> <span class="dv">2</span>),</span>
<span id="cb3-181"><a href="#cb3-181" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> base_size),</span>
<span id="cb3-182"><a href="#cb3-182" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.caption =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> base_size <span class="sc">-</span> <span class="dv">2</span>),</span>
<span id="cb3-183"><a href="#cb3-183" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> base_size),</span>
<span id="cb3-184"><a href="#cb3-184" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">10</span>)),</span>
<span id="cb3-185"><a href="#cb3-185" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">r =</span> <span class="dv">10</span>)),</span>
<span id="cb3-186"><a href="#cb3-186" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size <span class="sc">-</span> <span class="dv">1</span>),</span>
<span id="cb3-187"><a href="#cb3-187" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb3-188"><a href="#cb3-188" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb3-189"><a href="#cb3-189" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"gray80"</span>),</span>
<span id="cb3-190"><a href="#cb3-190" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-191"><a href="#cb3-191" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-192"><a href="#cb3-192" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb3-193"><a href="#cb3-193" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> base_size),</span>
<span id="cb3-194"><a href="#cb3-194" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size <span class="sc">-</span> <span class="dv">1</span>),</span>
<span id="cb3-195"><a href="#cb3-195" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb3-196"><a href="#cb3-196" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> base_size),</span>
<span id="cb3-197"><a href="#cb3-197" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"gray90"</span>, <span class="at">color =</span> <span class="st">"black"</span>)</span>
<span id="cb3-198"><a href="#cb3-198" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-199"><a href="#cb3-199" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-200"><a href="#cb3-200" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-201"><a href="#cb3-201" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-202"><a href="#cb3-202" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-205"><a href="#cb3-205" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-206"><a href="#cb3-206" aria-hidden="true" tabindex="-1"></a><span class="co"># fit models beforehand so tables and figures are in correct place.</span></span>
<span id="cb3-207"><a href="#cb3-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-208"><a href="#cb3-208" aria-hidden="true" tabindex="-1"></a>model1 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"model_reg_bayes.rds"</span>))</span>
<span id="cb3-209"><a href="#cb3-209" aria-hidden="true" tabindex="-1"></a>model2 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(</span>
<span id="cb3-210"><a href="#cb3-210" aria-hidden="true" tabindex="-1"></a>  <span class="st">"manuscript"</span>,</span>
<span id="cb3-211"><a href="#cb3-211" aria-hidden="true" tabindex="-1"></a>  <span class="st">"models"</span>,</span>
<span id="cb3-212"><a href="#cb3-212" aria-hidden="true" tabindex="-1"></a>  <span class="st">"model_beta_bayes_dis_run01.rds"</span></span>
<span id="cb3-213"><a href="#cb3-213" aria-hidden="true" tabindex="-1"></a>))</span>
<span id="cb3-214"><a href="#cb3-214" aria-hidden="true" tabindex="-1"></a>model3 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_zib_model0not1.rds"</span>))</span>
<span id="cb3-215"><a href="#cb3-215" aria-hidden="true" tabindex="-1"></a>model4 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_zoib_model.rds"</span>))</span>
<span id="cb3-216"><a href="#cb3-216" aria-hidden="true" tabindex="-1"></a>model5 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_ordbeta_phi_model.rds"</span>))</span>
<span id="cb3-217"><a href="#cb3-217" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-218"><a href="#cb3-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-221"><a href="#cb3-221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-222"><a href="#cb3-222" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: data-import</span></span>
<span id="cb3-223"><a href="#cb3-223" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-224"><a href="#cb3-224" aria-hidden="true" tabindex="-1"></a><span class="co">#| </span></span>
<span id="cb3-225"><a href="#cb3-225" aria-hidden="true" tabindex="-1"></a>fluency_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"data"</span>, <span class="st">"miko_data.csv"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb3-226"><a href="#cb3-226" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(</span>
<span id="cb3-227"><a href="#cb3-227" aria-hidden="true" tabindex="-1"></a>    <span class="at">Participant =</span> <span class="st">"ResponseID"</span>,</span>
<span id="cb3-228"><a href="#cb3-228" aria-hidden="true" tabindex="-1"></a>    <span class="at">Fluency =</span> <span class="st">"Condition"</span>,</span>
<span id="cb3-229"><a href="#cb3-229" aria-hidden="true" tabindex="-1"></a>    <span class="at">Time =</span> <span class="st">"name"</span>,</span>
<span id="cb3-230"><a href="#cb3-230" aria-hidden="true" tabindex="-1"></a>    <span class="at">Accuracy =</span> <span class="st">"value"</span></span>
<span id="cb3-231"><a href="#cb3-231" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb3-232"><a href="#cb3-232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-233"><a href="#cb3-233" aria-hidden="true" tabindex="-1"></a>    <span class="at">Accuracy =</span> Accuracy <span class="sc">/</span> <span class="dv">10</span>,</span>
<span id="cb3-234"><a href="#cb3-234" aria-hidden="true" tabindex="-1"></a>    <span class="at">Fluency =</span> <span class="fu">ifelse</span>(Fluency <span class="sc">==</span> <span class="dv">1</span>, <span class="st">"Fluent"</span>, <span class="st">"Disfluent"</span>)</span>
<span id="cb3-235"><a href="#cb3-235" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb3-236"><a href="#cb3-236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Time <span class="sc">==</span> <span class="st">"FreeCt1AVG"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-237"><a href="#cb3-237" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>X, <span class="sc">-</span>Time) <span class="sc">%&gt;%</span></span>
<span id="cb3-238"><a href="#cb3-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Participant =</span> <span class="fu">dense_rank</span>(Participant)) <span class="sc">%&gt;%</span></span>
<span id="cb3-239"><a href="#cb3-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(Accuracy, <span class="at">.after =</span> <span class="fu">last_col</span>())</span>
<span id="cb3-240"><a href="#cb3-240" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-241"><a href="#cb3-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-242"><a href="#cb3-242" aria-hidden="true" tabindex="-1"></a><span class="fu"># Introduction</span></span>
<span id="cb3-243"><a href="#cb3-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-244"><a href="#cb3-244" aria-hidden="true" tabindex="-1"></a>Many key outcomes in psychological research are naturally expressed as proportions or percentages. Examples include the proportion of correct responses on a test <span class="co">[</span><span class="ot">e.g., @kornell2008</span><span class="co">]</span>, the proportion of time a participant fixates on a particular stimulus in an eye-tracking task <span class="co">[</span><span class="ot">e.g., @james2025</span><span class="co">]</span>, or the proportion of respondents agreeing with a given statement or belief <span class="co">[</span><span class="ot">e.g., @Costello2024</span><span class="co">]</span>. Consider, for instance, a memory experiment in which participants read a short passage, complete a brief distractor task, and then take a final memory test consisting of 10 short‑answer questions. If each question carries a different point value (e.g., question 1 is worth 4 points, question 2 is worth 1 point), a meaningful outcome measure could be the proportion of points earned for each question relative to its maximum possible value.</span>
<span id="cb3-245"><a href="#cb3-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-246"><a href="#cb3-246" aria-hidden="true" tabindex="-1"></a>A key question arises from this example: how should proportional outcomes be analyzed? Researchers frequently default to linear models that assume Gaussian (normal) distributions, such as t-tests, ANOVAs, and linear regression. However, these models make strong assumptions: (1) residuals are normally distributed, (2) the outcome is unbounded (from $-\infty$ to $\infty$), and (3) variance is constant across the range of the data. These assumptions are rarely satisfied in practice <span class="co">[</span><span class="ot">@sladekova2024unicorns</span><span class="co">]</span>, and they are especially ill-suited for proportional outcomes, which are bounded between 0 and 1 and often exhibit heteroscedasticity—non-constant variance, particularly near the boundaries <span class="co">[</span><span class="ot">@smithson2006; @ferrari2004; @paolino2001</span><span class="co">]</span>. Violating these assumptions can lead to biased estimates and spurious inferences, especially when many observations cluster near 0 or 1.</span>
<span id="cb3-247"><a href="#cb3-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-248"><a href="#cb3-248" aria-hidden="true" tabindex="-1"></a>In some cases, it is possible to use a generalized version of the linear model (GLM) that relaxes the assumption of normality. For instance, binomial and Bernoulli models—often referred to as logistic regression when paired with a logit link—are well-suited for binary outcomes (e.g., 0 or 1) or success counts out of a fixed number of trials. However, these models require discretized data and may perform poorly when the data are continuous proportions or exhibit excess variability (i.e., overdispersion), especially near the scale’s boundaries.</span>
<span id="cb3-249"><a href="#cb3-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-250"><a href="#cb3-250" aria-hidden="true" tabindex="-1"></a>The challenges of analyzing proportional data are not new <span class="co">[</span><span class="ot">see @bartlett1936</span><span class="co">]</span>. Fortunately, several existing approaches address the limitations of commonly used models. One such approach is Beta regression, an extension of the generalized linear model that employs the Beta distribution <span class="co">[</span><span class="ot">@ferrari2004; @paolino2001</span><span class="co">]</span>. Beta regression offers a flexible and robust solution for modeling proportional data directly by accounting for boundary effects and over-dispersion, making it a valuable alternative to traditional binomial models. This approach is particularly well-suited for psychological research because it can handle both the bounded nature of proportional data and the non-constant variance often encountered in these datasets <span class="co">[</span><span class="ot">@sladekova2024unicorns</span><span class="co">]</span>. In addition, the direct modeling of proportions allows comparability across tasks and scales, and can be particularly valuable when only the proportional data is available, as is often the case with secondary data that lack item-level structure or point values.</span>
<span id="cb3-251"><a href="#cb3-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-252"><a href="#cb3-252" aria-hidden="true" tabindex="-1"></a>While in this paper we will focus on proportional-responses that lie between 0 and 1--it is important to note that our analysis applies to any bounded continuous scale. Any bounded scale can be mapped to lie within 0 and 1 without resulting in a loss of information as the transformation is linear.<span class="ot">[^1]</span> Consequently, a scale that has natural end points of -1,234 and +8,451--or any other end points on the real number line short of infinity—can be modeled using the approaches we describe in this paper.</span>
<span id="cb3-253"><a href="#cb3-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-254"><a href="#cb3-254" aria-hidden="true" tabindex="-1"></a><span class="ot">[^1]: </span>Specifically, for any continuous bounded variable $x$, we can rescale this variable to lie within 0 and 1 by using the formula $x' = \frac{x - \min(x)}{\max(x) - \min(x)}$ where $0 \leq x' \leq 1$.</span>
<span id="cb3-255"><a href="#cb3-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-256"><a href="#cb3-256" aria-hidden="true" tabindex="-1"></a><span class="fu">## A Beta Way Is Possible</span></span>
<span id="cb3-257"><a href="#cb3-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-258"><a href="#cb3-258" aria-hidden="true" tabindex="-1"></a>With the combination of open-source programming languages like R <span class="co">[</span><span class="ot">@R</span><span class="co">]</span> and their user-developed extensions, analyses such as Beta regression have become increasingly accessible. Yet, adoption of these methods--particularly in psychology--remains limited. One reason may be the lack of informative examples that directly apply to psychological research. Although recent years have seen a surge of interest in Beta regression <span class="co">[</span><span class="ot">@smithson2006; @heiss2021; @vuorre2019; @bendixen2023; @coretta2025bayesian</span><span class="co">]</span>, its adoption in psychology remains limited.</span>
<span id="cb3-259"><a href="#cb3-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-260"><a href="#cb3-260" aria-hidden="true" tabindex="-1"></a>While previous tutorials have discussed Beta regression, most have been limited in scope--focusing either on the basic model or offering only brief mentions of more applicable alternatives. This tutorial aims to fill that gap by offering a comprehensive and practical tutorial of Beta regression and its extensions. In addition to covering the standard Beta model, we walk through its extensions such as zero-inflated, zero-one-inflated, and ordered Beta regression. These models are important for researchers dealing with boundary values (e.g., exact 0s or 1s) or ordinal response structures.</span>
<span id="cb3-261"><a href="#cb3-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-262"><a href="#cb3-262" aria-hidden="true" tabindex="-1"></a>Beyond model specification, we place strong emphasis on interpreting results on the response scale--that is, in terms of probabilities and proportions--rather than relying on often difficult to interpret parameters. This focus makes the models more accessible and meaningful for psychological applications, where effects are often easier to communicate when framed on the original scale of the outcome (e.g., changes in recall accuracy or task performance). Throughout, we provide reproducible code and annotated examples to help readers implement and interpret these models in their own work.</span>
<span id="cb3-263"><a href="#cb3-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-264"><a href="#cb3-264" aria-hidden="true" tabindex="-1"></a>We begin the tutorial with a non-technical overview of the Beta distribution and its core parameters. We then walk through the process of estimating Beta regression models using the R package <span class="in">`brms`</span> <span class="co">[</span><span class="ot">@brms</span><span class="co">]</span>, illustrating each step with applied examples. To guide interpretation, we emphasize coefficients, predicted probabilities, and marginal effects calculated using the <span class="in">`marginaleffects`</span> package <span class="co">[</span><span class="ot">@margeffects2024</span><span class="co">]</span>. We also introduce several useful extensions--zero-inflated (ZIB), zero-one-inflated (ZOIB), and ordered Beta regression--that enable researchers to model outcomes that include boundary values. Finally, all code and materials used in this tutorial are fully reproducible and available via our GitHub repository: <span class="ot">&lt;https://github.com/jgeller112/Beta_regression_tutorial&gt;[^2]</span>.</span>
<span id="cb3-265"><a href="#cb3-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-266"><a href="#cb3-266" aria-hidden="true" tabindex="-1"></a><span class="ot">[^2]: </span>In this article, we try to limit code where possible; however, the online version has all the code needed to reproduce all analyses herein. Furthermore, to promote transparency and reproducibility, the tutorial was written in R version 4.5.1 (@R) using Quarto (v.1.5.54), an open-source publishing system that allows for dynamic and static documents. This allows figures, tables, and text to be programmatically included directly in the manuscript, ensuring that all results are seamlessly integrated into the document. In addition, we use the <span class="in">`rix`</span> <span class="co">[</span><span class="ot">@rix</span><span class="co">]</span> R package which harnesses the power of the <span class="in">`nix`</span> <span class="co">[</span><span class="ot">@nix</span><span class="co">]</span> ecosystem to to help with computational reproducibility. Not only does this give us a snapshot of the packages used to create the current manuscript, but it also takes a snapshot of system dependencies used at run-time. This way reproducers can easily re-use the exact same environment by installing the <span class="in">`nix`</span> package manager and using the included default.nix file to set up the right environment. The README file in the GitHub repository contains detailed information on how to set this up to reproduce the contents of the current manuscript, including a video.</span>
<span id="cb3-267"><a href="#cb3-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-268"><a href="#cb3-268" aria-hidden="true" tabindex="-1"></a><span class="fu">## Beta Distribution</span></span>
<span id="cb3-269"><a href="#cb3-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-270"><a href="#cb3-270" aria-hidden="true" tabindex="-1"></a>Proportional data pose some challenges for standard modeling approaches: The data are bounded between 0 and 1 and often exhibit non-constant variance (heteroscedasticity) <span class="co">[</span><span class="ot">@ferrari2004; @paolino2001</span><span class="co">]</span>. Common distributions used within the generalized linear model frameworks often fail to capture these properties adequately, which can necessitate alternative modeling strategies.</span>
<span id="cb3-271"><a href="#cb3-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-272"><a href="#cb3-272" aria-hidden="true" tabindex="-1"></a>While we do not have time to delve fully into its derivation, the Beta distribution is a preferred distribution for this type of response because of certain unique properties. The Beta distribution is defined as a distribution of the uncertainty of probabilities, which must lie within 0 and 1. As a consequence, the Beta distribution is the maximum entropy distribution for any bounded continuous random variable, which means that the Beta distribution can represent the full range of possibilities of such a scale.<span class="ot">[^3]</span> As a consequence, if we have a continuous scale with upper and lower bounds--and no other special conditions--the Beta distribution will in principle provide a very good approximation of the uncertainty of the scale.</span>
<span id="cb3-273"><a href="#cb3-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-274"><a href="#cb3-274" aria-hidden="true" tabindex="-1"></a><span class="ot">[^3]: </span>Technically, this maximum entropy condition is satisfied because the Beta(1,1) distribution is uniform over its support.</span>
<span id="cb3-275"><a href="#cb3-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-276"><a href="#cb3-276" aria-hidden="true" tabindex="-1"></a>Typically, the expected value (or mean) of the response variable is the central estimand scholars want to estimate. A model should specify how this expected value depends on explanatory variables through two main components: a linear predictor, which combines the explanatory variables in a linear form ($a+b_1x_1+b_2x_2$, etc.), and a link function, which connects the expected value of the response variable to the linear predictor (e.g, $E<span class="co">[</span><span class="ot">Y</span><span class="co">]</span> = g(a + b_1x_1 + b_2x_2)$). In addition, a random component specifies the distribution of the response variable around its expected value (such as Poisson or binomial distributions, which belong to the exponential family) <span class="co">[</span><span class="ot">@nelder1972</span><span class="co">]</span>. Together, these components provide a flexible framework for modeling data with different distributional properties.</span>
<span id="cb3-277"><a href="#cb3-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-278"><a href="#cb3-278" aria-hidden="true" tabindex="-1"></a>The Beta distribution is continuous and restricted to values between 0 and 1 (exclusive). Its two parameters--commonly called shape1 ($\alpha$) and shape2 ($\beta$)--govern the distribution’s location, skewness, and spread. By adjusting these parameters, the distribution can take many functional forms (e.g., it can be symmetric, skewed, U-shaped, or even approximately uniform; see @fig-curves).</span>
<span id="cb3-279"><a href="#cb3-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-280"><a href="#cb3-280" aria-hidden="true" tabindex="-1"></a>To illustrate, consider a test question worth seven points. Suppose a participant scores five out of seven. The number of points received (5) can be treated as $\alpha$, and the number of points missed (2) as $\beta$. The resulting Beta distribution would be skewed toward higher values, reflecting a high performance (yellow line in @fig-curves; "Beta(5, 2)"). Reversing these values would produce a distribution skewed toward lower values, representing poorer performance (green line in @fig-curves; "Beta(2, 5)").</span>
<span id="cb3-281"><a href="#cb3-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-284"><a href="#cb3-284" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-285"><a href="#cb3-285" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-curves</span></span>
<span id="cb3-286"><a href="#cb3-286" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Beta distributions with different shape1 and shape2 parameters."</span></span>
<span id="cb3-287"><a href="#cb3-287" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-asp: 0.52</span></span>
<span id="cb3-288"><a href="#cb3-288" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-289"><a href="#cb3-289" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to generate Beta distribution data</span></span>
<span id="cb3-290"><a href="#cb3-290" aria-hidden="true" tabindex="-1"></a>make_Beta_df <span class="ot">&lt;-</span> <span class="cf">function</span>(shape1, shape2, <span class="at">n =</span> <span class="dv">9000</span>) {</span>
<span id="cb3-291"><a href="#cb3-291" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.001</span>, <span class="fl">0.999</span>, <span class="at">length.out =</span> n)</span>
<span id="cb3-292"><a href="#cb3-292" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb3-293"><a href="#cb3-293" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> x,</span>
<span id="cb3-294"><a href="#cb3-294" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">dbeta</span>(x, shape1, shape2),</span>
<span id="cb3-295"><a href="#cb3-295" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"Beta("</span>, shape1, <span class="st">", "</span>, shape2, <span class="st">")"</span>)</span>
<span id="cb3-296"><a href="#cb3-296" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-297"><a href="#cb3-297" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-298"><a href="#cb3-298" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-299"><a href="#cb3-299" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine distributions</span></span>
<span id="cb3-300"><a href="#cb3-300" aria-hidden="true" tabindex="-1"></a>Beta_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb3-301"><a href="#cb3-301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_Beta_df</span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>),</span>
<span id="cb3-302"><a href="#cb3-302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_Beta_df</span>(<span class="dv">2</span>, <span class="dv">2</span>),</span>
<span id="cb3-303"><a href="#cb3-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_Beta_df</span>(<span class="dv">5</span>, <span class="dv">2</span>),</span>
<span id="cb3-304"><a href="#cb3-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_Beta_df</span>(<span class="dv">2</span>, <span class="dv">5</span>)</span>
<span id="cb3-305"><a href="#cb3-305" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span></span>
<span id="cb3-306"><a href="#cb3-306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">factor</span>(label))</span>
<span id="cb3-307"><a href="#cb3-307" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-308"><a href="#cb3-308" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(</span>
<span id="cb3-309"><a href="#cb3-309" aria-hidden="true" tabindex="-1"></a>  Beta_df,</span>
<span id="cb3-310"><a href="#cb3-310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> label, <span class="at">group =</span> label, <span class="at">color =</span> label)</span>
<span id="cb3-311"><a href="#cb3-311" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb3-312"><a href="#cb3-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb3-313"><a href="#cb3-313" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_okabe_ito</span>(</span>
<span id="cb3-314"><a href="#cb3-314" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"Distribution"</span></span>
<span id="cb3-315"><a href="#cb3-315" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-316"><a href="#cb3-316" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(</span>
<span id="cb3-317"><a href="#cb3-317" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Proportion"</span>,</span>
<span id="cb3-318"><a href="#cb3-318" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="fl">0.01</span>)</span>
<span id="cb3-319"><a href="#cb3-319" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-320"><a href="#cb3-320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(</span>
<span id="cb3-321"><a href="#cb3-321" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Density"</span>,</span>
<span id="cb3-322"><a href="#cb3-322" aria-hidden="true" tabindex="-1"></a>    <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">5</span>),</span>
<span id="cb3-323"><a href="#cb3-323" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="fl">0.01</span>)</span>
<span id="cb3-324"><a href="#cb3-324" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-325"><a href="#cb3-325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb3-326"><a href="#cb3-326" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>)</span>
<span id="cb3-327"><a href="#cb3-327" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-328"><a href="#cb3-328" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-329"><a href="#cb3-329" aria-hidden="true" tabindex="-1"></a><span class="fu">## I Can't Believe It's Not Beta</span></span>
<span id="cb3-330"><a href="#cb3-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-331"><a href="#cb3-331" aria-hidden="true" tabindex="-1"></a>While the standard parameterization of the Beta distribution uses $\alpha$ and $\beta$, a reparameterization to a mean ($\mu$) and precision ($\phi$) is more useful for regression models. The mean represents the expected value of the distribution, while the dispersion, which is inversely related to variance, reflects how concentrated the distribution is around the mean, with higher values indicating a narrower distribution and lower values indicating a wider one. The connections between the Beta distribution's parameters are shown in @eq-beta. Importantly, the variance depends on the average value of the response because uncertainty intervals need to adjust for how close the value of the response is to the boundary.</span>
<span id="cb3-332"><a href="#cb3-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-333"><a href="#cb3-333" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb3-334"><a href="#cb3-334" aria-hidden="true" tabindex="-1"></a>\begin{aligned}<span class="co">[</span><span class="ot">t</span><span class="co">]</span></span>
<span id="cb3-335"><a href="#cb3-335" aria-hidden="true" tabindex="-1"></a>\text{Shape 1:} &amp;&amp; a &amp;= \mu \phi <span class="sc">\\</span></span>
<span id="cb3-336"><a href="#cb3-336" aria-hidden="true" tabindex="-1"></a>\text{Shape 2:} &amp;&amp; b &amp;= (1 - \mu) \phi</span>
<span id="cb3-337"><a href="#cb3-337" aria-hidden="true" tabindex="-1"></a>\end{aligned}</span>
<span id="cb3-338"><a href="#cb3-338" aria-hidden="true" tabindex="-1"></a>\qquad\qquad\qquad</span>
<span id="cb3-339"><a href="#cb3-339" aria-hidden="true" tabindex="-1"></a>\begin{aligned}<span class="co">[</span><span class="ot">t</span><span class="co">]</span></span>
<span id="cb3-340"><a href="#cb3-340" aria-hidden="true" tabindex="-1"></a>\text{Mean:} &amp;&amp; \mu &amp;= \frac{a}{a + b} <span class="sc">\\</span></span>
<span id="cb3-341"><a href="#cb3-341" aria-hidden="true" tabindex="-1"></a>\text{Precision:} &amp;&amp; \phi &amp;= a + b <span class="sc">\\</span></span>
<span id="cb3-342"><a href="#cb3-342" aria-hidden="true" tabindex="-1"></a>\text{Variance:} &amp;&amp; var &amp;= \frac{\mu \cdot (1 - \mu)}{1 + \phi}</span>
<span id="cb3-343"><a href="#cb3-343" aria-hidden="true" tabindex="-1"></a>\end{aligned}</span>
<span id="cb3-344"><a href="#cb3-344" aria-hidden="true" tabindex="-1"></a>$$ {#eq-beta}</span>
<span id="cb3-345"><a href="#cb3-345" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-346"><a href="#cb3-346" aria-hidden="true" tabindex="-1"></a>Thus, Beta regression allows modeling both the mean and precision of the outcome distribution. To ensure that $\mu$ stays between 0 and 1, we apply a link function, which allows linear modeling of the mean on an unbounded scale. A common link-function choice is the logit, but other functions such as the probit or complementary log-log are possible.</span>
<span id="cb3-347"><a href="#cb3-347" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-348"><a href="#cb3-348" aria-hidden="true" tabindex="-1"></a>The logit function, $\text{logit}(\mu) = \log \left( \frac{\mu}{1 - \mu} \right)$ links the mean to log-odds which are unbounded, making linear modeling possible. The inverse of the logit, called the logistic function, maps the linear predictor $\eta$ back to the original scale of the data $\left(\mu = \frac{1}{1 + e^{-\eta}}\right)$. Similarly, the strictly positive dispersion parameter is usually modeled through a log link function, ensuring it remains positive.</span>
<span id="cb3-349"><a href="#cb3-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-350"><a href="#cb3-350" aria-hidden="true" tabindex="-1"></a>By accounting for the observations' natural limits and non-constant variance across different values, the Beta distribution is useful in psychology where outcomes like performance rates or response scales frequently exhibit these features.</span>
<span id="cb3-351"><a href="#cb3-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-352"><a href="#cb3-352" aria-hidden="true" tabindex="-1"></a><span class="fu">## Bayesian Approach to Beta Regression</span></span>
<span id="cb3-353"><a href="#cb3-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-354"><a href="#cb3-354" aria-hidden="true" tabindex="-1"></a>Beta regression models can be estimated with both frequentist and Bayesian methods. We adopt a Bayesian framework because it makes estimating and interpreting more complex models easier <span class="co">[</span><span class="ot">@gelman2013bayesian; @mcelreath2020statistical; @bayes_rules</span><span class="co">]</span>. Generally speaking, most Bayesian analyses can also be implemented with frequentist methods like maximum likelihood, but more complex techniques may require adjustments like bootstrapping. The main limitation of Bayesian modeling is that it is slower than frequentist approaches, but we note that modern Bayesian computation engines are reasonably fast and that explanatory modeling necessarily emphasizes deriving appropriate estimands over computational convenience. We use the R package <span class="in">`brms`</span> <span class="co">[</span><span class="ot">@brms</span><span class="co">]</span>, a high-level interface to the probabilistic programming language Stan <span class="co">[</span><span class="ot">@stan2023</span><span class="co">]</span>, because it uses standard R regression formula syntax but extends its scope while remaining accessible for non-expert users. The package also implements parallel processing that can dramatically shorten computational times for larger datasets.</span>
<span id="cb3-355"><a href="#cb3-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-356"><a href="#cb3-356" aria-hidden="true" tabindex="-1"></a>There are several important differences between our Bayesian analysis and the frequentist methods readers may be more familiar with—most notably, the absence of *t*- and *p*-values. To estimate models, the <span class="in">`brms`</span> package uses Stan’s computational algorithms to draw random samples from the posterior distribution, which represents uncertainty about the model parameters. This posterior is conceptually analogous to a frequentist sampling distribution.</span>
<span id="cb3-357"><a href="#cb3-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-358"><a href="#cb3-358" aria-hidden="true" tabindex="-1"></a>By default, Bayesian models run 4 chains with 2,000 iterations each. The first 1,000 iterations per chain are warmup and are discarded. The remaining 1,000 iterations per chain are retained as posterior draws, yielding 4,000 total post-warmup draws across all chains. From these draws, we can compute the posterior mean (analogous to a frequentist point estimate) and the 95% credible interval (Cr.I.), which is often compared to a confidence interval. In addition, we report the *probability of direction* (pd), which reflects the probability that a parameter is strictly positive or negative. When a uniform prior is used (all values equally likely in the prior), a pd of 95%, 97.5%, 99.5%, and 99.95% corresponds approximately to two-sided p-values of .10, .05, .01, and .001 (i.e., pd ≈ 1 − p/2 for symmetric posteriors with weak/flat priors) (see @fig-p-value-pd-comp for an illustrative comparison). For directional hypotheses, the pd can be interpreted as roughly equivalent to one minus the *p*-value <span class="co">[</span><span class="ot">@marsmanThreeInsightsBayesian2016</span><span class="co">]</span>.</span>
<span id="cb3-359"><a href="#cb3-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-362"><a href="#cb3-362" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-363"><a href="#cb3-363" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-p-value-pd-comp</span></span>
<span id="cb3-364"><a href="#cb3-364" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "A Bayesian posterior distribution (assuming a uniform prior) centered at a point estimate chosen so that the probability of direction (pd) equals 0.95, and a frequentist sampling distribution (under the null; centered at 0). In the Bayesian posterior distribution, the blue area represents the pd, and the red area represents the remaining 1 − pd of the distribution. In the frequentist sampling distribution, the red tail areas represent the rejection region at α = 0.10. In this example, the posterior mean lies exactly at the $1 - \\frac{\\alpha}{2}$ quantile of the null sampling distribution. For symmetric posteriors with flat priors, the pd is numerically equivalent to the one-sided $p$-value."</span></span>
<span id="cb3-365"><a href="#cb3-365" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-asp: 0.8</span></span>
<span id="cb3-366"><a href="#cb3-366" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb3-367"><a href="#cb3-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-368"><a href="#cb3-368" aria-hidden="true" tabindex="-1"></a><span class="co"># --- parameters ---</span></span>
<span id="cb3-369"><a href="#cb3-369" aria-hidden="true" tabindex="-1"></a>pd <span class="ot">&lt;-</span> <span class="fl">0.95</span></span>
<span id="cb3-370"><a href="#cb3-370" aria-hidden="true" tabindex="-1"></a>alpha_two <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> (<span class="dv">1</span> <span class="sc">-</span> pd) <span class="co"># matching two-sided alpha</span></span>
<span id="cb3-371"><a href="#cb3-371" aria-hidden="true" tabindex="-1"></a>mu <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(pd) <span class="co"># shift so P(X &gt; 0) = pd for N(mu, 1)</span></span>
<span id="cb3-372"><a href="#cb3-372" aria-hidden="true" tabindex="-1"></a>lim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="dv">6</span>)</span>
<span id="cb3-373"><a href="#cb3-373" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-374"><a href="#cb3-374" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Posterior-like panel ---</span></span>
<span id="cb3-375"><a href="#cb3-375" aria-hidden="true" tabindex="-1"></a>post <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x =</span> lim), <span class="fu">aes</span>(x)) <span class="sc">+</span></span>
<span id="cb3-376"><a href="#cb3-376" aria-hidden="true" tabindex="-1"></a>  <span class="co"># full curve</span></span>
<span id="cb3-377"><a href="#cb3-377" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb3-378"><a href="#cb3-378" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb3-379"><a href="#cb3-379" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> mu, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb3-380"><a href="#cb3-380" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb3-381"><a href="#cb3-381" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span></span>
<span id="cb3-382"><a href="#cb3-382" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-383"><a href="#cb3-383" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pd region (blue)</span></span>
<span id="cb3-384"><a href="#cb3-384" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb3-385"><a href="#cb3-385" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb3-386"><a href="#cb3-386" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> mu, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb3-387"><a href="#cb3-387" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, lim[<span class="dv">2</span>]),</span>
<span id="cb3-388"><a href="#cb3-388" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"area"</span>,</span>
<span id="cb3-389"><a href="#cb3-389" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"blue"</span>,</span>
<span id="cb3-390"><a href="#cb3-390" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb3-391"><a href="#cb3-391" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-392"><a href="#cb3-392" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remaining region (red)</span></span>
<span id="cb3-393"><a href="#cb3-393" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb3-394"><a href="#cb3-394" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb3-395"><a href="#cb3-395" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> mu, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb3-396"><a href="#cb3-396" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(lim[<span class="dv">1</span>], <span class="dv">0</span>),</span>
<span id="cb3-397"><a href="#cb3-397" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"area"</span>,</span>
<span id="cb3-398"><a href="#cb3-398" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"red"</span>,</span>
<span id="cb3-399"><a href="#cb3-399" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb3-400"><a href="#cb3-400" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-401"><a href="#cb3-401" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linewidth =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb3-402"><a href="#cb3-402" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb3-403"><a href="#cb3-403" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Posterior (illustrative Normal)"</span>,</span>
<span id="cb3-404"><a href="#cb3-404" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"pd = "</span>, <span class="fu">round</span>(pd <span class="sc">*</span> <span class="dv">100</span>), <span class="st">"%"</span>),</span>
<span id="cb3-405"><a href="#cb3-405" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb3-406"><a href="#cb3-406" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">""</span></span>
<span id="cb3-407"><a href="#cb3-407" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-408"><a href="#cb3-408" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> lim, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.45</span>)) <span class="sc">+</span></span>
<span id="cb3-409"><a href="#cb3-409" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>()</span>
<span id="cb3-410"><a href="#cb3-410" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-411"><a href="#cb3-411" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Frequentist panel ---</span></span>
<span id="cb3-412"><a href="#cb3-412" aria-hidden="true" tabindex="-1"></a>z_lo <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(alpha_two <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb3-413"><a href="#cb3-413" aria-hidden="true" tabindex="-1"></a>z_hi <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(<span class="dv">1</span> <span class="sc">-</span> alpha_two <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb3-414"><a href="#cb3-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-415"><a href="#cb3-415" aria-hidden="true" tabindex="-1"></a>freq <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x =</span> lim), <span class="fu">aes</span>(x)) <span class="sc">+</span></span>
<span id="cb3-416"><a href="#cb3-416" aria-hidden="true" tabindex="-1"></a>  <span class="co"># full curve</span></span>
<span id="cb3-417"><a href="#cb3-417" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb3-418"><a href="#cb3-418" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb3-419"><a href="#cb3-419" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb3-420"><a href="#cb3-420" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb3-421"><a href="#cb3-421" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span></span>
<span id="cb3-422"><a href="#cb3-422" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-423"><a href="#cb3-423" aria-hidden="true" tabindex="-1"></a>  <span class="co"># lower rejection tail (red)</span></span>
<span id="cb3-424"><a href="#cb3-424" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb3-425"><a href="#cb3-425" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb3-426"><a href="#cb3-426" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb3-427"><a href="#cb3-427" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(lim[<span class="dv">1</span>], z_lo),</span>
<span id="cb3-428"><a href="#cb3-428" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"area"</span>,</span>
<span id="cb3-429"><a href="#cb3-429" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"red"</span>,</span>
<span id="cb3-430"><a href="#cb3-430" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb3-431"><a href="#cb3-431" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-432"><a href="#cb3-432" aria-hidden="true" tabindex="-1"></a>  <span class="co"># upper rejection tail (red)</span></span>
<span id="cb3-433"><a href="#cb3-433" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb3-434"><a href="#cb3-434" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb3-435"><a href="#cb3-435" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb3-436"><a href="#cb3-436" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(z_hi, lim[<span class="dv">2</span>]),</span>
<span id="cb3-437"><a href="#cb3-437" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"area"</span>,</span>
<span id="cb3-438"><a href="#cb3-438" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"red"</span>,</span>
<span id="cb3-439"><a href="#cb3-439" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb3-440"><a href="#cb3-440" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-441"><a href="#cb3-441" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(z_lo, z_hi), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linewidth =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb3-442"><a href="#cb3-442" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linewidth =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb3-443"><a href="#cb3-443" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb3-444"><a href="#cb3-444" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Frequentist sampling under H0 (N(0,1))"</span>,</span>
<span id="cb3-445"><a href="#cb3-445" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"two-sided α = "</span>, <span class="fu">signif</span>(alpha_two, <span class="dv">2</span>)),</span>
<span id="cb3-446"><a href="#cb3-446" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb3-447"><a href="#cb3-447" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">""</span></span>
<span id="cb3-448"><a href="#cb3-448" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-449"><a href="#cb3-449" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> lim, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.45</span>)) <span class="sc">+</span></span>
<span id="cb3-450"><a href="#cb3-450" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>()</span>
<span id="cb3-451"><a href="#cb3-451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-452"><a href="#cb3-452" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Stack the plots ---</span></span>
<span id="cb3-453"><a href="#cb3-453" aria-hidden="true" tabindex="-1"></a>post <span class="sc">/</span> freq</span>
<span id="cb3-454"><a href="#cb3-454" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-455"><a href="#cb3-455" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-456"><a href="#cb3-456" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-457"><a href="#cb3-457" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-458"><a href="#cb3-458" aria-hidden="true" tabindex="-1"></a>For reasons of space, we refer readers unfamiliar to Bayesian data analysis to several existing books on the topic <span class="co">[</span><span class="ot">@gelman2013bayesian; @mcelreath2020statistical; @kruschke2015doing</span><span class="co">]</span>. In addition, we assume readers are familiar with R, but those in need of a refresher should find @wickhamDataScienceImport2023 useful.</span>
<span id="cb3-459"><a href="#cb3-459" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-460"><a href="#cb3-460" aria-hidden="true" tabindex="-1"></a><span class="fu"># Beta Regression Tutorial</span></span>
<span id="cb3-461"><a href="#cb3-461" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-462"><a href="#cb3-462" aria-hidden="true" tabindex="-1"></a><span class="fu">## Example Data</span></span>
<span id="cb3-463"><a href="#cb3-463" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-464"><a href="#cb3-464" aria-hidden="true" tabindex="-1"></a>Throughout this tutorial, we analyze data from a memory experiment examining whether the fluency of an instructor’s delivery affects recall performance <span class="co">[</span><span class="ot">@wilford2020, Experiment 1A</span><span class="co">]</span>. Instructor fluency—marked by expressive gestures, dynamic vocal tone, and confident pacing—has been shown to influence students’ perceptions of learning, often leading learners to rate fluent instructors more favorably <span class="co">[</span><span class="ot">@carpenter2013</span><span class="co">]</span>. However, previous research suggests that these impressions do not reliably translate into improved memory performance <span class="co">[</span><span class="ot">e.g., @carpenter2013; @toftness2017; @witherby2022</span><span class="co">]</span>. In contrast, @wilford2020 found that participants actually recalled more information after watching a fluent instructor compared to a disfluent one. This surprising finding makes the dataset a compelling case study for analyzing proportion data, as recall was scored out of 10 possible idea units per video.</span>
<span id="cb3-465"><a href="#cb3-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-466"><a href="#cb3-466" aria-hidden="true" tabindex="-1"></a>In Experiment 1A, participants watched two short instructional videos, each delivered either fluently or disfluently. Fluent videos featured instructors with smooth delivery and natural pacing, while disfluent videos included hesitations, monotone speech, and awkward pauses. After a distractor task, participants completed a free recall test, writing down as much content as they could remember from each video within a three-minute window. Their recall was then scored for the number of idea units correctly remembered.</span>
<span id="cb3-467"><a href="#cb3-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-470"><a href="#cb3-470" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-471"><a href="#cb3-471" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-data</span></span>
<span id="cb3-472"><a href="#cb3-472" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Data needed to run examples</span></span>
<span id="cb3-473"><a href="#cb3-473" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb3-474"><a href="#cb3-474" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-475"><a href="#cb3-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-476"><a href="#cb3-476" aria-hidden="true" tabindex="-1"></a><span class="co"># get data here from github</span></span>
<span id="cb3-477"><a href="#cb3-477" aria-hidden="true" tabindex="-1"></a>url <span class="ot">&lt;-</span> <span class="fu">str_glue</span>(</span>
<span id="cb3-478"><a href="#cb3-478" aria-hidden="true" tabindex="-1"></a>  <span class="st">"https://raw.githubusercontent.com/jgeller112/"</span>,</span>
<span id="cb3-479"><a href="#cb3-479" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Beta_regression_tutorial/refs/heads/main/"</span>,</span>
<span id="cb3-480"><a href="#cb3-480" aria-hidden="true" tabindex="-1"></a>  <span class="st">"manuscript/data/fluency_data.csv"</span></span>
<span id="cb3-481"><a href="#cb3-481" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-482"><a href="#cb3-482" aria-hidden="true" tabindex="-1"></a>fluency_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(url)</span>
<span id="cb3-483"><a href="#cb3-483" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-484"><a href="#cb3-484" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-485"><a href="#cb3-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-486"><a href="#cb3-486" aria-hidden="true" tabindex="-1"></a>Our primary outcome variable is the proportion of idea units recalled on the final test, calculated by dividing the number of correct units by 10. We show a sample of these data in @tbl-dataset. The dataset can be downloaded from Github (@lst-data). Because this is a bounded continuous variable (i.e., it ranges from 0 to 1), it violates the assumptions of typical linear regression models that treat outcomes as normally distributed. Despite this, it remains common in psychological research to analyze proportion data using models that assume normality. In what follows, we reproduce @wilford2020's analysis and then re-analyze the data using Beta regression and highlight how it can improve our inferences.</span>
<span id="cb3-487"><a href="#cb3-487" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-490"><a href="#cb3-490" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-491"><a href="#cb3-491" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-dataset</span></span>
<span id="cb3-492"><a href="#cb3-492" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Four observations from @wilford2020. Accuracy refers to the proportion of correctly recalled idea units."</span></span>
<span id="cb3-493"><a href="#cb3-493" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-494"><a href="#cb3-494" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-495"><a href="#cb3-495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-496"><a href="#cb3-496" aria-hidden="true" tabindex="-1"></a>fluency_data <span class="sc">|&gt;</span></span>
<span id="cb3-497"><a href="#cb3-497" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">4</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-498"><a href="#cb3-498" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-499"><a href="#cb3-499" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-500"><a href="#cb3-500" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-501"><a href="#cb3-501" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-502"><a href="#cb3-502" aria-hidden="true" tabindex="-1"></a><span class="fu">## Reanalysis of Wilford et al. Experiment 1A</span></span>
<span id="cb3-503"><a href="#cb3-503" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-504"><a href="#cb3-504" aria-hidden="true" tabindex="-1"></a>In their original analysis of Experiment 1A, @wilford2020 compared memory performance between fluent and disfluent instructor conditions using a traditional independent-samples t-test. They found that participants who watched the fluent instructor recalled significantly more idea units than those who viewed the disfluent version (see @fig-flu1).</span>
<span id="cb3-505"><a href="#cb3-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-508"><a href="#cb3-508" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-509"><a href="#cb3-509" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-flu1</span></span>
<span id="cb3-510"><a href="#cb3-510" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Raincloud plot depicting accuracy distributions for the Fluent and Disfluent conditions. Each condition shows individual data points, a density plot, and summary statistics to illustrate variability and central tendency."</span></span>
<span id="cb3-511"><a href="#cb3-511" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-512"><a href="#cb3-512" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(fluency_data, <span class="fu">aes</span>(<span class="at">x =</span> Fluency, <span class="at">y =</span> Accuracy, <span class="at">fill =</span> Fluency)) <span class="sc">+</span></span>
<span id="cb3-513"><a href="#cb3-513" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_rain</span>(</span>
<span id="cb3-514"><a href="#cb3-514" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">color =</span> Fluency),</span>
<span id="cb3-515"><a href="#cb3-515" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> <span class="dv">21</span>,</span>
<span id="cb3-516"><a href="#cb3-516" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> .<span class="dv">6</span>,</span>
<span id="cb3-517"><a href="#cb3-517" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.55</span>,</span>
<span id="cb3-518"><a href="#cb3-518" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="fl">0.1</span></span>
<span id="cb3-519"><a href="#cb3-519" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-520"><a href="#cb3-520" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Add black mean dot, nudged to center</span></span>
<span id="cb3-521"><a href="#cb3-521" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_summary</span>(</span>
<span id="cb3-522"><a href="#cb3-522" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> mean,</span>
<span id="cb3-523"><a href="#cb3-523" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"point"</span>,</span>
<span id="cb3-524"><a href="#cb3-524" aria-hidden="true" tabindex="-1"></a>    <span class="at">shape =</span> <span class="dv">21</span>,</span>
<span id="cb3-525"><a href="#cb3-525" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">3</span>,</span>
<span id="cb3-526"><a href="#cb3-526" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"black"</span>,</span>
<span id="cb3-527"><a href="#cb3-527" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb3-528"><a href="#cb3-528" aria-hidden="true" tabindex="-1"></a>    <span class="at">position =</span> <span class="fu">position_nudge</span>(<span class="at">x =</span> .<span class="dv">1</span>) <span class="co"># adjust as needed</span></span>
<span id="cb3-529"><a href="#cb3-529" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-530"><a href="#cb3-530" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">labels =</span> <span class="fu">label_percent</span>()) <span class="sc">+</span></span>
<span id="cb3-531"><a href="#cb3-531" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_okabe_ito</span>(<span class="at">aesthetics =</span> <span class="fu">c</span>(<span class="st">"fill"</span>, <span class="st">"color"</span>)) <span class="sc">+</span></span>
<span id="cb3-532"><a href="#cb3-532" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb3-533"><a href="#cb3-533" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb3-534"><a href="#cb3-534" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb3-535"><a href="#cb3-535" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>()</span>
<span id="cb3-536"><a href="#cb3-536" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-537"><a href="#cb3-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-538"><a href="#cb3-538" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-539"><a href="#cb3-539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-540"><a href="#cb3-540" aria-hidden="true" tabindex="-1"></a>We first replicate this analysis in a regression framework using brms. We model final test accuracy--the proportion of correctly recalled idea units across the videos--as the dependent variable. Our predictor is instructor fluency, with two levels: Fluent and Disfluent. We use treatment (dummy) coding, which is the default in R. This coding scheme sets the first level of a factor (in alphabetical order) as the reference level. In this case, Disfluent is the reference, and the coefficient for Fluent reflects the contrast between fluent and disfluent instructor conditions.</span>
<span id="cb3-541"><a href="#cb3-541" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-544"><a href="#cb3-544" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-545"><a href="#cb3-545" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-all</span></span>
<span id="cb3-546"><a href="#cb3-546" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Bayesian regression summaries for each model"</span></span>
<span id="cb3-547"><a href="#cb3-547" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-align: "center"</span></span>
<span id="cb3-548"><a href="#cb3-548" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-549"><a href="#cb3-549" aria-hidden="true" tabindex="-1"></a><span class="co">#| apa-note: "Link functions: b_mean = logit; b_phi = log; b_zoi (zero-one inflation) = logit; b_coi (conditional one-inflation) = logit. Asterisks reflect approximate two-sided p-values derived from the posterior pd. pd ≥ 0.975 (p ≤ .05) = `*`; pd ≥ 0.990 (p ≤ .01) = `**`; pd ≥ 0.998 (p ≤ .001) = `***`."</span></span>
<span id="cb3-550"><a href="#cb3-550" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-551"><a href="#cb3-551" aria-hidden="true" tabindex="-1"></a><span class="co"># Define your models</span></span>
<span id="cb3-552"><a href="#cb3-552" aria-hidden="true" tabindex="-1"></a>model_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb3-553"><a href="#cb3-553" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Bayesian LM"</span> <span class="ot">=</span> model1,</span>
<span id="cb3-554"><a href="#cb3-554" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Beta Regression"</span> <span class="ot">=</span> model2,</span>
<span id="cb3-555"><a href="#cb3-555" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ZIB"</span> <span class="ot">=</span> model3,</span>
<span id="cb3-556"><a href="#cb3-556" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ZOIB"</span> <span class="ot">=</span> model4,</span>
<span id="cb3-557"><a href="#cb3-557" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Ordered Beta"</span> <span class="ot">=</span> model5</span>
<span id="cb3-558"><a href="#cb3-558" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-559"><a href="#cb3-559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-560"><a href="#cb3-560" aria-hidden="true" tabindex="-1"></a><span class="co"># Process each model</span></span>
<span id="cb3-561"><a href="#cb3-561" aria-hidden="true" tabindex="-1"></a>cleaned_tables <span class="ot">&lt;-</span> <span class="fu">imap</span>(model_list, <span class="cf">function</span>(mod, name) {</span>
<span id="cb3-562"><a href="#cb3-562" aria-hidden="true" tabindex="-1"></a>  <span class="fu">model_parameters</span>(mod, <span class="at">centrality =</span> <span class="st">"mean"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-563"><a href="#cb3-563" aria-hidden="true" tabindex="-1"></a>    <span class="fu">filter</span>(Parameter <span class="sc">!=</span> <span class="st">"sigma"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-564"><a href="#cb3-564" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb3-565"><a href="#cb3-565" aria-hidden="true" tabindex="-1"></a>      <span class="at">Mean =</span> <span class="fu">sprintf</span>(<span class="st">"%.3f"</span>, Mean),</span>
<span id="cb3-566"><a href="#cb3-566" aria-hidden="true" tabindex="-1"></a>      <span class="at">Cr.I =</span> <span class="fu">str_glue</span>(<span class="st">"[{round(CI_low, 3)}, {round(CI_high, 3)}]"</span>),</span>
<span id="cb3-567"><a href="#cb3-567" aria-hidden="true" tabindex="-1"></a>      <span class="at">pd_star =</span> <span class="fu">case_when</span>(</span>
<span id="cb3-568"><a href="#cb3-568" aria-hidden="true" tabindex="-1"></a>        pd <span class="sc">&gt;=</span> <span class="fl">0.995</span> <span class="sc">~</span> <span class="fu">str_glue</span>(<span class="st">"{sprintf('%.3f', pd)}***"</span>),</span>
<span id="cb3-569"><a href="#cb3-569" aria-hidden="true" tabindex="-1"></a>        pd <span class="sc">&gt;=</span> <span class="fl">0.985</span> <span class="sc">~</span> <span class="fu">str_glue</span>(<span class="st">"{sprintf('%.3f', pd)}**"</span>),</span>
<span id="cb3-570"><a href="#cb3-570" aria-hidden="true" tabindex="-1"></a>        pd <span class="sc">&gt;=</span> <span class="fl">0.975</span> <span class="sc">~</span> <span class="fu">str_glue</span>(<span class="st">"{sprintf('%.3f', pd)}*"</span>),</span>
<span id="cb3-571"><a href="#cb3-571" aria-hidden="true" tabindex="-1"></a>        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="fu">sprintf</span>(<span class="st">"%.3f"</span>, pd)</span>
<span id="cb3-572"><a href="#cb3-572" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb3-573"><a href="#cb3-573" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span></span>
<span id="cb3-574"><a href="#cb3-574" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(</span>
<span id="cb3-575"><a href="#cb3-575" aria-hidden="true" tabindex="-1"></a>      <span class="at">cols =</span> <span class="fu">c</span>(Mean, Cr.I, pd_star),</span>
<span id="cb3-576"><a href="#cb3-576" aria-hidden="true" tabindex="-1"></a>      <span class="at">names_to =</span> <span class="st">"Stat"</span>,</span>
<span id="cb3-577"><a href="#cb3-577" aria-hidden="true" tabindex="-1"></a>      <span class="at">values_to =</span> <span class="st">"value"</span></span>
<span id="cb3-578"><a href="#cb3-578" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span></span>
<span id="cb3-579"><a href="#cb3-579" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb3-580"><a href="#cb3-580" aria-hidden="true" tabindex="-1"></a>      <span class="at">Stat =</span> <span class="fu">recode</span>(Stat, <span class="at">pd_star =</span> <span class="st">"pd"</span>),</span>
<span id="cb3-581"><a href="#cb3-581" aria-hidden="true" tabindex="-1"></a>      <span class="at">Model =</span> name</span>
<span id="cb3-582"><a href="#cb3-582" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span></span>
<span id="cb3-583"><a href="#cb3-583" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(Parameter, Stat, Model, value)</span>
<span id="cb3-584"><a href="#cb3-584" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb3-585"><a href="#cb3-585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-586"><a href="#cb3-586" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine all tables</span></span>
<span id="cb3-587"><a href="#cb3-587" aria-hidden="true" tabindex="-1"></a>pretty_table <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cleaned_tables) <span class="sc">|&gt;</span></span>
<span id="cb3-588"><a href="#cb3-588" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Model, <span class="at">values_from =</span> value) <span class="sc">|&gt;</span></span>
<span id="cb3-589"><a href="#cb3-589" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.character), <span class="sc">~</span> <span class="fu">replace_na</span>(.x, <span class="st">"-"</span>))) <span class="sc">|&gt;</span></span>
<span id="cb3-590"><a href="#cb3-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-591"><a href="#cb3-591" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Simplify parameter names</span></span>
<span id="cb3-592"><a href="#cb3-592" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Parameter =</span> <span class="fu">str_replace_all</span>(Parameter, <span class="st">"FluencyFluent"</span>, <span class="st">"Fluency"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb3-593"><a href="#cb3-593" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-594"><a href="#cb3-594" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set factor levels</span></span>
<span id="cb3-595"><a href="#cb3-595" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-596"><a href="#cb3-596" aria-hidden="true" tabindex="-1"></a>    <span class="at">Stat =</span> <span class="fu">factor</span>(Stat, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Mean"</span>, <span class="st">"Cr.I"</span>, <span class="st">"pd"</span>)),</span>
<span id="cb3-597"><a href="#cb3-597" aria-hidden="true" tabindex="-1"></a>    <span class="at">Parameter =</span> <span class="fu">factor</span>(</span>
<span id="cb3-598"><a href="#cb3-598" aria-hidden="true" tabindex="-1"></a>      Parameter,</span>
<span id="cb3-599"><a href="#cb3-599" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb3-600"><a href="#cb3-600" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_Intercept"</span>,</span>
<span id="cb3-601"><a href="#cb3-601" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_Fluency"</span>,</span>
<span id="cb3-602"><a href="#cb3-602" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_phi_Intercept"</span>,</span>
<span id="cb3-603"><a href="#cb3-603" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_phi_Fluency"</span>,</span>
<span id="cb3-604"><a href="#cb3-604" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_zi_Intercept"</span>,</span>
<span id="cb3-605"><a href="#cb3-605" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_zi_Fluency"</span>,</span>
<span id="cb3-606"><a href="#cb3-606" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_zoi_Intercept"</span>,</span>
<span id="cb3-607"><a href="#cb3-607" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_zoi_Fluency"</span>,</span>
<span id="cb3-608"><a href="#cb3-608" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_coi_Intercept"</span>,</span>
<span id="cb3-609"><a href="#cb3-609" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_coi_Fluency"</span></span>
<span id="cb3-610"><a href="#cb3-610" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb3-611"><a href="#cb3-611" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb3-612"><a href="#cb3-612" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-613"><a href="#cb3-613" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-614"><a href="#cb3-614" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Parameter, Stat) <span class="sc">|&gt;</span></span>
<span id="cb3-615"><a href="#cb3-615" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Parameter) <span class="sc">|&gt;</span></span>
<span id="cb3-616"><a href="#cb3-616" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Parameter =</span> <span class="fu">if_else</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>, <span class="fu">as.character</span>(Parameter), <span class="st">""</span>)) <span class="sc">|&gt;</span></span>
<span id="cb3-617"><a href="#cb3-617" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb3-618"><a href="#cb3-618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-619"><a href="#cb3-619" aria-hidden="true" tabindex="-1"></a><span class="co"># Display as styled table</span></span>
<span id="cb3-620"><a href="#cb3-620" aria-hidden="true" tabindex="-1"></a>pretty_table <span class="sc">|&gt;</span></span>
<span id="cb3-621"><a href="#cb3-621" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>(<span class="at">width =</span> <span class="fl">0.95</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-622"><a href="#cb3-622" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-623"><a href="#cb3-623" aria-hidden="true" tabindex="-1"></a>  <span class="fu">style_tt</span>(<span class="at">fontsize =</span> .<span class="dv">9</span>, <span class="at">align =</span> <span class="st">"c"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-624"><a href="#cb3-624" aria-hidden="true" tabindex="-1"></a>  <span class="fu">style_tt</span>(</span>
<span id="cb3-625"><a href="#cb3-625" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> <span class="fu">c</span>(<span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>, <span class="dv">5</span><span class="sc">:</span><span class="dv">6</span>, <span class="dv">8</span><span class="sc">:</span><span class="dv">9</span>, <span class="dv">11</span><span class="sc">:</span><span class="dv">12</span>, <span class="dv">14</span><span class="sc">:</span><span class="dv">15</span>, <span class="dv">17</span><span class="sc">:</span><span class="dv">18</span>, <span class="dv">20</span><span class="sc">:</span><span class="dv">21</span>, <span class="dv">23</span><span class="sc">:</span><span class="dv">24</span>, <span class="dv">26</span><span class="sc">:</span><span class="dv">27</span>, <span class="dv">29</span><span class="sc">:</span><span class="dv">30</span>),</span>
<span id="cb3-626"><a href="#cb3-626" aria-hidden="true" tabindex="-1"></a>    <span class="at">italic =</span> <span class="cn">TRUE</span>,</span>
<span id="cb3-627"><a href="#cb3-627" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontsize =</span> .<span class="dv">6</span></span>
<span id="cb3-628"><a href="#cb3-628" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-629"><a href="#cb3-629" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-630"><a href="#cb3-630" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-631"><a href="#cb3-631" aria-hidden="true" tabindex="-1"></a><span class="fu">### Regression Model</span></span>
<span id="cb3-632"><a href="#cb3-632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-633"><a href="#cb3-633" aria-hidden="true" tabindex="-1"></a>We first start by loading the <span class="in">`brms`</span> <span class="co">[</span><span class="ot">@brms</span><span class="co">]</span> and <span class="in">`cmdstanr`</span> <span class="co">[</span><span class="ot">@cmdstanr</span><span class="co">]</span> packages (@lst-loadbrms). We use the <span class="in">`cmdstanr`</span> backend for Stan <span class="co">[</span><span class="ot">@stan2023</span><span class="co">]</span> because it's faster than the default used to run models (i.e., <span class="in">`rstan`</span>),<span class="ot">[^4]</span> though all of these models can also be fit with <span class="in">`brms`</span> defaults.</span>
<span id="cb3-634"><a href="#cb3-634" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-635"><a href="#cb3-635" aria-hidden="true" tabindex="-1"></a><span class="ot">[^4]: </span>In order to use the <span class="in">`cmdstanr`</span> backend you will need to first install the package (see <span class="in">`https://mc-stan.org/cmdstanr/`</span>) and also run <span class="in">`cmdstanr::install_cmdstan()`</span> if you have not done so already.</span>
<span id="cb3-636"><a href="#cb3-636" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-639"><a href="#cb3-639" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-640"><a href="#cb3-640" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-loadbrms</span></span>
<span id="cb3-641"><a href="#cb3-641" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Load the `brms` and `cmdstanr` packages</span></span>
<span id="cb3-642"><a href="#cb3-642" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-643"><a href="#cb3-643" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb3-644"><a href="#cb3-644" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-645"><a href="#cb3-645" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="cb3-646"><a href="#cb3-646" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cmdstanr)</span>
<span id="cb3-647"><a href="#cb3-647" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-648"><a href="#cb3-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-651"><a href="#cb3-651" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-652"><a href="#cb3-652" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fit-gauss</span></span>
<span id="cb3-653"><a href="#cb3-653" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-gauss</span></span>
<span id="cb3-654"><a href="#cb3-654" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting a gaussian model with brm().</span></span>
<span id="cb3-655"><a href="#cb3-655" aria-hidden="true" tabindex="-1"></a><span class="co">#| hide: false</span></span>
<span id="cb3-656"><a href="#cb3-656" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-657"><a href="#cb3-657" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-658"><a href="#cb3-658" aria-hidden="true" tabindex="-1"></a>bayes_reg_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb3-659"><a href="#cb3-659" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency,</span>
<span id="cb3-660"><a href="#cb3-660" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb3-661"><a href="#cb3-661" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">gaussian</span>(),</span>
<span id="cb3-662"><a href="#cb3-662" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"model_reg_bayes"</span></span>
<span id="cb3-663"><a href="#cb3-663" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-664"><a href="#cb3-664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-665"><a href="#cb3-665" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-666"><a href="#cb3-666" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-669"><a href="#cb3-669" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-670"><a href="#cb3-670" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fit-gauss-parameters</span></span>
<span id="cb3-671"><a href="#cb3-671" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-672"><a href="#cb3-672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-673"><a href="#cb3-673" aria-hidden="true" tabindex="-1"></a>brms_model_new <span class="ot">&lt;-</span> <span class="fu">model_parameters</span>(bayes_reg_model, <span class="at">centrality =</span> <span class="st">"mean"</span>)</span>
<span id="cb3-674"><a href="#cb3-674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-675"><a href="#cb3-675" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-676"><a href="#cb3-676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-677"><a href="#cb3-677" aria-hidden="true" tabindex="-1"></a>We fit the model using the <span class="in">`brm()`</span> function from the <span class="in">`brms`</span> package (@lst-gauss). Although not shown here, we ran the models using four chains (the default), executed in parallel across four cores. When the model is run in @lst-gauss, the model summary output will appear in the R console. The output from <span class="in">`bayes_reg_model`</span> shows each parameter's posterior summary: The posterior distribution's mean and standard deviation (analogous to the frequentist standard error) and its 95% credible interval, which indicate the 95% of the most credible parameter values. In <span class="in">`brms`</span>, the reported Cr.I is an equal-tailed interval, meaning that the probability mass excluded from the interval is split equally between the lower and upper tails. Additionally, the output indicates numerical estimates of the sampling algorithm's performance: Rhat should be close to one, and the ESS (effective sample size) metrics should be as large as possible given the number of iterations specified (default is 4000). Generally, ESS <span class="sc">\&gt;</span>= 1000 is recommended <span class="co">[</span><span class="ot">@brms</span><span class="co">]</span>. For the models we present in this paper, convergence is trivial with standard linear models, though we note that these metrics are still important to pay attention to in case of model mis-fit.</span>
<span id="cb3-678"><a href="#cb3-678" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-679"><a href="#cb3-679" aria-hidden="true" tabindex="-1"></a>Our main question of interest is: Does instructor fluency have an effect on final test performance? In order to answer this question, we will have to look at the output summary produced by @lst-gauss. the <span class="in">`Intercept`</span> refers to the posterior mean accuracy in the disfluent condition, M = <span class="in">`r round(brms_model_new$Mean[1], 3)`</span> , as fluency was dummy-coded. The fluency coefficient (<span class="in">`FluencyFluent`</span>) reflects the mean posterior difference in recall accuracy between the fluent and disfluent conditions: *b* = <span class="in">`r round(brms_model_new$Mean[2], 3)`</span>. The 95% credible interval for this estimate spans from <span class="in">`r round(brms_model_new$CI_low[2], 3)`</span> to <span class="in">`r round(brms_model_new$CI_high[2], 3)`</span>. These values are shown in the "95% Cr.I" columns of the output. These results closely mirror the findings reported by @wilford2020 (Experiment 1A).</span>
<span id="cb3-680"><a href="#cb3-680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-683"><a href="#cb3-683" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-684"><a href="#cb3-684" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: show reg model</span></span>
<span id="cb3-685"><a href="#cb3-685" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(bayes_reg_model)</span>
<span id="cb3-686"><a href="#cb3-686" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-687"><a href="#cb3-687" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-688"><a href="#cb3-688" aria-hidden="true" tabindex="-1"></a>The output also includes the effective sample size (ESS) and R̂ (R-hat) values, both of which fall within acceptable ranges, indicating good model convergence. Throughout the tutorial, we focus on posterior mean estimates and their 95% credible intervals. In addition, we include the *pd* measure in the main summary table (@tbl-all), which is provided by the <span class="in">`bayestestR`</span> package <span class="co">[</span><span class="ot">@makowski2019a; @makowski2019</span><span class="co">]</span>. This measure offers an intuitive parallel to p-values that many readers may be familiar with.</span>
<span id="cb3-689"><a href="#cb3-689" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-692"><a href="#cb3-692" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-693"><a href="#cb3-693" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-hypo</span></span>
<span id="cb3-694"><a href="#cb3-694" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Getting BF using `hypothesis()` function from `brms`</span></span>
<span id="cb3-695"><a href="#cb3-695" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-696"><a href="#cb3-696" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-697"><a href="#cb3-697" aria-hidden="true" tabindex="-1"></a><span class="co"># test non-zero effect</span></span>
<span id="cb3-698"><a href="#cb3-698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-699"><a href="#cb3-699" aria-hidden="true" tabindex="-1"></a>fluency_effect <span class="ot">&lt;-</span> <span class="fu">hypothesis</span>(bayes_reg_model, <span class="st">"FluencyFluent &gt; 0"</span>)</span>
<span id="cb3-700"><a href="#cb3-700" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-701"><a href="#cb3-701" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-704"><a href="#cb3-704" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-705"><a href="#cb3-705" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-hypo</span></span>
<span id="cb3-706"><a href="#cb3-706" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Evidence ratio for the difference between fluent and disfluent conditions</span></span>
<span id="cb3-707"><a href="#cb3-707" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-708"><a href="#cb3-708" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-709"><a href="#cb3-709" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-710"><a href="#cb3-710" aria-hidden="true" tabindex="-1"></a><span class="co"># get hypothesis table</span></span>
<span id="cb3-711"><a href="#cb3-711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-712"><a href="#cb3-712" aria-hidden="true" tabindex="-1"></a>fluency_effect<span class="sc">$</span>hypothesis <span class="sc">|&gt;</span></span>
<span id="cb3-713"><a href="#cb3-713" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-714"><a href="#cb3-714" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb3-715"><a href="#cb3-715" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{CI.Lower}, {CI.Upper}]"</span>)</span>
<span id="cb3-716"><a href="#cb3-716" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-717"><a href="#cb3-717" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"Estimate"</span>, <span class="st">"SE"</span> <span class="ot">=</span> <span class="st">"Est.Error"</span>, <span class="st">"ER"</span> <span class="ot">=</span> <span class="st">"Evid.Ratio"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-718"><a href="#cb3-718" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Hypothesis, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, ER) <span class="sc">|&gt;</span></span>
<span id="cb3-719"><a href="#cb3-719" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-720"><a href="#cb3-720" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-721"><a href="#cb3-721" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-722"><a href="#cb3-722" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-723"><a href="#cb3-723" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-724"><a href="#cb3-724" aria-hidden="true" tabindex="-1"></a>For example, the fluency effect has a *pd* of .977, which indicates a high probability that the effect is positive rather than negative (akin to p \&lt; .05). However, *pd* does not tell us the probability that the effect is *non-zero*. To assess this, we can compare a model where the effect is greater than 0 to an alternative model where the effect is less than 0. The <span class="in">`hypothesis()`</span> function in <span class="in">`brms`</span> (see @lst-hypo) allows us to perform this comparison. Running the code returns a column called <span class="in">`ER`</span> (<span class="in">`Evid.Ratio`</span>), which contains a numerical value that represents the Bayes factor for a directional hypothesis (@tbl-hypo). The evidence ratio (ER) for the fluency effect is <span class="in">`r fluency_effect$hypothesis$Evid.Ratio`</span>, providing very strong evidence that the effect is both non-zero and positive <span class="co">[</span><span class="ot">@jeffries1961; @kass1995</span><span class="co">]</span>.</span>
<span id="cb3-725"><a href="#cb3-725" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-726"><a href="#cb3-726" aria-hidden="true" tabindex="-1"></a><span class="fu">## Beta Regression</span></span>
<span id="cb3-727"><a href="#cb3-727" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-728"><a href="#cb3-728" aria-hidden="true" tabindex="-1"></a>@wilford2020 observed that instructor fluency impacts actual learning, using a t-test. But recall this approach assumes normality of residuals and homoscedacity. These assumptions are unrealistic when the response values approach the scale boundaries <span class="co">[</span><span class="ot">@sladekova2024unicorns</span><span class="co">]</span>. Does the data we have meet those assumptions? We can use the function <span class="in">`check_model()`</span> from <span class="in">`easystats`</span> <span class="co">[</span><span class="ot">@easystats</span><span class="co">]</span> to check our assumptions easily. The code in @lst-check1 automatically produces @fig-reg-assump. We can see some issues with our data. Specifically, there appears to be violations of constant variance across the values of the scale (homoskedasticity). In plain terms, this type of model mis-specfication means that a standard OLS model can predict non-sensical values outside the bounds of the scale.</span>
<span id="cb3-729"><a href="#cb3-729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-732"><a href="#cb3-732" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-733"><a href="#cb3-733" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-reg-assump</span></span>
<span id="cb3-734"><a href="#cb3-734" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-check1</span></span>
<span id="cb3-735"><a href="#cb3-735" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Checking assumptions with the `check_model()` from `easystats` package .</span></span>
<span id="cb3-736"><a href="#cb3-736" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Two assumption checks for our OLS model: Normality (left) and Homoskedasticity (right)"</span></span>
<span id="cb3-737"><a href="#cb3-737" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-738"><a href="#cb3-738" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-739"><a href="#cb3-739" aria-hidden="true" tabindex="-1"></a><span class="fu">check_model</span>(bayes_reg_model, <span class="at">check =</span> <span class="fu">c</span>(<span class="st">"homogeneity"</span>, <span class="st">"normality"</span>))</span>
<span id="cb3-740"><a href="#cb3-740" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-741"><a href="#cb3-741" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-742"><a href="#cb3-742" aria-hidden="true" tabindex="-1"></a>We can also examine how well the data fits the model by performing a posterior predictive check using the <span class="in">`pp_check()`</span> function from <span class="in">`brms`</span>. A posterior predictive check involves looking at multiple draws or repetitions from the posterior distribution and plotting it against the observed data. Ideally, the predictive draws (the light blue lines) should show reasonable resemblance with the observed data (dark blue line). In our example (see @fig-ppcheckall (A)) the model-predicted density is slightly too peaked and narrow compared to the data. In addition, some of the draws extend into negative accuracy values.</span>
<span id="cb3-743"><a href="#cb3-743" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-746"><a href="#cb3-746" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-747"><a href="#cb3-747" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ppcheckall</span></span>
<span id="cb3-748"><a href="#cb3-748" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: The plots show 100 posterior predicted distributions with the label yrep (light blue), the distribution of accuracy in dark blue for regular regression (A) , Beta regression (B), and ZIB (C) models</span></span>
<span id="cb3-749"><a href="#cb3-749" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb3-750"><a href="#cb3-750" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb3-751"><a href="#cb3-751" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-752"><a href="#cb3-752" aria-hidden="true" tabindex="-1"></a><span class="co">#pp_check</span></span>
<span id="cb3-753"><a href="#cb3-753" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-754"><a href="#cb3-754" aria-hidden="true" tabindex="-1"></a>pp_check1 <span class="ot">&lt;-</span> <span class="fu">pp_check</span>(model1, <span class="at">ndraws =</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb3-755"><a href="#cb3-755" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>()</span>
<span id="cb3-756"><a href="#cb3-756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-757"><a href="#cb3-757" aria-hidden="true" tabindex="-1"></a>pp_check2 <span class="ot">&lt;-</span> <span class="fu">pp_check</span>(model2, <span class="at">ndraws =</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb3-758"><a href="#cb3-758" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb3-759"><a href="#cb3-759" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb3-760"><a href="#cb3-760" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-761"><a href="#cb3-761" aria-hidden="true" tabindex="-1"></a>pp_check3 <span class="ot">&lt;-</span> <span class="fu">pp_check</span>(model3, <span class="at">ndraws =</span> <span class="dv">100</span>) <span class="sc">+</span></span>
<span id="cb3-762"><a href="#cb3-762" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb3-763"><a href="#cb3-763" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">1</span>)</span>
<span id="cb3-764"><a href="#cb3-764" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-765"><a href="#cb3-765" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-766"><a href="#cb3-766" aria-hidden="true" tabindex="-1"></a><span class="co"># Use `plot_spacer()` for the empty 4th spot</span></span>
<span id="cb3-767"><a href="#cb3-767" aria-hidden="true" tabindex="-1"></a>combined_plot <span class="ot">&lt;-</span> (pp_check1 <span class="sc">+</span> pp_check2 <span class="sc">+</span> pp_check3) <span class="sc">+</span></span>
<span id="cb3-768"><a href="#cb3-768" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(</span>
<span id="cb3-769"><a href="#cb3-769" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag_levels =</span> <span class="st">'A'</span>,</span>
<span id="cb3-770"><a href="#cb3-770" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag_prefix =</span> <span class="st">"("</span>,</span>
<span id="cb3-771"><a href="#cb3-771" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag_suffix =</span> <span class="st">")"</span></span>
<span id="cb3-772"><a href="#cb3-772" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-773"><a href="#cb3-773" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-774"><a href="#cb3-774" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-775"><a href="#cb3-775" aria-hidden="true" tabindex="-1"></a>combined_plot</span>
<span id="cb3-776"><a href="#cb3-776" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-777"><a href="#cb3-777" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-778"><a href="#cb3-778" aria-hidden="true" tabindex="-1"></a>Given the outcome variable is proportional, one solution would be to run a Beta regression model. Again, we can create the Beta regression model in <span class="in">`brms`</span>. In <span class="in">`brms`</span>, we model each parameter independently. Recall from the introduction that in a Beta model we model two parameters--$\mu$ and $\phi$. We can easily do this by using the <span class="in">`bf()`</span> function from <span class="in">`brms`</span> (@lst-Beta01). <span class="in">`bf()`</span> facilitates the specification of several sub-models within the same formula call. We fit two formulas, one for $\mu$ and one for $\phi$ and store it in the <span class="in">`model_Beta_bayes`</span> object below. In the below <span class="in">`bf()`</span> call, we are modeling Fluency as a function of Accuracy only for the $\mu$ parameter. For the $\phi$ parameter, we are only modeling the intercept value. This is saying dispersion does not change as a function of fluency.</span>
<span id="cb3-779"><a href="#cb3-779" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-780"><a href="#cb3-780" aria-hidden="true" tabindex="-1"></a>To run our Beta regression model, we need to exclude 0s and 1s in our data set. If we try to run a model with our data <span class="in">`data_fluency`</span> we get an error: <span class="in">`Error: Family 'Beta' requires response greater than 0`</span>. This is because the Beta distribution only supports observations in the 0 to 1 interval *excluding exact 0s and 1s*. We need make sure there are no 0s and 1s in our dataset.</span>
<span id="cb3-781"><a href="#cb3-781" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-782"><a href="#cb3-782" aria-hidden="true" tabindex="-1"></a>The dataset contains nine 0s and one 1. One approach is to nudge our 0s towards .01 and our 1s to .99, or apply a special formula <span class="co">[</span><span class="ot">@smithson2006</span><span class="co">]</span> so values fall within the <span class="sc">\[</span>0, 1<span class="sc">\]</span> interval. We implore readers not to engage in this practice. @kubinec2022 showed that this practice can result in serious distortion of the outcome as the sample size 𝑁 grows larger, resulting in ever smaller values that are “nudged”. Because the Beta distribution is a non-linear model of the outcome, values that are very close to the boundary, such as 0.00001 or 0.99999, will be highly influential outliers.To run this Beta model we will remove the 0s and 1s, and later in this article we will show how to jointly model these scale end points with the rest of the data. The model from @lst-Beta01 uses a transformed <span class="in">`data_fluency`</span> object (called <span class="in">`data_Beta`</span>) where 0s and 1s are removed. When we run this code we should not get an error.</span>
<span id="cb3-783"><a href="#cb3-783" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-786"><a href="#cb3-786" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-787"><a href="#cb3-787" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: Beta_brms_01s</span></span>
<span id="cb3-788"><a href="#cb3-788" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-Beta01</span></span>
<span id="cb3-789"><a href="#cb3-789" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting a Beta model without 0s and 1s in brm().</span></span>
<span id="cb3-790"><a href="#cb3-790" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-791"><a href="#cb3-791" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-792"><a href="#cb3-792" aria-hidden="true" tabindex="-1"></a><span class="co"># set up model formual</span></span>
<span id="cb3-793"><a href="#cb3-793" aria-hidden="true" tabindex="-1"></a>model_Beta_bayes <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb3-794"><a href="#cb3-794" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># fit mu model</span></span>
<span id="cb3-795"><a href="#cb3-795" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> <span class="dv">1</span> <span class="co"># fit phi model</span></span>
<span id="cb3-796"><a href="#cb3-796" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-797"><a href="#cb3-797" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-798"><a href="#cb3-798" aria-hidden="true" tabindex="-1"></a><span class="co"># transform 0 to 0.1 and 1 to .99</span></span>
<span id="cb3-799"><a href="#cb3-799" aria-hidden="true" tabindex="-1"></a>data_Beta <span class="ot">&lt;-</span> fluency_data <span class="sc">|&gt;</span></span>
<span id="cb3-800"><a href="#cb3-800" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb3-801"><a href="#cb3-801" aria-hidden="true" tabindex="-1"></a>    Accuracy <span class="sc">!=</span> <span class="dv">0</span>,</span>
<span id="cb3-802"><a href="#cb3-802" aria-hidden="true" tabindex="-1"></a>    Accuracy <span class="sc">!=</span> <span class="dv">1</span></span>
<span id="cb3-803"><a href="#cb3-803" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-804"><a href="#cb3-804" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-805"><a href="#cb3-805" aria-hidden="true" tabindex="-1"></a>Beta_brms <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb3-806"><a href="#cb3-806" aria-hidden="true" tabindex="-1"></a>  model_Beta_bayes,</span>
<span id="cb3-807"><a href="#cb3-807" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_Beta,</span>
<span id="cb3-808"><a href="#cb3-808" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">Beta</span>(),</span>
<span id="cb3-809"><a href="#cb3-809" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"model_Beta_bayes_reg_01"</span></span>
<span id="cb3-810"><a href="#cb3-810" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-811"><a href="#cb3-811" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-812"><a href="#cb3-812" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-813"><a href="#cb3-813" aria-hidden="true" tabindex="-1"></a><span class="fu">### Model Parameters</span></span>
<span id="cb3-814"><a href="#cb3-814" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-815"><a href="#cb3-815" aria-hidden="true" tabindex="-1"></a>In @tbl-all under the Beta Regression column, the first set of coefficients represent how factors influence the $\mu$ parameter estimates (which is the mean of the Beta distribution), which are labeled with an underscore <span class="in">`b_`</span>. These coefficients are interpreted on the scale of the logit, meaning they represent linear changes on a nonlinear space. The intercept term <span class="in">`(b_Intercept)`</span> represents the log odds of the mean on accuracy for the fluent instructor. Log odds that are negative indicate that it is more likely a "success" (like getting the correct answer) will NOT happen than that it will happen. Similarly, regression coefficients in log odds forms that are negative indicate that an increase in that predictor leads to a decrease in the predicted probability of a "success".</span>
<span id="cb3-816"><a href="#cb3-816" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-817"><a href="#cb3-817" aria-hidden="true" tabindex="-1"></a>The other component we need to pay attention to is the dispersion or precision parameter coefficients labeled as <span class="in">`b_phi`</span> in @tbl-all. The dispersion ($\phi$) parameter tells us how precise our estimate is. Specifically, $\phi$ in Beta regression tells us about the variability of the response variable around its mean. Specifically, a higher dispersion parameter indicates a narrower distribution, reflecting less variability. Conversely, a lower dispersion parameter suggests a wider distribution, reflecting greater variability. The main difference between a dispersion parameter and the variance is that the dispersion has a different interpretation depending on the value of the outcome, as we show below. The best way to understand dispersion is to examine visual changes in the distribution as the dispersion increases or decreases.</span>
<span id="cb3-818"><a href="#cb3-818" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-819"><a href="#cb3-819" aria-hidden="true" tabindex="-1"></a>Understanding the dispersion parameter helps us gauge the precision of our predictions and the consistency of the response variable. In <span class="in">`Beta_brms`</span> we only modeled the dispersion of the intercept. When $\phi$ is not specified, the intercept is modeled by default (see @tbl-all). The intercept under the precision heading is not that interesting. It represents the overall dispersion in the outcome across all conditions. Instead, we can model different dispersions across levels of the <span class="in">`Fluency`</span> factor. To do so, we add <span class="in">`Fluency`</span> to the phi model in <span class="in">`bf()`</span>. We model the precision (<span class="in">`phi`</span>) of the <span class="in">`Fluency`</span> factor by using a <span class="in">`~`</span> and adding factors of interest to the right of it (@lst-Betadis).</span>
<span id="cb3-820"><a href="#cb3-820" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-823"><a href="#cb3-823" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-824"><a href="#cb3-824" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: brms-model-dis</span></span>
<span id="cb3-825"><a href="#cb3-825" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-Betadis</span></span>
<span id="cb3-826"><a href="#cb3-826" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting Beta model with dispersion in `brm()`.</span></span>
<span id="cb3-827"><a href="#cb3-827" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb3-828"><a href="#cb3-828" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-829"><a href="#cb3-829" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-830"><a href="#cb3-830" aria-hidden="true" tabindex="-1"></a>model_Beta_bayes_disp <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb3-831"><a href="#cb3-831" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># Model of the mean</span></span>
<span id="cb3-832"><a href="#cb3-832" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency <span class="co"># Model of the precision</span></span>
<span id="cb3-833"><a href="#cb3-833" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-834"><a href="#cb3-834" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-835"><a href="#cb3-835" aria-hidden="true" tabindex="-1"></a>Beta_brms_dis <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb3-836"><a href="#cb3-836" aria-hidden="true" tabindex="-1"></a>  model_Beta_bayes_disp,</span>
<span id="cb3-837"><a href="#cb3-837" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_Beta,</span>
<span id="cb3-838"><a href="#cb3-838" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">Beta</span>(),</span>
<span id="cb3-839"><a href="#cb3-839" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"model_Beta_bayes_dis_run01"</span></span>
<span id="cb3-840"><a href="#cb3-840" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-841"><a href="#cb3-841" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-842"><a href="#cb3-842" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-843"><a href="#cb3-843" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-844"><a href="#cb3-844" aria-hidden="true" tabindex="-1"></a>@tbl-all displays the model summary with the precision parameter added to our model as a function of fluency. It is important to note that the estimates are logged and not on the original scale (this is only the case when additional parameters are modeled). To interpret them on the original scale, we can exponentiate the log-transformed value--this transformation gets us back to our original scale. In the below model call, we set <span class="in">`exponentiate`</span> = TRUE.</span>
<span id="cb3-845"><a href="#cb3-845" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-848"><a href="#cb3-848" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-849"><a href="#cb3-849" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: phi-Beta</span></span>
<span id="cb3-850"><a href="#cb3-850" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-851"><a href="#cb3-851" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-852"><a href="#cb3-852" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-853"><a href="#cb3-853" aria-hidden="true" tabindex="-1"></a>Beta_model_dis_exp <span class="ot">&lt;-</span> Beta_brms_dis <span class="sc">|&gt;</span></span>
<span id="cb3-854"><a href="#cb3-854" aria-hidden="true" tabindex="-1"></a>  <span class="fu">model_parameters</span>(<span class="at">exponentiate =</span> <span class="cn">TRUE</span>, <span class="at">centrality =</span> <span class="st">"mean"</span>)</span>
<span id="cb3-855"><a href="#cb3-855" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-856"><a href="#cb3-856" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-859"><a href="#cb3-859" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-860"><a href="#cb3-860" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-phi-Beta-exp</span></span>
<span id="cb3-861"><a href="#cb3-861" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-862"><a href="#cb3-862" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Beta regression model summary for fluency factor with $\phi$ parameter exponentiated</span></span>
<span id="cb3-863"><a href="#cb3-863" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-864"><a href="#cb3-864" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-865"><a href="#cb3-865" aria-hidden="true" tabindex="-1"></a>Beta_model_dis_exp <span class="sc">|&gt;</span></span>
<span id="cb3-866"><a href="#cb3-866" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="fu">startsWith</span>(Parameter, <span class="st">"b_phi"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb3-867"><a href="#cb3-867" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-868"><a href="#cb3-868" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb3-869"><a href="#cb3-869" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{CI_low}, {CI_high}]"</span>)</span>
<span id="cb3-870"><a href="#cb3-870" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-871"><a href="#cb3-871" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Parameter, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, pd) <span class="sc">|&gt;</span></span>
<span id="cb3-872"><a href="#cb3-872" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-873"><a href="#cb3-873" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-874"><a href="#cb3-874" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-875"><a href="#cb3-875" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-876"><a href="#cb3-876" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-877"><a href="#cb3-877" aria-hidden="true" tabindex="-1"></a>The $\phi$ intercept represents the precision of the fluent condition. The $\phi$ coefficient for <span class="in">`FluencyFluent`</span> represents the change in that precision for performance between the fluent vs. disfluent conditions. The credible interval does not include 0, meaning that zero is not among the 95% most credible parameter values.</span>
<span id="cb3-878"><a href="#cb3-878" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-879"><a href="#cb3-879" aria-hidden="true" tabindex="-1"></a>It is important to note that these estimates are not the same as the marginal effects we discussed earlier. Changes in dispersion affect the spread or variability of the response distribution without necessarily altering its mean. This makes dispersion particularly relevant for research questions that focus on features of the distribution beyond the average—such as how concentrated responses are. For instance, high dispersion might indicate that individuals cluster at the extremes (e.g., very high or very low ratings), suggesting clustering in the outcome.</span>
<span id="cb3-880"><a href="#cb3-880" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-881"><a href="#cb3-881" aria-hidden="true" tabindex="-1"></a>A critical assumption of the GLM is homoscedasticity, which means constant variance of the errors. WIth Beta regression model we can include a dispersion parameter for Fluency. Properly accounting for dispersion is crucial because it impacts the precision of our mean estimates and, consequently, the significance of our coefficients. The inclusion of dispersion in the our model increased the uncertainty of the $\mu$ coefficient (see @fig-plt-diff). This suggests that failing to account for the dispersion of the variables might lead to biased estimates. This highlights the potential utility of an approach like Beta regression over a traditional approach as Beta regression can explicitly model dispersion and address issues of heteroscedasticity.</span>
<span id="cb3-882"><a href="#cb3-882" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-883"><a href="#cb3-883" aria-hidden="true" tabindex="-1"></a>It is only necessary to model the dispersion with covariates when there is reason to believe that this variation is substantively relevant to the research question. In case there is uncertainty about the best model, a relatively agnostic approach would be to compare models, for example with leave one out (loo) cross validation, to examine if a dispersion parameter should be considered in our model.<span class="ot">[^5]</span></span>
<span id="cb3-884"><a href="#cb3-884" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-885"><a href="#cb3-885" aria-hidden="true" tabindex="-1"></a><span class="ot">[^5]: </span>The model fit statistic LOO-CV can be compared for any set of fitted <span class="in">`brms`</span> models with the function <span class="in">`loo()`</span>.</span>
<span id="cb3-886"><a href="#cb3-886" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-889"><a href="#cb3-889" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-890"><a href="#cb3-890" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-plt-diff</span></span>
<span id="cb3-891"><a href="#cb3-891" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Comparison of posterior distributions for the risk difference in fluency: Simple model  (no dispersion for Fluency) vs. complex model with  dispersion"</span></span>
<span id="cb3-892"><a href="#cb3-892" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-893"><a href="#cb3-893" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-894"><a href="#cb3-894" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a model identifier to each dataset</span></span>
<span id="cb3-895"><a href="#cb3-895" aria-hidden="true" tabindex="-1"></a>pred_no <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(Beta_brms, <span class="at">variables =</span> <span class="st">"Fluency"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-896"><a href="#cb3-896" aria-hidden="true" tabindex="-1"></a>  <span class="fu">posterior_draws</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-897"><a href="#cb3-897" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">model =</span> <span class="st">"No Dispersion"</span>)</span>
<span id="cb3-898"><a href="#cb3-898" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-899"><a href="#cb3-899" aria-hidden="true" tabindex="-1"></a>pred_yes <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(Beta_brms_dis, <span class="at">variables =</span> <span class="st">"Fluency"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-900"><a href="#cb3-900" aria-hidden="true" tabindex="-1"></a>  <span class="fu">posterior_draws</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-901"><a href="#cb3-901" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">model =</span> <span class="st">"Dispersion"</span>)</span>
<span id="cb3-902"><a href="#cb3-902" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-903"><a href="#cb3-903" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine both datasets</span></span>
<span id="cb3-904"><a href="#cb3-904" aria-hidden="true" tabindex="-1"></a>pred_combined <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(pred_no, pred_yes)</span>
<span id="cb3-905"><a href="#cb3-905" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-906"><a href="#cb3-906" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pred_combined, <span class="fu">aes</span>(<span class="at">x =</span> draw, <span class="at">y =</span> model, <span class="at">fill =</span> model)) <span class="sc">+</span></span>
<span id="cb3-907"><a href="#cb3-907" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_halfeye</span>(<span class="at">.width =</span> <span class="fu">c</span>(.<span class="dv">8</span>, .<span class="dv">95</span>), <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb3-908"><a href="#cb3-908" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_okabe_ito</span>() <span class="sc">+</span></span>
<span id="cb3-909"><a href="#cb3-909" aria-hidden="true" tabindex="-1"></a>  <span class="co"># guides(fill = guide_legend(title = "Model")) +  # Remove or comment out</span></span>
<span id="cb3-910"><a href="#cb3-910" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb3-911"><a href="#cb3-911" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb3-912"><a href="#cb3-912" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Difference in Final Recall Performance"</span>,</span>
<span id="cb3-913"><a href="#cb3-913" aria-hidden="true" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">"80% and 95% credible intervals shown in black"</span></span>
<span id="cb3-914"><a href="#cb3-914" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-915"><a href="#cb3-915" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">"solid"</span>, <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb3-916"><a href="#cb3-916" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb3-917"><a href="#cb3-917" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb3-918"><a href="#cb3-918" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-919"><a href="#cb3-919" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-922"><a href="#cb3-922" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-923"><a href="#cb3-923" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: Beta-brms</span></span>
<span id="cb3-924"><a href="#cb3-924" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-925"><a href="#cb3-925" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb3-926"><a href="#cb3-926" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-927"><a href="#cb3-927" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-928"><a href="#cb3-928" aria-hidden="true" tabindex="-1"></a><span class="fu">model_parameters</span>(Beta_brms, <span class="st">"mean"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-929"><a href="#cb3-929" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-930"><a href="#cb3-930" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb3-931"><a href="#cb3-931" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{CI_low}, {CI_high}]"</span>)</span>
<span id="cb3-932"><a href="#cb3-932" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-933"><a href="#cb3-933" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Parameter, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, pd) <span class="sc">|&gt;</span></span>
<span id="cb3-934"><a href="#cb3-934" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-935"><a href="#cb3-935" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-936"><a href="#cb3-936" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-937"><a href="#cb3-937" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-938"><a href="#cb3-938" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-941"><a href="#cb3-941" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-942"><a href="#cb3-942" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: Beta-mu</span></span>
<span id="cb3-943"><a href="#cb3-943" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb3-944"><a href="#cb3-944" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-945"><a href="#cb3-945" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-946"><a href="#cb3-946" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-947"><a href="#cb3-947" aria-hidden="true" tabindex="-1"></a><span class="fu">model_parameters</span>(Beta_brms, <span class="st">"mean"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-948"><a href="#cb3-948" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">startsWith</span>(Parameter, <span class="st">"b_phi"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb3-949"><a href="#cb3-949" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-950"><a href="#cb3-950" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb3-951"><a href="#cb3-951" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{CI_low}, {CI_high}]"</span>)</span>
<span id="cb3-952"><a href="#cb3-952" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-953"><a href="#cb3-953" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Parameter, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, pd) <span class="sc">|&gt;</span></span>
<span id="cb3-954"><a href="#cb3-954" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-955"><a href="#cb3-955" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-956"><a href="#cb3-956" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-957"><a href="#cb3-957" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-958"><a href="#cb3-958" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-959"><a href="#cb3-959" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-960"><a href="#cb3-960" aria-hidden="true" tabindex="-1"></a><span class="fu">### Predicted Probabilities</span></span>
<span id="cb3-961"><a href="#cb3-961" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-962"><a href="#cb3-962" aria-hidden="true" tabindex="-1"></a>Parameter estimates are usually difficult to interpret on their own. We argue that researchers should not spend too much time interpreting raw coefficients from non-linear models. We report them in this tutorial for completeness. Instead researchers should discuss the effects of the predictor on the actual outcome of interest (in this case the 0-1 scale). The logit link allows us to transform back and forth between the scale of a linear model and the nonlinear scale of the outcome, which is bounded by 0 and 1. By using the inverse of the logit, we can easily transform our linear coefficients to obtain average effects on the scale of the proportions or percentages, which is usually what is interesting to applied researchers. In a simple case, we can do this manually, but when there are many factors in your model this can be quite complex.</span>
<span id="cb3-963"><a href="#cb3-963" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-964"><a href="#cb3-964" aria-hidden="true" tabindex="-1"></a>In our example, we can use the <span class="in">`plogis()`</span> function in base R to convert estimates from the log-odds (logit) scale to the probability scale. The intercept of our model is -0.918, which reflects the log-odds of the mean accuracy in the disfluent condition. If the estimated difference between the fluent and disfluent conditions is 0.24 on the log-odds scale, we first add this value to the intercept value (-0.918) to get the log-odds for the fluent condition: <span class="in">`-0.83 + 0.20 = -0.63`</span>. We then use <span class="in">`plogis()`</span> to convert both log-odds values to probabilities (Fluent = 35%, Disfluent = 30%).</span>
<span id="cb3-965"><a href="#cb3-965" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-966"><a href="#cb3-966" aria-hidden="true" tabindex="-1"></a>This is pretty easy to do manually, but when your model has many predictors, it can be quite cumbersome. To help us extract predictions from our model and visualize them we will use a package called **`marginaleffects`** <span class="co">[</span><span class="ot">@margeffects2024</span><span class="co">]</span> (see @lst-me). To get the proportions for each of our categorical predictors on the $\mu$ parameter we can use the function from the package called <span class="in">`predictions()`</span>. These are displayed in @tbl-predict-prob. These probabilities match what we calculated above.</span>
<span id="cb3-967"><a href="#cb3-967" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-970"><a href="#cb3-970" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-971"><a href="#cb3-971" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-972"><a href="#cb3-972" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-me</span></span>
<span id="cb3-973"><a href="#cb3-973" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Load the `marginaleffects` package.</span></span>
<span id="cb3-974"><a href="#cb3-974" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-975"><a href="#cb3-975" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(marginaleffects)</span>
<span id="cb3-976"><a href="#cb3-976" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-977"><a href="#cb3-977" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-980"><a href="#cb3-980" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-981"><a href="#cb3-981" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-pred</span></span>
<span id="cb3-982"><a href="#cb3-982" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Predictions from the Beta model for each level of Fluency.</span></span>
<span id="cb3-983"><a href="#cb3-983" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb3-984"><a href="#cb3-984" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-985"><a href="#cb3-985" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-986"><a href="#cb3-986" aria-hidden="true" tabindex="-1"></a><span class="fu">predictions</span>(</span>
<span id="cb3-987"><a href="#cb3-987" aria-hidden="true" tabindex="-1"></a>  Beta_brms,</span>
<span id="cb3-988"><a href="#cb3-988" aria-hidden="true" tabindex="-1"></a>  <span class="co"># need to specify the levels of the categorical predictor</span></span>
<span id="cb3-989"><a href="#cb3-989" aria-hidden="true" tabindex="-1"></a>  <span class="at">newdata =</span> <span class="fu">datagrid</span>(<span class="at">Fluency =</span> <span class="fu">c</span>(<span class="st">"Disfluent"</span>, <span class="st">"Fluent"</span>))</span>
<span id="cb3-990"><a href="#cb3-990" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-991"><a href="#cb3-991" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-992"><a href="#cb3-992" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-995"><a href="#cb3-995" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-996"><a href="#cb3-996" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-predict-prob</span></span>
<span id="cb3-997"><a href="#cb3-997" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Predicted probabilitiesfor fluency factor.</span></span>
<span id="cb3-998"><a href="#cb3-998" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap-location: top</span></span>
<span id="cb3-999"><a href="#cb3-999" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-1000"><a href="#cb3-1000" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-1001"><a href="#cb3-1001" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1002"><a href="#cb3-1002" aria-hidden="true" tabindex="-1"></a><span class="fu">predictions</span>(</span>
<span id="cb3-1003"><a href="#cb3-1003" aria-hidden="true" tabindex="-1"></a>  Beta_brms,</span>
<span id="cb3-1004"><a href="#cb3-1004" aria-hidden="true" tabindex="-1"></a>  <span class="at">newdata =</span> <span class="fu">datagrid</span>(<span class="at">Fluency =</span> <span class="fu">c</span>(<span class="st">"Disfluent"</span>, <span class="st">"Fluent"</span>))</span>
<span id="cb3-1005"><a href="#cb3-1005" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb3-1006"><a href="#cb3-1006" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Fluency =</span> <span class="fu">ifelse</span>(Fluency <span class="sc">==</span> <span class="st">"Fluent"</span>, <span class="st">"Fluent"</span>, <span class="st">"Disfluent"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb3-1007"><a href="#cb3-1007" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-1008"><a href="#cb3-1008" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb3-1009"><a href="#cb3-1009" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{conf.low}, {conf.high}]"</span>)</span>
<span id="cb3-1010"><a href="#cb3-1010" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-1011"><a href="#cb3-1011" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"estimate"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1012"><a href="#cb3-1012" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Fluency, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1013"><a href="#cb3-1013" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(Fluency) <span class="sc">|&gt;</span></span>
<span id="cb3-1014"><a href="#cb3-1014" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1015"><a href="#cb3-1015" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-1016"><a href="#cb3-1016" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1017"><a href="#cb3-1017" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1018"><a href="#cb3-1018" aria-hidden="true" tabindex="-1"></a>For the <span class="in">`Fluency`</span> factor, we can interpret <span class="in">`Mean`</span> as proportions or percentages. That is, participants who watched the fluent instructor scored on average 35% on the final exam compared to 30% for those who watched the disfluent instructor. We can also visualize these from <span class="in">`marginaleffects`</span> using the <span class="in">`plot_predictions()`</span> function (see @lst-plotBeta).</span>
<span id="cb3-1019"><a href="#cb3-1019" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1022"><a href="#cb3-1022" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1023"><a href="#cb3-1023" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot Beta</span></span>
<span id="cb3-1024"><a href="#cb3-1024" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-plotBeta</span></span>
<span id="cb3-1025"><a href="#cb3-1025" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Plot predicted probabilities using `plot_predictions()` from `marginaleffects`</span></span>
<span id="cb3-1026"><a href="#cb3-1026" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-1027"><a href="#cb3-1027" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1028"><a href="#cb3-1028" aria-hidden="true" tabindex="-1"></a>Beta_plot <span class="ot">&lt;-</span> <span class="fu">plot_predictions</span>(Beta_brms, <span class="at">by =</span> <span class="st">"Fluency"</span>)</span>
<span id="cb3-1029"><a href="#cb3-1029" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1030"><a href="#cb3-1030" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1031"><a href="#cb3-1031" aria-hidden="true" tabindex="-1"></a>The <span class="in">`plot_predictions()`</span> function will only display the point estimate with the 95% credible intervals. However, Bayesian estimation methods generate distributions for each parameter. This approach allows visualizing full uncertainty estimates beyond points and intervals. Using the <span class="in">`marginaleffects`</span> package, we can obtain samples from the posterior distribution with the <span class="in">`posterior_draws()`</span> function (see @lst-posdraws). We can then plot these results to illustrate the range of plausible values for our estimates at different levels of uncertainty (see @fig-draws-fluency).</span>
<span id="cb3-1032"><a href="#cb3-1032" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1035"><a href="#cb3-1035" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1036"><a href="#cb3-1036" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-posdraws</span></span>
<span id="cb3-1037"><a href="#cb3-1037" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Extracting posterior draws from the Beta regression model.</span></span>
<span id="cb3-1038"><a href="#cb3-1038" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-1039"><a href="#cb3-1039" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1040"><a href="#cb3-1040" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a model identifier to each dataset</span></span>
<span id="cb3-1041"><a href="#cb3-1041" aria-hidden="true" tabindex="-1"></a>pred_draws_Beta <span class="ot">&lt;-</span> <span class="fu">avg_predictions</span>(Beta_brms, <span class="at">variables =</span> <span class="st">"Fluency"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1042"><a href="#cb3-1042" aria-hidden="true" tabindex="-1"></a>  <span class="fu">posterior_draws</span>()</span>
<span id="cb3-1043"><a href="#cb3-1043" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1044"><a href="#cb3-1044" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1047"><a href="#cb3-1047" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1048"><a href="#cb3-1048" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-draws-fluency</span></span>
<span id="cb3-1049"><a href="#cb3-1049" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Predicted probablity posterior distributions by fluency</span></span>
<span id="cb3-1050"><a href="#cb3-1050" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot both models</span></span>
<span id="cb3-1051"><a href="#cb3-1051" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pred_draws_Beta, <span class="fu">aes</span>(<span class="at">x =</span> draw, <span class="at">y =</span> Fluency, <span class="at">fill =</span> Fluency)) <span class="sc">+</span></span>
<span id="cb3-1052"><a href="#cb3-1052" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_halfeye</span>(<span class="at">.width =</span> <span class="fu">c</span>(.<span class="dv">5</span>, .<span class="dv">8</span>, .<span class="dv">95</span>), <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb3-1053"><a href="#cb3-1053" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_okabe_ito</span>() <span class="sc">+</span></span>
<span id="cb3-1054"><a href="#cb3-1054" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">label_percent</span>()) <span class="sc">+</span></span>
<span id="cb3-1055"><a href="#cb3-1055" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb3-1056"><a href="#cb3-1056" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Accuracy"</span>,</span>
<span id="cb3-1057"><a href="#cb3-1057" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb3-1058"><a href="#cb3-1058" aria-hidden="true" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">"50%, 80% and 95% credible intervals shown in black"</span></span>
<span id="cb3-1059"><a href="#cb3-1059" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-1060"><a href="#cb3-1060" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">"solid"</span>, <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb3-1061"><a href="#cb3-1061" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb3-1062"><a href="#cb3-1062" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb3-1063"><a href="#cb3-1063" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1064"><a href="#cb3-1064" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1065"><a href="#cb3-1065" aria-hidden="true" tabindex="-1"></a><span class="fu">### Marginal Effects</span></span>
<span id="cb3-1066"><a href="#cb3-1066" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1067"><a href="#cb3-1067" aria-hidden="true" tabindex="-1"></a>Marginal effects provide a way to understand how changes in a predictor influence an outcome, holding all other factors constant in a specific manner. Technically, marginal effects are calculated using partial derivatives for continuous variables or finite differences for categorical and continuous variables, depending on the nature of the data and the research question. Substantively, these procedures translate raw regression coefficients back into effects that represent changes in the bounded outcome, such as an $x$% change in the value of a proportion.</span>
<span id="cb3-1068"><a href="#cb3-1068" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1069"><a href="#cb3-1069" aria-hidden="true" tabindex="-1"></a>There are various types of marginal effects, and their calculation can vary across software packages. For example, the popular <span class="in">`emmeans`</span> package <span class="co">[</span><span class="ot">@emmeans</span><span class="co">]</span> computes marginal effects by holding all predictors at their means. In this tutorial, we will use the <span class="in">`marginaleffects`</span> package <span class="co">[</span><span class="ot">@margeffects2024</span><span class="co">]</span>, which focuses on average marginal effects (AMEs) by default. AMEs summarize effects by generating predictions for each row of the original dataset and then averaging these predictions. This approach retains a strong connection to the original data while offering a straightforward summary of the effect of interest.</span>
<span id="cb3-1070"><a href="#cb3-1070" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1071"><a href="#cb3-1071" aria-hidden="true" tabindex="-1"></a>One practical application of AMEs is calculating the average difference between two groups or conditions (called the risk difference). Using the <span class="in">`avg_comparisons()`</span> function in the <span class="in">`marginaleffects`</span> package (@lst-avgcomp), we can compute this metric directly. By default, the function calculates the discrete difference between groups. The function can also compute other effect size metrics, such as odds ratios and risk ratios, depending on the research question. This flexibility makes it a powerful tool for interpreting regression results in a meaningful way.</span>
<span id="cb3-1072"><a href="#cb3-1072" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1075"><a href="#cb3-1075" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1076"><a href="#cb3-1076" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ame1</span></span>
<span id="cb3-1077"><a href="#cb3-1077" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-avgcomp</span></span>
<span id="cb3-1078"><a href="#cb3-1078" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: "Calculating the difference between probabilities with `avg_comparisons()`"</span></span>
<span id="cb3-1079"><a href="#cb3-1079" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-1080"><a href="#cb3-1080" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb3-1081"><a href="#cb3-1081" aria-hidden="true" tabindex="-1"></a><span class="co"># get risk difference by default</span></span>
<span id="cb3-1082"><a href="#cb3-1082" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1083"><a href="#cb3-1083" aria-hidden="true" tabindex="-1"></a>Beta_avg_comp <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(Beta_brms, <span class="at">comparison =</span> <span class="st">"difference"</span>)</span>
<span id="cb3-1084"><a href="#cb3-1084" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1085"><a href="#cb3-1085" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1088"><a href="#cb3-1088" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1089"><a href="#cb3-1089" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-ame1</span></span>
<span id="cb3-1090"><a href="#cb3-1090" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-1091"><a href="#cb3-1091" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Fluency difference</span></span>
<span id="cb3-1092"><a href="#cb3-1092" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-1093"><a href="#cb3-1093" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1094"><a href="#cb3-1094" aria-hidden="true" tabindex="-1"></a>Beta_avg_comp <span class="sc">|&gt;</span></span>
<span id="cb3-1095"><a href="#cb3-1095" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remove unwanted variables</span></span>
<span id="cb3-1096"><a href="#cb3-1096" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>predicted_lo, <span class="sc">-</span>predicted_hi, <span class="sc">-</span>tmp_idx, <span class="sc">-</span>predicted) <span class="sc">|&gt;</span></span>
<span id="cb3-1097"><a href="#cb3-1097" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-1098"><a href="#cb3-1098" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb3-1099"><a href="#cb3-1099" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{conf.low}, {conf.high}]"</span>)</span>
<span id="cb3-1100"><a href="#cb3-1100" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-1101"><a href="#cb3-1101" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Contrast"</span> <span class="ot">=</span> <span class="st">"contrast"</span>, <span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"estimate"</span>, <span class="st">"Term"</span> <span class="ot">=</span> <span class="st">"term"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1102"><a href="#cb3-1102" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Term, Contrast, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1103"><a href="#cb3-1103" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>(<span class="at">digits =</span> <span class="dv">2</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1104"><a href="#cb3-1104" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-1105"><a href="#cb3-1105" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1106"><a href="#cb3-1106" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1107"><a href="#cb3-1107" aria-hidden="true" tabindex="-1"></a>@tbl-ame1 presents the estimated difference for the fluency factor (Mean column). The difference between the fluent and disfluent conditions is 0.06, indicating that participants who watched a fluent instructor scored, on average, 6% higher on the final recall test than those who watched a disfluent instructor. However, the 95% credible interval includes 0 among the most credible values, suggesting we cannot rule out the possibility of a null or weakly negative effect.</span>
<span id="cb3-1108"><a href="#cb3-1108" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1109"><a href="#cb3-1109" aria-hidden="true" tabindex="-1"></a>To formally assess evidence in favor of the null hypothesis (i.e., that the effect is exactly zero), we can use the <span class="in">`hypothesis()`</span> function from the <span class="in">`brms`</span> package to compute an evidence ratio (ER), which is a Bayes factor for directional or point hypotheses.<span class="ot">[^6]</span>However, it is important to note that this procedure requires weakly informative or informative priors; the default priors in <span class="in">`brms`</span> are too diffuse and therefore cannot be used to accurately compute the ER in favor of a point null hypothesis.</span>
<span id="cb3-1110"><a href="#cb3-1110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1111"><a href="#cb3-1111" aria-hidden="true" tabindex="-1"></a><span class="ot">[^6]: </span><span class="in">`bayestestr`</span> also has functions that will allow users to compute Bayes Factors.</span>
<span id="cb3-1112"><a href="#cb3-1112" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1115"><a href="#cb3-1115" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1116"><a href="#cb3-1116" aria-hidden="true" tabindex="-1"></a><span class="co"># Proportions</span></span>
<span id="cb3-1117"><a href="#cb3-1117" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fl">0.35</span> <span class="co"># Fluent</span></span>
<span id="cb3-1118"><a href="#cb3-1118" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fl">0.30</span> <span class="co"># Disfluent</span></span>
<span id="cb3-1119"><a href="#cb3-1119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1120"><a href="#cb3-1120" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohen's h formula</span></span>
<span id="cb3-1121"><a href="#cb3-1121" aria-hidden="true" tabindex="-1"></a>cohens_h <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> (<span class="fu">asin</span>(<span class="fu">sqrt</span>(p1)) <span class="sc">-</span> <span class="fu">asin</span>(<span class="fu">sqrt</span>(p2)))</span>
<span id="cb3-1122"><a href="#cb3-1122" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1123"><a href="#cb3-1123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1124"><a href="#cb3-1124" aria-hidden="true" tabindex="-1"></a><span class="co">#CI lower and upper bounds</span></span>
<span id="cb3-1125"><a href="#cb3-1125" aria-hidden="true" tabindex="-1"></a>p1_lower <span class="ot">&lt;-</span> <span class="fl">0.328</span></span>
<span id="cb3-1126"><a href="#cb3-1126" aria-hidden="true" tabindex="-1"></a>p1_upper <span class="ot">&lt;-</span> <span class="fl">0.419</span></span>
<span id="cb3-1127"><a href="#cb3-1127" aria-hidden="true" tabindex="-1"></a>p2_lower <span class="ot">&lt;-</span> <span class="fl">0.250</span></span>
<span id="cb3-1128"><a href="#cb3-1128" aria-hidden="true" tabindex="-1"></a>p2_upper <span class="ot">&lt;-</span> <span class="fl">0.329</span></span>
<span id="cb3-1129"><a href="#cb3-1129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1130"><a href="#cb3-1130" aria-hidden="true" tabindex="-1"></a><span class="co"># Lower bound of h</span></span>
<span id="cb3-1131"><a href="#cb3-1131" aria-hidden="true" tabindex="-1"></a>h_lower <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> (<span class="fu">asin</span>(<span class="fu">sqrt</span>(p1_lower)) <span class="sc">-</span> <span class="fu">asin</span>(<span class="fu">sqrt</span>(p2_upper)))</span>
<span id="cb3-1132"><a href="#cb3-1132" aria-hidden="true" tabindex="-1"></a><span class="co"># Upper bound of h</span></span>
<span id="cb3-1133"><a href="#cb3-1133" aria-hidden="true" tabindex="-1"></a>h_upper <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> (<span class="fu">asin</span>(<span class="fu">sqrt</span>(p1_upper)) <span class="sc">-</span> <span class="fu">asin</span>(<span class="fu">sqrt</span>(p2_lower)))</span>
<span id="cb3-1134"><a href="#cb3-1134" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1135"><a href="#cb3-1135" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1136"><a href="#cb3-1136" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1137"><a href="#cb3-1137" aria-hidden="true" tabindex="-1"></a>In psychology, it is common to report effect size measures like Cohen's *d* [@cohen1977]. When working with proportions we can calculate something similar called Cohen's *h*. Taking our proportions, we can use the below equation (@eq-cohenh) to calculate Cohen's *h* along with the 95% Cr.I around it. Using this metric we see the effect size is small (<span class="in">`r round(cohens_h, 3)`</span>), 95% credible interval <span class="sc">\[</span><span class="in">`r round(h_lower,3)`</span>, <span class="in">`r round(h_upper,3)`</span><span class="sc">\]</span>.</span>
<span id="cb3-1138"><a href="#cb3-1138" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1139"><a href="#cb3-1139" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb3-1140"><a href="#cb3-1140" aria-hidden="true" tabindex="-1"></a>h = 2 \cdot \left( \arcsin\left(\sqrt{p_1}\right) - \arcsin\left(\sqrt{p_2}\right) \right)</span>
<span id="cb3-1141"><a href="#cb3-1141" aria-hidden="true" tabindex="-1"></a>$$ {#eq-cohenh}</span>
<span id="cb3-1142"><a href="#cb3-1142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1143"><a href="#cb3-1143" aria-hidden="true" tabindex="-1"></a><span class="fu">### Posterior Predictive Check</span></span>
<span id="cb3-1144"><a href="#cb3-1144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1145"><a href="#cb3-1145" aria-hidden="true" tabindex="-1"></a>@fig-ppcheckall (B) shows the predictive check for our Beta model. The model does a pretty good job at capturing the data (The draws are now between 0-1) and the predicted values from the model follow the observed data. However, it could be better.</span>
<span id="cb3-1146"><a href="#cb3-1146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1147"><a href="#cb3-1147" aria-hidden="true" tabindex="-1"></a><span class="fu">## Zero-Inflated Beta (ZIB) Regression</span></span>
<span id="cb3-1148"><a href="#cb3-1148" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1149"><a href="#cb3-1149" aria-hidden="true" tabindex="-1"></a>A limitation of the Beta regression model is that it can only accommodate values strictly between 0 and 1—it cannot handle values exactly equal to 0 or 1. In our dataset, we observed 9 rows where Accuracy equals zero. To fit a Beta regression model, we removed these values, but we have left out potentially valuable information from our model--especially if the end points of the scale are distinctive in some way. In our case, these 0s may be structural—that is, they represent real, systematic instances where participants failed to answer correctly (rather than random noise or measurement error). For example, the fluency of the instructor might be a key factor in predicting these zero responses. We will discuss two approaches for jointly modeling these end points with the continuous data. First, we can use a zero-inflated Beta (ZIB) model. This model still estimates the mean ($\mu$) and precision ($\phi$) of the Beta distribution for values between 0 and 1, but it also includes an additional parameter, $\alpha$, which captures the probability of observing structural 0s.</span>
<span id="cb3-1150"><a href="#cb3-1150" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1151"><a href="#cb3-1151" aria-hidden="true" tabindex="-1"></a>The zero-inflated Beta models a mixture of the data-generating process. The $\alpha$ parameter uses a logistic regression to model whether the data is 0 or not. Substantively, this could be a useful model when we think that 0s come from a process that is relatively distinct from the data that is greater than 0. For example, if we had a dataset with proportion of looks or eye fixations to certain areas on marketing materials, we might want a separate model for those that do not look at certain areas on the screen because individuals who do not look might be substantively different than those that look.</span>
<span id="cb3-1152"><a href="#cb3-1152" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1153"><a href="#cb3-1153" aria-hidden="true" tabindex="-1"></a>We can fit a ZIB model using <span class="in">`brms`</span> and use the <span class="in">`marginaleffects`</span> package to make inferences about our parameters of interest. Before we run a zero-inflated Beta model, we will need to transform our data again and remove the one 1 value in our data--we can keep our 0s. Similar to our Beta regression model we fit in <span class="in">`brms`</span>, we will use the <span class="in">`bf()`</span> function to fit several models. We fit our $\mu$ and $\phi$ parameters as well as our zero-inflated parameter ($\alpha$; here labeled as <span class="in">`zi`</span>). In <span class="in">`brms`</span> we can use the zero_inflated_Beta family (see @lst-zib).</span>
<span id="cb3-1154"><a href="#cb3-1154" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1157"><a href="#cb3-1157" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1158"><a href="#cb3-1158" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: zib brms model</span></span>
<span id="cb3-1159"><a href="#cb3-1159" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-zib</span></span>
<span id="cb3-1160"><a href="#cb3-1160" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting zib model with `brm()`</span></span>
<span id="cb3-1161"><a href="#cb3-1161" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-1162"><a href="#cb3-1162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1163"><a href="#cb3-1163" aria-hidden="true" tabindex="-1"></a><span class="co"># keep 0 but remove 1</span></span>
<span id="cb3-1164"><a href="#cb3-1164" aria-hidden="true" tabindex="-1"></a>data_Beta_0 <span class="ot">&lt;-</span> fluency_data <span class="sc">|&gt;</span></span>
<span id="cb3-1165"><a href="#cb3-1165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Accuracy <span class="sc">!=</span> <span class="dv">1</span>)</span>
<span id="cb3-1166"><a href="#cb3-1166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1167"><a href="#cb3-1167" aria-hidden="true" tabindex="-1"></a><span class="co"># set up model formual for zero-inflated Beta in brm</span></span>
<span id="cb3-1168"><a href="#cb3-1168" aria-hidden="true" tabindex="-1"></a>zib_model <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb3-1169"><a href="#cb3-1169" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># The mean of the 0-1 values, or mu</span></span>
<span id="cb3-1170"><a href="#cb3-1170" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency, <span class="co"># The precision of the 0-1 values, or phi</span></span>
<span id="cb3-1171"><a href="#cb3-1171" aria-hidden="true" tabindex="-1"></a>  zi <span class="sc">~</span> Fluency, <span class="co"># The zero-or-one-inflated part, or alpha</span></span>
<span id="cb3-1172"><a href="#cb3-1172" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">zero_inflated_beta</span>()</span>
<span id="cb3-1173"><a href="#cb3-1173" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-1174"><a href="#cb3-1174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1175"><a href="#cb3-1175" aria-hidden="true" tabindex="-1"></a><span class="co"># fit zib model with brm</span></span>
<span id="cb3-1176"><a href="#cb3-1176" aria-hidden="true" tabindex="-1"></a>fit_zi <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb3-1177"><a href="#cb3-1177" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> zib_model,</span>
<span id="cb3-1178"><a href="#cb3-1178" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_Beta_0,</span>
<span id="cb3-1179"><a href="#cb3-1179" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"bayes_zib_model0not1"</span></span>
<span id="cb3-1180"><a href="#cb3-1180" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-1181"><a href="#cb3-1181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1182"><a href="#cb3-1182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1183"><a href="#cb3-1183" aria-hidden="true" tabindex="-1"></a><span class="fu">### Posterior Predictive Check</span></span>
<span id="cb3-1184"><a href="#cb3-1184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1185"><a href="#cb3-1185" aria-hidden="true" tabindex="-1"></a>The ZIB model does a bit better at capturing the structure of the data then the Beta regression model (see @fig-ppcheckall). Specifically, the ZIB model more accurately captures the increased density of values near the lower end of the scale (i.e., near zero), which the standard Beta model underestimates. The ZIB model’s predictive distributions also align more closely with the observed data across the entire range, particularly in the peak and tail regions. This improved fit likely reflects the ZIB model’s ability to explicitly model excess 0s (or near-zero values) via its inflation component, allowing it to better account for features in the data that a standard Beta distribution cannot accommodate.</span>
<span id="cb3-1186"><a href="#cb3-1186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1189"><a href="#cb3-1189" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1190"><a href="#cb3-1190" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig ppcheckzib</span></span>
<span id="cb3-1191"><a href="#cb3-1191" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Posterior predictive check for the zib model</span></span>
<span id="cb3-1192"><a href="#cb3-1192" aria-hidden="true" tabindex="-1"></a>pp_check3 <span class="ot">&lt;-</span> <span class="fu">pp_check</span>(fit_zi, <span class="at">ndraws =</span> <span class="dv">100</span>) <span class="sc">+</span> <span class="co"># ndraws is nu</span></span>
<span id="cb3-1193"><a href="#cb3-1193" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>()</span>
<span id="cb3-1194"><a href="#cb3-1194" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1195"><a href="#cb3-1195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1196"><a href="#cb3-1196" aria-hidden="true" tabindex="-1"></a><span class="fu">### Predicted Probabilities and Marginal Effects</span></span>
<span id="cb3-1197"><a href="#cb3-1197" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1198"><a href="#cb3-1198" aria-hidden="true" tabindex="-1"></a>@tbl-all under the zero-inflated Beta regression column provides a summary of the posterior distribution for each parameter. As stated before, it is preferable to back-transform our estimates to get probabilities. To get the predicted probabilities we can again use the <span class="in">`avg_predictions()`</span> and <span class="in">`avg_comparisons()`</span> functions from <span class="in">`marginaleffects`</span> package <span class="co">[</span><span class="ot">@marginaleffects</span><span class="co">]</span> to get predicted probabilities and the probability difference between the levels of each factor. We can model the parameters separately using the <span class="in">`dpar`</span> argument setting to: $\mu$, $\phi$, $\alpha$. Here we look at the risk difference for Fluency under each parameter.</span>
<span id="cb3-1199"><a href="#cb3-1199" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1200"><a href="#cb3-1200" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Mean</span></span>
<span id="cb3-1201"><a href="#cb3-1201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1202"><a href="#cb3-1202" aria-hidden="true" tabindex="-1"></a>As shown in @tbl-marg-mu-brms, there is little evidence for an effect of Fluency -- the 95% Cr.I includes zero, suggesting substantial uncertainty about the direction and magnitude of the effect--that is, though most of the posterior density supports positive effects, nil and weakly negative effects cannot be ruled out.</span>
<span id="cb3-1203"><a href="#cb3-1203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1206"><a href="#cb3-1206" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1207"><a href="#cb3-1207" aria-hidden="true" tabindex="-1"></a>marg_zi_brms <span class="ot">&lt;-</span> fit_zi <span class="sc">|&gt;</span></span>
<span id="cb3-1208"><a href="#cb3-1208" aria-hidden="true" tabindex="-1"></a>  <span class="fu">avg_comparisons</span>(</span>
<span id="cb3-1209"><a href="#cb3-1209" aria-hidden="true" tabindex="-1"></a>    <span class="at">variables =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb3-1210"><a href="#cb3-1210" aria-hidden="true" tabindex="-1"></a>    <span class="at">dpar =</span> <span class="st">"mu"</span>,</span>
<span id="cb3-1211"><a href="#cb3-1211" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb3-1212"><a href="#cb3-1212" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-1213"><a href="#cb3-1213" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1214"><a href="#cb3-1214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1217"><a href="#cb3-1217" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1218"><a href="#cb3-1218" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-marg-mu-brms</span></span>
<span id="cb3-1219"><a href="#cb3-1219" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap:  Probablity fluency difference ($\mu$)</span></span>
<span id="cb3-1220"><a href="#cb3-1220" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-1221"><a href="#cb3-1221" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-1222"><a href="#cb3-1222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1223"><a href="#cb3-1223" aria-hidden="true" tabindex="-1"></a>marg_zi_brms <span class="sc">|&gt;</span></span>
<span id="cb3-1224"><a href="#cb3-1224" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(term, contrast, estimate, conf.low, conf.high) <span class="sc">|&gt;</span></span>
<span id="cb3-1225"><a href="#cb3-1225" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-1226"><a href="#cb3-1226" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb3-1227"><a href="#cb3-1227" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{conf.low}, {conf.high}]"</span>)</span>
<span id="cb3-1228"><a href="#cb3-1228" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-1229"><a href="#cb3-1229" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Contrast"</span> <span class="ot">=</span> <span class="st">"contrast"</span>, <span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"estimate"</span>, <span class="st">"Term"</span> <span class="ot">=</span> <span class="st">"term"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1230"><a href="#cb3-1230" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Term, Contrast, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1231"><a href="#cb3-1231" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-1232"><a href="#cb3-1232" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1233"><a href="#cb3-1233" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-1234"><a href="#cb3-1234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1235"><a href="#cb3-1235" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1236"><a href="#cb3-1236" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1237"><a href="#cb3-1237" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Dispersion</span></span>
<span id="cb3-1238"><a href="#cb3-1238" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1239"><a href="#cb3-1239" aria-hidden="true" tabindex="-1"></a>As shown in @tbl-marg-phi-brms, the posterior estimates suggest a credible effect of Fluency on dispersion ($\phi$), with disfluent responses showing greater variability. The 95% Cr.I for the fluency contrast does not include zero, indicating a high probability in differences in precision.</span>
<span id="cb3-1240"><a href="#cb3-1240" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1243"><a href="#cb3-1243" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1244"><a href="#cb3-1244" aria-hidden="true" tabindex="-1"></a>marg_phi_brms <span class="ot">&lt;-</span> fit_zi <span class="sc">|&gt;</span></span>
<span id="cb3-1245"><a href="#cb3-1245" aria-hidden="true" tabindex="-1"></a>  <span class="fu">avg_comparisons</span>(</span>
<span id="cb3-1246"><a href="#cb3-1246" aria-hidden="true" tabindex="-1"></a>    <span class="at">variables =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb3-1247"><a href="#cb3-1247" aria-hidden="true" tabindex="-1"></a>    <span class="at">dpar =</span> <span class="st">"phi"</span>,</span>
<span id="cb3-1248"><a href="#cb3-1248" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb3-1249"><a href="#cb3-1249" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-1250"><a href="#cb3-1250" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1251"><a href="#cb3-1251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1254"><a href="#cb3-1254" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1255"><a href="#cb3-1255" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-marg-phi-brms</span></span>
<span id="cb3-1256"><a href="#cb3-1256" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap:  Probablity fluency difference ($\phi$)</span></span>
<span id="cb3-1257"><a href="#cb3-1257" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-1258"><a href="#cb3-1258" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-1259"><a href="#cb3-1259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1260"><a href="#cb3-1260" aria-hidden="true" tabindex="-1"></a>marg_phi_brms <span class="sc">|&gt;</span></span>
<span id="cb3-1261"><a href="#cb3-1261" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(term, contrast, estimate, conf.low, conf.high) <span class="sc">|&gt;</span></span>
<span id="cb3-1262"><a href="#cb3-1262" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-1263"><a href="#cb3-1263" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb3-1264"><a href="#cb3-1264" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{conf.low}, {conf.high}]"</span>)</span>
<span id="cb3-1265"><a href="#cb3-1265" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-1266"><a href="#cb3-1266" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Contrast"</span> <span class="ot">=</span> <span class="st">"contrast"</span>, <span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"estimate"</span>, <span class="st">"Term"</span> <span class="ot">=</span> <span class="st">"term"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1267"><a href="#cb3-1267" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Term, Contrast, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1268"><a href="#cb3-1268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-1269"><a href="#cb3-1269" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1270"><a href="#cb3-1270" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-1271"><a href="#cb3-1271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1272"><a href="#cb3-1272" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1273"><a href="#cb3-1273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1274"><a href="#cb3-1274" aria-hidden="true" tabindex="-1"></a><span class="fu">### Zero-Inflation</span></span>
<span id="cb3-1275"><a href="#cb3-1275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1276"><a href="#cb3-1276" aria-hidden="true" tabindex="-1"></a>We can harness the power of <span class="in">`marginaleffects`</span> again and plot the posterior difference between the fluent and disfluent conditions (see @fig-zi-plot). In @fig-zi-plot, there is evidence that watching a lecture video with a fluent instructor reduces the probability of a zero response by approximately 13%. The 95% Cr.I for this effect does not include zero, suggesting a meaningful reduction in the likelihood of zero outcomes under fluent instruction. We can harness the power of <span class="in">`marginaleffects`</span> again and plot the posterior probability of each level (see @fig-zi-plot).</span>
<span id="cb3-1277"><a href="#cb3-1277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1280"><a href="#cb3-1280" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1281"><a href="#cb3-1281" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl marg-zib-brms</span></span>
<span id="cb3-1282"><a href="#cb3-1282" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb3-1283"><a href="#cb3-1283" aria-hidden="true" tabindex="-1"></a>marg_zi_brms <span class="ot">&lt;-</span> fit_zi <span class="sc">|&gt;</span></span>
<span id="cb3-1284"><a href="#cb3-1284" aria-hidden="true" tabindex="-1"></a>  <span class="fu">avg_comparisons</span>(</span>
<span id="cb3-1285"><a href="#cb3-1285" aria-hidden="true" tabindex="-1"></a>    <span class="at">variables =</span> <span class="fu">list</span>(<span class="at">Fluency =</span> <span class="st">"revpairwise"</span>),</span>
<span id="cb3-1286"><a href="#cb3-1286" aria-hidden="true" tabindex="-1"></a>    <span class="at">dpar =</span> <span class="st">"zi"</span>,</span>
<span id="cb3-1287"><a href="#cb3-1287" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb3-1288"><a href="#cb3-1288" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-1289"><a href="#cb3-1289" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1290"><a href="#cb3-1290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1293"><a href="#cb3-1293" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1294"><a href="#cb3-1294" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-zi-plot</span></span>
<span id="cb3-1295"><a href="#cb3-1295" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Visualization of the predicted difference for zero-inflated part of model"</span></span>
<span id="cb3-1296"><a href="#cb3-1296" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-1297"><a href="#cb3-1297" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1298"><a href="#cb3-1298" aria-hidden="true" tabindex="-1"></a>marg_zi_brms <span class="sc">|&gt;</span></span>
<span id="cb3-1299"><a href="#cb3-1299" aria-hidden="true" tabindex="-1"></a>  <span class="fu">posterior_draws</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-1300"><a href="#cb3-1300" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> draw, <span class="at">y =</span> contrast)) <span class="sc">+</span></span>
<span id="cb3-1301"><a href="#cb3-1301" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_halfeye</span>(<span class="at">.width =</span> <span class="fu">c</span>(.<span class="dv">5</span>, .<span class="dv">8</span>, .<span class="dv">95</span>), <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb3-1302"><a href="#cb3-1302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_okabe_ito</span>() <span class="sc">+</span></span>
<span id="cb3-1303"><a href="#cb3-1303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">label_percent</span>()) <span class="sc">+</span></span>
<span id="cb3-1304"><a href="#cb3-1304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb3-1305"><a href="#cb3-1305" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Accuracy"</span>,</span>
<span id="cb3-1306"><a href="#cb3-1306" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb3-1307"><a href="#cb3-1307" aria-hidden="true" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">"50%, 80% and 95% credible intervals shown in black"</span></span>
<span id="cb3-1308"><a href="#cb3-1308" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-1309"><a href="#cb3-1309" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">"solid"</span>, <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb3-1310"><a href="#cb3-1310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb3-1311"><a href="#cb3-1311" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb3-1312"><a href="#cb3-1312" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1313"><a href="#cb3-1313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1314"><a href="#cb3-1314" aria-hidden="true" tabindex="-1"></a><span class="fu">## Zero-One-Inflated Beta (ZOIB)</span></span>
<span id="cb3-1315"><a href="#cb3-1315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1316"><a href="#cb3-1316" aria-hidden="true" tabindex="-1"></a>The ZIB model works well if you have 0s in your data, but not 1s.<span class="ot">[^7]</span> In our previous examples we either got rid of both 0s and 1s (Beta regression), or removed the 1s (ZIB). Sometimes it is theoretically useful to model both 0s and 1s as separate processes or to consider these values as essentially similar parts of the continuous response, as we show later in the ordered Beta regression model. For example, this is important in visual analog scale data where there might be a prevalence of responses at the bounds <span class="co">[</span><span class="ot">@kong2016</span><span class="co">]</span>, in JOL tasks <span class="co">[</span><span class="ot">@wilford2020</span><span class="co">]</span>, or in a free-list task where individuals provide open responses to some question or topic which are then recoded to fall between 0-1 <span class="co">[</span><span class="ot">@bendixen2023</span><span class="co">]</span>. Here 0s and 1s are meaningful; 0 means item was not listed and 1 means the item was listed first.</span>
<span id="cb3-1317"><a href="#cb3-1317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1318"><a href="#cb3-1318" aria-hidden="true" tabindex="-1"></a><span class="ot">[^7]: </span>In cases where your data include exact 1s but no 0s, you can fit a one-inflated Beta regression model in <span class="in">`brms`</span> by setting the <span class="in">`coi`</span> parameter to <span class="in">`1`</span>. This tells the model that all point masses occur at 1, rather than being split between 0 and 1. In other words, <span class="in">`coi = 1`</span> assumes that any inflation in the data is due entirely to values at 1. In our data, we have exactly one value equal to 1<span class="sc">\[</span>\^6<span class="sc">\]</span>. While probably not significant to alter our findings, we can model 1s with a special type of model called the zero-one-inflated Beta (ZOIB) model <span class="co">[</span><span class="ot">@liu2015</span><span class="co">]</span> if we believe that both 0s and 1s are distinct outcomes.</span>
<span id="cb3-1319"><a href="#cb3-1319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1320"><a href="#cb3-1320" aria-hidden="true" tabindex="-1"></a>Similar to our Beta and zero-inflated models above, we can fit a ZOIB model in brms quite easily using the <span class="in">`zero_one_inflated_Beta`</span> family. In this model, we simultaneously estimate the mean ($\mu$) and precision ($\phi$) of the Beta distribution, a zero-one inflation parameter ($\alpha$) that represents the probability that an observation is either exactly 0 or 1 (i.e., 0 or 1 vs. not 0 or 1) and a conditional one-inflation parameter ($\gamma$) that represents the probability that, given an observation is at one of the endpoints, it is 1 (i.e, 1 vs. not 1). This specification captures the entire range of possible values while remaining constrained between 0 and 1. To get a better sense of how $\alpha$ and $\gamma$ control the distribution of values, @fig-zoib presents simulated data across combinations of these parameters. As $\alpha$ increases, we see a greater proportion of responses at the endpoints. As $\gamma$ increases, the proportion of endpoint responses at 1 grows relative to 0, making the spikes at 1 more prominent as $\gamma$ approaches 1. This visualization illustrates how the ZOIB model flexibly accounts for both the continuous portion of the distribution and the occurrence of exact 0s and 1s.</span>
<span id="cb3-1321"><a href="#cb3-1321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1324"><a href="#cb3-1324" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1325"><a href="#cb3-1325" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-zoib</span></span>
<span id="cb3-1326"><a href="#cb3-1326" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Simulated data from a ZOIB model illustrating the effects of the zero-one inflation parameter ($\alpha$) and the conditional one-inflation parameter ($\gamma$).</span></span>
<span id="cb3-1327"><a href="#cb3-1327" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb3-1328"><a href="#cb3-1328" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb3-1329"><a href="#cb3-1329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1330"><a href="#cb3-1330" aria-hidden="true" tabindex="-1"></a><span class="co"># Zero-one inflated Beta generator adapted from</span></span>
<span id="cb3-1331"><a href="#cb3-1331" aria-hidden="true" tabindex="-1"></a><span class="co"># https://vuorre.com/posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-Beta-models/</span></span>
<span id="cb3-1332"><a href="#cb3-1332" aria-hidden="true" tabindex="-1"></a>rzoib <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">n =</span> <span class="dv">1000</span>, <span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">gamma =</span> <span class="fl">0.45</span>, <span class="at">mu =</span> <span class="fl">0.7</span>, <span class="at">phi =</span> <span class="dv">3</span>) {</span>
<span id="cb3-1333"><a href="#cb3-1333" aria-hidden="true" tabindex="-1"></a>  a <span class="ot">&lt;-</span> mu <span class="sc">*</span> phi</span>
<span id="cb3-1334"><a href="#cb3-1334" aria-hidden="true" tabindex="-1"></a>  b <span class="ot">&lt;-</span> (<span class="dv">1</span> <span class="sc">-</span> mu) <span class="sc">*</span> phi</span>
<span id="cb3-1335"><a href="#cb3-1335" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb3-1336"><a href="#cb3-1336" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbinom</span>(n, <span class="dv">1</span>, alpha),</span>
<span id="cb3-1337"><a href="#cb3-1337" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbinom</span>(n, <span class="dv">1</span>, gamma),</span>
<span id="cb3-1338"><a href="#cb3-1338" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbeta</span>(n, a, b)</span>
<span id="cb3-1339"><a href="#cb3-1339" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-1340"><a href="#cb3-1340" aria-hidden="true" tabindex="-1"></a>  y</span>
<span id="cb3-1341"><a href="#cb3-1341" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb3-1342"><a href="#cb3-1342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1343"><a href="#cb3-1343" aria-hidden="true" tabindex="-1"></a><span class="co"># Optimized parameter grid</span></span>
<span id="cb3-1344"><a href="#cb3-1344" aria-hidden="true" tabindex="-1"></a>param_grid <span class="ot">&lt;-</span> <span class="fu">expand_grid</span>(</span>
<span id="cb3-1345"><a href="#cb3-1345" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.4</span>),</span>
<span id="cb3-1346"><a href="#cb3-1346" aria-hidden="true" tabindex="-1"></a>  <span class="at">gamma =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.9</span>),</span>
<span id="cb3-1347"><a href="#cb3-1347" aria-hidden="true" tabindex="-1"></a>  <span class="at">mu =</span> <span class="fl">0.7</span>,</span>
<span id="cb3-1348"><a href="#cb3-1348" aria-hidden="true" tabindex="-1"></a>  <span class="at">phi =</span> <span class="dv">3</span></span>
<span id="cb3-1349"><a href="#cb3-1349" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-1350"><a href="#cb3-1350" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1351"><a href="#cb3-1351" aria-hidden="true" tabindex="-1"></a><span class="co"># Simulate data</span></span>
<span id="cb3-1352"><a href="#cb3-1352" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb3-1353"><a href="#cb3-1353" aria-hidden="true" tabindex="-1"></a>sim_data <span class="ot">&lt;-</span> param_grid <span class="sc">%&gt;%</span></span>
<span id="cb3-1354"><a href="#cb3-1354" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-1355"><a href="#cb3-1355" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">pmap</span>(</span>
<span id="cb3-1356"><a href="#cb3-1356" aria-hidden="true" tabindex="-1"></a>      <span class="fu">list</span>(alpha, gamma, mu, phi),</span>
<span id="cb3-1357"><a href="#cb3-1357" aria-hidden="true" tabindex="-1"></a>      <span class="sc">~</span> <span class="fu">rzoib</span>(<span class="dv">1000</span>, ..<span class="dv">1</span>, ..<span class="dv">2</span>, ..<span class="dv">3</span>, ..<span class="dv">4</span>)</span>
<span id="cb3-1358"><a href="#cb3-1358" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb3-1359"><a href="#cb3-1359" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb3-1360"><a href="#cb3-1360" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(data) <span class="sc">%&gt;%</span></span>
<span id="cb3-1361"><a href="#cb3-1361" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Accuracy =</span> data)</span>
<span id="cb3-1362"><a href="#cb3-1362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1363"><a href="#cb3-1363" aria-hidden="true" tabindex="-1"></a><span class="co"># Classify type</span></span>
<span id="cb3-1364"><a href="#cb3-1364" aria-hidden="true" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> sim_data <span class="sc">%&gt;%</span></span>
<span id="cb3-1365"><a href="#cb3-1365" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-1366"><a href="#cb3-1366" aria-hidden="true" tabindex="-1"></a>    <span class="at">Type =</span> <span class="fu">case_when</span>(</span>
<span id="cb3-1367"><a href="#cb3-1367" aria-hidden="true" tabindex="-1"></a>      Accuracy <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"Zero"</span>,</span>
<span id="cb3-1368"><a href="#cb3-1368" aria-hidden="true" tabindex="-1"></a>      Accuracy <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"One"</span>,</span>
<span id="cb3-1369"><a href="#cb3-1369" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"Beta"</span></span>
<span id="cb3-1370"><a href="#cb3-1370" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb3-1371"><a href="#cb3-1371" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-1372"><a href="#cb3-1372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1373"><a href="#cb3-1373" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure 'Zero' comes before 'One'</span></span>
<span id="cb3-1374"><a href="#cb3-1374" aria-hidden="true" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> plot_data <span class="sc">%&gt;%</span></span>
<span id="cb3-1375"><a href="#cb3-1375" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Type =</span> <span class="fu">factor</span>(Type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Zero"</span>, <span class="st">"One"</span>, <span class="st">"Beta"</span>)))</span>
<span id="cb3-1376"><a href="#cb3-1376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1377"><a href="#cb3-1377" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb3-1378"><a href="#cb3-1378" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Beta histogram</span></span>
<span id="cb3-1379"><a href="#cb3-1379" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(</span>
<span id="cb3-1380"><a href="#cb3-1380" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> plot_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Type <span class="sc">==</span> <span class="st">"Beta"</span>),</span>
<span id="cb3-1381"><a href="#cb3-1381" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> Accuracy, <span class="at">y =</span> <span class="fu">after_stat</span>(count)),</span>
<span id="cb3-1382"><a href="#cb3-1382" aria-hidden="true" tabindex="-1"></a>    <span class="at">bins =</span> <span class="dv">50</span>,</span>
<span id="cb3-1383"><a href="#cb3-1383" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"gray60"</span>,</span>
<span id="cb3-1384"><a href="#cb3-1384" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"white"</span>,</span>
<span id="cb3-1385"><a href="#cb3-1385" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.8</span></span>
<span id="cb3-1386"><a href="#cb3-1386" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-1387"><a href="#cb3-1387" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Spikes at 0 and 1</span></span>
<span id="cb3-1388"><a href="#cb3-1388" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(</span>
<span id="cb3-1389"><a href="#cb3-1389" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> plot_data <span class="sc">%&gt;%</span></span>
<span id="cb3-1390"><a href="#cb3-1390" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(Type <span class="sc">!=</span> <span class="st">"Beta"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb3-1391"><a href="#cb3-1391" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(alpha, gamma, Accuracy, Type) <span class="sc">%&gt;%</span></span>
<span id="cb3-1392"><a href="#cb3-1392" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>),</span>
<span id="cb3-1393"><a href="#cb3-1393" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> Accuracy, <span class="at">y =</span> count, <span class="at">fill =</span> Type),</span>
<span id="cb3-1394"><a href="#cb3-1394" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="fl">0.01</span></span>
<span id="cb3-1395"><a href="#cb3-1395" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-1396"><a href="#cb3-1396" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb3-1397"><a href="#cb3-1397" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Zero"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"One"</span> <span class="ot">=</span> <span class="st">"blue"</span>),</span>
<span id="cb3-1398"><a href="#cb3-1398" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"Type"</span></span>
<span id="cb3-1399"><a href="#cb3-1399" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-1400"><a href="#cb3-1400" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(</span>
<span id="cb3-1401"><a href="#cb3-1401" aria-hidden="true" tabindex="-1"></a>    gamma <span class="sc">~</span> alpha,</span>
<span id="cb3-1402"><a href="#cb3-1402" aria-hidden="true" tabindex="-1"></a>    <span class="at">labeller =</span> label_both,</span>
<span id="cb3-1403"><a href="#cb3-1403" aria-hidden="true" tabindex="-1"></a>    <span class="at">scales =</span> <span class="st">"free_y"</span>,</span>
<span id="cb3-1404"><a href="#cb3-1404" aria-hidden="true" tabindex="-1"></a>    <span class="at">switch =</span> <span class="st">"y"</span></span>
<span id="cb3-1405"><a href="#cb3-1405" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-1406"><a href="#cb3-1406" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb3-1407"><a href="#cb3-1407" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Accuracy"</span>,</span>
<span id="cb3-1408"><a href="#cb3-1408" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Count"</span></span>
<span id="cb3-1409"><a href="#cb3-1409" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-1410"><a href="#cb3-1410" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb3-1411"><a href="#cb3-1411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">label_percent</span>())</span>
<span id="cb3-1412"><a href="#cb3-1412" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1413"><a href="#cb3-1413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1414"><a href="#cb3-1414" aria-hidden="true" tabindex="-1"></a>To fit a ZOIB model we use the <span class="in">`bf()`</span> function. We model each parameter as a function of Fluency. We then pass the <span class="in">`zoib_model`</span> to our <span class="in">`brm()`</span> function (see @lst-zoib). The summary of the output is in @tbl-all (under <span class="in">`ZOIB`</span>).</span>
<span id="cb3-1415"><a href="#cb3-1415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1418"><a href="#cb3-1418" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1419"><a href="#cb3-1419" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fit zoib model</span></span>
<span id="cb3-1420"><a href="#cb3-1420" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-zoib</span></span>
<span id="cb3-1421"><a href="#cb3-1421" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting a ZOIB model with `brm()`.</span></span>
<span id="cb3-1422"><a href="#cb3-1422" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-1423"><a href="#cb3-1423" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb3-1424"><a href="#cb3-1424" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb3-1425"><a href="#cb3-1425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1426"><a href="#cb3-1426" aria-hidden="true" tabindex="-1"></a><span class="co"># fit the zoib model</span></span>
<span id="cb3-1427"><a href="#cb3-1427" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1428"><a href="#cb3-1428" aria-hidden="true" tabindex="-1"></a>zoib_model <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb3-1429"><a href="#cb3-1429" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># The mean of the 0-1 values, or mu</span></span>
<span id="cb3-1430"><a href="#cb3-1430" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency, <span class="co"># The precision of the 0-1 values, or phi</span></span>
<span id="cb3-1431"><a href="#cb3-1431" aria-hidden="true" tabindex="-1"></a>  zoi <span class="sc">~</span> Fluency, <span class="co"># The zero-or-one-inflated part, or alpha</span></span>
<span id="cb3-1432"><a href="#cb3-1432" aria-hidden="true" tabindex="-1"></a>  coi <span class="sc">~</span> Fluency, <span class="co"># The one-inflated part, conditional on the 1s, or gamma</span></span>
<span id="cb3-1433"><a href="#cb3-1433" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">zero_one_inflated_beta</span>()</span>
<span id="cb3-1434"><a href="#cb3-1434" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-1435"><a href="#cb3-1435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1436"><a href="#cb3-1436" aria-hidden="true" tabindex="-1"></a>fit_zoib <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb3-1437"><a href="#cb3-1437" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> zoib_model,</span>
<span id="cb3-1438"><a href="#cb3-1438" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb3-1439"><a href="#cb3-1439" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"bayes_zoib_model"</span></span>
<span id="cb3-1440"><a href="#cb3-1440" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-1441"><a href="#cb3-1441" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1442"><a href="#cb3-1442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1445"><a href="#cb3-1445" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1446"><a href="#cb3-1446" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl zoib</span></span>
<span id="cb3-1447"><a href="#cb3-1447" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-1448"><a href="#cb3-1448" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-1449"><a href="#cb3-1449" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb3-1450"><a href="#cb3-1450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1451"><a href="#cb3-1451" aria-hidden="true" tabindex="-1"></a>zoib_model <span class="ot">&lt;-</span> <span class="fu">model_parameters</span>(fit_zoib, <span class="st">"mean"</span>)</span>
<span id="cb3-1452"><a href="#cb3-1452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1453"><a href="#cb3-1453" aria-hidden="true" tabindex="-1"></a>zoib_model <span class="sc">|&gt;</span></span>
<span id="cb3-1454"><a href="#cb3-1454" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-1455"><a href="#cb3-1455" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb3-1456"><a href="#cb3-1456" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{CI_low}, {CI_high}]"</span>)</span>
<span id="cb3-1457"><a href="#cb3-1457" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-1458"><a href="#cb3-1458" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Parameter, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, pd) <span class="sc">|&gt;</span></span>
<span id="cb3-1459"><a href="#cb3-1459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1460"><a href="#cb3-1460" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-1461"><a href="#cb3-1461" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1462"><a href="#cb3-1462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1463"><a href="#cb3-1463" aria-hidden="true" tabindex="-1"></a><span class="fu">### Model Parameters</span></span>
<span id="cb3-1464"><a href="#cb3-1464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1465"><a href="#cb3-1465" aria-hidden="true" tabindex="-1"></a>The output for the model is lengthy because we are estimating three distinct components, each with their own independent responses and sub-models. All the coefficients are on the logit scale, except $\phi$ , which is on the log scale. Thankfully drawing inferences for all these different parameters, plotting their distributions, and estimating their average marginal effects looks exactly the same--all the <span class="in">`brms`</span> and <span class="in">`marginaleffects`</span> functions we used work the same.</span>
<span id="cb3-1466"><a href="#cb3-1466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1467"><a href="#cb3-1467" aria-hidden="true" tabindex="-1"></a><span class="fu">### Predictions and Marginal Effects</span></span>
<span id="cb3-1468"><a href="#cb3-1468" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1469"><a href="#cb3-1469" aria-hidden="true" tabindex="-1"></a>With <span class="in">`marginaleffects`</span> we can choose marginalize over all the sub-models, averaged across the 0s, continuous responses, and 1s in the data, or we can model the parameters separately using the <span class="in">`dpar`</span> argument like we did above setting it to: $\mu$, $\phi$, $\alpha$, $\gamma$ (see below). Using <span class="in">`avg_predictions()`</span> and not setting <span class="in">`dpar`</span> we can get the predicted probabilities across all the sub-models. We can also plot the overall difference between fluency and disfluency for the whole model with <span class="in">`plot_predictions()`</span>.</span>
<span id="cb3-1470"><a href="#cb3-1470" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1471"><a href="#cb3-1471" aria-hidden="true" tabindex="-1"></a>In addition, we show below how one can extract the predicted probabilities and marginal effects for $\gamma$ (and a similar process for any other model component, <span class="in">`zoi`</span>, etc.):</span>
<span id="cb3-1472"><a href="#cb3-1472" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1475"><a href="#cb3-1475" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1476"><a href="#cb3-1476" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: avg coi</span></span>
<span id="cb3-1477"><a href="#cb3-1477" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-coi</span></span>
<span id="cb3-1478"><a href="#cb3-1478" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: "Extracting predicted probabilities and marginal effects for conditional-one parameter"</span></span>
<span id="cb3-1479"><a href="#cb3-1479" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-1480"><a href="#cb3-1480" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1481"><a href="#cb3-1481" aria-hidden="true" tabindex="-1"></a><span class="co"># get average predictions for coi param</span></span>
<span id="cb3-1482"><a href="#cb3-1482" aria-hidden="true" tabindex="-1"></a>coi_probs <span class="ot">&lt;-</span> <span class="fu">avg_predictions</span>(fit_zoib, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Fluency"</span>), <span class="at">dpar =</span> <span class="st">"coi"</span>)</span>
<span id="cb3-1483"><a href="#cb3-1483" aria-hidden="true" tabindex="-1"></a><span class="co"># get differene between the two conditions</span></span>
<span id="cb3-1484"><a href="#cb3-1484" aria-hidden="true" tabindex="-1"></a>coi_me <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(fit_zoib, <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Fluency"</span>), <span class="at">dpar =</span> <span class="st">"coi"</span>)</span>
<span id="cb3-1485"><a href="#cb3-1485" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1486"><a href="#cb3-1486" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1487"><a href="#cb3-1487" aria-hidden="true" tabindex="-1"></a><span class="fu">## Ordered Beta Regression</span></span>
<span id="cb3-1488"><a href="#cb3-1488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1489"><a href="#cb3-1489" aria-hidden="true" tabindex="-1"></a>Looking at the output from the ZOIB model (@tbl-all), we can see how running a model like this can become fairly complex as it is fitting distinct sub-models for each component of the scale. The ability to consider 0s and 1s as distinct processes from continuous values comes at a price in terms of complexity and interpretability. A simplified version of the zero-one-inflated beta (ZOIB) model, known as ordered Beta regression <span class="co">[</span><span class="ot">@kubinec2022; see also @Makowski2025CHOCO for a reparameterized version called the *Beta-Gate* model</span><span class="co">]</span>, has been recently proposed. The ordered Beta regression model exploits the fact that, for most analyses, the continuous values (between 0-1) and the discrete outcomes (e.g., 0 or 1) are *ordered*. For example, as a covariate $x$ increases or decreases, we should expect the bounded outcome $y$ to increase or decrease monotonically as well from 0 to $(0,1)$ to 1. The ZOIB model does not impose this restriction; a covariate could increase and the response $y$ could increase in its continuous values while *simultaneously* decreasing at *both* end points.<span class="ot">[^8]</span> This complexity is not immediately obvious when fitting the ZOIB, nor is it a potential relationship that many scholars want to consider when examining how covariates influence a bounded scale.</span>
<span id="cb3-1490"><a href="#cb3-1490" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1491"><a href="#cb3-1491" aria-hidden="true" tabindex="-1"></a><span class="ot">[^8]: </span>For a more complete description of this issue, we refer the reader to @kubinec2022.</span>
<span id="cb3-1492"><a href="#cb3-1492" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1493"><a href="#cb3-1493" aria-hidden="true" tabindex="-1"></a>To make the response ordered, the ordered Beta regression model estimates a weighted combination of a standard Beta regression model for continuous responses and a logit model for the discrete values of the response. By doing so, the amount of distinctiveness between the continuous responses and the discrete end points is a function of the data (and any informative priors) rather than strictly defined as fully distinct processes as in the ZOIB. For some datasets, the continuous and discrete responses will be fairly distinct, and in others less so. To give a slightly absured example, if a value of 0 meant that the subject was a fish, and a value of 1 meant that the subject was now a trombone, then the ordered Beta model would no longer be appropriate.</span>
<span id="cb3-1494"><a href="#cb3-1494" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1495"><a href="#cb3-1495" aria-hidden="true" tabindex="-1"></a>The weights that average together the two parts of the outcome (i.e., discrete and continuous) are determined by cutpoints that are estimated in conjunction with the data in a similar manner to what is known as an ordered logit model. An in-depth explanation of ordinal regression is beyond the scope of this tutorial <span class="co">[</span><span class="ot">but see @Fullerton2023; @bürkner2019</span><span class="co">]</span>. At a basic level, ordinal regression models are useful for outcome variables that are categorical in nature and have some inherent ordering (e.g., Likert scale items). To preserve this ordering, ordinal models rely on the cumulative probability distribution. Within an ordinal regression model it is assumed that there is a continuous but unobserved latent variable that determines which of $k$ ordinal responses will be selected. For example on a typical Likert scale from 'Strongly Disagree' to 'Strongly Agree', you could assume that there is a continuous, unobserved variable called 'Agreement'.</span>
<span id="cb3-1496"><a href="#cb3-1496" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1497"><a href="#cb3-1497" aria-hidden="true" tabindex="-1"></a>While we cannot measure Agreement directly, the ordinal response gives us some indication about where participants are on the continuous Agreement scale. $k - 1$ cutoffs are then estimated to indicate the point on the continuous Agreement scale at which your Agreement level is high enough to push you into the next ordinal category (say Agree to Strongly Agree). Coefficients in the model estimate how much different predictors change the estimated *continuous* scale (here, Agreement). Since there’s only one underlying process, there’s only one set of coefficients to work with (proportional odds assumption). In an ordered Beta regression, three ordered categories are modeled: (1) exactly zero, (2) somewhere between zero and one, and (3) exactly one. In an ordered Beta regression, (1) and (2) are modeled with cumulative logits, where one cutpoint is the the boundary between Exactly 0 and Between 0 and 1 and the other cutpoint is the boundary between *Between 0 and 1* and *Exactly 1.* The continuous values in the middle, 0 to 1 (3), are modeled as a vanilla Beta regression with parameters reflecting the mean response on the logit scale as we have described previously. Ultimately, employing cutpoints allows for a smooth transition between the bounds and the continuous values, permitting both to be considered together rather than modeled separately as the ZOIB requires.</span>
<span id="cb3-1498"><a href="#cb3-1498" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1499"><a href="#cb3-1499" aria-hidden="true" tabindex="-1"></a>The ordered Beta regression model has shown to be more efficient and less biased than some of the methods discussed <span class="co">[</span><span class="ot">@kubinec2022</span><span class="co">]</span> herein and has seen increasing use across the biomedical and social sciences <span class="co">[</span><span class="ot">@wilkesQuantifyingCoextinctionsEcosystem2024; @martinVocalComplexitySocially2024; @smithEcologicalMomentaryAssessment2024; @shresthaBigFiveTraits2024; @nouvianGlyphosateImpairsAversive2023</span><span class="co">]</span> because it produces only a single set of coefficient estimates in a similar manner to a standard Beta regression or OLS.</span>
<span id="cb3-1500"><a href="#cb3-1500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1501"><a href="#cb3-1501" aria-hidden="true" tabindex="-1"></a><span class="fu">### Fitting an Ordered Beta Regression</span></span>
<span id="cb3-1502"><a href="#cb3-1502" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1503"><a href="#cb3-1503" aria-hidden="true" tabindex="-1"></a>To fit an ordered Beta regression in a Bayesian context we use the <span class="in">`ordbetareg`</span> <span class="co">[</span><span class="ot">@ordbetareg</span><span class="co">]</span> package. <span class="in">`ordbetareg`</span> is a front-end to the <span class="in">`brms`</span> package that we described earlier; in addition to the functions available in the package, most <span class="in">`brms`</span> functions and plots, including the diverse array of regression modeling options, will work with <span class="in">`ordbetareg`</span> models. (We note that the <span class="in">`ordBeta`</span> model is also available as a maximum-likelihood variant in the R package <span class="in">`glmmTMB`</span>.) We first load the <span class="in">`ordbetareg`</span> package (see @lst-ordBeta).</span>
<span id="cb3-1504"><a href="#cb3-1504" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1507"><a href="#cb3-1507" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1508"><a href="#cb3-1508" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-1509"><a href="#cb3-1509" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-ordBeta</span></span>
<span id="cb3-1510"><a href="#cb3-1510" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Load `ordbetareg`</span></span>
<span id="cb3-1511"><a href="#cb3-1511" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load ordbetareg</span></span>
<span id="cb3-1512"><a href="#cb3-1512" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb3-1513"><a href="#cb3-1513" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1514"><a href="#cb3-1514" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ordbetareg)</span>
<span id="cb3-1515"><a href="#cb3-1515" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1516"><a href="#cb3-1516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1517"><a href="#cb3-1517" aria-hidden="true" tabindex="-1"></a>The <span class="in">`ordbetareg`</span> package uses <span class="in">`brms`</span> on the front-end so all the arguments we used previously apply here. Instead of the <span class="in">`brm()`</span> function we use <span class="in">`ordbetareg()`</span>. To fit a model where dispersion does not vary as a function of fluency we can use the below code (see @lst-fitordBeta).</span>
<span id="cb3-1518"><a href="#cb3-1518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1521"><a href="#cb3-1521" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1522"><a href="#cb3-1522" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: brms-ordered-Beta-model</span></span>
<span id="cb3-1523"><a href="#cb3-1523" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-fitordBeta</span></span>
<span id="cb3-1524"><a href="#cb3-1524" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting ordered Beta model with `ordbetareg()`</span></span>
<span id="cb3-1525"><a href="#cb3-1525" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb3-1526"><a href="#cb3-1526" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-1527"><a href="#cb3-1527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1528"><a href="#cb3-1528" aria-hidden="true" tabindex="-1"></a>ord_fit_brms <span class="ot">&lt;-</span> <span class="fu">ordbetareg</span>(</span>
<span id="cb3-1529"><a href="#cb3-1529" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency,</span>
<span id="cb3-1530"><a href="#cb3-1530" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb3-1531"><a href="#cb3-1531" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"bayes_ordbeta_model"</span></span>
<span id="cb3-1532"><a href="#cb3-1532" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-1533"><a href="#cb3-1533" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1534"><a href="#cb3-1534" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1535"><a href="#cb3-1535" aria-hidden="true" tabindex="-1"></a>However, if we want dispersion to vary as a function of fluency we can easily do that (see @lst-ordBetadisp). Note the addition of the <span class="in">`phi_reg`</span> argument in <span class="in">`m.phi`</span>. This argument allows us to include a model that explicitly models the dispersion parameter. Because we are modeling $\phi$ as a function of fluency, we set the the argument to <span class="in">`both.`</span></span>
<span id="cb3-1536"><a href="#cb3-1536" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1539"><a href="#cb3-1539" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1540"><a href="#cb3-1540" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: brms ordered Beta model phi</span></span>
<span id="cb3-1541"><a href="#cb3-1541" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-ordBetadisp</span></span>
<span id="cb3-1542"><a href="#cb3-1542" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting ordered Beta model with dispersion using `ordbetareg()`</span></span>
<span id="cb3-1543"><a href="#cb3-1543" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb3-1544"><a href="#cb3-1544" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb3-1545"><a href="#cb3-1545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1546"><a href="#cb3-1546" aria-hidden="true" tabindex="-1"></a>ord_beta_phi <span class="ot">&lt;-</span> <span class="fu">bf</span>(Accuracy <span class="sc">~</span> Fluency, phi <span class="sc">~</span> Fluency)</span>
<span id="cb3-1547"><a href="#cb3-1547" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1548"><a href="#cb3-1548" aria-hidden="true" tabindex="-1"></a>m.phi <span class="ot">&lt;-</span> <span class="fu">ordbetareg</span>(</span>
<span id="cb3-1549"><a href="#cb3-1549" aria-hidden="true" tabindex="-1"></a>  ord_beta_phi,</span>
<span id="cb3-1550"><a href="#cb3-1550" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb3-1551"><a href="#cb3-1551" aria-hidden="true" tabindex="-1"></a>  <span class="at">phi_reg =</span> <span class="st">'both'</span>,</span>
<span id="cb3-1552"><a href="#cb3-1552" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"bayes_ordbeta_phi_model"</span></span>
<span id="cb3-1553"><a href="#cb3-1553" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-1554"><a href="#cb3-1554" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1555"><a href="#cb3-1555" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1556"><a href="#cb3-1556" aria-hidden="true" tabindex="-1"></a><span class="fu">## Marginal Effects</span></span>
<span id="cb3-1557"><a href="#cb3-1557" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1558"><a href="#cb3-1558" aria-hidden="true" tabindex="-1"></a>@tbl-all presents the overall model summary (under <span class="in">`Ordered Beta`</span>). We can use <span class="in">`marginaleffects`</span> to calculate differences on the response scale that average over (or marginalize over) all our parameters.</span>
<span id="cb3-1559"><a href="#cb3-1559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1562"><a href="#cb3-1562" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1563"><a href="#cb3-1563" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fit ordBeta</span></span>
<span id="cb3-1564"><a href="#cb3-1564" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1565"><a href="#cb3-1565" aria-hidden="true" tabindex="-1"></a>ord_beta_avg <span class="ot">&lt;-</span> m.phi <span class="sc">|&gt;</span></span>
<span id="cb3-1566"><a href="#cb3-1566" aria-hidden="true" tabindex="-1"></a>  <span class="fu">avg_comparisons</span>(</span>
<span id="cb3-1567"><a href="#cb3-1567" aria-hidden="true" tabindex="-1"></a>    <span class="at">variables =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb3-1568"><a href="#cb3-1568" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb3-1569"><a href="#cb3-1569" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb3-1570"><a href="#cb3-1570" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1571"><a href="#cb3-1571" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1574"><a href="#cb3-1574" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1575"><a href="#cb3-1575" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-ordBeta</span></span>
<span id="cb3-1576"><a href="#cb3-1576" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Marginal effect for fluency in ordered Beta model"</span></span>
<span id="cb3-1577"><a href="#cb3-1577" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-1578"><a href="#cb3-1578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1579"><a href="#cb3-1579" aria-hidden="true" tabindex="-1"></a>ord_beta_avg <span class="sc">|&gt;</span></span>
<span id="cb3-1580"><a href="#cb3-1580" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(term, contrast, estimate, conf.low, conf.high) <span class="sc">|&gt;</span></span>
<span id="cb3-1581"><a href="#cb3-1581" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb3-1582"><a href="#cb3-1582" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb3-1583"><a href="#cb3-1583" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{conf.low}, {conf.high}]"</span>)</span>
<span id="cb3-1584"><a href="#cb3-1584" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb3-1585"><a href="#cb3-1585" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Contrast"</span> <span class="ot">=</span> <span class="st">"contrast"</span>, <span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"estimate"</span>, <span class="st">"Term"</span> <span class="ot">=</span> <span class="st">"term"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1586"><a href="#cb3-1586" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Term, Contrast, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1587"><a href="#cb3-1587" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-1588"><a href="#cb3-1588" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1589"><a href="#cb3-1589" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-1590"><a href="#cb3-1590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1591"><a href="#cb3-1591" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1592"><a href="#cb3-1592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1593"><a href="#cb3-1593" aria-hidden="true" tabindex="-1"></a>In @tbl-ordBeta the credible interval is close enough to zero relative to its uncertainty that we can conclude there likely aren’t substantial differences between the conditions after taking dispersion and the 0s and 1s in our data into account.</span>
<span id="cb3-1594"><a href="#cb3-1594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1595"><a href="#cb3-1595" aria-hidden="true" tabindex="-1"></a><span class="fu">### Cutpoints</span></span>
<span id="cb3-1596"><a href="#cb3-1596" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1597"><a href="#cb3-1597" aria-hidden="true" tabindex="-1"></a>The model cutpoints are not reported by default in the summary output, but we can access them with the R package <span class="in">`posterior`</span> <span class="co">[</span><span class="ot">@posterior2025a</span><span class="co">]</span> and the functions <span class="in">`as_draws`</span> and <span class="in">`summary_draws`</span>.</span>
<span id="cb3-1598"><a href="#cb3-1598" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1601"><a href="#cb3-1601" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1602"><a href="#cb3-1602" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-cutpoints</span></span>
<span id="cb3-1603"><a href="#cb3-1603" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Cutzero and cutone parameter summary</span></span>
<span id="cb3-1604"><a href="#cb3-1604" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb3-1605"><a href="#cb3-1605" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb3-1606"><a href="#cb3-1606" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1607"><a href="#cb3-1607" aria-hidden="true" tabindex="-1"></a><span class="fu">as_draws</span>(m.phi, <span class="fu">c</span>(<span class="st">"cutzero"</span>, <span class="st">"cutone"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb3-1608"><a href="#cb3-1608" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize_draws</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-1609"><a href="#cb3-1609" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(variable, mean, q5, q95) <span class="sc">|&gt;</span></span>
<span id="cb3-1610"><a href="#cb3-1610" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(mean, q5, q95), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">2</span>))) <span class="sc">|&gt;</span></span>
<span id="cb3-1611"><a href="#cb3-1611" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{q5}, {q95}]"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb3-1612"><a href="#cb3-1612" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Parameter"</span> <span class="ot">=</span> <span class="st">"variable"</span>, <span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"mean"</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1613"><a href="#cb3-1613" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Parameter, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb3-1614"><a href="#cb3-1614" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb3-1615"><a href="#cb3-1615" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb3-1616"><a href="#cb3-1616" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1617"><a href="#cb3-1617" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1618"><a href="#cb3-1618" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1619"><a href="#cb3-1619" aria-hidden="true" tabindex="-1"></a>In @tbl-cutpoints, <span class="in">`cutzero`</span> is the first cutpoint (the difference between 0 and continuous values) and <span class="in">`cutone`</span> is the second cutpoint (the difference between the continuous values and 1). These cutpoints are on the logit scale and as such the numbers do not have a simple substantive meaning. In general, as the cutpoints increase in absolute value (away from zero), then the discrete/boundary observations are more distinct from the continuous values. This will happen if there is a clear gap or bunching in the outcome around the bounds. This type of empirical feature of the distribution may be useful to scholars if they want to study differences in how people perceive the ends of the scale versus the middle. It is possible, though beyond the scope of this article, to model the location of the cutpoints with hierarchical (non-linear) covariates in <span class="in">`brms`</span>.</span>
<span id="cb3-1620"><a href="#cb3-1620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1621"><a href="#cb3-1621" aria-hidden="true" tabindex="-1"></a><span class="fu">### Model Fit</span></span>
<span id="cb3-1622"><a href="#cb3-1622" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1623"><a href="#cb3-1623" aria-hidden="true" tabindex="-1"></a>The best way to visualize model fit is to plot the full predictive distribution relative to the original outcome. Because ordered Beta regression is a mixed discrete/continuous model, a separate plotting function, <span class="in">`pp_check_ordBetareg`</span>, is included in the <span class="in">`ordbetareg`</span> package that accurately handles the unique features of this distribution. The default plot in <span class="in">`brms`</span> will collapse these two features of the outcome together, which will make the fit look worse than it actually is. The <span class="in">`ordbetareg`</span> function returns a list with two plots, <span class="in">`discrete`</span> and <span class="in">`continuous`</span>, which can either be printed and plotted or further modified as <span class="in">`ggplot2`</span> objects (see @fig-ppcheckord).</span>
<span id="cb3-1624"><a href="#cb3-1624" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1627"><a href="#cb3-1627" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1628"><a href="#cb3-1628" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ppcheckord</span></span>
<span id="cb3-1629"><a href="#cb3-1629" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 12</span></span>
<span id="cb3-1630"><a href="#cb3-1630" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb3-1631"><a href="#cb3-1631" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Posterior predictive check for ordered Beta regression model. A. Discrete posterior check. B. Continuous posterior check.</span></span>
<span id="cb3-1632"><a href="#cb3-1632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1633"><a href="#cb3-1633" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">pp_check_ordbeta</span>(</span>
<span id="cb3-1634"><a href="#cb3-1634" aria-hidden="true" tabindex="-1"></a>  m.phi,</span>
<span id="cb3-1635"><a href="#cb3-1635" aria-hidden="true" tabindex="-1"></a>  <span class="at">ndraws =</span> <span class="dv">100</span>,</span>
<span id="cb3-1636"><a href="#cb3-1636" aria-hidden="true" tabindex="-1"></a>  <span class="at">outcome_label =</span> <span class="st">"Final Test Accuracy"</span></span>
<span id="cb3-1637"><a href="#cb3-1637" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-1638"><a href="#cb3-1638" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1639"><a href="#cb3-1639" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plots<span class="sc">$</span>discrete, plots<span class="sc">$</span>continuous, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>))</span>
<span id="cb3-1640"><a href="#cb3-1640" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1641"><a href="#cb3-1641" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1642"><a href="#cb3-1642" aria-hidden="true" tabindex="-1"></a>The discrete plot, which is a bar graph, shows that the posterior distribution accurately captures the number of different types of responses (discrete or continuous) in the data. For the continuous plot shown as a density plot with one line per posterior draw, the model does a very good job at capturing the distribution.</span>
<span id="cb3-1643"><a href="#cb3-1643" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1644"><a href="#cb3-1644" aria-hidden="true" tabindex="-1"></a>Overall, it is clear from the posterior distribution plot that the ordered Beta model fits the data well. To fully understand model fit, both of these plots need to be inspected as they are conceptually distinct.</span>
<span id="cb3-1645"><a href="#cb3-1645" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1646"><a href="#cb3-1646" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Model Visualization</span></span>
<span id="cb3-1647"><a href="#cb3-1647" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1648"><a href="#cb3-1648" aria-hidden="true" tabindex="-1"></a><span class="in">`ordbetareg`</span> provides a neat visualization function called <span class="in">`plot_heiss()`</span> <span class="co">[</span><span class="ot">@yeEnforcingBoundariesChinas2023</span><span class="co">]</span> that can represent dispersion in the entire outcome as a function of discrete covariates. This function produces a plot of predicted proportions across the range of our <span class="in">`Fluency`</span> factor. In @fig-heiss we get predicted proportions for Fluency across the bounded scale. Looking at the figure we can see there is much overlap between instructors in the middle portion ($\mu$) . However, we do see some small differences at the zero bounds.</span>
<span id="cb3-1649"><a href="#cb3-1649" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1652"><a href="#cb3-1652" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb3-1653"><a href="#cb3-1653" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-heiss</span></span>
<span id="cb3-1654"><a href="#cb3-1654" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Heiss plot of predicted probabilities across the scale (0-100)"</span></span>
<span id="cb3-1655"><a href="#cb3-1655" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1656"><a href="#cb3-1656" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_heiss</span>(</span>
<span id="cb3-1657"><a href="#cb3-1657" aria-hidden="true" tabindex="-1"></a>  ord_fit_brms,</span>
<span id="cb3-1658"><a href="#cb3-1658" aria-hidden="true" tabindex="-1"></a>  <span class="at">grouping_fac =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb3-1659"><a href="#cb3-1659" aria-hidden="true" tabindex="-1"></a>  <span class="at">recode_group_labels =</span> <span class="fu">c</span>(<span class="st">"Fluency"</span>, <span class="st">"Disfluency"</span>)</span>
<span id="cb3-1660"><a href="#cb3-1660" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb3-1661"><a href="#cb3-1661" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb3-1662"><a href="#cb3-1662" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1663"><a href="#cb3-1663" aria-hidden="true" tabindex="-1"></a><span class="fu">### Ordered Beta Scale</span></span>
<span id="cb3-1664"><a href="#cb3-1664" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1665"><a href="#cb3-1665" aria-hidden="true" tabindex="-1"></a>In the <span class="in">`ordbetareg`</span> function there is a <span class="in">`true_bound`</span> argument. In the case where you data in not bounded between 0-1, you can use the argument to specify the bounds of the argument to fit the ordered Beta regression. For example, you data might be bounded between 1 and 7. If so, you can model it as such and <span class="in">`ordbetareg`</span> will convert the model predictions back to the true bounds after estimation.</span>
<span id="cb3-1666"><a href="#cb3-1666" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1667"><a href="#cb3-1667" aria-hidden="true" tabindex="-1"></a><span class="fu"># Discussion</span></span>
<span id="cb3-1668"><a href="#cb3-1668" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1669"><a href="#cb3-1669" aria-hidden="true" tabindex="-1"></a>The use of Beta regression in psychology, and the social sciences in general, is rare. With this tutorial, we hope to turn the tides. Beta regression models are an attractive alternative to models that impose unrealistic assumptions like normality, linearity, homoscedasticity, and unbounded data. Beyond these models, there are a diverse array of different models that can be used depending on your outcome of interest.</span>
<span id="cb3-1670"><a href="#cb3-1670" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1671"><a href="#cb3-1671" aria-hidden="true" tabindex="-1"></a>Throughout this tutorial our main aim was to help guide researchers in running analyses with proportional or percentage outcomes using Beta regression and some of its alternatives. In the current example, we used real data from @wilford2020 and discussed how to fit these models in R, interpret model parameters, extract predicted probabilities and marginal effects, and visualize the results.</span>
<span id="cb3-1672"><a href="#cb3-1672" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1673"><a href="#cb3-1673" aria-hidden="true" tabindex="-1"></a>Comparing our analysis with that of @wilford2020, we demonstrated that using traditional approaches (e.g., t-tests) to analyze accuracy data can lead to inaccurate inferences. Although we successfully reproduced one of their key findings, our use of Beta regression and its extensions revealed important nuances in the results. With a traditional Beta regression model–which accounts for both the mean and the precision (dispersion)–we observed similar effects of instructor fluency on performance. However, the standard Beta model does not accommodate boundary values (i.e., 0s and 1s).</span>
<span id="cb3-1674"><a href="#cb3-1674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1675"><a href="#cb3-1675" aria-hidden="true" tabindex="-1"></a>When we applied a ZIB model, which explicitly accounts for structural 0s, we found no effect of fluency on the mean ($\mu$) part of the model. Instead, the effect of fluency emerged in the structural zero (inflated zero; $\alpha$) component. This pattern was consistent when using a zero-one-inflated Beta (ZOIB) model. Furthermore, we fit an ordered Beta regression model <span class="co">[</span><span class="ot">@kubinec2022</span><span class="co">]</span>, which appropriately models the full range of values, including 0s and 1s. Here, we did not observe a reliable effect of fluency on the mean once we accounted for dispersion.</span>
<span id="cb3-1676"><a href="#cb3-1676" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1677"><a href="#cb3-1677" aria-hidden="true" tabindex="-1"></a>These analyses emphasize the importance of fitting a model that aligns with the nature of the data. The simplest and recommended approach when dealing with data that contains 0s and/or 1s is to fit an ordered Beta model, assuming the process is truly continuous. However, if you believe the process is distinct in nature, a ZIB or ZOIB model might be a better choice. Ultimately, this decision should be guided by theory.</span>
<span id="cb3-1678"><a href="#cb3-1678" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1679"><a href="#cb3-1679" aria-hidden="true" tabindex="-1"></a>For instance, if we believe fluency influences the structural zero part of the model, we might want to model this process separately using a ZIB or ZOIB. With the current dataset, fluency might affect specific aspects of performance (such as the likelihood of complete failure) rather than general performance levels. This effect could be due to participant disengagement during the disfluent lecture. If students fail to pay attention because of features of disfluency, they may miss relevant information, leading to a floor effect at the test. If this is the case, we would want to model this appropriately. However, if we believe fluency effects general performance levels, a model that takes in to account the entire process accounting for the 0s and 1s might be appropriate.</span>
<span id="cb3-1680"><a href="#cb3-1680" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1681"><a href="#cb3-1681" aria-hidden="true" tabindex="-1"></a>In the discussion section of @wilford2020, they were unable to offer a tenable explanation for performance differences based on instructor fluency. A model that accounts for the excess 0s in the dataset provides one testable explanation: watching a disfluent lecture may lead to lapses in attention, resulting in poorer performance in that group. These lapses, in turn, contribute to the observed differences in the fluent condition. This modeling approach opens a promising avenue for future research--one that would have remained inaccessible otherwise.</span>
<span id="cb3-1682"><a href="#cb3-1682" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1683"><a href="#cb3-1683" aria-hidden="true" tabindex="-1"></a>Not everyone will be eager to implement the techniques discussed herein. In such cases, the key question becomes: What is the least problematic approach to handling proportional data? One reasonable option is to fit multiple models tailored to the specific characteristics of your data. For example, if your data contain 0s, you might fit two models: a traditional linear model excluding the 0s, and a logistic model to account for the zero versus non-zero distinction. If your data contain both 0s and 1s, you could fit separate models for the 0s and 1s in addition to the OLS model. There are many defensible strategies to choose from depending on the context. However, we do not recommend transforming the values of your data (e.g., 0s to .01 and 1s to .99) or ignoring the properties of your data simply to fit traditional statistical models.</span>
<span id="cb3-1684"><a href="#cb3-1684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1685"><a href="#cb3-1685" aria-hidden="true" tabindex="-1"></a>In this tutorial, we demonstrated how to analyze these models from a Bayesian perspective. While we recognize that not everyone identifies as a Bayesian, implementing these models using a Bayesian framework is relatively straightforward--it requires only a single package, lowering the barrier to entry. For those who prefer frequentist analyses, several R packages are available. For standard Beta regression, the <span class="in">`betareg`</span> package <span class="co">[</span><span class="ot">@betareg</span><span class="co">]</span> is a solid option, while more complex models such as zero-inflated and ordered Beta regressions can be implemented using <span class="in">`glmmTMB`</span> <span class="co">[</span><span class="ot">@glmmTMB</span><span class="co">]</span>. For fitting zero-one models, there are is a new implementation in the <span class="in">`betareg`</span> package <span class="co">[</span><span class="ot">@betareg</span><span class="co">]</span> called the XBX regression model <span class="co">[</span><span class="ot">@kosmidis2025</span><span class="co">]</span> that allows you to model these types of data.</span>
<span id="cb3-1686"><a href="#cb3-1686" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1687"><a href="#cb3-1687" aria-hidden="true" tabindex="-1"></a><span class="fu">## Conclusion</span></span>
<span id="cb3-1688"><a href="#cb3-1688" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1689"><a href="#cb3-1689" aria-hidden="true" tabindex="-1"></a>Overall, this tutorial emphasizes the importance of modeling the data you have. Although the example provided is relatively simple (a one-factor model with two levels), we hope it demonstrates that even with a basic dataset, there is much nuance in interpretation and inference. Properly modeling your data can lead to deeper insights, far beyond what traditional measures might offer. With the tools introduced in this tutorial, researchers now have the means to analyze their data effectively, uncover patterns, make accurate predictions, and support their findings with robust statistical evidence. By applying these modeling techniques, researchers can improve the validity and reliability of their studies, ultimately leading to more informed decisions and advancements in their respective fields.</span>
<span id="cb3-1690"><a href="#cb3-1690" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1691"><a href="#cb3-1691" aria-hidden="true" tabindex="-1"></a><span class="fu"># References</span></span>
<span id="cb3-1692"><a href="#cb3-1692" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-1693"><a href="#cb3-1693" aria-hidden="true" tabindex="-1"></a>::: {#refs}</span>
<span id="cb3-1694"><a href="#cb3-1694" aria-hidden="true" tabindex="-1"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>