<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.6.43">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jason Geller">
<meta name="author" content="Robert Kubinec">
<meta name="author" content="Chelsea M. Parlett Pelleriti">
<meta name="author" content="Matti Vuorre">
<meta name="keywords" content="beta regression, beta distribution, R, tutorial, psychology, learning and memory">
<meta name="description" content="BETA REGRESSION TUTORIAL">

<title>A Beta Way: A Tutorial For Using Beta Regression in Psychological Research</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="ms_files/libs/clipboard/clipboard.min.js"></script>
<script src="ms_files/libs/quarto-html/quarto.js"></script>
<script src="ms_files/libs/quarto-html/popper.min.js"></script>
<script src="ms_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="ms_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="ms_files/libs/quarto-html/quarto-syntax-highlighting-d4d76bf8491c20bad77d141916dc28e1.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="ms_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="ms_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="ms_files/libs/bootstrap/bootstrap-e2b52ab01165ad98e739bac1fbc1eb77.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">

  <script>window.backupDefine = window.define; window.define = undefined;</script><script src="https://cdn.jsdelivr.net/npm/katex@latest/dist/katex.min.js"></script>
  <script>document.addEventListener("DOMContentLoaded", function () {
 var mathElements = document.getElementsByClassName("math");
 var macros = [];
 for (var i = 0; i < mathElements.length; i++) {
  var texText = mathElements[i].firstChild;
  if (mathElements[i].tagName == "SPAN") {
   katex.render(texText.data, mathElements[i], {
    displayMode: mathElements[i].classList.contains('display'),
    throwOnError: false,
    macros: macros,
    fleqn: false
   });
}}});
  </script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@latest/dist/katex.min.css">

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="_extensions/wjschne/apaquarto/apa.css">
</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#motivating-example" id="toc-motivating-example" class="nav-link active" data-scroll-target="#motivating-example">Motivating Example</a>
  <ul class="collapse">
  <li><a href="#data-and-methods" id="toc-data-and-methods" class="nav-link" data-scroll-target="#data-and-methods">Data and Methods</a></li>
  <li><a href="#reanalysis-of-wilford-et-al.-experiment-1a" id="toc-reanalysis-of-wilford-et-al.-experiment-1a" class="nav-link" data-scroll-target="#reanalysis-of-wilford-et-al.-experiment-1a">Reanalysis of Wilford et al.&nbsp;Experiment 1A</a>
  <ul class="collapse">
  <li><a href="#data" id="toc-data" class="nav-link" data-scroll-target="#data">Data</a></li>
  </ul></li>
  <li><a href="#bayesian-regression" id="toc-bayesian-regression" class="nav-link" data-scroll-target="#bayesian-regression">Bayesian Regression</a></li>
  <li><a href="#beta-regression-1" id="toc-beta-regression-1" class="nav-link" data-scroll-target="#beta-regression-1">Beta Regression</a>
  <ul class="collapse">
  <li><a href="#mean-mu" id="toc-mean-mu" class="nav-link" data-scroll-target="#mean-mu">Mean <span class="math inline">\mu</span></a></li>
  <li><a href="#precision-phi-component" id="toc-precision-phi-component" class="nav-link" data-scroll-target="#precision-phi-component">Precision (<span class="math inline">\phi</span>) Component</a></li>
  <li><a href="#posterior-predictive-check" id="toc-posterior-predictive-check" class="nav-link" data-scroll-target="#posterior-predictive-check">Posterior Predictive Check</a></li>
  </ul></li>
  <li><a href="#zero-inflated-beta-zib-regression" id="toc-zero-inflated-beta-zib-regression" class="nav-link" data-scroll-target="#zero-inflated-beta-zib-regression">Zero-Inflated Beta (ZIB) Regression</a>
  <ul class="collapse">
  <li><a href="#predicted-probabilities-and-marginal-effects" id="toc-predicted-probabilities-and-marginal-effects" class="nav-link" data-scroll-target="#predicted-probabilities-and-marginal-effects">Predicted Probabilities and Marginal Effects</a></li>
  <li><a href="#zero-inflated-alpha" id="toc-zero-inflated-alpha" class="nav-link" data-scroll-target="#zero-inflated-alpha">Zero-Inflated (<span class="math inline">\alpha</span>)</a></li>
  </ul></li>
  <li><a href="#zero-one-inflated-beta-zoib" id="toc-zero-one-inflated-beta-zoib" class="nav-link" data-scroll-target="#zero-one-inflated-beta-zoib">Zero-One-Inflated Beta (ZOIB)</a>
  <ul class="collapse">
  <li><a href="#model-parameters-1" id="toc-model-parameters-1" class="nav-link" data-scroll-target="#model-parameters-1">Model Parameters</a></li>
  <li><a href="#predictions-and-marginal-effects" id="toc-predictions-and-marginal-effects" class="nav-link" data-scroll-target="#predictions-and-marginal-effects">Predictions and Marginal Effects</a></li>
  <li><a href="#gamma" id="toc-gamma" class="nav-link" data-scroll-target="#gamma"><span class="math inline">\gamma</span></a></li>
  </ul></li>
  <li><a href="#ordered-beta-regression" id="toc-ordered-beta-regression" class="nav-link" data-scroll-target="#ordered-beta-regression">Ordered Beta Regression</a>
  <ul class="collapse">
  <li><a href="#fitting-an-ordered-beta-regression" id="toc-fitting-an-ordered-beta-regression" class="nav-link" data-scroll-target="#fitting-an-ordered-beta-regression">Fitting an Ordered Beta Regression</a></li>
  </ul></li>
  <li><a href="#marginal-effects-1" id="toc-marginal-effects-1" class="nav-link" data-scroll-target="#marginal-effects-1">Marginal effects</a>
  <ul class="collapse">
  <li><a href="#cutpoints" id="toc-cutpoints" class="nav-link" data-scroll-target="#cutpoints">Cutpoints</a></li>
  <li><a href="#ordered-beta-regression-model-fit" id="toc-ordered-beta-regression-model-fit" class="nav-link" data-scroll-target="#ordered-beta-regression-model-fit">Ordered Beta Regression Model Fit</a></li>
  <li><a href="#ordered-beta-scale" id="toc-ordered-beta-scale" class="nav-link" data-scroll-target="#ordered-beta-scale">Ordered Beta Scale</a></li>
  </ul></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion">Discussion</a>
  <ul class="collapse">
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion">Conclusion</a></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="../ms.pdf"><i class="bi bi-file-pdf"></i>PDF (apaquarto)</a></li><li><a href="ms.docx"><i class="bi bi-file-word"></i>MS Word (apaquarto)</a></li></ul></div></nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content" id="quarto-document-content">




<br>

<br>

<section id="title" class="level1 title unnumbered unlisted">
<h1 class="title unnumbered unlisted">A Beta Way: A Tutorial For Using Beta Regression in Psychological Research</h1>
<div class="Author">
<br>

<p>Jason Geller<sup>1</sup>, Robert Kubinec<sup>2</sup>, Chelsea M. Parlett Pelleriti<sup>3</sup>, and Matti Vuorre<sup>4</sup></p>
<p><sup>1</sup>Department of Psychology and Neuroscience, Boston College</p>
<p><sup>2</sup>University of South Carolina</p>
<p><sup>3</sup>School</p>
<p><sup>4</sup>Tilburg University</p>
</div>
<br>

<br>

</section>
<section id="author-note" class="level1 unnumbered unlisted AuthorNote">
<h1 class="unnumbered unlisted AuthorNote">Author Note</h1>
<p>Jason Geller <img src="_extensions/wjschne/apaquarto/ORCID-iD_icon-vector.svg" id="orchid" class="img-fluid" style="width:4.23mm" alt="Orcid ID Logo: A green circle with white letters ID"> <a href="https://orcid.org/0000-0002-7459-4505">https://orcid.org/0000-0002-7459-4505</a></p>
<p>Robert Kubinec <img src="_extensions/wjschne/apaquarto/ORCID-iD_icon-vector.svg" id="orchid" class="img-fluid" style="width:4.23mm" alt="Orcid ID Logo: A green circle with white letters ID"> <a href="https://orcid.org/0000-0001-6655-4119">https://orcid.org/0000-0001-6655-4119</a></p>
<p>Chelsea M. Parlett Pelleriti <img src="_extensions/wjschne/apaquarto/ORCID-iD_icon-vector.svg" id="orchid" class="img-fluid" style="width:4.23mm" alt="Orcid ID Logo: A green circle with white letters ID"> <a href="https://orcid.org/0000-0001-9301-1398">https://orcid.org/0000-0001-9301-1398</a></p>
<p>Matti Vuorre <img src="_extensions/wjschne/apaquarto/ORCID-iD_icon-vector.svg" id="orchid" class="img-fluid" style="width:4.23mm" alt="Orcid ID Logo: A green circle with white letters ID"> <a href="https://orcid.org/0000-0001-5052-066X">https://orcid.org/0000-0001-5052-066X</a></p>
<p>No preregistration for this paper. Data, code, and materials for this manusscript can be found at. The authors have no conflicts of interest to disclose.</p>
<p>Author roles were classified using the Contributor Role Taxonomy (CRediT; https://credit.niso.org/) as follows: <em>Jason Geller</em><strong>: </strong>conceptualization, writing – original draft, data curation, writing – review &amp; editing, and formal analysis. <em>Robert Kubinec</em><strong>: </strong>writing – original draft, writing – review &amp; editing, and formal analysis. <em>Chelsea M. Parlett Pelleriti</em><strong>: </strong>writing – original draft, writing – review &amp; editing, and formal analysis. <em>Matti Vuorre</em><strong>: </strong>writing – original draft, writing – review &amp; editing, and formal analysis</p>
<p>Correspondence concerning this article should be addressed to Jason Geller, Department of Psychology and Neuroscience, Boston College, McGuinn 300, Chestnut Hill, MA 2467, USA, Email: <a href="mailto:drjasongeller@gmail.com">drjasongeller@gmail.com</a></p>
<br>
</section>
<section id="abstract" class="level1 unnumbered unlisted AuthorNote">
<h1 class="unnumbered unlisted AuthorNote">Abstract</h1>
<div class="AbstractFirstParagraph">
<p>Rates, percentages, and proportions are common outcomes in psychology and the social sciences. These outcomes are often analyzed using models that assume normality, but this practice overlooks important features of the data, such as their natural bounds at 0 and 1. As a result, estimates can become distorted. In contrast, treating such outcomes as beta-distributed respects these limits and can yield more accurate estimates. Despite these advantages, the use of beta models in applied research remains limited. Our goal is to provide researchers with practical guidance for adopting beta regression models, illustrated with an example drawn from the psychological literature. We begin by introducing the beta distribution and beta regression, emphasizing key components and assumptions. Next, using data from a learning and memory study, we demonstrate how to fit a Beta regression model in R with the Bayesian package <code>brms</code> and how to interpret results on the response scale. We also discuss model extensions, including zero-inflated, zero- and one-inflated, and ordered beta models. To promote wider adoption of these methods, we provide detailed code and materials at https://github.com/jgeller112/beta_regression_tutorial.</p>
</div>
<p><em>Keywords</em>: beta regression, beta distribution, R, tutorial, psychology, learning and memory</p>
<p><em>Word Count</em>: 9773</p>
<br>
</section>
<section id="firstheader" class="level1 title unnumbered unlisted">
<h1 class="title unnumbered unlisted">A Beta Way: A Tutorial For Using Beta Regression in Psychological Research</h1>
<p>Many important outcomes in psychological research can be expressed as proportions or percentages: the proportion of correct responses on a test, the proportion of looks on a particular stimulus in an eye-tracking task, or the percentage of agreement with a given statement. As a practical example from educational and cognitive research, one popular way to assess learning is by looking at the proportion of correct responses on a test. To illustrate, consider a memory experiment where participants read a short passage on a specific topic. After a brief distractor task, they complete a final memory test consisting of 10 short-answer questions, each assigned a different point value (e.g., question 1 might be worth 4 points, while question 2 might be worth 1 point). The primary outcome measure could be the proportion of points earned on each question relative to the total possible points for each question.</p>
<p>A key question arises: how should proportional data like this be analyzed? In psychology, such outcomes are often analyzed using models from the general linear model (GLM) or generalized linear model (GLiM) frameworks. GLMs — including t-tests, ANOVAs, and linear regression — assume the outcome is normally distributed, unbounded, and exhibits constant variance. However, these assumptions are frequently violated when working with proportional data, which are bounded between 0 and 1 and tend to show non-constant variance, especially near the boundaries <span class="citation" data-cites="smithson2006 ferrari2004 paolino2001">(<a href="#ref-ferrari2004" role="doc-biblioref">Ferrari &amp; Cribari-Neto, 2004</a>; <a href="#ref-paolino2001" role="doc-biblioref">Paolino, 2001</a>; <a href="#ref-smithson2006" role="doc-biblioref">Smithson &amp; Verkuilen, 2006</a>)</span>. This can lead to biased estimates and invalid inferences, making general linear models an unsuitable choice for analyzing proportions.</p>
<p>Another option are GLiMs, which extend the linear model framework to accommodate non-normal outcome distributions and different link functions. For example, binomial or Bernoulli models with a logit link (commonly referred to as logistic regression) are well-suited for binary outcomes or counts of successes out of a fixed number of trials. However, these models may still fall short when data exhibit overdispersion or cluster near 0 or 1.</p>
<p>The challenges of analyzing proportional data are not new <span class="citation" data-cites="bartlett1936">(see <a href="#ref-bartlett1936" role="doc-biblioref">Bartlett, 1936</a>)</span>. Fortunately, several alternative approaches address the limitations of commonly used models. One such approach is Beta regression, an extension of the GLiM that employs the Beta distribution (described in-depth below) <span class="citation" data-cites="ferrari2004 paolino2001">(<a href="#ref-ferrari2004" role="doc-biblioref">Ferrari &amp; Cribari-Neto, 2004</a>; <a href="#ref-paolino2001" role="doc-biblioref">Paolino, 2001</a>)</span>. Beta regression offers a flexible and robust solution for modeling proportional data by accounting for boundary effects and overdispersion, making it a valuable alternative to traditional binomial models. This approach is particularly well-suited for psychological research because it can handle both the bounded nature of proportional data and the non-constant variance often encountered in these datasets.</p>
<section id="a-beta-way-forward" class="level2">
<h2 data-anchor-id="a-beta-way-forward">A Beta Way Forward</h2>
<p>With the combination of open-source programming languages like R <span class="citation" data-cites="R">(<a href="#ref-R" role="doc-biblioref">R Core Team, 2024</a>)</span> and a vibrant community of package developers, analyses such as Beta regression have become increasingly accessible. Yet, adoption of these methods—particularly in psychology—remains limited. One reason may be the lack of resources tailored to the needs of psychologists. Although recent years have seen a surge of interest in Beta regression with excellent tutorials and guides <span class="citation" data-cites="smithson2006 heiss2021 vuorre2019 bendixen2023 coretta2025bayesian">(<a href="#ref-bendixen2023" role="doc-biblioref">Bendixen &amp; Purzycki, 2023</a>; <a href="#ref-coretta2025bayesian" role="doc-biblioref">Coretta &amp; Bürkner, 2025</a>; <a href="#ref-heiss2021" role="doc-biblioref">Heiss, 2021</a>; <a href="#ref-smithson2006" role="doc-biblioref">Smithson &amp; Verkuilen, 2006</a>; <a href="#ref-vuorre2019" role="doc-biblioref">Vuorre, 2019</a>)</span>, its adoption in psychology remains limited.</p>
<p>While previous tutorials have introduced Beta regression, most have been limited in scope—focusing either on the basic model or offering only brief mentions of more complex alternatives. This tutorial aims to fill that gap by offering a comprehensive and practical overview of Beta regression and its extensions. In addition to covering the standard Beta model, we walk through advanced variants such as zero-inflated, zero-one-inflated, and ordered Beta regression. These models are critical for researchers dealing with boundary values (e.g., exact 0s or 1s) or ordinal response structures, yet they remain underutilized in psychology.</p>
<p>Beyond model specification, we place strong emphasis on interpreting results on the response scale—that is, in terms of predicted probabilities and proportions—rather than relying solely on log-odds or latent parameters. This focus makes the models more accessible and meaningful for psychological applications, where effect sizes are often easier to communicate when framed in terms of expected outcomes (e.g., changes in recall accuracy or task performance). Throughout, we provide reproducible code and annotated examples to help readers implement and interpret these models in their own work</p>
<p>In this tutorial, we provide a comprehensive and accessible introduction to Beta regression for psychological research. We begin with a non-technical overview of the Beta distribution and its core parameters, laying the foundation for understanding how the model works. We then walk through the process of fitting Beta regression models using the R package <code>brms</code>, illustrating each step with applied examples. To help researchers make sense of model output, we offer guidance on interpreting coefficients, predicted probabilities, and marginal effects using the <code>marginaleffects</code> package <span class="citation" data-cites="margeffects2024">(<a href="#ref-margeffects2024" role="doc-biblioref">Arel-Bundock et al., 2024</a>)</span>. We also introduce several useful extensions—such as zero-inflated (ZIB), zero-one-inflated (ZOIB), and ordered Beta regression models—that enable researchers to model outcomes that include exact boundary values like 0 and 1. Finally, all code and materials used in this tutorial are fully reproducible and available via our GitHub repository: <a href="https://github.com/jgeller112/beta_regression_tutorial" class="uri">https://github.com/jgeller112/beta_regression_tutorial</a><a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a></p>
</section>
<section id="beta-distribution" class="level2">
<h2 data-anchor-id="beta-distribution">Beta Distribution</h2>
<p>Proportional data pose some challenges for standard modeling approaches. That is, the data are bounded between 0 and 1 and often exhibit non-constant variance (heteroscedasticity) <span class="citation" data-cites="ferrari2004 paolino2001">(<a href="#ref-ferrari2004" role="doc-biblioref">Ferrari &amp; Cribari-Neto, 2004</a>; <a href="#ref-paolino2001" role="doc-biblioref">Paolino, 2001</a>)</span>. Common distributions used within the GLM or GLiM frameworks often fail to capture these properties adequately, which can necessitate alternative modeling strategies.</p>
<p>In any statistical model, the expected value (or mean) of the response variable serves as the central estimand we aim to describe and predict. The model specifies how this expected value depends on explanatory variables through several components: a random component that specifies the distribution of the response variable around its expected value (such as a Poisson or binomial, which are part of the exponential family), a linear predictor that combines explanatory variables in a linear form, and a link function that connects the mean of the response variable to the linear predictor (also referred to as the expected value) <span class="citation" data-cites="nelder1972">(<a href="#ref-nelder1972" role="doc-biblioref">Nelder &amp; Wedderburn, 1972</a>)</span>. Together, these components provide a flexible framework for modeling data with different distributional properties.</p>
<p>The Beta distribution offers a compelling solution <span class="citation" data-cites="paolino2001">(<a href="#ref-paolino2001" role="doc-biblioref">Paolino, 2001</a>)</span>. The Beta distribution is continuous, restricted to values between 0 and 1 (exclusive), and highly flexible. Its two shape parameters—commonly called shape1 (<span class="math inline">\alpha</span>) and shape2 (<span class="math inline">\beta</span>)—govern the distribution’s location, skewness, and spread. By adjusting its two shape parameters it can take on many functional forms (e.g., it can be symmetric, skewed, U-shaped, or even approximately uniform; see <a href="#fig-curves" class="quarto-xref" aria-expanded="false">Figure&nbsp;1</a>).</p>
<p>To see this intuitively, consider a test question worth six points. Suppose a participant scores four out of six. The number of points received (4) can be treated as <span class="math inline">\alpha</span>, and the number of points missed (2) as <span class="math inline">\beta</span>. The resulting Beta distribution would be skewed toward higher values, reflecting a high performance. Reversing these values would produce a distribution skewed toward lower values, representing poorer performance (see <a href="#fig-beta-dist" class="quarto-xref" aria-expanded="false">Figure&nbsp;2</a>). These parameterizations offer a natural way to visualize performance data as probability densities.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-curves" class="quarto-float quarto-figure quarto-figure-center" data-fignum="1" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-curves-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;1</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Beta distributions with different shape1 and shape2 parameters</p>
</div>
</figcaption>
<div aria-describedby="fig-curves-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-curves-1.png" class="img-fluid figure-img" width="768">
</div>
</figure>
</div>
</div>
</div>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-beta-dist" class="quarto-float quarto-figure quarto-figure-center" data-fignum="2" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-beta-dist-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;2</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>A. beta distribution with 4 correct (<code>shape1</code>) and 2 incorrect (<code>shape2</code>) responses (<code>shape2</code>) on one test question. B. beta distribution with 2 correct (<code>shape1</code>) and 4 incorrect (<code>shape2</code>) responses on one test question</p>
</div>
</figcaption>
<div aria-describedby="fig-beta-dist-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-beta-dist-1.png" class="img-fluid figure-img" width="768">
</div>
</figure>
</div>
</div>
</div>
</section>
<section id="beta-regression" class="level2">
<h2 data-anchor-id="beta-regression">Beta Regression</h2>
<p>While <span class="math inline">\alpha</span> and <span class="math inline">\beta</span> are useful for defining the distribution, regression modeling typically uses a reparameterized form involving the <strong>mean</strong> <span class="math inline">\mu</span> and <strong>precision</strong> <span class="math inline">\phi</span> . The mean (<span class="math inline">\mu</span>) represents the expected value of the distribution, while the dispersion (<span class="math inline">\phi</span>), which is inversely related to variance, reflects how concentrated the distribution is around the mean, with higher values of <span class="math inline">\phi</span> representing a narrower distribution while lower values of <span class="math inline">\phi</span> represent a wider, more spread out, distribution. These are connected to <span class="math inline">\alpha</span> and <span class="math inline">\beta</span> as follows:</p>
<p><span class="math display">
\begin{aligned}[t]
\text{Shape 1:} &amp;&amp; a &amp;= \mu \phi \\
\text{Shape 2:} &amp;&amp; b &amp;= (1 - \mu) \phi
\end{aligned}
\qquad\qquad\qquad
\begin{aligned}[t]
\text{Mean:} &amp;&amp; \mu &amp;= \frac{a}{a + b} \\
\text{Precision:} &amp;&amp; \phi &amp;= a + b
\end{aligned}
</span></p>
<p>The variance can then be calculated as a function of <span class="math inline">\mu</span> and <span class="math inline">\phi</span>:</p>
<p><span class="math display">
\frac{\mu \cdot (1 - \mu)}{1 + \phi}
</span></p>
<p>Importantly, <em>the variance depends on the average value of the response</em>, which is what allows the model to non-linearly adjust to the bounds of the outcome.</p>
<p>In Beta regression, the goal is to model the mean of the response variable (often denoted <span class="math inline">\mu</span>) as a function of predictor variables, while optionally modeling a precision or dispersion parameter (often denoted <span class="math inline">\phi</span>) that captures variability around the mean. To ensure that <span class="math inline">\mu</span> stays between 0 and 1, we apply a link function, which transforms the mean so that it can be modeled on an unbounded scale. A common choice is the logit link (which is used by default), but other links (such as the probit or complementary log-log) are also available; the logit is not required.</p>
<p>The logit link is defined as <span class="math inline">\text{logit}(\mu) = \log \left( \frac{\mu}{1 - \mu} \right)</span>. This transforms the mean proportion, <span class="math inline">\mu</span>, into log-odds, which can take any real value between <span class="math inline">-\infty</span> and <span class="math inline">+\infty</span>. This makes it suitable for linear modeling. The inverse of the logit, called the logistic function, maps predictions back onto the 0 to 1 scale: <span class="math inline">\left[ \mu = \frac{1}{1 + e^{-\eta}} \right]</span> where <span class="math inline">\eta</span> is the linear predictor (a combination of your model’s coefficients and predictors). Similarly, the dispersion parameter, which must be positive, is often modeled using a log link: <span class="math inline">\log(\phi) = \text{linear predictor}</span>. This ensures the estimated dispersion is strictly positive.</p>
<p>Beta regression is designed specifically for modeling bounded, continuous outcomes. It respects the (0, 1) range of proportional data, accounts for varying variance across levels of the mean, and provides a coherent statistical foundation for estimating effects on proportions and percentages. This makes it especially useful in psychology, where outcomes like accuracy, recall rates, or rating scales often fall within this bounded range.</p>
</section>
<section id="bayesian-approach-to-beta-regression" class="level2">
<h2 data-anchor-id="bayesian-approach-to-beta-regression">Bayesian Approach to Beta Regression</h2>
<p>Beta regression models can be estimated from either a frequentist or Bayesian perspective. In this tutorial, we adopt a Bayesian framework—not because it is inherently superior, but because it offers practical advantages for fitting both simple and complex models <span class="citation" data-cites="gelman2013bayesian mcelreath2020statistical bayes_rules">(<a href="#ref-gelman2013bayesian" role="doc-biblioref">Gelman et al., 2013</a>; <a href="#ref-bayes_rules" role="doc-biblioref">Johnson et al., n.d.</a>; <a href="#ref-mcelreath2020statistical" role="doc-biblioref">McElreath, 2020</a>)</span>, especially in the context of psychological data. The R package <code>brms</code> <span class="citation" data-cites="brms">(<a href="#ref-brms" role="doc-biblioref">Bürkner, 2017</a>)</span>, which interfaces with the probabilistic programming language Stan <span class="citation" data-cites="stan2023">(<a href="#ref-stan2023" role="doc-biblioref">Team, 2023</a>)</span>, makes Bayesian modeling accessible and flexible, even for users with limited prior experience in Bayesian statistics.</p>
<p>At the heart of Bayesian estimation is the idea of updating beliefs. This invovles first specifying prior distributions that represent our beliefs about the parameters before observing any data. These priors are then updated through the likelihood function using the observed data, producing posterior distributions. The posterior reflects what we believe about the parameters after taking the data into account. Unlike frequentist approaches that yield point estimates and confidence intervals, Bayesian methods return full distributions over parameters. These distributions allow researchers to express uncertainty more directly, make probability-based claims, and explore the full range of plausible values for each estimate.</p>
<p>The models we fit throughout this tutorial are estimated using <code>brms</code>, which supports a wide range of regression families, including the Beta distribution and its extensions. It also allows for flexible model specification, including hierarchical structures, non-default link functions, and separate sub-models for different components.</p>
<p>Behind the scenes, <code>brms</code> uses Markov Chain Monte Carlo (MCMC) sampling to draw from the posterior distribution. This process generates a range of representative values for each parameter, allowing researchers to capture and quantify uncertainty. The resulting output can be summarized in several ways <span class="citation" data-cites="makowski2019a">(see <a href="#ref-makowski2019a" role="doc-biblioref">Makowski, Ben-Shachar, Chen, et al., 2019</a>)</span>. Point estimates, such as the posterior mean or median, serve as central tendency measures of the parameter distributions. Credible intervals, typically the 95% Cr.I., indicate the range of values within which a parameter is likely to fall given the data and prior assumptions. In addition, researchers can examine the probability of direction (pd), which reflects the proportion of the posterior distribution that lies entirely above or below zero, offering an intuitive summary of effect direction. Finally, Bayes Factors can be used to quantify the relative evidence for one model or hypothesis over another.</p>
<p>Throughout the tutorial, we will demonstrate how to fit these Bayesian Beta regression models using the <code>brm()</code> function. We will also focus on interpreting the results in terms of probabilities and proportions on the original response scale, rather than staying solely on the logit or latent scale.</p>
</section>
</section>
<section id="motivating-example" class="level1">
<h1>Motivating Example</h1>
<section id="data-and-methods" class="level2">
<h2 data-anchor-id="data-and-methods">Data and Methods</h2>
<p>To demonstrate how Beta regression can be applied in practice, we re-analyze data from a memory experiment conducted by <span class="citation" data-cites="wilford2020">Wilford et al. (<a href="#ref-wilford2020" role="doc-biblioref">2020</a>)</span>; Experiment 1A. This study investigated whether the fluency of an instructor’s delivery influenced participants’ memory performance on a final recall test. By working through this dataset, we illustrate each stage of Beta modeling—from basic regression to more advanced extensions—and show how Beta regression can offer more accurate and interpretable inferences for bounded outcome variables.</p>
<p>The idea that extrinsic cues, such as how easily information is processed, influence metamemory has been widely documented. In particular, a growing body of research has examined the role of instructor fluency—characterized by dynamic vocal intonation, purposeful gestures, mobility, and natural pacing—in shaping students’ perceptions of learning. Although fluent instructors are often judged as more effective, several studies have shown that fluency has little to no impact on actual memory performance <span class="citation" data-cites="carpenter2013 witherby2022">Toftness et al. (<a href="#ref-toftness2017" role="doc-biblioref">2017</a>)</span>. However, Wilford et al.&nbsp;(2020) reported the opposite: in a series of experiments, they found that participants recalled more information when watching a fluent versus a disfluent instructor. This contradictory pattern makes the dataset especially useful for illustrating statistical approaches to analyzing proportion data.</p>
<p>In Experiment 1A of Wilford et al.&nbsp;(2020), participants watched two brief instructional videos—one on the genetics of calico cats, the other on the biology of skin wrinkling. Each video was presented in either a fluent or disfluent style. The fluent condition featured instructors with smooth delivery, clear vocal modulation, and confident pacing. The disfluent condition included hesitations, monotone delivery, and unnatural pauses. After viewing the videos, participants made judgments of learning (JOLs), rating how likely they were to remember the material on a scale from 0 to 100. They also answered questions about the instructor and self-reported how much they believed they had learned. Following a distractor task, participants completed a free recall test, writing down as much information as they could remember from each video within a three-minute time limit. Each response was scored out of 10 possible idea units per video.</p>
<p>Our primary outcome variable is the proportion of idea units recalled on the final test, calculated by dividing the number of correct units by 10. Because this is a bounded continuous variable (i.e., it ranges from 0 to 1), it violates the assumptions of models like linear regression that treat outcomes as unbounded and homoscedastic. Despite this, it remains common in psychological research to analyze proportion data using models that assume normality. In what follows, we reproduce Wilford et al.’s original analysis and then re-analyze the data using Beta regression. Our goal is to highlight how Beta regression—and its extensions—can offer a more robust, flexible, and interpretable framework for modeling proportional outcomes.</p>
</section>
<section id="reanalysis-of-wilford-et-al.-experiment-1a" class="level2">
<h2 data-anchor-id="reanalysis-of-wilford-et-al.-experiment-1a">Reanalysis of Wilford et al.&nbsp;Experiment 1A</h2>
<p>In their original analysis of Experiment 1A, <span class="citation" data-cites="wilford2020">Wilford et al. (<a href="#ref-wilford2020" role="doc-biblioref">2020</a>)</span> compared memory performance between fluent and disfluent instructor conditions using a traditional independent-samples t-test. They found that participants who watched the fluent instructor recalled significantly more idea units than those who viewed the disfluent version (see <a href="#fig-flu1" class="quarto-xref" aria-expanded="false">Figure&nbsp;3</a>).</p>
<p>In our reanalysis, we take a Bayesian regression approach to replicate and extend these findings. Specifically, we model final test accuracy—defined as the number of correctly recalled idea units out of 10, converted to a proportion by dividing by 10—as the dependent variable. Our primary predictor is instructor fluency, with two levels: Fluent and Disfluent.</p>
<p>We use treatment (dummy) coding, which is the default in R. This coding scheme sets the first level of a factor (in alphabetical order) as the reference level. In this case, Disfluent is the reference, and the coefficient for Fluent reflects the contrast between fluent and disfluent instructor conditions.</p>
<p>We estimate the model using the <code>brm()</code> function from the <code>brms</code> package, with default (weakly informative) priors. This choice mirrors a common starting point for applied researchers and ensures that the results are primarily driven by the data. By using Bayesian estimation, we can go beyond simple group comparisons to quantify uncertainty, interpret results on the probability scale, and visualize the posterior distribution of the fluency effect.</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="1" data-custom-style="FigureWithoutNote">
<div id="tbl-dataset" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="1">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-dataset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;1</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>First six rows of dataset</p>
</div>
</figcaption>
<div aria-describedby="tbl-dataset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_e75ire0in9zxuk7vpi57(i, j, css_id) {
          var table = document.getElementById("tinytable_e75ire0in9zxuk7vpi57");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_e75ire0in9zxuk7vpi57');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_e75ire0in9zxuk7vpi57(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_e75ire0in9zxuk7vpi57");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 6, j: 0 }, { i: 6, j: 1 }, { i: 6, j: 2 },  ], css_id: 'tinytable_css_mpfjskcuh7qoiff6kuup',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 },  ], css_id: 'tinytable_css_jbvb25ea4mh6xf61h2wj',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_e75ire0in9zxuk7vpi57(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_mpfjskcuh7qoiff6kuup, .table th.tinytable_css_mpfjskcuh7qoiff6kuup { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_jbvb25ea4mh6xf61h2wj, .table th.tinytable_css_jbvb25ea4mh6xf61h2wj { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_e75ire0in9zxuk7vpi57" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Participant</th>
                <th scope="col">Fluency</th>
                <th scope="col">Accuracy</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>1</td>
                  <td>Fluent</td>
                  <td>0.45</td>
                </tr>
                <tr>
                  <td>8</td>
                  <td>Fluent</td>
                  <td>0.30</td>
                </tr>
                <tr>
                  <td>10</td>
                  <td>Fluent</td>
                  <td>0.40</td>
                </tr>
                <tr>
                  <td>18</td>
                  <td>Fluent</td>
                  <td>0.15</td>
                </tr>
                <tr>
                  <td>19</td>
                  <td>Fluent</td>
                  <td>0.50</td>
                </tr>
                <tr>
                  <td>14</td>
                  <td>Fluent</td>
                  <td>0.70</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-flu1" class="quarto-float quarto-figure quarto-figure-center" data-fignum="3" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-flu1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;3</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Fluency accuracy: Raw data (points), half-boxplot, and Mean marker (diamond)</p>
</div>
</figcaption>
<div aria-describedby="fig-flu1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-flu1-1.png" class="img-fluid figure-img" width="768">
</div>
</figure>
</div>
</div>
</div>
<section id="data" class="level3">
<h3 data-anchor-id="data">Data</h3>
<p>Before we fit the regression model, we will read in the data.</p>
<p>The first few rows of the data are displayed in <a href="#tbl-dataset" class="quarto-xref" aria-expanded="false">Table&nbsp;1</a>. The dataset contains three columns:</p>
<ul>
<li><p><strong><code>Participant:</code></strong> Participant ID number</p></li>
<li><p><strong><code>Fluency</code>:</strong> Fluent vs.&nbsp;Disfluent</p></li>
<li><p><strong><code>Accuracy</code>:</strong> Proportion recalled (idea units)</p></li>
</ul>
<div class="cell FigureWithNote" data-tbl-align="center" data-apa-note="Mean[95% Cr.I](pd); mu = logit; phi = log; zi= logit;  zoi = logit; coi = logit" prefix="" data-tblnum="2" data-custom-style="FigureWithNote">
<div id="tbl-all" class="cell quarto-float quarto-figure quarto-figure-center" data-tblnum="2" prefix="" data-apa-note="Mean[95% Cr.I](pd); mu = logit; phi = log; zi= logit;  zoi = logit; coi = logit" data-tbl-align="center">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-all-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;2</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Regression model summaries for each model</p>
</div>
</figcaption>
<div aria-describedby="tbl-all-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="Figures/combined_table.png" class="img-fluid figure-img"></p>
</figure>
</div>
</div>
</div>
</figure>
</div>
<div>
<div class="FigureNote NoIndent" data-custom-style="FigureNote">
<em>Note</em>. Mean<a href="pd">95% Cr.I</a>; mu = logit; phi = log; zi= logit; zoi = logit; coi = logit
</div>
</div>
</div>
</section>
</section>
<section id="bayesian-regression" class="level2">
<h2 data-anchor-id="bayesian-regression">Bayesian Regression</h2>
<p>To fit our Bayesian regression models, we will be using the <code>brms</code> <span class="citation" data-cites="brms">(<a href="#ref-brms" role="doc-biblioref">Bürkner, 2017</a>)</span> R package. For reference, lists each model that was run, along with the model parameters and posterior summaries, including the mean, 95% credible intervals (Cr.I), and posterior probabilities of direction (pd). For the following analyses we will be using default priors provided by <code>brms</code>, which are non-informative or weak. This will get us something tantamount to a frequentist model with maximum likelihood estimates most of the readers should be familiar with. For the purposes of this tutorial, we will not be getting into the minutiae of Bayesian data analysis (i.e., setting informative priors, MCMC sampling, etc,). For a more in-depth look into Bayesian data analysis I refer the reader to <span class="citation" data-cites="mcelreath2020statistical">McElreath (<a href="#ref-mcelreath2020statistical" role="doc-biblioref">2020</a>)</span> and <span class="citation" data-cites="bayes_rules">Johnson et al. (<a href="#ref-bayes_rules" role="doc-biblioref">n.d.</a>)</span>.</p>
<p>To create a regression model from in <code>brms</code> we will use the <code>brm()</code> function and fit a model looking at final test accuracy (<code>Accuracy</code>) as a function of instructor fluency (<code>fluency</code>). The syntax is similar to how you run a regression using the popular <code>lm</code> function in R. Here we are concerned with modeling mean performance differences between the fluency and disfluency conditions.</p>
<p>We first start by loading the <code>brms</code> <span class="citation" data-cites="brms">(<a href="#ref-brms" role="doc-biblioref">Bürkner, 2017</a>)</span> and <code>cmdstanr</code> <span class="citation" data-cites="cmdstanr">(<a href="#ref-cmdstanr" role="doc-biblioref">Gabry et al., 2024</a>)</span> packages. We use the <code>cmdstanr</code> backend for Stan <span class="citation" data-cites="stan2023">(<a href="#ref-stan2023" role="doc-biblioref">Team, 2023</a>)</span> because it’s faster and more modern than the defaults used to run models.<a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a></p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load brms and cmdstanr</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cmdstanr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We fit the model using the <code>brm()</code> function, as shown in the code below. To speed up processing we set <code>cores</code> = 4, and use the <code>cmdstand</code> backend. The output provides estimates for each effect: the coefficient (<code>b_,</code>) representing the mean of the posterior distribution), its estimation error (the standard deviation of the posterior distribution), and the 95% credible interval (Cr.I), which indicate the most credible values within the distribution. We inferred evidence for an effect when the 95% Cr.I did not include zero.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fit ols reg</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>bayes_ols_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency,</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">backend =</span> <span class="st">"cmdstanr"</span>,</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"model_ols_bayes"</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell tbl-cap-location-top" data-after-caption-space="0pt">

</div>
<p><a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a> displays the summary of the Bayesian regression model. To make the output more readable, each model parameter is labeled with a prefix before the variable name. The <code>b_Intercept</code> value refers to the accuracy in the disfluent condition (because we dummy coded our variable). The Fluency coefficient (<code>b_FluencyFluent</code>) highlights the difference between the fluent and disfluent conditions, <em>b</em> = 0.084, 95% Cr.Is = [0.002,0.166], pd =1. Our results are very similar to the results reported by <span class="citation" data-cites="wilford2020">Wilford et al. (<a href="#ref-wilford2020" role="doc-biblioref">2020</a>)</span> : <em>t</em>(94) = 2.00, <em>p = .048,</em> 95% CI [0.06, 16.56].</p>
<p>There are some notable differences between our Bayesian analysis and the frequenist analyses readers might be more sued to seeing, such as the absence of <em>t</em>- and <em>p</em>-values. There is a metric in the table that is included with models fit when using the <code>bayestestr</code> <span class="citation" data-cites="makowski2019">(<a href="#ref-makowski2019" role="doc-biblioref">Makowski, Ben-Shachar, &amp; Lüdecke, 2019</a>)</span> function from <code>easystats</code> called probability of direction (pd) that gives an indication of how much of the posterior distribution estimate is in one direction (positive or negative). The pd measure appears to correlated with <em>p</em>-values (see <span class="citation" data-cites="makowski2019 makowski2019a">(<a href="#ref-makowski2019" role="doc-biblioref">Makowski, Ben-Shachar, &amp; Lüdecke, 2019</a>; <a href="#ref-makowski2019a" role="doc-biblioref">Makowski, Ben-Shachar, Chen, et al., 2019</a>)</span>). A pd of 95%, 97.5%, 99.5% and 99.95% correspond approximately to two-sided p-value of respectively .1, .05, .01 and .001. In addition to the pd value, one can look at the 95% credible interval (sometimes called highest probability density, depending on the package being used) to see if it includes 0–if it does not then the effect can be said to be significant. In the table below, the 95% credible intervals are located in the 95% Cr.Is column.</p>
</section>
<section id="beta-regression-1" class="level2">
<h2 data-anchor-id="beta-regression-1">Beta Regression</h2>
<p><span class="citation" data-cites="wilford2020">Wilford et al. (<a href="#ref-wilford2020" role="doc-biblioref">2020</a>)</span>, using a traditional GLM approach, observed that instructor fluency impacts actual learning. Keep in mind the traditional approach assumes normality of residuals and homoscadacity or constant variance. These assumptions are tricky to maintain when the continuous response approaches either the upper or lower boundary of the scale and are almost never true (see <span class="citation" data-cites="sladekova2024unicorns">(<a href="#ref-sladekova2024unicorns" role="doc-biblioref">Sladekova &amp; Field, 2024</a>)</span>. Does the data we have meet those assumptions? Using <code>easystats</code> <span class="citation" data-cites="easystats">(<a href="#ref-easystats" role="doc-biblioref">Lüdecke et al., 2022</a>)</span> and the <code>check_model()</code> function, we can easily check this. In <a href="#fig-ols-assump" class="quarto-xref" aria-expanded="false">Figure&nbsp;4</a> , we see there definitely some issues with our model. Specifically, there appears to be violations of normality constant variance (homogeneity).</p>
<p>We can also examine how well the data fits the model by performing a posterior predictive check. In <a href="#fig-ppcheck-normal" class="quarto-xref" aria-expanded="false">Figure&nbsp;5</a>, multiple draws or repetitions from the posterior distribution are plotted (light blue lines) against the observed data. (dark green). Ideally, the predictive draws (the light blue lines) should show reasonable resemblance with the observed data (dark blue line). In our example, the model-predicted density is slightly too peaked and narrow compared to the data. In addition, some of the draws extend into negative accuracy values.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-ols-assump" class="quarto-float quarto-figure quarto-figure-center" data-fignum="4" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-ols-assump-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;4</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Two assumption checks for our OLS model: Normality (left) and Homogeneity (right)</p>
</div>
</figcaption>
<div aria-describedby="fig-ols-assump-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-ols-assump-1.png" class="img-fluid figure-img" width="768">
</div>
</figure>
</div>
</div>
</div>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-ppcheck-normal" class="quarto-float quarto-figure quarto-figure-center" data-fignum="5" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-ppcheck-normal-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;5</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Posterior predictive check</p>
</div>
</figcaption>
<div aria-describedby="fig-ppcheck-normal-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-ppcheck-normal-1.png" class="img-fluid figure-img" width="768">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>Given the outcome variable is proportional, one solution would be to run a beta regression model.</p>
</div>
<p>Again, we can create the beta regression model in <code>brms</code>. In <code>brms</code>, we model each parameter independently. Recall from the introduction that in a Beta model we model two parameters—<span class="math inline">\mu</span> and <span class="math inline">\phi</span>. We can easily do this by using the <code>bf()</code> function from <code>brms</code>. <code>bf()</code> facilitates the specification of several sub-models within the same formula call. We fit two formulas, one for <span class="math inline">\mu</span> and one for <span class="math inline">\phi</span> and store it in the <code>model_beta_bayes</code> object below. In the below <code>bf()</code> call, we are modeling Fluency as a function of Accuracy only for the <span class="math inline">\mu</span> parameter. For the <span class="math inline">\phi</span> parameter, we are only modeling the intercept value. This is saying dispersion does not change as a function of fluency.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fit model with mu and phi</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>model_beta_bayes <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># fit mu model</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> <span class="dv">1</span> <span class="co"># fit phi model</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>If we try to run <code>beta_brms</code>, however, we get an error: <code>Error: Family 'beta' requires response greater than 0</code>. This is by design. If your remember, the Beta distribution can only model responses in the interval [0-1], but not responses that are exactly 0 or 1. We need make sure there are no zeros and ones in our dataset.</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="3" data-custom-style="FigureWithoutNote">
<div id="tbl-01s" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="3">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-01s-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;3</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Number of zeros and ones in our dataset</p>
</div>
</figcaption>
<div aria-describedby="tbl-01s-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_9hui8sc080gm9804hnm0(i, j, css_id) {
          var table = document.getElementById("tinytable_9hui8sc080gm9804hnm0");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_9hui8sc080gm9804hnm0');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_9hui8sc080gm9804hnm0(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_9hui8sc080gm9804hnm0");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 2, j: 0 }, { i: 2, j: 1 },  ], css_id: 'tinytable_css_2wzp63lodc0cq5jiu6yk',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 },  ], css_id: 'tinytable_css_qr4pn6v3xyv8hnvo8b6m',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_9hui8sc080gm9804hnm0(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_2wzp63lodc0cq5jiu6yk, .table th.tinytable_css_2wzp63lodc0cq5jiu6yk { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_qr4pn6v3xyv8hnvo8b6m, .table th.tinytable_css_qr4pn6v3xyv8hnvo8b6m { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_9hui8sc080gm9804hnm0" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Accuracy</th>
                <th scope="col">Count</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>0</td>
                  <td>9</td>
                </tr>
                <tr>
                  <td>1</td>
                  <td>1</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p><a href="#tbl-01s" class="quarto-xref" aria-expanded="false">Table&nbsp;3</a> shows we have 9 rows with accuracy of 0, and 1 row with an accuracy of exactly 1. To run a Beta regression, a common hack is to nudge our 0s towards .01 and our 1s to .99 so they fall within the interval of [0-1]. The model <code>beta_brms</code> uses data where 0s and 1s are changed to .01 and .99, respectively.</p>
</div>
<p>We can rerun our model replacing <code>fluency_data</code> with <code>data_beta</code> in the below function call.</p>
<p>No errors this time! We will perform the Beta regression using the nudged values of .01 and .99 values and report our results.</p>
<section id="model-parameters" class="level4">
<h4 data-anchor-id="model-parameters">Model Parameters.</h4>
<p><a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a> displays the summary from all out Bayesian implementation. <a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a> The <span class="math inline">\mu</span> parameter estimates, which are labeled with an underscore <code>b_</code>while <span class="math inline">\phi</span> parameter coefficients are tagged with b<code>_phi</code> in the <code>Parameter</code> column.</p>
</section>
<section id="mean-mu" class="level3">
<h3 data-anchor-id="mean-mu">Mean <span class="math inline">\mu</span></h3>
<p>In <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a> under the Beta Regression column, the first set of coefficients represent how factors influence the <span class="math inline">\mu</span> parameter, which is the mean of the beta distribution. These coefficients are interpreted on the scale of the logit, meaning they represent linear changes on a nonlinear space. The intercept term <code>(b_Intercept)</code> represents the log odds of the mean on accuracy for the fluent instructor. Log odds that are negative indicate that it is more likely a “success” (like getting the correct answer) will NOT happen than that it will happen. Similarly, regression coefficients in log odds forms that are negative indicate that an increase in that predictor leads to a decrease in the predicted probability of a “success”.</p>
<section id="predicted-probabilities" class="level4">
<h4 data-anchor-id="predicted-probabilities">Predicted Probabilities.</h4>
<p>Parameter estimates are usually difficult to intercept on their own. We argue that researchers should not spend too much time interpreting single model estimates. We report them in this tutorial for completeness. Instead researchers should discuss the effects of the predictor on the actual outcome of interest (in this case the 0-1 scale). The logit link allows us to transform back and forth between the scale of a linear model and the nonlinear scale of the outcome, which is bounded by 0 and 1. By using the inverse of the logit, we can easily transform our linear coefficients to obtain average effects on the scale of the proportions or percentages, which is usually what is interesting to applied researchers. In a simple case, we can do this manually, but when there are many factors in your model this can be quite complex.</p>
<p>In our example, we can use the <code>plogis()</code> function in base R to convert estimates from the log-odds (logit) scale to the probability scale. The intercept of our model is -0.918, which reflects the log-odds of the mean accuracy in the disfluent condition. If the estimated difference between the fluent and disfluent conditions is 0.24 on the log-odds scale, we first add this value to the intercept to get the log-odds for the fluent condition: <code>-0.918 + 0.24 = -0.678</code>. We then use <code>plogis()</code> to convert both log-odds values to probabilities:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Log-odds for each condition</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>log_odds_disfluent <span class="ot">&lt;-</span> <span class="sc">-</span><span class="fl">1.036</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>log_odds_fluent <span class="ot">&lt;-</span> log_odds_disfluent <span class="sc">+</span> <span class="fl">0.24</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert to probabilities</span></span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a>prob_disfluent <span class="ot">&lt;-</span> <span class="fu">plogis</span>(log_odds_disfluent)</span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a>prob_fluent <span class="ot">&lt;-</span> <span class="fu">plogis</span>(log_odds_fluent)</span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a><span class="co"># View results</span></span>
<span id="cb4-10"><a href="#cb4-10" aria-hidden="true" tabindex="-1"></a>prob_disfluent <span class="co"># ≈ 0.262</span></span>
<span id="cb4-11"><a href="#cb4-11" aria-hidden="true" tabindex="-1"></a>prob_fluent <span class="co"># ≈ 0.355</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This is pretty easy to do manually, but when your model has many predictors, it can be quite cumbersome. To help us extract predictions from our model and visualize them we will use a package called <strong><code>marginaleffects</code></strong> <span class="citation" data-cites="margeffects2024">(<a href="#ref-margeffects2024" role="doc-biblioref">Arel-Bundock et al., 2024</a>)</span>. <a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a> To get the proportions for each of our categorical predictors on the mu parameter we can use the function from the package called <code>predictions()</code>. These are displayed in <a href="#tbl-predict-prob" class="quarto-xref" aria-expanded="false">Table&nbsp;4</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load marginaleffects package</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(marginaleffects)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># use predictions to get predicted probs for each condition</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">redictions</span>(</span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a>  beta_brms,</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">newdata =</span> <span class="fu">datagrid</span>(<span class="at">Fluency =</span> <span class="fu">c</span>(<span class="st">"Disfluent"</span>, <span class="st">"Fluent"</span>)) <span class="co"># need to specify the levels of the categorical predictor</span></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="4" data-tbl-cap-location="top" data-custom-style="FigureWithoutNote">
<div id="tbl-predict-prob" class="cell quarto-float quarto-figure quarto-figure-center" data-after-caption-space="0pt" prefix="" data-tbl-cap-location="top" data-tblnum="4">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-predict-prob-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;4</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Predicted probablities for fluency factor</p>
</div>
</figcaption>
<div aria-describedby="tbl-predict-prob-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_0w0cvrcb25l3rfbpfgqs(i, j, css_id) {
          var table = document.getElementById("tinytable_0w0cvrcb25l3rfbpfgqs");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_0w0cvrcb25l3rfbpfgqs');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_0w0cvrcb25l3rfbpfgqs(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_0w0cvrcb25l3rfbpfgqs");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 2, j: 0 }, { i: 2, j: 1 }, { i: 2, j: 2 },  ], css_id: 'tinytable_css_uo1i6gkxnrtuq4o7jtms',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 },  ], css_id: 'tinytable_css_9nkffednyr5vokkcpeya',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_0w0cvrcb25l3rfbpfgqs(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_uo1i6gkxnrtuq4o7jtms, .table th.tinytable_css_uo1i6gkxnrtuq4o7jtms { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_9nkffednyr5vokkcpeya, .table th.tinytable_css_9nkffednyr5vokkcpeya { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_0w0cvrcb25l3rfbpfgqs" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Fluency</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>Disfluent</td>
                  <td>0.264</td>
                  <td>[0.213, 0.322]</td>
                </tr>
                <tr>
                  <td>Fluent</td>
                  <td>0.357</td>
                  <td>[0.293, 0.423]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>For the <code>Fluency</code> factor, we can interpret the <code>Mean</code> column in terms of proportions or percentages. That is, participants who watched the fluent instructor scored on average 36% on the final exam compared to 26% for those who watched the disfluent instructor.</p>
</div>
<p>We can also easily visualize these from <code>marginaleffects</code> using the <code>plot_predictions</code> function. After using this function, the proportions are visualized in <a href="#fig-plot-pre" class="quarto-xref" aria-expanded="false">Figure&nbsp;6</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>beta_plot <span class="ot">&lt;-</span> <span class="fu">plot_predictions</span>(beta_brms, <span class="at">condition =</span> <span class="st">"Fluency"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-plot-pre" class="quarto-float quarto-figure quarto-figure-center" data-fignum="6" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-plot-pre-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;6</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Predicted probablities for fluency factor</p>
</div>
</figcaption>
<div aria-describedby="fig-plot-pre-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-plot-pre-1.png" class="img-fluid figure-img" width="768">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>Readers might notice that <a href="#fig-plot-pre" class="quarto-xref" aria-expanded="false">Figure&nbsp;6</a> displays a single point estimate along with its 95% Cr.I.. In Bayesian data analysis, MCMC sampling generates a distribution for each parameter of interest. This approach allows us to visualize the uncertainty surrounding our estimates. Using the <code>marginaleffects</code> package, we can obtain samples from the posterior distribution with the <code>posterior_draws()</code> function. We can then plot these results to illustrate the range of plausible values for our estimates at different levels of uncertainty (e.g., 80% or 95%) (see <a href="#fig-draws-fluency" class="quarto-xref" aria-expanded="false">Figure&nbsp;7</a>).</p>
</div>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-draws-fluency" class="quarto-float quarto-figure quarto-figure-center" data-fignum="7" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-draws-fluency-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;7</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Predicted probablity posterior distributions by fluency</p>
</div>
</figcaption>
<div aria-describedby="fig-draws-fluency-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-draws-fluency-1.png" class="img-fluid figure-img" width="768">
</div>
</figure>
</div>
</div>
</div>
</section>
<section id="marginal-effects" class="level4">
<h4 data-anchor-id="marginal-effects">Marginal Effects.</h4>
<p>Marginal effects provide a way to understand how changes in a predictor influence an outcome, holding all other factors constant in a specific manner. Technically, marginal effects are calculated using partial derivatives for continuous variables or finite differences for categorical and continuous variables, depending on the nature of the data and the research question. Substantively, these effects translate regression coefficients into a form that can be interpreted directly on the outcome scale of interest.</p>
<p>There are various types of marginal effects, and their calculation can vary across software packages. For example, the popular <code>emmeans</code> package [] computes marginal effects by holding all predictors at their means. In this tutorial, we will use the <code>marginaleffects</code> package, which focuses on average marginal effects (AMEs) by default. AMEs summarize effects by generating predictions for each row of the original dataset and then averaging these predictions. This approach retains a strong connection to the original data while offering a straightforward summary of the effect of interest.</p>
<p>One practical application of AMEs is calculating the risk difference for categorical variables. The risk difference represents the difference in average outcomes between two groups or conditions. Using the <code>avg_comparisons()</code> function in the <code>marginaleffects</code> package, we can compute this metric directly. By default, the function calculates the discrete difference between groups. It can also compute other effect size metrics, such as odds ratios and risk ratios, depending on the research question (see <a href="#tbl-or" class="quarto-xref" aria-expanded="false">Table&nbsp;6</a> for more details). This flexibility makes it a powerful tool for interpreting regression results in a meaningful way.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get risk difference by default</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>beta_avg_comp <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(beta_brms)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="5" data-custom-style="FigureWithoutNote">
<div id="tbl-ame1" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="5">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-ame1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;5</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Risk difference for instructor fluency</p>
</div>
</figcaption>
<div aria-describedby="tbl-ame1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_l1yloe9mbz8rlozikyfl(i, j, css_id) {
          var table = document.getElementById("tinytable_l1yloe9mbz8rlozikyfl");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_l1yloe9mbz8rlozikyfl');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_l1yloe9mbz8rlozikyfl(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_l1yloe9mbz8rlozikyfl");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 }, { i: 1, j: 2 }, { i: 1, j: 3 },  ], css_id: 'tinytable_css_fnu3oqsk9mjvfhiczspf',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_xroaa8qbzmtq0w2q95w8',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_l1yloe9mbz8rlozikyfl(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_fnu3oqsk9mjvfhiczspf, .table th.tinytable_css_fnu3oqsk9mjvfhiczspf { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_xroaa8qbzmtq0w2q95w8, .table th.tinytable_css_xroaa8qbzmtq0w2q95w8 { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_l1yloe9mbz8rlozikyfl" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Term</th>
                <th scope="col">Contrast</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>Fluency</td>
                  <td>Fluent - Disfluent</td>
                  <td>0.093</td>
                  <td>[0.009, 0.176]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p><a href="#tbl-ame1" class="quarto-xref" aria-expanded="false">Table&nbsp;5</a> displays the risk difference for the fluency factor (<code>estimate</code> column). The difference between the fluent and disfluent conditions is .09. That is, participants who watched a fluent instructor scored 9% higher on the final recall test than participants who watched the disfluent instructor. Our credible interval [-.0174, -0.011] shows that zero is not among the 95% most credible values.</p>
</div>
<p>We can also get the odds ratio with <code>avg_comparisons</code> (see <a href="#tbl-or" class="quarto-xref" aria-expanded="false">Table&nbsp;6</a>).</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="6" data-custom-style="FigureWithoutNote">
<div id="tbl-or" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="6">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-or-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;6</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Odds ratio for fluency factor</p>
</div>
</figcaption>
<div aria-describedby="tbl-or-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_zsc7j238al1ecawmj6j9(i, j, css_id) {
          var table = document.getElementById("tinytable_zsc7j238al1ecawmj6j9");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_zsc7j238al1ecawmj6j9');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_zsc7j238al1ecawmj6j9(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_zsc7j238al1ecawmj6j9");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 }, { i: 1, j: 2 }, { i: 1, j: 3 },  ], css_id: 'tinytable_css_920k6c8idyrmzqmyjp6u',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_mnuwtkdqiv42fstlpb3g',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_zsc7j238al1ecawmj6j9(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_920k6c8idyrmzqmyjp6u, .table th.tinytable_css_920k6c8idyrmzqmyjp6u { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_mnuwtkdqiv42fstlpb3g, .table th.tinytable_css_mnuwtkdqiv42fstlpb3g { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_zsc7j238al1ecawmj6j9" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Term</th>
                <th scope="col">Contrast</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>Fluency</td>
                  <td>ln(odds(Fluent) / odds(Disfluent))</td>
                  <td>1.552</td>
                  <td>[1.041, 2.307]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>In psychology, it is common to report effect size measures like Cohen’s <em>d</em> <span class="citation" data-cites="cohen1977">(<a href="#ref-cohen1977" role="doc-biblioref">Cohen, 1977</a>)</span>. When working with proportions we can calculate something similar called Cohen’s <em>h</em>. Taking our proportions, we can use the below equation to calculate Cohen’s <em>h</em> along with the 95% Cr.Is around it.</p>
</div>
<p><span class="math display">
h = 2 \cdot \left( \arcsin\left(\sqrt{p_1}\right) - \arcsin\left(\sqrt{p_2}\right) \right)
</span></p>
<p>Using this metric we see the effect size is small (0.17), 95% Cr.I [-0.002, 0.361].</p>
</section>
</section>
<section id="precision-phi-component" class="level3">
<h3 data-anchor-id="precision-phi-component">Precision (<span class="math inline">\phi</span>) Component</h3>
<p>The other component we need to pay attention to is the dispersion or precision parameter coefficients labeled as `<code>b_phi</code> in <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a>. The dispersion (<span class="math inline">\phi</span>) parameter tells us how precise our estimate is. Specifically, <span class="math inline">\phi</span> in Beta regression tells us about the variability of the response variable around its mean. Specifically, a higher dispersion parameter indicates a narrower distribution, reflecting less variability. Conversely, a lower dispersion parameter suggests a wider distribution, reflecting greater variability. The main difference between a dispersion parameter and the variance is that the dispersion has a different interpretation depending on the value of the outcome, as we show below. The best way to understand dispersion is to examine visual changes in the distribution as the dispersion increases or decreases.</p>
<p>Understanding the dispersion parameter helps us gauge the precision of our predictions and the consistency of the response variable. In <code>beta_brms</code> we only modeled the dispersion of the intercept. When <span class="math inline">\phi</span> is not specified, the intercept is modeled by default (see <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a>).</p>
<p>The intercept under the precision heading is not that interesting. It represents the overall dispersion in the model. We can model the dispersion of the <code>Fluency</code> factor—this allows dispersion to differ between the fluent and disfluent conditions. To do this we add <code>Fluency</code> to the phi model in <code>bf()</code>. We model the precision of the <code>Fluency</code> factor by using a <code>~</code> and adding factors of interest to the right of it.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>model_beta_bayes_disp <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># fit mu model</span></span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency <span class="co"># fit phi model with fluency</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>beta_brms_dis <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>  model_beta_bayes_disp,</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_beta,</span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">Beta</span>(),</span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">chain =</span> <span class="dv">4</span>,</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">backend =</span> <span class="st">"cmdstanr"</span>,</span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"model_beta_bayes_dis_run"</span> <span class="co"># Save this so it doesn't have to always rerun</span></span>
<span id="cb10-9"><a href="#cb10-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a> displays the model summary with the precision parameter added to our model as a function of fluency. It is important to note that the estimates are logged and not on the original scale (this is only the case when additional parameters are modeled). To interpret them on the original scale, we can exponentiate the log-transformed value—this transformation gets us back to our original scale . In below model call, we set <code>exponentiate</code> = TRUE.</p>
<div class="cell" data-after-caption-space="0pt">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a>beta_model_dis_exp <span class="ot">&lt;-</span> beta_brms_dis <span class="sc">|&gt;</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">model_parameters</span>(<span class="at">exponentiate =</span> <span class="cn">TRUE</span>, <span class="at">centrality =</span> <span class="st">"mean"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="7" data-custom-style="FigureWithoutNote">
<div id="tbl-phi-beta-exp" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="7">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-phi-beta-exp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;7</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Beta regression model summary for fluency factor with <span class="math inline">\phi</span> parameter exponentiated</p>
</div>
</figcaption>
<div aria-describedby="tbl-phi-beta-exp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_k4o33c9tajegkgx7g2fo(i, j, css_id) {
          var table = document.getElementById("tinytable_k4o33c9tajegkgx7g2fo");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_k4o33c9tajegkgx7g2fo');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_k4o33c9tajegkgx7g2fo(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_k4o33c9tajegkgx7g2fo");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 2, j: 0 }, { i: 2, j: 1 }, { i: 2, j: 2 }, { i: 2, j: 3 },  ], css_id: 'tinytable_css_winokfjx3fegk8p8kqr8',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_e8pz7jpfirdrjgobzwgc',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_k4o33c9tajegkgx7g2fo(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_winokfjx3fegk8p8kqr8, .table th.tinytable_css_winokfjx3fegk8p8kqr8 { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_e8pz7jpfirdrjgobzwgc, .table th.tinytable_css_e8pz7jpfirdrjgobzwgc { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_k4o33c9tajegkgx7g2fo" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Parameter</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
                <th scope="col">pd</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>b_phi_Intercept</td>
                  <td>2.41</td>
                  <td>[1.655, 3.365]</td>
                  <td>1</td>
                </tr>
                <tr>
                  <td>b_phi_FluencyFluent</td>
                  <td>2.52</td>
                  <td>[1.483, 4.267]</td>
                  <td>1</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>The <span class="math inline">\phi</span> intercept represents the precision of the fluent condition. The <span class="math inline">\phi</span> coefficient for <code>Fluency1</code> represents the change in that precision for performance between the fluent vs.&nbsp;disfluent conditions. The Cr.I [.235, .684] do not include 0 so our results are statistically significant.</p>
</div>
<p>It is important to note that these estimates are not the same as the marginal effects we discussed earlier. Changes in dispersion will change the shape of the distribution but not necessarily the average value of the response. This makes dispersion most interesting for research questions that focus on other features of the distribution besides the mean, such as the level of polarization in an outcome.</p>
<p>A critical assumption of the GLM is homoscedasticity, which means constant variance of the errors. Here we see one of the benefits of a beta regression model: we can include a dispersion parameter for Fluency. Properly accounting for dispersion is crucial because it impacts the precision of our mean estimates and, consequently, the significance of our coefficients. The inclusion of dispersion in the our model increased the uncertainty of the <span class="math inline">\mu</span> coefficient (see <a href="#fig-plt-diff" class="quarto-xref" aria-expanded="false">Figure&nbsp;8</a>)–our Cr.I is now very close to zero. This suggests that failing to account for the dispersion of the variables might lead to biased estimates. This highlights the potential utility of an approach like Beta regression over a traditional approach as Beta regression can explicitly model dispersion and address issues of heteroscedasticity.</p>
<p>We won’t always need to include dispersion parameters for each of our variables. We advise conducting very simple model comparisons like leave one out (loo) cross validation to examine if a dispersion parameter should be considered in our model.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-plt-diff" class="quarto-float quarto-figure quarto-figure-center" data-fignum="8" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-plt-diff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;8</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Comparison of posterior distributions for the risk difference in fluency: Simple model (no dispersion for Fluency) vs.&nbsp;complex model with dispersion</p>
</div>
</figcaption>
<div aria-describedby="fig-plt-diff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-plt-diff-1.png" class="img-fluid figure-img" width="768">
</div>
</figure>
</div>
</div>
</div>
</section>
<section id="posterior-predictive-check" class="level3">
<h3 data-anchor-id="posterior-predictive-check">Posterior Predictive Check</h3>
<p>It is a always a good idea to check how well your data fits the model. The <code>pp_check()</code> function allows us to examine the fit between our data and the model. In <a href="#fig-post-pred" class="quarto-xref" aria-expanded="false">Figure&nbsp;9</a>, multiple draws or repetitions from the posterior distribution are plotted (light blue lines) against the observed data. (dark blue). Ideally, the predictive draws (the light blue lines) should show reasonable resemblance with the observed data (dark blue line). We see it does a pretty good job capturing the data. Ideally, the predictive draws (the light blue lines) should show reasonable resemblance with the observed data (dark blue line). We see it does a pretty good job capturing the data.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-post-pred" class="quarto-float quarto-figure quarto-figure-center" data-fignum="9" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-post-pred-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;9</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Posterior predictive check for our Beta model with 100 draws.</p>
</div>
</figcaption>
<div aria-describedby="fig-post-pred-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-post-pred-1.png" class="img-fluid figure-img" width="768">
</div>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="zero-inflated-beta-zib-regression" class="level2">
<h2 data-anchor-id="zero-inflated-beta-zib-regression">Zero-Inflated Beta (ZIB) Regression</h2>
<p>A limitation of the beta regression model is it can can only model values between 0 and 1, but not exactly 0 or 1. In our dataset we have 9 rows with <code>Accuracy</code> equal to zero.</p>
<p>To use the Beta distribution we nudged our zeros to 0.01–which is never a good idea in practice. In our case it might be important to model the structural zeros in our data, as fluency of instructor might be an important factor in predicting the zeros in our model. There is a model called the zero-inflated beta (ZIB) model that takes into account the structural 0s in our data. We’ll still model the <span class="math inline">\mu</span> and <span class="math inline">\phi</span> (or mean and precision) of the Beta distribution, but now we’ll also add one new special parameter: <span class="math inline">\alpha</span>.</p>
<p>With zero-inflated regression, we’re actually modelling a mixture of the data-generating process. The <span class="math inline">\alpha</span> parameter uses a logistic regression to model whether the data is 0 or not. Substantively, this could be a useful model when we think that 0s come from a process that is relatively distinct from the data that is greater than 0. For example, if we had a dataset of with proportion of looks or eye fixations to certain areas on marketing materials, we might want a separate model for those that do not look at certain areas on the screen because individuals who do not look might be substantively different than those that look.</p>
<p>We can fit a ZIB model using <code>brms</code> and use the <code>marginaleffects</code> package to make inferences about our parameters of interest. Before we run a zero-inflated beta model, we will need to transform our data again and nudge our 1s to .99–we can keep our zeros. Similar to our Beta regression model we fit in <code>brms</code>, we will use the <code>bf()</code> function to fit several models. We fit our <span class="math inline">\mu</span> and <span class="math inline">\phi</span> parameters as well as our zero-inflated parameter (<span class="math inline">\alpha</span>; here labeled as <code>zi</code>). In <code>brms</code> we can use the zero_inflated_beta family.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="co"># fit zero-inflated beta in brms</span></span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a>zib_model <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># The mean of the 0-1 values, or mu</span></span>
<span id="cb12-5"><a href="#cb12-5" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency, <span class="co"># The precision of the 0-1 values, or phi</span></span>
<span id="cb12-6"><a href="#cb12-6" aria-hidden="true" tabindex="-1"></a>  zi <span class="sc">~</span> Fluency, <span class="co"># The zero-or-one-inflated part, or alpha</span></span>
<span id="cb12-7"><a href="#cb12-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">zero_inflated_beta</span>()</span>
<span id="cb12-8"><a href="#cb12-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Below we pass <code>zib_model</code> to the <code>brm</code> function.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>fit_zi <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> zib_model,</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_beta_0,</span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">chain =</span> <span class="dv">4</span>,</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">backend =</span> <span class="st">"cmdstanr"</span>,</span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"model_beta_bayes_zib"</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="predicted-probabilities-and-marginal-effects" class="level3">
<h3 data-anchor-id="predicted-probabilities-and-marginal-effects">Predicted Probabilities and Marginal Effects</h3>
<p><a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a> under the zero-inflated Beta regression column provides a summary of the posterior distribution for each parameter. As stated before, it is preferable to back-transform our estimates to get probabilities. To get the predicted probabilities we can use the <code>avg_predictions()</code> function and to get risk difference between the levels of each factor we can use the <code>avg_comparisons()</code> function from <code>marginaleffects</code> package <span class="citation" data-cites="marginaleffects">(<a href="#ref-marginaleffects" role="doc-biblioref">Arel-Bundock, 2024</a>)</span>. We can model the parameters separately using the <code>dpar</code> argument setting to: <span class="math inline">\mu</span>, <span class="math inline">\phi</span>, <span class="math inline">\alpha</span>. Here we look at the risk difference for Fluency under each parameter.</p>
<section id="mean-mu-1" class="level4">
<h4 data-anchor-id="mean-mu-1">Mean (<span class="math inline">\mu</span>).</h4>
<p>Looking at <a href="#tbl-marg-mu-brms" class="quarto-xref" aria-expanded="false">Table&nbsp;8</a> there is no significant effect for Fluency.</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="8" data-custom-style="FigureWithoutNote">
<div id="tbl-marg-mu-brms" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="8">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-marg-mu-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;8</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Risk difference (<span class="math inline">\mu</span>) for fluency factor</p>
</div>
</figcaption>
<div aria-describedby="tbl-marg-mu-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_c2au28vrn27k47zjfbnx(i, j, css_id) {
          var table = document.getElementById("tinytable_c2au28vrn27k47zjfbnx");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_c2au28vrn27k47zjfbnx');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_c2au28vrn27k47zjfbnx(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_c2au28vrn27k47zjfbnx");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 }, { i: 1, j: 2 }, { i: 1, j: 3 },  ], css_id: 'tinytable_css_fehfukdx4mgxiwwyb9nu',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_s143130mcsg29cplwot2',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_c2au28vrn27k47zjfbnx(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_fehfukdx4mgxiwwyb9nu, .table th.tinytable_css_fehfukdx4mgxiwwyb9nu { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_s143130mcsg29cplwot2, .table th.tinytable_css_s143130mcsg29cplwot2 { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_c2au28vrn27k47zjfbnx" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Term</th>
                <th scope="col">Contrast</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>Fluency</td>
                  <td>Fluent - Disfluent</td>
                  <td>0.006</td>
                  <td>[-0.08, 0.087]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
</section>
<section id="dispersion-phi" class="level4">
<h4 data-anchor-id="dispersion-phi">Dispersion (<span class="math inline">\phi</span>).</h4>
<p>Looking at <a href="#tbl-marg-phi-brms" class="quarto-xref" aria-expanded="false">Table&nbsp;9</a>, there is a significant effect of fluency on dispersion, with disfluency having a larger variation than fluency.</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="9" data-custom-style="FigureWithoutNote">
<div id="tbl-marg-phi-brms" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="9">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-marg-phi-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;9</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Risk difference (<span class="math inline">\phi</span>) for fluency factor</p>
</div>
</figcaption>
<div aria-describedby="tbl-marg-phi-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_jir7b60gshvba4972j5j(i, j, css_id) {
          var table = document.getElementById("tinytable_jir7b60gshvba4972j5j");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_jir7b60gshvba4972j5j');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_jir7b60gshvba4972j5j(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_jir7b60gshvba4972j5j");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 }, { i: 1, j: 2 }, { i: 1, j: 3 },  ], css_id: 'tinytable_css_58pvwhpqg36qb6tbloyo',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_bj7jv7z2477n43uhyx3o',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_jir7b60gshvba4972j5j(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_58pvwhpqg36qb6tbloyo, .table th.tinytable_css_58pvwhpqg36qb6tbloyo { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_bj7jv7z2477n43uhyx3o, .table th.tinytable_css_bj7jv7z2477n43uhyx3o { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_jir7b60gshvba4972j5j" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Term</th>
                <th scope="col">Contrast</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>Fluency</td>
                  <td>Fluent - Disfluent</td>
                  <td>4.4</td>
                  <td>[1.286, 8.153]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
</section>
</section>
<section id="zero-inflated-alpha" class="level3">
<h3 data-anchor-id="zero-inflated-alpha">Zero-Inflated (<span class="math inline">\alpha</span>)</h3>
<p>In <a href="#tbl-marg-zib-brms" class="quarto-xref" aria-expanded="false">Table&nbsp;10</a> we see that watching a lecture video with a fluent instructor reduces the proportion of zeros by about 13%. The Cr.I does not include zero.</p>
<div id="tbl-marg-zib-brms" class="cell quarto-float quarto-figure quarto-figure-center FigureWithoutNote" data-tblnum="10" prefix="" data-custom-style="FigureWithoutNote">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl quarto-uncaptioned" id="tbl-marg-zib-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;10
</figcaption>
<div aria-describedby="tbl-marg-zib-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div>

</div>
</div>
</figure>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>If we wanted we could easily visualize the zero-inflated part of the model (see <a href="#fig-zi-plot" class="quarto-xref" aria-expanded="false">Figure&nbsp;10</a>).</p>
</div>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-zi-plot" class="quarto-float quarto-figure quarto-figure-center" data-fignum="10" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-zi-plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;10</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Visualization of predicted probablites for zero-inflated part of model</p>
</div>
</figcaption>
<div aria-describedby="fig-zi-plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-zi-plot-1.png" class="img-fluid figure-img" width="768">
</div>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="zero-one-inflated-beta-zoib" class="level2">
<h2 data-anchor-id="zero-one-inflated-beta-zoib">Zero-One-Inflated Beta (ZOIB)</h2>
<p>The ZIB model works well if you have 0s in your data, but not 1s.<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a> In our previous example, we nudged our 1s to .99. <span class="citation" data-cites="kubinec2022">Kubinec (<a href="#ref-kubinec2022" role="doc-biblioref">2022</a>)</span> showed that this practice can result in serious distortion of the outcome as the sample size 𝑁 grows larger, resulting in ever smaller values that are “nudged”. Because the Beta distribution is a non-linear model of the outcome, values that are very close to the boundary, such as 0.00001 or 0.99999, will be highly influential outliers. Sometimes it is theoretically useful to model both zeros and ones as separate processes or to consider these values as essentially similar parts of the continuous response, as we show later in the ordered Beta regression model. For example, this is important in visual analog scale data where there might be a prevalence of responses at the bounds <span class="citation" data-cites="kong2016">(<a href="#ref-kong2016" role="doc-biblioref">Kong &amp; Edwards, 2016</a>)</span>, in JOL tasks <span class="citation" data-cites="wilford2020">(<a href="#ref-wilford2020" role="doc-biblioref">Wilford et al., 2020</a>)</span>, or in a free-list task where individuals provide open responses to some question or topic which are then recoded to fall between 0-1 <span class="citation" data-cites="bendixen2023">(<a href="#ref-bendixen2023" role="doc-biblioref">Bendixen &amp; Purzycki, 2023</a>)</span>. Here 0s and 1s are meaningful; 0 means item was not listed and 1 means the item was listed first.</p>
<p>In our data, we have exactly one value equal to 1<a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a>. While probably not significant to alter our findings, we can model ones with a special type of model called the zero-one-inflated beta (ZOIB) model <span class="citation" data-cites="liu2015">(<a href="#ref-liu2015" role="doc-biblioref">Liu &amp; Kong, 2015</a>)</span> if we believe that both 0s and 1s are distinct outcomes.</p>
<p>Similar to our beta and zero-inflated models above, we can fit a ZOIB model in brms quite easily using the <code>zero_one_inflated_beta</code> family. In this model, we simultaneously estimate the mean (<span class="math inline">\mu</span>) and precision (<span class="math inline">\phi</span>) of the Beta distribution, a zero-one inflation parameter (<span class="math inline">\alpha</span>) that represents the probability that an observation is either exactly 0 or 1 (i.e., 0 or 1 vs.&nbsp;not 0 or 1) and a conditional one-inflation parameter (<span class="math inline">\gamma</span>) that represents the probability that, given an observation is at one of the endpoints, it is 1 (i.e, 1 vs.&nbsp;not 1). This specification captures the entire range of possible values while remaining constrained between 0 and 1. To get a better sense of how <span class="math inline">\alpha</span> and <span class="math inline">\gamma</span> control the distribution of values, <a href="#fig-zoib" class="quarto-xref" aria-expanded="false">Figure&nbsp;11</a> presents simulated data across combinations of these parameters. As <span class="math inline">\alpha</span> increases, we see a greater proportion of responses at the endpoints. As <span class="math inline">\gamma</span> increases, the proportion of endpoint responses at 1 grows relative to 0, making the spikes at 1 more prominent as <span class="math inline">\gamma</span> approaches 1. This visualization illustrates how the ZOIB model flexibly accounts for both the continuous portion of the distribution and the occurrence of exact 0s and 1s.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-zoib" class="quarto-float quarto-figure quarto-figure-center" data-fignum="11" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-zoib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;11</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Simulated data from a ZOIB model illustrating the effects of the zero-one inflation parameter (<span class="math inline">\alpha</span>) and the conditional one-inflation parameter (<span class="math inline">\gamma</span>).</p>
</div>
</figcaption>
<div aria-describedby="fig-zoib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-zoib-1.png" class="img-fluid figure-img" width="768">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>To fit a ZOIB model we use the <code>bf()</code> function. We model each parameter as a function of Fluency.</p>
</div>
<div class="cell">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fit the zoib model</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>zoib_model <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># The mean of the 0-1 values, or mu</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency, <span class="co"># The precision of the 0-1 values, or phi</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  zoi <span class="sc">~</span> Fluency, <span class="co"># The zero-or-one-inflated part, or alpha</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  coi <span class="sc">~</span> Fluency, <span class="co"># The one-inflated part, conditional on the 0s, or gamma</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">zero_one_inflated_beta</span>()</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>We then pass the <code>zoib_model</code> to our <code>brm()</code> function. The summary of the output is in <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a> (under <code>ZOIB</code>).</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># run the zoib mode using brm</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>fit_zoib <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> zoib_model,</span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">chains =</span> <span class="dv">4</span>,</span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">backend =</span> <span class="st">"cmdstanr"</span>,</span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"model_beta_zoib_1"</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="model-parameters-1" class="level3">
<h3 data-anchor-id="model-parameters-1">Model Parameters</h3>
<p>The output for the model is pretty lengthy we are estimating four parameters each with their own independent responses and sub-models. All the coefficients are on the logit scale, except <span class="math inline">\phi</span> , which is on the log scale. Thankfully drawing inferences for all these different parameters, plotting their distributions, and estimating their average marginal effects looks exactly the same—all the <code>brms</code> and <code>marginaleffects</code> functions we used work the same.</p>
</section>
<section id="predictions-and-marginal-effects" class="level3">
<h3 data-anchor-id="predictions-and-marginal-effects">Predictions and Marginal Effects</h3>
<p>With <code>marginaleffects</code> we can choose marginalize over all the sub-models, averaged across the 0s, continuous responses, and 1s in the data, or we can model the parameters separately using the <code>dpar</code> argument like we did above setting it to: <span class="math inline">\mu</span>, <span class="math inline">\phi</span>, <span class="math inline">\alpha</span>, <span class="math inline">\gamma</span> (see below). Using <code>avg_predictions()</code> and not setting <code>dpar</code> we can get the predicted probabilities across all the sub-models. We can also plot the overall difference between fluency and disfluency for the whole model with <code>plot_predictions()</code>.</p>
<p>We wont highlight all the parameters for this model <span class="math inline">\mu</span>, <span class="math inline">\phi</span> and <span class="math inline">\alpha</span> are the same as above, but below I show how one can extract the predicted probabilities and marginal effects for <span class="math inline">\gamma</span> .</p>
</section>
<section id="gamma" class="level3">
<h3 data-anchor-id="gamma"><span class="math inline">\gamma</span></h3>
</section>
</section>
<section id="ordered-beta-regression" class="level2">
<h2 data-anchor-id="ordered-beta-regression">Ordered Beta Regression</h2>
<p>Looking at the output from the ZOIB model (<a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a>), we can see how running a model like this can become vastly complex and computational intensive as it is fitting sub-models for each parameter. The ability to consider 0s and 1s as distinct processes from continuous values comes at a price in terms of complexity. A special version of the ZOIB was recently developed called ordered beta regression <span class="citation" data-cites="kubinec2022">(<a href="#ref-kubinec2022" role="doc-biblioref">Kubinec, 2022</a>)</span>. The ordered beta regression model allows for the analysis of continuous data (between 0-1) and discrete outcomes (e.g., 0 or 1) without requiring that either be fully distinct from the other. In the simplest sense, the ordered beta regression model is a hybrid model that estimates a weighted combination of a beta regression model for continuous responses and a logit model for the discrete values of the response.</p>
<p>The weights that average together the two parts of the outcome (i.e., discrete and continuous) are determined by cutpoints that are estimated in conjunction with the data in a similar manner to what is known as an ordered logit model. An in-depth explanation of ordinal regression is beyond the scope of this tutorial <span class="citation" data-cites="Fullerton2023 bürkner2019">(<a href="#ref-bürkner2019" role="doc-biblioref">Bürkner &amp; Vuorre, 2019</a>; but see <a href="#ref-Fullerton2023" role="doc-biblioref">Fullerton &amp; Anderson, 2021</a>)</span>. At a basic level, ordinal regression models are useful for outcome variables that are categorical in nature and have some inherent ordering (e.g., Likert scale items). To preserve this ordering, ordinal models rely on the cumulative probability distribution. Within an ordinal regression model it is assumed that there is a continuous but unobserved latent variable that determines which of <span class="math inline">k</span> ordinal responses will be selected. For example on a typical Likert scale from ‘Strongly Disagree’ to ‘Strongly Agree’, you could assume that there is a continuous, unobserved variable called ‘Agreement’. While we cannot measure Agreement directly, the ordinal response gives us some indication about where participants are on the continuous Agreement scale. <span class="math inline">k - 1</span> cutoffs are then estimated to indicate the point on the continuous Agreement scale at which your Agreement level is high enough to push you into the next ordinal category (say Agree to Strongly Agree). Coefficients in the model estimate how much different predictors change the estimated <em>continuous</em> scale (here, Agreement). Since there’s only one underlying process, there’s only one set of coefficients to work with (proportional odds assumption). In an ordered beta regression, three ordered categories are modeled: (1) exactly zero, (2) somewhere between zero and one, and (3) exactly one. In an ordered beta regression, (1) and (2) are modeled with cumulative logits, where one cutpoint is the the boundary between Exactly 0 and Between 0 and 1 and the other cutpoint is the boundary between <em>Between 0 and 1</em> and <em>Exactly 1.</em> Somewhere between 0-1 (3) is modeled as a beta regression with parameters reflecting the mean response on the logit scale. Ultimately, employing cutpoints allows for a smooth transition between the bounds and the continuous values, permitting both to be considered together rather than modeled separately as the ZOIB requires.</p>
<p>The ordered beta regression model has shown to be more efficient and less biased than some of the methods discussed <span class="citation" data-cites="kubinec2022">(<a href="#ref-kubinec2022" role="doc-biblioref">Kubinec, 2022</a>)</span> herein and has seen increasing usage across the biomedical and social sciences <span class="citation" data-cites="wilkesQuantifyingCoextinctionsEcosystem2024 martinVocalComplexitySocially2024 smithEcologicalMomentaryAssessment2024 shresthaBigFiveTraits2024 nouvianGlyphosateImpairsAversive2023">(<a href="#ref-martinVocalComplexitySocially2024" role="doc-biblioref">Martin et al., 2024</a>; <a href="#ref-nouvianGlyphosateImpairsAversive2023" role="doc-biblioref">Nouvian et al., 2023</a>; <a href="#ref-shresthaBigFiveTraits2024" role="doc-biblioref">Shrestha et al., 2024</a>; <a href="#ref-smithEcologicalMomentaryAssessment2024" role="doc-biblioref">Smith et al., 2024</a>; <a href="#ref-wilkesQuantifyingCoextinctionsEcosystem2024" role="doc-biblioref">Wilkes et al., 2024</a>)</span> because it produces only a single set of coefficient estimates in a similar manner to a standard beta regression or OLS.</p>
<section id="fitting-an-ordered-beta-regression" class="level3">
<h3 data-anchor-id="fitting-an-ordered-beta-regression">Fitting an Ordered Beta Regression</h3>
<p>To fit an ordered Beta regression in a Bayesian context we use the <code>ordbetareg</code> <span class="citation" data-cites="ordbetareg">(<a href="#ref-ordbetareg" role="doc-biblioref">Kubinec, 2023</a>)</span> package. <code>ordbetareg</code> is a front-end to the <code>brms</code> package that we described earlier; in addition to the functions available in the package, most <code>brms</code> functions and plots, including the diverse array of regression modeling options, will work with <code>ordbetareg</code> models.</p>
<p>We first load the <code>ordbetareg</code> package. You can download it from CRAN or from here: <a href="https://github.com/saudiwin/ordbetareg_pack" class="uri">https://github.com/saudiwin/ordbetareg_pack</a>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># load ordbetareg package need to have it installed</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ordbetareg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>The <code>ordbetareg</code> package uses <code>brms</code> on the front-end so all the arguments we used previously apply here. Instead of the <code>brm()</code> function we use <code>ordbetareg()</code>.</p>
<p>To fit a model where dispersion does not vary as a function of fluency we can use the below code.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co"># use ordbetareg to fit model</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>ord_fit_brms <span class="ot">&lt;-</span> <span class="fu">ordbetareg</span>(</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency,</span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">chains =</span> <span class="dv">4</span>,</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">backend =</span> <span class="st">"cmdstanr"</span>,</span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"model_beta_ordbeta"</span></span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>However, if we want dispersion to vary as a function of fluency we can easily do that. Note the addition of the <code>phi_reg</code> argument in <code>m.phi</code>. This argument allows us to include a model that explicitly models the dispersion parameter. Because we are modeling <span class="math inline">\phi</span> as a function of fluency, we set the the argument to <code>both</code>.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>m.phi <span class="ot">&lt;-</span> <span class="fu">ordbetareg</span>(</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>  ord_beta_phi,</span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb18-4"><a href="#cb18-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">cores =</span> <span class="dv">4</span>,</span>
<span id="cb18-5"><a href="#cb18-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">chains =</span> <span class="dv">4</span>,</span>
<span id="cb18-6"><a href="#cb18-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">backend =</span> <span class="st">"cmdstanr"</span>,</span>
<span id="cb18-7"><a href="#cb18-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> <span class="st">"model_beta_ordbeta_phi"</span>,</span>
<span id="cb18-8"><a href="#cb18-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">phi_reg =</span> <span class="st">'both'</span> <span class="co"># log phi</span></span>
<span id="cb18-9"><a href="#cb18-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="marginal-effects-1" class="level2">
<h2 data-anchor-id="marginal-effects-1">Marginal effects</h2>
<p><a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a> presents the overall model summary (under <code>Ordered Beta</code>). We can use <code>marginaleffects</code> to calculate differences on the response scale that average over (or marginalize over) all our parameters.</p>
<div class="cell FigureWithoutNote" prefix="" data-tblnum="11" data-custom-style="FigureWithoutNote">
<div id="tbl-ordbeta" class="cell quarto-float quarto-figure quarto-figure-center" data-tblnum="11" prefix="">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl quarto-uncaptioned" id="tbl-ordbeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;11
</figcaption>
<div aria-describedby="tbl-ordbeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_i6v4jsvxvdz33z2trf3x(i, j, css_id) {
          var table = document.getElementById("tinytable_i6v4jsvxvdz33z2trf3x");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_i6v4jsvxvdz33z2trf3x');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_i6v4jsvxvdz33z2trf3x(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_i6v4jsvxvdz33z2trf3x");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 1, j: 0 }, { i: 1, j: 1 }, { i: 1, j: 2 }, { i: 1, j: 3 },  ], css_id: 'tinytable_css_ldpjdkeyopzbu1t6vddu',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 }, { i: 0, j: 3 },  ], css_id: 'tinytable_css_2e8ywk9yl7i4qcv1qimu',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_i6v4jsvxvdz33z2trf3x(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_ldpjdkeyopzbu1t6vddu, .table th.tinytable_css_ldpjdkeyopzbu1t6vddu { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_2e8ywk9yl7i4qcv1qimu, .table th.tinytable_css_2e8ywk9yl7i4qcv1qimu { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_i6v4jsvxvdz33z2trf3x" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Term</th>
                <th scope="col">Contrast</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>Fluency</td>
                  <td>Fluent - Disfluent</td>
                  <td>0.061</td>
                  <td>[-0.014, 0.138]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>In <a href="#tbl-ordbeta" class="quarto-xref" aria-expanded="false">Table&nbsp;11</a> the credible interval is close enough to zero relative to its uncertainty that we can conclude there likely aren’t substantial differences between the conditions after taking dispersion and the zeros and ones in our data into account.</p>
</div>
<section id="cutpoints" class="level3">
<h3 data-anchor-id="cutpoints">Cutpoints</h3>
<p>The model cutpoints are not reported by default in the summary output, but we can access them with the R package <code>posterior</code> and the functions <code>as_draws</code> and <code>summary_draws</code>.</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="12" data-custom-style="FigureWithoutNote">
<div id="tbl-cutpoints" class="cell quarto-float quarto-figure quarto-figure-center" prefix="" data-after-caption-space="0pt" data-tblnum="12">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-cutpoints-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;12</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Cutzero and cutone parameter summary</p>
</div>
</figcaption>
<div aria-describedby="tbl-cutpoints-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script>

      function styleCell_o93f0zmp6486kf10cw7e(i, j, css_id) {
          var table = document.getElementById("tinytable_o93f0zmp6486kf10cw7e");
          var cell = table.rows[i]?.cells[j];  // Safe navigation to avoid errors
          if (cell) {
              console.log(`Styling cell at (${i}, ${j}) with class ${css_id}`);
              cell.classList.add(css_id);
          } else {
              console.warn(`Cell at (${i}, ${j}) not found.`);
          }
      }
      function insertSpanRow(i, colspan, content) {
        var table = document.getElementById('tinytable_o93f0zmp6486kf10cw7e');
        var newRow = table.insertRow(i);
        var newCell = newRow.insertCell(0);
        newCell.setAttribute("colspan", colspan);
        // newCell.innerText = content;
        // this may be unsafe, but innerText does not interpret <br>
        newCell.innerHTML = content;
      }
      function spanCell_o93f0zmp6486kf10cw7e(i, j, rowspan, colspan) {
        var table = document.getElementById("tinytable_o93f0zmp6486kf10cw7e");
        const targetRow = table.rows[i];
        const targetCell = targetRow.cells[j];
        for (let r = 0; r < rowspan; r++) {
          // Only start deleting cells to the right for the first row (r == 0)
          if (r === 0) {
            // Delete cells to the right of the target cell in the first row
            for (let c = colspan - 1; c > 0; c--) {
              if (table.rows[i + r].cells[j + c]) {
                table.rows[i + r].deleteCell(j + c);
              }
            }
          }
          // For rows below the first, delete starting from the target column
          if (r > 0) {
            for (let c = colspan - 1; c >= 0; c--) {
              if (table.rows[i + r] && table.rows[i + r].cells[j]) {
                table.rows[i + r].deleteCell(j);
              }
            }
          }
        }
        // Set rowspan and colspan of the target cell
        targetCell.rowSpan = rowspan;
        targetCell.colSpan = colspan;
      }
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: 2, j: 0 }, { i: 2, j: 1 }, { i: 2, j: 2 },  ], css_id: 'tinytable_css_kmtjn5voxxhphrpbjbrp',}, 
          { positions: [ { i: 0, j: 0 }, { i: 0, j: 1 }, { i: 0, j: 2 },  ], css_id: 'tinytable_css_bsd14vxeyzy5ivxl8nym',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  styleCell_o93f0zmp6486kf10cw7e(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <style>
      /* tinytable css entries after */
      .table td.tinytable_css_kmtjn5voxxhphrpbjbrp, .table th.tinytable_css_kmtjn5voxxhphrpbjbrp { border-bottom: solid #d3d8dc 0.1em; }
      .table td.tinytable_css_bsd14vxeyzy5ivxl8nym, .table th.tinytable_css_bsd14vxeyzy5ivxl8nym { border-top: solid #d3d8dc 0.1em; border-bottom: solid #d3d8dc 0.05em; }
    </style>
    <div class="container">
      <table class="table table-borderless" id="tinytable_o93f0zmp6486kf10cw7e" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        <thead>
        
              <tr>
                <th scope="col">Parameter</th>
                <th scope="col">Mean</th>
                <th scope="col">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td>cutzero</td>
                  <td>-2.98</td>
                  <td>[-3.58, -2.42]</td>
                </tr>
                <tr>
                  <td>cutone</td>
                  <td>1.85</td>
                  <td>[1.65, 2.07]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>In <a href="#tbl-cutpoints" class="quarto-xref" aria-expanded="false">Table&nbsp;12</a>, <code>cutzero</code> is the first cutpoint (the difference between 0 and continuous values) and <code>cutone</code> is the second cutpoint (the difference between the continuous values and 1). These cutpoints are on the logit scale and as such the numbers do not have a simple substantive meaning. In general, as the cutpoints increase in absolute value (away from zero), then the discrete/boundary observations are more distinct from the continuous values. This will happen if there is a clear gap or bunching in the outcome around the bounds. This type of empirical feature of the distribution may be useful to scholars if they want to study differences in how people perceive the ends of the scale versus the middle.</p>
</div>
</section>
<section id="ordered-beta-regression-model-fit" class="level3">
<h3 data-anchor-id="ordered-beta-regression-model-fit">Ordered Beta Regression Model Fit</h3>
<p>The best way to visualize model fit is to plot the full predictive distribution relative to the original outcome. Because ordered beta regression is a mixed discrete/continuous model, a separate plotting function, <code>pp_check_ordbetareg</code>, is included in the <code>ordbetareg</code> package that accurately handles the unique features of this distribution. The default plot in <code>brms</code> will collapse these two features of the outcome together, which will make the fit look worse than it actually is. The <code>ordbetareg</code> function returns a list with two plots, <code>discrete</code> and <code>continuous</code>, which can either be printed and plotted or further modified as <code>ggplot2</code> objects (see <a href="#fig-ppcheckord" class="quarto-xref" aria-expanded="false">Figure&nbsp;12</a>).</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-ppcheckord" class="quarto-float quarto-figure quarto-figure-center" data-fignum="12" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-ppcheckord-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;12</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Posterior predictive check for ordered beta regression model. A. Discrete posterior check. B. Continuous posterior check.</p>
</div>
</figcaption>
<div aria-describedby="fig-ppcheckord-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-ppcheckord-1.png" class="img-fluid figure-img" width="1152">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>The discrete plot which is a bar graph, shows that the posterior distribution accurately captures the number of different types of responses (discrete or continuous) in the data. For the continuous plot shown as a density plot with one line per posterior draw, the model does a very good job at capturing the distribution.</p>
</div>
<p>Overall, it is clear from the posterior distribution plot that the ordered beta model fits the data well. To fully understand model fit, both of these plots need to be inspected as they are conceptually distinct.</p>
<section id="model-visualization" class="level4">
<h4 data-anchor-id="model-visualization">Model Visualization.</h4>
<p><code>ordbetareg</code> provides a visualization function called <code>plot_hess</code>. This function produces a plot of predicted proportions across the range of our <code>Fluency</code> factor. In <a href="#fig-heiss" class="quarto-xref" aria-expanded="false">Figure&nbsp;13</a> we get predicted proportions for Fluency across the bounded scale. Looking at the figure we can see there is much overlap between instructors in the middle portion (<span class="math inline">\mu</span>) . However, we do see some small differences at the zero bounds.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-heiss" class="quarto-float quarto-figure quarto-figure-center" data-fignum="13" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-heiss-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;13</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Heiss Plot of Predicted Probablities across the scale (0-100)</p>
</div>
</figcaption>
<div aria-describedby="fig-heiss-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-heiss-1.png" class="img-fluid figure-img" width="768">
</div>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="ordered-beta-scale" class="level3">
<h3 data-anchor-id="ordered-beta-scale">Ordered Beta Scale</h3>
<p>In the <code>ordbetareg</code> function there is a <code>true_bound</code> argument. In the case where you data in not bounded between 0-1, you can use the argument to specify the bounds of the argument to fit the ordered beta regression. For example, you data might be bounded between 1 and 7. If so, you can model it as such.</p>
</section>
</section>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
<p>The use of Beta regression in psychology, and the social sciences in general, is rare. With this tutorial, we hope to turn the tides. Beta regression models are an attractive alternative to models that impose unrealistic assumptions like normality, linearity, homoscedasticity, and unbounded data. Beyond these models, there are a diverse array of different models that can be used depending on your outcome of interest.</p>
<p>Throughout this tutorial our main aim was to help guide researchers in running analyses with proportional or percentage outcomes using Beta regression and some of it’s alternatives. In the current example, we used real data from <span class="citation" data-cites="wilford2020">Wilford et al. (<a href="#ref-wilford2020" role="doc-biblioref">2020</a>)</span> and discussed how to fit these models in R, interpret model parameters, extract predicted probabilities and marginal effects, and visualize the results.</p>
<p>Comparing our analysis with that of <span class="citation" data-cites="wilford2020">Wilford et al. (<a href="#ref-wilford2020" role="doc-biblioref">2020</a>)</span>, we demonstrated that using traditional approaches (e.g., <em>t</em>-tests) to analyze accuracy data can lead to inaccurate inferences. Although we successfully reproduced one of their key findings, our use of Beta regression and its extensions revealed important nuances in the results. With a traditional Beta regression model—which accounts for both the mean and the precision (dispersion)—we observed similar effects of instructor fluency on performance. However, the standard Beta model does not accommodate boundary values (i.e., 0s and 1s).</p>
<p>When we applied a zero-inflated Beta regression model, which explicitly accounts for structural zeros, we found no effect of fluency on the mean (<span class="math inline">\mu</span>) part of the model. Instead, the effect of fluency emerged in the structural zero (inflated zero; <span class="math inline">\alpha</span>) component. This pattern was consistent when using a zero-one-inflated Beta (ZOIB) model. Furthermore, we fit an ordered Beta regression model <span class="citation" data-cites="kubinec2022">(<a href="#ref-kubinec2022" role="doc-biblioref">Kubinec, 2022</a>)</span>, which appropriately models the full range of values, including 0s and 1s. Here, we did not observe a reliable effect of fluency on the mean once we accounted for dispersion.</p>
<p>These analyses emphasize the importance of fitting a model that aligns with the nature of the data. The simplest and recommended approach when dealing with data that contains zeros and/or ones is to fit an ordered Beta model, assuming the process is truly continuous. However, if you believe the process is distinct in nature, a ZIB or ZOIB model might be a better choice. Ultimately, this decision should be guided by theory.</p>
<p>For instance, if we believe fluency influences the structural zero part of the model, we might want to model this process separately using a ZIB or ZOIB. With the current dataset, fluency might affect specific aspects of performance (such as the likelihood of complete failure) rather than general performance levels. This effect could be due to participant disengagement during the disfluent lecture. If students fail to pay attention because of features of disfluency, they may miss relevant information, leading to a floor effect at the test. If this is the case, we would want to model this appropriately. However, if we believe fluency effects general performance levels, a model that takes in to account the entire process accounting for the zeros and ones might be appropriate.</p>
<p>In the discussion section of <span class="citation" data-cites="wilford2020">Wilford et al. (<a href="#ref-wilford2020" role="doc-biblioref">2020</a>)</span>, they were unable to offer a tenable explanation for performance differences based on instructor fluency. A model that accounts for the excess zeros in the dataset provides one testable explanation: watching a disfluent lecture may lead to lapses in attention, resulting in poorer performance in that group. These lapses, in turn, contribute to the observed differences in the fluent condition. This modeling approach opens a promising avenue for future research—one that would have remained inaccessible otherwise.</p>
<p>Not everyone will be eager to implement the techniques discussed herein. In such cases, the key question becomes: What is the least problematic approach to handling proportional data? One reasonable option is to fit multiple models tailored to the specific characteristics of your data. For example, if your data contain zeros, you might fit two models: a traditional OLS regression excluding the zeros, and a logistic model to account for the zero versus non-zero distinction. If your data contain both zeros and ones, you could fit separate models for the zeros and ones in addition to the OLS model. There are many defensible strategies to choose from depending on the context. However, we do not recommend transforming the values of your data (e.g., 0s to .01 and 1s to .99) or ignoring the properties of your data simply to fit traditional statistical models.</p>
<p>In this tutorial, we demonstrated how to analyze these models from a Bayesian perspective. While we recognize that not everyone identifies as a Bayesian, implementing these models using a Bayesian framework is relatively straightforward—it requires only a single package, lowering the barrier to entry. For those who prefer frequentist analyses, several R packages are available. For standard beta regression, the <code>betareg</code> package <span class="citation" data-cites="betareg">(<a href="#ref-betareg" role="doc-biblioref">Cribari-Neto &amp; Zeileis, 2010</a>)</span> is a solid option, while more complex models such as zero-inflated and ordered beta regressions can be implemented using <code>glmmTMB</code> <span class="citation" data-cites="glmmTMB">(<a href="#ref-glmmTMB" role="doc-biblioref">Brooks et al., 2017</a>)</span>. For fitting zero-one models, there is a new implementation in <span class="citation" data-cites="betareg">Cribari-Neto and Zeileis (<a href="#ref-betareg" role="doc-biblioref">2010</a>)</span>, that allows you to model these types of data.</p>
<section id="conclusion" class="level2">
<h2 data-anchor-id="conclusion">Conclusion</h2>
<p>Overall, this tutorial emphasizes the importance of modeling the data you have. Although the example provided is relatively simple (a one-factor model with two levels), we hope it demonstrates that even with a basic dataset, there is much nuance in interpretation and inference. Properly modeling your data can lead to deeper insights, far beyond what traditional measures might offer. With the tools introduced in this tutorial, researchers now have the means to analyze their data effectively, uncover patterns, make accurate predictions, and support their findings with robust statistical evidence. By applying these modeling techniques, researchers can improve the validity and reliability of their studies, ultimately leading to more informed decisions and advancements in their respective fields.</p>
</section>
<section id="references" class="level2">
<h2 data-anchor-id="references">References</h2>
<!-- References will auto-populate in the refs div below -->
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list">
<div id="ref-marginaleffects" class="csl-entry" role="listitem">
Arel-Bundock, V. (2024). <em>Marginaleffects: Predictions, comparisons, slopes, marginal means, and hypothesis tests</em>. <a href="https://CRAN.R-project.org/package=marginaleffects">https://CRAN.R-project.org/package=marginaleffects</a>
</div>
<div id="ref-margeffects2024" class="csl-entry" role="listitem">
Arel-Bundock, V., Greifer, N., &amp; Heiss, A. (2024). How to interpret statistical models using <span class="nocase">marginaleffects</span> for <span>R</span> and <span>Python</span>. <em>Journal of Statistical Software</em>, <em>111</em>(9), 1–32. <a href="https://doi.org/10.18637/jss.v111.i09">https://doi.org/10.18637/jss.v111.i09</a>
</div>
<div id="ref-bartlett1936" class="csl-entry" role="listitem">
Bartlett, M. S. (1936). The Square Root Transformation in Analysis of Variance. <em>Journal of the Royal Statistical Society Series B: Statistical Methodology</em>, <em>3</em>(1), 68–78. <a href="https://doi.org/10.2307/2983678">https://doi.org/10.2307/2983678</a>
</div>
<div id="ref-bendixen2023" class="csl-entry" role="listitem">
Bendixen, T., &amp; Purzycki, B. G. (2023). Cognitive and cultural models in psychological science: A tutorial on modeling free-list data as a dependent variable in Bayesian regression. <em>Psychological Methods</em>. <a href="https://doi.org/10.1037/met0000553">https://doi.org/10.1037/met0000553</a>
</div>
<div id="ref-glmmTMB" class="csl-entry" role="listitem">
Brooks, M. E., Kristensen, K., van, K. J., Magnusson, A., Berg, C. W., Nielsen, A., Skaug, H. J., Maechler, M., &amp; Bolker, B. M. (2017). <em><span></span>glmmTMB<span></span> balances speed and flexibility among packages for zero-inflated generalized linear mixed modeling</em>. <em>9</em>. <a href="https://doi.org/10.32614/RJ-2017-066">https://doi.org/10.32614/RJ-2017-066</a>
</div>
<div id="ref-brms" class="csl-entry" role="listitem">
Bürkner, P.-C. (2017). <em><span></span>Brms<span></span>: An <span></span>r<span></span> package for <span></span>bayesian<span></span> multilevel models using <span></span>stan<span></span></em>. <em>80</em>. <a href="https://doi.org/10.18637/jss.v080.i01">https://doi.org/10.18637/jss.v080.i01</a>
</div>
<div id="ref-bürkner2019" class="csl-entry" role="listitem">
Bürkner, P.-C., &amp; Vuorre, M. (2019). Ordinal Regression Models in Psychology: A Tutorial. <em>Advances in Methods and Practices in Psychological Science</em>, <em>2</em>(1), 77–101. <a href="https://doi.org/10.1177/2515245918823199">https://doi.org/10.1177/2515245918823199</a>
</div>
<div id="ref-carpenter2013" class="csl-entry" role="listitem">
Carpenter, S. K., Wilford, M. M., Kornell, N., &amp; Mullaney, K. M. (2013). Appearances can be deceiving: instructor fluency increases perceptions of learning without increasing actual learning. <em>Psychonomic Bulletin &amp; Review</em>, <em>20</em>(6), 1350–1356. <a href="https://doi.org/10.3758/s13423-013-0442-z">https://doi.org/10.3758/s13423-013-0442-z</a>
</div>
<div id="ref-cohen1977" class="csl-entry" role="listitem">
Cohen, J. (1977). <em>Statistical power analysis for the behavioral sciences, rev. ed.</em> Lawrence Erlbaum Associates, Inc.
</div>
<div id="ref-coretta2025bayesian" class="csl-entry" role="listitem">
Coretta, S., &amp; Bürkner, P.-C. (2025). <em>Bayesian beta regressions with brms in r: A tutorial for phoneticians</em>. <a href="https://doi.org/10.31219/osf.io/f9rqg_v1" class="uri">https://doi.org/10.31219/osf.io/f9rqg_v1</a>.
</div>
<div id="ref-betareg" class="csl-entry" role="listitem">
Cribari-Neto, F., &amp; Zeileis, A. (2010). <em>Beta regression in <span></span>r<span></span></em>. <em>34</em>. <a href="https://doi.org/10.18637/jss.v034.i02">https://doi.org/10.18637/jss.v034.i02</a>
</div>
<div id="ref-nix" class="csl-entry" role="listitem">
Dolstra, E., &amp; contributors, T. N. (2006). <em>Nix</em> [Computer software]. <a href="https://nixos.org/">https://nixos.org/</a>
</div>
<div id="ref-ferrari2004" class="csl-entry" role="listitem">
Ferrari, S., &amp; Cribari-Neto, F. (2004). Beta Regression for Modelling Rates and Proportions. <em>Journal of Applied Statistics</em>, <em>31</em>(7), 799–815. <a href="https://doi.org/10.1080/0266476042000214501">https://doi.org/10.1080/0266476042000214501</a>
</div>
<div id="ref-Fullerton2023" class="csl-entry" role="listitem">
Fullerton, A. S., &amp; Anderson, K. F. (2021). Ordered Regression Models: a Tutorial. <em>Prevention Science</em>, <em>24</em>(3), 431–443. <a href="https://doi.org/10.1007/s11121-021-01302-y">https://doi.org/10.1007/s11121-021-01302-y</a>
</div>
<div id="ref-cmdstanr" class="csl-entry" role="listitem">
Gabry, J., Češnovar, R., Johnson, A., &amp; Bronder, S. (2024). <em>Cmdstanr: R interface to ’CmdStan’</em>. <a href="https://mc-stan.org/cmdstanr/">https://mc-stan.org/cmdstanr/</a>
</div>
<div id="ref-gelman2013bayesian" class="csl-entry" role="listitem">
Gelman, A., Carlin, J. B., Stern, H. S., Dunson, D. B., Vehtari, A., &amp; Rubin, D. B. (2013). <em>Bayesian data analysis</em> (Third). CRC. <a href="https://stat.columbia.edu/~gelman/book/">https://stat.columbia.edu/~gelman/book/</a>
</div>
<div id="ref-heiss2021" class="csl-entry" role="listitem">
Heiss, A. (2021). <em>A guide to modeling proportions with bayesian beta and zero-inflated beta regression models</em>. <a href="http://dx.doi.org/10.59350/7p1a4-0tw75">http://dx.doi.org/10.59350/7p1a4-0tw75</a>
</div>
<div id="ref-bayes_rules" class="csl-entry" role="listitem">
Johnson, A., Ott, M., &amp; Dogucu, M. (n.d.). <em>Bayes rules!: An introduction to applied bayesian modeling</em>. Routledge &amp; CRC Press.
</div>
<div id="ref-kong2016" class="csl-entry" role="listitem">
Kong, E. J., &amp; Edwards, J. (2016). Individual differences in categorical perception of speech: Cue weighting and executive function. <em>Journal of Phonetics</em>, <em>59</em>, 40–57. <a href="https://doi.org/10.1016/j.wocn.2016.08.006">https://doi.org/10.1016/j.wocn.2016.08.006</a>
</div>
<div id="ref-kubinec2022" class="csl-entry" role="listitem">
Kubinec, R. (2022). Ordered Beta Regression: A Parsimonious, Well-Fitting Model for Continuous Data with Lower and Upper Bounds. <em>Political Analysis</em>, <em>31</em>(4), 519–536. <a href="https://doi.org/10.1017/pan.2022.20">https://doi.org/10.1017/pan.2022.20</a>
</div>
<div id="ref-ordbetareg" class="csl-entry" role="listitem">
Kubinec, R. (2023). <em>Ordbetareg: Ordered beta regression models with ’brms’</em>. <a href="https://CRAN.R-project.org/package=ordbetareg">https://CRAN.R-project.org/package=ordbetareg</a>
</div>
<div id="ref-liu2015" class="csl-entry" role="listitem">
Liu, F., &amp; Kong, Y. (2015). zoib: An R Package for Bayesian Inference for Beta Regression and Zero/One Inflated Beta Regression. <em>The R Journal</em>, <em>7</em>(2), 34. <a href="https://doi.org/10.32614/rj-2015-019">https://doi.org/10.32614/rj-2015-019</a>
</div>
<div id="ref-ggeffects-2" class="csl-entry" role="listitem">
Lüdecke, D. (2018). <em>Ggeffects: Tidy data frames of marginal effects from regression models.</em> <em>3</em>, 772. <a href="https://doi.org/10.21105/joss.00772">https://doi.org/10.21105/joss.00772</a>
</div>
<div id="ref-easystats" class="csl-entry" role="listitem">
Lüdecke, D., Ben-Shachar, M. S., Patil, I., Wiernik, B. M., Bacher, E., Thériault, R., &amp; Makowski, D. (2022). <em>Easystats: Framework for easy statistical modeling, visualization, and reporting</em>. <a href="https://easystats.github.io/easystats/">https://easystats.github.io/easystats/</a>
</div>
<div id="ref-makowski2019a" class="csl-entry" role="listitem">
Makowski, D., Ben-Shachar, M. S., Chen, S. H. A., &amp; Lüdecke, D. (2019). Indices of effect existence and significance in the bayesian framework. <em>Frontiers in Psychology</em>, <em>10</em>. <a href="https://doi.org/10.3389/fpsyg.2019.02767">https://doi.org/10.3389/fpsyg.2019.02767</a>
</div>
<div id="ref-makowski2019" class="csl-entry" role="listitem">
Makowski, D., Ben-Shachar, M., &amp; Lüdecke, D. (2019). bayestestR: Describing effects and their uncertainty, existence and significance within the bayesian framework. <em>Journal of Open Source Software</em>, <em>4</em>(40), 1541. <a href="https://doi.org/10.21105/joss.01541">https://doi.org/10.21105/joss.01541</a>
</div>
<div id="ref-martinVocalComplexitySocially2024" class="csl-entry" role="listitem">
Martin, K., Cornero, F. M., Clayton, N. S., Adam, O., Obin, N., &amp; Dufour, V. (2024). Vocal complexity in a socially complex corvid: Gradation, diversity and lack of common call repertoire in male rooks. <em>Royal Society Open Science</em>, <em>11</em>(1), 231713. <a href="https://doi.org/10.1098/rsos.231713">https://doi.org/10.1098/rsos.231713</a>
</div>
<div id="ref-mcelreath2020statistical" class="csl-entry" role="listitem">
McElreath, R. (2020). <em>Statistical rethinking: A bayesian course with examples in r and STAN</em> (2nd ed.). Chapman; Hall/CRC. <a href="https://doi.org/10.1201/9780429029608">https://doi.org/10.1201/9780429029608</a>
</div>
<div id="ref-nelder1972" class="csl-entry" role="listitem">
Nelder, J. A., &amp; Wedderburn, R. W. M. (1972). Generalized linear models. <em>Journal of the Royal Statistical Society. Series A (General)</em>, <em>135</em>(3), 370–384. <a href="https://doi.org/10.2307/2344614">https://doi.org/10.2307/2344614</a>
</div>
<div id="ref-nouvianGlyphosateImpairsAversive2023" class="csl-entry" role="listitem">
Nouvian, M., Foster, J. J., &amp; Weidenmüller, A. (2023). Glyphosate impairs aversive learning in bumblebees. <em>Science of The Total Environment</em>, <em>898</em>, 165527. <a href="https://www.sciencedirect.com/science/article/pii/S0048969723041505">https://www.sciencedirect.com/science/article/pii/S0048969723041505</a>
</div>
<div id="ref-paolino2001" class="csl-entry" role="listitem">
Paolino, P. (2001). Maximum Likelihood Estimation of Models with Beta-Distributed Dependent Variables. <em>Political Analysis</em>, <em>9</em>(4), 325–346. <a href="https://doi.org/10.1093/oxfordjournals.pan.a004873">https://doi.org/10.1093/oxfordjournals.pan.a004873</a>
</div>
<div id="ref-R" class="csl-entry" role="listitem">
R Core Team. (2024). <em>R: A language and environment for statistical computing</em>. R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>
</div>
<div id="ref-rix" class="csl-entry" role="listitem">
Rodrigues, B., &amp; Baumann, P. (2025). <em>Rix: Reproducible data science environments with ’nix’</em>. <a href="https://docs.ropensci.org/rix/">https://docs.ropensci.org/rix/</a>
</div>
<div id="ref-shresthaBigFiveTraits2024" class="csl-entry" role="listitem">
Shrestha, S., Sigdel, K., Pokharel, M., &amp; Columbus, S. (2024). Big five traits predict between- and within-person variation in loneliness. <em>European Journal of Personality</em>, 08902070241239834. <a href="https://doi.org/10.1177/08902070241239834">https://doi.org/10.1177/08902070241239834</a>
</div>
<div id="ref-sladekova2024unicorns" class="csl-entry" role="listitem">
Sladekova, M., &amp; Field, A. P. (2024). <em>In search of unicorns: Assessing statistical assumptions in real psychology datasets</em>. <a href="https://doi.org/10.31234/osf.io/4rznt">https://doi.org/10.31234/osf.io/4rznt</a>
</div>
<div id="ref-smithEcologicalMomentaryAssessment2024" class="csl-entry" role="listitem">
Smith, K. E., Panlilio, L. V., Feldman, J. D., Grundmann, O., Dunn, K. E., McCurdy, C. R., Garcia-Romeu, A., &amp; Epstein, D. H. (2024). Ecological momentary assessment of self-reported kratom use, effects, and motivations among US adults. <em>JAMA Network Open</em>, <em>7</em>(1), e2353401. <a href="https://doi.org/10.1001/jamanetworkopen.2023.53401">https://doi.org/10.1001/jamanetworkopen.2023.53401</a>
</div>
<div id="ref-smithson2006" class="csl-entry" role="listitem">
Smithson, M., &amp; Verkuilen, J. (2006). A better lemon squeezer? Maximum-likelihood regression with beta-distributed dependent variables. <em>Psychological Methods</em>, <em>11</em>(1), 54–71. <a href="https://doi.org/10.1037/1082-989x.11.1.54">https://doi.org/10.1037/1082-989x.11.1.54</a>
</div>
<div id="ref-stan2023" class="csl-entry" role="listitem">
Team, S. D. (2023). <em>Stan: A probabilistic programming language</em>. <a href="https://mc-stan.org">https://mc-stan.org</a>
</div>
<div id="ref-toftness2017" class="csl-entry" role="listitem">
Toftness, A. R., Carpenter, S. K., Geller, J., Lauber, S., Johnson, M., &amp; Armstrong, P. I. (2017). Instructor fluency leads to higher confidence in learning, but not better learning. <em>Metacognition and Learning</em>, <em>13</em>(1), 1–14. <a href="https://doi.org/10.1007/s11409-017-9175-0">https://doi.org/10.1007/s11409-017-9175-0</a>
</div>
<div id="ref-vuorre2019" class="csl-entry" role="listitem">
Vuorre, M. (2019, February 18). <em>How to Analyze Visual Analog (Slider) Scale Data?</em> <a href="https://vuorre.com/posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-beta-models">https://vuorre.com/posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-beta-models</a>
</div>
<div id="ref-wilford2020" class="csl-entry" role="listitem">
Wilford, M. M., Kurpad, N., Platt, M., &amp; Weinstein-Jones, Y. (2020). Lecturer fluency can impact students’ judgments of learning and actual learning performance. <em>Applied Cognitive Psychology</em>, <em>34</em>(6), 1444–1456. <a href="https://doi.org/10.1002/acp.3724">https://doi.org/10.1002/acp.3724</a>
</div>
<div id="ref-wilkesQuantifyingCoextinctionsEcosystem2024" class="csl-entry" role="listitem">
Wilkes, L. N., Barner, A. K., Keyes, A. A., Morton, D., Byrnes, J. E. K., &amp; Dee, L. E. (2024). Quantifying co‐extinctions and ecosystem service vulnerability in coastal ecosystems experiencing climate warming. <em>Global Change Biology</em>, <em>30</em>(7), e17422. <a href="https://doi.org/10.1111/gcb.17422">https://doi.org/10.1111/gcb.17422</a>
</div>
<div id="ref-witherby2022" class="csl-entry" role="listitem">
Witherby, A. E., &amp; Carpenter, S. K. (2022). The impact of lecture fluency and technology fluency on students<span>’</span> online learning and evaluations of instructors. <em>Journal of Applied Research in Memory and Cognition</em>, <em>11</em>(4), 500–509. <a href="https://doi.org/10.1037/mac0000003">https://doi.org/10.1037/mac0000003</a>
</div>
</div>
</section>
</section>


<div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p>In this article, we try to limit code where possible; however, the online version has all the code needed to reproduce all analyses herein. Furthermore, to promote transparency and reproduciblity, the tutorial was written in R version 4.4.3 (<span class="citation" data-cites="R">R Core Team (<a href="#ref-R" role="doc-biblioref">2024</a>)</span>) using Quarto (v.1.5.54), an open-source publishing system that allows for dynamic and static documents. This allows figures, tables, and text to be programmatically included directly in the manuscript, ensuring that all results are seamlessly integrated into the document. In addition, we use the <code>rix</code> <span class="citation" data-cites="rix">(<a href="#ref-rix" role="doc-biblioref">Rodrigues &amp; Baumann, 2025</a>)</span> R package which harnesses the power of the <code>nix</code> <span class="citation" data-cites="nix">(<a href="#ref-nix" role="doc-biblioref">Dolstra &amp; contributors, 2006</a>)</span> ecosystem to to help with computational reproducibility. Not only does this give us a snapshot of the packages used to create the current manuscript, but it also takes a snapshot of system dependencies used at run-time. This way reproducers can easily re-use the exact same environment by installing the <code>nix</code> package manager and using the included default.nix file to set up the right environment. The README file in the GitHub repository contains detailed information on how to set this up to reproduce the contents of the current manuscript, including a video.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>In order to use the <code>cmdstanr</code> backend you will need to first install the package and also run <code>cmdstanr::install_cmdstan()</code> if you have not done so already.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3"><p>We have chain diagnostics included like Rhat and ESS which indicates how the MCMC sampling performed. For more information check out Gelman et al., 2013; Kruschke, 2014; McElreath, 2020)<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4"><p><code>ggeffects</code> is another great package to extract marginal effects and plot <span class="citation" data-cites="ggeffects-2">(<a href="#ref-ggeffects-2" role="doc-biblioref">Lüdecke, 2018</a>)</span><a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5"><p>In cases where there are large amounts of 1s but no zeros you can fit a one-inflated model. Currently you can do so with the <code>gamlss</code> package in R<a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn6"><p>In cases where your data include exact 1s but no 0s, you can fit a one-inflated beta regression model in <code>brms</code> by setting the <code>coi</code> parameter to <code>1</code>. This tells the model that all point masses occur at 1, rather than being split between 0 and 1. In other words, <code>coi = 1</code> assumes that any inflation in the data is due entirely to values at 1.<a href="#fnref6" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>