<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.34">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jason Geller">
<meta name="author" content="Robert Kubinec">
<meta name="author" content="Chelsea M. Parlett Pelleriti">
<meta name="author" content="Matti Vuorre">
<meta name="keywords" content="beta regression, beta distribution, R tutorial, psychology, learning and memory">
<meta name="description" content="BETA REGRESSION TUTORIAL">

<title>A Beta Way: A Tutorial For Using Beta Regression in Psychological Research</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
html { -webkit-text-size-adjust: 100%; }
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
/* CSS for citations */
div.csl-bib-body { }
div.csl-entry {
  clear: both;
  margin-bottom: 0em;
}
.hanging-indent div.csl-entry {
  margin-left:2em;
  text-indent:-2em;
}
div.csl-left-margin {
  min-width:2em;
  float:left;
}
div.csl-right-inline {
  margin-left:2em;
  padding-left:1em;
}
div.csl-indent {
  margin-left: 2em;
}</style>


<script src="ms_files/libs/clipboard/clipboard.min.js"></script>
<script src="ms_files/libs/quarto-html/quarto.js" type="module"></script>
<script src="ms_files/libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="ms_files/libs/quarto-html/popper.min.js"></script>
<script src="ms_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="ms_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="ms_files/libs/quarto-html/quarto-syntax-highlighting-c8ad9e5dbd60b7b70b38521ab19b7da4.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="ms_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="ms_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="ms_files/libs/bootstrap/bootstrap-c6467a2f2170b0c2dd38189614536510.min.css" rel="stylesheet" append-hash="true" id="quarto-bootstrap" data-mode="light">
<script async="" src="https://hypothes.is/embed.js"></script>
<script>
  window.document.addEventListener("DOMContentLoaded", function (_event) {
    document.body.classList.add('hypothesis-enabled');
  });
</script>

  <script>window.backupDefine = window.define; window.define = undefined;</script><script src="https://cdn.jsdelivr.net/npm/katex@latest/dist/katex.min.js"></script>
  <script>document.addEventListener("DOMContentLoaded", function () {
 var mathElements = document.getElementsByClassName("math");
 var macros = [];
 for (var i = 0; i < mathElements.length; i++) {
  var texText = mathElements[i].firstChild;
  if (mathElements[i].tagName == "SPAN") {
   katex.render(texText.data, mathElements[i], {
    displayMode: mathElements[i].classList.contains('display'),
    throwOnError: false,
    macros: macros,
    fleqn: false
   });
}}});
  </script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script><link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@latest/dist/katex.min.css">

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="_extensions/apaquarto/apa.css">
</head>

<body class="quarto-light">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Manuscript Outline</h2>
   
  <ul class="collapse">
  <li><a href="#introduction" id="toc-introduction" class="nav-link active" data-scroll-target="#introduction">Introduction</a>
  <ul class="collapse">
  <li><a href="#a-beta-way-is-possible" id="toc-a-beta-way-is-possible" class="nav-link" data-scroll-target="#a-beta-way-is-possible">A Beta Way Is Possible</a></li>
  <li><a href="#beta-distribution" id="toc-beta-distribution" class="nav-link" data-scroll-target="#beta-distribution">Beta Distribution</a></li>
  <li><a href="#i-cant-believe-its-not-beta" id="toc-i-cant-believe-its-not-beta" class="nav-link" data-scroll-target="#i-cant-believe-its-not-beta">I Can’t Believe It’s Not beta</a></li>
  <li><a href="#bayesian-approach-to-beta-regression" id="toc-bayesian-approach-to-beta-regression" class="nav-link" data-scroll-target="#bayesian-approach-to-beta-regression">Bayesian Approach to Beta Regression</a></li>
  </ul></li>
  <li><a href="#beta-regression-tutorial" id="toc-beta-regression-tutorial" class="nav-link" data-scroll-target="#beta-regression-tutorial">Beta Regression Tutorial</a>
  <ul class="collapse">
  <li><a href="#example-data" id="toc-example-data" class="nav-link" data-scroll-target="#example-data">Example Data</a></li>
  <li><a href="#reanalysis-of-wilford-et-al.-experiment-1a" id="toc-reanalysis-of-wilford-et-al.-experiment-1a" class="nav-link" data-scroll-target="#reanalysis-of-wilford-et-al.-experiment-1a">Reanalysis of Wilford et al.&nbsp;Experiment 1A</a></li>
  <li><a href="#distributional-regression---beta" id="toc-distributional-regression---beta" class="nav-link" data-scroll-target="#distributional-regression---beta">Distributional Regression - Beta</a></li>
  <li><a href="#zero-inflated-beta-zib-regression" id="toc-zero-inflated-beta-zib-regression" class="nav-link" data-scroll-target="#zero-inflated-beta-zib-regression">Zero-Inflated beta (ZIB) Regression</a></li>
  <li><a href="#zero-one-inflated-beta-zoib" id="toc-zero-one-inflated-beta-zoib" class="nav-link" data-scroll-target="#zero-one-inflated-beta-zoib">Zero-One-Inflated beta (ZOIB)</a></li>
  <li><a href="#ordered-beta-regression" id="toc-ordered-beta-regression" class="nav-link" data-scroll-target="#ordered-beta-regression">Ordered Beta Regression</a></li>
  </ul></li>
  <li><a href="#discussion" id="toc-discussion" class="nav-link" data-scroll-target="#discussion">Discussion</a>
  <ul class="collapse">
  <li><a href="#conclusion" id="toc-conclusion" class="nav-link" data-scroll-target="#conclusion">Conclusion</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
<div class="quarto-alternate-formats"><h2>Other Formats</h2><ul><li><a href="ms.pdf"><i class="bi bi-file-pdf"></i>PDF (apaquarto)</a></li><li><a href="ms.docx"><i class="bi bi-file-word"></i>MS Word (apaquarto)</a></li></ul></div></nav>
</div>
<main class="content" id="quarto-document-content"><div class="quarto-title-block"><div class="quarto-title-tools-only"><h1></h1><button type="button" class="btn code-tools-button dropdown-toggle" id="quarto-code-tools-menu" data-bs-toggle="dropdown" aria-expanded="false"><i class="bi"></i> Code</button><ul class="dropdown-menu dropdown-menu-end" aria-labelelledby="quarto-code-tools-menu"><li><a id="quarto-show-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Show All Code</a></li><li><a id="quarto-hide-all-code" class="dropdown-item" href="javascript:void(0)" role="button">Hide All Code</a></li><li><hr class="dropdown-divider"></li><li><a id="quarto-view-source" class="dropdown-item" href="javascript:void(0)" role="button">View Source</a></li></ul></div></div>




<br>

<br>

<section id="title" class="level1 title unnumbered unlisted">
<h1 class="title unnumbered unlisted">A Beta Way: A Tutorial For Using Beta Regression in Psychological Research</h1>
<div class="Author">
<br>

<p>Jason Geller<sup>1</sup>, Robert Kubinec<sup>2</sup>, Chelsea M. Parlett Pelleriti<sup>3</sup>, and Matti Vuorre<sup>4</sup></p>
<p><sup>1</sup>Department of Psychology and Neuroscience, Boston College</p>
<p><sup>2</sup>University of South Carolina</p>
<p><sup>3</sup>Canva</p>
<p><sup>4</sup>Tilburg University</p>
</div>
<br>

<br>

</section>
<section id="author-note" class="level1 unnumbered unlisted AuthorNote">
<h1 class="unnumbered unlisted AuthorNote">Author Note</h1>
<p>Jason Geller <img src="_extensions/apaquarto/ORCID-iD_icon-vector.svg" id="orchid" class="img-fluid" style="width:4.23mm" alt="Orcid ID Logo: A green circle with white letters ID"> <a href="https://orcid.org/0000-0002-7459-4505">https://orcid.org/0000-0002-7459-4505</a></p>
<p>Robert Kubinec <img src="_extensions/apaquarto/ORCID-iD_icon-vector.svg" id="orchid" class="img-fluid" style="width:4.23mm" alt="Orcid ID Logo: A green circle with white letters ID"> <a href="https://orcid.org/0000-0001-6655-4119">https://orcid.org/0000-0001-6655-4119</a></p>
<p>Chelsea M. Parlett Pelleriti <img src="_extensions/apaquarto/ORCID-iD_icon-vector.svg" id="orchid" class="img-fluid" style="width:4.23mm" alt="Orcid ID Logo: A green circle with white letters ID"> <a href="https://orcid.org/0000-0001-9301-1398">https://orcid.org/0000-0001-9301-1398</a></p>
<p>Matti Vuorre <img src="_extensions/apaquarto/ORCID-iD_icon-vector.svg" id="orchid" class="img-fluid" style="width:4.23mm" alt="Orcid ID Logo: A green circle with white letters ID"> <a href="https://orcid.org/0000-0001-5052-066X">https://orcid.org/0000-0001-5052-066X</a></p>
<p>The analyses herein were not preregistered. Data, code, and materials for this manuscript can be found at https://doi.org/10.5281/zenodo.1583059. The authors have no conflicts of interest to disclose.</p>
<p>Author roles were classified using the Contributor Role Taxonomy (CRediT; https://credit.niso.org/) as follows: <em>Jason Geller</em><strong>: </strong>conceptualization, data curation, formal analysis, project administration, resources, visualization, and writing – original draft. <em>Robert Kubinec</em><strong>: </strong>formal analysis, validation, and Writing - review &amp; editing. <em>Chelsea M. Parlett Pelleriti</em><strong>: </strong>formal analysis and Writing - review &amp; editing. <em>Matti Vuorre</em><strong>: </strong>formal analysis and Writing - review &amp; editing</p>
<p>Correspondence concerning this article should be addressed to Jason Geller, Department of Psychology and Neuroscience, Boston College, McGuinn 300z, Chestnut Hill, MA 2467, USA, Email: <a href="mailto:drjasongeller@gmail.com">drjasongeller@gmail.com</a></p>
<br>
</section>
<section id="abstract" class="level1 unnumbered unlisted AuthorNote">
<h1 class="unnumbered unlisted AuthorNote">Abstract</h1>
<div class="AbstractFirstParagraph">
<p>Rates, percentages, and proportions are common outcomes in psychology and the social sciences. These outcomes are often analyzed using models that assume normality, but this practice overlooks important features of the data, such as their natural bounds at 0 and 1. As a result, estimates can become distorted. In contrast, treating such outcomes as beta-distributed respects these limits and can yield more accurate estimates. Despite these advantages, the use of beta models in applied research remains limited. Our goal is to provide researchers with practical guidance for adopting beta regression models, illustrated with an example drawn from the psychological literature. We begin by introducing the beta distribution and beta regression, emphasizing key components and assumptions. Next, using data from a learning and memory study, we demonstrate how to fit a beta regression model in R with the Bayesian package {brms} and how to interpret results on the response scale. We also discuss model extensions, including zero-inflated, zero- and one-inflated, and ordered beta models. Basic familiarity with regression modeling and R is assumed. To promote wider adoption of these methods, we provide detailed code and materials at https://doi.org/10.5281/zenodo.15830595.</p>
</div>
<p><em>Keywords</em>: beta regression, beta distribution, R tutorial, psychology, learning and memory</p>
<p><em>Word Count</em>: 27827</p>
<br>
</section>
<section id="firstheader" class="level1 title unnumbered unlisted">
<h1 class="title unnumbered unlisted">A Beta Way: A Tutorial For Using Beta Regression in Psychological Research</h1>
<div class="callout callout-style-default callout-warning callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Preprint
</div>
</div>
<div class="callout-body-container callout-body">
<p>This manuscript is currently <strong>under review</strong> and has not been peer-reviewed. Content is <strong>subject to change</strong>. Please feel free to provide feedback!</p>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>path <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the directory if it doesn't exist</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(path)) {</span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(path, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>Many outcomes in psychological research are naturally expressed as proportions or percentages. These include the proportion of correct responses on a test (e.g., <span class="citation" data-cites="kornell2008">Kornell and Bjork (2008)</span>), the proportion of time a participant fixates on a stimulus in an eye-tracking task <span class="citation" data-cites="james2025">(e.g., James et al., 2025)</span>, or the proportion of respondents endorsing a particular belief <span class="citation" data-cites="Costello2024">(e.g., Costello et al., 2024)</span>. A central question for studies involving proportional data is how to analyze these outcomes in a statistically appropriate and interpretable way.</p>
<p>Researchers frequently default to linear models that assume Gaussian (normal) distributions, such as t-tests, ANOVAs, and linear regression. However, these models make strong assumptions: (1) residuals are normally distributed, (2) the outcome is unbounded (from <span class="math inline">-\infty</span> to <span class="math inline">\infty</span>), and (3) variance is constant across the range of the data. These assumptions are rarely satisfied in practice <span class="citation" data-cites="sladekova2024unicorns">(Sladekova &amp; Field, 2024)</span>, and they are especially ill-suited for proportional outcomes, which are bounded between 0 and 1 and often exhibit heteroscedasticity—non-constant variance, particularly near the boundaries <span class="citation" data-cites="smithson2006 ferrari2004 paolino2001">(Ferrari &amp; Cribari-Neto, 2004; Paolino, 2001; Smithson &amp; Verkuilen, 2006)</span>. Violating these assumptions can lead to biased estimates and spurious inferences, especially when many observations cluster near 0 or 1.</p>
<p>In some cases, a generalized linear model (GLM) can relax the assumption of normality. For example, binomial and Bernoulli models—often referred to as logistic regression when used with a logit link—are appropriate when the outcome is binary (0/1) or represents the number of successes out of a fixed number of trials. These models, however, rely on data that arise from discrete trial structures. They tend to perform poorly when the observed proportions are truly continuous or when the data show extra variability (overdispersion), particularly when many values occur near the boundaries of the scale (0 and 1).</p>
<p>The challenges of analyzing proportional data are not new <span class="citation" data-cites="bartlett1936">(see Bartlett, 1936)</span>. Fortunately, several existing approaches address the limitations of commonly used models. One such approach is beta regression, an extension of the generalized linear model that employs the beta distribution <span class="citation" data-cites="ferrari2004 paolino2001">(Ferrari &amp; Cribari-Neto, 2004; Paolino, 2001)</span>. Beta regression offers a flexible and robust solution for modeling proportional data directly by accounting for boundary effects and over-dispersion, making it a valuable alternative to traditional binomial models. This approach is particularly well-suited for psychological research because it can handle both the bounded nature of proportional data and the non-constant variance often encountered in these datasets <span class="citation" data-cites="sladekova2024unicorns">(Sladekova &amp; Field, 2024)</span>. In addition, the direct modeling of proportions allows comparability across tasks and scales, and can be particularly valuable when only the proportional data is available, as is often the case with secondary data that lack item-level structure or point values. While usage of these models has faced obstacles due to theoretical and computational limitations, as we argue in this paper, beta regression and its extensions now provide an accessible and more robust method to traditional linear modeling.</p>
<p>While in this paper we will focus on proportional-responses that lie between 0 and 1–it is important to note that our analysis applies to any bounded continuous scale. Any bounded scale can be mapped to lie within 0 and 1 without resulting in a loss of information as the transformation is linear.<a href="#fn1" class="footnote-ref" id="fnref1" role="doc-noteref"><sup>1</sup></a> Consequently, a scale that has natural end points of -1,234 and +8,451–or any other end points on the real number line short of infinity—can be modeled using the approaches we describe in this paper.</p>
<section id="a-beta-way-is-possible" class="level2">
<h2 data-anchor-id="a-beta-way-is-possible">A Beta Way Is Possible</h2>
<p>With the widespread availability of open-source software such as R <span class="citation" data-cites="R">(R Core Team, 2024)</span> and its extensive ecosystem of user-developed packages, advanced models like beta regression have become increasingly accessible to applied researchers. Yet, their adoption in psychology remains relatively limited. One contributing factor may be the lack of domain-specific examples that demonstrate how these models address common challenges in psychological data. Although recent years have seen a growing interest in beta regression, and a number of useful tutorials are available <span class="citation" data-cites="smithson2006 vuorre2019 heiss2021 bendixen2023 coretta2025bayesian">(Bendixen &amp; Purzycki, 2023; Coretta &amp; Bürkner, 2025; Heiss, 2021; e.g., Smithson &amp; Verkuilen, 2006; Vuorre, 2019)</span>, these resources often either focus on basic implementation or briefly mention extensions without detailing how they can be applied to psychologically relevant research questions.</p>
<p>The present tutorial aims to help bridge this gap by providing a comprehensive, applied introduction to beta regression and several of its extensions. In addition to the standard beta model, we walk through zero-inflated, zero-one-inflated, and ordered beta regression. These models are particularly useful for researchers working with proportion outcomes that include boundary values (e.g., exact 0s or 1s) or responses with an inherent ordinal structure. Our goal is to offer practical guidance that enables psychological researchers to implement, interpret, and report these models in ways that directly support their empirical questions.</p>
<p>Beyond model specification, we place strong emphasis on interpreting results on the response scale–that is, in terms of probabilities and proportions–rather than relying on often difficult to interpret parameters. This focus makes the models more accessible and meaningful for psychological applications, where effects are often easier to communicate when framed on the original scale of the outcome (e.g., changes in recall accuracy or task performance). Throughout, we provide reproducible code and annotated examples to help readers implement and interpret these models in their own work.</p>
<p>We begin the tutorial with a non-technical overview of the beta distribution and its core parameters. We then walk through the process of estimating beta regression models using the R package {brms} <span class="citation" data-cites="brms">(Bürkner, 2017)</span>, illustrating each step with applied examples. To guide interpretation, we emphasize coefficients, predicted probabilities, and marginal effects calculated using the {marginaleffects} package <span class="citation" data-cites="margeffects2024">(Arel-Bundock et al., 2024)</span>. We also introduce several useful extensions–zero-inflated (ZIB), zero-one-inflated (ZOIB), and ordered beta regression–that enable researchers to model outcomes that include boundary values. Finally, all code and materials used in this tutorial are fully reproducible and available via our GitHub repository: <a href="https://github.com/jgeller112/Beta_regression_tutorial" class="uri">https://github.com/jgeller112/beta_regression_tutorial</a> and on Zenodo (https://doi.org/10.5281/zenodo.15830595) <a href="#fn2" class="footnote-ref" id="fnref2" role="doc-noteref"><sup>2</sup></a>.</p>
</section>
<section id="beta-distribution" class="level2">
<h2 data-anchor-id="beta-distribution">Beta Distribution</h2>
<p>Proportional data pose some challenges for standard modeling approaches: The data are bounded between 0 and 1 and often exhibit non-constant variance (heteroscedasticity) <span class="citation" data-cites="ferrari2004 paolino2001">(Ferrari &amp; Cribari-Neto, 2004; Paolino, 2001)</span>. Common distributions used within the generalized linear model frameworks often fail to capture these properties adequately, which can necessitate alternative modeling strategies.</p>
<p>While we do not have time to delve fully into its derivation, the beta distribution is useful for modeling bounded continuous scales because it is the distribution for the probability of an event. Given that a probability can take on any value from near 0 (the event will not occur with certainty) to 1 (the event will occur with certainty), the beta distribution can likewise take on virtually any value in that bounded interval. As a consequence, the beta distribution is the maximum entropy distribution for <em>any</em> bounded continuous random variable, which means that the beta distribution can represent the full range of possibilities of such a scale.<a href="#fn3" class="footnote-ref" id="fnref3" role="doc-noteref"><sup>3</sup></a> As a consequence, if we have a continuous scale with upper and lower bounds–and no other special conditions–the beta distribution will in principle provide a very good approximation of the uncertainty of the scale.</p>
<p>Typically, the expected value (or mean) of the response variable is the central estimand scholars want to estimate. A model should specify how this expected value depends on explanatory variables through two main components: a linear predictor, which combines the explanatory variables in a linear form (<span class="math inline">a+b_1x_1+b_2x_2</span>, etc.), and a link function, which connects the expected value of the response variable to the linear predictor (e.g, <span class="math inline">E[Y] = g(a + b_1x_1 + b_2x_2)</span>). In addition, a random component specifies the distribution of the response variable around its expected value (such as Poisson or binomial distributions, which belong to the exponential family) <span class="citation" data-cites="nelder1972">(Nelder &amp; Wedderburn, 1972)</span>. Together, these components provide a flexible framework for modeling data with different distributional properties.</p>
<p>The beta distribution is continuous and restricted to values between 0 and 1 (exclusive). Its two parameters–commonly called shape1 (<span class="math inline">\alpha</span>) and shape2 (<span class="math inline">\beta</span>)–govern the distribution’s location, skewness, and spread. By adjusting these parameters, the distribution can take many functional forms (e.g., it can be symmetric, skewed, U-shaped, or even approximately uniform; see <a href="#fig-curves" class="quarto-xref" aria-expanded="false">Figure&nbsp;1</a>).</p>
<p>To illustrate, consider a test question worth seven points. Suppose a participant scores five out of seven. The number of points received (5) can be treated as <span class="math inline">\alpha</span>, and the number of points missed (2) as <span class="math inline">\beta</span>. The resulting beta distribution would be skewed toward higher values, reflecting a high performance (pink line in <a href="#fig-curves" class="quarto-xref" aria-expanded="false">Figure&nbsp;1</a>; “beta(5, 2)”). Reversing these values would produce a distribution skewed toward lower values, representing poorer performance (orange line in <a href="#fig-curves" class="quarto-xref" aria-expanded="false">Figure&nbsp;1</a>; “beta(2, 5)”).</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to generate beta distribution data</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>make_beta_df <span class="ot">&lt;-</span> <span class="cf">function</span>(shape1, shape2, <span class="at">n =</span> <span class="dv">9000</span>) {</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.001</span>, <span class="fl">0.999</span>, <span class="at">length.out =</span> n)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> x,</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">dbeta</span>(x, shape1, shape2),</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"beta("</span>, shape1, <span class="st">", "</span>, shape2, <span class="st">")"</span>)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb2-10"><a href="#cb2-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-11"><a href="#cb2-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine distributions</span></span>
<span id="cb2-12"><a href="#cb2-12" aria-hidden="true" tabindex="-1"></a>beta_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb2-13"><a href="#cb2-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_beta_df</span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>),</span>
<span id="cb2-14"><a href="#cb2-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_beta_df</span>(<span class="dv">2</span>, <span class="dv">2</span>),</span>
<span id="cb2-15"><a href="#cb2-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_beta_df</span>(<span class="dv">5</span>, <span class="dv">2</span>),</span>
<span id="cb2-16"><a href="#cb2-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_beta_df</span>(<span class="dv">2</span>, <span class="dv">5</span>)</span>
<span id="cb2-17"><a href="#cb2-17" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb2-18"><a href="#cb2-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">factor</span>(label))</span>
<span id="cb2-19"><a href="#cb2-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-20"><a href="#cb2-20" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(</span>
<span id="cb2-21"><a href="#cb2-21" aria-hidden="true" tabindex="-1"></a>  beta_df,</span>
<span id="cb2-22"><a href="#cb2-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> label, <span class="at">group =</span> label, <span class="at">color =</span> label)</span>
<span id="cb2-23"><a href="#cb2-23" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb2-24"><a href="#cb2-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb2-25"><a href="#cb2-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_okabe_ito</span>(</span>
<span id="cb2-26"><a href="#cb2-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">order =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">6</span>, <span class="dv">7</span>), <span class="co"># whichever correspond to desired colors</span></span>
<span id="cb2-27"><a href="#cb2-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.7</span>,</span>
<span id="cb2-28"><a href="#cb2-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"Distribution"</span></span>
<span id="cb2-29"><a href="#cb2-29" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2-30"><a href="#cb2-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(</span>
<span id="cb2-31"><a href="#cb2-31" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Proportion"</span>,</span>
<span id="cb2-32"><a href="#cb2-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="fl">0.01</span>)</span>
<span id="cb2-33"><a href="#cb2-33" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2-34"><a href="#cb2-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(</span>
<span id="cb2-35"><a href="#cb2-35" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Density"</span>,</span>
<span id="cb2-36"><a href="#cb2-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">5</span>),</span>
<span id="cb2-37"><a href="#cb2-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="fl">0.01</span>)</span>
<span id="cb2-38"><a href="#cb2-38" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb2-39"><a href="#cb2-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb2-40"><a href="#cb2-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-curves" class="quarto-float quarto-figure quarto-figure-center" data-fignum="1" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-curves-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;1</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>beta distributions with different shape1 and shape2 parameters.</p>
</div>
</figcaption>
<div aria-describedby="fig-curves-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-curves-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
</section>
<section id="i-cant-believe-its-not-beta" class="level2">
<h2 data-anchor-id="i-cant-believe-its-not-beta">I Can’t Believe It’s Not beta</h2>
<p>While the standard parameterization of the beta distribution uses <span class="math inline">\alpha</span> and <span class="math inline">\beta</span>, a reparameterization to a mean (<span class="math inline">\mu</span>) and precision (<span class="math inline">\phi</span>) is more useful for regression models. The mean represents the expected value of the distribution, while the dispersion, which is inversely related to variance, reflects how concentrated the distribution is around the mean, with higher values indicating a narrower distribution and lower values indicating a wider one. The connections between the beta distribution’s parameters are shown in <a href="#eq-beta" class="quarto-xref">Equation&nbsp;1</a>. Importantly, the variance depends on the average value of the response because uncertainty intervals need to adjust for how close the value of the response is to the boundary.</p>
<p><span id="eq-beta"><span class="math display">
\begin{aligned}[t]
\text{Shape 1:} &amp;&amp; a &amp;= \mu \phi \\
\text{Shape 2:} &amp;&amp; b &amp;= (1 - \mu) \phi
\end{aligned}
\qquad\qquad\qquad
\begin{aligned}[t]
\text{Mean:} &amp;&amp; \mu &amp;= \frac{a}{a + b} \\
\text{Precision:} &amp;&amp; \phi &amp;= a + b \\
\text{Variance:} &amp;&amp; var &amp;= \frac{\mu \cdot (1 - \mu)}{1 + \phi}
\end{aligned}
\tag{1}</span></span></p>
<p>Thus, beta regression allows modeling both the mean and precision of the outcome distribution. To ensure that <span class="math inline">\mu</span> stays between 0 and 1, we apply a link function, which allows linear modeling of the mean on an unbounded scale. A common link-function choice is the logit, but other functions such as the probit or complementary log-log are possible.</p>
<p>The logit function, <span class="math inline">\text{logit}(\mu) = \log \left( \frac{\mu}{1 - \mu} \right)</span> links the mean to log-odds which are unbounded, making linear modeling possible. The logit here no longer carries the same literal <em>odds</em> interpretation because there are no corresponding counts of “successes” and “failures.” Instead, the logit transform here simply maps the mean of the distribution to the real line. The inverse of the logit, called the logistic function, maps the linear predictor <span class="math inline">\eta</span> back to the original scale of the data <span class="math inline">\left(\mu = \frac{1}{1 + e^{-\eta}}\right)</span>. The coefficients describe how predictors shift the <em>average proportion</em> on the logit scale. Similarly, the strictly positive dispersion parameter is usually modeled through a log link function, ensuring it remains positive.</p>
<p>By accounting for the observations’ natural limits and non-constant variance, the beta distribution is useful in psychology where outcomes like performance rates or response scales frequently exhibit these features.</p>
</section>
<section id="bayesian-approach-to-beta-regression" class="level2">
<h2 data-anchor-id="bayesian-approach-to-beta-regression">Bayesian Approach to Beta Regression</h2>
<p>Beta regression models can be estimated using either frequentist or Bayesian methods. In this paper, we adopt a Bayesian framework because it facilitates the estimation and interpretation of more complex models <span class="citation" data-cites="gelman2013bayesian mcelreath2020statistical bayes_rules">(Gelman et al., 2013; Johnson et al., 2022; McElreath, 2020)</span>. Additionally, the use of Bayesin statistics in psychology has been steadily growing <span class="citation" data-cites="Pfadt2025">(Pfadt et al., 2025)</span>. In principle, frequentist methods like maximum likelihood can be framed as Bayesian models with uninformative priors, and as a result, the modeling perspective we put forward in this paper can apply to either approach. Nonetheless, we note that in non-linear and hierarchical models, frequentist estimation may require additional adjustments such as bootstrapping to obtain proper uncertainty intervals, whereas Bayesian modeling handles these extensions more naturally via exploration of the full joint posterior distribution.<a href="#fn4" class="footnote-ref" id="fnref4" role="doc-noteref"><sup>4</sup></a></p>
<p>There are several important differences between our Bayesian analysis and the frequentist methods readers may be more familiar with—most notably, the absence of <em>t</em>- and <em>p</em>-values. To estimate models, the {brms} package uses Stan’s computational algorithms to draw random samples from the posterior distribution, which represents uncertainty about the model parameters. This posterior is conceptually analogous to a frequentist sampling distribution. By default, Bayesian models run 4 chains with 2,000 iterations each.<a href="#fn5" class="footnote-ref" id="fnref5" role="doc-noteref"><sup>5</sup></a> The first 1,000 iterations per chain are warmup and are discarded. The remaining 1,000 iterations per chain are retained as posterior draws, yielding 4,000 total post-warmup draws across all chains. From these draws, we can compute the posterior mean (analogous to a frequentist point estimate) and the 95% credible interval (Cr.I.), which is often compared to a confidence interval.</p>
<p>In addition, an important part of Bayesian analyses is prior specification. Priors encode our assumptions about plausible parameter values before observing the data and allow the model to regularize estimates, especially when data are sparse or parameters are weakly identified. To help bridge the conceptual gap for users more familiar with frequentist models, we begin with the default priors (flat/non-informative, or weakly informative in some cases) provided by {brms}. These priors are intentionally non-informative, and in many applications produce results that closely align with frequentist estimates, while still offering the flexibility and interpretive advantages of a Bayesian framework. We strongly urge readers to consider prior specification strongly in all their work.</p>
<p>To ease readers into Bayesian data analysis we provide a metric known as the <em>probability of direction</em> (pd), which reflects the probability that a parameter is positive or negative. When a uniform prior is used (all values equally likely in the prior), pds of 95%, 97.5%, 99.5%, and 99.95% corresponds approximately to two-sided p-values of .10, .05, .01, and .001 (i.e., pd ≈ 1 − p/2 for symmetric posteriors with weak/flat priors) (see <a href="#fig-p-value-pd-comp" class="quarto-xref" aria-expanded="false">Figure&nbsp;2</a> for an illustrative comparison). For directional hypotheses, the pd can be interpreted as roughly equivalent to one minus the <em>p</em>-value <span class="citation" data-cites="marsmanThreeInsightsBayesian2016">(Marsman &amp; Wagenmakers, 2016)</span>.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="co"># --- parameters ---</span></span>
<span id="cb3-2"><a href="#cb3-2" aria-hidden="true" tabindex="-1"></a>pd <span class="ot">&lt;-</span> <span class="fl">0.95</span></span>
<span id="cb3-3"><a href="#cb3-3" aria-hidden="true" tabindex="-1"></a>alpha_two <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> (<span class="dv">1</span> <span class="sc">-</span> pd) <span class="co"># matching two-sided alpha</span></span>
<span id="cb3-4"><a href="#cb3-4" aria-hidden="true" tabindex="-1"></a>mu <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(pd) <span class="co"># shift so P(X &gt; 0) = pd for N(mu, 1)</span></span>
<span id="cb3-5"><a href="#cb3-5" aria-hidden="true" tabindex="-1"></a>lim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="dv">6</span>)</span>
<span id="cb3-6"><a href="#cb3-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-7"><a href="#cb3-7" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Posterior-like panel ---</span></span>
<span id="cb3-8"><a href="#cb3-8" aria-hidden="true" tabindex="-1"></a>post <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x =</span> lim), <span class="fu">aes</span>(x)) <span class="sc">+</span></span>
<span id="cb3-9"><a href="#cb3-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># full curve</span></span>
<span id="cb3-10"><a href="#cb3-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb3-11"><a href="#cb3-11" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb3-12"><a href="#cb3-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> mu, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb3-13"><a href="#cb3-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb3-14"><a href="#cb3-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span></span>
<span id="cb3-15"><a href="#cb3-15" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-16"><a href="#cb3-16" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pd region (blue)</span></span>
<span id="cb3-17"><a href="#cb3-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb3-18"><a href="#cb3-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb3-19"><a href="#cb3-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> mu, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb3-20"><a href="#cb3-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, lim[<span class="dv">2</span>]),</span>
<span id="cb3-21"><a href="#cb3-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"area"</span>,</span>
<span id="cb3-22"><a href="#cb3-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"blue"</span>,</span>
<span id="cb3-23"><a href="#cb3-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb3-24"><a href="#cb3-24" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-25"><a href="#cb3-25" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remaining region (red)</span></span>
<span id="cb3-26"><a href="#cb3-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb3-27"><a href="#cb3-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb3-28"><a href="#cb3-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> mu, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb3-29"><a href="#cb3-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(lim[<span class="dv">1</span>], <span class="dv">0</span>),</span>
<span id="cb3-30"><a href="#cb3-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"area"</span>,</span>
<span id="cb3-31"><a href="#cb3-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"red"</span>,</span>
<span id="cb3-32"><a href="#cb3-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb3-33"><a href="#cb3-33" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-34"><a href="#cb3-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linewidth =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb3-35"><a href="#cb3-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb3-36"><a href="#cb3-36" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Posterior (illustrative Normal)"</span>,</span>
<span id="cb3-37"><a href="#cb3-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"pd = "</span>, <span class="fu">round</span>(pd <span class="sc">*</span> <span class="dv">100</span>), <span class="st">"%"</span>),</span>
<span id="cb3-38"><a href="#cb3-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb3-39"><a href="#cb3-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Density"</span></span>
<span id="cb3-40"><a href="#cb3-40" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-41"><a href="#cb3-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> lim, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.45</span>)) <span class="sc">+</span></span>
<span id="cb3-42"><a href="#cb3-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>()</span>
<span id="cb3-43"><a href="#cb3-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-44"><a href="#cb3-44" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Frequentist panel ---</span></span>
<span id="cb3-45"><a href="#cb3-45" aria-hidden="true" tabindex="-1"></a>z_lo <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(alpha_two <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb3-46"><a href="#cb3-46" aria-hidden="true" tabindex="-1"></a>z_hi <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(<span class="dv">1</span> <span class="sc">-</span> alpha_two <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb3-47"><a href="#cb3-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-48"><a href="#cb3-48" aria-hidden="true" tabindex="-1"></a>freq <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x =</span> lim), <span class="fu">aes</span>(x)) <span class="sc">+</span></span>
<span id="cb3-49"><a href="#cb3-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># full curve</span></span>
<span id="cb3-50"><a href="#cb3-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb3-51"><a href="#cb3-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb3-52"><a href="#cb3-52" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb3-53"><a href="#cb3-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb3-54"><a href="#cb3-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span></span>
<span id="cb3-55"><a href="#cb3-55" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-56"><a href="#cb3-56" aria-hidden="true" tabindex="-1"></a>  <span class="co"># lower rejection tail (red)</span></span>
<span id="cb3-57"><a href="#cb3-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb3-58"><a href="#cb3-58" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb3-59"><a href="#cb3-59" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb3-60"><a href="#cb3-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(lim[<span class="dv">1</span>], z_lo),</span>
<span id="cb3-61"><a href="#cb3-61" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"area"</span>,</span>
<span id="cb3-62"><a href="#cb3-62" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"red"</span>,</span>
<span id="cb3-63"><a href="#cb3-63" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb3-64"><a href="#cb3-64" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-65"><a href="#cb3-65" aria-hidden="true" tabindex="-1"></a>  <span class="co"># upper rejection tail (red)</span></span>
<span id="cb3-66"><a href="#cb3-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb3-67"><a href="#cb3-67" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb3-68"><a href="#cb3-68" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb3-69"><a href="#cb3-69" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(z_hi, lim[<span class="dv">2</span>]),</span>
<span id="cb3-70"><a href="#cb3-70" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"area"</span>,</span>
<span id="cb3-71"><a href="#cb3-71" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"red"</span>,</span>
<span id="cb3-72"><a href="#cb3-72" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb3-73"><a href="#cb3-73" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-74"><a href="#cb3-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(z_lo, z_hi), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linewidth =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb3-75"><a href="#cb3-75" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linewidth =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb3-76"><a href="#cb3-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb3-77"><a href="#cb3-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Frequentist sampling under H0 (N(0,1))"</span>,</span>
<span id="cb3-78"><a href="#cb3-78" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"two-sided α = "</span>, <span class="fu">signif</span>(alpha_two, <span class="dv">2</span>)),</span>
<span id="cb3-79"><a href="#cb3-79" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Parameter Values"</span>,</span>
<span id="cb3-80"><a href="#cb3-80" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Density"</span></span>
<span id="cb3-81"><a href="#cb3-81" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb3-82"><a href="#cb3-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> lim, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.45</span>)) <span class="sc">+</span></span>
<span id="cb3-83"><a href="#cb3-83" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>()</span>
<span id="cb3-84"><a href="#cb3-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb3-85"><a href="#cb3-85" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Stack the plots ---</span></span>
<span id="cb3-86"><a href="#cb3-86" aria-hidden="true" tabindex="-1"></a>post <span class="sc">/</span> freq</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-p-value-pd-comp" class="quarto-float quarto-figure quarto-figure-center" data-fignum="2" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-p-value-pd-comp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;2</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>A Bayesian posterior distribution (assuming a uniform prior) centered at a point estimate chosen so that the probability of direction (pd) equals 0.95, and a frequentist sampling distribution (under the null; centered at 0). In the Bayesian posterior distribution, the blue area represents the pd, and the red area represents the remaining 1 − pd of the distribution. In the frequentist sampling distribution, the red tail areas represent the rejection region at α = 0.10. In this example, the posterior mean lies exactly at the <span class="math inline">1 - \frac{\alpha}{2}</span> quantile of the null sampling distribution. For symmetric posteriors with flat priors, the pd is numerically equivalent to the one-sided <span class="math inline">p</span>-value.</p>
</div>
</figcaption>
<div aria-describedby="fig-p-value-pd-comp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-p-value-pd-comp-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>For reasons of space, we refer readers unfamiliar to Bayesian data analysis to several existing books on the topic <span class="citation" data-cites="gelman2013bayesian mcelreath2020statistical kruschke2015doing">(Gelman et al., 2013; Kruschke, 2015; McElreath, 2020)</span>. In addition, we assume readers are familiar with R, but those in need of a refresher should find <span class="citation" data-cites="wickhamDataScienceImport2023">Wickham et al. (2023)</span> useful.</p>
</div>
</section>
</section>
<section id="beta-regression-tutorial" class="level1">
<h1>Beta Regression Tutorial</h1>
<section id="example-data" class="level2">
<h2 data-anchor-id="example-data">Example Data</h2>
<p>Throughout this tutorial, we analyze data from a memory experiment examining whether the fluency of an instructor’s delivery affects recall performance <span class="citation" data-cites="wilford2020">(Wilford et al., 2020, Experiment 1A)</span>. Instructor fluency—marked by expressive gestures, dynamic vocal tone, and confident pacing—has been shown to influence students’ perceptions of learning, often leading learners to rate fluent instructors more favorably <span class="citation" data-cites="carpenter2013">(Carpenter et al., 2013)</span>. However, previous research suggests that these impressions do not reliably translate into improved memory performance <span class="citation" data-cites="carpenter2013 toftness2017 witherby2022">(e.g., Carpenter et al., 2013; Toftness et al., 2017; Witherby &amp; Carpenter, 2022)</span>. In contrast, <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span> found that participants actually recalled more information after watching a fluent instructor compared to a disfluent one. This surprising finding makes the dataset a compelling case study for analyzing proportion data, as recall was scored out of 10 possible idea units per video.</p>
<p>In Experiment 1A, ninety-six participants watched two short instructional videos, each delivered either fluently or disfluently. Fluent videos featured instructors with smooth delivery and natural pacing, while disfluent videos included hesitations, monotone speech, and awkward pauses. After a distractor task, participants completed a free recall test, writing down as much content as they could remember from each video within a three-minute window. Their recall was then scored for the number of idea units correctly remembered.</p>
<div class="cell">
<div id="lst-data" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-data-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;1: Data needed to run examples
</figcaption>
<div aria-describedby="lst-data-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get data here from project folder</span></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a>fluency_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"data"</span>, <span class="st">"fluency_data.csv"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<p>Our primary outcome variable is the proportion of idea units recalled on the final test, calculated by dividing the number of correct units by 10. We show a sample of these data in <a href="#tbl-dataset" class="quarto-xref" aria-expanded="false">Table&nbsp;1</a>. The dataset can be downloaded from GitHub (<a href="#lst-data" class="quarto-xref">Listing&nbsp;1</a>). Because this is a bounded continuous variable (i.e., it ranges from 0 to 1), it violates the assumptions of typical linear regression models that assume normally distributed residual errors. Despite this, it remains common in psychological research to analyze proportion data using models that assume normality. In what follows, we reproduce <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span>’s analysis and then re-analyze the data using beta regression and highlight how it can improve our inferences.</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="1" data-custom-style="FigureWithoutNote">
<div id="tbl-dataset" class="cell quarto-float quarto-figure quarto-figure-center" data-tblnum="1" data-after-caption-space="0pt" prefix="">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-dataset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;1</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Four observations from <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span>. Accuracy refers to the proportion of correctly recalled idea units.</p>
</div>
</figcaption>
<div aria-describedby="tbl-dataset-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script src="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.js"></script>

    <script>
      // Create table-specific functions using external factory
      const tableFns_nv425rtuzevxbi96xw97 = TinyTable.createTableFunctions("tinytable_nv425rtuzevxbi96xw97");
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: '4', j: 1 }, { i: '4', j: 2 }, { i: '4', j: 3 } ], css_id: 'tinytable_css_c2ij2gyxaqhn45q40otz',}, 
          { positions: [ { i: '0', j: 1 }, { i: '0', j: 2 }, { i: '0', j: 3 } ], css_id: 'tinytable_css_vlt0rvpjyj85iahvzzbr',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  tableFns_nv425rtuzevxbi96xw97.styleCell(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.css">
    <style>
    /* tinytable css entries after */
    #tinytable_nv425rtuzevxbi96xw97 td.tinytable_css_c2ij2gyxaqhn45q40otz, #tinytable_nv425rtuzevxbi96xw97 th.tinytable_css_c2ij2gyxaqhn45q40otz {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 0; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.1em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    #tinytable_nv425rtuzevxbi96xw97 td.tinytable_css_vlt0rvpjyj85iahvzzbr, #tinytable_nv425rtuzevxbi96xw97 th.tinytable_css_vlt0rvpjyj85iahvzzbr {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 1; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.05em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    </style>
    <div class="container">
      <table class="tinytable" id="tinytable_nv425rtuzevxbi96xw97" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        
        <thead>
              <tr>
                <th scope="col" data-row="0" data-col="1">Participant</th>
                <th scope="col" data-row="0" data-col="2">Fluency</th>
                <th scope="col" data-row="0" data-col="3">Accuracy</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td data-row="1" data-col="1">64</td>
                  <td data-row="1" data-col="2">Disfluent</td>
                  <td data-row="1" data-col="3">0.10</td>
                </tr>
                <tr>
                  <td data-row="2" data-col="1">30</td>
                  <td data-row="2" data-col="2">Fluent</td>
                  <td data-row="2" data-col="3">0.60</td>
                </tr>
                <tr>
                  <td data-row="3" data-col="1">12</td>
                  <td data-row="3" data-col="2">Fluent</td>
                  <td data-row="3" data-col="3">0.10</td>
                </tr>
                <tr>
                  <td data-row="4" data-col="1">37</td>
                  <td data-row="4" data-col="2">Fluent</td>
                  <td data-row="4" data-col="3">0.35</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
</section>
<section id="reanalysis-of-wilford-et-al.-experiment-1a" class="level2">
<h2 data-anchor-id="reanalysis-of-wilford-et-al.-experiment-1a">Reanalysis of Wilford et al.&nbsp;Experiment 1A</h2>
<p>In their original analysis of Experiment 1A, <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span> compared memory performance between fluent and disfluent instructor conditions using a traditional independent-samples t-test on mean accuracy for 96 participants. They found that participants who watched the fluent instructor recalled significantly more idea units than those who viewed the disfluent version (see <a href="#fig-flu1" class="quarto-xref" aria-expanded="false">Figure&nbsp;3</a>).</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a>fluency_data <span class="sc">|&gt;</span></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Accuracy, <span class="at">y =</span> Fluency, <span class="at">fill =</span> Fluency)) <span class="sc">+</span></span>
<span id="cb5-4"><a href="#cb5-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_dotsinterval</span>(</span>
<span id="cb5-5"><a href="#cb5-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">point_interval =</span> <span class="st">"mean_qi"</span>,</span>
<span id="cb5-6"><a href="#cb5-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">.width =</span> <span class="fl">0.95</span>,</span>
<span id="cb5-7"><a href="#cb5-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">slab_linewidth =</span> <span class="dv">0</span></span>
<span id="cb5-8"><a href="#cb5-8" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb5-9"><a href="#cb5-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">label_percent</span>()) <span class="sc">+</span></span>
<span id="cb5-10"><a href="#cb5-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="cn">NULL</span>, <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb5-11"><a href="#cb5-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_okabe_ito</span>(<span class="at">breaks =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb5-12"><a href="#cb5-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-flu1" class="quarto-float quarto-figure quarto-figure-center" data-fignum="3" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-flu1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;3</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Dot plot depicting accuracy distributions for the Fluent and Disfluent conditions. Each condition shows individual data points and summary statistics (mean and 95% percentile interval) to illustrate variability and central tendency.</p>
</div>
</figcaption>
<div aria-describedby="fig-flu1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-flu1-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>We first replicate this analysis in a regression framework using {brms}. We model final test mean accuracy–the proportion of correctly recalled idea units across the videos–as the dependent variable. Our predictor is instructor fluency, with two levels: Fluent and Disfluent. We use treatment (dummy) coding, which is the default in R. This coding scheme sets the first level of a factor (in alphabetical order) as the reference level. In this case, Disfluent is the reference, and the coefficient for Fluent reflects the contrast between fluent and disfluent instructor conditions.</p>
</div>
<section id="regression-analysis" class="level3">
<h3 data-anchor-id="regression-analysis">Regression Analysis</h3>
<p>We first start by loading the {brms} <span class="citation" data-cites="brms">(Bürkner, 2017)</span> and {cmdstanr} <span class="citation" data-cites="cmdstanr">(Gabry et al., 2024)</span> packages (<a href="#lst-loadbrms" class="quarto-xref">Listing&nbsp;2</a>). We use the cmdstanr backend for Stan <span class="citation" data-cites="stan2023">(Team, 2023)</span> because it’s faster than the default used to run models (i.e., <code>rstan</code>),<a href="#fn6" class="footnote-ref" id="fnref6" role="doc-noteref"><sup>6</sup></a> though all of these models can also be fit with <code>brms</code> defaults.</p>
<div class="cell">
<div id="lst-loadbrms" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-loadbrms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;2: Load the {brms} and {cmdstanr} packages
</figcaption>
<div aria-describedby="lst-loadbrms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cmdstanr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<div class="cell" data-hide="false">
<div id="lst-gauss" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-gauss-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;3: Fitting a gaussian model with brm().
</figcaption>
<div aria-describedby="lst-gauss-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a>bayes_reg_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency,</span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb7-4"><a href="#cb7-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">gaussian</span>(),</span>
<span id="cb7-5"><a href="#cb7-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"model_reg_bayes"</span>)</span>
<span id="cb7-6"><a href="#cb7-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<p>We fit the model using the <code>brm()</code> function from the {brms} package (<a href="#lst-gauss" class="quarto-xref">Listing&nbsp;3</a>). Although not shown here, we ran the models using four chains (the default), executed in parallel across four cores. When the model is run in <a href="#lst-gauss" class="quarto-xref">Listing&nbsp;3</a>, the model summary output will appear in the R console. The output from <code>bayes_reg_model</code> shows each parameter’s posterior summary: The posterior distribution’s mean and standard deviation (analogous to the frequentist standard error) and its 95% credible interval, which indicate the 95% of the most credible parameter values. In {brms}, the reported Cr.I is an equal-tailed interval, meaning that the probability mass excluded from the interval is split equally between the lower and upper tails. Additionally, the output indicates numerical estimates of the sampling algorithm’s performance: Rhat should be close to one, and the ESS (effective sample size) metrics should be as large as possible given the number of iterations specified (default is 4000). Generally, ESS &gt;= 1000 is recommended <span class="citation" data-cites="brms">(Bürkner, 2017)</span>. For the models we present in this paper, convergence is trivial with standard linear models, though we note that these metrics are still important to pay attention to in case of model misfit.</p>
<p>Our main question of interest is: what is the causal effect of instructor fluency on final test performance? In order to answer this question, we will have to look at the output summary produced by <a href="#lst-gauss" class="quarto-xref">Listing&nbsp;3</a> (also see <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;8</a> under Bayesian LM). the <code>Intercept</code> refers to the posterior mean accuracy in the disfluent condition, <em>M</em> = 0.256 , as fluency was dummy-coded. The fluency coefficient (<code>FluencyFluent</code>) reflects the mean posterior difference in recall accuracy between the fluent and disfluent conditions: <em>b</em> = 0.085. The 95% Cr.I for this estimate spans from 0 to 0.167. These values are shown in the “95% Cr.I” columns of the output. These results closely mirror the findings reported by <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span> (Experiment 1A).</p>
<div class="cell">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(bayes_reg_model)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stdout">
<pre><code> Family: gaussian 
  Links: mu = identity 
Formula: Accuracy ~ Fluency 
   Data: fluency_data (Number of observations: 96) 

Regression Coefficients:
              Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
Intercept         0.26      0.03     0.20     0.31 1.00     4190     2492
FluencyFluent     0.08      0.04     0.00     0.17 1.00     4386     3152

Further Distributional Parameters:
      Estimate Est.Error l-95% CI u-95% CI Rhat Bulk_ESS Tail_ESS
sigma     0.21      0.02     0.18     0.24 1.00     4259     2944</code></pre>
</div>
</div>
<p>The output also includes the effective sample size (ESS) and R̂ (R-hat) values, both of which fall within acceptable ranges, indicating good model convergence. Throughout the tutorial, we focus primarily on posterior mean estimates and their 95% credible intervals. In addition, we report the pd measure in the main summary table (<a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;8</a>), provided by the {bayestestR} package <span class="citation" data-cites="makowski2019a makowski2019">(Makowski, Ben-Shachar, Chen, et al., 2019; Makowski, Ben-Shachar, &amp; Lüdecke, 2019)</span>. This measure offers an intuitive parallel to <em>p</em>-values, which many readers may find familiar. For example, the fluency effect has a pd of .977, indicating a high probability that the effect is positive rather than negative.</p>
<p>Importantly, pd does not indicate whether an effect is meaningfully different from a point value—it only reflects the proportion of the posterior in one direction. To address questions of practical significance, readers can consider the Region of Practical Equivalence (ROPE) with the Cr.Is <span class="citation" data-cites="kruschke2015doing">(Kruschke, 2015)</span>. Unlike a hypothesis test of a point null (e.g., exactly zero), the ROPE defines a range of values that are deemed too small to be of substantive importance. As a rule of thumb <span class="citation" data-cites="kruschke2018">(see Kruschke, 2018)</span>, if more than 95% of the posterior distribution lies inside the ROPE, the effect can be considered practically equivalent to that negligible range. If less than 5% lies inside, the effect can be considered meaningfully different. Intermediate cases are typically labeled undecided.</p>
<p>The <code>rope()</code> function in the {bayestestR} package computes the proportion of the posterior within the ROPE to facilitate this evaluation. By default, from bayesian models fit via {brms}, the package determines a ROPE based on the data (roughly reflecting “negligible” effects), but these defaults should be used cautiously. The choice of ROPE should always be guided by theoretical considerations, previous research, and the substantive context of the study. In <a href="#lst-ropebrms" class="quarto-xref">Listing&nbsp;4</a>, we show how to compute this using {bayestestR}. Running the function with default settings suggests that less than 5% lies within the default ROPE (indicating the effect is larger than .02) (see <a href="#fig-rope" class="quarto-xref" aria-expanded="false">Figure&nbsp;4</a>).</p>
<div class="cell">
<div id="lst-ropebrms" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-ropebrms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;4: Getting ROPE from <code>bayes_reg_model</code> obect using <code>rope</code> function from {bayestestR}
</figcaption>
<div aria-describedby="lst-ropebrms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a>brms_rope <span class="ot">&lt;-</span> bayestestR<span class="sc">::</span><span class="fu">rope</span>(bayes_reg_model, <span class="at">ci =</span> .<span class="dv">95</span>, <span class="at">ci_method =</span> <span class="st">"ETI"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># echo: false</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plot</span>(brms_rope)</span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the ROPE percentage</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>rope_percentage <span class="ot">&lt;-</span> <span class="fu">round</span>(brms_rope<span class="sc">$</span>ROPE_Percentage[<span class="dv">2</span>], <span class="dv">3</span>) <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the percentage text to the plot</span></span>
<span id="cb11-9"><a href="#cb11-9" aria-hidden="true" tabindex="-1"></a>p_annotated <span class="ot">&lt;-</span> p <span class="sc">+</span></span>
<span id="cb11-10"><a href="#cb11-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(</span>
<span id="cb11-11"><a href="#cb11-11" aria-hidden="true" tabindex="-1"></a>    <span class="st">"text"</span>,</span>
<span id="cb11-12"><a href="#cb11-12" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">0</span>, <span class="co"># Since your ROPE is -0.02 to 0.02, center is 0</span></span>
<span id="cb11-13"><a href="#cb11-13" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="cn">Inf</span>,</span>
<span id="cb11-14"><a href="#cb11-14" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(rope_percentage, <span class="st">"%"</span>),</span>
<span id="cb11-15"><a href="#cb11-15" aria-hidden="true" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="fl">1.2</span>,</span>
<span id="cb11-16"><a href="#cb11-16" aria-hidden="true" tabindex="-1"></a>    <span class="at">hjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb11-17"><a href="#cb11-17" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">4</span>,</span>
<span id="cb11-18"><a href="#cb11-18" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb11-19"><a href="#cb11-19" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontface =</span> <span class="st">"bold"</span></span>
<span id="cb11-20"><a href="#cb11-20" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb11-21"><a href="#cb11-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb11-22"><a href="#cb11-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_okabe_ito</span>() <span class="sc">+</span></span>
<span id="cb11-23"><a href="#cb11-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_okabe_ito</span>()</span>
<span id="cb11-24"><a href="#cb11-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-25"><a href="#cb11-25" aria-hidden="true" tabindex="-1"></a>p_annotated</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-rope" class="quarto-float quarto-figure quarto-figure-center" data-fignum="4" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-rope-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;4</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Posterior distribution for the fluency effect showing the ROPE(shaded area) with 95% credible interval (orange) and 100% credible interval (blue). The percentage indicates the proportion of the posterior within the ROPE.</p>
</div>
</figcaption>
<div aria-describedby="fig-rope-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-rope-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
<section id="assumptions" class="level4">
<h4 data-anchor-id="assumptions">Assumptions.</h4>
<p><span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span> observed that instructor fluency impacts actual learning, using a standard <em>t</em>-test on the mean accuracy. But recall this approach assumes normality of residuals and homoskedasticity. These assumptions are unrealistic when the response values approach the scale boundaries <span class="citation" data-cites="sladekova2024unicorns">(Sladekova &amp; Field, 2024)</span>. Does the data we have meet those assumptions? We can use the function <code>check_model()</code> from {easystats} <span class="citation" data-cites="easystats">(Lüdecke et al., 2022)</span> to check our assumptions easily. The code in <a href="#lst-check1" class="quarto-xref">Listing&nbsp;5</a> produces <a href="#fig-reg-assump" class="quarto-xref" aria-expanded="false">Figure&nbsp;5</a>. We can see some issues with our model. Specifically, there appears to be violations of constant variance across the values of the scale (homoskedasticity). In plain terms, this type of model mis-specfication means that a standard OLS model can predict non-sensical values outside the bounds of the scale.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div id="lst-check1" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-check1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;5: Checking assumptions with the <code>check_model()</code> from {easystats} package .
</figcaption>
<div aria-describedby="lst-check1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">check_model</span>(bayes_reg_model, <span class="at">check =</span> <span class="fu">c</span>(<span class="st">"homogeneity"</span>, <span class="st">"qq"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
<div class="cell-output-display">
<div id="fig-reg-assump" class="quarto-float quarto-figure quarto-figure-center" data-fignum="5" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-reg-assump-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;5</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Two assumption checks for our OLS model: Normality (right) and Homoskedasticity (left)</p>
</div>
</figcaption>
<div aria-describedby="fig-reg-assump-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-reg-assump-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>We can also examine how well the data fits the model by performing a posterior predictive check using the <code>pp_check()</code> function from {brms}. A posterior predictive check involves looking at multiple model-predicted values and plotting them against the observed data. Ideally, the predicted values (the light blue lines) should show reasonable resemblance with the observed data (dark blue line). In our example (see <a href="#fig-ppcheckall" class="quarto-xref" aria-expanded="false">Figure&nbsp;12</a> (A)) the model-predicted density is slightly too peaked and narrow compared to the data. In addition, some of the predicted accuracy values are negative.</p>
</div>
</section>
</section>
</section>
<section id="distributional-regression---beta" class="level2">
<h2 data-anchor-id="distributional-regression---beta">Distributional Regression - Beta</h2>
<p>It is important to note that there are several justifiable approaches for addressing the distributional issues observed in the data. For instance, one could analyze median accuracy instead of the mean, use some type of robust estimator for heterogeneity, or apply non-parametric methods to relax some of the model assumptions. Alternatively, we can address these issues directly by fitting distributional models <span class="citation" data-cites="kruschke2013 kneib2023">(Kneib et al., 2023; Kruschke, 2013)</span>. A key advantage of distributional models is that they are not limited to modeling only the mean or median of the outcome, but can also model parameters such as the variance (or other shape parameters) as functions of predictors. This allows examining how instructor fluency may influence not only average performance, but also the variability in performance across students. If we wanted to keep our mean accuracy variable and continue to use a Guassian model, we could use a distributional approach and model the effect of fluency on <span class="math inline">\sigma</span>.</p>
<p>Given the outcome variable is proportional, another solution would be to run a beta regression model. Again, we can create the beta regression model in {brms}. In {brms}, we model each parameter independently. Recall from the introduction that in a beta model we model two parameters–<span class="math inline">\mu</span> and <span class="math inline">\phi</span>. Again we do this by using the <code>bf()</code> function from {brms} (<a href="#lst-beta01" class="quarto-xref">Listing&nbsp;6</a>). We specify two formulas, one for <span class="math inline">\mu</span> and one for <span class="math inline">\phi</span> and store it in the <code>model_beta_bayes</code> object below. In the below <code>bf()</code> call, we are modeling accuracy as a function of fluency only for the <span class="math inline">\mu</span> parameter. For the <span class="math inline">\phi</span> parameter, we are only modeling the intercept value. This is saying dispersion does not change as a function of fluency.</p>
<p>To run our beta regression model, we need to exclude 0s and 1s in our data set. If we try to run a model with our data <code>data_fluency</code> we get an error: <code>Error: Family 'beta' requires response greater than 0</code>. This is because the beta distribution only supports observations in the 0 to 1 interval <em>excluding exact 0s and 1s</em>. We need make sure there are no 0s and 1s in our dataset.</p>
<p>The dataset contains nine 0s and one 1. One approach is to nudge our 0s towards .01 and our 1s to .99, or apply a special formula <span class="citation" data-cites="smithson2006">(Smithson &amp; Verkuilen, 2006)</span> so values fall within the [0, 1] interval. We implore readers not to engage in this practice. <span class="citation" data-cites="kubinec2022">Kubinec (2022)</span> showed that this practice can result in serious distortion of the outcome as the sample size 𝑁 grows larger, resulting in ever smaller values that are “nudged”. Because the beta distribution is a non-linear model of the outcome, values that are very close to the boundary, such as 0.00001 or 0.99999, will be highly influential outliers. To run this beta model we will remove the 0s and 1s, and later in this article we will show how to jointly model these scale end points with the rest of the data. The model from <a href="#lst-beta01" class="quarto-xref">Listing&nbsp;6</a> uses a transformed <code>data_fluency</code> object (called <code>data_beta</code>) where 0s and 1s are removed. When we run this code we should not get an error.</p>
<div class="cell">
<div id="lst-beta01" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-beta01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;6: Fitting a beta model without 0s and 1s in brm().
</figcaption>
<div aria-describedby="lst-beta01-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># set up model formual</span></span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>model_beta_bayes <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># fit mu model</span></span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> <span class="dv">1</span> <span class="co"># fit phi model</span></span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a><span class="co"># transform 0 to 0.1 and 1 to .99</span></span>
<span id="cb13-8"><a href="#cb13-8" aria-hidden="true" tabindex="-1"></a>data_beta <span class="ot">&lt;-</span> fluency_data <span class="sc">|&gt;</span></span>
<span id="cb13-9"><a href="#cb13-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb13-10"><a href="#cb13-10" aria-hidden="true" tabindex="-1"></a>    Accuracy <span class="sc">!=</span> <span class="dv">0</span>,</span>
<span id="cb13-11"><a href="#cb13-11" aria-hidden="true" tabindex="-1"></a>    Accuracy <span class="sc">!=</span> <span class="dv">1</span></span>
<span id="cb13-12"><a href="#cb13-12" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb13-13"><a href="#cb13-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb13-14"><a href="#cb13-14" aria-hidden="true" tabindex="-1"></a>beta_brms <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb13-15"><a href="#cb13-15" aria-hidden="true" tabindex="-1"></a>  model_beta_bayes,</span>
<span id="cb13-16"><a href="#cb13-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_beta,</span>
<span id="cb13-17"><a href="#cb13-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">Beta</span>(),</span>
<span id="cb13-18"><a href="#cb13-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>,<span class="st">"models"</span>, <span class="st">"model_beta_bayes_reg_01"</span>)</span>
<span id="cb13-19"><a href="#cb13-19" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<section id="model-parameters" class="level5">
<h5 data-anchor-id="model-parameters">Model Parameters.</h5>
<p>In <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;8</a>, under the beta regression column, the coefficient with <code>b_</code> represents how fluency of instructor influences the <span class="math inline">\mu</span> parameter estimates (which is the mean of the distribution here). These coefficients are linear on the logit-scale, but not on the raw accuracy scale. The intercept term <code>(b_Intercept)</code> represents the log odds of the mean on accuracy for the fluent instructor. Log odds that are negative indicate that it is more likely a “success” (like getting the correct answer) will not happen than that it will happen. Similarly, regression coefficients in log odds forms that are negative indicate that an increase in that predictor leads to a decrease in the predicted probability of a “success”.</p>
<p>The other component we need to pay attention to is the dispersion or precision parameter coefficients labeled as <code>phi</code> in <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;8</a>. The dispersion (<span class="math inline">\phi</span>) parameter tells us how precise our estimate is. Specifically, <span class="math inline">\phi</span> in beta regression tells us about the variability of the response variable around its mean. Specifically, a higher dispersion parameter indicates a narrower distribution, reflecting less variability. Conversely, a lower dispersion parameter suggests a wider distribution, reflecting greater variability. The main difference between a dispersion parameter and the variance is that the dispersion has a different interpretation depending on the value of the outcome, as we show below. The best way to understand dispersion is to examine visual changes in the distribution as the dispersion increases or decreases.</p>
<p>Understanding the dispersion parameter helps us gauge the precision of our predictions and the consistency of the response variable. In <code>beta_brms</code> we only modeled the dispersion of the intercept. When <span class="math inline">\phi</span> is not specified, the intercept is modeled by default (see <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;8</a>). It represents the overall dispersion in the outcome across all conditions. Instead, we can model different dispersions across levels of the <code>Fluency</code> factor. To do so, we add <code>Fluency</code> to the phi model in <code>bf()</code>. We model the precision (<code>phi</code>) of the <code>Fluency</code> factor by using a <code>~</code> and adding factors of interest to the right of it (<a href="#lst-betadis" class="quarto-xref">Listing&nbsp;7</a>).</p>
<div class="cell">
<div id="lst-betadis" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-betadis-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;7: Fitting beta model with dispersion.
</figcaption>
<div aria-describedby="lst-betadis-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a>model_beta_bayes_disp <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># Model of the mean</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency <span class="co"># Model of the precision</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>beta_brms_dis <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>  model_beta_bayes_disp,</span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_beta,</span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">Beta</span>(),</span>
<span id="cb14-10"><a href="#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>,<span class="st">"models"</span>, <span class="st">"model_beta_bayes_dis_run01"</span>)</span>
<span id="cb14-11"><a href="#cb14-11" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<p><a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;8</a> displays the model summary with the precision parameter labeled as <code>phi_Fluency</code>. Since <span class="math inline">\phi</span> is modeled on the log scale, the coefficients represent changes in log-<span class="math inline">\phi</span> rather than φ itself. To see the effect in the original units, we convert the values back by exponentiating. Thus, the effect of the Fluent condition can be understood by comparing the exponentiated predicted <span class="math inline">\phi</span> in the Fluent condition to that in the baseline condition.</p>
<p>The <span class="math inline">\phi</span> parameters are estimated on the log scale. The term <span class="math inline">\beta^{(\phi)}_{\text{Intercept}}</span> represents the log-precision for the reference (disfluent) condition. The coefficient <span class="math inline">\beta^{(\phi)}_{\text{FluencyFluent}}</span> represents the change in log-precision when moving from the disfluent to the fluent condition.</p>
<p>To obtain precision on the original scale, we exponentiate the linear predictor:</p>
<p><span class="math display">
\phi_{\text{disfluent}} = \exp\big(\beta^{(\phi)}_{\text{Intercept}}\big), \qquad
\phi_{\text{fluent}} = \exp\big(\beta^{(\phi)}_{\text{Intercept}} + \beta^{(\phi)}_{\text{Fluency}}\big).
</span></p>
<p>The coefficient <span class="math inline">\beta^{(\phi)}_{\text{Fluency}}</span> therefore describes a <em>multiplicative</em> change in precision. Specifically,</p>
<p><span class="math display">
\frac{\phi_{\text{fluent}}}{\phi_{\text{disfluent}}} = \exp\big(\beta^{(\phi)}_{\text{Fluency}}\big).
</span></p>
<p>Because the 95% credible interval for <span class="math inline">\beta^{(\phi)}_{\text{Fluency}}</span> does not include zero, we infer that there is a credible difference in precision between the fluent and disfluent conditions.</p>
<p>It is important to note that these estimates are not the same as the marginal effects we discussed earlier. Changes in dispersion affect the spread or variability of the response distribution without necessarily altering its mean. This makes dispersion particularly relevant for research questions that focus on features of the distribution beyond the average—such as how concentrated responses are. For instance, high dispersion might indicate that individuals cluster at the extremes (e.g., very high or very low ratings), suggesting clustering in the outcome.</p>
<p>A critical assumption of the linear model is homoscedasticity, which means constant variance of the errors. With beta regression model we can include a dispersion parameter for Fluency. Properly accounting for dispersion is crucial because it impacts the precision of our mean estimates and, consequently, the substantive inferences we might make about the coefficients. The inclusion of dispersion in the model increased the uncertainty of the <span class="math inline">\mu</span> coefficient (see <a href="#fig-plt-diff" class="quarto-xref" aria-expanded="false">Figure&nbsp;6</a>). This highlights the potential utility of an approach like beta regression over a traditional approach as beta regression can explicitly model dispersion and address issues of heteroscedasticity.</p>
<p>While it is advisable to model precision, if there is uncertainty about the best model, a relatively agnostic approach would be to compare models, for example with leave one out (loo) cross validation, to examine if a dispersion parameter should be considered in our model.<a href="#fn7" class="footnote-ref" id="fnref7" role="doc-noteref"><sup>7</sup></a></p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-plt-diff" class="quarto-float quarto-figure quarto-figure-center" data-fignum="6" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-plt-diff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;6</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Comparison of posterior distributions for the risk difference in fluency: Simple model (no dispersion for Fluency) vs.&nbsp;complex model with dispersion</p>
</div>
</figcaption>
<div aria-describedby="fig-plt-diff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-plt-diff-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
</section>
<section id="predicted-probabilities" class="level3">
<h3 data-anchor-id="predicted-probabilities">Predicted Probabilities</h3>
<p>Parameter estimates can be difficult to interpret, and researchers can instead discuss effects on the actual outcome scale (in this case the 0-1 scale). The logit link allows us to transform back and forth between the scale of a linear model and the nonlinear scale of the outcome, which is bounded by 0 and 1. By using the inverse of the logit, we can easily transform our linear coefficients to obtain average effects on the scale of the proportions or percentages, which is usually easier to interpret. In a simple case, we can do this manually, but when there are many factors in your model this can be quite complex.</p>
<p>In our example, we can use the <code>plogis()</code> function in base R to convert estimates from the logit scale to the probability scale. The intercept of our model is -0.918, which reflects the logit of the mean accuracy in the disfluent condition. If the estimated difference between the fluent and disfluent conditions is 0.24 on the logit scale, we first add this value to the intercept value (-0.918) to get the logit for the fluent condition: <code>-0.83 + 0.20 = -0.63</code>. We then use <code>plogis()</code> to convert both logit values to probabilities (Fluent = 35%, Disfluent = 30%).</p>
<p>With single coefficients this calculation is trivial, but in more complex models with interactions, it can be quite cumbersome. To help us extract predictions from our model and visualize them we will use a package called {marginaleffects} <span class="citation" data-cites="margeffects2024">(Arel-Bundock et al., 2024)</span> (see <a href="#lst-me" class="quarto-xref">Listing&nbsp;8</a>). To get the proportions for each of our categorical predictors on the <span class="math inline">\mu</span> parameter we can use the function from the package called <code>avg_predictions()</code>. These are displayed in <a href="#tbl-predict-prob" class="quarto-xref" aria-expanded="false">Table&nbsp;2</a>. These probabilities match what we calculated above.</p>
<div class="cell">
<div id="lst-me" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-me-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;8: Load the {marginaleffects} package.
</figcaption>
<div aria-describedby="lst-me-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(marginaleffects)</span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">marginaleffects_posterior_center =</span> mean) <span class="co"># make sure returns mean</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<div class="cell">
<div id="lst-pred" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-pred-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;9: Predictions from the beta model for each level of Fluency (<span class="math inline">\mu</span>).
</figcaption>
<div aria-describedby="lst-pred-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">avg_predictions</span>(</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>  beta_brms_dis,</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>  <span class="co"># need to specify the levels of the categorical predictor</span></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">variables =</span> <span class="st">"Fluency"</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="2" data-tbl-cap-location="top" data-custom-style="FigureWithoutNote">
<div id="tbl-predict-prob" class="cell quarto-float quarto-figure quarto-figure-center" data-tbl-cap-location="top" data-tblnum="2" data-after-caption-space="0pt" prefix="">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-predict-prob-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;2</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Predicted probabilities for fluency factor (<span class="math inline">\mu</span>).</p>
</div>
</figcaption>
<div aria-describedby="tbl-predict-prob-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script src="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.js"></script>

    <script>
      // Create table-specific functions using external factory
      const tableFns_3czftfozgeussbprn2ss = TinyTable.createTableFunctions("tinytable_3czftfozgeussbprn2ss");
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: '2', j: 1 }, { i: '2', j: 2 }, { i: '2', j: 3 } ], css_id: 'tinytable_css_gc6o77oii04opttvuwyj',}, 
          { positions: [ { i: '0', j: 1 }, { i: '0', j: 2 }, { i: '0', j: 3 } ], css_id: 'tinytable_css_agfp8isb187vl6zj24lt',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  tableFns_3czftfozgeussbprn2ss.styleCell(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.css">
    <style>
    /* tinytable css entries after */
    #tinytable_3czftfozgeussbprn2ss td.tinytable_css_gc6o77oii04opttvuwyj, #tinytable_3czftfozgeussbprn2ss th.tinytable_css_gc6o77oii04opttvuwyj {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 0; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.1em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    #tinytable_3czftfozgeussbprn2ss td.tinytable_css_agfp8isb187vl6zj24lt, #tinytable_3czftfozgeussbprn2ss th.tinytable_css_agfp8isb187vl6zj24lt {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 1; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.05em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    </style>
    <div class="container">
      <table class="tinytable" id="tinytable_3czftfozgeussbprn2ss" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        
        <thead>
              <tr>
                <th scope="col" data-row="0" data-col="1">Fluency</th>
                <th scope="col" data-row="0" data-col="2">Mean</th>
                <th scope="col" data-row="0" data-col="3">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td data-row="1" data-col="1">Disfluent</td>
                  <td data-row="1" data-col="2">0.304</td>
                  <td data-row="1" data-col="3">[0.249, 0.363]</td>
                </tr>
                <tr>
                  <td data-row="2" data-col="1">Fluent</td>
                  <td data-row="2" data-col="2">0.348</td>
                  <td data-row="2" data-col="3">[0.301, 0.398]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>For the <code>Fluency</code> factor, we can interpret the estimates under the <code>Mean</code> column as a proportion or percentage. That is, participants who watched the fluent instructor scored on average 35% on the final exam compared to 30% for those who watched the disfluent instructor. We can also visualize these from {marginaleffects} using the <code>plot_predictions()</code> function (see <a href="#lst-plotbeta" class="quarto-xref">Listing&nbsp;10</a>).</p>
</div>
<div class="cell">
<div id="lst-plotbeta" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-plotbeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;10: Plot predicted probabilities using <code>plot_predictions()</code> from {marginaleffects}
</figcaption>
<div aria-describedby="lst-plotbeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a>beta_plot <span class="ot">&lt;-</span> <span class="fu">plot_predictions</span>(beta_brms_dis, <span class="at">by =</span> <span class="st">"Fluency"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<p>The <code>plot_predictions()</code> function will only display the point estimate with the 95% credible interval. However, Bayesian estimation methods generate distributions for each parameter. This approach allows visualizing full uncertainty estimates beyond points and intervals. Using the {marginaleffects} package, we can obtain samples from the posterior distribution with the <code>posterior_draws()</code> function (see <a href="#lst-posdraws" class="quarto-xref">Listing&nbsp;11</a>). We can then plot these results to illustrate the range of plausible values for our estimates at different levels of uncertainty (see <a href="#fig-draws-fluency" class="quarto-xref" aria-expanded="false">Figure&nbsp;7</a>).</p>
<div class="cell">
<div id="lst-posdraws" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-posdraws-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;11: Extracting posterior draws from the beta regression model.
</figcaption>
<div aria-describedby="lst-posdraws-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a model identifier to each dataset</span></span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a>pred_draws_beta <span class="ot">&lt;-</span> <span class="fu">avg_predictions</span>(beta_brms_dis, <span class="at">variables =</span> <span class="st">"Fluency"</span>) <span class="sc">|&gt;</span></span>
<span id="cb18-3"><a href="#cb18-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">posterior_draws</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot both models</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pred_draws_beta, <span class="fu">aes</span>(<span class="at">x =</span> draw, <span class="at">y =</span> Fluency, <span class="at">fill =</span> Fluency)) <span class="sc">+</span></span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_halfeye</span>(<span class="at">.width =</span> <span class="fu">c</span>(.<span class="dv">5</span>, .<span class="dv">8</span>, .<span class="dv">95</span>), <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_okabe_ito</span>() <span class="sc">+</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">label_percent</span>()) <span class="sc">+</span></span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Accuracy"</span>,</span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">"50%, 80% and 95% credible intervals shown in black"</span></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">"solid"</span>, <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-draws-fluency" class="quarto-float quarto-figure quarto-figure-center" data-fignum="7" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-draws-fluency-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;7</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Predicted probablity posterior distributions by instructor fluency</p>
</div>
</figcaption>
<div aria-describedby="fig-draws-fluency-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-draws-fluency-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
</section>
<section id="marginal-effects" class="level3">
<h3 data-anchor-id="marginal-effects">Marginal Effects</h3>
<p>Marginal effects offer an interpretable way to quantify how changes in a predictor influence an outcome, while holding other factors constant in a specific manner. In recent years, there has been a thrust to move away from reporting regression coefficients alone, focusing instead on estimates that are easier to interpret and communicate—particularly in non-linear models <span class="citation" data-cites="mccabe2021 rohrer2025models">(McCabe et al., 2021; Rohrer &amp; Arel-Bundock, 2025)</span>. Technically, marginal effects are computed as partial derivatives for continuous variables or as finite differences for categorical (and sometimes continuous) predictors, depending on the structure of the data and the research question. Substantively, these procedures translate raw regression coefficients into quantities that reflect changes in the bounded outcome—for example, an <span class="math inline">x</span>% change in the value of a proportion.</p>
<p>There are several types of marginal effects, and their computation can vary across software packages. For example, the popular {emmeans} package <span class="citation" data-cites="emmeans">(Lenth, 2025)</span> computes marginal effects by holding all predictors at their means (marginal effects at the mean; MEM). In this tutorial, we use the {marginaleffects} package <span class="citation" data-cites="margeffects2024">(Arel-Bundock et al., 2024)</span>, which by default computes average marginal effects (AMEs). AMEs are based on counterfactual predictions: the dataset is conceptually replicated across all unique values of the predictor of interest, predictions are generated for each row under each counterfactual scenario, and the resulting differences are then averaged. This approach maintains a strong connection to the observed data—because predictions are made using each participant’s actual values—while providing a clear and interpretable summary of the effect of interest.</p>
<p>One practical use of AMEs is to estimate the average difference between two groups or conditions which corresponds to the average treatment effect (ATE). Using the <code>avg_comparisons()</code> function in the {marginaleffects}package (<a href="#lst-avgcomp" class="quarto-xref">Listing&nbsp;12</a>), we can compute this quantity directly. By default, the function returns the discrete difference between groups. When we take the difference in proportions between two groups it is called the risk difference. Depending on the audience and modeling goals, the function can also produce alternative effect size metrics, such as odds ratios or risk ratios. This flexibility makes it a powerful approach for summarizing and communicating regression results.</p>
<div class="cell">
<div id="lst-avgcomp" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-avgcomp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;12: Calculating the difference between probabilities (<span class="math inline">\mu</span>) for fluency with <code>avg_comparisons()</code>
</figcaption>
<div aria-describedby="lst-avgcomp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get risk difference by default</span></span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>beta_avg_comp <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a>  beta_brms_dis,</span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">variables =</span> <span class="st">"Fluency"</span>, <span class="co"># counterfactural difference</span></span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="3" data-custom-style="FigureWithoutNote">
<div id="tbl-ame1" class="cell quarto-float quarto-figure quarto-figure-center" data-tblnum="3" data-after-caption-space="0pt" prefix="">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-ame1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;3</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Fluency difference (<span class="math inline">\mu</span>)</p>
</div>
</figcaption>
<div aria-describedby="tbl-ame1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script src="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.js"></script>

    <script>
      // Create table-specific functions using external factory
      const tableFns_1zsfeolo65siqxuil3dv = TinyTable.createTableFunctions("tinytable_1zsfeolo65siqxuil3dv");
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: '1', j: 1 }, { i: '1', j: 2 }, { i: '1', j: 3 }, { i: '1', j: 4 } ], css_id: 'tinytable_css_pus0t6rarlmarj8md2mm',}, 
          { positions: [ { i: '0', j: 1 }, { i: '0', j: 2 }, { i: '0', j: 3 }, { i: '0', j: 4 } ], css_id: 'tinytable_css_vsvr5m6h481wwtqouisy',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  tableFns_1zsfeolo65siqxuil3dv.styleCell(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.css">
    <style>
    /* tinytable css entries after */
    #tinytable_1zsfeolo65siqxuil3dv td.tinytable_css_pus0t6rarlmarj8md2mm, #tinytable_1zsfeolo65siqxuil3dv th.tinytable_css_pus0t6rarlmarj8md2mm {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 0; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.1em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    #tinytable_1zsfeolo65siqxuil3dv td.tinytable_css_vsvr5m6h481wwtqouisy, #tinytable_1zsfeolo65siqxuil3dv th.tinytable_css_vsvr5m6h481wwtqouisy {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 1; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.05em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    </style>
    <div class="container">
      <table class="tinytable" id="tinytable_1zsfeolo65siqxuil3dv" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        
        <thead>
              <tr>
                <th scope="col" data-row="0" data-col="1">Contrast</th>
                <th scope="col" data-row="0" data-col="2">Mean</th>
                <th scope="col" data-row="0" data-col="3">95% Cr.I</th>
                <th scope="col" data-row="0" data-col="4">pd</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td data-row="1" data-col="1">Fluent - Disfluent</td>
                  <td data-row="1" data-col="2">0.044</td>
                  <td data-row="1" data-col="3">[-0.036, 0.118]</td>
                  <td data-row="1" data-col="4">0.87</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<section id="mu-difference" class="level4">
<h4 data-anchor-id="mu-difference"><span class="math inline">\mu</span> Difference.</h4>
<p><a href="#tbl-ame1" class="quarto-xref" aria-expanded="false">Table&nbsp;3</a> presents the estimated difference for the <code>Fluency</code> factor (Mean column). The difference between the fluent and disfluent conditions is 0.06, indicating that participants who watched a fluent instructor scored, on average, 6% higher on the final recall test than those who watched a disfluent instructor. However, the 95% credible interval includes 0 among the most credible values, suggesting we cannot rule out the possibility of a null or weakly negative effect. The pd also indicates the effect is non-significant (<em>p</em> &gt; .05). In the basic regression model above, the ROPE value was extracted using the {bayestestR} package.</p>
<div class="cell">
<div id="lst-merope" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-merope-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;13: Extracting ROPE using {marginaleffects} for <span class="math inline">\mu</span> difference
</figcaption>
<div aria-describedby="lst-merope-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get risk difference by default</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>beta_avg_comp_rope <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>  beta_brms_dis,</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">variables =</span> <span class="st">"Fluency"</span>, <span class="co"># counterfactural difference</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">comparison =</span> <span class="st">"difference"</span>,</span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">equivalence=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.02</span>, .<span class="dv">02</span>)</span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<p>,The ROPE can also be obtained using the {marginaleffects} package by specifying the <code>equivalence</code> bounds in <code>avg_comparisons()</code>. Running the code in <a href="#lst-merope" class="quarto-xref">Listing&nbsp;13</a> returns a ROPE probability of 23%, indicating that there is insufficient evidence to conclude that the effect exceeds 2% or is practically equivalent.</p>
</section>
<section id="phi-difference" class="level4">
<h4 data-anchor-id="phi-difference"><span class="math inline">\phi</span> difference.</h4>
<p>In addition, with {marginaleffects}, we can get the actual precision difference between the two groups on <span class="math inline">\phi</span> using similar code to above by setting <code>dpar</code> to “phi” (<a href="#lst-avgphi" class="quarto-xref">Listing&nbsp;14</a>).</p>
<div class="cell">
<div id="lst-avgphi" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-avgphi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;14: Calculating <span class="math inline">\phi</span> difference with <code>avg_comparisons()</code>
</figcaption>
<div aria-describedby="lst-avgphi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get risk difference by default</span></span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>beta_avg_phi <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>  beta_brms_dis,</span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">variables =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb22-6"><a href="#cb22-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">dpar =</span> <span class="st">"phi"</span>,</span>
<span id="cb22-7"><a href="#cb22-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb22-8"><a href="#cb22-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<div class="cell">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Proportions</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fl">0.35</span> <span class="co"># Fluent</span></span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fl">0.30</span> <span class="co"># Disfluent</span></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohen's h formula</span></span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a>cohens_h <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> (<span class="fu">asin</span>(<span class="fu">sqrt</span>(p1)) <span class="sc">-</span> <span class="fu">asin</span>(<span class="fu">sqrt</span>(p2)))</span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a><span class="co"># CI lower and upper bounds</span></span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a>p1_lower <span class="ot">&lt;-</span> <span class="fl">0.328</span></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a>p1_upper <span class="ot">&lt;-</span> <span class="fl">0.419</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>p2_lower <span class="ot">&lt;-</span> <span class="fl">0.250</span></span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a>p2_upper <span class="ot">&lt;-</span> <span class="fl">0.329</span></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Lower bound of h</span></span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a>h_lower <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> (<span class="fu">asin</span>(<span class="fu">sqrt</span>(p1_lower)) <span class="sc">-</span> <span class="fu">asin</span>(<span class="fu">sqrt</span>(p2_upper)))</span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a><span class="co"># Upper bound of h</span></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a>h_upper <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> (<span class="fu">asin</span>(<span class="fu">sqrt</span>(p1_upper)) <span class="sc">-</span> <span class="fu">asin</span>(<span class="fu">sqrt</span>(p2_lower)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</section>
<section id="effect-size" class="level4">
<h4 data-anchor-id="effect-size">Effect Size.</h4>
<p>In psychology, it is common to report effect size measures like Cohen’s <em>d</em> <span class="citation" data-cites="cohen1977">(Cohen, 1977)</span>. When working with proportions we can calculate something similar called Cohen’s <em>h</em>. Taking our proportions, we can use the below equation (<a href="#eq-cohenh" class="quarto-xref">Equation&nbsp;2</a>) to calculate Cohen’s <em>h</em> along with the 95% Cr.I around it. Using this metric we see the effect size is small (0.107), 95% credible interval [-0.002, 0.361].</p>
<p><span id="eq-cohenh"><span class="math display">
h = 2 \cdot \left( \arcsin\left(\sqrt{p_1}\right) - \arcsin\left(\sqrt{p_2}\right) \right)
\tag{2}</span></span></p>
</section>
</section>
<section id="posterior-predictive-check" class="level3">
<h3 data-anchor-id="posterior-predictive-check">Posterior Predictive Check</h3>
<p><a href="#fig-ppcheckall" class="quarto-xref" aria-expanded="false">Figure&nbsp;12</a> (B) shows the predictive check for our beta model. The model’s predictions generally conform to the data as the predictions are now between constrained to the 0-1 interval. However, the model’s predictive performance can be improved the bounds of the scale are explicitly specified.</p>
</section>
</section>
<section id="zero-inflated-beta-zib-regression" class="level2">
<h2 data-anchor-id="zero-inflated-beta-zib-regression">Zero-Inflated beta (ZIB) Regression</h2>
<p>A limitation of the beta regression model is that it can only accommodate values strictly between 0 and 1—a probability cannot take on values of 0 (the event will not occur with certainty) or 1 (the event will occur with certainty). In our dataset, we observed 9 rows where Accuracy equals zero. To fit a beta regression model, we removed these values, but we have left out potentially valuable information from our model–especially if the end points of the scale are distinctive in some way. In our case, these 0s may be structural—that is, they represent real, systematic instances where participants failed to answer correctly (rather than random noise or measurement error). For example, the fluency of the instructor might be a key factor in predicting these zero responses. We will discuss two approaches for jointly modeling these end points with the continuous data. First, we can use a zero-inflated beta (ZIB) model. This model still estimates the mean (<span class="math inline">\mu</span>) and precision (<span class="math inline">\phi</span>) of the beta distribution for values between 0 and 1, but it also includes an additional parameter, <span class="math inline">\alpha</span>, which captures the probability of observing structural 0s.</p>
<p>The zero-inflated beta models a mixture of the data-generating process. The <span class="math inline">\alpha</span> parameter uses a logistic regression to model whether the data is 0 or not. Substantively, this could be a useful model when we think that 0s come from a process that is relatively distinct from the data that is greater than 0. For example, if we had a dataset with proportion of looks or eye fixations to certain areas on marketing materials, we might want a separate model for those that do not look at certain areas on the screen because individuals who do not look might be substantively different than those that look.</p>
<p>We can fit a ZIB model using <code>brms()</code> and use the {marginaleffects} package to make inferences about our parameters of interest. Before we run a zero-inflated beta model, we will need to transform our data again and remove the one 1 value in our data–we can keep our 0s. Similar to our beta regression model we fit in <code>brms</code>, we will use the <code>bf()</code> function to fit several models. We fit our <span class="math inline">\mu</span> and <span class="math inline">\phi</span> parameters as well as our zero-inflated parameter (<span class="math inline">\alpha</span>; here labeled as <code>zi</code>). In <code>brms</code> we can use the zero_inflated_beta family (see <a href="#lst-zib" class="quarto-xref">Listing&nbsp;15</a>).</p>
<div class="cell">
<div id="lst-zib" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-zib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;15: Fitting zib model with <code>brm()</code>
</figcaption>
<div aria-describedby="lst-zib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># keep 0 but remove 1</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a>data_beta_0 <span class="ot">&lt;-</span> fluency_data <span class="sc">|&gt;</span></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Accuracy <span class="sc">!=</span> <span class="dv">1</span>)</span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a><span class="co"># set up model formual for zero-inflated beta in brm</span></span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>zib_model <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># The mean of the 0-1 values, or mu</span></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency, <span class="co"># The precision of the 0-1 values, or phi</span></span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>  zi <span class="sc">~</span> Fluency, <span class="co"># The zero-or-one-inflated part, or alpha</span></span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">zero_inflated_beta</span>()</span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb24-12"><a href="#cb24-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-13"><a href="#cb24-13" aria-hidden="true" tabindex="-1"></a><span class="co"># fit zib model with brm</span></span>
<span id="cb24-14"><a href="#cb24-14" aria-hidden="true" tabindex="-1"></a>fit_zi <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb24-15"><a href="#cb24-15" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> zib_model,</span>
<span id="cb24-16"><a href="#cb24-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_beta_0,</span>
<span id="cb24-17"><a href="#cb24-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>,<span class="st">"models"</span>, <span class="st">"bayes_zib_model0not1.rds"</span>)</span>
<span id="cb24-18"><a href="#cb24-18" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<section id="posterior-predictive-check-1" class="level3">
<h3 data-anchor-id="posterior-predictive-check-1">Posterior Predictive Check</h3>
<p>The ZIB model does a bit better at capturing the structure of the data then the beta regression model (see <a href="#fig-ppcheckall" class="quarto-xref" aria-expanded="false">Figure&nbsp;12</a>). Specifically, the ZIB model more accurately captures the increased density of values near the lower end of the scale (i.e., near zero), which the standard beta model underestimates. The ZIB model’s predictive distributions also align more closely with the observed data across the entire range, particularly in the peak and tail regions. This improved fit likely reflects the ZIB model’s ability to explicitly model excess 0s (or near-zero values) via its inflation component, allowing it to better account for features in the data that a standard beta distribution cannot accommodate.</p>
</section>
<section id="predicted-probabilities-and-marginal-effects" class="level3">
<h3 data-anchor-id="predicted-probabilities-and-marginal-effects">Predicted Probabilities and Marginal Effects</h3>
<p><a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;8</a>, under the zero-inflated beta regression column, provides a summary of the posterior distribution for each parameter. As stated before, it is preferable to back-transform our estimates to get probabilities. To get the predicted probabilities we can again use the <code>avg_predictions()</code> and <code>avg_comparisons()</code> functions from {marginaleffects} package <span class="citation" data-cites="marginaleffects">(Arel-Bundock, 2024)</span> to get predicted probabilities and the probability difference between the levels of each factor. We can model the parameters separately using the <code>dpar</code> argument setting to: <span class="math inline">\mu</span>, <span class="math inline">\phi</span>, <span class="math inline">\alpha</span>. Here we look at the risk difference for Fluency under each parameter. If one were interested in the average effect for the entire model, the <code>dpar</code> argument could be removed.</p>
<section id="mu" class="level4">
<h4 data-anchor-id="mu">Mu.</h4>
<p>As shown in <a href="#tbl-marg-mu-brms" class="quarto-xref" aria-expanded="false">Table&nbsp;4</a>, there is little evidence for an effect of Fluency – the 95% Cr.I includes zero, suggesting substantial uncertainty about the direction and magnitude of the effect–that is, though most of the posterior density supports positive effects, nil and weakly negative effects cannot be ruled out.</p>
<div class="cell">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a>marg_zi_brms <span class="ot">&lt;-</span> fit_zi <span class="sc">|&gt;</span></span>
<span id="cb25-2"><a href="#cb25-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">avg_comparisons</span>(</span>
<span id="cb25-3"><a href="#cb25-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">variables =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb25-4"><a href="#cb25-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">dpar =</span> <span class="st">"mu"</span>,</span>
<span id="cb25-5"><a href="#cb25-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb25-6"><a href="#cb25-6" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="4" data-custom-style="FigureWithoutNote">
<div id="tbl-marg-mu-brms" class="cell quarto-float quarto-figure quarto-figure-center" data-tblnum="4" data-after-caption-space="0pt" prefix="">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-marg-mu-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;4</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Probablity fluency difference (<span class="math inline">\mu</span>)</p>
</div>
</figcaption>
<div aria-describedby="tbl-marg-mu-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script src="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.js"></script>

    <script>
      // Create table-specific functions using external factory
      const tableFns_krwgirsosrqz09x11sc6 = TinyTable.createTableFunctions("tinytable_krwgirsosrqz09x11sc6");
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: '1', j: 1 }, { i: '1', j: 2 }, { i: '1', j: 3 }, { i: '1', j: 4 } ], css_id: 'tinytable_css_kl0rhgvxirpkh3i93m9y',}, 
          { positions: [ { i: '0', j: 1 }, { i: '0', j: 2 }, { i: '0', j: 3 }, { i: '0', j: 4 } ], css_id: 'tinytable_css_yyp7ngnfewztz188k8ql',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  tableFns_krwgirsosrqz09x11sc6.styleCell(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.css">
    <style>
    /* tinytable css entries after */
    #tinytable_krwgirsosrqz09x11sc6 td.tinytable_css_kl0rhgvxirpkh3i93m9y, #tinytable_krwgirsosrqz09x11sc6 th.tinytable_css_kl0rhgvxirpkh3i93m9y {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 0; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.1em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    #tinytable_krwgirsosrqz09x11sc6 td.tinytable_css_yyp7ngnfewztz188k8ql, #tinytable_krwgirsosrqz09x11sc6 th.tinytable_css_yyp7ngnfewztz188k8ql {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 1; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.05em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    </style>
    <div class="container">
      <table class="tinytable" id="tinytable_krwgirsosrqz09x11sc6" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        
        <thead>
              <tr>
                <th scope="col" data-row="0" data-col="1">Contrast</th>
                <th scope="col" data-row="0" data-col="2">Mean</th>
                <th scope="col" data-row="0" data-col="3">95% Cr.I</th>
                <th scope="col" data-row="0" data-col="4">pd</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td data-row="1" data-col="1">Fluent - Disfluent</td>
                  <td data-row="1" data-col="2">0.043</td>
                  <td data-row="1" data-col="3">[-0.031, 0.118]</td>
                  <td data-row="1" data-col="4">0.875</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
</section>
<section id="dispersion" class="level4">
<h4 data-anchor-id="dispersion">Dispersion.</h4>
<p>As shown in <a href="#tbl-marg-phi-brms" class="quarto-xref" aria-expanded="false">Table&nbsp;5</a>, the posterior estimates suggest a credible effect of Fluency on dispersion (<span class="math inline">\phi</span>), with disfluent responses showing greater variability. The 95% Cr.I for the fluency contrast does not include zero, indicating a high probability in differences in precision.</p>
<div class="cell">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb26"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a>marg_phi_brms <span class="ot">&lt;-</span> fit_zi <span class="sc">|&gt;</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">avg_comparisons</span>(</span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">variables =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">dpar =</span> <span class="st">"phi"</span>,</span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="5" data-custom-style="FigureWithoutNote">
<div id="tbl-marg-phi-brms" class="cell quarto-float quarto-figure quarto-figure-center" data-tblnum="5" data-after-caption-space="0pt" prefix="">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-marg-phi-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;5</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Probablity fluency difference (<span class="math inline">\phi</span>)</p>
</div>
</figcaption>
<div aria-describedby="tbl-marg-phi-brms-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script src="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.js"></script>

    <script>
      // Create table-specific functions using external factory
      const tableFns_ir6tds8g8289xluq7udc = TinyTable.createTableFunctions("tinytable_ir6tds8g8289xluq7udc");
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: '1', j: 1 }, { i: '1', j: 2 }, { i: '1', j: 3 }, { i: '1', j: 4 } ], css_id: 'tinytable_css_710s7qymd1qccwiznfq5',}, 
          { positions: [ { i: '0', j: 1 }, { i: '0', j: 2 }, { i: '0', j: 3 }, { i: '0', j: 4 } ], css_id: 'tinytable_css_l5n8d0j56wdxzi1bri1p',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  tableFns_ir6tds8g8289xluq7udc.styleCell(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.css">
    <style>
    /* tinytable css entries after */
    #tinytable_ir6tds8g8289xluq7udc td.tinytable_css_710s7qymd1qccwiznfq5, #tinytable_ir6tds8g8289xluq7udc th.tinytable_css_710s7qymd1qccwiznfq5 {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 0; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.1em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    #tinytable_ir6tds8g8289xluq7udc td.tinytable_css_l5n8d0j56wdxzi1bri1p, #tinytable_ir6tds8g8289xluq7udc th.tinytable_css_l5n8d0j56wdxzi1bri1p {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 1; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.05em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    </style>
    <div class="container">
      <table class="tinytable" id="tinytable_ir6tds8g8289xluq7udc" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        
        <thead>
              <tr>
                <th scope="col" data-row="0" data-col="1">Contrast</th>
                <th scope="col" data-row="0" data-col="2">Mean</th>
                <th scope="col" data-row="0" data-col="3">95% Cr.I</th>
                <th scope="col" data-row="0" data-col="4">pd</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td data-row="1" data-col="1">Fluent - Disfluent</td>
                  <td data-row="1" data-col="2">2.68</td>
                  <td data-row="1" data-col="3">[-0.894, 6.45]</td>
                  <td data-row="1" data-col="4">0.926</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
</section>
</section>
<section id="zero-inflation" class="level3">
<h3 data-anchor-id="zero-inflation">Zero-Inflation</h3>
<p>We can use {marginaleffects} to estimate and plot the posterior difference between the fluent and disfluent conditions (see <a href="#fig-zi-plot" class="quarto-xref" aria-expanded="false">Figure&nbsp;8</a>). In <a href="#fig-zi-plot" class="quarto-xref" aria-expanded="false">Figure&nbsp;8</a>, the posterior distribution for this contrast lies mostly below zero, indicating that a fluent instructor is associated with a lower probability of zero responses. The estimated reduction is approximately 13%. The 95% credible interval does not include zero, which indicates that the data provide consistent evidence for a reduction in zero responses under fluent instruction.</p>
<div class="cell">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a>marg_zi_brms <span class="ot">&lt;-</span> fit_zi <span class="sc">|&gt;</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">avg_comparisons</span>(</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">variables =</span> <span class="fu">list</span>(<span class="at">Fluency =</span> <span class="st">"revpairwise"</span>),</span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">dpar =</span> <span class="st">"zi"</span>,</span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<div class="cell-output-display">
<div id="fig-zi-plot" class="quarto-float quarto-figure quarto-figure-center" data-fignum="8" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-zi-plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;8</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Visualization of the predicted difference for zero-inflated part of model</p>
</div>
</figcaption>
<div aria-describedby="fig-zi-plot-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-zi-plot-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
</section>
</section>
<section id="zero-one-inflated-beta-zoib" class="level2">
<h2 data-anchor-id="zero-one-inflated-beta-zoib">Zero-One-Inflated beta (ZOIB)</h2>
<p>The ZIB model works well if there are 0s in your data, but not 1s.<a href="#fn8" class="footnote-ref" id="fnref8" role="doc-noteref"><sup>8</sup></a> In our previous examples we either got rid of both 0s and 1s (beta regression), or removed the 1s (ZIB). Sometimes it is theoretically useful to model both 0s and 1s as separate processes or to consider these values as essentially similar parts of the continuous response, as we show later in the ordered beta regression model. For example, this is important in visual analog scale data where there might be a prevalence of responses at the bounds <span class="citation" data-cites="kong2016">(Kong &amp; Edwards, 2016)</span>, in JOL tasks <span class="citation" data-cites="wilford2020">(Wilford et al., 2020)</span>, or in a free-list task where individuals provide open responses to some question or topic which are then recoded to fall between 0-1 <span class="citation" data-cites="bendixen2023">(Bendixen &amp; Purzycki, 2023)</span>. Here 0s and 1s are meaningful; 0 means item was not listed and 1 means the item was listed first.</p>
<p>Similar to the beta and zero-inflated models discussed above, we can fit a zero-and-one‐inflated beta (ZOIB) model in {brms} using the <code>zero_one_inflated_beta</code> family. This formulation simultaneously estimates the mean <span class="math inline">\mu</span> and precision <span class="math inline">\phi</span> of the Beta component, as well as two inflation parameters: <span class="math inline">\alpha</span> , the probability that an observation is at either boundary (0 or 1), and <span class="math inline">\gamma</span> , the conditional probability that, given an observation falls on a boundary, it takes the value 1 rather than 0. In other words, <span class="math inline">\alpha</span> determines how often responses occur exactly at the endpoints, and <span class="math inline">\gamma</span> determines the balance between zeros and ones among those endpoint values. This specification allows the model to capture both the continuous variation in the interior of the (0, 1) interval and the presence of exact boundary values.</p>
<p>To illustrate how <span class="math inline">\alpha</span> and <span class="math inline">\gamma</span> shape the distribution, <a href="#fig-zoib" class="quarto-xref" aria-expanded="false">Figure&nbsp;9</a> displays simulated data across a range of parameter combinations. As <span class="math inline">\alpha</span> increases, more responses occur at the endpoints. As <span class="math inline">\gamma</span> increases, the proportion of those endpoint responses that are 1 increases relative to 0, producing increasingly pronounced spikes at 1 as <span class="math inline">\gamma</span> approaches 1. Together, these parameters give the ZOIB model the flexibility to represent datasets with mixtures of continuous values and exact zeros and ones.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb28"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb28-1"><a href="#cb28-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Zero-one inflated beta generator adapted from</span></span>
<span id="cb28-2"><a href="#cb28-2" aria-hidden="true" tabindex="-1"></a><span class="co"># https://vuorre.com/posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-beta-models/</span></span>
<span id="cb28-3"><a href="#cb28-3" aria-hidden="true" tabindex="-1"></a>rzoib <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">n =</span> <span class="dv">1000</span>, <span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">gamma =</span> <span class="fl">0.45</span>, <span class="at">mu =</span> <span class="fl">0.7</span>, <span class="at">phi =</span> <span class="dv">3</span>) {</span>
<span id="cb28-4"><a href="#cb28-4" aria-hidden="true" tabindex="-1"></a>  a <span class="ot">&lt;-</span> mu <span class="sc">*</span> phi</span>
<span id="cb28-5"><a href="#cb28-5" aria-hidden="true" tabindex="-1"></a>  b <span class="ot">&lt;-</span> (<span class="dv">1</span> <span class="sc">-</span> mu) <span class="sc">*</span> phi</span>
<span id="cb28-6"><a href="#cb28-6" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb28-7"><a href="#cb28-7" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbinom</span>(n, <span class="dv">1</span>, alpha),</span>
<span id="cb28-8"><a href="#cb28-8" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbinom</span>(n, <span class="dv">1</span>, gamma),</span>
<span id="cb28-9"><a href="#cb28-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbeta</span>(n, a, b)</span>
<span id="cb28-10"><a href="#cb28-10" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb28-11"><a href="#cb28-11" aria-hidden="true" tabindex="-1"></a>  y</span>
<span id="cb28-12"><a href="#cb28-12" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb28-13"><a href="#cb28-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-14"><a href="#cb28-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Optimized parameter grid</span></span>
<span id="cb28-15"><a href="#cb28-15" aria-hidden="true" tabindex="-1"></a>param_grid <span class="ot">&lt;-</span> <span class="fu">expand_grid</span>(</span>
<span id="cb28-16"><a href="#cb28-16" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.4</span>),</span>
<span id="cb28-17"><a href="#cb28-17" aria-hidden="true" tabindex="-1"></a>  <span class="at">gamma =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.9</span>),</span>
<span id="cb28-18"><a href="#cb28-18" aria-hidden="true" tabindex="-1"></a>  <span class="at">mu =</span> <span class="fl">0.7</span>,</span>
<span id="cb28-19"><a href="#cb28-19" aria-hidden="true" tabindex="-1"></a>  <span class="at">phi =</span> <span class="dv">3</span></span>
<span id="cb28-20"><a href="#cb28-20" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb28-21"><a href="#cb28-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-22"><a href="#cb28-22" aria-hidden="true" tabindex="-1"></a><span class="co"># Simulate data</span></span>
<span id="cb28-23"><a href="#cb28-23" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb28-24"><a href="#cb28-24" aria-hidden="true" tabindex="-1"></a>sim_data <span class="ot">&lt;-</span> param_grid <span class="sc">%&gt;%</span></span>
<span id="cb28-25"><a href="#cb28-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb28-26"><a href="#cb28-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">pmap</span>(</span>
<span id="cb28-27"><a href="#cb28-27" aria-hidden="true" tabindex="-1"></a>      <span class="fu">list</span>(alpha, gamma, mu, phi),</span>
<span id="cb28-28"><a href="#cb28-28" aria-hidden="true" tabindex="-1"></a>      <span class="sc">~</span> <span class="fu">rzoib</span>(<span class="dv">1000</span>, ..<span class="dv">1</span>, ..<span class="dv">2</span>, ..<span class="dv">3</span>, ..<span class="dv">4</span>)</span>
<span id="cb28-29"><a href="#cb28-29" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb28-30"><a href="#cb28-30" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb28-31"><a href="#cb28-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(data) <span class="sc">%&gt;%</span></span>
<span id="cb28-32"><a href="#cb28-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Accuracy =</span> data)</span>
<span id="cb28-33"><a href="#cb28-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-34"><a href="#cb28-34" aria-hidden="true" tabindex="-1"></a><span class="co"># Classify type</span></span>
<span id="cb28-35"><a href="#cb28-35" aria-hidden="true" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> sim_data <span class="sc">%&gt;%</span></span>
<span id="cb28-36"><a href="#cb28-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb28-37"><a href="#cb28-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">Type =</span> <span class="fu">case_when</span>(</span>
<span id="cb28-38"><a href="#cb28-38" aria-hidden="true" tabindex="-1"></a>      Accuracy <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"Zero"</span>,</span>
<span id="cb28-39"><a href="#cb28-39" aria-hidden="true" tabindex="-1"></a>      Accuracy <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"One"</span>,</span>
<span id="cb28-40"><a href="#cb28-40" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"beta"</span></span>
<span id="cb28-41"><a href="#cb28-41" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb28-42"><a href="#cb28-42" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb28-43"><a href="#cb28-43" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-44"><a href="#cb28-44" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure 'Zero' comes before 'One'</span></span>
<span id="cb28-45"><a href="#cb28-45" aria-hidden="true" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> plot_data <span class="sc">%&gt;%</span></span>
<span id="cb28-46"><a href="#cb28-46" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Type =</span> <span class="fu">factor</span>(Type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Zero"</span>, <span class="st">"One"</span>, <span class="st">"beta"</span>)))</span>
<span id="cb28-47"><a href="#cb28-47" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb28-48"><a href="#cb28-48" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb28-49"><a href="#cb28-49" aria-hidden="true" tabindex="-1"></a>  <span class="co"># beta histogram</span></span>
<span id="cb28-50"><a href="#cb28-50" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(</span>
<span id="cb28-51"><a href="#cb28-51" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> plot_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Type <span class="sc">==</span> <span class="st">"beta"</span>),</span>
<span id="cb28-52"><a href="#cb28-52" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> Accuracy, <span class="at">y =</span> <span class="fu">after_stat</span>(count)),</span>
<span id="cb28-53"><a href="#cb28-53" aria-hidden="true" tabindex="-1"></a>    <span class="at">bins =</span> <span class="dv">50</span>,</span>
<span id="cb28-54"><a href="#cb28-54" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"gray60"</span>,</span>
<span id="cb28-55"><a href="#cb28-55" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"white"</span>,</span>
<span id="cb28-56"><a href="#cb28-56" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.8</span></span>
<span id="cb28-57"><a href="#cb28-57" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb28-58"><a href="#cb28-58" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Spikes at 0 and 1</span></span>
<span id="cb28-59"><a href="#cb28-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(</span>
<span id="cb28-60"><a href="#cb28-60" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> plot_data <span class="sc">%&gt;%</span></span>
<span id="cb28-61"><a href="#cb28-61" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(Type <span class="sc">!=</span> <span class="st">"beta"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb28-62"><a href="#cb28-62" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(alpha, gamma, Accuracy, Type) <span class="sc">%&gt;%</span></span>
<span id="cb28-63"><a href="#cb28-63" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>),</span>
<span id="cb28-64"><a href="#cb28-64" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> Accuracy, <span class="at">y =</span> count, <span class="at">fill =</span> Type),</span>
<span id="cb28-65"><a href="#cb28-65" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="fl">0.01</span></span>
<span id="cb28-66"><a href="#cb28-66" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb28-67"><a href="#cb28-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb28-68"><a href="#cb28-68" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Zero"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"One"</span> <span class="ot">=</span> <span class="st">"blue"</span>),</span>
<span id="cb28-69"><a href="#cb28-69" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"Type"</span></span>
<span id="cb28-70"><a href="#cb28-70" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb28-71"><a href="#cb28-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(</span>
<span id="cb28-72"><a href="#cb28-72" aria-hidden="true" tabindex="-1"></a>    gamma <span class="sc">~</span> alpha,</span>
<span id="cb28-73"><a href="#cb28-73" aria-hidden="true" tabindex="-1"></a>    <span class="at">labeller =</span> label_both,</span>
<span id="cb28-74"><a href="#cb28-74" aria-hidden="true" tabindex="-1"></a>    <span class="at">scales =</span> <span class="st">"free_y"</span>,</span>
<span id="cb28-75"><a href="#cb28-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">switch =</span> <span class="st">"y"</span></span>
<span id="cb28-76"><a href="#cb28-76" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb28-77"><a href="#cb28-77" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb28-78"><a href="#cb28-78" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Accuracy"</span>,</span>
<span id="cb28-79"><a href="#cb28-79" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Count"</span></span>
<span id="cb28-80"><a href="#cb28-80" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb28-81"><a href="#cb28-81" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb28-82"><a href="#cb28-82" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">label_percent</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-zoib" class="quarto-float quarto-figure quarto-figure-center" data-fignum="9" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-zoib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;9</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Simulated data from a ZOIB model illustrating the effects of the zero-one inflation parameter (<span class="math inline">\alpha</span>) and the conditional one-inflation parameter (<span class="math inline">\gamma</span>).</p>
</div>
</figcaption>
<div aria-describedby="fig-zoib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-zoib-1.png" class="img-fluid figure-img" width="2400">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>To fit a ZOIB model we use the <code>bf()</code> function. We model each parameter as a function of Fluency. We then pass the <code>zoib_model</code> to our <code>brm()</code> function (see <a href="#lst-zoib" class="quarto-xref">Listing&nbsp;16</a>). The summary of the output is in <a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;8</a> (under <code>ZOIB</code>).</p>
</div>
<div class="cell">
<div id="lst-zoib" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-zoib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;16: Fitting a ZOIB model with <code>brm()</code>.
</figcaption>
<div aria-describedby="lst-zoib-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fit the zoib model</span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>zoib_model <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># The mean of the 0-1 values, or mu</span></span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency, <span class="co"># The precision of the 0-1 values, or phi</span></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a>  zoi <span class="sc">~</span> Fluency, <span class="co"># The zero-or-one-inflated part, or alpha</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>  coi <span class="sc">~</span> Fluency, <span class="co"># The one-inflated part, conditional on the 1s, or gamma</span></span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">zero_one_inflated_beta</span>()</span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a>fit_zoib <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> zoib_model,</span>
<span id="cb29-12"><a href="#cb29-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb29-13"><a href="#cb29-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_zoib_model"</span>)</span>
<span id="cb29-14"><a href="#cb29-14" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<section id="model-parameters-1" class="level3">
<h3 data-anchor-id="model-parameters-1">Model Parameters</h3>
<p>The output for the model is lengthy because we are estimating four distinct components, each with their own independent responses and sub-models. All the coefficients are on the logit scale, except <span class="math inline">\phi</span> , which is on the log scale. Thankfully drawing inferences for all these different parameters, plotting their distributions, and estimating their average marginal effects looks exactly the same–all the <code>brms</code> and {marginaleffects} functions we used work the same.</p>
</section>
<section id="predictions-and-marginal-effects" class="level3">
<h3 data-anchor-id="predictions-and-marginal-effects">Predictions and Marginal Effects</h3>
<p>With {marginaleffects} we can choose marginalize over all the sub-models, averaged across the 0s, continuous responses, and 1s in the data, or we can model the parameters separately using the <code>dpar</code> argument like we did above setting it to: <span class="math inline">\mu</span>, <span class="math inline">\phi</span>, <span class="math inline">\alpha</span>, <span class="math inline">\gamma</span> (see below). Using <code>avg_predictions()</code> and not setting <code>dpar</code> we can get the predicted probabilities across all the sub-models. We can also plot the overall difference between fluency and disfluency for the whole model with <code>plot_predictions()</code>.</p>
<p>In addition, we show below how one can extract the predicted probabilities and marginal effects for <span class="math inline">\gamma</span> (and a similar process for any other model component, <code>zoi</code>, etc.):</p>
<div class="cell">
<div id="lst-coi" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-coi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;17: Extracting predicted probabilities and marginal effects for conditional-one parameter
</figcaption>
<div aria-describedby="lst-coi-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb30"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb30-1"><a href="#cb30-1" aria-hidden="true" tabindex="-1"></a><span class="co"># get average predictions for coi param</span></span>
<span id="cb30-2"><a href="#cb30-2" aria-hidden="true" tabindex="-1"></a>coi_probs <span class="ot">&lt;-</span> <span class="fu">avg_predictions</span>(fit_zoib, <span class="at">variables =</span> <span class="st">"Fluency"</span>, <span class="at">dpar =</span> <span class="st">"coi"</span>)</span>
<span id="cb30-3"><a href="#cb30-3" aria-hidden="true" tabindex="-1"></a><span class="co"># get differene between the two conditions</span></span>
<span id="cb30-4"><a href="#cb30-4" aria-hidden="true" tabindex="-1"></a>coi_me <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(fit_zoib, <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Fluency"</span>), <span class="at">dpar =</span> <span class="st">"coi"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
</section>
</section>
<section id="ordered-beta-regression" class="level2">
<h2 data-anchor-id="ordered-beta-regression">Ordered Beta Regression</h2>
<p>Looking at the output from the ZOIB model (<a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;8</a>), we can see how running a model like this can become fairly complex as it is fitting distinct sub-models for each component of the scale. The ability to consider 0s and 1s as distinct processes from continuous values comes at a price in terms of complexity and interpretability. A simplified version of the zero-one-inflated beta (ZOIB) model, known as ordered beta regression <span class="citation" data-cites="kubinec2022 Makowski2025CHOCO">(Kubinec, 2022; see also Makowski et al., 2025 for a reparameterized version called the <em>beta-Gate</em> model)</span>, has been recently proposed. The ordered beta regression model exploits the fact that, for most analyses, the continuous values (between 0-1) and the discrete outcomes (e.g., 0 or 1) are <em>ordered</em>. For example, as a covariate <span class="math inline">x</span> increases or decreases, we should expect the bounded outcome <span class="math inline">y</span> to increase or decrease monotonically as well from 0 to <span class="math inline">(0,1)</span> to 1. The ZOIB model does not impose this restriction; a covariate could increase and the response <span class="math inline">y</span> could increase in its continuous values while <em>simultaneously</em> decreasing at <em>both</em> end points.<a href="#fn9" class="footnote-ref" id="fnref9" role="doc-noteref"><sup>9</sup></a> This complexity is not immediately obvious when fitting the ZOIB, nor is it a potential relationship that many scholars want to consider when examining how covariates influence a bounded scale.</p>
<p>To make the response ordered, the ordered beta regression model estimates a weighted combination of a standard beta regression model for continuous responses and a logit model for the discrete values of the response. By doing so, the amount of distinctiveness between the continuous responses and the discrete end points is a function of the data (and any informative priors) rather than strictly defined as fully distinct processes as in the ZOIB. For some datasets, the continuous and discrete responses will be fairly distinct, and in others less so.</p>
<p>The weights that average together the two parts of the outcome (i.e., discrete and continuous) are determined by cutpoints that are estimated in conjunction with the data in a similar manner to what is known as an ordered logit model. An in-depth explanation of ordinal regression is beyond the scope of this tutorial <span class="citation" data-cites="Fullerton2023 bürkner2019">(Bürkner &amp; Vuorre, 2019; but see Fullerton &amp; Anderson, 2021)</span>. At a basic level, ordinal regression models are useful for outcome variables that are categorical in nature and have some inherent ordering (e.g., Likert scale items). To preserve this ordering, ordinal models rely on the cumulative probability distribution. Within an ordinal regression model it is assumed that there is a continuous but unobserved latent variable that determines which of <span class="math inline">k</span> ordinal responses will be selected. For example on a typical Likert scale from ‘Strongly Disagree’ to ‘Strongly Agree’, you could assume that there is a continuous, unobserved variable called ‘Agreement’.</p>
<p>While we cannot measure Agreement directly, the ordinal response gives us some indication about where participants are on the continuous Agreement scale. <span class="math inline">k - 1</span> cutoffs are then estimated to indicate the point on the continuous Agreement scale at which your Agreement level is high enough to push you into the next ordinal category (say Agree to Strongly Agree). Coefficients in the model estimate how much different predictors change the estimated <em>continuous</em> scale (here, Agreement). Since there’s only one underlying process, there’s only one set of coefficients to work with (proportional odds assumption).</p>
<p>In an ordered beta regression, three ordered categories are modeled: (1) exactly zero, (2) somewhere between zero and one, and (3) exactly one. In an ordered beta regression, (1) and (2) are modeled with cumulative logits, where one cutpoint is the the boundary between Exactly 0 and Between 0 and 1 and the other cutpoint is the boundary between <em>Between 0 and 1</em> and <em>Exactly 1.</em> The continuous values in the middle, 0 to 1 (3), are modeled as a vanilla beta regression with parameters reflecting the mean response on the logit scale as we have described previously. Ultimately, employing cutpoints allows for a smooth transition between the bounds and the continuous values, permitting both to be considered together rather than modeled separately as the ZOIB requires.</p>
<p>The ordered beta regression model has shown to be more efficient and less biased than some of the methods discussed <span class="citation" data-cites="kubinec2022">(Kubinec, 2022)</span> herein and has seen increasing use across the biomedical and social sciences <span class="citation" data-cites="wilkesQuantifyingCoextinctionsEcosystem2024 martinVocalComplexitySocially2024 smithEcologicalMomentaryAssessment2024 shresthaBigFiveTraits2024 nouvianGlyphosateImpairsAversive2023">(Martin et al., 2024; Nouvian et al., 2023; Shrestha et al., 2024; Smith et al., 2024; Wilkes et al., 2024)</span> because it produces only a single set of coefficient estimates in a similar manner to a standard beta regression or OLS.<a href="#fn10" class="footnote-ref" id="fnref10" role="doc-noteref"><sup>10</sup></a></p>
<section id="fitting-an-ordered-beta-regression" class="level3">
<h3 data-anchor-id="fitting-an-ordered-beta-regression">Fitting an Ordered Beta Regression</h3>
<p>To fit an ordered beta regression in a Bayesian context we use the {ordbetareg} <span class="citation" data-cites="ordbetareg">(Kubinec, 2023)</span> package. {ordbetareg} is a front-end to the <code>brms</code> package that we described earlier; in addition to the functions available in the package, most <code>brms</code> functions and plots, including the diverse array of regression modeling options, will work with {ordbetareg} models. (We note that the <code>ordbeta</code> model is also available as a maximum-likelihood variant in the R package {glmmTMB}.) We first load the {ordbetareg} package (see <a href="#lst-ordbeta" class="quarto-xref">Listing&nbsp;18</a>).</p>
<div class="cell">
<div id="lst-ordbeta" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-ordbeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;18: Load {ordbetareg}
</figcaption>
<div aria-describedby="lst-ordbeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ordbetareg)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<p>The {ordbetareg} package uses <code>brms</code> on the front-end so all the arguments we used previously apply here. Instead of the <code>brm()</code> function we use <code>ordbetareg()</code>. To fit a model where dispersion does not vary as a function of fluency we can use the below code (see <a href="#lst-fitordbeta" class="quarto-xref">Listing&nbsp;19</a>).</p>
<div class="cell">
<div id="lst-fitordbeta" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-fitordbeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;19: Fitting ordered beta model with <code>ordbetareg()</code>
</figcaption>
<div aria-describedby="lst-fitordbeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>ord_fit_brms <span class="ot">&lt;-</span> <span class="fu">ordbetareg</span>(</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency,</span>
<span id="cb32-3"><a href="#cb32-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb32-4"><a href="#cb32-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>,<span class="st">"models"</span>, <span class="st">"bayes_ordbeta_model"</span>)</span>
<span id="cb32-5"><a href="#cb32-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<p>However, if we want dispersion to vary as a function of fluency we can easily do that (see <a href="#lst-ordbetadisp" class="quarto-xref">Listing&nbsp;20</a>). Note the addition of the <code>phi_reg</code> argument in <code>m.phi</code>. This argument allows us to include a model that explicitly models the dispersion parameter. Because we are modeling <span class="math inline">\phi</span> as a function of fluency, we set the the argument to <code>both.</code></p>
<div class="cell">
<div id="lst-ordbetadisp" class="r cell-code listing quarto-float quarto-figure quarto-figure-left">
<figure class="quarto-float quarto-float-lst figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-lst" id="lst-ordbetadisp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Listing&nbsp;20: Fitting ordered beta model with dispersion using <code>ordbetareg()</code>
</figcaption>
<div aria-describedby="lst-ordbetadisp-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a>ord_beta_phi <span class="ot">&lt;-</span> <span class="fu">bf</span>(Accuracy <span class="sc">~</span> Fluency, phi <span class="sc">~</span> Fluency)</span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a>m.phi <span class="ot">&lt;-</span> <span class="fu">ordbetareg</span>(</span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a>  ord_beta_phi,</span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">phi_reg =</span> <span class="st">'both'</span>,</span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_ordbeta_phi_model"</span>)</span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</figure>
</div>
</div>
<section id="marginal-effects-1" class="level4">
<h4 data-anchor-id="marginal-effects-1">Marginal Effects.</h4>
<p><a href="#tbl-all" class="quarto-xref" aria-expanded="false">Table&nbsp;8</a> presents the posterior summary. We can use {marginaleffects} to calculate differences on the response scale that average over (or marginalize over) all our parameters.</p>
<div class="cell">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb34"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb34-1"><a href="#cb34-1" aria-hidden="true" tabindex="-1"></a>ord_beta_avg <span class="ot">&lt;-</span> m.phi <span class="sc">|&gt;</span></span>
<span id="cb34-2"><a href="#cb34-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">avg_comparisons</span>(</span>
<span id="cb34-3"><a href="#cb34-3" aria-hidden="true" tabindex="-1"></a>    <span class="at">variables =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb34-4"><a href="#cb34-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb34-5"><a href="#cb34-5" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="6" data-custom-style="FigureWithoutNote">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a>p_d <span class="ot">&lt;-</span> ord_beta_avg <span class="sc">|&gt;</span></span>
<span id="cb35-2"><a href="#cb35-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">p_direction</span>() <span class="sc">|&gt;</span></span>
<span id="cb35-3"><a href="#cb35-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb35-4"><a href="#cb35-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pd)</span>
<span id="cb35-5"><a href="#cb35-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb35-6"><a href="#cb35-6" aria-hidden="true" tabindex="-1"></a>ord_beta_avg <span class="sc">|&gt;</span></span>
<span id="cb35-7"><a href="#cb35-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(term, contrast, estimate, conf.low, conf.high) <span class="sc">|&gt;</span></span>
<span id="cb35-8"><a href="#cb35-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb35-9"><a href="#cb35-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb35-10"><a href="#cb35-10" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{conf.low}, {conf.high}]"</span>)</span>
<span id="cb35-11"><a href="#cb35-11" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb35-12"><a href="#cb35-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Contrast"</span> <span class="ot">=</span> <span class="st">"contrast"</span>, <span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"estimate"</span>, <span class="st">"Term"</span> <span class="ot">=</span> <span class="st">"term"</span>) <span class="sc">|&gt;</span></span>
<span id="cb35-13"><a href="#cb35-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(p_d) <span class="sc">|&gt;</span></span>
<span id="cb35-14"><a href="#cb35-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Contrast, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, pd) <span class="sc">|&gt;</span></span>
<span id="cb35-15"><a href="#cb35-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb35-16"><a href="#cb35-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb35-17"><a href="#cb35-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="tbl-ordbeta" class="cell quarto-float quarto-figure quarto-figure-center" data-tblnum="6" data-after-caption-space="0pt" prefix="">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-ordbeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;6</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Marginal effect of fluency ordered beta model</p>
</div>
</figcaption>
<div aria-describedby="tbl-ordbeta-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script src="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.js"></script>

    <script>
      // Create table-specific functions using external factory
      const tableFns_dbpuiruckbl8wo2gpln2 = TinyTable.createTableFunctions("tinytable_dbpuiruckbl8wo2gpln2");
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: '1', j: 1 }, { i: '1', j: 2 }, { i: '1', j: 3 }, { i: '1', j: 4 } ], css_id: 'tinytable_css_iqv7cdzdcahm6umr7nms',}, 
          { positions: [ { i: '0', j: 1 }, { i: '0', j: 2 }, { i: '0', j: 3 }, { i: '0', j: 4 } ], css_id: 'tinytable_css_mi3505jpo06se5cxmo9m',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  tableFns_dbpuiruckbl8wo2gpln2.styleCell(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.css">
    <style>
    /* tinytable css entries after */
    #tinytable_dbpuiruckbl8wo2gpln2 td.tinytable_css_iqv7cdzdcahm6umr7nms, #tinytable_dbpuiruckbl8wo2gpln2 th.tinytable_css_iqv7cdzdcahm6umr7nms {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 0; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.1em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    #tinytable_dbpuiruckbl8wo2gpln2 td.tinytable_css_mi3505jpo06se5cxmo9m, #tinytable_dbpuiruckbl8wo2gpln2 th.tinytable_css_mi3505jpo06se5cxmo9m {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 1; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.05em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    </style>
    <div class="container">
      <table class="tinytable" id="tinytable_dbpuiruckbl8wo2gpln2" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        
        <thead>
              <tr>
                <th scope="col" data-row="0" data-col="1">Contrast</th>
                <th scope="col" data-row="0" data-col="2">Mean</th>
                <th scope="col" data-row="0" data-col="3">95% Cr.I</th>
                <th scope="col" data-row="0" data-col="4">pd</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td data-row="1" data-col="1">Fluent - Disfluent</td>
                  <td data-row="1" data-col="2">0.061</td>
                  <td data-row="1" data-col="3">[-0.016, 0.139]</td>
                  <td data-row="1" data-col="4">0.94</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>In <a href="#tbl-ordbeta" class="quarto-xref" aria-expanded="false">Table&nbsp;6</a> the credible interval is close enough to zero relative to its uncertainty that we can conclude there likely aren’t differences between the conditions after taking dispersion and the 0s and 1s in our data into account.</p>
</div>
</section>
<section id="cutpoints" class="level4">
<h4 data-anchor-id="cutpoints">Cutpoints.</h4>
<p>The model cutpoints are not reported by default in the summary output, but we can access them with the R package <code>posterior</code> <span class="citation" data-cites="posterior2025a">(Bürkner et al., 2025)</span> and the functions <code>as_draws</code> and <code>summary_draws</code>.</p>
<div class="cell FigureWithoutNote" data-after-caption-space="0pt" prefix="" data-tblnum="7" data-custom-style="FigureWithoutNote">
<div id="tbl-cutpoints" class="cell quarto-float quarto-figure quarto-figure-center" data-tblnum="7" data-after-caption-space="0pt" prefix="">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-cutpoints-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;7</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Cutzero and cutone parameter summary</p>
</div>
</figcaption>
<div aria-describedby="tbl-cutpoints-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script src="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.js"></script>

    <script>
      // Create table-specific functions using external factory
      const tableFns_k5emcq2uabt4svtttx06 = TinyTable.createTableFunctions("tinytable_k5emcq2uabt4svtttx06");
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: '2', j: 1 }, { i: '2', j: 2 }, { i: '2', j: 3 } ], css_id: 'tinytable_css_ckgl26kxd2s61a1p991g',}, 
          { positions: [ { i: '0', j: 1 }, { i: '0', j: 2 }, { i: '0', j: 3 } ], css_id: 'tinytable_css_jqp8u2fx8qcrrlut6t96',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  tableFns_k5emcq2uabt4svtttx06.styleCell(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.css">
    <style>
    /* tinytable css entries after */
    #tinytable_k5emcq2uabt4svtttx06 td.tinytable_css_ckgl26kxd2s61a1p991g, #tinytable_k5emcq2uabt4svtttx06 th.tinytable_css_ckgl26kxd2s61a1p991g {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 0; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.1em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    #tinytable_k5emcq2uabt4svtttx06 td.tinytable_css_jqp8u2fx8qcrrlut6t96, #tinytable_k5emcq2uabt4svtttx06 th.tinytable_css_jqp8u2fx8qcrrlut6t96 {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 1; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.05em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%;  }
    </style>
    <div class="container">
      <table class="tinytable" id="tinytable_k5emcq2uabt4svtttx06" style="width: auto; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        
        <thead>
              <tr>
                <th scope="col" data-row="0" data-col="1">Parameter</th>
                <th scope="col" data-row="0" data-col="2">Mean</th>
                <th scope="col" data-row="0" data-col="3">95% Cr.I</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td data-row="1" data-col="1">cutzero</td>
                  <td data-row="1" data-col="2">-2.97</td>
                  <td data-row="1" data-col="3">[-3.57, -2.44]</td>
                </tr>
                <tr>
                  <td data-row="2" data-col="1">cutone</td>
                  <td data-row="2" data-col="2">1.85</td>
                  <td data-row="2" data-col="3">[1.64, 2.07]</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>In <a href="#tbl-cutpoints" class="quarto-xref" aria-expanded="false">Table&nbsp;7</a>, <code>cutzero</code> is the first cutpoint (the difference between 0 and continuous values) and <code>cutone</code> is the second cutpoint (the difference between the continuous values and 1). These cutpoints are on the logit scale and as such the numbers do not have a simple substantive meaning. In general, as the cutpoints increase in absolute value (away from zero), then the discrete/boundary observations are more distinct from the continuous values. This will happen if there is a clear gap or bunching in the outcome around the bounds. This type of empirical feature of the distribution may be useful to scholars if they want to study differences in how people perceive the ends of the scale versus the middle. It is possible, though beyond the scope of this article, to model the location of the cutpoints with hierarchical (non-linear) covariates in <code>brms</code>. In the most recent version of <code>ordbeta</code>, it is possible to test the influence of different factors on these boundaries.</p>
</div>
</section>
</section>
<section id="model-fit" class="level3">
<h3 data-anchor-id="model-fit">Model Fit</h3>
<p>The best way to visualize model fit is to plot the full predictive distribution relative to the original outcome. Because ordered beta regression is a mixed discrete/continuous model, a separate plotting function, <code>pp_check_ordbetareg</code>, is included in the {ordbetareg} package that accurately handles the unique features of this distribution. The default plot in <code>brms</code> will collapse these two features of the outcome together, which will make the fit look worse than it actually is. The {ordbetareg} function returns a list with two plots, <code>discrete</code> and <code>continuous</code>, which can either be printed and plotted or further modified as {ggplot2} objects (see <a href="#fig-ppcheckord" class="quarto-xref" aria-expanded="false">Figure&nbsp;10</a>).</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">pp_check_ordbeta</span>(</span>
<span id="cb36-2"><a href="#cb36-2" aria-hidden="true" tabindex="-1"></a>  m.phi,</span>
<span id="cb36-3"><a href="#cb36-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">ndraws =</span> <span class="dv">100</span>,</span>
<span id="cb36-4"><a href="#cb36-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">outcome_label =</span> <span class="st">"Final Test Accuracy"</span></span>
<span id="cb36-5"><a href="#cb36-5" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb36-6"><a href="#cb36-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-7"><a href="#cb36-7" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> plots <span class="sc">+</span> <span class="fu">theme_publication</span>()</span>
<span id="cb36-8"><a href="#cb36-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb36-9"><a href="#cb36-9" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plots<span class="sc">$</span>discrete, plots<span class="sc">$</span>continuous, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-ppcheckord" class="quarto-float quarto-figure quarto-figure-center" data-fignum="10" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-ppcheckord-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;10</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Posterior predictive check for ordered beta regression model. A. Discrete posterior check. B. Continuous posterior check.</p>
</div>
</figcaption>
<div aria-describedby="fig-ppcheckord-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-ppcheckord-1.png" class="img-fluid figure-img" width="3600">
</div>
</figure>
</div>
</div>
</div>
<div class="AfterWithoutNote" data-custom-style="AfterWithoutNote">
<p>The discrete plot, which is a bar graph, shows that the posterior distribution accurately captures the number of different types of responses (discrete or continuous) in the data. For the continuous plot shown as a density plot with one line per posterior draw, the model does a very good job at capturing the distribution.</p>
</div>
<p>Overall, it is clear from the posterior distribution plot that the ordered beta model fits the data well. To fully understand model fit, both of these plots need to be inspected as they are conceptually distinct.</p>
</section>
<section id="model-visualization" class="level3">
<h3 data-anchor-id="model-visualization">Model Visualization</h3>
<p>{ordbetareg} provides a useful visualization function called <code>plot_heiss()</code> <span class="citation" data-cites="yeEnforcingBoundariesChinas2023">(Ye &amp; Heiss, 2023)</span> that can represent dispersion in the entire outcome as a function of discrete covariates. This function produces a plot of predicted proportions across the range of our <code>Fluency</code> factor. In <a href="#fig-heiss" class="quarto-xref" aria-expanded="false">Figure&nbsp;11</a> we get predicted proportions for Fluency across the bounded scale. Looking at the figure we can see there is much overlap between instructors in the middle portion (<span class="math inline">\mu</span>) . However, we do see some small differences at the zero bounds.</p>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_heiss</span>(</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a>  ord_fit_brms,</span>
<span id="cb37-3"><a href="#cb37-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">grouping_fac =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb37-4"><a href="#cb37-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">recode_group_labels =</span> <span class="fu">c</span>(<span class="st">"Disfluency"</span>, <span class="st">"Fluency"</span>),</span>
<span id="cb37-5"><a href="#cb37-5" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-heiss" class="quarto-float quarto-figure quarto-figure-center" data-fignum="11" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-heiss-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;11</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Heiss plot of predicted probabilities across the scale (0-100)</p>
</div>
</figcaption>
<div aria-describedby="fig-heiss-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-heiss-1.png" class="img-fluid figure-img" width="1800">
</div>
</figure>
</div>
</div>
</div>
</section>
<section id="ordered-beta-scale" class="level3">
<h3 data-anchor-id="ordered-beta-scale">Ordered Beta Scale</h3>
<p>In the {ordbetareg} function there is a <code>true_bound</code> argument. In cases where your data is not bounded between 0-1, this argument can be used to specify the bounds of the argument to fit the ordered beta regression. For example, the response data might be bounded between 1 and 7. If so, {ordbetareg} can model it within the [0,1] interval and {ordbetareg} will convert the model predictions back to the true bounds after estimation.</p>
<div class="cell">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># fit models from ran models so tables and figures are in correct place.</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>model1 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"model_reg_bayes.rds"</span>))</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>model2 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"model_beta_bayes_dis_run01.rds"</span>))</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>model3 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_zib_model0not1.rds"</span>))</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a>model4 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_zoib_model.rds"</span>))</span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>model5 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_ordbeta_phi_model.rds"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
<div class="cell FigureWithNote" data-tbl-align="center" data-after-caption-space="0pt" data-apa-note="Link functions: b_mean = logit; b_phi = log; b_zoi (zero-one inflation) = logit; b_coi (conditional one-inflation) = logit. Asterisks reflect approximate two-sided p-values derived from the posterior pd. pd ≥ 0.975 (p ≤ .05) = `*`; pd ≥ 0.990 (p ≤ .01) = `**`; pd ≥ 0.998 (p ≤ .001) = `***`." prefix="" data-tblnum="8" data-custom-style="FigureWithNote">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define your models</span></span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a>model_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Bayesian LM"</span> <span class="ot">=</span> model1,</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Beta Regression"</span> <span class="ot">=</span> model2,</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ZIB"</span> <span class="ot">=</span> model3,</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ZOIB"</span> <span class="ot">=</span> model4,</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Ordered Beta"</span> <span class="ot">=</span> model5</span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Process each model</span></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a>cleaned_tables <span class="ot">&lt;-</span> <span class="fu">imap</span>(model_list, <span class="cf">function</span>(mod, name) {</span>
<span id="cb39-12"><a href="#cb39-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">model_parameters</span>(mod, <span class="at">centrality =</span> <span class="st">"mean"</span>) <span class="sc">|&gt;</span></span>
<span id="cb39-13"><a href="#cb39-13" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb39-14"><a href="#cb39-14" aria-hidden="true" tabindex="-1"></a>      <span class="at">Mean =</span> <span class="fu">sprintf</span>(<span class="st">"%.3f"</span>, Mean),</span>
<span id="cb39-15"><a href="#cb39-15" aria-hidden="true" tabindex="-1"></a>      <span class="at">Cr.I =</span> <span class="fu">str_glue</span>(<span class="st">"[{round(CI_low, 3)}, {round(CI_high, 3)}]"</span>),</span>
<span id="cb39-16"><a href="#cb39-16" aria-hidden="true" tabindex="-1"></a>      <span class="at">pd_star =</span> <span class="fu">case_when</span>(</span>
<span id="cb39-17"><a href="#cb39-17" aria-hidden="true" tabindex="-1"></a>        pd <span class="sc">&gt;=</span> <span class="fl">0.995</span> <span class="sc">~</span> <span class="fu">str_glue</span>(<span class="st">"{sprintf('%.3f', pd)}***"</span>),</span>
<span id="cb39-18"><a href="#cb39-18" aria-hidden="true" tabindex="-1"></a>        pd <span class="sc">&gt;=</span> <span class="fl">0.985</span> <span class="sc">~</span> <span class="fu">str_glue</span>(<span class="st">"{sprintf('%.3f', pd)}**"</span>),</span>
<span id="cb39-19"><a href="#cb39-19" aria-hidden="true" tabindex="-1"></a>        pd <span class="sc">&gt;=</span> <span class="fl">0.975</span> <span class="sc">~</span> <span class="fu">str_glue</span>(<span class="st">"{sprintf('%.3f', pd)}*"</span>),</span>
<span id="cb39-20"><a href="#cb39-20" aria-hidden="true" tabindex="-1"></a>        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="fu">sprintf</span>(<span class="st">"%.3f"</span>, pd)</span>
<span id="cb39-21"><a href="#cb39-21" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb39-22"><a href="#cb39-22" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span></span>
<span id="cb39-23"><a href="#cb39-23" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(</span>
<span id="cb39-24"><a href="#cb39-24" aria-hidden="true" tabindex="-1"></a>      <span class="at">cols =</span> <span class="fu">c</span>(Mean, Cr.I, pd_star),</span>
<span id="cb39-25"><a href="#cb39-25" aria-hidden="true" tabindex="-1"></a>      <span class="at">names_to =</span> <span class="st">"Stat"</span>,</span>
<span id="cb39-26"><a href="#cb39-26" aria-hidden="true" tabindex="-1"></a>      <span class="at">values_to =</span> <span class="st">"value"</span></span>
<span id="cb39-27"><a href="#cb39-27" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span></span>
<span id="cb39-28"><a href="#cb39-28" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb39-29"><a href="#cb39-29" aria-hidden="true" tabindex="-1"></a>      <span class="at">Stat =</span> <span class="fu">recode</span>(Stat, <span class="at">pd_star =</span> <span class="st">"pd"</span>),</span>
<span id="cb39-30"><a href="#cb39-30" aria-hidden="true" tabindex="-1"></a>      <span class="at">Model =</span> name</span>
<span id="cb39-31"><a href="#cb39-31" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span></span>
<span id="cb39-32"><a href="#cb39-32" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(Parameter, Stat, Model, value)</span>
<span id="cb39-33"><a href="#cb39-33" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb39-34"><a href="#cb39-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-35"><a href="#cb39-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine all tables</span></span>
<span id="cb39-36"><a href="#cb39-36" aria-hidden="true" tabindex="-1"></a>pretty_table <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cleaned_tables) <span class="sc">|&gt;</span></span>
<span id="cb39-37"><a href="#cb39-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Model, <span class="at">values_from =</span> value) <span class="sc">|&gt;</span></span>
<span id="cb39-38"><a href="#cb39-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.character), <span class="sc">~</span> <span class="fu">replace_na</span>(.x, <span class="st">"-"</span>))) <span class="sc">|&gt;</span></span>
<span id="cb39-39"><a href="#cb39-39" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-40"><a href="#cb39-40" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Simplify parameter names</span></span>
<span id="cb39-41"><a href="#cb39-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Parameter =</span> <span class="fu">str_replace_all</span>(Parameter, <span class="st">"FluencyFluent"</span>, <span class="st">"Fluency"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb39-42"><a href="#cb39-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-43"><a href="#cb39-43" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set factor levels</span></span>
<span id="cb39-44"><a href="#cb39-44" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb39-45"><a href="#cb39-45" aria-hidden="true" tabindex="-1"></a>    <span class="at">Stat =</span> <span class="fu">factor</span>(Stat, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Mean"</span>, <span class="st">"Cr.I"</span>, <span class="st">"pd"</span>)),</span>
<span id="cb39-46"><a href="#cb39-46" aria-hidden="true" tabindex="-1"></a>    <span class="at">Parameter =</span> <span class="fu">factor</span>(</span>
<span id="cb39-47"><a href="#cb39-47" aria-hidden="true" tabindex="-1"></a>      Parameter,</span>
<span id="cb39-48"><a href="#cb39-48" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb39-49"><a href="#cb39-49" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_Intercept"</span>,</span>
<span id="cb39-50"><a href="#cb39-50" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_Fluency"</span>,</span>
<span id="cb39-51"><a href="#cb39-51" aria-hidden="true" tabindex="-1"></a>        <span class="st">"sigma"</span>,</span>
<span id="cb39-52"><a href="#cb39-52" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_phi_Intercept"</span>,</span>
<span id="cb39-53"><a href="#cb39-53" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_phi_Fluency"</span>,</span>
<span id="cb39-54"><a href="#cb39-54" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_zi_Intercept"</span>,</span>
<span id="cb39-55"><a href="#cb39-55" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_zi_Fluency"</span>,</span>
<span id="cb39-56"><a href="#cb39-56" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_zoi_Intercept"</span>,</span>
<span id="cb39-57"><a href="#cb39-57" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_zoi_Fluency"</span>,</span>
<span id="cb39-58"><a href="#cb39-58" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_coi_Intercept"</span>,</span>
<span id="cb39-59"><a href="#cb39-59" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_coi_Fluency"</span></span>
<span id="cb39-60"><a href="#cb39-60" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb39-61"><a href="#cb39-61" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb39-62"><a href="#cb39-62" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb39-63"><a href="#cb39-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-64"><a href="#cb39-64" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Parameter, Stat) <span class="sc">|&gt;</span></span>
<span id="cb39-65"><a href="#cb39-65" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Parameter) <span class="sc">|&gt;</span></span>
<span id="cb39-66"><a href="#cb39-66" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Parameter =</span> <span class="fu">if_else</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>, <span class="fu">as.character</span>(Parameter), <span class="st">""</span>)) <span class="sc">|&gt;</span></span>
<span id="cb39-67"><a href="#cb39-67" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb39-68"><a href="#cb39-68" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb39-69"><a href="#cb39-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Display as styled table</span></span>
<span id="cb39-70"><a href="#cb39-70" aria-hidden="true" tabindex="-1"></a>pretty_table <span class="sc">|&gt;</span></span>
<span id="cb39-71"><a href="#cb39-71" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>(<span class="at">width =</span> <span class="fl">0.95</span>) <span class="sc">|&gt;</span></span>
<span id="cb39-72"><a href="#cb39-72" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb39-73"><a href="#cb39-73" aria-hidden="true" tabindex="-1"></a>  <span class="fu">style_tt</span>(<span class="at">fontsize =</span> .<span class="dv">9</span>, <span class="at">align =</span> <span class="st">"c"</span>) <span class="sc">|&gt;</span></span>
<span id="cb39-74"><a href="#cb39-74" aria-hidden="true" tabindex="-1"></a>  <span class="fu">style_tt</span>(</span>
<span id="cb39-75"><a href="#cb39-75" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> <span class="fu">c</span>(</span>
<span id="cb39-76"><a href="#cb39-76" aria-hidden="true" tabindex="-1"></a>      <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>,</span>
<span id="cb39-77"><a href="#cb39-77" aria-hidden="true" tabindex="-1"></a>      <span class="dv">5</span><span class="sc">:</span><span class="dv">6</span>,</span>
<span id="cb39-78"><a href="#cb39-78" aria-hidden="true" tabindex="-1"></a>      <span class="dv">8</span><span class="sc">:</span><span class="dv">9</span>,</span>
<span id="cb39-79"><a href="#cb39-79" aria-hidden="true" tabindex="-1"></a>      <span class="dv">11</span><span class="sc">:</span><span class="dv">12</span>,</span>
<span id="cb39-80"><a href="#cb39-80" aria-hidden="true" tabindex="-1"></a>      <span class="dv">14</span><span class="sc">:</span><span class="dv">15</span>,</span>
<span id="cb39-81"><a href="#cb39-81" aria-hidden="true" tabindex="-1"></a>      <span class="dv">17</span><span class="sc">:</span><span class="dv">18</span>,</span>
<span id="cb39-82"><a href="#cb39-82" aria-hidden="true" tabindex="-1"></a>      <span class="dv">20</span><span class="sc">:</span><span class="dv">21</span>,</span>
<span id="cb39-83"><a href="#cb39-83" aria-hidden="true" tabindex="-1"></a>      <span class="dv">23</span><span class="sc">:</span><span class="dv">24</span>,</span>
<span id="cb39-84"><a href="#cb39-84" aria-hidden="true" tabindex="-1"></a>      <span class="dv">26</span><span class="sc">:</span><span class="dv">27</span>,</span>
<span id="cb39-85"><a href="#cb39-85" aria-hidden="true" tabindex="-1"></a>      <span class="dv">29</span><span class="sc">:</span><span class="dv">30</span>,</span>
<span id="cb39-86"><a href="#cb39-86" aria-hidden="true" tabindex="-1"></a>      <span class="dv">32</span><span class="sc">:</span><span class="dv">33</span></span>
<span id="cb39-87"><a href="#cb39-87" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb39-88"><a href="#cb39-88" aria-hidden="true" tabindex="-1"></a>    <span class="at">italic =</span> <span class="cn">TRUE</span>,</span>
<span id="cb39-89"><a href="#cb39-89" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontsize =</span> .<span class="dv">6</span></span>
<span id="cb39-90"><a href="#cb39-90" aria-hidden="true" tabindex="-1"></a>  )</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div id="tbl-all" class="cell quarto-float quarto-figure quarto-figure-center" data-tblnum="8" data-after-caption-space="0pt" data-tbl-align="center" data-apa-note="Link functions: b_mean = logit; b_phi = log; b_zoi (zero-one inflation) = logit; b_coi (conditional one-inflation) = logit. Asterisks reflect approximate two-sided p-values derived from the posterior pd. pd ≥ 0.975 (p ≤ .05) = `*`; pd ≥ 0.990 (p ≤ .01) = `**`; pd ≥ 0.998 (p ≤ .001) = `***`." prefix="">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-all-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Table&nbsp;8</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>Bayesian regression summaries for each model</p>
</div>
</figcaption>
<div aria-describedby="tbl-all-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="cell-output-display">
<!-- preamble start -->

    <script src="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.js"></script>

    <script>
      // Create table-specific functions using external factory
      const tableFns_mukqpsbi6dg5lyyxv6y0 = TinyTable.createTableFunctions("tinytable_mukqpsbi6dg5lyyxv6y0");
      // tinytable span after
      window.addEventListener('load', function () {
          var cellsToStyle = [
            // tinytable style arrays after
          { positions: [ { i: '2', j: 1 }, { i: '3', j: 1 }, { i: '5', j: 1 }, { i: '6', j: 1 }, { i: '8', j: 1 }, { i: '9', j: 1 }, { i: '11', j: 1 }, { i: '12', j: 1 }, { i: '14', j: 1 }, { i: '15', j: 1 }, { i: '17', j: 1 }, { i: '18', j: 1 }, { i: '20', j: 1 }, { i: '21', j: 1 }, { i: '23', j: 1 }, { i: '24', j: 1 }, { i: '26', j: 1 }, { i: '27', j: 1 }, { i: '29', j: 1 }, { i: '30', j: 1 }, { i: '32', j: 1 }, { i: '2', j: 2 }, { i: '3', j: 2 }, { i: '5', j: 2 }, { i: '6', j: 2 }, { i: '8', j: 2 }, { i: '9', j: 2 }, { i: '11', j: 2 }, { i: '12', j: 2 }, { i: '14', j: 2 }, { i: '15', j: 2 }, { i: '17', j: 2 }, { i: '18', j: 2 }, { i: '20', j: 2 }, { i: '21', j: 2 }, { i: '23', j: 2 }, { i: '24', j: 2 }, { i: '26', j: 2 }, { i: '27', j: 2 }, { i: '29', j: 2 }, { i: '30', j: 2 }, { i: '32', j: 2 }, { i: '2', j: 3 }, { i: '3', j: 3 }, { i: '5', j: 3 }, { i: '6', j: 3 }, { i: '8', j: 3 }, { i: '9', j: 3 }, { i: '11', j: 3 }, { i: '12', j: 3 }, { i: '14', j: 3 }, { i: '15', j: 3 }, { i: '17', j: 3 }, { i: '18', j: 3 }, { i: '20', j: 3 }, { i: '21', j: 3 }, { i: '23', j: 3 }, { i: '24', j: 3 }, { i: '26', j: 3 }, { i: '27', j: 3 }, { i: '29', j: 3 }, { i: '30', j: 3 }, { i: '32', j: 3 }, { i: '2', j: 4 }, { i: '3', j: 4 }, { i: '5', j: 4 }, { i: '6', j: 4 }, { i: '8', j: 4 }, { i: '9', j: 4 }, { i: '11', j: 4 }, { i: '12', j: 4 }, { i: '14', j: 4 }, { i: '15', j: 4 }, { i: '17', j: 4 }, { i: '18', j: 4 }, { i: '20', j: 4 }, { i: '21', j: 4 }, { i: '23', j: 4 }, { i: '24', j: 4 }, { i: '26', j: 4 }, { i: '27', j: 4 }, { i: '29', j: 4 }, { i: '30', j: 4 }, { i: '32', j: 4 }, { i: '2', j: 5 }, { i: '3', j: 5 }, { i: '5', j: 5 }, { i: '6', j: 5 }, { i: '8', j: 5 }, { i: '9', j: 5 }, { i: '11', j: 5 }, { i: '12', j: 5 }, { i: '14', j: 5 }, { i: '15', j: 5 }, { i: '17', j: 5 }, { i: '18', j: 5 }, { i: '20', j: 5 }, { i: '21', j: 5 }, { i: '23', j: 5 }, { i: '24', j: 5 }, { i: '26', j: 5 }, { i: '27', j: 5 }, { i: '29', j: 5 }, { i: '30', j: 5 }, { i: '32', j: 5 }, { i: '2', j: 6 }, { i: '3', j: 6 }, { i: '5', j: 6 }, { i: '6', j: 6 }, { i: '8', j: 6 }, { i: '9', j: 6 }, { i: '11', j: 6 }, { i: '12', j: 6 }, { i: '14', j: 6 }, { i: '15', j: 6 }, { i: '17', j: 6 }, { i: '18', j: 6 }, { i: '20', j: 6 }, { i: '21', j: 6 }, { i: '23', j: 6 }, { i: '24', j: 6 }, { i: '26', j: 6 }, { i: '27', j: 6 }, { i: '29', j: 6 }, { i: '30', j: 6 }, { i: '32', j: 6 }, { i: '2', j: 7 }, { i: '3', j: 7 }, { i: '5', j: 7 }, { i: '6', j: 7 }, { i: '8', j: 7 }, { i: '9', j: 7 }, { i: '11', j: 7 }, { i: '12', j: 7 }, { i: '14', j: 7 }, { i: '15', j: 7 }, { i: '17', j: 7 }, { i: '18', j: 7 }, { i: '20', j: 7 }, { i: '21', j: 7 }, { i: '23', j: 7 }, { i: '24', j: 7 }, { i: '26', j: 7 }, { i: '27', j: 7 }, { i: '29', j: 7 }, { i: '30', j: 7 }, { i: '32', j: 7 } ], css_id: 'tinytable_css_0bog2xgnuka8a61xzltx',}, 
          { positions: [ { i: '1', j: 1 }, { i: '4', j: 1 }, { i: '7', j: 1 }, { i: '10', j: 1 }, { i: '13', j: 1 }, { i: '16', j: 1 }, { i: '19', j: 1 }, { i: '22', j: 1 }, { i: '25', j: 1 }, { i: '28', j: 1 }, { i: '31', j: 1 }, { i: '1', j: 2 }, { i: '4', j: 2 }, { i: '7', j: 2 }, { i: '10', j: 2 }, { i: '13', j: 2 }, { i: '16', j: 2 }, { i: '19', j: 2 }, { i: '22', j: 2 }, { i: '25', j: 2 }, { i: '28', j: 2 }, { i: '31', j: 2 }, { i: '1', j: 3 }, { i: '4', j: 3 }, { i: '7', j: 3 }, { i: '10', j: 3 }, { i: '13', j: 3 }, { i: '16', j: 3 }, { i: '19', j: 3 }, { i: '22', j: 3 }, { i: '25', j: 3 }, { i: '28', j: 3 }, { i: '31', j: 3 }, { i: '1', j: 4 }, { i: '4', j: 4 }, { i: '7', j: 4 }, { i: '10', j: 4 }, { i: '13', j: 4 }, { i: '16', j: 4 }, { i: '19', j: 4 }, { i: '22', j: 4 }, { i: '25', j: 4 }, { i: '28', j: 4 }, { i: '31', j: 4 }, { i: '1', j: 5 }, { i: '4', j: 5 }, { i: '7', j: 5 }, { i: '10', j: 5 }, { i: '13', j: 5 }, { i: '16', j: 5 }, { i: '19', j: 5 }, { i: '22', j: 5 }, { i: '25', j: 5 }, { i: '28', j: 5 }, { i: '31', j: 5 }, { i: '1', j: 6 }, { i: '4', j: 6 }, { i: '7', j: 6 }, { i: '10', j: 6 }, { i: '13', j: 6 }, { i: '16', j: 6 }, { i: '19', j: 6 }, { i: '22', j: 6 }, { i: '25', j: 6 }, { i: '28', j: 6 }, { i: '31', j: 6 }, { i: '1', j: 7 }, { i: '4', j: 7 }, { i: '7', j: 7 }, { i: '10', j: 7 }, { i: '13', j: 7 }, { i: '16', j: 7 }, { i: '19', j: 7 }, { i: '22', j: 7 }, { i: '25', j: 7 }, { i: '28', j: 7 }, { i: '31', j: 7 } ], css_id: 'tinytable_css_yonz1m2h99kfp86eisp4',}, 
          { positions: [ { i: '33', j: 1 }, { i: '33', j: 2 }, { i: '33', j: 3 }, { i: '33', j: 4 }, { i: '33', j: 5 }, { i: '33', j: 6 }, { i: '33', j: 7 } ], css_id: 'tinytable_css_2ocf6grd3707ou34u44h',}, 
          { positions: [ { i: '0', j: 1 }, { i: '0', j: 2 }, { i: '0', j: 3 }, { i: '0', j: 4 }, { i: '0', j: 5 }, { i: '0', j: 6 }, { i: '0', j: 7 } ], css_id: 'tinytable_css_x1ysjec8gnopge223kk4',}, 
          ];

          // Loop over the arrays to style the cells
          cellsToStyle.forEach(function (group) {
              group.positions.forEach(function (cell) {
                  tableFns_mukqpsbi6dg5lyyxv6y0.styleCell(cell.i, cell.j, group.css_id);
              });
          });
      });
    </script>

    <link rel="stylesheet" href="https://cdn.jsdelivr.net/gh/vincentarelbundock/tinytable@main/inst/tinytable.css">
    <style>
    /* tinytable css entries after */
    #tinytable_mukqpsbi6dg5lyyxv6y0 td.tinytable_css_0bog2xgnuka8a61xzltx, #tinytable_mukqpsbi6dg5lyyxv6y0 th.tinytable_css_0bog2xgnuka8a61xzltx { font-style: italic; text-align: center; font-size: 0.9em }
    #tinytable_mukqpsbi6dg5lyyxv6y0 td.tinytable_css_yonz1m2h99kfp86eisp4, #tinytable_mukqpsbi6dg5lyyxv6y0 th.tinytable_css_yonz1m2h99kfp86eisp4 { text-align: center; font-size: 0.9em }
    #tinytable_mukqpsbi6dg5lyyxv6y0 td.tinytable_css_2ocf6grd3707ou34u44h, #tinytable_mukqpsbi6dg5lyyxv6y0 th.tinytable_css_2ocf6grd3707ou34u44h {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 0; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.1em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%; ; font-style: italic; text-align: center; font-size: 0.9em }
    #tinytable_mukqpsbi6dg5lyyxv6y0 td.tinytable_css_x1ysjec8gnopge223kk4, #tinytable_mukqpsbi6dg5lyyxv6y0 th.tinytable_css_x1ysjec8gnopge223kk4 {  position: relative; --border-bottom: 1; --border-left: 0; --border-right: 0; --border-top: 1; --line-color-bottom: black; --line-color-left: black; --line-color-right: black; --line-color-top: black; --line-width-bottom: 0.05em; --line-width-left: 0.1em; --line-width-right: 0.1em; --line-width-top: 0.1em; --trim-bottom-left: 0%; --trim-bottom-right: 0%; --trim-left-bottom: 0%; --trim-left-top: 0%; --trim-right-bottom: 0%; --trim-right-top: 0%; --trim-top-left: 0%; --trim-top-right: 0%; ; text-align: center; font-size: 0.9em }
    </style>
    <div class="container">
      <table class="tinytable" id="tinytable_mukqpsbi6dg5lyyxv6y0" style="table-layout: fixed; width: 95% !important; margin-left: auto; margin-right: auto;" data-quarto-disable-processing="true">
        
        <thead>
              <tr>
                <th scope="col" data-row="0" data-col="1">Parameter</th>
                <th scope="col" data-row="0" data-col="2">Stat</th>
                <th scope="col" data-row="0" data-col="3">Bayesian LM</th>
                <th scope="col" data-row="0" data-col="4">Beta Regression</th>
                <th scope="col" data-row="0" data-col="5">ZIB</th>
                <th scope="col" data-row="0" data-col="6">ZOIB</th>
                <th scope="col" data-row="0" data-col="7">Ordered Beta</th>
              </tr>
        </thead>
        
        <tbody>
                <tr>
                  <td data-row="1" data-col="1">b_Intercept</td>
                  <td data-row="1" data-col="2">Mean</td>
                  <td data-row="1" data-col="3">0.256</td>
                  <td data-row="1" data-col="4">-0.831</td>
                  <td data-row="1" data-col="5">-0.828</td>
                  <td data-row="1" data-col="6">-0.826</td>
                  <td data-row="1" data-col="7">-0.865</td>
                </tr>
                <tr>
                  <td data-row="2" data-col="1"></td>
                  <td data-row="2" data-col="2">Cr.I</td>
                  <td data-row="2" data-col="3">[0.201, 0.313]</td>
                  <td data-row="2" data-col="4">[-1.102, -0.561]</td>
                  <td data-row="2" data-col="5">[-1.1, -0.564]</td>
                  <td data-row="2" data-col="6">[-1.093, -0.556]</td>
                  <td data-row="2" data-col="7">[-1.124, -0.606]</td>
                </tr>
                <tr>
                  <td data-row="3" data-col="1"></td>
                  <td data-row="3" data-col="2">pd</td>
                  <td data-row="3" data-col="3">1.000***</td>
                  <td data-row="3" data-col="4">1.000***</td>
                  <td data-row="3" data-col="5">1.000***</td>
                  <td data-row="3" data-col="6">1.000***</td>
                  <td data-row="3" data-col="7">1.000***</td>
                </tr>
                <tr>
                  <td data-row="4" data-col="1">b_Fluency</td>
                  <td data-row="4" data-col="2">Mean</td>
                  <td data-row="4" data-col="3">0.085</td>
                  <td data-row="4" data-col="4">0.201</td>
                  <td data-row="4" data-col="5">0.198</td>
                  <td data-row="4" data-col="6">0.198</td>
                  <td data-row="4" data-col="7">0.261</td>
                </tr>
                <tr>
                  <td data-row="5" data-col="1"></td>
                  <td data-row="5" data-col="2">Cr.I</td>
                  <td data-row="5" data-col="3">[0, 0.167]</td>
                  <td data-row="5" data-col="4">[-0.156, 0.552]</td>
                  <td data-row="5" data-col="5">[-0.14, 0.543]</td>
                  <td data-row="5" data-col="6">[-0.151, 0.54]</td>
                  <td data-row="5" data-col="7">[-0.069, 0.597]</td>
                </tr>
                <tr>
                  <td data-row="6" data-col="1"></td>
                  <td data-row="6" data-col="2">pd</td>
                  <td data-row="6" data-col="3">0.976*</td>
                  <td data-row="6" data-col="4">0.871</td>
                  <td data-row="6" data-col="5">0.875</td>
                  <td data-row="6" data-col="6">0.870</td>
                  <td data-row="6" data-col="7">0.940</td>
                </tr>
                <tr>
                  <td data-row="7" data-col="1">sigma</td>
                  <td data-row="7" data-col="2">Mean</td>
                  <td data-row="7" data-col="3">0.208</td>
                  <td data-row="7" data-col="4">-</td>
                  <td data-row="7" data-col="5">-</td>
                  <td data-row="7" data-col="6">-</td>
                  <td data-row="7" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="8" data-col="1"></td>
                  <td data-row="8" data-col="2">Cr.I</td>
                  <td data-row="8" data-col="3">[0.182, 0.24]</td>
                  <td data-row="8" data-col="4">-</td>
                  <td data-row="8" data-col="5">-</td>
                  <td data-row="8" data-col="6">-</td>
                  <td data-row="8" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="9" data-col="1"></td>
                  <td data-row="9" data-col="2">pd</td>
                  <td data-row="9" data-col="3">1.000***</td>
                  <td data-row="9" data-col="4">-</td>
                  <td data-row="9" data-col="5">-</td>
                  <td data-row="9" data-col="6">-</td>
                  <td data-row="9" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="10" data-col="1">b_phi_Intercept</td>
                  <td data-row="10" data-col="2">Mean</td>
                  <td data-row="10" data-col="3">-</td>
                  <td data-row="10" data-col="4">1.599</td>
                  <td data-row="10" data-col="5">1.603</td>
                  <td data-row="10" data-col="6">1.600</td>
                  <td data-row="10" data-col="7">1.619</td>
                </tr>
                <tr>
                  <td data-row="11" data-col="1"></td>
                  <td data-row="11" data-col="2">Cr.I</td>
                  <td data-row="11" data-col="3">-</td>
                  <td data-row="11" data-col="4">[1.172, 1.986]</td>
                  <td data-row="11" data-col="5">[1.183, 1.984]</td>
                  <td data-row="11" data-col="6">[1.176, 1.985]</td>
                  <td data-row="11" data-col="7">[1.204, 1.996]</td>
                </tr>
                <tr>
                  <td data-row="12" data-col="1"></td>
                  <td data-row="12" data-col="2">pd</td>
                  <td data-row="12" data-col="3">-</td>
                  <td data-row="12" data-col="4">1.000***</td>
                  <td data-row="12" data-col="5">1.000***</td>
                  <td data-row="12" data-col="6">1.000***</td>
                  <td data-row="12" data-col="7">1.000***</td>
                </tr>
                <tr>
                  <td data-row="13" data-col="1">b_phi_Fluency</td>
                  <td data-row="13" data-col="2">Mean</td>
                  <td data-row="13" data-col="3">-</td>
                  <td data-row="13" data-col="4">0.429</td>
                  <td data-row="13" data-col="5">0.424</td>
                  <td data-row="13" data-col="6">0.434</td>
                  <td data-row="13" data-col="7">0.396</td>
                </tr>
                <tr>
                  <td data-row="14" data-col="1"></td>
                  <td data-row="14" data-col="2">Cr.I</td>
                  <td data-row="14" data-col="3">-</td>
                  <td data-row="14" data-col="4">[-0.157, 0.996]</td>
                  <td data-row="14" data-col="5">[-0.155, 0.993]</td>
                  <td data-row="14" data-col="6">[-0.131, 1.021]</td>
                  <td data-row="14" data-col="7">[-0.202, 0.966]</td>
                </tr>
                <tr>
                  <td data-row="15" data-col="1"></td>
                  <td data-row="15" data-col="2">pd</td>
                  <td data-row="15" data-col="3">-</td>
                  <td data-row="15" data-col="4">0.928</td>
                  <td data-row="15" data-col="5">0.926</td>
                  <td data-row="15" data-col="6">0.929</td>
                  <td data-row="15" data-col="7">0.907</td>
                </tr>
                <tr>
                  <td data-row="16" data-col="1">b_zi_Intercept</td>
                  <td data-row="16" data-col="2">Mean</td>
                  <td data-row="16" data-col="3">-</td>
                  <td data-row="16" data-col="4">-</td>
                  <td data-row="16" data-col="5">-1.670</td>
                  <td data-row="16" data-col="6">-</td>
                  <td data-row="16" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="17" data-col="1"></td>
                  <td data-row="17" data-col="2">Cr.I</td>
                  <td data-row="17" data-col="3">-</td>
                  <td data-row="17" data-col="4">-</td>
                  <td data-row="17" data-col="5">[-2.476, -0.953]</td>
                  <td data-row="17" data-col="6">-</td>
                  <td data-row="17" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="18" data-col="1"></td>
                  <td data-row="18" data-col="2">pd</td>
                  <td data-row="18" data-col="3">-</td>
                  <td data-row="18" data-col="4">-</td>
                  <td data-row="18" data-col="5">1.000***</td>
                  <td data-row="18" data-col="6">-</td>
                  <td data-row="18" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="19" data-col="1">b_zi_Fluency</td>
                  <td data-row="19" data-col="2">Mean</td>
                  <td data-row="19" data-col="3">-</td>
                  <td data-row="19" data-col="4">-</td>
                  <td data-row="19" data-col="5">-2.115</td>
                  <td data-row="19" data-col="6">-</td>
                  <td data-row="19" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="20" data-col="1"></td>
                  <td data-row="20" data-col="2">Cr.I</td>
                  <td data-row="20" data-col="3">-</td>
                  <td data-row="20" data-col="4">-</td>
                  <td data-row="20" data-col="5">[-4.5, -0.34]</td>
                  <td data-row="20" data-col="6">-</td>
                  <td data-row="20" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="21" data-col="1"></td>
                  <td data-row="21" data-col="2">pd</td>
                  <td data-row="21" data-col="3">-</td>
                  <td data-row="21" data-col="4">-</td>
                  <td data-row="21" data-col="5">0.993**</td>
                  <td data-row="21" data-col="6">-</td>
                  <td data-row="21" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="22" data-col="1">b_zoi_Intercept</td>
                  <td data-row="22" data-col="2">Mean</td>
                  <td data-row="22" data-col="3">-</td>
                  <td data-row="22" data-col="4">-</td>
                  <td data-row="22" data-col="5">-</td>
                  <td data-row="22" data-col="6">-1.538</td>
                  <td data-row="22" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="23" data-col="1"></td>
                  <td data-row="23" data-col="2">Cr.I</td>
                  <td data-row="23" data-col="3">-</td>
                  <td data-row="23" data-col="4">-</td>
                  <td data-row="23" data-col="5">-</td>
                  <td data-row="23" data-col="6">[-2.303, -0.856]</td>
                  <td data-row="23" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="24" data-col="1"></td>
                  <td data-row="24" data-col="2">pd</td>
                  <td data-row="24" data-col="3">-</td>
                  <td data-row="24" data-col="4">-</td>
                  <td data-row="24" data-col="5">-</td>
                  <td data-row="24" data-col="6">1.000***</td>
                  <td data-row="24" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="25" data-col="1">b_zoi_Fluency</td>
                  <td data-row="25" data-col="2">Mean</td>
                  <td data-row="25" data-col="3">-</td>
                  <td data-row="25" data-col="4">-</td>
                  <td data-row="25" data-col="5">-</td>
                  <td data-row="25" data-col="6">-2.268</td>
                  <td data-row="25" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="26" data-col="1"></td>
                  <td data-row="26" data-col="2">Cr.I</td>
                  <td data-row="26" data-col="3">-</td>
                  <td data-row="26" data-col="4">-</td>
                  <td data-row="26" data-col="5">-</td>
                  <td data-row="26" data-col="6">[-4.737, -0.439]</td>
                  <td data-row="26" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="27" data-col="1"></td>
                  <td data-row="27" data-col="2">pd</td>
                  <td data-row="27" data-col="3">-</td>
                  <td data-row="27" data-col="4">-</td>
                  <td data-row="27" data-col="5">-</td>
                  <td data-row="27" data-col="6">0.995***</td>
                  <td data-row="27" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="28" data-col="1">b_coi_Intercept</td>
                  <td data-row="28" data-col="2">Mean</td>
                  <td data-row="28" data-col="3">-</td>
                  <td data-row="28" data-col="4">-</td>
                  <td data-row="28" data-col="5">-</td>
                  <td data-row="28" data-col="6">-2.062</td>
                  <td data-row="28" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="29" data-col="1"></td>
                  <td data-row="29" data-col="2">Cr.I</td>
                  <td data-row="29" data-col="3">-</td>
                  <td data-row="29" data-col="4">-</td>
                  <td data-row="29" data-col="5">-</td>
                  <td data-row="29" data-col="6">[-4.484, -0.311]</td>
                  <td data-row="29" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="30" data-col="1"></td>
                  <td data-row="30" data-col="2">pd</td>
                  <td data-row="30" data-col="3">-</td>
                  <td data-row="30" data-col="4">-</td>
                  <td data-row="30" data-col="5">-</td>
                  <td data-row="30" data-col="6">0.991**</td>
                  <td data-row="30" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="31" data-col="1">b_coi_Fluency</td>
                  <td data-row="31" data-col="2">Mean</td>
                  <td data-row="31" data-col="3">-</td>
                  <td data-row="31" data-col="4">-</td>
                  <td data-row="31" data-col="5">-</td>
                  <td data-row="31" data-col="6">0.157</td>
                  <td data-row="31" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="32" data-col="1"></td>
                  <td data-row="32" data-col="2">Cr.I</td>
                  <td data-row="32" data-col="3">-</td>
                  <td data-row="32" data-col="4">-</td>
                  <td data-row="32" data-col="5">-</td>
                  <td data-row="32" data-col="6">[-7.175, 5.776]</td>
                  <td data-row="32" data-col="7">-</td>
                </tr>
                <tr>
                  <td data-row="33" data-col="1"></td>
                  <td data-row="33" data-col="2">pd</td>
                  <td data-row="33" data-col="3">-</td>
                  <td data-row="33" data-col="4">-</td>
                  <td data-row="33" data-col="5">-</td>
                  <td data-row="33" data-col="6">0.565</td>
                  <td data-row="33" data-col="7">-</td>
                </tr>
        </tbody>
      </table>
    </div>
<!-- hack to avoid NA insertion in last line -->
</div>
</div>
</figure>
</div>
<div>
<div class="FigureNote NoIndent" data-custom-style="FigureNote">
<em>Note</em>. Link functions: b_mean = logit; b_phi = log; b_zoi (zero-one inflation) = logit; b_coi (conditional one-inflation) = logit. Asterisks reflect approximate two-sided p-values derived from the posterior pd. pd ≥ 0.975 (p ≤ .05) = <code>*</code>; pd ≥ 0.990 (p ≤ .01) = <code>**</code>; pd ≥ 0.998 (p ≤ .001) = <code>***</code>.
</div>
</div>
</div>
<div class="cell FigureWithoutNote" data-custom-style="FigureWithoutNote">
<details class="code-fold">
<summary>Show R code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>pp_check1 <span class="ot">&lt;-</span> <span class="fu">pp_check</span>(</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>  model1,</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">"dens_overlay_grouped"</span>,</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">ndraws =</span> <span class="dv">100</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>()</span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>pp_check2 <span class="ot">&lt;-</span> <span class="fu">pp_check</span>(</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>  model2,</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">"dens_overlay_grouped"</span>,</span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>  <span class="at">ndraws =</span> <span class="dv">100</span></span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb40-17"><a href="#cb40-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">6</span>)) <span class="co"># example limit – adjust as needed</span></span>
<span id="cb40-18"><a href="#cb40-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-19"><a href="#cb40-19" aria-hidden="true" tabindex="-1"></a>pp_check3 <span class="ot">&lt;-</span> <span class="fu">pp_check</span>(</span>
<span id="cb40-20"><a href="#cb40-20" aria-hidden="true" tabindex="-1"></a>  model3,</span>
<span id="cb40-21"><a href="#cb40-21" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">"dens_overlay_grouped"</span>,</span>
<span id="cb40-22"><a href="#cb40-22" aria-hidden="true" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb40-23"><a href="#cb40-23" aria-hidden="true" tabindex="-1"></a>  <span class="at">ndraws =</span> <span class="dv">100</span></span>
<span id="cb40-24"><a href="#cb40-24" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb40-25"><a href="#cb40-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb40-26"><a href="#cb40-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb40-27"><a href="#cb40-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">6</span>))</span>
<span id="cb40-28"><a href="#cb40-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-29"><a href="#cb40-29" aria-hidden="true" tabindex="-1"></a><span class="co"># Use `plot_spacer()` for the empty 4th spot</span></span>
<span id="cb40-30"><a href="#cb40-30" aria-hidden="true" tabindex="-1"></a>combined_plot <span class="ot">&lt;-</span> (pp_check1 <span class="sc">/</span> pp_check2 <span class="sc">/</span> pp_check3) <span class="sc">+</span></span>
<span id="cb40-31"><a href="#cb40-31" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(</span>
<span id="cb40-32"><a href="#cb40-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag_levels =</span> <span class="st">'A'</span>,</span>
<span id="cb40-33"><a href="#cb40-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag_prefix =</span> <span class="st">"("</span>,</span>
<span id="cb40-34"><a href="#cb40-34" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag_suffix =</span> <span class="st">")"</span></span>
<span id="cb40-35"><a href="#cb40-35" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb40-36"><a href="#cb40-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb40-37"><a href="#cb40-37" aria-hidden="true" tabindex="-1"></a>combined_plot</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-ppcheckall" class="quarto-float quarto-figure quarto-figure-center" data-fignum="12" prefix="">
<figure class="quarto-float quarto-float-fig figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-fig" id="fig-ppcheckall-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<div class="FigureTitle" data-custom-style="FigureTitle">
<p>Figure&nbsp;12</p>
</div>
<div class="Caption" data-custom-style="Caption">
<p>The plots show 100 posterior predicted distributions with the label yrep (light blue), the distribution of accuracy as function of fluency in dark blue for regular regression (A) , beta regression (B), ZIB (C) and ZOIB (D).</p>
</div>
</figcaption>
<div aria-describedby="fig-ppcheckall-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<img src="ms_files/figure-html/fig-ppcheckall-1.png" class="img-fluid figure-img" width="2400">
</div>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="discussion" class="level1">
<h1>Discussion</h1>
<p>The use of beta regression in psychology, and the social sciences in general, is rare. With this tutorial, we hope to turn the tides. Beta regression models are an attractive alternative to models that impose unrealistic assumptions like normality, linearity, homoscedasticity, and unbounded data. Beyond these models, there are a diverse array of different models that can be used depending on your outcome of interest.</p>
<p>Throughout this tutorial our main aim was to help guide researchers in running analyses with proportional or percentage outcomes using beta regression and some of its alternatives. In the current example, we used real data from <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span> and discussed how to fit these models in R, interpret model parameters, extract predicted probabilities and marginal effects, and visualize the results using the {marginaleffects} package.</p>
<p>Comparing our analysis with that of <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span>, we demonstrated that using a traditional approache (e.g., <em>t</em>-test) to analyze mean accuracy data can lead to biased inferences. Although we successfully reproduced one of their key findings, our use of beta regression and its extensions revealed important nuances in the results. With a traditional beta regression model–which accounts for both the mean and the precision (dispersion)–we observed similar effects of instructor fluency on performance. However, the standard beta model does not accommodate boundary values (i.e., 0s and 1s).</p>
<p>When we applied a ZIB model, which explicitly accounts for structural 0s, we found no effect of fluency on the mean (<span class="math inline">\mu</span>) part of the model. Instead, the effect of fluency emerged in the structural zero (inflated zero; <span class="math inline">\alpha</span>) component. This pattern was consistent when using a zero-one-inflated beta (ZOIB) model. Furthermore, we fit an ordered beta regression model <span class="citation" data-cites="kubinec2022">(Kubinec, 2022)</span>, which appropriately models the full range of values, including 0s and 1s. Here, we did not observe a reliable effect of fluency on the mean once we accounted for dispersion.</p>
<p>These analyses emphasize the importance of fitting a model that aligns with the nature of the data. The simplest and recommended approach when dealing with data that contains 0s and/or 1s is to fit an ordered beta model, assuming the process is truly continuous. However, if you believe the process is distinct in nature, a ZIB or ZOIB model might be a better choice. Ultimately, this decision should be guided by theory.</p>
<p>For instance, if we believe fluency influences the boundaries (0 and 1), we might want to model this process separately using a ZIB or ZOIB. With the current dataset, fluency might affect specific aspects of performance (such as the likelihood of complete failure) rather than general performance levels. This effect could be due to participant disengagement during the disfluent lecture. If students fail to pay attention because of features of disfluency, they may miss relevant information, leading to a floor effect at the test. Follwoing from this, disfluency would be expected to influence the boundary (O) and not the continuous part of the model. If this is the case, we would want to model this appropriately. However, if we believe fluency effects general performance levels (the continuous part), a model that takes in to account the entire process accounting for the 0s and 1s might be appropriate.</p>
<p>In the discussion section of <span class="citation" data-cites="wilford2020">Wilford et al. (2020)</span>, they were unable to offer a tenable explanation for performance differences based on instructor fluency. A model that accounts for the excess 0s in the dataset provides one testable explanation: watching a disfluent lecture may lead to lapses in attention, resulting in poorer performance in that group. These lapses, in turn, contribute to the observed differences in the fluent condition. This modeling approach opens a promising avenue for future research–one that would have remained inaccessible otherwise.</p>
<p>Not everyone will be eager to implement the techniques discussed herein. In such cases, the key question becomes: What is the least problematic approach to handling proportional data? One reasonable option is to fit multiple models tailored to the specific characteristics of your data. For example, if your data contain 0s, you might fit two models: a traditional linear model excluding the 0s, and a logistic model to account for the zero versus non-zero distinction. If your data contain both 0s and 1s, you could fit separate models for the 0s and 1s in addition to the OLS model. There are many defensible strategies to choose from depending on the context. However, we do not recommend transforming the values of your data (e.g., 0s to .01 and 1s to .99) or ignoring the properties of your data simply to fit traditional statistical models.</p>
<p>In this tutorial, we demonstrated how to analyze these models from a Bayesian perspective. While we recognize that not everyone identifies as a Bayesian, implementing these models using a Bayesian framework is relatively straightforward–it requires only a single package, lowering the barrier to entry. For those who prefer frequentist analyses, several R packages are available. For example, the {betareg} package <span class="citation" data-cites="betareg">(Cribari-Neto &amp; Zeileis, 2010)</span> {glmmTMB} <span class="citation" data-cites="glmmTMB">(Brooks et al., 2017)</span> and {gamlss} <span class="citation" data-cites="gamlss">(2005)</span> are nice options. To this end, I have included supplemental materials demonstrating how to use frequentist packages to analyze the data presented herein.</p>
<section id="conclusion" class="level2">
<h2 data-anchor-id="conclusion">Conclusion</h2>
<p>Overall, this tutorial emphasizes the importance of modeling the data you have. Although the example provided is relatively simple (a one-factor model with two levels), we hope it demonstrates that even with a basic dataset, there is much nuance in interpretation and inference. Properly modeling your data can lead to deeper insights, far beyond what traditional measures might offer. With the tools introduced in this tutorial, researchers now have the means to analyze their data effectively, uncover patterns, make accurate predictions, and support their findings with robust statistical evidence. By applying these modeling techniques, researchers can improve the validity and reliability of their studies, ultimately leading to more informed decisions and advancements in their respective fields.</p>
</section>
</section>
<section id="references" class="level1">
<h1>References</h1>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" data-line-spacing="2" role="list">
<div id="ref-marginaleffects" class="csl-entry" role="listitem">
Arel-Bundock, V. (2024). <em>Marginaleffects: Predictions, comparisons, slopes, marginal means, and hypothesis tests</em>. <a href="https://CRAN.R-project.org/package=marginaleffects">https://CRAN.R-project.org/package=marginaleffects</a>
</div>
<div id="ref-margeffects2024" class="csl-entry" role="listitem">
Arel-Bundock, V., Greifer, N., &amp; Heiss, A. (2024). How to interpret statistical models using <span class="nocase">marginaleffects</span> for <span>R</span> and <span>Python</span>. <em>Journal of Statistical Software</em>, <em>111</em>(9), 1–32. <a href="https://doi.org/10.18637/jss.v111.i09">https://doi.org/10.18637/jss.v111.i09</a>
</div>
<div id="ref-bartlett1936" class="csl-entry" role="listitem">
Bartlett, M. S. (1936). The square root transformation in analysis of variance. <em>Journal of the Royal Statistical Society Series B: Statistical Methodology</em>, <em>3</em>(1), 68–78. <a href="https://doi.org/10.2307/2983678">https://doi.org/10.2307/2983678</a>
</div>
<div id="ref-bendixen2023" class="csl-entry" role="listitem">
Bendixen, T., &amp; Purzycki, B. G. (2023). Cognitive and cultural models in psychological science: A tutorial on modeling free-list data as a dependent variable in bayesian regression. <em>Psychological Methods</em>. <a href="https://doi.org/10.1037/met0000553">https://doi.org/10.1037/met0000553</a>
</div>
<div id="ref-glmmTMB" class="csl-entry" role="listitem">
Brooks, M. E., Kristensen, K., van, K. J., Magnusson, A., Berg, C. W., Nielsen, A., Skaug, H. J., Maechler, M., &amp; Bolker, B. M. (2017). <em><span></span>glmmTMB<span></span> balances speed and flexibility among packages for zero-inflated generalized linear mixed modeling</em>. <em>9</em>. <a href="https://doi.org/10.32614/RJ-2017-066">https://doi.org/10.32614/RJ-2017-066</a>
</div>
<div id="ref-brms" class="csl-entry" role="listitem">
Bürkner, P.-C. (2017). <em><span></span>Brms<span></span>: An <span></span>r<span></span> package for <span></span>bayesian<span></span> multilevel models using <span></span>stan<span></span></em>. <em>80</em>. <a href="https://doi.org/10.18637/jss.v080.i01">https://doi.org/10.18637/jss.v080.i01</a>
</div>
<div id="ref-posterior2025a" class="csl-entry" role="listitem">
Bürkner, P.-C., Gabry, J., Kay, M., &amp; Vehtari, A. (2025). <em><span class="nocase">posterior</span>: Tools for working with posterior distributions</em>. <a href="https://mc-stan.org/posterior/">https://mc-stan.org/posterior/</a>
</div>
<div id="ref-bürkner2019" class="csl-entry" role="listitem">
Bürkner, P.-C., &amp; Vuorre, M. (2019). Ordinal regression models in psychology: A tutorial. <em>Advances in Methods and Practices in Psychological Science</em>, <em>2</em>(1), 77–101. <a href="https://doi.org/10.1177/2515245918823199">https://doi.org/10.1177/2515245918823199</a>
</div>
<div id="ref-carpenter2013" class="csl-entry" role="listitem">
Carpenter, S. K., Wilford, M. M., Kornell, N., &amp; Mullaney, K. M. (2013). Appearances can be deceiving: Instructor fluency increases perceptions of learning without increasing actual learning. <em>Psychonomic Bulletin &amp; Review</em>, <em>20</em>(6), 1350–1356. <a href="https://doi.org/10.3758/s13423-013-0442-z">https://doi.org/10.3758/s13423-013-0442-z</a>
</div>
<div id="ref-cohen1977" class="csl-entry" role="listitem">
Cohen, J. (1977). <em>Statistical power analysis for the behavioral sciences, rev. ed.</em> Lawrence Erlbaum Associates, Inc.
</div>
<div id="ref-coretta2025bayesian" class="csl-entry" role="listitem">
Coretta, S., &amp; Bürkner, P.-C. (2025). <em>Bayesian beta regressions with brms in r: A tutorial for phoneticians</em>. <a href="https://doi.org/10.31219/osf.io/f9rqg_v1" class="uri">https://doi.org/10.31219/osf.io/f9rqg_v1</a>.
</div>
<div id="ref-Costello2024" class="csl-entry" role="listitem">
Costello, T. H. et al. (2024). Durably reducing conspiracy beliefs through dialogues with AI. <em>Science</em>, <em>385</em>, eadq1814. <a href="https://doi.org/10.1126/science.adq1814">https://doi.org/10.1126/science.adq1814</a>
</div>
<div id="ref-betareg" class="csl-entry" role="listitem">
Cribari-Neto, F., &amp; Zeileis, A. (2010). <em>Beta regression in <span></span>r<span></span></em>. <em>34</em>. <a href="https://doi.org/10.18637/jss.v034.i02">https://doi.org/10.18637/jss.v034.i02</a>
</div>
<div id="ref-nix" class="csl-entry" role="listitem">
Dolstra, E., &amp; contributors, T. N. (2006). <em>Nix</em> [Computer software]. <a href="https://nixos.org/">https://nixos.org/</a>
</div>
<div id="ref-ferrari2004" class="csl-entry" role="listitem">
Ferrari, S., &amp; Cribari-Neto, F. (2004). Beta regression for modelling rates and proportions. <em>Journal of Applied Statistics</em>, <em>31</em>(7), 799–815. <a href="https://doi.org/10.1080/0266476042000214501">https://doi.org/10.1080/0266476042000214501</a>
</div>
<div id="ref-Fullerton2023" class="csl-entry" role="listitem">
Fullerton, A. S., &amp; Anderson, K. F. (2021). Ordered regression models: A tutorial. <em>Prevention Science</em>, <em>24</em>(3), 431–443. <a href="https://doi.org/10.1007/s11121-021-01302-y">https://doi.org/10.1007/s11121-021-01302-y</a>
</div>
<div id="ref-cmdstanr" class="csl-entry" role="listitem">
Gabry, J., Češnovar, R., Johnson, A., &amp; Bronder, S. (2024). <em>Cmdstanr: R interface to ’CmdStan’</em>. <a href="https://mc-stan.org/cmdstanr/">https://mc-stan.org/cmdstanr/</a>
</div>
<div id="ref-gelman2013bayesian" class="csl-entry" role="listitem">
Gelman, A., Carlin, J. B., Stern, H. S., Dunson, D. B., Vehtari, A., &amp; Rubin, D. B. (2013). <em>Bayesian data analysis</em> (Third). CRC. <a href="https://stat.columbia.edu/~gelman/book/">https://stat.columbia.edu/~gelman/book/</a>
</div>
<div id="ref-heiss2021" class="csl-entry" role="listitem">
Heiss, A. (2021). <em>A guide to modeling proportions with bayesian beta and zero-inflated beta regression models</em>. <a href="http://dx.doi.org/10.59350/7p1a4-0tw75">http://dx.doi.org/10.59350/7p1a4-0tw75</a>
</div>
<div id="ref-james2025" class="csl-entry" role="listitem">
James, A. N., Ryskin, R., Hartshorne, J. K., Backs, H., Bala, N., Barcenas-Meade, L., Bhattarai, S., Charles, T., Copoulos, G., Coss, C., Eisert, A., Furuhashi, E., Ginell, K., Guttman-McCabe, A., Harrison, E. (Chaz)., Hoban, L., Hwang, W. A., Iannetta, C., Koenig, K. M., … Leeuw, J. R. de. (2025). What paradigms can webcam eye-tracking be used for? Attempted replications of five cognitive science experiments. <em>Collabra: Psychology</em>, <em>11</em>(1). <a href="https://doi.org/10.1525/collabra.140755">https://doi.org/10.1525/collabra.140755</a>
</div>
<div id="ref-bayes_rules" class="csl-entry" role="listitem">
Johnson, A., Ott, M., &amp; Dogucu, M. (2022). <em>Bayes rules!: An introduction to applied bayesian modeling</em>. Routledge &amp; CRC Press.
</div>
<div id="ref-kneib2023" class="csl-entry" role="listitem">
Kneib, T., Silbersdorff, A., &amp; Säfken, B. (2023). Rage against the mean <span></span> a review of distributional regression approaches. <em>Econometrics and Statistics</em>, <em>26</em>, 99–123. <a href="https://doi.org/10.1016/j.ecosta.2021.07.006">https://doi.org/10.1016/j.ecosta.2021.07.006</a>
</div>
<div id="ref-kong2016" class="csl-entry" role="listitem">
Kong, E. J., &amp; Edwards, J. (2016). Individual differences in categorical perception of speech: Cue weighting and executive function. <em>Journal of Phonetics</em>, <em>59</em>, 40–57. <a href="https://doi.org/10.1016/j.wocn.2016.08.006">https://doi.org/10.1016/j.wocn.2016.08.006</a>
</div>
<div id="ref-kornell2008" class="csl-entry" role="listitem">
Kornell, N., &amp; Bjork, R. A. (2008). Learning concepts and categories. <em>Psychological Science</em>, <em>19</em>(6), 585–592. <a href="https://doi.org/10.1111/j.1467-9280.2008.02127.x">https://doi.org/10.1111/j.1467-9280.2008.02127.x</a>
</div>
<div id="ref-kosmidis2025" class="csl-entry" role="listitem">
Kosmidis, I., &amp; Zeileis, A. (2025). Extended-support beta regression for [0, 1] responses. <em>Journal of the Royal Statistical Society Series C: Applied Statistics</em>. <a href="https://doi.org/10.1093/jrsssc/qlaf039">https://doi.org/10.1093/jrsssc/qlaf039</a>
</div>
<div id="ref-kruschke2013" class="csl-entry" role="listitem">
Kruschke, J. K. (2013). Bayesian estimation supersedes the t test. <em>Journal of Experimental Psychology: General</em>, <em>142</em>(2), 573–603. <a href="https://doi.org/10.1037/a0029146">https://doi.org/10.1037/a0029146</a>
</div>
<div id="ref-kruschke2015doing" class="csl-entry" role="listitem">
Kruschke, J. K. (2015). <em>Doing bayesian data analysis: A tutorial with r, JAGS, and stan</em> (2nd ed.). Academic Press.
</div>
<div id="ref-kruschke2018" class="csl-entry" role="listitem">
Kruschke, J. K. (2018). Rejecting or accepting parameter values in bayesian estimation. <em>Advances in Methods and Practices in Psychological Science</em>, <em>1</em>(2), 270–280. <a href="https://doi.org/10.1177/2515245918771304">https://doi.org/10.1177/2515245918771304</a>
</div>
<div id="ref-kubinec2022" class="csl-entry" role="listitem">
Kubinec, R. (2022). Ordered beta regression: A parsimonious, well-fitting model for continuous data with lower and upper bounds. <em>Political Analysis</em>, <em>31</em>(4), 519–536. <a href="https://doi.org/10.1017/pan.2022.20">https://doi.org/10.1017/pan.2022.20</a>
</div>
<div id="ref-ordbetareg" class="csl-entry" role="listitem">
Kubinec, R. (2023). <em>Ordbetareg: Ordered beta regression models with ’brms’</em>. <a href="https://CRAN.R-project.org/package=ordbetareg">https://CRAN.R-project.org/package=ordbetareg</a>
</div>
<div id="ref-emmeans" class="csl-entry" role="listitem">
Lenth, R. V. (2025). <em>Emmeans: Estimated marginal means, aka least-squares means</em>. <a href="https://doi.org/10.32614/CRAN.package.emmeans">https://doi.org/10.32614/CRAN.package.emmeans</a>
</div>
<div id="ref-liu2015" class="csl-entry" role="listitem">
Liu, F., &amp; Kong, Y. (2015). Zoib: An r package for bayesian inference for beta regression and zero/one inflated beta regression. <em>The R Journal</em>, <em>7</em>(2), 34. <a href="https://doi.org/10.32614/rj-2015-019">https://doi.org/10.32614/rj-2015-019</a>
</div>
<div id="ref-easystats" class="csl-entry" role="listitem">
Lüdecke, D., Ben-Shachar, M. S., Patil, I., Wiernik, B. M., Bacher, E., Thériault, R., &amp; Makowski, D. (2022). <em>Easystats: Framework for easy statistical modeling, visualization, and reporting</em>. <a href="https://easystats.github.io/easystats/">https://easystats.github.io/easystats/</a>
</div>
<div id="ref-makowski2019a" class="csl-entry" role="listitem">
Makowski, D., Ben-Shachar, M. S., Chen, S. H. A., &amp; Lüdecke, D. (2019). Indices of effect existence and significance in the bayesian framework. <em>Frontiers in Psychology</em>, <em>10</em>. <a href="https://doi.org/10.3389/fpsyg.2019.02767">https://doi.org/10.3389/fpsyg.2019.02767</a>
</div>
<div id="ref-makowski2019" class="csl-entry" role="listitem">
Makowski, D., Ben-Shachar, M., &amp; Lüdecke, D. (2019). bayestestR: Describing effects and their uncertainty, existence and significance within the bayesian framework. <em>Journal of Open Source Software</em>, <em>4</em>(40), 1541. <a href="https://doi.org/10.21105/joss.01541">https://doi.org/10.21105/joss.01541</a>
</div>
<div id="ref-Makowski2025CHOCO" class="csl-entry" role="listitem">
Makowski, D., Neves, A., &amp; Field, A. P. (2025). <em>Introducing the choice-confidence (CHOCO) model for bimodal data from subjective ratings: Application to the effect of attractiveness on reality beliefs about AI-generated faces</em>. <a href="https://doi.org/10.31234/osf.io/z68v3_v1">https://doi.org/10.31234/osf.io/z68v3_v1</a>
</div>
<div id="ref-marsmanThreeInsightsBayesian2016" class="csl-entry" role="listitem">
Marsman, M., &amp; Wagenmakers, E.-J. (2016). Three <span>Insights</span> from a <span>Bayesian Interpretation</span> of the <span>One-Sided P Value</span>. <em>Educational and Psychological Measurement</em>. <a href="https://doi.org/10.1177/0013164416669201">https://doi.org/10.1177/0013164416669201</a>
</div>
<div id="ref-martinVocalComplexitySocially2024" class="csl-entry" role="listitem">
Martin, K., Cornero, F. M., Clayton, N. S., Adam, O., Obin, N., &amp; Dufour, V. (2024). Vocal complexity in a socially complex corvid: Gradation, diversity and lack of common call repertoire in male rooks. <em>Royal Society Open Science</em>, <em>11</em>(1), 231713. <a href="https://doi.org/10.1098/rsos.231713">https://doi.org/10.1098/rsos.231713</a>
</div>
<div id="ref-mccabe2021" class="csl-entry" role="listitem">
McCabe, C. J., Halvorson, M. A., King, K. M., Cao, X., &amp; Kim, D. S. (2021). Interpreting interaction effects in generalized linear models of nonlinear probabilities and counts. <em>Multivariate Behavioral Research</em>, <em>57</em>(2-3), 243–263. <a href="https://doi.org/10.1080/00273171.2020.1868966">https://doi.org/10.1080/00273171.2020.1868966</a>
</div>
<div id="ref-mcelreath2020statistical" class="csl-entry" role="listitem">
McElreath, R. (2020). <em>Statistical rethinking: A bayesian course with examples in r and STAN</em> (2nd ed.). Chapman; Hall/CRC. <a href="https://doi.org/10.1201/9780429029608">https://doi.org/10.1201/9780429029608</a>
</div>
<div id="ref-nelder1972" class="csl-entry" role="listitem">
Nelder, J. A., &amp; Wedderburn, R. W. M. (1972). Generalized linear models. <em>Journal of the Royal Statistical Society. Series A (General)</em>, <em>135</em>(3), 370–384. <a href="https://doi.org/10.2307/2344614">https://doi.org/10.2307/2344614</a>
</div>
<div id="ref-nouvianGlyphosateImpairsAversive2023" class="csl-entry" role="listitem">
Nouvian, M., Foster, J. J., &amp; Weidenmüller, A. (2023). Glyphosate impairs aversive learning in bumblebees. <em>Science of The Total Environment</em>, <em>898</em>, 165527. <a href="https://www.sciencedirect.com/science/article/pii/S0048969723041505">https://www.sciencedirect.com/science/article/pii/S0048969723041505</a>
</div>
<div id="ref-paolino2001" class="csl-entry" role="listitem">
Paolino, P. (2001). Maximum likelihood estimation of models with beta-distributed dependent variables. <em>Political Analysis</em>, <em>9</em>(4), 325–346. <a href="https://doi.org/10.1093/oxfordjournals.pan.a004873">https://doi.org/10.1093/oxfordjournals.pan.a004873</a>
</div>
<div id="ref-Pfadt2025" class="csl-entry" role="listitem">
Pfadt, J. M., Bartoš, F., Godmann, H. R., Waaijers, M., Groot, L., Heo, I., &amp; Wagenmakers, E. …. (2025). A methodological metamorphosis: The rapid rise of bayesian inference and open science practices in psychology. <em>Preprint</em>. <a href="https://doi.org/10.31234/osf.io/ck3js_v1">https://doi.org/10.31234/osf.io/ck3js_v1</a>
</div>
<div id="ref-R" class="csl-entry" role="listitem">
R Core Team. (2024). <em>R: A language and environment for statistical computing</em>. R Foundation for Statistical Computing. <a href="https://www.R-project.org/">https://www.R-project.org/</a>
</div>
<div id="ref-gamlss" class="csl-entry" role="listitem">
R. A. Rigby, &amp; D. M. Stasinopoulos. (2005). <em>Generalized additive models for location, scale and shape,(with discussion)</em>. <em>54</em>, 507–554.
</div>
<div id="ref-rix" class="csl-entry" role="listitem">
Rodrigues, B., &amp; Baumann, P. (2025). <em>Rix: Reproducible data science environments with ’nix’</em>. <a href="https://docs.ropensci.org/rix/">https://docs.ropensci.org/rix/</a>
</div>
<div id="ref-rohrer2025models" class="csl-entry" role="listitem">
Rohrer, J. M., &amp; Arel-Bundock, V. (2025). <em>Models as prediction machines: How to convert confusing coefficients into clear quantities</em>. <a href="https://doi.org/10.31234/osf.io/g4s2a_v1">https://doi.org/10.31234/osf.io/g4s2a_v1</a>
</div>
<div id="ref-shresthaBigFiveTraits2024" class="csl-entry" role="listitem">
Shrestha, S., Sigdel, K., Pokharel, M., &amp; Columbus, S. (2024). Big five traits predict between- and within-person variation in loneliness. <em>European Journal of Personality</em>, 08902070241239834. <a href="https://doi.org/10.1177/08902070241239834">https://doi.org/10.1177/08902070241239834</a>
</div>
<div id="ref-sladekova2024unicorns" class="csl-entry" role="listitem">
Sladekova, M., &amp; Field, A. P. (2024). <em>In search of unicorns: Assessing statistical assumptions in real psychology datasets</em>. <a href="https://doi.org/10.31234/osf.io/4rznt">https://doi.org/10.31234/osf.io/4rznt</a>
</div>
<div id="ref-smithEcologicalMomentaryAssessment2024" class="csl-entry" role="listitem">
Smith, K. E., Panlilio, L. V., Feldman, J. D., Grundmann, O., Dunn, K. E., McCurdy, C. R., Garcia-Romeu, A., &amp; Epstein, D. H. (2024). Ecological momentary assessment of self-reported kratom use, effects, and motivations among US adults. <em>JAMA Network Open</em>, <em>7</em>(1), e2353401. <a href="https://doi.org/10.1001/jamanetworkopen.2023.53401">https://doi.org/10.1001/jamanetworkopen.2023.53401</a>
</div>
<div id="ref-smithson2006" class="csl-entry" role="listitem">
Smithson, M., &amp; Verkuilen, J. (2006). A better lemon squeezer? Maximum-likelihood regression with beta-distributed dependent variables. <em>Psychological Methods</em>, <em>11</em>(1), 54–71. <a href="https://doi.org/10.1037/1082-989x.11.1.54">https://doi.org/10.1037/1082-989x.11.1.54</a>
</div>
<div id="ref-stan2023" class="csl-entry" role="listitem">
Team, S. D. (2023). <em>Stan: A probabilistic programming language</em>. <a href="https://mc-stan.org">https://mc-stan.org</a>
</div>
<div id="ref-toftness2017" class="csl-entry" role="listitem">
Toftness, A. R., Carpenter, S. K., Geller, J., Lauber, S., Johnson, M., &amp; Armstrong, P. I. (2017). Instructor fluency leads to higher confidence in learning, but not better learning. <em>Metacognition and Learning</em>, <em>13</em>(1), 1–14. <a href="https://doi.org/10.1007/s11409-017-9175-0">https://doi.org/10.1007/s11409-017-9175-0</a>
</div>
<div id="ref-vuorre2019" class="csl-entry" role="listitem">
Vuorre, M. (2019, February 18). <em>How to analyze visual analog (slider) scale data?</em> <a href="https://vuorre.com/posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-beta-models">https://vuorre.com/posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-beta-models</a>
</div>
<div id="ref-wickhamDataScienceImport2023" class="csl-entry" role="listitem">
Wickham, H., Çetinkaya-Rundel, M., &amp; Grolemund, G. (2023). <em>R for <span>Data Science</span>: <span>Import</span>, <span>Tidy</span>, <span>Transform</span>, <span>Visualize</span>, and <span>Model Data</span></em>. O’Reilly. <a href="https://r4ds.hadley.nz/">https://r4ds.hadley.nz/</a>
</div>
<div id="ref-wilford2020" class="csl-entry" role="listitem">
Wilford, M. M., Kurpad, N., Platt, M., &amp; Weinstein-Jones, Y. (2020). Lecturer fluency can impact students’ judgments of learning and actual learning performance. <em>Applied Cognitive Psychology</em>, <em>34</em>(6), 1444–1456. <a href="https://doi.org/10.1002/acp.3724">https://doi.org/10.1002/acp.3724</a>
</div>
<div id="ref-wilkesQuantifyingCoextinctionsEcosystem2024" class="csl-entry" role="listitem">
Wilkes, L. N., Barner, A. K., Keyes, A. A., Morton, D., Byrnes, J. E. K., &amp; Dee, L. E. (2024). Quantifying co‐extinctions and ecosystem service vulnerability in coastal ecosystems experiencing climate warming. <em>Global Change Biology</em>, <em>30</em>(7), e17422. <a href="https://doi.org/10.1111/gcb.17422">https://doi.org/10.1111/gcb.17422</a>
</div>
<div id="ref-witherby2022" class="csl-entry" role="listitem">
Witherby, A. E., &amp; Carpenter, S. K. (2022). The impact of lecture fluency and technology fluency on students<span>’</span> online learning and evaluations of instructors. <em>Journal of Applied Research in Memory and Cognition</em>, <em>11</em>(4), 500–509. <a href="https://doi.org/10.1037/mac0000003">https://doi.org/10.1037/mac0000003</a>
</div>
<div id="ref-yeEnforcingBoundariesChinas2023" class="csl-entry" role="listitem">
Ye, M., &amp; Heiss, A. (2023). Enforcing <span>Boundaries</span>: <span>China</span>’s <span>Overseas NGO Law</span> and <span>Operational Constraints</span> for <span>Global Civil Society</span>. <em>Working Paper</em>. <a href="https://stats.andrewheiss.com/compassionate-clam/manuscript/output/manuscript.html">https://stats.andrewheiss.com/compassionate-clam/manuscript/output/manuscript.html</a>
</div>
</div>
<!-- -->

</section>


<div id="quarto-appendix" class="default"><section id="footnotes" class="footnotes footnotes-end-of-document" role="doc-endnotes"><h2 class="anchored quarto-appendix-heading">Footnotes</h2>

<ol>
<li id="fn1"><p>Specifically, for any continuous bounded variable <span class="math inline">x</span>, we can rescale this variable to lie within 0 and 1 by using the formula <span class="math inline">x' = \frac{x - \min(x)}{\max(x) - \min(x)}</span> where <span class="math inline">0 \leq x' \leq 1</span>.<a href="#fnref1" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn2"><p>In this article, we try to limit code where possible; however, the online version has all the code needed to reproduce all analyses herein. Furthermore, to promote transparency and reproducibility, the tutorial was written in R version 4.5.1 (<span class="citation" data-cites="R">R Core Team (2024)</span>) using Quarto (v.1.5.54), an open-source publishing system that allows for dynamic and static documents. This allows figures, tables, and text to be programmatically included directly in the manuscript, ensuring that all results are seamlessly integrated into the document. In addition, we use the <code>rix</code> <span class="citation" data-cites="rix">(Rodrigues &amp; Baumann, 2025)</span> R package which harnesses the power of the <code>nix</code> <span class="citation" data-cites="nix">(Dolstra &amp; contributors, 2006)</span> ecosystem to to help with computational reproducibility. Not only does this give us a snapshot of the packages used to create the current manuscript, but it also takes a snapshot of system dependencies used at run-time. This way reproducers can easily re-use the exact same environment by installing the <code>nix</code> package manager and using the included default.nix file to set up the right environment. The README file in the GitHub repository contains detailed information on how to set this up to reproduce the contents of the current manuscript, including a video.<a href="#fnref2" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn3"><p>Technically, this maximum entropy condition is satisfied because the beta(1,1) distribution is uniform over its support. In addition, we assume that the scale has been re-scaled to the <span class="math inline">[0,1]</span> interval as we describe above.<a href="#fnref3" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn4"><p>A common concern is that Bayesian methods are slower than frequentist ones. While this is true in general, modern Bayesian computation engines are efficient, and in explanatory modeling our priority is to specify appropriate estimands rather than to optimize for computation speed. Moreover, we use the {brms} package <span class="citation" data-cites="brms">(Bürkner, 2017)</span>, which provides a high-level interface to Stan <span class="citation" data-cites="stan2023">(Team, 2023)</span> and uses familiar R formula syntax (similar to <code>lm()</code>), making advanced Bayesian modeling accessible to researchers with standard regression backgrounds. The package also supports parallelization, which substantially reduces computation time for large datasets.<a href="#fnref4" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn5"><p>The Hamiltonian Monte Carlo sampler employed by Stan, which we also use in this paper, can converge with significantly fewer iterations, though rapid convergence depends on model complexity, which is why we use a more conservative standard in this paper.<a href="#fnref5" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn6"><p>In order to use the cmdstanr backend you will need to first install the package (see <code>https://mc-stan.org/cmdstanr/</code>) and also run <code>cmdstanr::install_cmdstan()</code> if you have not done so already.<a href="#fnref6" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn7"><p>The model fit statistic LOO-CV can be compared for any set of fitted brms models with the function <code>loo()</code>.<a href="#fnref7" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn8"><p>In cases where your data include exact 1s but no 0s, you can fit a one-inflated beta regression model in {brms} by setting the <code>coi</code> parameter to <code>1</code>. This tells the model that all point masses occur at 1, rather than being split between 0 and 1. In other words, <code>coi = 1</code> assumes that any inflation in the data is due entirely to values at 1. In our data, we have exactly one value equal to 1[^6]. While probably not significant to alter our findings, we can model 1s with a special type of model called the zero-one-inflated beta (ZOIB) model <span class="citation" data-cites="liu2015">(Liu &amp; Kong, 2015)</span> if we believe that both 0s and 1s are distinct outcomes.<a href="#fnref8" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn9"><p>For a more complete description of this issue, we refer the reader to <span class="citation" data-cites="kubinec2022">Kubinec (2022)</span>.<a href="#fnref9" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
<li id="fn10"><p>Please note that there are other models available that can model this continuous process like the beta-gate model <span class="citation" data-cites="Makowski2025CHOCO">(Makowski et al., 2025)</span> and the censored extended beta regression model <span class="citation" data-cites="kosmidis2025">(Kosmidis &amp; Zeileis, 2025)</span>.<a href="#fnref10" class="footnote-back" role="doc-backlink">↩︎</a></p></li>
</ol>
</section></div></main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
    const viewSource = window.document.getElementById('quarto-view-source') ||
                       window.document.getElementById('quarto-code-tools-source');
    if (viewSource) {
      const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
      viewSource.addEventListener("click", function(e) {
        if (sourceUrl) {
          // rstudio viewer pane
          if (/\bcapabilities=\b/.test(window.location)) {
            window.open(sourceUrl);
          } else {
            window.location.href = sourceUrl;
          }
        } else {
          const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
          modal.show();
        }
        return false;
      });
    }
    function toggleCodeHandler(show) {
      return function(e) {
        const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
        for (let i=0; i<detailsSrc.length; i++) {
          const details = detailsSrc[i].parentElement;
          if (show) {
            details.open = true;
          } else {
            details.removeAttribute("open");
          }
        }
        const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
        const fromCls = show ? "hidden" : "unhidden";
        const toCls = show ? "unhidden" : "hidden";
        for (let i=0; i<cellCodeDivs.length; i++) {
          const codeDiv = cellCodeDivs[i];
          if (codeDiv.classList.contains(fromCls)) {
            codeDiv.classList.remove(fromCls);
            codeDiv.classList.add(toCls);
          } 
        }
        return false;
      }
    }
    const hideAllCode = window.document.getElementById("quarto-hide-all-code");
    if (hideAllCode) {
      hideAllCode.addEventListener("click", toggleCodeHandler(false));
    }
    const showAllCode = window.document.getElementById("quarto-show-all-code");
    if (showAllCode) {
      showAllCode.addEventListener("click", toggleCodeHandler(true));
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb41" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb41-1"><a href="#cb41-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb41-2"><a href="#cb41-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "A Beta Way: A Tutorial For Using Beta Regression in Psychological Research"</span></span>
<span id="cb41-3"><a href="#cb41-3" aria-hidden="true" tabindex="-1"></a><span class="an">shorttitle:</span><span class="co"> "Beta Regression Tutorial"</span></span>
<span id="cb41-4"><a href="#cb41-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span></span>
<span id="cb41-5"><a href="#cb41-5" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Jason Geller</span></span>
<span id="cb41-6"><a href="#cb41-6" aria-hidden="true" tabindex="-1"></a><span class="co">    corresponding: true</span></span>
<span id="cb41-7"><a href="#cb41-7" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0002-7459-4505</span></span>
<span id="cb41-8"><a href="#cb41-8" aria-hidden="true" tabindex="-1"></a><span class="co">    email: drjasongeller@gmail.com</span></span>
<span id="cb41-9"><a href="#cb41-9" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliations:</span></span>
<span id="cb41-10"><a href="#cb41-10" aria-hidden="true" tabindex="-1"></a><span class="co">      - name: Boston College</span></span>
<span id="cb41-11"><a href="#cb41-11" aria-hidden="true" tabindex="-1"></a><span class="co">        department: Department of Psychology and Neuroscience</span></span>
<span id="cb41-12"><a href="#cb41-12" aria-hidden="true" tabindex="-1"></a><span class="co">        address: McGuinn 300z</span></span>
<span id="cb41-13"><a href="#cb41-13" aria-hidden="true" tabindex="-1"></a><span class="co">        city: Chestnut Hill</span></span>
<span id="cb41-14"><a href="#cb41-14" aria-hidden="true" tabindex="-1"></a><span class="co">        region: MA</span></span>
<span id="cb41-15"><a href="#cb41-15" aria-hidden="true" tabindex="-1"></a><span class="co">        country: USA</span></span>
<span id="cb41-16"><a href="#cb41-16" aria-hidden="true" tabindex="-1"></a><span class="co">        postal-code: 02467</span></span>
<span id="cb41-17"><a href="#cb41-17" aria-hidden="true" tabindex="-1"></a><span class="co">    roles:</span></span>
<span id="cb41-18"><a href="#cb41-18" aria-hidden="true" tabindex="-1"></a><span class="co">      - Conceptualization</span></span>
<span id="cb41-19"><a href="#cb41-19" aria-hidden="true" tabindex="-1"></a><span class="co">      - Data curation</span></span>
<span id="cb41-20"><a href="#cb41-20" aria-hidden="true" tabindex="-1"></a><span class="co">      - Formal analysis</span></span>
<span id="cb41-21"><a href="#cb41-21" aria-hidden="true" tabindex="-1"></a><span class="co">      - Project administration</span></span>
<span id="cb41-22"><a href="#cb41-22" aria-hidden="true" tabindex="-1"></a><span class="co">      - Resources</span></span>
<span id="cb41-23"><a href="#cb41-23" aria-hidden="true" tabindex="-1"></a><span class="co">      - Visualization</span></span>
<span id="cb41-24"><a href="#cb41-24" aria-hidden="true" tabindex="-1"></a><span class="co">      - Writing - original draft</span></span>
<span id="cb41-25"><a href="#cb41-25" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Robert Kubinec</span></span>
<span id="cb41-26"><a href="#cb41-26" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0001-6655-4119</span></span>
<span id="cb41-27"><a href="#cb41-27" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliations:</span></span>
<span id="cb41-28"><a href="#cb41-28" aria-hidden="true" tabindex="-1"></a><span class="co">      - name: University of South Carolina</span></span>
<span id="cb41-29"><a href="#cb41-29" aria-hidden="true" tabindex="-1"></a><span class="co">    roles:</span></span>
<span id="cb41-30"><a href="#cb41-30" aria-hidden="true" tabindex="-1"></a><span class="co">     - Formal analysis</span></span>
<span id="cb41-31"><a href="#cb41-31" aria-hidden="true" tabindex="-1"></a><span class="co">     - Validation</span></span>
<span id="cb41-32"><a href="#cb41-32" aria-hidden="true" tabindex="-1"></a><span class="co">     - Writing - review &amp; editing</span></span>
<span id="cb41-33"><a href="#cb41-33" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Chelsea M. Parlett Pelleriti</span></span>
<span id="cb41-34"><a href="#cb41-34" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0001-9301-1398</span></span>
<span id="cb41-35"><a href="#cb41-35" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliations:</span></span>
<span id="cb41-36"><a href="#cb41-36" aria-hidden="true" tabindex="-1"></a><span class="co">      - name: Canva</span></span>
<span id="cb41-37"><a href="#cb41-37" aria-hidden="true" tabindex="-1"></a><span class="co">    roles:</span></span>
<span id="cb41-38"><a href="#cb41-38" aria-hidden="true" tabindex="-1"></a><span class="co">     - Formal analysis</span></span>
<span id="cb41-39"><a href="#cb41-39" aria-hidden="true" tabindex="-1"></a><span class="co">     - Writing - review &amp; editing</span></span>
<span id="cb41-40"><a href="#cb41-40" aria-hidden="true" tabindex="-1"></a><span class="co">  - name: Matti Vuorre</span></span>
<span id="cb41-41"><a href="#cb41-41" aria-hidden="true" tabindex="-1"></a><span class="co">    orcid: 0000-0001-5052-066X</span></span>
<span id="cb41-42"><a href="#cb41-42" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliations:</span></span>
<span id="cb41-43"><a href="#cb41-43" aria-hidden="true" tabindex="-1"></a><span class="co">      - name: Tilburg University</span></span>
<span id="cb41-44"><a href="#cb41-44" aria-hidden="true" tabindex="-1"></a><span class="co">    roles:</span></span>
<span id="cb41-45"><a href="#cb41-45" aria-hidden="true" tabindex="-1"></a><span class="co">     - Formal analysis</span></span>
<span id="cb41-46"><a href="#cb41-46" aria-hidden="true" tabindex="-1"></a><span class="co">     - Writing - review &amp; editing</span></span>
<span id="cb41-47"><a href="#cb41-47" aria-hidden="true" tabindex="-1"></a><span class="an">author-note:</span></span>
<span id="cb41-48"><a href="#cb41-48" aria-hidden="true" tabindex="-1"></a><span class="co">  status-changes:</span></span>
<span id="cb41-49"><a href="#cb41-49" aria-hidden="true" tabindex="-1"></a><span class="co">    affiliation-change: ~</span></span>
<span id="cb41-50"><a href="#cb41-50" aria-hidden="true" tabindex="-1"></a><span class="co">    deceased: ~</span></span>
<span id="cb41-51"><a href="#cb41-51" aria-hidden="true" tabindex="-1"></a><span class="co">  disclosures:</span></span>
<span id="cb41-52"><a href="#cb41-52" aria-hidden="true" tabindex="-1"></a><span class="co">    # Example: This study was registered at X (Identifier Y).</span></span>
<span id="cb41-53"><a href="#cb41-53" aria-hidden="true" tabindex="-1"></a><span class="co">    study-registration: "The analyses herein were not preregistered."</span></span>
<span id="cb41-54"><a href="#cb41-54" aria-hidden="true" tabindex="-1"></a><span class="co">    # Acknowledge and cite data/materials to be shared.</span></span>
<span id="cb41-55"><a href="#cb41-55" aria-hidden="true" tabindex="-1"></a><span class="co">    data-sharing: Data, code, and materials for this manuscript can be found at https://doi.org/10.5281/zenodo.1583059.</span></span>
<span id="cb41-56"><a href="#cb41-56" aria-hidden="true" tabindex="-1"></a><span class="co">    related-report: ~</span></span>
<span id="cb41-57"><a href="#cb41-57" aria-hidden="true" tabindex="-1"></a><span class="co">    conflict-of-interest: The authors have no conflicts of interest to disclose.</span></span>
<span id="cb41-58"><a href="#cb41-58" aria-hidden="true" tabindex="-1"></a><span class="co">    financial-support: ~</span></span>
<span id="cb41-59"><a href="#cb41-59" aria-hidden="true" tabindex="-1"></a><span class="co">    authorship-agreements: ~</span></span>
<span id="cb41-60"><a href="#cb41-60" aria-hidden="true" tabindex="-1"></a><span class="an">abstract:</span><span class="co"> "Rates, percentages, and proportions are common outcomes in psychology and the social sciences. These outcomes are often analyzed using models that assume normality, but this practice overlooks important features of the data, such as their natural bounds at 0 and 1. As a result, estimates can become distorted. In contrast, treating such outcomes as beta-distributed respects these limits and can yield more accurate estimates. Despite these advantages, the use of beta models in applied research remains limited. Our goal is to provide researchers with practical guidance for adopting beta regression models, illustrated with an example drawn from the psychological literature. We begin by introducing the beta distribution and beta regression, emphasizing key components and assumptions. Next, using data from a learning and memory study, we demonstrate how to fit a beta regression model in R with the Bayesian package {brms} and how to interpret results on the response scale. We also discuss model extensions, including zero-inflated, zero- and one-inflated, and ordered beta models. Basic familiarity with regression modeling and R is assumed. To promote wider adoption of these methods, we provide detailed code and materials at https://doi.org/10.5281/zenodo.15830595."</span></span>
<span id="cb41-61"><a href="#cb41-61" aria-hidden="true" tabindex="-1"></a><span class="an">keywords:</span><span class="co"> [beta regression, beta distribution, R tutorial, psychology, learning and memory]</span></span>
<span id="cb41-62"><a href="#cb41-62" aria-hidden="true" tabindex="-1"></a><span class="an">floatsintext:</span><span class="co"> true</span></span>
<span id="cb41-63"><a href="#cb41-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Numbered lines (.pdf and .docx only)</span></span>
<span id="cb41-64"><a href="#cb41-64" aria-hidden="true" tabindex="-1"></a><span class="an">numbered-lines:</span><span class="co"> true</span></span>
<span id="cb41-65"><a href="#cb41-65" aria-hidden="true" tabindex="-1"></a><span class="co"># File with references</span></span>
<span id="cb41-66"><a href="#cb41-66" aria-hidden="true" tabindex="-1"></a><span class="an">suppress-title-page:</span><span class="co"> false</span></span>
<span id="cb41-67"><a href="#cb41-67" aria-hidden="true" tabindex="-1"></a><span class="co"># Link citations to references</span></span>
<span id="cb41-68"><a href="#cb41-68" aria-hidden="true" tabindex="-1"></a><span class="an">link-citations:</span><span class="co"> false</span></span>
<span id="cb41-69"><a href="#cb41-69" aria-hidden="true" tabindex="-1"></a><span class="co"># Masks references that appear in the masked-citations list</span></span>
<span id="cb41-70"><a href="#cb41-70" aria-hidden="true" tabindex="-1"></a><span class="an">mask:</span><span class="co"> false</span></span>
<span id="cb41-71"><a href="#cb41-71" aria-hidden="true" tabindex="-1"></a><span class="an">word-count:</span><span class="co"> true</span></span>
<span id="cb41-72"><a href="#cb41-72" aria-hidden="true" tabindex="-1"></a><span class="co"># Language options. See https://quarto.org/docs/authoring/language.html</span></span>
<span id="cb41-73"><a href="#cb41-73" aria-hidden="true" tabindex="-1"></a><span class="an">lang:</span><span class="co"> en</span></span>
<span id="cb41-74"><a href="#cb41-74" aria-hidden="true" tabindex="-1"></a><span class="an">language:</span></span>
<span id="cb41-75"><a href="#cb41-75" aria-hidden="true" tabindex="-1"></a><span class="co">  citation-last-author-separator: "and"</span></span>
<span id="cb41-76"><a href="#cb41-76" aria-hidden="true" tabindex="-1"></a><span class="co">  citation-masked-author: "Masked Citation"</span></span>
<span id="cb41-77"><a href="#cb41-77" aria-hidden="true" tabindex="-1"></a><span class="co">  citation-masked-date: "n.d."</span></span>
<span id="cb41-78"><a href="#cb41-78" aria-hidden="true" tabindex="-1"></a><span class="co">  citation-masked-title: "Masked Title"</span></span>
<span id="cb41-79"><a href="#cb41-79" aria-hidden="true" tabindex="-1"></a><span class="co">  email: "Email"</span></span>
<span id="cb41-80"><a href="#cb41-80" aria-hidden="true" tabindex="-1"></a><span class="co">  title-block-author-note: "Author Note"</span></span>
<span id="cb41-81"><a href="#cb41-81" aria-hidden="true" tabindex="-1"></a><span class="co">  title-block-correspondence-note: "Correspondence concerning this article should be addressed to"</span></span>
<span id="cb41-82"><a href="#cb41-82" aria-hidden="true" tabindex="-1"></a><span class="co">  title-block-role-introduction: "Author roles were classified using the Contributor Role Taxonomy (CRediT; https://credit.niso.org/) as follows:"</span></span>
<span id="cb41-83"><a href="#cb41-83" aria-hidden="true" tabindex="-1"></a><span class="an">bibliography:</span><span class="co"> bibliography.bib</span></span>
<span id="cb41-84"><a href="#cb41-84" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-85"><a href="#cb41-85" aria-hidden="true" tabindex="-1"></a><span class="an">execute:</span></span>
<span id="cb41-86"><a href="#cb41-86" aria-hidden="true" tabindex="-1"></a><span class="co">  echo: false</span></span>
<span id="cb41-87"><a href="#cb41-87" aria-hidden="true" tabindex="-1"></a><span class="co">  cache: false</span></span>
<span id="cb41-88"><a href="#cb41-88" aria-hidden="true" tabindex="-1"></a><span class="co">  warning: false</span></span>
<span id="cb41-89"><a href="#cb41-89" aria-hidden="true" tabindex="-1"></a><span class="co">  message: false</span></span>
<span id="cb41-90"><a href="#cb41-90" aria-hidden="true" tabindex="-1"></a><span class="co">  results: hide</span></span>
<span id="cb41-91"><a href="#cb41-91" aria-hidden="true" tabindex="-1"></a><span class="co">  freeze: auto</span></span>
<span id="cb41-92"><a href="#cb41-92" aria-hidden="true" tabindex="-1"></a><span class="co">  fig-align: center</span></span>
<span id="cb41-93"><a href="#cb41-93" aria-hidden="true" tabindex="-1"></a><span class="co">  tbl-align: center</span></span>
<span id="cb41-94"><a href="#cb41-94" aria-hidden="true" tabindex="-1"></a><span class="co">  code-line-numbers: true</span></span>
<span id="cb41-95"><a href="#cb41-95" aria-hidden="true" tabindex="-1"></a><span class="co">  keep-with-next: true</span></span>
<span id="cb41-96"><a href="#cb41-96" aria-hidden="true" tabindex="-1"></a><span class="co">  code-overflow: wrap</span></span>
<span id="cb41-97"><a href="#cb41-97" aria-hidden="true" tabindex="-1"></a><span class="co">  fig-width: 6</span></span>
<span id="cb41-98"><a href="#cb41-98" aria-hidden="true" tabindex="-1"></a><span class="co">  fig-asp: 0.8</span></span>
<span id="cb41-99"><a href="#cb41-99" aria-hidden="true" tabindex="-1"></a><span class="co">  fig-pos: "h"</span></span>
<span id="cb41-100"><a href="#cb41-100" aria-hidden="true" tabindex="-1"></a><span class="co">  tbl-pos: "h"</span></span>
<span id="cb41-101"><a href="#cb41-101" aria-hidden="true" tabindex="-1"></a><span class="co">  fig-dpi: 300</span></span>
<span id="cb41-102"><a href="#cb41-102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-103"><a href="#cb41-103" aria-hidden="true" tabindex="-1"></a><span class="an">knitr:</span></span>
<span id="cb41-104"><a href="#cb41-104" aria-hidden="true" tabindex="-1"></a><span class="co">  opts_chunk:</span></span>
<span id="cb41-105"><a href="#cb41-105" aria-hidden="true" tabindex="-1"></a><span class="co">    dev: "ragg_png"</span></span>
<span id="cb41-106"><a href="#cb41-106" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb41-107"><a href="#cb41-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-108"><a href="#cb41-108" aria-hidden="true" tabindex="-1"></a>::: callout-warning</span>
<span id="cb41-109"><a href="#cb41-109" aria-hidden="true" tabindex="-1"></a><span class="fu">## Preprint</span></span>
<span id="cb41-110"><a href="#cb41-110" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-111"><a href="#cb41-111" aria-hidden="true" tabindex="-1"></a>This manuscript is currently **under review** and has not been peer-reviewed. Content is **subject to change**. Please feel free to provide feedback!</span>
<span id="cb41-112"><a href="#cb41-112" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb41-113"><a href="#cb41-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-116"><a href="#cb41-116" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-117"><a href="#cb41-117" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: packages_1</span></span>
<span id="cb41-118"><a href="#cb41-118" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-119"><a href="#cb41-119" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb41-120"><a href="#cb41-120" aria-hidden="true" tabindex="-1"></a><span class="co">#| warning: false</span></span>
<span id="cb41-121"><a href="#cb41-121" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-122"><a href="#cb41-122" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(easystats)</span>
<span id="cb41-123"><a href="#cb41-123" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb41-124"><a href="#cb41-124" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(scales)</span>
<span id="cb41-125"><a href="#cb41-125" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tinytable)</span>
<span id="cb41-126"><a href="#cb41-126" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(collapse)</span>
<span id="cb41-127"><a href="#cb41-127" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(marginaleffects)</span>
<span id="cb41-128"><a href="#cb41-128" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(extraDistr)</span>
<span id="cb41-129"><a href="#cb41-129" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="cb41-130"><a href="#cb41-130" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(posterior)</span>
<span id="cb41-131"><a href="#cb41-131" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(bayesplot)</span>
<span id="cb41-132"><a href="#cb41-132" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggdist)</span>
<span id="cb41-133"><a href="#cb41-133" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span>
<span id="cb41-134"><a href="#cb41-134" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(glmmTMB)</span>
<span id="cb41-135"><a href="#cb41-135" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggrain)</span>
<span id="cb41-136"><a href="#cb41-136" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span>
<span id="cb41-137"><a href="#cb41-137" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggokabeito)</span>
<span id="cb41-138"><a href="#cb41-138" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ordbetareg)</span>
<span id="cb41-139"><a href="#cb41-139" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb41-140"><a href="#cb41-140" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb41-141"><a href="#cb41-141" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cmdstanr)</span>
<span id="cb41-142"><a href="#cb41-142" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(webshot2)</span>
<span id="cb41-143"><a href="#cb41-143" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(performance)</span>
<span id="cb41-144"><a href="#cb41-144" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(parameters)</span>
<span id="cb41-145"><a href="#cb41-145" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb41-146"><a href="#cb41-146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-147"><a href="#cb41-147" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(</span>
<span id="cb41-148"><a href="#cb41-148" aria-hidden="true" tabindex="-1"></a>  <span class="at">scipen =</span> <span class="dv">999</span>,</span>
<span id="cb41-149"><a href="#cb41-149" aria-hidden="true" tabindex="-1"></a>  <span class="at">marginaleffects_posterior_center =</span> mean,</span>
<span id="cb41-150"><a href="#cb41-150" aria-hidden="true" tabindex="-1"></a>  <span class="at">brms.short_summary =</span> <span class="cn">TRUE</span>,</span>
<span id="cb41-151"><a href="#cb41-151" aria-hidden="true" tabindex="-1"></a>  <span class="at">rms.backend =</span> <span class="fu">Sys.getenv</span>(<span class="st">"BRMS_BACKEND"</span>, <span class="st">"cmdstanr"</span>),</span>
<span id="cb41-152"><a href="#cb41-152" aria-hidden="true" tabindex="-1"></a>  <span class="at">brms.threads =</span> <span class="fu">as.numeric</span>(<span class="fu">Sys.getenv</span>(<span class="st">"BRMS_THREADS"</span>, <span class="dv">2</span>)),</span>
<span id="cb41-153"><a href="#cb41-153" aria-hidden="true" tabindex="-1"></a>  <span class="at">mc.cores =</span> <span class="fu">as.numeric</span>(<span class="fu">Sys.getenv</span>(<span class="st">"MAX_CORES"</span>, <span class="dv">4</span>))</span>
<span id="cb41-154"><a href="#cb41-154" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-155"><a href="#cb41-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-156"><a href="#cb41-156" aria-hidden="true" tabindex="-1"></a>is_html <span class="ot">&lt;-</span> knitr<span class="sc">::</span><span class="fu">is_html_output</span>()</span>
<span id="cb41-157"><a href="#cb41-157" aria-hidden="true" tabindex="-1"></a>opts_chunk<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb41-158"><a href="#cb41-158" aria-hidden="true" tabindex="-1"></a>  <span class="at">eval =</span> <span class="cn">TRUE</span>,</span>
<span id="cb41-159"><a href="#cb41-159" aria-hidden="true" tabindex="-1"></a>  <span class="at">cache =</span> <span class="cn">TRUE</span>,</span>
<span id="cb41-160"><a href="#cb41-160" aria-hidden="true" tabindex="-1"></a>  <span class="at">warning =</span> <span class="cn">FALSE</span>,</span>
<span id="cb41-161"><a href="#cb41-161" aria-hidden="true" tabindex="-1"></a>  <span class="at">error =</span> <span class="cn">FALSE</span>,</span>
<span id="cb41-162"><a href="#cb41-162" aria-hidden="true" tabindex="-1"></a>  <span class="at">message =</span> <span class="cn">FALSE</span>,</span>
<span id="cb41-163"><a href="#cb41-163" aria-hidden="true" tabindex="-1"></a>  <span class="at">echo =</span> is_html</span>
<span id="cb41-164"><a href="#cb41-164" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-165"><a href="#cb41-165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-166"><a href="#cb41-166" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">666</span>)</span>
<span id="cb41-167"><a href="#cb41-167" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-168"><a href="#cb41-168" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-171"><a href="#cb41-171" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-172"><a href="#cb41-172" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ggplot-theme</span></span>
<span id="cb41-173"><a href="#cb41-173" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-174"><a href="#cb41-174" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-175"><a href="#cb41-175" aria-hidden="true" tabindex="-1"></a>IS_HTML <span class="ot">&lt;-</span> knitr<span class="sc">::</span><span class="fu">is_html_output</span>()</span>
<span id="cb41-176"><a href="#cb41-176" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-177"><a href="#cb41-177" aria-hidden="true" tabindex="-1"></a>theme_publication <span class="ot">&lt;-</span> <span class="cf">function</span>(</span>
<span id="cb41-178"><a href="#cb41-178" aria-hidden="true" tabindex="-1"></a>  <span class="at">base_size =</span> <span class="fu">ifelse</span>(IS_HTML, <span class="dv">12</span>, <span class="dv">14</span>),</span>
<span id="cb41-179"><a href="#cb41-179" aria-hidden="true" tabindex="-1"></a>  <span class="at">base_family =</span> <span class="st">"Times"</span></span>
<span id="cb41-180"><a href="#cb41-180" aria-hidden="true" tabindex="-1"></a>) {</span>
<span id="cb41-181"><a href="#cb41-181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(</span>
<span id="cb41-182"><a href="#cb41-182" aria-hidden="true" tabindex="-1"></a>    <span class="at">text =</span> <span class="fu">element_text</span>(</span>
<span id="cb41-183"><a href="#cb41-183" aria-hidden="true" tabindex="-1"></a>      <span class="at">size =</span> base_size,</span>
<span id="cb41-184"><a href="#cb41-184" aria-hidden="true" tabindex="-1"></a>      <span class="at">family =</span> base_family,</span>
<span id="cb41-185"><a href="#cb41-185" aria-hidden="true" tabindex="-1"></a>      <span class="at">color =</span> <span class="st">"black"</span></span>
<span id="cb41-186"><a href="#cb41-186" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb41-187"><a href="#cb41-187" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> base_size <span class="sc">+</span> <span class="dv">2</span>),</span>
<span id="cb41-188"><a href="#cb41-188" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.subtitle =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> base_size),</span>
<span id="cb41-189"><a href="#cb41-189" aria-hidden="true" tabindex="-1"></a>    <span class="at">plot.caption =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">size =</span> base_size <span class="sc">-</span> <span class="dv">2</span>),</span>
<span id="cb41-190"><a href="#cb41-190" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> base_size),</span>
<span id="cb41-191"><a href="#cb41-191" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">t =</span> <span class="dv">10</span>)),</span>
<span id="cb41-192"><a href="#cb41-192" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">margin =</span> <span class="fu">margin</span>(<span class="at">r =</span> <span class="dv">10</span>)),</span>
<span id="cb41-193"><a href="#cb41-193" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size <span class="sc">-</span> <span class="dv">1</span>),</span>
<span id="cb41-194"><a href="#cb41-194" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.line =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb41-195"><a href="#cb41-195" aria-hidden="true" tabindex="-1"></a>    <span class="at">axis.ticks =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"black"</span>),</span>
<span id="cb41-196"><a href="#cb41-196" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.major =</span> <span class="fu">element_line</span>(<span class="at">color =</span> <span class="st">"gray80"</span>),</span>
<span id="cb41-197"><a href="#cb41-197" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb41-198"><a href="#cb41-198" aria-hidden="true" tabindex="-1"></a>    <span class="at">panel.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb41-199"><a href="#cb41-199" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb41-200"><a href="#cb41-200" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.title =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> base_size),</span>
<span id="cb41-201"><a href="#cb41-201" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.text =</span> <span class="fu">element_text</span>(<span class="at">size =</span> base_size <span class="sc">-</span> <span class="dv">1</span>),</span>
<span id="cb41-202"><a href="#cb41-202" aria-hidden="true" tabindex="-1"></a>    <span class="at">legend.background =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb41-203"><a href="#cb41-203" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.text =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">size =</span> base_size),</span>
<span id="cb41-204"><a href="#cb41-204" aria-hidden="true" tabindex="-1"></a>    <span class="at">strip.background =</span> <span class="fu">element_rect</span>(<span class="at">fill =</span> <span class="st">"gray90"</span>, <span class="at">color =</span> <span class="st">"black"</span>)</span>
<span id="cb41-205"><a href="#cb41-205" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-206"><a href="#cb41-206" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb41-207"><a href="#cb41-207" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-208"><a href="#cb41-208" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-209"><a href="#cb41-209" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-212"><a href="#cb41-212" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-213"><a href="#cb41-213" aria-hidden="true" tabindex="-1"></a>path <span class="ot">&lt;-</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>)</span>
<span id="cb41-214"><a href="#cb41-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-215"><a href="#cb41-215" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the directory if it doesn't exist</span></span>
<span id="cb41-216"><a href="#cb41-216" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="sc">!</span><span class="fu">dir.exists</span>(path)) {</span>
<span id="cb41-217"><a href="#cb41-217" aria-hidden="true" tabindex="-1"></a>  <span class="fu">dir.create</span>(path, <span class="at">recursive =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-218"><a href="#cb41-218" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb41-219"><a href="#cb41-219" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-220"><a href="#cb41-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-223"><a href="#cb41-223" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-224"><a href="#cb41-224" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: data-import</span></span>
<span id="cb41-225"><a href="#cb41-225" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-226"><a href="#cb41-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-227"><a href="#cb41-227" aria-hidden="true" tabindex="-1"></a>fluency_data <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"data"</span>, <span class="st">"miko_data.csv"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb41-228"><a href="#cb41-228" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(</span>
<span id="cb41-229"><a href="#cb41-229" aria-hidden="true" tabindex="-1"></a>    <span class="at">Participant =</span> <span class="st">"ResponseID"</span>,</span>
<span id="cb41-230"><a href="#cb41-230" aria-hidden="true" tabindex="-1"></a>    <span class="at">Fluency =</span> <span class="st">"Condition"</span>,</span>
<span id="cb41-231"><a href="#cb41-231" aria-hidden="true" tabindex="-1"></a>    <span class="at">Time =</span> <span class="st">"name"</span>,</span>
<span id="cb41-232"><a href="#cb41-232" aria-hidden="true" tabindex="-1"></a>    <span class="at">Accuracy =</span> <span class="st">"value"</span></span>
<span id="cb41-233"><a href="#cb41-233" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb41-234"><a href="#cb41-234" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-235"><a href="#cb41-235" aria-hidden="true" tabindex="-1"></a>    <span class="at">Accuracy =</span> Accuracy <span class="sc">/</span> <span class="dv">10</span>,</span>
<span id="cb41-236"><a href="#cb41-236" aria-hidden="true" tabindex="-1"></a>    <span class="at">Fluency =</span> <span class="fu">ifelse</span>(Fluency <span class="sc">==</span> <span class="dv">1</span>, <span class="st">"Fluent"</span>, <span class="st">"Disfluent"</span>)</span>
<span id="cb41-237"><a href="#cb41-237" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb41-238"><a href="#cb41-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Time <span class="sc">==</span> <span class="st">"FreeCt1AVG"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb41-239"><a href="#cb41-239" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>X, <span class="sc">-</span>Time) <span class="sc">%&gt;%</span></span>
<span id="cb41-240"><a href="#cb41-240" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Participant =</span> <span class="fu">dense_rank</span>(Participant)) <span class="sc">%&gt;%</span></span>
<span id="cb41-241"><a href="#cb41-241" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(Accuracy, <span class="at">.after =</span> <span class="fu">last_col</span>())</span>
<span id="cb41-242"><a href="#cb41-242" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-243"><a href="#cb41-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-244"><a href="#cb41-244" aria-hidden="true" tabindex="-1"></a><span class="fu"># Introduction</span></span>
<span id="cb41-245"><a href="#cb41-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-246"><a href="#cb41-246" aria-hidden="true" tabindex="-1"></a>Many outcomes in psychological research are naturally expressed as proportions or percentages. These include the proportion of correct responses on a test (e.g., @kornell2008), the proportion of time a participant fixates on a stimulus in an eye-tracking task <span class="co">[</span><span class="ot">e.g., @james2025</span><span class="co">]</span>, or the proportion of respondents endorsing a particular belief <span class="co">[</span><span class="ot">e.g., @Costello2024</span><span class="co">]</span>. A central question for studies involving proportional data is how to analyze these outcomes in a statistically appropriate and interpretable way.</span>
<span id="cb41-247"><a href="#cb41-247" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-248"><a href="#cb41-248" aria-hidden="true" tabindex="-1"></a>Researchers frequently default to linear models that assume Gaussian (normal) distributions, such as t-tests, ANOVAs, and linear regression. However, these models make strong assumptions: (1) residuals are normally distributed, (2) the outcome is unbounded (from $-\infty$ to $\infty$), and (3) variance is constant across the range of the data. These assumptions are rarely satisfied in practice <span class="co">[</span><span class="ot">@sladekova2024unicorns</span><span class="co">]</span>, and they are especially ill-suited for proportional outcomes, which are bounded between 0 and 1 and often exhibit heteroscedasticity—non-constant variance, particularly near the boundaries <span class="co">[</span><span class="ot">@smithson2006; @ferrari2004; @paolino2001</span><span class="co">]</span>. Violating these assumptions can lead to biased estimates and spurious inferences, especially when many observations cluster near 0 or 1.</span>
<span id="cb41-249"><a href="#cb41-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-250"><a href="#cb41-250" aria-hidden="true" tabindex="-1"></a>In some cases, a generalized linear model (GLM) can relax the assumption of normality. For example, binomial and Bernoulli models—often referred to as logistic regression when used with a logit link—are appropriate when the outcome is binary (0/1) or represents the number of successes out of a fixed number of trials. These models, however, rely on data that arise from discrete trial structures. They tend to perform poorly when the observed proportions are truly continuous or when the data show extra variability (overdispersion), particularly when many values occur near the boundaries of the scale (0 and 1).</span>
<span id="cb41-251"><a href="#cb41-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-252"><a href="#cb41-252" aria-hidden="true" tabindex="-1"></a>The challenges of analyzing proportional data are not new <span class="co">[</span><span class="ot">see @bartlett1936</span><span class="co">]</span>. Fortunately, several existing approaches address the limitations of commonly used models. One such approach is beta regression, an extension of the generalized linear model that employs the beta distribution <span class="co">[</span><span class="ot">@ferrari2004; @paolino2001</span><span class="co">]</span>. Beta regression offers a flexible and robust solution for modeling proportional data directly by accounting for boundary effects and over-dispersion, making it a valuable alternative to traditional binomial models. This approach is particularly well-suited for psychological research because it can handle both the bounded nature of proportional data and the non-constant variance often encountered in these datasets <span class="co">[</span><span class="ot">@sladekova2024unicorns</span><span class="co">]</span>. In addition, the direct modeling of proportions allows comparability across tasks and scales, and can be particularly valuable when only the proportional data is available, as is often the case with secondary data that lack item-level structure or point values. While usage of these models has faced obstacles due to theoretical and computational limitations, as we argue in this paper, beta regression and its extensions now provide an accessible and more robust method to traditional linear modeling.</span>
<span id="cb41-253"><a href="#cb41-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-254"><a href="#cb41-254" aria-hidden="true" tabindex="-1"></a>While in this paper we will focus on proportional-responses that lie between 0 and 1--it is important to note that our analysis applies to any bounded continuous scale. Any bounded scale can be mapped to lie within 0 and 1 without resulting in a loss of information as the transformation is linear.<span class="ot">[^1]</span> Consequently, a scale that has natural end points of -1,234 and +8,451--or any other end points on the real number line short of infinity—can be modeled using the approaches we describe in this paper.</span>
<span id="cb41-255"><a href="#cb41-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-256"><a href="#cb41-256" aria-hidden="true" tabindex="-1"></a><span class="ot">[^1]: </span>Specifically, for any continuous bounded variable $x$, we can rescale this variable to lie within 0 and 1 by using the formula $x' = \frac{x - \min(x)}{\max(x) - \min(x)}$ where $0 \leq x' \leq 1$.</span>
<span id="cb41-257"><a href="#cb41-257" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-258"><a href="#cb41-258" aria-hidden="true" tabindex="-1"></a><span class="fu">## A Beta Way Is Possible</span></span>
<span id="cb41-259"><a href="#cb41-259" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-260"><a href="#cb41-260" aria-hidden="true" tabindex="-1"></a>With the widespread availability of open-source software such as R <span class="co">[</span><span class="ot">@R</span><span class="co">]</span> and its extensive ecosystem of user-developed packages, advanced models like beta regression have become increasingly accessible to applied researchers. Yet, their adoption in psychology remains relatively limited. One contributing factor may be the lack of domain-specific examples that demonstrate how these models address common challenges in psychological data. Although recent years have seen a growing interest in beta regression, and a number of useful tutorials are available <span class="co">[</span><span class="ot">e.g., @smithson2006; @vuorre2019; @heiss2021; @bendixen2023; @coretta2025bayesian</span><span class="co">]</span>, these resources often either focus on basic implementation or briefly mention extensions without detailing how they can be applied to psychologically relevant research questions.</span>
<span id="cb41-261"><a href="#cb41-261" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-262"><a href="#cb41-262" aria-hidden="true" tabindex="-1"></a>The present tutorial aims to help bridge this gap by providing a comprehensive, applied introduction to beta regression and several of its extensions. In addition to the standard beta model, we walk through zero-inflated, zero-one-inflated, and ordered beta regression. These models are particularly useful for researchers working with proportion outcomes that include boundary values (e.g., exact 0s or 1s) or responses with an inherent ordinal structure. Our goal is to offer practical guidance that enables psychological researchers to implement, interpret, and report these models in ways that directly support their empirical questions.</span>
<span id="cb41-263"><a href="#cb41-263" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-264"><a href="#cb41-264" aria-hidden="true" tabindex="-1"></a>Beyond model specification, we place strong emphasis on interpreting results on the response scale--that is, in terms of probabilities and proportions--rather than relying on often difficult to interpret parameters. This focus makes the models more accessible and meaningful for psychological applications, where effects are often easier to communicate when framed on the original scale of the outcome (e.g., changes in recall accuracy or task performance). Throughout, we provide reproducible code and annotated examples to help readers implement and interpret these models in their own work.</span>
<span id="cb41-265"><a href="#cb41-265" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-266"><a href="#cb41-266" aria-hidden="true" tabindex="-1"></a>We begin the tutorial with a non-technical overview of the beta distribution and its core parameters. We then walk through the process of estimating beta regression models using the R package {brms} <span class="co">[</span><span class="ot">@brms</span><span class="co">]</span>, illustrating each step with applied examples. To guide interpretation, we emphasize coefficients, predicted probabilities, and marginal effects calculated using the {marginaleffects} package <span class="co">[</span><span class="ot">@margeffects2024</span><span class="co">]</span>. We also introduce several useful extensions--zero-inflated (ZIB), zero-one-inflated (ZOIB), and ordered beta regression--that enable researchers to model outcomes that include boundary values. Finally, all code and materials used in this tutorial are fully reproducible and available via our GitHub repository: <span class="co">[</span><span class="ot">https://github.com/jgeller112/beta_regression_tutorial</span><span class="co">](https://github.com/jgeller112/Beta_regression_tutorial)</span>{.uri} and on Zenodo (https://doi.org/10.5281/zenodo.15830595) <span class="ot">[^2]</span>.</span>
<span id="cb41-267"><a href="#cb41-267" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-268"><a href="#cb41-268" aria-hidden="true" tabindex="-1"></a><span class="ot">[^2]: </span>In this article, we try to limit code where possible; however, the online version has all the code needed to reproduce all analyses herein. Furthermore, to promote transparency and reproducibility, the tutorial was written in R version 4.5.1 (@R) using Quarto (v.1.5.54), an open-source publishing system that allows for dynamic and static documents. This allows figures, tables, and text to be programmatically included directly in the manuscript, ensuring that all results are seamlessly integrated into the document. In addition, we use the <span class="in">`rix`</span> <span class="co">[</span><span class="ot">@rix</span><span class="co">]</span> R package which harnesses the power of the <span class="in">`nix`</span> <span class="co">[</span><span class="ot">@nix</span><span class="co">]</span> ecosystem to to help with computational reproducibility. Not only does this give us a snapshot of the packages used to create the current manuscript, but it also takes a snapshot of system dependencies used at run-time. This way reproducers can easily re-use the exact same environment by installing the <span class="in">`nix`</span> package manager and using the included default.nix file to set up the right environment. The README file in the GitHub repository contains detailed information on how to set this up to reproduce the contents of the current manuscript, including a video.</span>
<span id="cb41-269"><a href="#cb41-269" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-270"><a href="#cb41-270" aria-hidden="true" tabindex="-1"></a><span class="fu">## Beta Distribution</span></span>
<span id="cb41-271"><a href="#cb41-271" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-272"><a href="#cb41-272" aria-hidden="true" tabindex="-1"></a>Proportional data pose some challenges for standard modeling approaches: The data are bounded between 0 and 1 and often exhibit non-constant variance (heteroscedasticity) <span class="co">[</span><span class="ot">@ferrari2004; @paolino2001</span><span class="co">]</span>. Common distributions used within the generalized linear model frameworks often fail to capture these properties adequately, which can necessitate alternative modeling strategies.</span>
<span id="cb41-273"><a href="#cb41-273" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-274"><a href="#cb41-274" aria-hidden="true" tabindex="-1"></a>While we do not have time to delve fully into its derivation, the beta distribution is useful for modeling bounded continuous scales because it is the distribution for the probability of an event. Given that a probability can take on any value from near 0 (the event will not occur with certainty) to 1 (the event will occur with certainty), the beta distribution can likewise take on virtually any value in that bounded interval. As a consequence, the beta distribution is the maximum entropy distribution for *any* bounded continuous random variable, which means that the beta distribution can represent the full range of possibilities of such a scale.<span class="ot">[^3]</span> As a consequence, if we have a continuous scale with upper and lower bounds--and no other special conditions--the beta distribution will in principle provide a very good approximation of the uncertainty of the scale.</span>
<span id="cb41-275"><a href="#cb41-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-276"><a href="#cb41-276" aria-hidden="true" tabindex="-1"></a><span class="ot">[^3]: </span>Technically, this maximum entropy condition is satisfied because the beta(1,1) distribution is uniform over its support. In addition, we assume that the scale has been re-scaled to the $<span class="co">[</span><span class="ot">0,1</span><span class="co">]</span>$ interval as we describe above.</span>
<span id="cb41-277"><a href="#cb41-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-278"><a href="#cb41-278" aria-hidden="true" tabindex="-1"></a>Typically, the expected value (or mean) of the response variable is the central estimand scholars want to estimate. A model should specify how this expected value depends on explanatory variables through two main components: a linear predictor, which combines the explanatory variables in a linear form ($a+b_1x_1+b_2x_2$, etc.), and a link function, which connects the expected value of the response variable to the linear predictor (e.g, $E<span class="co">[</span><span class="ot">Y</span><span class="co">]</span> = g(a + b_1x_1 + b_2x_2)$). In addition, a random component specifies the distribution of the response variable around its expected value (such as Poisson or binomial distributions, which belong to the exponential family) <span class="co">[</span><span class="ot">@nelder1972</span><span class="co">]</span>. Together, these components provide a flexible framework for modeling data with different distributional properties.</span>
<span id="cb41-279"><a href="#cb41-279" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-280"><a href="#cb41-280" aria-hidden="true" tabindex="-1"></a>The beta distribution is continuous and restricted to values between 0 and 1 (exclusive). Its two parameters--commonly called shape1 ($\alpha$) and shape2 ($\beta$)--govern the distribution’s location, skewness, and spread. By adjusting these parameters, the distribution can take many functional forms (e.g., it can be symmetric, skewed, U-shaped, or even approximately uniform; see @fig-curves).</span>
<span id="cb41-281"><a href="#cb41-281" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-282"><a href="#cb41-282" aria-hidden="true" tabindex="-1"></a>To illustrate, consider a test question worth seven points. Suppose a participant scores five out of seven. The number of points received (5) can be treated as $\alpha$, and the number of points missed (2) as $\beta$. The resulting beta distribution would be skewed toward higher values, reflecting a high performance (pink line in @fig-curves; "beta(5, 2)"). Reversing these values would produce a distribution skewed toward lower values, representing poorer performance (orange line in @fig-curves; "beta(2, 5)").</span>
<span id="cb41-283"><a href="#cb41-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-286"><a href="#cb41-286" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-287"><a href="#cb41-287" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-curves</span></span>
<span id="cb41-288"><a href="#cb41-288" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "beta distributions with different shape1 and shape2 parameters."</span></span>
<span id="cb41-289"><a href="#cb41-289" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-asp: 0.52</span></span>
<span id="cb41-290"><a href="#cb41-290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-291"><a href="#cb41-291" aria-hidden="true" tabindex="-1"></a><span class="co"># Function to generate beta distribution data</span></span>
<span id="cb41-292"><a href="#cb41-292" aria-hidden="true" tabindex="-1"></a>make_beta_df <span class="ot">&lt;-</span> <span class="cf">function</span>(shape1, shape2, <span class="at">n =</span> <span class="dv">9000</span>) {</span>
<span id="cb41-293"><a href="#cb41-293" aria-hidden="true" tabindex="-1"></a>  x <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fl">0.001</span>, <span class="fl">0.999</span>, <span class="at">length.out =</span> n)</span>
<span id="cb41-294"><a href="#cb41-294" aria-hidden="true" tabindex="-1"></a>  <span class="fu">data.frame</span>(</span>
<span id="cb41-295"><a href="#cb41-295" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> x,</span>
<span id="cb41-296"><a href="#cb41-296" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="fu">dbeta</span>(x, shape1, shape2),</span>
<span id="cb41-297"><a href="#cb41-297" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(<span class="st">"beta("</span>, shape1, <span class="st">", "</span>, shape2, <span class="st">")"</span>)</span>
<span id="cb41-298"><a href="#cb41-298" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-299"><a href="#cb41-299" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb41-300"><a href="#cb41-300" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-301"><a href="#cb41-301" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine distributions</span></span>
<span id="cb41-302"><a href="#cb41-302" aria-hidden="true" tabindex="-1"></a>beta_df <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(</span>
<span id="cb41-303"><a href="#cb41-303" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_beta_df</span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>),</span>
<span id="cb41-304"><a href="#cb41-304" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_beta_df</span>(<span class="dv">2</span>, <span class="dv">2</span>),</span>
<span id="cb41-305"><a href="#cb41-305" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_beta_df</span>(<span class="dv">5</span>, <span class="dv">2</span>),</span>
<span id="cb41-306"><a href="#cb41-306" aria-hidden="true" tabindex="-1"></a>  <span class="fu">make_beta_df</span>(<span class="dv">2</span>, <span class="dv">5</span>)</span>
<span id="cb41-307"><a href="#cb41-307" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb41-308"><a href="#cb41-308" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">label =</span> <span class="fu">factor</span>(label))</span>
<span id="cb41-309"><a href="#cb41-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-310"><a href="#cb41-310" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(</span>
<span id="cb41-311"><a href="#cb41-311" aria-hidden="true" tabindex="-1"></a>  beta_df,</span>
<span id="cb41-312"><a href="#cb41-312" aria-hidden="true" tabindex="-1"></a>  <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> y, <span class="at">label =</span> label, <span class="at">group =</span> label, <span class="at">color =</span> label)</span>
<span id="cb41-313"><a href="#cb41-313" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb41-314"><a href="#cb41-314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="fl">0.9</span>) <span class="sc">+</span></span>
<span id="cb41-315"><a href="#cb41-315" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_okabe_ito</span>(</span>
<span id="cb41-316"><a href="#cb41-316" aria-hidden="true" tabindex="-1"></a>    <span class="at">order =</span> <span class="fu">c</span>(<span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">6</span>, <span class="dv">7</span>), <span class="co"># whichever correspond to desired colors</span></span>
<span id="cb41-317"><a href="#cb41-317" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.7</span>,</span>
<span id="cb41-318"><a href="#cb41-318" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"Distribution"</span></span>
<span id="cb41-319"><a href="#cb41-319" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-320"><a href="#cb41-320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(</span>
<span id="cb41-321"><a href="#cb41-321" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Proportion"</span>,</span>
<span id="cb41-322"><a href="#cb41-322" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="fl">0.01</span>)</span>
<span id="cb41-323"><a href="#cb41-323" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-324"><a href="#cb41-324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(</span>
<span id="cb41-325"><a href="#cb41-325" aria-hidden="true" tabindex="-1"></a>    <span class="st">"Density"</span>,</span>
<span id="cb41-326"><a href="#cb41-326" aria-hidden="true" tabindex="-1"></a>    <span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">5</span>),</span>
<span id="cb41-327"><a href="#cb41-327" aria-hidden="true" tabindex="-1"></a>    <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="fl">0.01</span>)</span>
<span id="cb41-328"><a href="#cb41-328" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-329"><a href="#cb41-329" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb41-330"><a href="#cb41-330" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>)</span>
<span id="cb41-331"><a href="#cb41-331" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-332"><a href="#cb41-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-333"><a href="#cb41-333" aria-hidden="true" tabindex="-1"></a><span class="fu">## I Can't Believe It's Not beta</span></span>
<span id="cb41-334"><a href="#cb41-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-335"><a href="#cb41-335" aria-hidden="true" tabindex="-1"></a>While the standard parameterization of the beta distribution uses $\alpha$ and $\beta$, a reparameterization to a mean ($\mu$) and precision ($\phi$) is more useful for regression models. The mean represents the expected value of the distribution, while the dispersion, which is inversely related to variance, reflects how concentrated the distribution is around the mean, with higher values indicating a narrower distribution and lower values indicating a wider one. The connections between the beta distribution's parameters are shown in @eq-beta. Importantly, the variance depends on the average value of the response because uncertainty intervals need to adjust for how close the value of the response is to the boundary.</span>
<span id="cb41-336"><a href="#cb41-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-337"><a href="#cb41-337" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb41-338"><a href="#cb41-338" aria-hidden="true" tabindex="-1"></a>\begin{aligned}<span class="co">[</span><span class="ot">t</span><span class="co">]</span></span>
<span id="cb41-339"><a href="#cb41-339" aria-hidden="true" tabindex="-1"></a>\text{Shape 1:} &amp;&amp; a &amp;= \mu \phi <span class="sc">\\</span></span>
<span id="cb41-340"><a href="#cb41-340" aria-hidden="true" tabindex="-1"></a>\text{Shape 2:} &amp;&amp; b &amp;= (1 - \mu) \phi</span>
<span id="cb41-341"><a href="#cb41-341" aria-hidden="true" tabindex="-1"></a>\end{aligned}</span>
<span id="cb41-342"><a href="#cb41-342" aria-hidden="true" tabindex="-1"></a>\qquad\qquad\qquad</span>
<span id="cb41-343"><a href="#cb41-343" aria-hidden="true" tabindex="-1"></a>\begin{aligned}<span class="co">[</span><span class="ot">t</span><span class="co">]</span></span>
<span id="cb41-344"><a href="#cb41-344" aria-hidden="true" tabindex="-1"></a>\text{Mean:} &amp;&amp; \mu &amp;= \frac{a}{a + b} <span class="sc">\\</span></span>
<span id="cb41-345"><a href="#cb41-345" aria-hidden="true" tabindex="-1"></a>\text{Precision:} &amp;&amp; \phi &amp;= a + b <span class="sc">\\</span></span>
<span id="cb41-346"><a href="#cb41-346" aria-hidden="true" tabindex="-1"></a>\text{Variance:} &amp;&amp; var &amp;= \frac{\mu \cdot (1 - \mu)}{1 + \phi}</span>
<span id="cb41-347"><a href="#cb41-347" aria-hidden="true" tabindex="-1"></a>\end{aligned}</span>
<span id="cb41-348"><a href="#cb41-348" aria-hidden="true" tabindex="-1"></a>$$ {#eq-beta}</span>
<span id="cb41-349"><a href="#cb41-349" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-350"><a href="#cb41-350" aria-hidden="true" tabindex="-1"></a>Thus, beta regression allows modeling both the mean and precision of the outcome distribution. To ensure that $\mu$ stays between 0 and 1, we apply a link function, which allows linear modeling of the mean on an unbounded scale. A common link-function choice is the logit, but other functions such as the probit or complementary log-log are possible.</span>
<span id="cb41-351"><a href="#cb41-351" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-352"><a href="#cb41-352" aria-hidden="true" tabindex="-1"></a>The logit function, $\text{logit}(\mu) = \log \left( \frac{\mu}{1 - \mu} \right)$ links the mean to log-odds which are unbounded, making linear modeling possible. The logit here no longer carries the same literal *odds* interpretation because there are no corresponding counts of “successes” and “failures.” Instead, the logit transform here simply maps the mean of the distribution to the real line. The inverse of the logit, called the logistic function, maps the linear predictor $\eta$ back to the original scale of the data $\left(\mu = \frac{1}{1 + e^{-\eta}}\right)$. The coefficients describe how predictors shift the *average proportion* on the logit scale. Similarly, the strictly positive dispersion parameter is usually modeled through a log link function, ensuring it remains positive.</span>
<span id="cb41-353"><a href="#cb41-353" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-354"><a href="#cb41-354" aria-hidden="true" tabindex="-1"></a>By accounting for the observations' natural limits and non-constant variance, the beta distribution is useful in psychology where outcomes like performance rates or response scales frequently exhibit these features.</span>
<span id="cb41-355"><a href="#cb41-355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-356"><a href="#cb41-356" aria-hidden="true" tabindex="-1"></a><span class="fu">## Bayesian Approach to Beta Regression</span></span>
<span id="cb41-357"><a href="#cb41-357" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-358"><a href="#cb41-358" aria-hidden="true" tabindex="-1"></a>Beta regression models can be estimated using either frequentist or Bayesian methods. In this paper, we adopt a Bayesian framework because it facilitates the estimation and interpretation of more complex models <span class="co">[</span><span class="ot">@gelman2013bayesian; @mcelreath2020statistical; @bayes_rules</span><span class="co">]</span>. Additionally, the use of Bayesin statistics in psychology has been steadily growing <span class="co">[</span><span class="ot">@Pfadt2025</span><span class="co">]</span>. In principle, frequentist methods like maximum likelihood can be framed as Bayesian models with uninformative priors, and as a result, the modeling perspective we put forward in this paper can apply to either approach. Nonetheless, we note that in non-linear and hierarchical models, frequentist estimation may require additional adjustments such as bootstrapping to obtain proper uncertainty intervals, whereas Bayesian modeling handles these extensions more naturally via exploration of the full joint posterior distribution.<span class="ot">[^4]</span></span>
<span id="cb41-359"><a href="#cb41-359" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-360"><a href="#cb41-360" aria-hidden="true" tabindex="-1"></a><span class="ot">[^4]: </span>A common concern is that Bayesian methods are slower than frequentist ones. While this is true in general, modern Bayesian computation engines are efficient, and in explanatory modeling our priority is to specify appropriate estimands rather than to optimize for computation speed. Moreover, we use the {brms} package <span class="co">[</span><span class="ot">@brms</span><span class="co">]</span>, which provides a high-level interface to Stan <span class="co">[</span><span class="ot">@stan2023</span><span class="co">]</span> and uses familiar R formula syntax (similar to <span class="in">`lm()`</span>), making advanced Bayesian modeling accessible to researchers with standard regression backgrounds. The package also supports parallelization, which substantially reduces computation time for large datasets.</span>
<span id="cb41-361"><a href="#cb41-361" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-362"><a href="#cb41-362" aria-hidden="true" tabindex="-1"></a>There are several important differences between our Bayesian analysis and the frequentist methods readers may be more familiar with—most notably, the absence of *t*- and *p*-values. To estimate models, the {brms} package uses Stan’s computational algorithms to draw random samples from the posterior distribution, which represents uncertainty about the model parameters. This posterior is conceptually analogous to a frequentist sampling distribution. By default, Bayesian models run 4 chains with 2,000 iterations each.<span class="ot">[^5]</span> The first 1,000 iterations per chain are warmup and are discarded. The remaining 1,000 iterations per chain are retained as posterior draws, yielding 4,000 total post-warmup draws across all chains. From these draws, we can compute the posterior mean (analogous to a frequentist point estimate) and the 95% credible interval (Cr.I.), which is often compared to a confidence interval.</span>
<span id="cb41-363"><a href="#cb41-363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-364"><a href="#cb41-364" aria-hidden="true" tabindex="-1"></a><span class="ot">[^5]: </span>The Hamiltonian Monte Carlo sampler employed by Stan, which we also use in this paper, can converge with significantly fewer iterations, though rapid convergence depends on model complexity, which is why we use a more conservative standard in this paper.</span>
<span id="cb41-365"><a href="#cb41-365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-366"><a href="#cb41-366" aria-hidden="true" tabindex="-1"></a>In addition, an important part of Bayesian analyses is prior specification. Priors encode our assumptions about plausible parameter values before observing the data and allow the model to regularize estimates, especially when data are sparse or parameters are weakly identified. To help bridge the conceptual gap for users more familiar with frequentist models, we begin with the default priors (flat/non-informative, or weakly informative in some cases) provided by {brms}. These priors are intentionally non-informative, and in many applications produce results that closely align with frequentist estimates, while still offering the flexibility and interpretive advantages of a Bayesian framework. We strongly urge readers to consider prior specification strongly in all their work.</span>
<span id="cb41-367"><a href="#cb41-367" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-368"><a href="#cb41-368" aria-hidden="true" tabindex="-1"></a>To ease readers into Bayesian data analysis we provide a metric known as the *probability of direction* (pd), which reflects the probability that a parameter is positive or negative. When a uniform prior is used (all values equally likely in the prior), pds of 95%, 97.5%, 99.5%, and 99.95% corresponds approximately to two-sided p-values of .10, .05, .01, and .001 (i.e., pd ≈ 1 − p/2 for symmetric posteriors with weak/flat priors) (see @fig-p-value-pd-comp for an illustrative comparison). For directional hypotheses, the pd can be interpreted as roughly equivalent to one minus the *p*-value <span class="co">[</span><span class="ot">@marsmanThreeInsightsBayesian2016</span><span class="co">]</span>.</span>
<span id="cb41-369"><a href="#cb41-369" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-372"><a href="#cb41-372" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-373"><a href="#cb41-373" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-p-value-pd-comp</span></span>
<span id="cb41-374"><a href="#cb41-374" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "A Bayesian posterior distribution (assuming a uniform prior) centered at a point estimate chosen so that the probability of direction (pd) equals 0.95, and a frequentist sampling distribution (under the null; centered at 0). In the Bayesian posterior distribution, the blue area represents the pd, and the red area represents the remaining 1 − pd of the distribution. In the frequentist sampling distribution, the red tail areas represent the rejection region at α = 0.10. In this example, the posterior mean lies exactly at the $1 - \\frac{\\alpha}{2}$ quantile of the null sampling distribution. For symmetric posteriors with flat priors, the pd is numerically equivalent to the one-sided $p$-value."</span></span>
<span id="cb41-375"><a href="#cb41-375" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-asp: 0.8</span></span>
<span id="cb41-376"><a href="#cb41-376" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb41-377"><a href="#cb41-377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-378"><a href="#cb41-378" aria-hidden="true" tabindex="-1"></a><span class="co"># --- parameters ---</span></span>
<span id="cb41-379"><a href="#cb41-379" aria-hidden="true" tabindex="-1"></a>pd <span class="ot">&lt;-</span> <span class="fl">0.95</span></span>
<span id="cb41-380"><a href="#cb41-380" aria-hidden="true" tabindex="-1"></a>alpha_two <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> (<span class="dv">1</span> <span class="sc">-</span> pd) <span class="co"># matching two-sided alpha</span></span>
<span id="cb41-381"><a href="#cb41-381" aria-hidden="true" tabindex="-1"></a>mu <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(pd) <span class="co"># shift so P(X &gt; 0) = pd for N(mu, 1)</span></span>
<span id="cb41-382"><a href="#cb41-382" aria-hidden="true" tabindex="-1"></a>lim <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">4</span>, <span class="dv">6</span>)</span>
<span id="cb41-383"><a href="#cb41-383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-384"><a href="#cb41-384" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Posterior-like panel ---</span></span>
<span id="cb41-385"><a href="#cb41-385" aria-hidden="true" tabindex="-1"></a>post <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x =</span> lim), <span class="fu">aes</span>(x)) <span class="sc">+</span></span>
<span id="cb41-386"><a href="#cb41-386" aria-hidden="true" tabindex="-1"></a>  <span class="co"># full curve</span></span>
<span id="cb41-387"><a href="#cb41-387" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb41-388"><a href="#cb41-388" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb41-389"><a href="#cb41-389" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> mu, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb41-390"><a href="#cb41-390" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb41-391"><a href="#cb41-391" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span></span>
<span id="cb41-392"><a href="#cb41-392" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-393"><a href="#cb41-393" aria-hidden="true" tabindex="-1"></a>  <span class="co"># pd region (blue)</span></span>
<span id="cb41-394"><a href="#cb41-394" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb41-395"><a href="#cb41-395" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb41-396"><a href="#cb41-396" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> mu, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb41-397"><a href="#cb41-397" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(<span class="dv">0</span>, lim[<span class="dv">2</span>]),</span>
<span id="cb41-398"><a href="#cb41-398" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"area"</span>,</span>
<span id="cb41-399"><a href="#cb41-399" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"blue"</span>,</span>
<span id="cb41-400"><a href="#cb41-400" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb41-401"><a href="#cb41-401" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-402"><a href="#cb41-402" aria-hidden="true" tabindex="-1"></a>  <span class="co"># remaining region (red)</span></span>
<span id="cb41-403"><a href="#cb41-403" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb41-404"><a href="#cb41-404" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb41-405"><a href="#cb41-405" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> mu, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb41-406"><a href="#cb41-406" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(lim[<span class="dv">1</span>], <span class="dv">0</span>),</span>
<span id="cb41-407"><a href="#cb41-407" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"area"</span>,</span>
<span id="cb41-408"><a href="#cb41-408" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"red"</span>,</span>
<span id="cb41-409"><a href="#cb41-409" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb41-410"><a href="#cb41-410" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-411"><a href="#cb41-411" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linewidth =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb41-412"><a href="#cb41-412" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb41-413"><a href="#cb41-413" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Posterior (illustrative Normal)"</span>,</span>
<span id="cb41-414"><a href="#cb41-414" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"pd = "</span>, <span class="fu">round</span>(pd <span class="sc">*</span> <span class="dv">100</span>), <span class="st">"%"</span>),</span>
<span id="cb41-415"><a href="#cb41-415" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">""</span>,</span>
<span id="cb41-416"><a href="#cb41-416" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Density"</span></span>
<span id="cb41-417"><a href="#cb41-417" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-418"><a href="#cb41-418" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> lim, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.45</span>)) <span class="sc">+</span></span>
<span id="cb41-419"><a href="#cb41-419" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>()</span>
<span id="cb41-420"><a href="#cb41-420" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-421"><a href="#cb41-421" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Frequentist panel ---</span></span>
<span id="cb41-422"><a href="#cb41-422" aria-hidden="true" tabindex="-1"></a>z_lo <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(alpha_two <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb41-423"><a href="#cb41-423" aria-hidden="true" tabindex="-1"></a>z_hi <span class="ot">&lt;-</span> <span class="fu">qnorm</span>(<span class="dv">1</span> <span class="sc">-</span> alpha_two <span class="sc">/</span> <span class="dv">2</span>)</span>
<span id="cb41-424"><a href="#cb41-424" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-425"><a href="#cb41-425" aria-hidden="true" tabindex="-1"></a>freq <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(<span class="fu">data.frame</span>(<span class="at">x =</span> lim), <span class="fu">aes</span>(x)) <span class="sc">+</span></span>
<span id="cb41-426"><a href="#cb41-426" aria-hidden="true" tabindex="-1"></a>  <span class="co"># full curve</span></span>
<span id="cb41-427"><a href="#cb41-427" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb41-428"><a href="#cb41-428" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb41-429"><a href="#cb41-429" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb41-430"><a href="#cb41-430" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb41-431"><a href="#cb41-431" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">1</span></span>
<span id="cb41-432"><a href="#cb41-432" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-433"><a href="#cb41-433" aria-hidden="true" tabindex="-1"></a>  <span class="co"># lower rejection tail (red)</span></span>
<span id="cb41-434"><a href="#cb41-434" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb41-435"><a href="#cb41-435" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb41-436"><a href="#cb41-436" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb41-437"><a href="#cb41-437" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(lim[<span class="dv">1</span>], z_lo),</span>
<span id="cb41-438"><a href="#cb41-438" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"area"</span>,</span>
<span id="cb41-439"><a href="#cb41-439" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"red"</span>,</span>
<span id="cb41-440"><a href="#cb41-440" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb41-441"><a href="#cb41-441" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-442"><a href="#cb41-442" aria-hidden="true" tabindex="-1"></a>  <span class="co"># upper rejection tail (red)</span></span>
<span id="cb41-443"><a href="#cb41-443" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_function</span>(</span>
<span id="cb41-444"><a href="#cb41-444" aria-hidden="true" tabindex="-1"></a>    <span class="at">fun =</span> dnorm,</span>
<span id="cb41-445"><a href="#cb41-445" aria-hidden="true" tabindex="-1"></a>    <span class="at">args =</span> <span class="fu">list</span>(<span class="at">mean =</span> <span class="dv">0</span>, <span class="at">sd =</span> <span class="dv">1</span>),</span>
<span id="cb41-446"><a href="#cb41-446" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlim =</span> <span class="fu">c</span>(z_hi, lim[<span class="dv">2</span>]),</span>
<span id="cb41-447"><a href="#cb41-447" aria-hidden="true" tabindex="-1"></a>    <span class="at">geom =</span> <span class="st">"area"</span>,</span>
<span id="cb41-448"><a href="#cb41-448" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"red"</span>,</span>
<span id="cb41-449"><a href="#cb41-449" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.3</span></span>
<span id="cb41-450"><a href="#cb41-450" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-451"><a href="#cb41-451" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="fu">c</span>(z_lo, z_hi), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linewidth =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb41-452"><a href="#cb41-452" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linewidth =</span> <span class="fl">0.8</span>) <span class="sc">+</span></span>
<span id="cb41-453"><a href="#cb41-453" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb41-454"><a href="#cb41-454" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Frequentist sampling under H0 (N(0,1))"</span>,</span>
<span id="cb41-455"><a href="#cb41-455" aria-hidden="true" tabindex="-1"></a>    <span class="at">subtitle =</span> <span class="fu">paste0</span>(<span class="st">"two-sided α = "</span>, <span class="fu">signif</span>(alpha_two, <span class="dv">2</span>)),</span>
<span id="cb41-456"><a href="#cb41-456" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Parameter Values"</span>,</span>
<span id="cb41-457"><a href="#cb41-457" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Density"</span></span>
<span id="cb41-458"><a href="#cb41-458" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-459"><a href="#cb41-459" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">xlim =</span> lim, <span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="fl">0.45</span>)) <span class="sc">+</span></span>
<span id="cb41-460"><a href="#cb41-460" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>()</span>
<span id="cb41-461"><a href="#cb41-461" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-462"><a href="#cb41-462" aria-hidden="true" tabindex="-1"></a><span class="co"># --- Stack the plots ---</span></span>
<span id="cb41-463"><a href="#cb41-463" aria-hidden="true" tabindex="-1"></a>post <span class="sc">/</span> freq</span>
<span id="cb41-464"><a href="#cb41-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-465"><a href="#cb41-465" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-466"><a href="#cb41-466" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-467"><a href="#cb41-467" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-468"><a href="#cb41-468" aria-hidden="true" tabindex="-1"></a>For reasons of space, we refer readers unfamiliar to Bayesian data analysis to several existing books on the topic <span class="co">[</span><span class="ot">@gelman2013bayesian; @mcelreath2020statistical; @kruschke2015doing</span><span class="co">]</span>. In addition, we assume readers are familiar with R, but those in need of a refresher should find @wickhamDataScienceImport2023 useful.</span>
<span id="cb41-469"><a href="#cb41-469" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-470"><a href="#cb41-470" aria-hidden="true" tabindex="-1"></a><span class="fu"># Beta Regression Tutorial</span></span>
<span id="cb41-471"><a href="#cb41-471" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-472"><a href="#cb41-472" aria-hidden="true" tabindex="-1"></a><span class="fu">## Example Data</span></span>
<span id="cb41-473"><a href="#cb41-473" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-474"><a href="#cb41-474" aria-hidden="true" tabindex="-1"></a>Throughout this tutorial, we analyze data from a memory experiment examining whether the fluency of an instructor’s delivery affects recall performance <span class="co">[</span><span class="ot">@wilford2020, Experiment 1A</span><span class="co">]</span>. Instructor fluency—marked by expressive gestures, dynamic vocal tone, and confident pacing—has been shown to influence students’ perceptions of learning, often leading learners to rate fluent instructors more favorably <span class="co">[</span><span class="ot">@carpenter2013</span><span class="co">]</span>. However, previous research suggests that these impressions do not reliably translate into improved memory performance <span class="co">[</span><span class="ot">e.g., @carpenter2013; @toftness2017; @witherby2022</span><span class="co">]</span>. In contrast, @wilford2020 found that participants actually recalled more information after watching a fluent instructor compared to a disfluent one. This surprising finding makes the dataset a compelling case study for analyzing proportion data, as recall was scored out of 10 possible idea units per video.</span>
<span id="cb41-475"><a href="#cb41-475" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-476"><a href="#cb41-476" aria-hidden="true" tabindex="-1"></a>In Experiment 1A, ninety-six participants watched two short instructional videos, each delivered either fluently or disfluently. Fluent videos featured instructors with smooth delivery and natural pacing, while disfluent videos included hesitations, monotone speech, and awkward pauses. After a distractor task, participants completed a free recall test, writing down as much content as they could remember from each video within a three-minute window. Their recall was then scored for the number of idea units correctly remembered.</span>
<span id="cb41-477"><a href="#cb41-477" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-480"><a href="#cb41-480" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-481"><a href="#cb41-481" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-data</span></span>
<span id="cb41-482"><a href="#cb41-482" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Data needed to run examples</span></span>
<span id="cb41-483"><a href="#cb41-483" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb41-484"><a href="#cb41-484" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-485"><a href="#cb41-485" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-486"><a href="#cb41-486" aria-hidden="true" tabindex="-1"></a><span class="co"># get data here from project folder</span></span>
<span id="cb41-487"><a href="#cb41-487" aria-hidden="true" tabindex="-1"></a>fluency_data <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"data"</span>, <span class="st">"fluency_data.csv"</span>))</span>
<span id="cb41-488"><a href="#cb41-488" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-489"><a href="#cb41-489" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-490"><a href="#cb41-490" aria-hidden="true" tabindex="-1"></a>Our primary outcome variable is the proportion of idea units recalled on the final test, calculated by dividing the number of correct units by 10. We show a sample of these data in @tbl-dataset. The dataset can be downloaded from GitHub (@lst-data). Because this is a bounded continuous variable (i.e., it ranges from 0 to 1), it violates the assumptions of typical linear regression models that assume normally distributed residual errors. Despite this, it remains common in psychological research to analyze proportion data using models that assume normality. In what follows, we reproduce @wilford2020's analysis and then re-analyze the data using beta regression and highlight how it can improve our inferences.</span>
<span id="cb41-491"><a href="#cb41-491" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-494"><a href="#cb41-494" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-495"><a href="#cb41-495" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-dataset</span></span>
<span id="cb41-496"><a href="#cb41-496" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Four observations from @wilford2020. Accuracy refers to the proportion of correctly recalled idea units."</span></span>
<span id="cb41-497"><a href="#cb41-497" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-498"><a href="#cb41-498" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb41-499"><a href="#cb41-499" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-500"><a href="#cb41-500" aria-hidden="true" tabindex="-1"></a>fluency_data <span class="sc">|&gt;</span></span>
<span id="cb41-501"><a href="#cb41-501" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sample_n</span>(<span class="dv">4</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-502"><a href="#cb41-502" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-503"><a href="#cb41-503" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-504"><a href="#cb41-504" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-505"><a href="#cb41-505" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-506"><a href="#cb41-506" aria-hidden="true" tabindex="-1"></a><span class="fu">## Reanalysis of Wilford et al. Experiment 1A</span></span>
<span id="cb41-507"><a href="#cb41-507" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-508"><a href="#cb41-508" aria-hidden="true" tabindex="-1"></a>In their original analysis of Experiment 1A, @wilford2020 compared memory performance between fluent and disfluent instructor conditions using a traditional independent-samples t-test on mean accuracy for 96 participants. They found that participants who watched the fluent instructor recalled significantly more idea units than those who viewed the disfluent version (see @fig-flu1).</span>
<span id="cb41-509"><a href="#cb41-509" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-512"><a href="#cb41-512" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-513"><a href="#cb41-513" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-flu1</span></span>
<span id="cb41-514"><a href="#cb41-514" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Dot plot depicting accuracy distributions for the Fluent and Disfluent conditions. Each condition shows individual data points and summary statistics (mean and 95% percentile interval) to illustrate variability and central tendency."</span></span>
<span id="cb41-515"><a href="#cb41-515" aria-hidden="true" tabindex="-1"></a><span class="co">#|</span></span>
<span id="cb41-516"><a href="#cb41-516" aria-hidden="true" tabindex="-1"></a>fluency_data <span class="sc">|&gt;</span></span>
<span id="cb41-517"><a href="#cb41-517" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> Accuracy, <span class="at">y =</span> Fluency, <span class="at">fill =</span> Fluency)) <span class="sc">+</span></span>
<span id="cb41-518"><a href="#cb41-518" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_dotsinterval</span>(</span>
<span id="cb41-519"><a href="#cb41-519" aria-hidden="true" tabindex="-1"></a>    <span class="at">point_interval =</span> <span class="st">"mean_qi"</span>,</span>
<span id="cb41-520"><a href="#cb41-520" aria-hidden="true" tabindex="-1"></a>    <span class="at">.width =</span> <span class="fl">0.95</span>,</span>
<span id="cb41-521"><a href="#cb41-521" aria-hidden="true" tabindex="-1"></a>    <span class="at">slab_linewidth =</span> <span class="dv">0</span></span>
<span id="cb41-522"><a href="#cb41-522" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-523"><a href="#cb41-523" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">label_percent</span>()) <span class="sc">+</span></span>
<span id="cb41-524"><a href="#cb41-524" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="cn">NULL</span>, <span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb41-525"><a href="#cb41-525" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_okabe_ito</span>(<span class="at">breaks =</span> <span class="cn">NULL</span>) <span class="sc">+</span></span>
<span id="cb41-526"><a href="#cb41-526" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>()</span>
<span id="cb41-527"><a href="#cb41-527" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-528"><a href="#cb41-528" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-529"><a href="#cb41-529" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-530"><a href="#cb41-530" aria-hidden="true" tabindex="-1"></a>We first replicate this analysis in a regression framework using {brms}. We model final test mean accuracy--the proportion of correctly recalled idea units across the videos--as the dependent variable. Our predictor is instructor fluency, with two levels: Fluent and Disfluent. We use treatment (dummy) coding, which is the default in R. This coding scheme sets the first level of a factor (in alphabetical order) as the reference level. In this case, Disfluent is the reference, and the coefficient for Fluent reflects the contrast between fluent and disfluent instructor conditions.</span>
<span id="cb41-531"><a href="#cb41-531" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-532"><a href="#cb41-532" aria-hidden="true" tabindex="-1"></a><span class="fu">### Regression Analysis</span></span>
<span id="cb41-533"><a href="#cb41-533" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-534"><a href="#cb41-534" aria-hidden="true" tabindex="-1"></a>We first start by loading the {brms} <span class="co">[</span><span class="ot">@brms</span><span class="co">]</span> and {cmdstanr} <span class="co">[</span><span class="ot">@cmdstanr</span><span class="co">]</span> packages (@lst-loadbrms). We use the cmdstanr backend for Stan <span class="co">[</span><span class="ot">@stan2023</span><span class="co">]</span> because it's faster than the default used to run models (i.e., <span class="in">`rstan`</span>),<span class="ot">[^6]</span> though all of these models can also be fit with <span class="in">`brms`</span> defaults.</span>
<span id="cb41-535"><a href="#cb41-535" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-536"><a href="#cb41-536" aria-hidden="true" tabindex="-1"></a><span class="ot">[^6]: </span>In order to use the cmdstanr backend you will need to first install the package (see <span class="in">`https://mc-stan.org/cmdstanr/`</span>) and also run <span class="in">`cmdstanr::install_cmdstan()`</span> if you have not done so already.</span>
<span id="cb41-537"><a href="#cb41-537" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-540"><a href="#cb41-540" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-541"><a href="#cb41-541" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-loadbrms</span></span>
<span id="cb41-542"><a href="#cb41-542" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Load the {brms} and {cmdstanr} packages</span></span>
<span id="cb41-543"><a href="#cb41-543" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-544"><a href="#cb41-544" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb41-545"><a href="#cb41-545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-546"><a href="#cb41-546" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(brms)</span>
<span id="cb41-547"><a href="#cb41-547" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cmdstanr)</span>
<span id="cb41-548"><a href="#cb41-548" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-549"><a href="#cb41-549" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-552"><a href="#cb41-552" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-553"><a href="#cb41-553" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fit-gauss</span></span>
<span id="cb41-554"><a href="#cb41-554" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-gauss</span></span>
<span id="cb41-555"><a href="#cb41-555" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting a gaussian model with brm().</span></span>
<span id="cb41-556"><a href="#cb41-556" aria-hidden="true" tabindex="-1"></a><span class="co">#| hide: false</span></span>
<span id="cb41-557"><a href="#cb41-557" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-558"><a href="#cb41-558" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-559"><a href="#cb41-559" aria-hidden="true" tabindex="-1"></a>bayes_reg_model <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb41-560"><a href="#cb41-560" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency,</span>
<span id="cb41-561"><a href="#cb41-561" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb41-562"><a href="#cb41-562" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">gaussian</span>(),</span>
<span id="cb41-563"><a href="#cb41-563" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"model_reg_bayes"</span>)</span>
<span id="cb41-564"><a href="#cb41-564" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-565"><a href="#cb41-565" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-566"><a href="#cb41-566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-569"><a href="#cb41-569" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-570"><a href="#cb41-570" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fit-gauss-parameters</span></span>
<span id="cb41-571"><a href="#cb41-571" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-572"><a href="#cb41-572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-573"><a href="#cb41-573" aria-hidden="true" tabindex="-1"></a>brms_model_new <span class="ot">&lt;-</span> <span class="fu">model_parameters</span>(bayes_reg_model, <span class="at">centrality =</span> <span class="st">"mean"</span>)</span>
<span id="cb41-574"><a href="#cb41-574" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-575"><a href="#cb41-575" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-576"><a href="#cb41-576" aria-hidden="true" tabindex="-1"></a>We fit the model using the <span class="in">`brm()`</span> function from the {brms} package (@lst-gauss). Although not shown here, we ran the models using four chains (the default), executed in parallel across four cores. When the model is run in @lst-gauss, the model summary output will appear in the R console. The output from <span class="in">`bayes_reg_model`</span> shows each parameter's posterior summary: The posterior distribution's mean and standard deviation (analogous to the frequentist standard error) and its 95% credible interval, which indicate the 95% of the most credible parameter values. In {brms}, the reported Cr.I is an equal-tailed interval, meaning that the probability mass excluded from the interval is split equally between the lower and upper tails. Additionally, the output indicates numerical estimates of the sampling algorithm's performance: Rhat should be close to one, and the ESS (effective sample size) metrics should be as large as possible given the number of iterations specified (default is 4000). Generally, ESS <span class="sc">\&gt;</span>= 1000 is recommended <span class="co">[</span><span class="ot">@brms</span><span class="co">]</span>. For the models we present in this paper, convergence is trivial with standard linear models, though we note that these metrics are still important to pay attention to in case of model misfit.</span>
<span id="cb41-577"><a href="#cb41-577" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-578"><a href="#cb41-578" aria-hidden="true" tabindex="-1"></a>Our main question of interest is: what is the causal effect of instructor fluency on final test performance? In order to answer this question, we will have to look at the output summary produced by @lst-gauss (also see @tbl-all under Bayesian LM). the <span class="in">`Intercept`</span> refers to the posterior mean accuracy in the disfluent condition, *M* = `r round(brms_model_new$Mean[1], 3)` , as fluency was dummy-coded. The fluency coefficient (`FluencyFluent`) reflects the mean posterior difference in recall accuracy between the fluent and disfluent conditions: *b* = <span class="in">`r round(brms_model_new$Mean[2], 3)`</span>. The 95% Cr.I for this estimate spans from <span class="in">`r round(brms_model_new$CI_low[2], 3)`</span> to <span class="in">`r round(brms_model_new$CI_high[2], 3)`</span>. These values are shown in the "95% Cr.I" columns of the output. These results closely mirror the findings reported by @wilford2020 (Experiment 1A).</span>
<span id="cb41-579"><a href="#cb41-579" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-582"><a href="#cb41-582" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-583"><a href="#cb41-583" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: show reg model</span></span>
<span id="cb41-584"><a href="#cb41-584" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(bayes_reg_model)</span>
<span id="cb41-585"><a href="#cb41-585" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-586"><a href="#cb41-586" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-587"><a href="#cb41-587" aria-hidden="true" tabindex="-1"></a>The output also includes the effective sample size (ESS) and R̂ (R-hat) values, both of which fall within acceptable ranges, indicating good model convergence. Throughout the tutorial, we focus primarily on posterior mean estimates and their 95% credible intervals. In addition, we report the pd measure in the main summary table (@tbl-all), provided by the {bayestestR} package <span class="co">[</span><span class="ot">@makowski2019a; @makowski2019</span><span class="co">]</span>. This measure offers an intuitive parallel to *p*-values, which many readers may find familiar. For example, the fluency effect has a pd of .977, indicating a high probability that the effect is positive rather than negative.</span>
<span id="cb41-588"><a href="#cb41-588" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-589"><a href="#cb41-589" aria-hidden="true" tabindex="-1"></a>Importantly, pd does not indicate whether an effect is meaningfully different from a point value—it only reflects the proportion of the posterior in one direction. To address questions of practical significance, readers can consider the Region of Practical Equivalence (ROPE) with the Cr.Is <span class="co">[</span><span class="ot">@kruschke2015doing</span><span class="co">]</span>. Unlike a hypothesis test of a point null (e.g., exactly zero), the ROPE defines a range of values that are deemed too small to be of substantive importance. As a rule of thumb <span class="co">[</span><span class="ot">see @kruschke2018</span><span class="co">]</span>, if more than 95% of the posterior distribution lies inside the ROPE, the effect can be considered practically equivalent to that negligible range. If less than 5% lies inside, the effect can be considered meaningfully different. Intermediate cases are typically labeled undecided.</span>
<span id="cb41-590"><a href="#cb41-590" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-591"><a href="#cb41-591" aria-hidden="true" tabindex="-1"></a>The <span class="in">`rope()`</span> function in the {bayestestR} package computes the proportion of the posterior within the ROPE to facilitate this evaluation. By default, from bayesian models fit via {brms}, the package determines a ROPE based on the data (roughly reflecting “negligible” effects), but these defaults should be used cautiously. The choice of ROPE should always be guided by theoretical considerations, previous research, and the substantive context of the study. In @lst-ropebrms, we show how to compute this using {bayestestR}. Running the function with default settings suggests that less than 5% lies within the default ROPE (indicating the effect is larger than .02) (see @fig-rope).</span>
<span id="cb41-592"><a href="#cb41-592" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-595"><a href="#cb41-595" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-596"><a href="#cb41-596" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: rope-brms</span></span>
<span id="cb41-597"><a href="#cb41-597" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-ropebrms</span></span>
<span id="cb41-598"><a href="#cb41-598" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Getting ROPE from `bayes_reg_model` obect using `rope` function from {bayestestR}</span></span>
<span id="cb41-599"><a href="#cb41-599" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-600"><a href="#cb41-600" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-601"><a href="#cb41-601" aria-hidden="true" tabindex="-1"></a>brms_rope <span class="ot">&lt;-</span> bayestestR<span class="sc">::</span><span class="fu">rope</span>(bayes_reg_model, <span class="at">ci =</span> .<span class="dv">95</span>, <span class="at">ci_method =</span> <span class="st">"ETI"</span>)</span>
<span id="cb41-602"><a href="#cb41-602" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-603"><a href="#cb41-603" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-606"><a href="#cb41-606" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-607"><a href="#cb41-607" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-rope</span></span>
<span id="cb41-608"><a href="#cb41-608" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Posterior distribution for the fluency effect showing the ROPE(shaded area) with 95% credible interval (orange) and 100% credible interval (blue). The percentage indicates the proportion of the posterior within the ROPE."</span></span>
<span id="cb41-609"><a href="#cb41-609" aria-hidden="true" tabindex="-1"></a><span class="co"># echo: false</span></span>
<span id="cb41-610"><a href="#cb41-610" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-611"><a href="#cb41-611" aria-hidden="true" tabindex="-1"></a>p <span class="ot">&lt;-</span> <span class="fu">plot</span>(brms_rope)</span>
<span id="cb41-612"><a href="#cb41-612" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-613"><a href="#cb41-613" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract the ROPE percentage</span></span>
<span id="cb41-614"><a href="#cb41-614" aria-hidden="true" tabindex="-1"></a>rope_percentage <span class="ot">&lt;-</span> <span class="fu">round</span>(brms_rope<span class="sc">$</span>ROPE_Percentage[<span class="dv">2</span>], <span class="dv">3</span>) <span class="sc">*</span> <span class="dv">100</span></span>
<span id="cb41-615"><a href="#cb41-615" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-616"><a href="#cb41-616" aria-hidden="true" tabindex="-1"></a><span class="co"># Add the percentage text to the plot</span></span>
<span id="cb41-617"><a href="#cb41-617" aria-hidden="true" tabindex="-1"></a>p_annotated <span class="ot">&lt;-</span> p <span class="sc">+</span></span>
<span id="cb41-618"><a href="#cb41-618" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(</span>
<span id="cb41-619"><a href="#cb41-619" aria-hidden="true" tabindex="-1"></a>    <span class="st">"text"</span>,</span>
<span id="cb41-620"><a href="#cb41-620" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="dv">0</span>, <span class="co"># Since your ROPE is -0.02 to 0.02, center is 0</span></span>
<span id="cb41-621"><a href="#cb41-621" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="cn">Inf</span>,</span>
<span id="cb41-622"><a href="#cb41-622" aria-hidden="true" tabindex="-1"></a>    <span class="at">label =</span> <span class="fu">paste0</span>(rope_percentage, <span class="st">"%"</span>),</span>
<span id="cb41-623"><a href="#cb41-623" aria-hidden="true" tabindex="-1"></a>    <span class="at">vjust =</span> <span class="fl">1.2</span>,</span>
<span id="cb41-624"><a href="#cb41-624" aria-hidden="true" tabindex="-1"></a>    <span class="at">hjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb41-625"><a href="#cb41-625" aria-hidden="true" tabindex="-1"></a>    <span class="at">size =</span> <span class="dv">4</span>,</span>
<span id="cb41-626"><a href="#cb41-626" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"black"</span>,</span>
<span id="cb41-627"><a href="#cb41-627" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontface =</span> <span class="st">"bold"</span></span>
<span id="cb41-628"><a href="#cb41-628" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-629"><a href="#cb41-629" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb41-630"><a href="#cb41-630" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_okabe_ito</span>() <span class="sc">+</span></span>
<span id="cb41-631"><a href="#cb41-631" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_okabe_ito</span>()</span>
<span id="cb41-632"><a href="#cb41-632" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-633"><a href="#cb41-633" aria-hidden="true" tabindex="-1"></a>p_annotated</span>
<span id="cb41-634"><a href="#cb41-634" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-635"><a href="#cb41-635" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-636"><a href="#cb41-636" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Assumptions</span></span>
<span id="cb41-637"><a href="#cb41-637" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-638"><a href="#cb41-638" aria-hidden="true" tabindex="-1"></a>@wilford2020 observed that instructor fluency impacts actual learning, using a standard *t*-test on the mean accuracy. But recall this approach assumes normality of residuals and homoskedasticity. These assumptions are unrealistic when the response values approach the scale boundaries <span class="co">[</span><span class="ot">@sladekova2024unicorns</span><span class="co">]</span>. Does the data we have meet those assumptions? We can use the function <span class="in">`check_model()`</span> from {easystats} <span class="co">[</span><span class="ot">@easystats</span><span class="co">]</span> to check our assumptions easily. The code in @lst-check1 produces @fig-reg-assump. We can see some issues with our model. Specifically, there appears to be violations of constant variance across the values of the scale (homoskedasticity). In plain terms, this type of model mis-specfication means that a standard OLS model can predict non-sensical values outside the bounds of the scale.</span>
<span id="cb41-639"><a href="#cb41-639" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-642"><a href="#cb41-642" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-643"><a href="#cb41-643" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-reg-assump</span></span>
<span id="cb41-644"><a href="#cb41-644" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-check1</span></span>
<span id="cb41-645"><a href="#cb41-645" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Checking assumptions with the `check_model()` from {easystats} package .</span></span>
<span id="cb41-646"><a href="#cb41-646" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Two assumption checks for our OLS model: Normality (right) and Homoskedasticity (left)"</span></span>
<span id="cb41-647"><a href="#cb41-647" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-648"><a href="#cb41-648" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-649"><a href="#cb41-649" aria-hidden="true" tabindex="-1"></a><span class="fu">check_model</span>(bayes_reg_model, <span class="at">check =</span> <span class="fu">c</span>(<span class="st">"homogeneity"</span>, <span class="st">"qq"</span>))</span>
<span id="cb41-650"><a href="#cb41-650" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-651"><a href="#cb41-651" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-652"><a href="#cb41-652" aria-hidden="true" tabindex="-1"></a>We can also examine how well the data fits the model by performing a posterior predictive check using the <span class="in">`pp_check()`</span> function from {brms}. A posterior predictive check involves looking at multiple model-predicted values and plotting them against the observed data. Ideally, the predicted values (the light blue lines) should show reasonable resemblance with the observed data (dark blue line). In our example (see @fig-ppcheckall (A)) the model-predicted density is slightly too peaked and narrow compared to the data. In addition, some of the predicted accuracy values are negative.</span>
<span id="cb41-653"><a href="#cb41-653" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-654"><a href="#cb41-654" aria-hidden="true" tabindex="-1"></a><span class="fu">## Distributional Regression - Beta</span></span>
<span id="cb41-655"><a href="#cb41-655" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-656"><a href="#cb41-656" aria-hidden="true" tabindex="-1"></a>It is important to note that there are several justifiable approaches for addressing the distributional issues observed in the data. For instance, one could analyze median accuracy instead of the mean, use some type of robust estimator for heterogeneity, or apply non-parametric methods to relax some of the model assumptions. Alternatively, we can address these issues directly by fitting distributional models <span class="co">[</span><span class="ot">@kruschke2013; @kneib2023</span><span class="co">]</span>. A key advantage of distributional models is that they are not limited to modeling only the mean or median of the outcome, but can also model parameters such as the variance (or other shape parameters) as functions of predictors. This allows examining how instructor fluency may influence not only average performance, but also the variability in performance across students. If we wanted to keep our mean accuracy variable and continue to use a Guassian model, we could use a distributional approach and model the effect of fluency on $\sigma$.</span>
<span id="cb41-657"><a href="#cb41-657" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-658"><a href="#cb41-658" aria-hidden="true" tabindex="-1"></a>Given the outcome variable is proportional, another solution would be to run a beta regression model. Again, we can create the beta regression model in {brms}. In {brms}, we model each parameter independently. Recall from the introduction that in a beta model we model two parameters--$\mu$ and $\phi$. Again we do this by using the <span class="in">`bf()`</span> function from {brms} (@lst-beta01). We specify two formulas, one for $\mu$ and one for $\phi$ and store it in the <span class="in">`model_beta_bayes`</span> object below. In the below <span class="in">`bf()`</span> call, we are modeling accuracy as a function of fluency only for the $\mu$ parameter. For the $\phi$ parameter, we are only modeling the intercept value. This is saying dispersion does not change as a function of fluency.</span>
<span id="cb41-659"><a href="#cb41-659" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-660"><a href="#cb41-660" aria-hidden="true" tabindex="-1"></a>To run our beta regression model, we need to exclude 0s and 1s in our data set. If we try to run a model with our data <span class="in">`data_fluency`</span> we get an error: <span class="in">`Error: Family 'beta' requires response greater than 0`</span>. This is because the beta distribution only supports observations in the 0 to 1 interval *excluding exact 0s and 1s*. We need make sure there are no 0s and 1s in our dataset.</span>
<span id="cb41-661"><a href="#cb41-661" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-662"><a href="#cb41-662" aria-hidden="true" tabindex="-1"></a>The dataset contains nine 0s and one 1. One approach is to nudge our 0s towards .01 and our 1s to .99, or apply a special formula <span class="co">[</span><span class="ot">@smithson2006</span><span class="co">]</span> so values fall within the <span class="sc">\[</span>0, 1<span class="sc">\]</span> interval. We implore readers not to engage in this practice. @kubinec2022 showed that this practice can result in serious distortion of the outcome as the sample size 𝑁 grows larger, resulting in ever smaller values that are “nudged”. Because the beta distribution is a non-linear model of the outcome, values that are very close to the boundary, such as 0.00001 or 0.99999, will be highly influential outliers. To run this beta model we will remove the 0s and 1s, and later in this article we will show how to jointly model these scale end points with the rest of the data. The model from @lst-beta01 uses a transformed <span class="in">`data_fluency`</span> object (called <span class="in">`data_beta`</span>) where 0s and 1s are removed. When we run this code we should not get an error.</span>
<span id="cb41-663"><a href="#cb41-663" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-666"><a href="#cb41-666" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-667"><a href="#cb41-667" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: beta_brms_01s</span></span>
<span id="cb41-668"><a href="#cb41-668" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-beta01</span></span>
<span id="cb41-669"><a href="#cb41-669" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting a beta model without 0s and 1s in brm().</span></span>
<span id="cb41-670"><a href="#cb41-670" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-671"><a href="#cb41-671" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-672"><a href="#cb41-672" aria-hidden="true" tabindex="-1"></a><span class="co"># set up model formual</span></span>
<span id="cb41-673"><a href="#cb41-673" aria-hidden="true" tabindex="-1"></a>model_beta_bayes <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb41-674"><a href="#cb41-674" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># fit mu model</span></span>
<span id="cb41-675"><a href="#cb41-675" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> <span class="dv">1</span> <span class="co"># fit phi model</span></span>
<span id="cb41-676"><a href="#cb41-676" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-677"><a href="#cb41-677" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-678"><a href="#cb41-678" aria-hidden="true" tabindex="-1"></a><span class="co"># transform 0 to 0.1 and 1 to .99</span></span>
<span id="cb41-679"><a href="#cb41-679" aria-hidden="true" tabindex="-1"></a>data_beta <span class="ot">&lt;-</span> fluency_data <span class="sc">|&gt;</span></span>
<span id="cb41-680"><a href="#cb41-680" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(</span>
<span id="cb41-681"><a href="#cb41-681" aria-hidden="true" tabindex="-1"></a>    Accuracy <span class="sc">!=</span> <span class="dv">0</span>,</span>
<span id="cb41-682"><a href="#cb41-682" aria-hidden="true" tabindex="-1"></a>    Accuracy <span class="sc">!=</span> <span class="dv">1</span></span>
<span id="cb41-683"><a href="#cb41-683" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-684"><a href="#cb41-684" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-685"><a href="#cb41-685" aria-hidden="true" tabindex="-1"></a>beta_brms <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb41-686"><a href="#cb41-686" aria-hidden="true" tabindex="-1"></a>  model_beta_bayes,</span>
<span id="cb41-687"><a href="#cb41-687" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_beta,</span>
<span id="cb41-688"><a href="#cb41-688" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">Beta</span>(),</span>
<span id="cb41-689"><a href="#cb41-689" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>,<span class="st">"models"</span>, <span class="st">"model_beta_bayes_reg_01"</span>)</span>
<span id="cb41-690"><a href="#cb41-690" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-691"><a href="#cb41-691" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-692"><a href="#cb41-692" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-693"><a href="#cb41-693" aria-hidden="true" tabindex="-1"></a><span class="fu">##### Model Parameters</span></span>
<span id="cb41-694"><a href="#cb41-694" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-695"><a href="#cb41-695" aria-hidden="true" tabindex="-1"></a>In @tbl-all, under the beta regression column, the coefficient with <span class="in">`b_`</span> represents how fluency of instructor influences the $\mu$ parameter estimates (which is the mean of the distribution here). These coefficients are linear on the logit-scale, but not on the raw accuracy scale. The intercept term <span class="in">`(b_Intercept)`</span> represents the log odds of the mean on accuracy for the fluent instructor. Log odds that are negative indicate that it is more likely a "success" (like getting the correct answer) will not happen than that it will happen. Similarly, regression coefficients in log odds forms that are negative indicate that an increase in that predictor leads to a decrease in the predicted probability of a "success".</span>
<span id="cb41-696"><a href="#cb41-696" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-697"><a href="#cb41-697" aria-hidden="true" tabindex="-1"></a>The other component we need to pay attention to is the dispersion or precision parameter coefficients labeled as <span class="in">`phi`</span> in @tbl-all. The dispersion ($\phi$) parameter tells us how precise our estimate is. Specifically, $\phi$ in beta regression tells us about the variability of the response variable around its mean. Specifically, a higher dispersion parameter indicates a narrower distribution, reflecting less variability. Conversely, a lower dispersion parameter suggests a wider distribution, reflecting greater variability. The main difference between a dispersion parameter and the variance is that the dispersion has a different interpretation depending on the value of the outcome, as we show below. The best way to understand dispersion is to examine visual changes in the distribution as the dispersion increases or decreases.</span>
<span id="cb41-698"><a href="#cb41-698" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-699"><a href="#cb41-699" aria-hidden="true" tabindex="-1"></a>Understanding the dispersion parameter helps us gauge the precision of our predictions and the consistency of the response variable. In <span class="in">`beta_brms`</span> we only modeled the dispersion of the intercept. When $\phi$ is not specified, the intercept is modeled by default (see @tbl-all). It represents the overall dispersion in the outcome across all conditions. Instead, we can model different dispersions across levels of the <span class="in">`Fluency`</span> factor. To do so, we add <span class="in">`Fluency`</span> to the phi model in <span class="in">`bf()`</span>. We model the precision (<span class="in">`phi`</span>) of the <span class="in">`Fluency`</span> factor by using a <span class="in">`~`</span> and adding factors of interest to the right of it (@lst-betadis).</span>
<span id="cb41-700"><a href="#cb41-700" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-703"><a href="#cb41-703" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-704"><a href="#cb41-704" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: brms-model-dis</span></span>
<span id="cb41-705"><a href="#cb41-705" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-betadis</span></span>
<span id="cb41-706"><a href="#cb41-706" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting beta model with dispersion.</span></span>
<span id="cb41-707"><a href="#cb41-707" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb41-708"><a href="#cb41-708" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-709"><a href="#cb41-709" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-710"><a href="#cb41-710" aria-hidden="true" tabindex="-1"></a>model_beta_bayes_disp <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb41-711"><a href="#cb41-711" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># Model of the mean</span></span>
<span id="cb41-712"><a href="#cb41-712" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency <span class="co"># Model of the precision</span></span>
<span id="cb41-713"><a href="#cb41-713" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-714"><a href="#cb41-714" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-715"><a href="#cb41-715" aria-hidden="true" tabindex="-1"></a>beta_brms_dis <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb41-716"><a href="#cb41-716" aria-hidden="true" tabindex="-1"></a>  model_beta_bayes_disp,</span>
<span id="cb41-717"><a href="#cb41-717" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_beta,</span>
<span id="cb41-718"><a href="#cb41-718" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">Beta</span>(),</span>
<span id="cb41-719"><a href="#cb41-719" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>,<span class="st">"models"</span>, <span class="st">"model_beta_bayes_dis_run01"</span>)</span>
<span id="cb41-720"><a href="#cb41-720" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-721"><a href="#cb41-721" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-722"><a href="#cb41-722" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-723"><a href="#cb41-723" aria-hidden="true" tabindex="-1"></a>@tbl-all displays the model summary with the precision parameter labeled as <span class="in">`phi_Fluency`</span>. Since $\phi$ is modeled on the log scale, the coefficients represent changes in log-$\phi$ rather than φ itself. To see the effect in the original units, we convert the values back by exponentiating. Thus, the effect of the Fluent condition can be understood by comparing the exponentiated predicted $\phi$ in the Fluent condition to that in the baseline condition.</span>
<span id="cb41-724"><a href="#cb41-724" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-725"><a href="#cb41-725" aria-hidden="true" tabindex="-1"></a>The $\phi$ parameters are estimated on the log scale. The term $\beta^{(\phi)}_{\text{Intercept}}$ represents the log-precision for the reference (disfluent) condition. The coefficient $\beta^{(\phi)}_{\text{FluencyFluent}}$ represents the change in log-precision when moving from the disfluent to the fluent condition.</span>
<span id="cb41-726"><a href="#cb41-726" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-727"><a href="#cb41-727" aria-hidden="true" tabindex="-1"></a>To obtain precision on the original scale, we exponentiate the linear predictor:</span>
<span id="cb41-728"><a href="#cb41-728" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-729"><a href="#cb41-729" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb41-730"><a href="#cb41-730" aria-hidden="true" tabindex="-1"></a>\phi_{\text{disfluent}} = \exp\big(\beta^{(\phi)}_{\text{Intercept}}\big), \qquad</span>
<span id="cb41-731"><a href="#cb41-731" aria-hidden="true" tabindex="-1"></a>\phi_{\text{fluent}} = \exp\big(\beta^{(\phi)}_{\text{Intercept}} + \beta^{(\phi)}_{\text{Fluency}}\big).</span>
<span id="cb41-732"><a href="#cb41-732" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb41-733"><a href="#cb41-733" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-734"><a href="#cb41-734" aria-hidden="true" tabindex="-1"></a>The coefficient $\beta^{(\phi)}_{\text{Fluency}}$ therefore describes a *multiplicative* change in precision. Specifically,</span>
<span id="cb41-735"><a href="#cb41-735" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-736"><a href="#cb41-736" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb41-737"><a href="#cb41-737" aria-hidden="true" tabindex="-1"></a>\frac{\phi_{\text{fluent}}}{\phi_{\text{disfluent}}} = \exp\big(\beta^{(\phi)}_{\text{Fluency}}\big).</span>
<span id="cb41-738"><a href="#cb41-738" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb41-739"><a href="#cb41-739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-740"><a href="#cb41-740" aria-hidden="true" tabindex="-1"></a>Because the 95% credible interval for $\beta^{(\phi)}_{\text{Fluency}}$ does not include zero, we infer that there is a credible difference in precision between the fluent and disfluent conditions.</span>
<span id="cb41-741"><a href="#cb41-741" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-742"><a href="#cb41-742" aria-hidden="true" tabindex="-1"></a>It is important to note that these estimates are not the same as the marginal effects we discussed earlier. Changes in dispersion affect the spread or variability of the response distribution without necessarily altering its mean. This makes dispersion particularly relevant for research questions that focus on features of the distribution beyond the average—such as how concentrated responses are. For instance, high dispersion might indicate that individuals cluster at the extremes (e.g., very high or very low ratings), suggesting clustering in the outcome.</span>
<span id="cb41-743"><a href="#cb41-743" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-744"><a href="#cb41-744" aria-hidden="true" tabindex="-1"></a>A critical assumption of the linear model is homoscedasticity, which means constant variance of the errors. With beta regression model we can include a dispersion parameter for Fluency. Properly accounting for dispersion is crucial because it impacts the precision of our mean estimates and, consequently, the substantive inferences we might make about the coefficients. The inclusion of dispersion in the model increased the uncertainty of the $\mu$ coefficient (see @fig-plt-diff). This highlights the potential utility of an approach like beta regression over a traditional approach as beta regression can explicitly model dispersion and address issues of heteroscedasticity.</span>
<span id="cb41-745"><a href="#cb41-745" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-746"><a href="#cb41-746" aria-hidden="true" tabindex="-1"></a>While it is advisable to model precision, if there is uncertainty about the best model, a relatively agnostic approach would be to compare models, for example with leave one out (loo) cross validation, to examine if a dispersion parameter should be considered in our model.<span class="ot">[^7]</span></span>
<span id="cb41-747"><a href="#cb41-747" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-748"><a href="#cb41-748" aria-hidden="true" tabindex="-1"></a><span class="ot">[^7]: </span>The model fit statistic LOO-CV can be compared for any set of fitted brms models with the function <span class="in">`loo()`</span>.</span>
<span id="cb41-749"><a href="#cb41-749" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-752"><a href="#cb41-752" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-753"><a href="#cb41-753" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-plt-diff</span></span>
<span id="cb41-754"><a href="#cb41-754" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Comparison of posterior distributions for the risk difference in fluency: Simple model  (no dispersion for Fluency) vs. complex model with dispersion"</span></span>
<span id="cb41-755"><a href="#cb41-755" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-756"><a href="#cb41-756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-757"><a href="#cb41-757" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a model identifier to each dataset</span></span>
<span id="cb41-758"><a href="#cb41-758" aria-hidden="true" tabindex="-1"></a>pred_no <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(beta_brms, <span class="at">variables =</span> <span class="st">"Fluency"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-759"><a href="#cb41-759" aria-hidden="true" tabindex="-1"></a>  <span class="fu">posterior_draws</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-760"><a href="#cb41-760" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">model =</span> <span class="st">"No Dispersion"</span>)</span>
<span id="cb41-761"><a href="#cb41-761" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-762"><a href="#cb41-762" aria-hidden="true" tabindex="-1"></a>pred_yes <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(beta_brms_dis, <span class="at">variables =</span> <span class="st">"Fluency"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-763"><a href="#cb41-763" aria-hidden="true" tabindex="-1"></a>  <span class="fu">posterior_draws</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-764"><a href="#cb41-764" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">model =</span> <span class="st">"Dispersion"</span>)</span>
<span id="cb41-765"><a href="#cb41-765" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-766"><a href="#cb41-766" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine both datasets</span></span>
<span id="cb41-767"><a href="#cb41-767" aria-hidden="true" tabindex="-1"></a>pred_combined <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(pred_no, pred_yes)</span>
<span id="cb41-768"><a href="#cb41-768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-769"><a href="#cb41-769" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pred_combined, <span class="fu">aes</span>(<span class="at">x =</span> draw, <span class="at">y =</span> model, <span class="at">fill =</span> model)) <span class="sc">+</span></span>
<span id="cb41-770"><a href="#cb41-770" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_halfeye</span>(<span class="at">.width =</span> <span class="fu">c</span>(.<span class="dv">8</span>, .<span class="dv">95</span>), <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb41-771"><a href="#cb41-771" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_okabe_ito</span>() <span class="sc">+</span></span>
<span id="cb41-772"><a href="#cb41-772" aria-hidden="true" tabindex="-1"></a>  <span class="co"># guides(fill = guide_legend(title = "Model")) +  # Remove or comment out</span></span>
<span id="cb41-773"><a href="#cb41-773" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb41-774"><a href="#cb41-774" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Difference in Final Recall Performance"</span>,</span>
<span id="cb41-775"><a href="#cb41-775" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Model"</span>,</span>
<span id="cb41-776"><a href="#cb41-776" aria-hidden="true" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">"80% and 95% credible intervals shown in black"</span></span>
<span id="cb41-777"><a href="#cb41-777" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-778"><a href="#cb41-778" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb41-779"><a href="#cb41-779" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">"solid"</span>, <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb41-780"><a href="#cb41-780" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb41-781"><a href="#cb41-781" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb41-782"><a href="#cb41-782" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-783"><a href="#cb41-783" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-786"><a href="#cb41-786" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-787"><a href="#cb41-787" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: beta-brms</span></span>
<span id="cb41-788"><a href="#cb41-788" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-789"><a href="#cb41-789" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb41-790"><a href="#cb41-790" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb41-791"><a href="#cb41-791" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-792"><a href="#cb41-792" aria-hidden="true" tabindex="-1"></a><span class="fu">model_parameters</span>(beta_brms, <span class="st">"mean"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-793"><a href="#cb41-793" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-794"><a href="#cb41-794" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb41-795"><a href="#cb41-795" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{CI_low}, {CI_high}]"</span>)</span>
<span id="cb41-796"><a href="#cb41-796" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb41-797"><a href="#cb41-797" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Parameter, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, pd) <span class="sc">|&gt;</span></span>
<span id="cb41-798"><a href="#cb41-798" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-799"><a href="#cb41-799" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-800"><a href="#cb41-800" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-801"><a href="#cb41-801" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-802"><a href="#cb41-802" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-805"><a href="#cb41-805" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-806"><a href="#cb41-806" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: beta-mu</span></span>
<span id="cb41-807"><a href="#cb41-807" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb41-808"><a href="#cb41-808" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-809"><a href="#cb41-809" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb41-810"><a href="#cb41-810" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-811"><a href="#cb41-811" aria-hidden="true" tabindex="-1"></a><span class="fu">model_parameters</span>(beta_brms, <span class="st">"mean"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-812"><a href="#cb41-812" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="sc">!</span><span class="fu">startsWith</span>(Parameter, <span class="st">"b_phi"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb41-813"><a href="#cb41-813" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-814"><a href="#cb41-814" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb41-815"><a href="#cb41-815" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{CI_low}, {CI_high}]"</span>)</span>
<span id="cb41-816"><a href="#cb41-816" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb41-817"><a href="#cb41-817" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Parameter, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, pd) <span class="sc">|&gt;</span></span>
<span id="cb41-818"><a href="#cb41-818" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-819"><a href="#cb41-819" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-820"><a href="#cb41-820" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-821"><a href="#cb41-821" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-822"><a href="#cb41-822" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-823"><a href="#cb41-823" aria-hidden="true" tabindex="-1"></a><span class="fu">### Predicted Probabilities</span></span>
<span id="cb41-824"><a href="#cb41-824" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-825"><a href="#cb41-825" aria-hidden="true" tabindex="-1"></a>Parameter estimates can be difficult to interpret, and researchers can instead discuss effects on the actual outcome scale (in this case the 0-1 scale). The logit link allows us to transform back and forth between the scale of a linear model and the nonlinear scale of the outcome, which is bounded by 0 and 1. By using the inverse of the logit, we can easily transform our linear coefficients to obtain average effects on the scale of the proportions or percentages, which is usually easier to interpret. In a simple case, we can do this manually, but when there are many factors in your model this can be quite complex.</span>
<span id="cb41-826"><a href="#cb41-826" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-827"><a href="#cb41-827" aria-hidden="true" tabindex="-1"></a>In our example, we can use the <span class="in">`plogis()`</span> function in base R to convert estimates from the logit scale to the probability scale. The intercept of our model is -0.918, which reflects the logit of the mean accuracy in the disfluent condition. If the estimated difference between the fluent and disfluent conditions is 0.24 on the logit scale, we first add this value to the intercept value (-0.918) to get the logit for the fluent condition: <span class="in">`-0.83 + 0.20 = -0.63`</span>. We then use <span class="in">`plogis()`</span> to convert both logit values to probabilities (Fluent = 35%, Disfluent = 30%).</span>
<span id="cb41-828"><a href="#cb41-828" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-829"><a href="#cb41-829" aria-hidden="true" tabindex="-1"></a>With single coefficients this calculation is trivial, but in more complex models with interactions, it can be quite cumbersome. To help us extract predictions from our model and visualize them we will use a package called {marginaleffects} <span class="co">[</span><span class="ot">@margeffects2024</span><span class="co">]</span> (see @lst-me). To get the proportions for each of our categorical predictors on the $\mu$ parameter we can use the function from the package called <span class="in">`avg_predictions()`</span>. These are displayed in @tbl-predict-prob. These probabilities match what we calculated above.</span>
<span id="cb41-830"><a href="#cb41-830" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-833"><a href="#cb41-833" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-834"><a href="#cb41-834" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-835"><a href="#cb41-835" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-me</span></span>
<span id="cb41-836"><a href="#cb41-836" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Load the {marginaleffects} package.</span></span>
<span id="cb41-837"><a href="#cb41-837" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-838"><a href="#cb41-838" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(marginaleffects)</span>
<span id="cb41-839"><a href="#cb41-839" aria-hidden="true" tabindex="-1"></a><span class="fu">options</span>(<span class="at">marginaleffects_posterior_center =</span> mean) <span class="co"># make sure returns mean</span></span>
<span id="cb41-840"><a href="#cb41-840" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-841"><a href="#cb41-841" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-844"><a href="#cb41-844" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-845"><a href="#cb41-845" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-pred</span></span>
<span id="cb41-846"><a href="#cb41-846" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Predictions from the beta model for each level of Fluency ($\mu$).</span></span>
<span id="cb41-847"><a href="#cb41-847" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb41-848"><a href="#cb41-848" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-849"><a href="#cb41-849" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-850"><a href="#cb41-850" aria-hidden="true" tabindex="-1"></a><span class="fu">avg_predictions</span>(</span>
<span id="cb41-851"><a href="#cb41-851" aria-hidden="true" tabindex="-1"></a>  beta_brms_dis,</span>
<span id="cb41-852"><a href="#cb41-852" aria-hidden="true" tabindex="-1"></a>  <span class="co"># need to specify the levels of the categorical predictor</span></span>
<span id="cb41-853"><a href="#cb41-853" aria-hidden="true" tabindex="-1"></a>  <span class="at">variables =</span> <span class="st">"Fluency"</span></span>
<span id="cb41-854"><a href="#cb41-854" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-855"><a href="#cb41-855" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-856"><a href="#cb41-856" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-859"><a href="#cb41-859" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-860"><a href="#cb41-860" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-predict-prob</span></span>
<span id="cb41-861"><a href="#cb41-861" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Predicted probabilities for fluency factor ($\mu$).</span></span>
<span id="cb41-862"><a href="#cb41-862" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap-location: top</span></span>
<span id="cb41-863"><a href="#cb41-863" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-864"><a href="#cb41-864" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb41-865"><a href="#cb41-865" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-866"><a href="#cb41-866" aria-hidden="true" tabindex="-1"></a><span class="fu">avg_predictions</span>(</span>
<span id="cb41-867"><a href="#cb41-867" aria-hidden="true" tabindex="-1"></a>  beta_brms_dis,</span>
<span id="cb41-868"><a href="#cb41-868" aria-hidden="true" tabindex="-1"></a>  <span class="at">variables =</span> <span class="st">"Fluency"</span></span>
<span id="cb41-869"><a href="#cb41-869" aria-hidden="true" tabindex="-1"></a>) <span class="sc">|&gt;</span></span>
<span id="cb41-870"><a href="#cb41-870" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Fluency =</span> <span class="fu">ifelse</span>(Fluency <span class="sc">==</span> <span class="st">"Fluent"</span>, <span class="st">"Fluent"</span>, <span class="st">"Disfluent"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb41-871"><a href="#cb41-871" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-872"><a href="#cb41-872" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb41-873"><a href="#cb41-873" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{conf.low}, {conf.high}]"</span>)</span>
<span id="cb41-874"><a href="#cb41-874" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb41-875"><a href="#cb41-875" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"estimate"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-876"><a href="#cb41-876" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Fluency, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-877"><a href="#cb41-877" aria-hidden="true" tabindex="-1"></a>  <span class="fu">relocate</span>(Fluency) <span class="sc">|&gt;</span></span>
<span id="cb41-878"><a href="#cb41-878" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-879"><a href="#cb41-879" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-880"><a href="#cb41-880" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-881"><a href="#cb41-881" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-882"><a href="#cb41-882" aria-hidden="true" tabindex="-1"></a>For the <span class="in">`Fluency`</span> factor, we can interpret the estimates under the <span class="in">`Mean`</span> column as a proportion or percentage. That is, participants who watched the fluent instructor scored on average 35% on the final exam compared to 30% for those who watched the disfluent instructor. We can also visualize these from {marginaleffects} using the <span class="in">`plot_predictions()`</span> function (see @lst-plotbeta).</span>
<span id="cb41-883"><a href="#cb41-883" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-886"><a href="#cb41-886" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-887"><a href="#cb41-887" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: plot beta</span></span>
<span id="cb41-888"><a href="#cb41-888" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-plotbeta</span></span>
<span id="cb41-889"><a href="#cb41-889" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Plot predicted probabilities using `plot_predictions()` from {marginaleffects}</span></span>
<span id="cb41-890"><a href="#cb41-890" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-891"><a href="#cb41-891" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-892"><a href="#cb41-892" aria-hidden="true" tabindex="-1"></a>beta_plot <span class="ot">&lt;-</span> <span class="fu">plot_predictions</span>(beta_brms_dis, <span class="at">by =</span> <span class="st">"Fluency"</span>)</span>
<span id="cb41-893"><a href="#cb41-893" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-894"><a href="#cb41-894" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-895"><a href="#cb41-895" aria-hidden="true" tabindex="-1"></a>The <span class="in">`plot_predictions()`</span> function will only display the point estimate with the 95% credible interval. However, Bayesian estimation methods generate distributions for each parameter. This approach allows visualizing full uncertainty estimates beyond points and intervals. Using the {marginaleffects} package, we can obtain samples from the posterior distribution with the <span class="in">`posterior_draws()`</span> function (see @lst-posdraws). We can then plot these results to illustrate the range of plausible values for our estimates at different levels of uncertainty (see @fig-draws-fluency).</span>
<span id="cb41-896"><a href="#cb41-896" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-899"><a href="#cb41-899" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-900"><a href="#cb41-900" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-posdraws</span></span>
<span id="cb41-901"><a href="#cb41-901" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Extracting posterior draws from the beta regression model.</span></span>
<span id="cb41-902"><a href="#cb41-902" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-903"><a href="#cb41-903" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-904"><a href="#cb41-904" aria-hidden="true" tabindex="-1"></a><span class="co"># Add a model identifier to each dataset</span></span>
<span id="cb41-905"><a href="#cb41-905" aria-hidden="true" tabindex="-1"></a>pred_draws_beta <span class="ot">&lt;-</span> <span class="fu">avg_predictions</span>(beta_brms_dis, <span class="at">variables =</span> <span class="st">"Fluency"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-906"><a href="#cb41-906" aria-hidden="true" tabindex="-1"></a>  <span class="fu">posterior_draws</span>()</span>
<span id="cb41-907"><a href="#cb41-907" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-908"><a href="#cb41-908" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-911"><a href="#cb41-911" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-912"><a href="#cb41-912" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-draws-fluency</span></span>
<span id="cb41-913"><a href="#cb41-913" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Predicted probablity posterior distributions by instructor fluency</span></span>
<span id="cb41-914"><a href="#cb41-914" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot both models</span></span>
<span id="cb41-915"><a href="#cb41-915" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(pred_draws_beta, <span class="fu">aes</span>(<span class="at">x =</span> draw, <span class="at">y =</span> Fluency, <span class="at">fill =</span> Fluency)) <span class="sc">+</span></span>
<span id="cb41-916"><a href="#cb41-916" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_halfeye</span>(<span class="at">.width =</span> <span class="fu">c</span>(.<span class="dv">5</span>, .<span class="dv">8</span>, .<span class="dv">95</span>), <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb41-917"><a href="#cb41-917" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_okabe_ito</span>() <span class="sc">+</span></span>
<span id="cb41-918"><a href="#cb41-918" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">label_percent</span>()) <span class="sc">+</span></span>
<span id="cb41-919"><a href="#cb41-919" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb41-920"><a href="#cb41-920" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Accuracy"</span>,</span>
<span id="cb41-921"><a href="#cb41-921" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb41-922"><a href="#cb41-922" aria-hidden="true" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">"50%, 80% and 95% credible intervals shown in black"</span></span>
<span id="cb41-923"><a href="#cb41-923" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-924"><a href="#cb41-924" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">"solid"</span>, <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb41-925"><a href="#cb41-925" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb41-926"><a href="#cb41-926" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb41-927"><a href="#cb41-927" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb41-928"><a href="#cb41-928" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-929"><a href="#cb41-929" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-930"><a href="#cb41-930" aria-hidden="true" tabindex="-1"></a><span class="fu">### Marginal Effects</span></span>
<span id="cb41-931"><a href="#cb41-931" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-932"><a href="#cb41-932" aria-hidden="true" tabindex="-1"></a>Marginal effects offer an interpretable way to quantify how changes in a predictor influence an outcome, while holding other factors constant in a specific manner. In recent years, there has been a thrust to move away from reporting regression coefficients alone, focusing instead on estimates that are easier to interpret and communicate—particularly in non-linear models <span class="co">[</span><span class="ot">@mccabe2021; @rohrer2025models</span><span class="co">]</span>. Technically, marginal effects are computed as partial derivatives for continuous variables or as finite differences for categorical (and sometimes continuous) predictors, depending on the structure of the data and the research question. Substantively, these procedures translate raw regression coefficients into quantities that reflect changes in the bounded outcome—for example, an $x$% change in the value of a proportion.</span>
<span id="cb41-933"><a href="#cb41-933" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-934"><a href="#cb41-934" aria-hidden="true" tabindex="-1"></a>There are several types of marginal effects, and their computation can vary across software packages. For example, the popular {emmeans} package <span class="co">[</span><span class="ot">@emmeans</span><span class="co">]</span> computes marginal effects by holding all predictors at their means (marginal effects at the mean; MEM). In this tutorial, we use the {marginaleffects} package <span class="co">[</span><span class="ot">@margeffects2024</span><span class="co">]</span>, which by default computes average marginal effects (AMEs). AMEs are based on counterfactual predictions: the dataset is conceptually replicated across all unique values of the predictor of interest, predictions are generated for each row under each counterfactual scenario, and the resulting differences are then averaged. This approach maintains a strong connection to the observed data—because predictions are made using each participant’s actual values—while providing a clear and interpretable summary of the effect of interest.</span>
<span id="cb41-935"><a href="#cb41-935" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-936"><a href="#cb41-936" aria-hidden="true" tabindex="-1"></a>One practical use of AMEs is to estimate the average difference between two groups or conditions which corresponds to the average treatment effect (ATE). Using the <span class="in">`avg_comparisons()`</span> function in the {marginaleffects}package (@lst-avgcomp), we can compute this quantity directly. By default, the function returns the discrete difference between groups. When we take the difference in proportions between two groups it is called the risk difference. Depending on the audience and modeling goals, the function can also produce alternative effect size metrics, such as odds ratios or risk ratios. This flexibility makes it a powerful approach for summarizing and communicating regression results.</span>
<span id="cb41-937"><a href="#cb41-937" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-940"><a href="#cb41-940" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-941"><a href="#cb41-941" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ame1</span></span>
<span id="cb41-942"><a href="#cb41-942" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-avgcomp</span></span>
<span id="cb41-943"><a href="#cb41-943" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Calculating the difference between probabilities ($\mu$) for fluency with `avg_comparisons()`</span></span>
<span id="cb41-944"><a href="#cb41-944" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-945"><a href="#cb41-945" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-946"><a href="#cb41-946" aria-hidden="true" tabindex="-1"></a><span class="co"># get risk difference by default</span></span>
<span id="cb41-947"><a href="#cb41-947" aria-hidden="true" tabindex="-1"></a>beta_avg_comp <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(</span>
<span id="cb41-948"><a href="#cb41-948" aria-hidden="true" tabindex="-1"></a>  beta_brms_dis,</span>
<span id="cb41-949"><a href="#cb41-949" aria-hidden="true" tabindex="-1"></a>  <span class="at">variables =</span> <span class="st">"Fluency"</span>, <span class="co"># counterfactural difference</span></span>
<span id="cb41-950"><a href="#cb41-950" aria-hidden="true" tabindex="-1"></a>  <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb41-951"><a href="#cb41-951" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-952"><a href="#cb41-952" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-953"><a href="#cb41-953" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-956"><a href="#cb41-956" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-957"><a href="#cb41-957" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-ame1</span></span>
<span id="cb41-958"><a href="#cb41-958" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-959"><a href="#cb41-959" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Fluency difference ($\mu$)</span></span>
<span id="cb41-960"><a href="#cb41-960" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb41-961"><a href="#cb41-961" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-962"><a href="#cb41-962" aria-hidden="true" tabindex="-1"></a>p_d <span class="ot">&lt;-</span> beta_avg_comp <span class="sc">|&gt;</span></span>
<span id="cb41-963"><a href="#cb41-963" aria-hidden="true" tabindex="-1"></a>  <span class="fu">p_direction</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-964"><a href="#cb41-964" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-965"><a href="#cb41-965" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pd)</span>
<span id="cb41-966"><a href="#cb41-966" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-967"><a href="#cb41-967" aria-hidden="true" tabindex="-1"></a>beta_avg_comp <span class="sc">|&gt;</span></span>
<span id="cb41-968"><a href="#cb41-968" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-969"><a href="#cb41-969" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb41-970"><a href="#cb41-970" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{conf.low}, {conf.high}]"</span>)</span>
<span id="cb41-971"><a href="#cb41-971" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb41-972"><a href="#cb41-972" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Contrast"</span> <span class="ot">=</span> <span class="st">"contrast"</span>, <span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"estimate"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-973"><a href="#cb41-973" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(p_d) <span class="sc">|&gt;</span></span>
<span id="cb41-974"><a href="#cb41-974" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Contrast, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, pd) <span class="sc">|&gt;</span></span>
<span id="cb41-975"><a href="#cb41-975" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>(<span class="at">digits =</span> <span class="dv">2</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-976"><a href="#cb41-976" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-977"><a href="#cb41-977" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-978"><a href="#cb41-978" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-979"><a href="#cb41-979" aria-hidden="true" tabindex="-1"></a><span class="fu">#### $\mu$ Difference</span></span>
<span id="cb41-980"><a href="#cb41-980" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-981"><a href="#cb41-981" aria-hidden="true" tabindex="-1"></a>@tbl-ame1 presents the estimated difference for the <span class="in">`Fluency`</span> factor (Mean column). The difference between the fluent and disfluent conditions is 0.06, indicating that participants who watched a fluent instructor scored, on average, 6% higher on the final recall test than those who watched a disfluent instructor. However, the 95% credible interval includes 0 among the most credible values, suggesting we cannot rule out the possibility of a null or weakly negative effect. The pd also indicates the effect is non-significant (*p* <span class="sc">\&gt;</span> .05). In the basic regression model above, the ROPE value was extracted using the {bayestestR} package.</span>
<span id="cb41-982"><a href="#cb41-982" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-985"><a href="#cb41-985" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-986"><a href="#cb41-986" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-merope</span></span>
<span id="cb41-987"><a href="#cb41-987" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Extracting ROPE using {marginaleffects} for $\mu$ difference</span></span>
<span id="cb41-988"><a href="#cb41-988" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true </span></span>
<span id="cb41-989"><a href="#cb41-989" aria-hidden="true" tabindex="-1"></a><span class="co"># get risk difference by default</span></span>
<span id="cb41-990"><a href="#cb41-990" aria-hidden="true" tabindex="-1"></a>beta_avg_comp_rope <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(</span>
<span id="cb41-991"><a href="#cb41-991" aria-hidden="true" tabindex="-1"></a>  beta_brms_dis,</span>
<span id="cb41-992"><a href="#cb41-992" aria-hidden="true" tabindex="-1"></a>  <span class="at">variables =</span> <span class="st">"Fluency"</span>, <span class="co"># counterfactural difference</span></span>
<span id="cb41-993"><a href="#cb41-993" aria-hidden="true" tabindex="-1"></a>  <span class="at">comparison =</span> <span class="st">"difference"</span>,</span>
<span id="cb41-994"><a href="#cb41-994" aria-hidden="true" tabindex="-1"></a>  <span class="at">equivalence=</span><span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.02</span>, .<span class="dv">02</span>)</span>
<span id="cb41-995"><a href="#cb41-995" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-996"><a href="#cb41-996" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-997"><a href="#cb41-997" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-998"><a href="#cb41-998" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-999"><a href="#cb41-999" aria-hidden="true" tabindex="-1"></a>,The ROPE can also be obtained using the {marginaleffects} package by specifying the <span class="in">`equivalence`</span> bounds in <span class="in">`avg_comparisons()`</span>. Running the code in @lst-merope returns a ROPE probability of 23%, indicating that there is insufficient evidence to conclude that the effect exceeds 2% or is practically equivalent.</span>
<span id="cb41-1000"><a href="#cb41-1000" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1001"><a href="#cb41-1001" aria-hidden="true" tabindex="-1"></a><span class="fu">#### $\phi$ difference</span></span>
<span id="cb41-1002"><a href="#cb41-1002" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1003"><a href="#cb41-1003" aria-hidden="true" tabindex="-1"></a>In addition, with {marginaleffects}, we can get the actual precision difference between the two groups on $\phi$ using similar code to above by setting <span class="in">`dpar`</span> to "phi" (@lst-avgphi).</span>
<span id="cb41-1004"><a href="#cb41-1004" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1007"><a href="#cb41-1007" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1008"><a href="#cb41-1008" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: ame-phi</span></span>
<span id="cb41-1009"><a href="#cb41-1009" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-avgphi</span></span>
<span id="cb41-1010"><a href="#cb41-1010" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Calculating $\phi$ difference  with `avg_comparisons()`</span></span>
<span id="cb41-1011"><a href="#cb41-1011" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-1012"><a href="#cb41-1012" aria-hidden="true" tabindex="-1"></a><span class="co"># get risk difference by default</span></span>
<span id="cb41-1013"><a href="#cb41-1013" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1014"><a href="#cb41-1014" aria-hidden="true" tabindex="-1"></a>beta_avg_phi <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(</span>
<span id="cb41-1015"><a href="#cb41-1015" aria-hidden="true" tabindex="-1"></a>  beta_brms_dis,</span>
<span id="cb41-1016"><a href="#cb41-1016" aria-hidden="true" tabindex="-1"></a>  <span class="at">variables =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb41-1017"><a href="#cb41-1017" aria-hidden="true" tabindex="-1"></a>  <span class="at">dpar =</span> <span class="st">"phi"</span>,</span>
<span id="cb41-1018"><a href="#cb41-1018" aria-hidden="true" tabindex="-1"></a>  <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb41-1019"><a href="#cb41-1019" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-1020"><a href="#cb41-1020" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1021"><a href="#cb41-1021" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1022"><a href="#cb41-1022" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1025"><a href="#cb41-1025" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1026"><a href="#cb41-1026" aria-hidden="true" tabindex="-1"></a><span class="co"># Proportions</span></span>
<span id="cb41-1027"><a href="#cb41-1027" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fl">0.35</span> <span class="co"># Fluent</span></span>
<span id="cb41-1028"><a href="#cb41-1028" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> <span class="fl">0.30</span> <span class="co"># Disfluent</span></span>
<span id="cb41-1029"><a href="#cb41-1029" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1030"><a href="#cb41-1030" aria-hidden="true" tabindex="-1"></a><span class="co"># Cohen's h formula</span></span>
<span id="cb41-1031"><a href="#cb41-1031" aria-hidden="true" tabindex="-1"></a>cohens_h <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> (<span class="fu">asin</span>(<span class="fu">sqrt</span>(p1)) <span class="sc">-</span> <span class="fu">asin</span>(<span class="fu">sqrt</span>(p2)))</span>
<span id="cb41-1032"><a href="#cb41-1032" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1033"><a href="#cb41-1033" aria-hidden="true" tabindex="-1"></a><span class="co"># CI lower and upper bounds</span></span>
<span id="cb41-1034"><a href="#cb41-1034" aria-hidden="true" tabindex="-1"></a>p1_lower <span class="ot">&lt;-</span> <span class="fl">0.328</span></span>
<span id="cb41-1035"><a href="#cb41-1035" aria-hidden="true" tabindex="-1"></a>p1_upper <span class="ot">&lt;-</span> <span class="fl">0.419</span></span>
<span id="cb41-1036"><a href="#cb41-1036" aria-hidden="true" tabindex="-1"></a>p2_lower <span class="ot">&lt;-</span> <span class="fl">0.250</span></span>
<span id="cb41-1037"><a href="#cb41-1037" aria-hidden="true" tabindex="-1"></a>p2_upper <span class="ot">&lt;-</span> <span class="fl">0.329</span></span>
<span id="cb41-1038"><a href="#cb41-1038" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1039"><a href="#cb41-1039" aria-hidden="true" tabindex="-1"></a><span class="co"># Lower bound of h</span></span>
<span id="cb41-1040"><a href="#cb41-1040" aria-hidden="true" tabindex="-1"></a>h_lower <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> (<span class="fu">asin</span>(<span class="fu">sqrt</span>(p1_lower)) <span class="sc">-</span> <span class="fu">asin</span>(<span class="fu">sqrt</span>(p2_upper)))</span>
<span id="cb41-1041"><a href="#cb41-1041" aria-hidden="true" tabindex="-1"></a><span class="co"># Upper bound of h</span></span>
<span id="cb41-1042"><a href="#cb41-1042" aria-hidden="true" tabindex="-1"></a>h_upper <span class="ot">&lt;-</span> <span class="dv">2</span> <span class="sc">*</span> (<span class="fu">asin</span>(<span class="fu">sqrt</span>(p1_upper)) <span class="sc">-</span> <span class="fu">asin</span>(<span class="fu">sqrt</span>(p2_lower)))</span>
<span id="cb41-1043"><a href="#cb41-1043" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1044"><a href="#cb41-1044" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1045"><a href="#cb41-1045" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Effect Size</span></span>
<span id="cb41-1046"><a href="#cb41-1046" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1047"><a href="#cb41-1047" aria-hidden="true" tabindex="-1"></a>In psychology, it is common to report effect size measures like Cohen's *d* [@cohen1977]. When working with proportions we can calculate something similar called Cohen's *h*. Taking our proportions, we can use the below equation (@eq-cohenh) to calculate Cohen's *h* along with the 95% Cr.I around it. Using this metric we see the effect size is small (<span class="in">`r round(cohens_h, 3)`</span>), 95% credible interval <span class="sc">\[</span><span class="in">`r round(h_lower,3)`</span>, <span class="in">`r round(h_upper,3)`</span><span class="sc">\]</span>.</span>
<span id="cb41-1048"><a href="#cb41-1048" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1049"><a href="#cb41-1049" aria-hidden="true" tabindex="-1"></a>$$</span>
<span id="cb41-1050"><a href="#cb41-1050" aria-hidden="true" tabindex="-1"></a>h = 2 \cdot \left( \arcsin\left(\sqrt{p_1}\right) - \arcsin\left(\sqrt{p_2}\right) \right)</span>
<span id="cb41-1051"><a href="#cb41-1051" aria-hidden="true" tabindex="-1"></a>$$ {#eq-cohenh}</span>
<span id="cb41-1052"><a href="#cb41-1052" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1053"><a href="#cb41-1053" aria-hidden="true" tabindex="-1"></a><span class="fu">### Posterior Predictive Check</span></span>
<span id="cb41-1054"><a href="#cb41-1054" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1055"><a href="#cb41-1055" aria-hidden="true" tabindex="-1"></a>@fig-ppcheckall (B) shows the predictive check for our beta model. The model's predictions generally conform to the data as the predictions are now between constrained to the 0-1 interval. However, the model's predictive performance can be improved the bounds of the scale are explicitly specified.</span>
<span id="cb41-1056"><a href="#cb41-1056" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1057"><a href="#cb41-1057" aria-hidden="true" tabindex="-1"></a><span class="fu">## Zero-Inflated beta (ZIB) Regression</span></span>
<span id="cb41-1058"><a href="#cb41-1058" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1059"><a href="#cb41-1059" aria-hidden="true" tabindex="-1"></a>A limitation of the beta regression model is that it can only accommodate values strictly between 0 and 1—a probability cannot take on values of 0 (the event will not occur with certainty) or 1 (the event will occur with certainty). In our dataset, we observed 9 rows where Accuracy equals zero. To fit a beta regression model, we removed these values, but we have left out potentially valuable information from our model--especially if the end points of the scale are distinctive in some way. In our case, these 0s may be structural—that is, they represent real, systematic instances where participants failed to answer correctly (rather than random noise or measurement error). For example, the fluency of the instructor might be a key factor in predicting these zero responses. We will discuss two approaches for jointly modeling these end points with the continuous data. First, we can use a zero-inflated beta (ZIB) model. This model still estimates the mean ($\mu$) and precision ($\phi$) of the beta distribution for values between 0 and 1, but it also includes an additional parameter, $\alpha$, which captures the probability of observing structural 0s.</span>
<span id="cb41-1060"><a href="#cb41-1060" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1061"><a href="#cb41-1061" aria-hidden="true" tabindex="-1"></a>The zero-inflated beta models a mixture of the data-generating process. The $\alpha$ parameter uses a logistic regression to model whether the data is 0 or not. Substantively, this could be a useful model when we think that 0s come from a process that is relatively distinct from the data that is greater than 0. For example, if we had a dataset with proportion of looks or eye fixations to certain areas on marketing materials, we might want a separate model for those that do not look at certain areas on the screen because individuals who do not look might be substantively different than those that look.</span>
<span id="cb41-1062"><a href="#cb41-1062" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1063"><a href="#cb41-1063" aria-hidden="true" tabindex="-1"></a>We can fit a ZIB model using <span class="in">`brms()`</span> and use the {marginaleffects} package to make inferences about our parameters of interest. Before we run a zero-inflated beta model, we will need to transform our data again and remove the one 1 value in our data--we can keep our 0s. Similar to our beta regression model we fit in <span class="in">`brms`</span>, we will use the <span class="in">`bf()`</span> function to fit several models. We fit our $\mu$ and $\phi$ parameters as well as our zero-inflated parameter ($\alpha$; here labeled as <span class="in">`zi`</span>). In <span class="in">`brms`</span> we can use the zero_inflated_beta family (see @lst-zib).</span>
<span id="cb41-1064"><a href="#cb41-1064" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1067"><a href="#cb41-1067" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1068"><a href="#cb41-1068" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: zib brms model</span></span>
<span id="cb41-1069"><a href="#cb41-1069" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-zib</span></span>
<span id="cb41-1070"><a href="#cb41-1070" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting zib model with `brm()`</span></span>
<span id="cb41-1071"><a href="#cb41-1071" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-1072"><a href="#cb41-1072" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1073"><a href="#cb41-1073" aria-hidden="true" tabindex="-1"></a><span class="co"># keep 0 but remove 1</span></span>
<span id="cb41-1074"><a href="#cb41-1074" aria-hidden="true" tabindex="-1"></a>data_beta_0 <span class="ot">&lt;-</span> fluency_data <span class="sc">|&gt;</span></span>
<span id="cb41-1075"><a href="#cb41-1075" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Accuracy <span class="sc">!=</span> <span class="dv">1</span>)</span>
<span id="cb41-1076"><a href="#cb41-1076" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1077"><a href="#cb41-1077" aria-hidden="true" tabindex="-1"></a><span class="co"># set up model formual for zero-inflated beta in brm</span></span>
<span id="cb41-1078"><a href="#cb41-1078" aria-hidden="true" tabindex="-1"></a>zib_model <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb41-1079"><a href="#cb41-1079" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># The mean of the 0-1 values, or mu</span></span>
<span id="cb41-1080"><a href="#cb41-1080" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency, <span class="co"># The precision of the 0-1 values, or phi</span></span>
<span id="cb41-1081"><a href="#cb41-1081" aria-hidden="true" tabindex="-1"></a>  zi <span class="sc">~</span> Fluency, <span class="co"># The zero-or-one-inflated part, or alpha</span></span>
<span id="cb41-1082"><a href="#cb41-1082" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">zero_inflated_beta</span>()</span>
<span id="cb41-1083"><a href="#cb41-1083" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-1084"><a href="#cb41-1084" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1085"><a href="#cb41-1085" aria-hidden="true" tabindex="-1"></a><span class="co"># fit zib model with brm</span></span>
<span id="cb41-1086"><a href="#cb41-1086" aria-hidden="true" tabindex="-1"></a>fit_zi <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb41-1087"><a href="#cb41-1087" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> zib_model,</span>
<span id="cb41-1088"><a href="#cb41-1088" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> data_beta_0,</span>
<span id="cb41-1089"><a href="#cb41-1089" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>,<span class="st">"models"</span>, <span class="st">"bayes_zib_model0not1.rds"</span>)</span>
<span id="cb41-1090"><a href="#cb41-1090" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-1091"><a href="#cb41-1091" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1092"><a href="#cb41-1092" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1093"><a href="#cb41-1093" aria-hidden="true" tabindex="-1"></a><span class="fu">### Posterior Predictive Check</span></span>
<span id="cb41-1094"><a href="#cb41-1094" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1095"><a href="#cb41-1095" aria-hidden="true" tabindex="-1"></a>The ZIB model does a bit better at capturing the structure of the data then the beta regression model (see @fig-ppcheckall). Specifically, the ZIB model more accurately captures the increased density of values near the lower end of the scale (i.e., near zero), which the standard beta model underestimates. The ZIB model’s predictive distributions also align more closely with the observed data across the entire range, particularly in the peak and tail regions. This improved fit likely reflects the ZIB model’s ability to explicitly model excess 0s (or near-zero values) via its inflation component, allowing it to better account for features in the data that a standard beta distribution cannot accommodate.</span>
<span id="cb41-1096"><a href="#cb41-1096" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1097"><a href="#cb41-1097" aria-hidden="true" tabindex="-1"></a><span class="fu">### Predicted Probabilities and Marginal Effects</span></span>
<span id="cb41-1098"><a href="#cb41-1098" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1099"><a href="#cb41-1099" aria-hidden="true" tabindex="-1"></a>@tbl-all, under the zero-inflated beta regression column, provides a summary of the posterior distribution for each parameter. As stated before, it is preferable to back-transform our estimates to get probabilities. To get the predicted probabilities we can again use the <span class="in">`avg_predictions()`</span> and <span class="in">`avg_comparisons()`</span> functions from {marginaleffects} package <span class="co">[</span><span class="ot">@marginaleffects</span><span class="co">]</span> to get predicted probabilities and the probability difference between the levels of each factor. We can model the parameters separately using the <span class="in">`dpar`</span> argument setting to: $\mu$, $\phi$, $\alpha$. Here we look at the risk difference for Fluency under each parameter. If one were interested in the average effect for the entire model, the <span class="in">`dpar`</span> argument could be removed.</span>
<span id="cb41-1100"><a href="#cb41-1100" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1101"><a href="#cb41-1101" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Mu</span></span>
<span id="cb41-1102"><a href="#cb41-1102" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1103"><a href="#cb41-1103" aria-hidden="true" tabindex="-1"></a>As shown in @tbl-marg-mu-brms, there is little evidence for an effect of Fluency -- the 95% Cr.I includes zero, suggesting substantial uncertainty about the direction and magnitude of the effect--that is, though most of the posterior density supports positive effects, nil and weakly negative effects cannot be ruled out.</span>
<span id="cb41-1104"><a href="#cb41-1104" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1107"><a href="#cb41-1107" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1108"><a href="#cb41-1108" aria-hidden="true" tabindex="-1"></a>marg_zi_brms <span class="ot">&lt;-</span> fit_zi <span class="sc">|&gt;</span></span>
<span id="cb41-1109"><a href="#cb41-1109" aria-hidden="true" tabindex="-1"></a>  <span class="fu">avg_comparisons</span>(</span>
<span id="cb41-1110"><a href="#cb41-1110" aria-hidden="true" tabindex="-1"></a>    <span class="at">variables =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb41-1111"><a href="#cb41-1111" aria-hidden="true" tabindex="-1"></a>    <span class="at">dpar =</span> <span class="st">"mu"</span>,</span>
<span id="cb41-1112"><a href="#cb41-1112" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb41-1113"><a href="#cb41-1113" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-1114"><a href="#cb41-1114" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1115"><a href="#cb41-1115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1118"><a href="#cb41-1118" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1119"><a href="#cb41-1119" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-marg-mu-brms</span></span>
<span id="cb41-1120"><a href="#cb41-1120" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap:  Probablity fluency difference ($\mu$)</span></span>
<span id="cb41-1121"><a href="#cb41-1121" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-1122"><a href="#cb41-1122" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb41-1123"><a href="#cb41-1123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1124"><a href="#cb41-1124" aria-hidden="true" tabindex="-1"></a>p_d <span class="ot">&lt;-</span> marg_zi_brms <span class="sc">|&gt;</span></span>
<span id="cb41-1125"><a href="#cb41-1125" aria-hidden="true" tabindex="-1"></a>  <span class="fu">p_direction</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-1126"><a href="#cb41-1126" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-1127"><a href="#cb41-1127" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pd)</span>
<span id="cb41-1128"><a href="#cb41-1128" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1129"><a href="#cb41-1129" aria-hidden="true" tabindex="-1"></a>marg_zi_brms <span class="sc">|&gt;</span></span>
<span id="cb41-1130"><a href="#cb41-1130" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(term, contrast, estimate, conf.low, conf.high) <span class="sc">|&gt;</span></span>
<span id="cb41-1131"><a href="#cb41-1131" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-1132"><a href="#cb41-1132" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb41-1133"><a href="#cb41-1133" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{conf.low}, {conf.high}]"</span>)</span>
<span id="cb41-1134"><a href="#cb41-1134" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb41-1135"><a href="#cb41-1135" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Contrast"</span> <span class="ot">=</span> <span class="st">"contrast"</span>, <span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"estimate"</span>, <span class="st">"Term"</span> <span class="ot">=</span> <span class="st">"term"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1136"><a href="#cb41-1136" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(p_d) <span class="sc">|&gt;</span></span>
<span id="cb41-1137"><a href="#cb41-1137" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Contrast, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, pd) <span class="sc">|&gt;</span></span>
<span id="cb41-1138"><a href="#cb41-1138" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-1139"><a href="#cb41-1139" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1140"><a href="#cb41-1140" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-1141"><a href="#cb41-1141" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1142"><a href="#cb41-1142" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1143"><a href="#cb41-1143" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Dispersion</span></span>
<span id="cb41-1144"><a href="#cb41-1144" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1145"><a href="#cb41-1145" aria-hidden="true" tabindex="-1"></a>As shown in @tbl-marg-phi-brms, the posterior estimates suggest a credible effect of Fluency on dispersion ($\phi$), with disfluent responses showing greater variability. The 95% Cr.I for the fluency contrast does not include zero, indicating a high probability in differences in precision.</span>
<span id="cb41-1146"><a href="#cb41-1146" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1149"><a href="#cb41-1149" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1150"><a href="#cb41-1150" aria-hidden="true" tabindex="-1"></a>marg_phi_brms <span class="ot">&lt;-</span> fit_zi <span class="sc">|&gt;</span></span>
<span id="cb41-1151"><a href="#cb41-1151" aria-hidden="true" tabindex="-1"></a>  <span class="fu">avg_comparisons</span>(</span>
<span id="cb41-1152"><a href="#cb41-1152" aria-hidden="true" tabindex="-1"></a>    <span class="at">variables =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb41-1153"><a href="#cb41-1153" aria-hidden="true" tabindex="-1"></a>    <span class="at">dpar =</span> <span class="st">"phi"</span>,</span>
<span id="cb41-1154"><a href="#cb41-1154" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb41-1155"><a href="#cb41-1155" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-1156"><a href="#cb41-1156" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1157"><a href="#cb41-1157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1160"><a href="#cb41-1160" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1161"><a href="#cb41-1161" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-marg-phi-brms</span></span>
<span id="cb41-1162"><a href="#cb41-1162" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap:  Probablity fluency difference ($\phi$)</span></span>
<span id="cb41-1163"><a href="#cb41-1163" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-1164"><a href="#cb41-1164" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb41-1165"><a href="#cb41-1165" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1166"><a href="#cb41-1166" aria-hidden="true" tabindex="-1"></a>p_d <span class="ot">&lt;-</span> marg_phi_brms <span class="sc">|&gt;</span></span>
<span id="cb41-1167"><a href="#cb41-1167" aria-hidden="true" tabindex="-1"></a>  <span class="fu">p_direction</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-1168"><a href="#cb41-1168" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-1169"><a href="#cb41-1169" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pd)</span>
<span id="cb41-1170"><a href="#cb41-1170" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1171"><a href="#cb41-1171" aria-hidden="true" tabindex="-1"></a>marg_phi_brms <span class="sc">|&gt;</span></span>
<span id="cb41-1172"><a href="#cb41-1172" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(term, contrast, estimate, conf.low, conf.high) <span class="sc">|&gt;</span></span>
<span id="cb41-1173"><a href="#cb41-1173" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-1174"><a href="#cb41-1174" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb41-1175"><a href="#cb41-1175" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{conf.low}, {conf.high}]"</span>)</span>
<span id="cb41-1176"><a href="#cb41-1176" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb41-1177"><a href="#cb41-1177" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Contrast"</span> <span class="ot">=</span> <span class="st">"contrast"</span>, <span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"estimate"</span>, <span class="st">"Term"</span> <span class="ot">=</span> <span class="st">"term"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1178"><a href="#cb41-1178" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(p_d) <span class="sc">|&gt;</span></span>
<span id="cb41-1179"><a href="#cb41-1179" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Contrast, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, pd) <span class="sc">|&gt;</span></span>
<span id="cb41-1180"><a href="#cb41-1180" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-1181"><a href="#cb41-1181" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1182"><a href="#cb41-1182" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-1183"><a href="#cb41-1183" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1184"><a href="#cb41-1184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1185"><a href="#cb41-1185" aria-hidden="true" tabindex="-1"></a><span class="fu">### Zero-Inflation</span></span>
<span id="cb41-1186"><a href="#cb41-1186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1187"><a href="#cb41-1187" aria-hidden="true" tabindex="-1"></a>We can use {marginaleffects} to estimate and plot the posterior difference between the fluent and disfluent conditions (see @fig-zi-plot). In @fig-zi-plot, the posterior distribution for this contrast lies mostly below zero, indicating that a fluent instructor is associated with a lower probability of zero responses. The estimated reduction is approximately 13%. The 95% credible interval does not include zero, which indicates that the data provide consistent evidence for a reduction in zero responses under fluent instruction.</span>
<span id="cb41-1188"><a href="#cb41-1188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1191"><a href="#cb41-1191" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1192"><a href="#cb41-1192" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl marg-zib-brms</span></span>
<span id="cb41-1193"><a href="#cb41-1193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1194"><a href="#cb41-1194" aria-hidden="true" tabindex="-1"></a>marg_zi_brms <span class="ot">&lt;-</span> fit_zi <span class="sc">|&gt;</span></span>
<span id="cb41-1195"><a href="#cb41-1195" aria-hidden="true" tabindex="-1"></a>  <span class="fu">avg_comparisons</span>(</span>
<span id="cb41-1196"><a href="#cb41-1196" aria-hidden="true" tabindex="-1"></a>    <span class="at">variables =</span> <span class="fu">list</span>(<span class="at">Fluency =</span> <span class="st">"revpairwise"</span>),</span>
<span id="cb41-1197"><a href="#cb41-1197" aria-hidden="true" tabindex="-1"></a>    <span class="at">dpar =</span> <span class="st">"zi"</span>,</span>
<span id="cb41-1198"><a href="#cb41-1198" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb41-1199"><a href="#cb41-1199" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-1200"><a href="#cb41-1200" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1201"><a href="#cb41-1201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1204"><a href="#cb41-1204" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1205"><a href="#cb41-1205" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-zi-plot</span></span>
<span id="cb41-1206"><a href="#cb41-1206" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Visualization of the predicted difference for zero-inflated part of model"</span></span>
<span id="cb41-1207"><a href="#cb41-1207" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-1208"><a href="#cb41-1208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1209"><a href="#cb41-1209" aria-hidden="true" tabindex="-1"></a>marg_zi_brms <span class="sc">|&gt;</span></span>
<span id="cb41-1210"><a href="#cb41-1210" aria-hidden="true" tabindex="-1"></a>  <span class="fu">posterior_draws</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-1211"><a href="#cb41-1211" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> draw, <span class="at">y =</span> contrast)) <span class="sc">+</span></span>
<span id="cb41-1212"><a href="#cb41-1212" aria-hidden="true" tabindex="-1"></a>  <span class="fu">stat_halfeye</span>(<span class="at">.width =</span> <span class="fu">c</span>(.<span class="dv">5</span>, .<span class="dv">8</span>, .<span class="dv">95</span>), <span class="at">alpha =</span> <span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb41-1213"><a href="#cb41-1213" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_okabe_ito</span>() <span class="sc">+</span></span>
<span id="cb41-1214"><a href="#cb41-1214" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">label_percent</span>()) <span class="sc">+</span></span>
<span id="cb41-1215"><a href="#cb41-1215" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb41-1216"><a href="#cb41-1216" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Accuracy"</span>,</span>
<span id="cb41-1217"><a href="#cb41-1217" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb41-1218"><a href="#cb41-1218" aria-hidden="true" tabindex="-1"></a>    <span class="at">caption =</span> <span class="st">"50%, 80% and 95% credible intervals shown in black"</span></span>
<span id="cb41-1219"><a href="#cb41-1219" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-1220"><a href="#cb41-1220" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_vline</span>(<span class="at">xintercept =</span> <span class="dv">0</span>, <span class="at">color =</span> <span class="st">"black"</span>, <span class="at">linetype =</span> <span class="st">"solid"</span>, <span class="at">size =</span> <span class="fl">1.2</span>) <span class="sc">+</span></span>
<span id="cb41-1221"><a href="#cb41-1221" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_discrete</span>(<span class="at">expand =</span> <span class="fu">expansion</span>(<span class="at">mult =</span> <span class="fl">0.1</span>)) <span class="sc">+</span></span>
<span id="cb41-1222"><a href="#cb41-1222" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb41-1223"><a href="#cb41-1223" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>)</span>
<span id="cb41-1224"><a href="#cb41-1224" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1225"><a href="#cb41-1225" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1226"><a href="#cb41-1226" aria-hidden="true" tabindex="-1"></a><span class="fu">## Zero-One-Inflated beta (ZOIB)</span></span>
<span id="cb41-1227"><a href="#cb41-1227" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1228"><a href="#cb41-1228" aria-hidden="true" tabindex="-1"></a>The ZIB model works well if there are 0s in your data, but not 1s.<span class="ot">[^8]</span> In our previous examples we either got rid of both 0s and 1s (beta regression), or removed the 1s (ZIB). Sometimes it is theoretically useful to model both 0s and 1s as separate processes or to consider these values as essentially similar parts of the continuous response, as we show later in the ordered beta regression model. For example, this is important in visual analog scale data where there might be a prevalence of responses at the bounds <span class="co">[</span><span class="ot">@kong2016</span><span class="co">]</span>, in JOL tasks <span class="co">[</span><span class="ot">@wilford2020</span><span class="co">]</span>, or in a free-list task where individuals provide open responses to some question or topic which are then recoded to fall between 0-1 <span class="co">[</span><span class="ot">@bendixen2023</span><span class="co">]</span>. Here 0s and 1s are meaningful; 0 means item was not listed and 1 means the item was listed first.</span>
<span id="cb41-1229"><a href="#cb41-1229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1230"><a href="#cb41-1230" aria-hidden="true" tabindex="-1"></a><span class="ot">[^8]: </span>In cases where your data include exact 1s but no 0s, you can fit a one-inflated beta regression model in {brms} by setting the <span class="in">`coi`</span> parameter to <span class="in">`1`</span>. This tells the model that all point masses occur at 1, rather than being split between 0 and 1. In other words, <span class="in">`coi = 1`</span> assumes that any inflation in the data is due entirely to values at 1. In our data, we have exactly one value equal to 1<span class="sc">\[</span>\^6<span class="sc">\]</span>. While probably not significant to alter our findings, we can model 1s with a special type of model called the zero-one-inflated beta (ZOIB) model <span class="co">[</span><span class="ot">@liu2015</span><span class="co">]</span> if we believe that both 0s and 1s are distinct outcomes.</span>
<span id="cb41-1231"><a href="#cb41-1231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1232"><a href="#cb41-1232" aria-hidden="true" tabindex="-1"></a>Similar to the beta and zero-inflated models discussed above, we can fit a zero-and-one‐inflated beta (ZOIB) model in {brms} using the <span class="in">`zero_one_inflated_beta`</span> family. This formulation simultaneously estimates the mean $\mu$ and precision $\phi$ of the Beta component, as well as two inflation parameters: $\alpha$ , the probability that an observation is at either boundary (0 or 1), and $\gamma$ , the conditional probability that, given an observation falls on a boundary, it takes the value 1 rather than 0. In other words, $\alpha$ determines how often responses occur exactly at the endpoints, and $\gamma$ determines the balance between zeros and ones among those endpoint values. This specification allows the model to capture both the continuous variation in the interior of the (0, 1) interval and the presence of exact boundary values.</span>
<span id="cb41-1233"><a href="#cb41-1233" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1234"><a href="#cb41-1234" aria-hidden="true" tabindex="-1"></a>To illustrate how $\alpha$ and $\gamma$ shape the distribution, @fig-zoib displays simulated data across a range of parameter combinations. As $\alpha$ increases, more responses occur at the endpoints. As $\gamma$ increases, the proportion of those endpoint responses that are 1 increases relative to 0, producing increasingly pronounced spikes at 1 as $\gamma$ approaches 1. Together, these parameters give the ZOIB model the flexibility to represent datasets with mixtures of continuous values and exact zeros and ones.</span>
<span id="cb41-1235"><a href="#cb41-1235" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1238"><a href="#cb41-1238" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1239"><a href="#cb41-1239" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-zoib</span></span>
<span id="cb41-1240"><a href="#cb41-1240" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Simulated data from a ZOIB model illustrating the effects of the zero-one inflation parameter ($\alpha$) and the conditional one-inflation parameter ($\gamma$).</span></span>
<span id="cb41-1241"><a href="#cb41-1241" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb41-1242"><a href="#cb41-1242" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb41-1243"><a href="#cb41-1243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1244"><a href="#cb41-1244" aria-hidden="true" tabindex="-1"></a><span class="co"># Zero-one inflated beta generator adapted from</span></span>
<span id="cb41-1245"><a href="#cb41-1245" aria-hidden="true" tabindex="-1"></a><span class="co"># https://vuorre.com/posts/2019-02-18-analyze-analog-scale-ratings-with-zero-one-inflated-beta-models/</span></span>
<span id="cb41-1246"><a href="#cb41-1246" aria-hidden="true" tabindex="-1"></a>rzoib <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">n =</span> <span class="dv">1000</span>, <span class="at">alpha =</span> <span class="fl">0.1</span>, <span class="at">gamma =</span> <span class="fl">0.45</span>, <span class="at">mu =</span> <span class="fl">0.7</span>, <span class="at">phi =</span> <span class="dv">3</span>) {</span>
<span id="cb41-1247"><a href="#cb41-1247" aria-hidden="true" tabindex="-1"></a>  a <span class="ot">&lt;-</span> mu <span class="sc">*</span> phi</span>
<span id="cb41-1248"><a href="#cb41-1248" aria-hidden="true" tabindex="-1"></a>  b <span class="ot">&lt;-</span> (<span class="dv">1</span> <span class="sc">-</span> mu) <span class="sc">*</span> phi</span>
<span id="cb41-1249"><a href="#cb41-1249" aria-hidden="true" tabindex="-1"></a>  y <span class="ot">&lt;-</span> <span class="fu">ifelse</span>(</span>
<span id="cb41-1250"><a href="#cb41-1250" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbinom</span>(n, <span class="dv">1</span>, alpha),</span>
<span id="cb41-1251"><a href="#cb41-1251" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbinom</span>(n, <span class="dv">1</span>, gamma),</span>
<span id="cb41-1252"><a href="#cb41-1252" aria-hidden="true" tabindex="-1"></a>    <span class="fu">rbeta</span>(n, a, b)</span>
<span id="cb41-1253"><a href="#cb41-1253" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-1254"><a href="#cb41-1254" aria-hidden="true" tabindex="-1"></a>  y</span>
<span id="cb41-1255"><a href="#cb41-1255" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb41-1256"><a href="#cb41-1256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1257"><a href="#cb41-1257" aria-hidden="true" tabindex="-1"></a><span class="co"># Optimized parameter grid</span></span>
<span id="cb41-1258"><a href="#cb41-1258" aria-hidden="true" tabindex="-1"></a>param_grid <span class="ot">&lt;-</span> <span class="fu">expand_grid</span>(</span>
<span id="cb41-1259"><a href="#cb41-1259" aria-hidden="true" tabindex="-1"></a>  <span class="at">alpha =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.4</span>),</span>
<span id="cb41-1260"><a href="#cb41-1260" aria-hidden="true" tabindex="-1"></a>  <span class="at">gamma =</span> <span class="fu">c</span>(<span class="fl">0.1</span>, <span class="fl">0.9</span>),</span>
<span id="cb41-1261"><a href="#cb41-1261" aria-hidden="true" tabindex="-1"></a>  <span class="at">mu =</span> <span class="fl">0.7</span>,</span>
<span id="cb41-1262"><a href="#cb41-1262" aria-hidden="true" tabindex="-1"></a>  <span class="at">phi =</span> <span class="dv">3</span></span>
<span id="cb41-1263"><a href="#cb41-1263" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-1264"><a href="#cb41-1264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1265"><a href="#cb41-1265" aria-hidden="true" tabindex="-1"></a><span class="co"># Simulate data</span></span>
<span id="cb41-1266"><a href="#cb41-1266" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb41-1267"><a href="#cb41-1267" aria-hidden="true" tabindex="-1"></a>sim_data <span class="ot">&lt;-</span> param_grid <span class="sc">%&gt;%</span></span>
<span id="cb41-1268"><a href="#cb41-1268" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-1269"><a href="#cb41-1269" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> <span class="fu">pmap</span>(</span>
<span id="cb41-1270"><a href="#cb41-1270" aria-hidden="true" tabindex="-1"></a>      <span class="fu">list</span>(alpha, gamma, mu, phi),</span>
<span id="cb41-1271"><a href="#cb41-1271" aria-hidden="true" tabindex="-1"></a>      <span class="sc">~</span> <span class="fu">rzoib</span>(<span class="dv">1000</span>, ..<span class="dv">1</span>, ..<span class="dv">2</span>, ..<span class="dv">3</span>, ..<span class="dv">4</span>)</span>
<span id="cb41-1272"><a href="#cb41-1272" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb41-1273"><a href="#cb41-1273" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">%&gt;%</span></span>
<span id="cb41-1274"><a href="#cb41-1274" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unnest</span>(data) <span class="sc">%&gt;%</span></span>
<span id="cb41-1275"><a href="#cb41-1275" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">Accuracy =</span> data)</span>
<span id="cb41-1276"><a href="#cb41-1276" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1277"><a href="#cb41-1277" aria-hidden="true" tabindex="-1"></a><span class="co"># Classify type</span></span>
<span id="cb41-1278"><a href="#cb41-1278" aria-hidden="true" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> sim_data <span class="sc">%&gt;%</span></span>
<span id="cb41-1279"><a href="#cb41-1279" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-1280"><a href="#cb41-1280" aria-hidden="true" tabindex="-1"></a>    <span class="at">Type =</span> <span class="fu">case_when</span>(</span>
<span id="cb41-1281"><a href="#cb41-1281" aria-hidden="true" tabindex="-1"></a>      Accuracy <span class="sc">==</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"Zero"</span>,</span>
<span id="cb41-1282"><a href="#cb41-1282" aria-hidden="true" tabindex="-1"></a>      Accuracy <span class="sc">==</span> <span class="dv">1</span> <span class="sc">~</span> <span class="st">"One"</span>,</span>
<span id="cb41-1283"><a href="#cb41-1283" aria-hidden="true" tabindex="-1"></a>      <span class="cn">TRUE</span> <span class="sc">~</span> <span class="st">"beta"</span></span>
<span id="cb41-1284"><a href="#cb41-1284" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb41-1285"><a href="#cb41-1285" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-1286"><a href="#cb41-1286" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1287"><a href="#cb41-1287" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure 'Zero' comes before 'One'</span></span>
<span id="cb41-1288"><a href="#cb41-1288" aria-hidden="true" tabindex="-1"></a>plot_data <span class="ot">&lt;-</span> plot_data <span class="sc">%&gt;%</span></span>
<span id="cb41-1289"><a href="#cb41-1289" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Type =</span> <span class="fu">factor</span>(Type, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Zero"</span>, <span class="st">"One"</span>, <span class="st">"beta"</span>)))</span>
<span id="cb41-1290"><a href="#cb41-1290" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1291"><a href="#cb41-1291" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb41-1292"><a href="#cb41-1292" aria-hidden="true" tabindex="-1"></a>  <span class="co"># beta histogram</span></span>
<span id="cb41-1293"><a href="#cb41-1293" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(</span>
<span id="cb41-1294"><a href="#cb41-1294" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> plot_data <span class="sc">%&gt;%</span> <span class="fu">filter</span>(Type <span class="sc">==</span> <span class="st">"beta"</span>),</span>
<span id="cb41-1295"><a href="#cb41-1295" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> Accuracy, <span class="at">y =</span> <span class="fu">after_stat</span>(count)),</span>
<span id="cb41-1296"><a href="#cb41-1296" aria-hidden="true" tabindex="-1"></a>    <span class="at">bins =</span> <span class="dv">50</span>,</span>
<span id="cb41-1297"><a href="#cb41-1297" aria-hidden="true" tabindex="-1"></a>    <span class="at">fill =</span> <span class="st">"gray60"</span>,</span>
<span id="cb41-1298"><a href="#cb41-1298" aria-hidden="true" tabindex="-1"></a>    <span class="at">color =</span> <span class="st">"white"</span>,</span>
<span id="cb41-1299"><a href="#cb41-1299" aria-hidden="true" tabindex="-1"></a>    <span class="at">alpha =</span> <span class="fl">0.8</span></span>
<span id="cb41-1300"><a href="#cb41-1300" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-1301"><a href="#cb41-1301" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Spikes at 0 and 1</span></span>
<span id="cb41-1302"><a href="#cb41-1302" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(</span>
<span id="cb41-1303"><a href="#cb41-1303" aria-hidden="true" tabindex="-1"></a>    <span class="at">data =</span> plot_data <span class="sc">%&gt;%</span></span>
<span id="cb41-1304"><a href="#cb41-1304" aria-hidden="true" tabindex="-1"></a>      <span class="fu">filter</span>(Type <span class="sc">!=</span> <span class="st">"beta"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb41-1305"><a href="#cb41-1305" aria-hidden="true" tabindex="-1"></a>      <span class="fu">group_by</span>(alpha, gamma, Accuracy, Type) <span class="sc">%&gt;%</span></span>
<span id="cb41-1306"><a href="#cb41-1306" aria-hidden="true" tabindex="-1"></a>      <span class="fu">summarise</span>(<span class="at">count =</span> <span class="fu">n</span>(), <span class="at">.groups =</span> <span class="st">"drop"</span>),</span>
<span id="cb41-1307"><a href="#cb41-1307" aria-hidden="true" tabindex="-1"></a>    <span class="fu">aes</span>(<span class="at">x =</span> Accuracy, <span class="at">y =</span> count, <span class="at">fill =</span> Type),</span>
<span id="cb41-1308"><a href="#cb41-1308" aria-hidden="true" tabindex="-1"></a>    <span class="at">width =</span> <span class="fl">0.01</span></span>
<span id="cb41-1309"><a href="#cb41-1309" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-1310"><a href="#cb41-1310" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(</span>
<span id="cb41-1311"><a href="#cb41-1311" aria-hidden="true" tabindex="-1"></a>    <span class="at">values =</span> <span class="fu">c</span>(<span class="st">"Zero"</span> <span class="ot">=</span> <span class="st">"red"</span>, <span class="st">"One"</span> <span class="ot">=</span> <span class="st">"blue"</span>),</span>
<span id="cb41-1312"><a href="#cb41-1312" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="st">"Type"</span></span>
<span id="cb41-1313"><a href="#cb41-1313" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-1314"><a href="#cb41-1314" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(</span>
<span id="cb41-1315"><a href="#cb41-1315" aria-hidden="true" tabindex="-1"></a>    gamma <span class="sc">~</span> alpha,</span>
<span id="cb41-1316"><a href="#cb41-1316" aria-hidden="true" tabindex="-1"></a>    <span class="at">labeller =</span> label_both,</span>
<span id="cb41-1317"><a href="#cb41-1317" aria-hidden="true" tabindex="-1"></a>    <span class="at">scales =</span> <span class="st">"free_y"</span>,</span>
<span id="cb41-1318"><a href="#cb41-1318" aria-hidden="true" tabindex="-1"></a>    <span class="at">switch =</span> <span class="st">"y"</span></span>
<span id="cb41-1319"><a href="#cb41-1319" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-1320"><a href="#cb41-1320" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(</span>
<span id="cb41-1321"><a href="#cb41-1321" aria-hidden="true" tabindex="-1"></a>    <span class="at">x =</span> <span class="st">"Accuracy"</span>,</span>
<span id="cb41-1322"><a href="#cb41-1322" aria-hidden="true" tabindex="-1"></a>    <span class="at">y =</span> <span class="st">"Count"</span></span>
<span id="cb41-1323"><a href="#cb41-1323" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">+</span></span>
<span id="cb41-1324"><a href="#cb41-1324" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb41-1325"><a href="#cb41-1325" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">labels =</span> <span class="fu">label_percent</span>())</span>
<span id="cb41-1326"><a href="#cb41-1326" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1327"><a href="#cb41-1327" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1328"><a href="#cb41-1328" aria-hidden="true" tabindex="-1"></a>To fit a ZOIB model we use the <span class="in">`bf()`</span> function. We model each parameter as a function of Fluency. We then pass the <span class="in">`zoib_model`</span> to our <span class="in">`brm()`</span> function (see @lst-zoib). The summary of the output is in @tbl-all (under <span class="in">`ZOIB`</span>).</span>
<span id="cb41-1329"><a href="#cb41-1329" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1332"><a href="#cb41-1332" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1333"><a href="#cb41-1333" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fit zoib model</span></span>
<span id="cb41-1334"><a href="#cb41-1334" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-zoib</span></span>
<span id="cb41-1335"><a href="#cb41-1335" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting a ZOIB model with `brm()`.</span></span>
<span id="cb41-1336"><a href="#cb41-1336" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-1337"><a href="#cb41-1337" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb41-1338"><a href="#cb41-1338" aria-hidden="true" tabindex="-1"></a><span class="co">#| message: false</span></span>
<span id="cb41-1339"><a href="#cb41-1339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1340"><a href="#cb41-1340" aria-hidden="true" tabindex="-1"></a><span class="co"># fit the zoib model</span></span>
<span id="cb41-1341"><a href="#cb41-1341" aria-hidden="true" tabindex="-1"></a>zoib_model <span class="ot">&lt;-</span> <span class="fu">bf</span>(</span>
<span id="cb41-1342"><a href="#cb41-1342" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency, <span class="co"># The mean of the 0-1 values, or mu</span></span>
<span id="cb41-1343"><a href="#cb41-1343" aria-hidden="true" tabindex="-1"></a>  phi <span class="sc">~</span> Fluency, <span class="co"># The precision of the 0-1 values, or phi</span></span>
<span id="cb41-1344"><a href="#cb41-1344" aria-hidden="true" tabindex="-1"></a>  zoi <span class="sc">~</span> Fluency, <span class="co"># The zero-or-one-inflated part, or alpha</span></span>
<span id="cb41-1345"><a href="#cb41-1345" aria-hidden="true" tabindex="-1"></a>  coi <span class="sc">~</span> Fluency, <span class="co"># The one-inflated part, conditional on the 1s, or gamma</span></span>
<span id="cb41-1346"><a href="#cb41-1346" aria-hidden="true" tabindex="-1"></a>  <span class="at">family =</span> <span class="fu">zero_one_inflated_beta</span>()</span>
<span id="cb41-1347"><a href="#cb41-1347" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-1348"><a href="#cb41-1348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1349"><a href="#cb41-1349" aria-hidden="true" tabindex="-1"></a>fit_zoib <span class="ot">&lt;-</span> <span class="fu">brm</span>(</span>
<span id="cb41-1350"><a href="#cb41-1350" aria-hidden="true" tabindex="-1"></a>  <span class="at">formula =</span> zoib_model,</span>
<span id="cb41-1351"><a href="#cb41-1351" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb41-1352"><a href="#cb41-1352" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_zoib_model"</span>)</span>
<span id="cb41-1353"><a href="#cb41-1353" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-1354"><a href="#cb41-1354" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1355"><a href="#cb41-1355" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1358"><a href="#cb41-1358" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1359"><a href="#cb41-1359" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl zoib</span></span>
<span id="cb41-1360"><a href="#cb41-1360" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-1361"><a href="#cb41-1361" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb41-1362"><a href="#cb41-1362" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb41-1363"><a href="#cb41-1363" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1364"><a href="#cb41-1364" aria-hidden="true" tabindex="-1"></a>zoib_model <span class="ot">&lt;-</span> <span class="fu">model_parameters</span>(fit_zoib, <span class="st">"mean"</span>)</span>
<span id="cb41-1365"><a href="#cb41-1365" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1366"><a href="#cb41-1366" aria-hidden="true" tabindex="-1"></a>zoib_model <span class="sc">|&gt;</span></span>
<span id="cb41-1367"><a href="#cb41-1367" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-1368"><a href="#cb41-1368" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb41-1369"><a href="#cb41-1369" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{CI_low}, {CI_high}]"</span>)</span>
<span id="cb41-1370"><a href="#cb41-1370" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb41-1371"><a href="#cb41-1371" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Parameter, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, pd) <span class="sc">|&gt;</span></span>
<span id="cb41-1372"><a href="#cb41-1372" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1373"><a href="#cb41-1373" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-1374"><a href="#cb41-1374" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1375"><a href="#cb41-1375" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1376"><a href="#cb41-1376" aria-hidden="true" tabindex="-1"></a><span class="fu">### Model Parameters</span></span>
<span id="cb41-1377"><a href="#cb41-1377" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1378"><a href="#cb41-1378" aria-hidden="true" tabindex="-1"></a>The output for the model is lengthy because we are estimating four distinct components, each with their own independent responses and sub-models. All the coefficients are on the logit scale, except $\phi$ , which is on the log scale. Thankfully drawing inferences for all these different parameters, plotting their distributions, and estimating their average marginal effects looks exactly the same--all the <span class="in">`brms`</span> and {marginaleffects} functions we used work the same.</span>
<span id="cb41-1379"><a href="#cb41-1379" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1380"><a href="#cb41-1380" aria-hidden="true" tabindex="-1"></a><span class="fu">### Predictions and Marginal Effects</span></span>
<span id="cb41-1381"><a href="#cb41-1381" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1382"><a href="#cb41-1382" aria-hidden="true" tabindex="-1"></a>With {marginaleffects} we can choose marginalize over all the sub-models, averaged across the 0s, continuous responses, and 1s in the data, or we can model the parameters separately using the <span class="in">`dpar`</span> argument like we did above setting it to: $\mu$, $\phi$, $\alpha$, $\gamma$ (see below). Using <span class="in">`avg_predictions()`</span> and not setting <span class="in">`dpar`</span> we can get the predicted probabilities across all the sub-models. We can also plot the overall difference between fluency and disfluency for the whole model with <span class="in">`plot_predictions()`</span>.</span>
<span id="cb41-1383"><a href="#cb41-1383" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1384"><a href="#cb41-1384" aria-hidden="true" tabindex="-1"></a>In addition, we show below how one can extract the predicted probabilities and marginal effects for $\gamma$ (and a similar process for any other model component, <span class="in">`zoi`</span>, etc.):</span>
<span id="cb41-1385"><a href="#cb41-1385" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1388"><a href="#cb41-1388" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1389"><a href="#cb41-1389" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: avg coi</span></span>
<span id="cb41-1390"><a href="#cb41-1390" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-coi</span></span>
<span id="cb41-1391"><a href="#cb41-1391" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: "Extracting predicted probabilities and marginal effects for conditional-one parameter"</span></span>
<span id="cb41-1392"><a href="#cb41-1392" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-1393"><a href="#cb41-1393" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1394"><a href="#cb41-1394" aria-hidden="true" tabindex="-1"></a><span class="co"># get average predictions for coi param</span></span>
<span id="cb41-1395"><a href="#cb41-1395" aria-hidden="true" tabindex="-1"></a>coi_probs <span class="ot">&lt;-</span> <span class="fu">avg_predictions</span>(fit_zoib, <span class="at">variables =</span> <span class="st">"Fluency"</span>, <span class="at">dpar =</span> <span class="st">"coi"</span>)</span>
<span id="cb41-1396"><a href="#cb41-1396" aria-hidden="true" tabindex="-1"></a><span class="co"># get differene between the two conditions</span></span>
<span id="cb41-1397"><a href="#cb41-1397" aria-hidden="true" tabindex="-1"></a>coi_me <span class="ot">&lt;-</span> <span class="fu">avg_comparisons</span>(fit_zoib, <span class="at">variables =</span> <span class="fu">c</span>(<span class="st">"Fluency"</span>), <span class="at">dpar =</span> <span class="st">"coi"</span>)</span>
<span id="cb41-1398"><a href="#cb41-1398" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1399"><a href="#cb41-1399" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1400"><a href="#cb41-1400" aria-hidden="true" tabindex="-1"></a><span class="fu">## Ordered Beta Regression</span></span>
<span id="cb41-1401"><a href="#cb41-1401" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1402"><a href="#cb41-1402" aria-hidden="true" tabindex="-1"></a>Looking at the output from the ZOIB model (@tbl-all), we can see how running a model like this can become fairly complex as it is fitting distinct sub-models for each component of the scale. The ability to consider 0s and 1s as distinct processes from continuous values comes at a price in terms of complexity and interpretability. A simplified version of the zero-one-inflated beta (ZOIB) model, known as ordered beta regression <span class="co">[</span><span class="ot">@kubinec2022; see also @Makowski2025CHOCO for a reparameterized version called the *beta-Gate* model</span><span class="co">]</span>, has been recently proposed. The ordered beta regression model exploits the fact that, for most analyses, the continuous values (between 0-1) and the discrete outcomes (e.g., 0 or 1) are *ordered*. For example, as a covariate $x$ increases or decreases, we should expect the bounded outcome $y$ to increase or decrease monotonically as well from 0 to $(0,1)$ to 1. The ZOIB model does not impose this restriction; a covariate could increase and the response $y$ could increase in its continuous values while *simultaneously* decreasing at *both* end points.<span class="ot">[^9]</span> This complexity is not immediately obvious when fitting the ZOIB, nor is it a potential relationship that many scholars want to consider when examining how covariates influence a bounded scale.</span>
<span id="cb41-1403"><a href="#cb41-1403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1404"><a href="#cb41-1404" aria-hidden="true" tabindex="-1"></a><span class="ot">[^9]: </span>For a more complete description of this issue, we refer the reader to @kubinec2022.</span>
<span id="cb41-1405"><a href="#cb41-1405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1406"><a href="#cb41-1406" aria-hidden="true" tabindex="-1"></a>To make the response ordered, the ordered beta regression model estimates a weighted combination of a standard beta regression model for continuous responses and a logit model for the discrete values of the response. By doing so, the amount of distinctiveness between the continuous responses and the discrete end points is a function of the data (and any informative priors) rather than strictly defined as fully distinct processes as in the ZOIB. For some datasets, the continuous and discrete responses will be fairly distinct, and in others less so.</span>
<span id="cb41-1407"><a href="#cb41-1407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1408"><a href="#cb41-1408" aria-hidden="true" tabindex="-1"></a>The weights that average together the two parts of the outcome (i.e., discrete and continuous) are determined by cutpoints that are estimated in conjunction with the data in a similar manner to what is known as an ordered logit model. An in-depth explanation of ordinal regression is beyond the scope of this tutorial <span class="co">[</span><span class="ot">but see @Fullerton2023; @bürkner2019</span><span class="co">]</span>. At a basic level, ordinal regression models are useful for outcome variables that are categorical in nature and have some inherent ordering (e.g., Likert scale items). To preserve this ordering, ordinal models rely on the cumulative probability distribution. Within an ordinal regression model it is assumed that there is a continuous but unobserved latent variable that determines which of $k$ ordinal responses will be selected. For example on a typical Likert scale from 'Strongly Disagree' to 'Strongly Agree', you could assume that there is a continuous, unobserved variable called 'Agreement'.</span>
<span id="cb41-1409"><a href="#cb41-1409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1410"><a href="#cb41-1410" aria-hidden="true" tabindex="-1"></a>While we cannot measure Agreement directly, the ordinal response gives us some indication about where participants are on the continuous Agreement scale. $k - 1$ cutoffs are then estimated to indicate the point on the continuous Agreement scale at which your Agreement level is high enough to push you into the next ordinal category (say Agree to Strongly Agree). Coefficients in the model estimate how much different predictors change the estimated *continuous* scale (here, Agreement). Since there’s only one underlying process, there’s only one set of coefficients to work with (proportional odds assumption).</span>
<span id="cb41-1411"><a href="#cb41-1411" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1412"><a href="#cb41-1412" aria-hidden="true" tabindex="-1"></a>In an ordered beta regression, three ordered categories are modeled: (1) exactly zero, (2) somewhere between zero and one, and (3) exactly one. In an ordered beta regression, (1) and (2) are modeled with cumulative logits, where one cutpoint is the the boundary between Exactly 0 and Between 0 and 1 and the other cutpoint is the boundary between *Between 0 and 1* and *Exactly 1.* The continuous values in the middle, 0 to 1 (3), are modeled as a vanilla beta regression with parameters reflecting the mean response on the logit scale as we have described previously. Ultimately, employing cutpoints allows for a smooth transition between the bounds and the continuous values, permitting both to be considered together rather than modeled separately as the ZOIB requires.</span>
<span id="cb41-1413"><a href="#cb41-1413" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1414"><a href="#cb41-1414" aria-hidden="true" tabindex="-1"></a>The ordered beta regression model has shown to be more efficient and less biased than some of the methods discussed <span class="co">[</span><span class="ot">@kubinec2022</span><span class="co">]</span> herein and has seen increasing use across the biomedical and social sciences <span class="co">[</span><span class="ot">@wilkesQuantifyingCoextinctionsEcosystem2024; @martinVocalComplexitySocially2024; @smithEcologicalMomentaryAssessment2024; @shresthaBigFiveTraits2024; @nouvianGlyphosateImpairsAversive2023</span><span class="co">]</span> because it produces only a single set of coefficient estimates in a similar manner to a standard beta regression or OLS.<span class="ot">[^10]</span></span>
<span id="cb41-1415"><a href="#cb41-1415" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1416"><a href="#cb41-1416" aria-hidden="true" tabindex="-1"></a><span class="ot">[^10]: </span>Please note that there are other models available that can model this continuous process like the beta-gate model <span class="co">[</span><span class="ot">@Makowski2025CHOCO</span><span class="co">]</span> and the censored extended beta regression model <span class="co">[</span><span class="ot">@kosmidis2025</span><span class="co">]</span>.</span>
<span id="cb41-1417"><a href="#cb41-1417" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1418"><a href="#cb41-1418" aria-hidden="true" tabindex="-1"></a><span class="fu">### Fitting an Ordered Beta Regression</span></span>
<span id="cb41-1419"><a href="#cb41-1419" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1420"><a href="#cb41-1420" aria-hidden="true" tabindex="-1"></a>To fit an ordered beta regression in a Bayesian context we use the {ordbetareg} <span class="co">[</span><span class="ot">@ordbetareg</span><span class="co">]</span> package. {ordbetareg} is a front-end to the <span class="in">`brms`</span> package that we described earlier; in addition to the functions available in the package, most <span class="in">`brms`</span> functions and plots, including the diverse array of regression modeling options, will work with {ordbetareg} models. (We note that the <span class="in">`ordbeta`</span> model is also available as a maximum-likelihood variant in the R package {glmmTMB}.) We first load the {ordbetareg} package (see @lst-ordbeta).</span>
<span id="cb41-1421"><a href="#cb41-1421" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1424"><a href="#cb41-1424" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1425"><a href="#cb41-1425" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-1426"><a href="#cb41-1426" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-ordbeta</span></span>
<span id="cb41-1427"><a href="#cb41-1427" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Load {ordbetareg}</span></span>
<span id="cb41-1428"><a href="#cb41-1428" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: load ordbetareg</span></span>
<span id="cb41-1429"><a href="#cb41-1429" aria-hidden="true" tabindex="-1"></a><span class="co">#| eval: false</span></span>
<span id="cb41-1430"><a href="#cb41-1430" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1431"><a href="#cb41-1431" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ordbetareg)</span>
<span id="cb41-1432"><a href="#cb41-1432" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1433"><a href="#cb41-1433" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1434"><a href="#cb41-1434" aria-hidden="true" tabindex="-1"></a>The {ordbetareg} package uses <span class="in">`brms`</span> on the front-end so all the arguments we used previously apply here. Instead of the <span class="in">`brm()`</span> function we use <span class="in">`ordbetareg()`</span>. To fit a model where dispersion does not vary as a function of fluency we can use the below code (see @lst-fitordbeta).</span>
<span id="cb41-1435"><a href="#cb41-1435" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1438"><a href="#cb41-1438" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1439"><a href="#cb41-1439" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: brms-ordered-beta-model</span></span>
<span id="cb41-1440"><a href="#cb41-1440" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-fitordbeta</span></span>
<span id="cb41-1441"><a href="#cb41-1441" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting ordered beta model with `ordbetareg()`</span></span>
<span id="cb41-1442"><a href="#cb41-1442" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb41-1443"><a href="#cb41-1443" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-1444"><a href="#cb41-1444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1445"><a href="#cb41-1445" aria-hidden="true" tabindex="-1"></a>ord_fit_brms <span class="ot">&lt;-</span> <span class="fu">ordbetareg</span>(</span>
<span id="cb41-1446"><a href="#cb41-1446" aria-hidden="true" tabindex="-1"></a>  Accuracy <span class="sc">~</span> Fluency,</span>
<span id="cb41-1447"><a href="#cb41-1447" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb41-1448"><a href="#cb41-1448" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>,<span class="st">"models"</span>, <span class="st">"bayes_ordbeta_model"</span>)</span>
<span id="cb41-1449"><a href="#cb41-1449" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-1450"><a href="#cb41-1450" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1451"><a href="#cb41-1451" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1452"><a href="#cb41-1452" aria-hidden="true" tabindex="-1"></a>However, if we want dispersion to vary as a function of fluency we can easily do that (see @lst-ordbetadisp). Note the addition of the <span class="in">`phi_reg`</span> argument in <span class="in">`m.phi`</span>. This argument allows us to include a model that explicitly models the dispersion parameter. Because we are modeling $\phi$ as a function of fluency, we set the the argument to <span class="in">`both.`</span></span>
<span id="cb41-1453"><a href="#cb41-1453" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1456"><a href="#cb41-1456" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1457"><a href="#cb41-1457" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: brms ordered beta model phi</span></span>
<span id="cb41-1458"><a href="#cb41-1458" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-label: lst-ordbetadisp</span></span>
<span id="cb41-1459"><a href="#cb41-1459" aria-hidden="true" tabindex="-1"></a><span class="co">#| lst-cap: Fitting ordered beta model with dispersion using `ordbetareg()`</span></span>
<span id="cb41-1460"><a href="#cb41-1460" aria-hidden="true" tabindex="-1"></a><span class="co">#| results: hide</span></span>
<span id="cb41-1461"><a href="#cb41-1461" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: true</span></span>
<span id="cb41-1462"><a href="#cb41-1462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1463"><a href="#cb41-1463" aria-hidden="true" tabindex="-1"></a>ord_beta_phi <span class="ot">&lt;-</span> <span class="fu">bf</span>(Accuracy <span class="sc">~</span> Fluency, phi <span class="sc">~</span> Fluency)</span>
<span id="cb41-1464"><a href="#cb41-1464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1465"><a href="#cb41-1465" aria-hidden="true" tabindex="-1"></a>m.phi <span class="ot">&lt;-</span> <span class="fu">ordbetareg</span>(</span>
<span id="cb41-1466"><a href="#cb41-1466" aria-hidden="true" tabindex="-1"></a>  ord_beta_phi,</span>
<span id="cb41-1467"><a href="#cb41-1467" aria-hidden="true" tabindex="-1"></a>  <span class="at">data =</span> fluency_data,</span>
<span id="cb41-1468"><a href="#cb41-1468" aria-hidden="true" tabindex="-1"></a>  <span class="at">phi_reg =</span> <span class="st">'both'</span>,</span>
<span id="cb41-1469"><a href="#cb41-1469" aria-hidden="true" tabindex="-1"></a>  <span class="at">file =</span> here<span class="sc">::</span><span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_ordbeta_phi_model"</span>)</span>
<span id="cb41-1470"><a href="#cb41-1470" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-1471"><a href="#cb41-1471" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1472"><a href="#cb41-1472" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1473"><a href="#cb41-1473" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Marginal Effects</span></span>
<span id="cb41-1474"><a href="#cb41-1474" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1475"><a href="#cb41-1475" aria-hidden="true" tabindex="-1"></a>@tbl-all presents the posterior summary. We can use {marginaleffects} to calculate differences on the response scale that average over (or marginalize over) all our parameters.</span>
<span id="cb41-1476"><a href="#cb41-1476" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1479"><a href="#cb41-1479" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1480"><a href="#cb41-1480" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fit ordbeta</span></span>
<span id="cb41-1481"><a href="#cb41-1481" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1482"><a href="#cb41-1482" aria-hidden="true" tabindex="-1"></a>ord_beta_avg <span class="ot">&lt;-</span> m.phi <span class="sc">|&gt;</span></span>
<span id="cb41-1483"><a href="#cb41-1483" aria-hidden="true" tabindex="-1"></a>  <span class="fu">avg_comparisons</span>(</span>
<span id="cb41-1484"><a href="#cb41-1484" aria-hidden="true" tabindex="-1"></a>    <span class="at">variables =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb41-1485"><a href="#cb41-1485" aria-hidden="true" tabindex="-1"></a>    <span class="at">comparison =</span> <span class="st">"difference"</span></span>
<span id="cb41-1486"><a href="#cb41-1486" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-1487"><a href="#cb41-1487" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1488"><a href="#cb41-1488" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1491"><a href="#cb41-1491" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1492"><a href="#cb41-1492" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-ordbeta</span></span>
<span id="cb41-1493"><a href="#cb41-1493" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Marginal effect of fluency ordered beta model"</span></span>
<span id="cb41-1494"><a href="#cb41-1494" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb41-1495"><a href="#cb41-1495" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1496"><a href="#cb41-1496" aria-hidden="true" tabindex="-1"></a>p_d <span class="ot">&lt;-</span> ord_beta_avg <span class="sc">|&gt;</span></span>
<span id="cb41-1497"><a href="#cb41-1497" aria-hidden="true" tabindex="-1"></a>  <span class="fu">p_direction</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-1498"><a href="#cb41-1498" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-1499"><a href="#cb41-1499" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(pd)</span>
<span id="cb41-1500"><a href="#cb41-1500" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1501"><a href="#cb41-1501" aria-hidden="true" tabindex="-1"></a>ord_beta_avg <span class="sc">|&gt;</span></span>
<span id="cb41-1502"><a href="#cb41-1502" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(term, contrast, estimate, conf.low, conf.high) <span class="sc">|&gt;</span></span>
<span id="cb41-1503"><a href="#cb41-1503" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-1504"><a href="#cb41-1504" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.numeric), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">3</span>)))),</span>
<span id="cb41-1505"><a href="#cb41-1505" aria-hidden="true" tabindex="-1"></a>    <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{conf.low}, {conf.high}]"</span>)</span>
<span id="cb41-1506"><a href="#cb41-1506" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb41-1507"><a href="#cb41-1507" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Contrast"</span> <span class="ot">=</span> <span class="st">"contrast"</span>, <span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"estimate"</span>, <span class="st">"Term"</span> <span class="ot">=</span> <span class="st">"term"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1508"><a href="#cb41-1508" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_cols</span>(p_d) <span class="sc">|&gt;</span></span>
<span id="cb41-1509"><a href="#cb41-1509" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Contrast, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>, pd) <span class="sc">|&gt;</span></span>
<span id="cb41-1510"><a href="#cb41-1510" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-1511"><a href="#cb41-1511" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">digits =</span> <span class="dv">3</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1512"><a href="#cb41-1512" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-1513"><a href="#cb41-1513" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1514"><a href="#cb41-1514" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1515"><a href="#cb41-1515" aria-hidden="true" tabindex="-1"></a>In @tbl-ordbeta the credible interval is close enough to zero relative to its uncertainty that we can conclude there likely aren’t differences between the conditions after taking dispersion and the 0s and 1s in our data into account.</span>
<span id="cb41-1516"><a href="#cb41-1516" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1517"><a href="#cb41-1517" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Cutpoints</span></span>
<span id="cb41-1518"><a href="#cb41-1518" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1519"><a href="#cb41-1519" aria-hidden="true" tabindex="-1"></a>The model cutpoints are not reported by default in the summary output, but we can access them with the R package <span class="in">`posterior`</span> <span class="co">[</span><span class="ot">@posterior2025a</span><span class="co">]</span> and the functions <span class="in">`as_draws`</span> and <span class="in">`summary_draws`</span>.</span>
<span id="cb41-1520"><a href="#cb41-1520" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1523"><a href="#cb41-1523" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1524"><a href="#cb41-1524" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-cutpoints</span></span>
<span id="cb41-1525"><a href="#cb41-1525" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: Cutzero and cutone parameter summary</span></span>
<span id="cb41-1526"><a href="#cb41-1526" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb41-1527"><a href="#cb41-1527" aria-hidden="true" tabindex="-1"></a><span class="co">#| echo: false</span></span>
<span id="cb41-1528"><a href="#cb41-1528" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1529"><a href="#cb41-1529" aria-hidden="true" tabindex="-1"></a><span class="fu">as_draws</span>(m.phi, <span class="fu">c</span>(<span class="st">"cutzero"</span>, <span class="st">"cutone"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb41-1530"><a href="#cb41-1530" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarize_draws</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-1531"><a href="#cb41-1531" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(variable, mean, q5, q95) <span class="sc">|&gt;</span></span>
<span id="cb41-1532"><a href="#cb41-1532" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">c</span>(mean, q5, q95), <span class="sc">~</span> <span class="fu">round</span>(.x, <span class="dv">2</span>))) <span class="sc">|&gt;</span></span>
<span id="cb41-1533"><a href="#cb41-1533" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">str_glue</span>(<span class="st">"[{q5}, {q95}]"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb41-1534"><a href="#cb41-1534" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="st">"Parameter"</span> <span class="ot">=</span> <span class="st">"variable"</span>, <span class="st">"Mean"</span> <span class="ot">=</span> <span class="st">"mean"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1535"><a href="#cb41-1535" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(Parameter, Mean, <span class="st">`</span><span class="at">95% Cr.I</span><span class="st">`</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1536"><a href="#cb41-1536" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>() <span class="sc">|&gt;</span></span>
<span id="cb41-1537"><a href="#cb41-1537" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>)</span>
<span id="cb41-1538"><a href="#cb41-1538" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1539"><a href="#cb41-1539" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1540"><a href="#cb41-1540" aria-hidden="true" tabindex="-1"></a>In @tbl-cutpoints, <span class="in">`cutzero`</span> is the first cutpoint (the difference between 0 and continuous values) and <span class="in">`cutone`</span> is the second cutpoint (the difference between the continuous values and 1). These cutpoints are on the logit scale and as such the numbers do not have a simple substantive meaning. In general, as the cutpoints increase in absolute value (away from zero), then the discrete/boundary observations are more distinct from the continuous values. This will happen if there is a clear gap or bunching in the outcome around the bounds. This type of empirical feature of the distribution may be useful to scholars if they want to study differences in how people perceive the ends of the scale versus the middle. It is possible, though beyond the scope of this article, to model the location of the cutpoints with hierarchical (non-linear) covariates in <span class="in">`brms`</span>. In the most recent version of <span class="in">`ordbeta`</span>, it is possible to test the influence of different factors on these boundaries.</span>
<span id="cb41-1541"><a href="#cb41-1541" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1542"><a href="#cb41-1542" aria-hidden="true" tabindex="-1"></a><span class="fu">### Model Fit</span></span>
<span id="cb41-1543"><a href="#cb41-1543" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1544"><a href="#cb41-1544" aria-hidden="true" tabindex="-1"></a>The best way to visualize model fit is to plot the full predictive distribution relative to the original outcome. Because ordered beta regression is a mixed discrete/continuous model, a separate plotting function, <span class="in">`pp_check_ordbetareg`</span>, is included in the {ordbetareg} package that accurately handles the unique features of this distribution. The default plot in <span class="in">`brms`</span> will collapse these two features of the outcome together, which will make the fit look worse than it actually is. The {ordbetareg} function returns a list with two plots, <span class="in">`discrete`</span> and <span class="in">`continuous`</span>, which can either be printed and plotted or further modified as {ggplot2} objects (see @fig-ppcheckord).</span>
<span id="cb41-1545"><a href="#cb41-1545" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1548"><a href="#cb41-1548" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1549"><a href="#cb41-1549" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ppcheckord</span></span>
<span id="cb41-1550"><a href="#cb41-1550" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 12</span></span>
<span id="cb41-1551"><a href="#cb41-1551" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 8</span></span>
<span id="cb41-1552"><a href="#cb41-1552" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: Posterior predictive check for ordered beta regression model. A. Discrete posterior check. B. Continuous posterior check.</span></span>
<span id="cb41-1553"><a href="#cb41-1553" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1554"><a href="#cb41-1554" aria-hidden="true" tabindex="-1"></a>plots <span class="ot">&lt;-</span> <span class="fu">pp_check_ordbeta</span>(</span>
<span id="cb41-1555"><a href="#cb41-1555" aria-hidden="true" tabindex="-1"></a>  m.phi,</span>
<span id="cb41-1556"><a href="#cb41-1556" aria-hidden="true" tabindex="-1"></a>  <span class="at">ndraws =</span> <span class="dv">100</span>,</span>
<span id="cb41-1557"><a href="#cb41-1557" aria-hidden="true" tabindex="-1"></a>  <span class="at">outcome_label =</span> <span class="st">"Final Test Accuracy"</span></span>
<span id="cb41-1558"><a href="#cb41-1558" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-1559"><a href="#cb41-1559" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1560"><a href="#cb41-1560" aria-hidden="true" tabindex="-1"></a>plot <span class="ot">&lt;-</span> plots <span class="sc">+</span> <span class="fu">theme_publication</span>()</span>
<span id="cb41-1561"><a href="#cb41-1561" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1562"><a href="#cb41-1562" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_grid</span>(plots<span class="sc">$</span>discrete, plots<span class="sc">$</span>continuous, <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"A"</span>, <span class="st">"B"</span>))</span>
<span id="cb41-1563"><a href="#cb41-1563" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1564"><a href="#cb41-1564" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1565"><a href="#cb41-1565" aria-hidden="true" tabindex="-1"></a>The discrete plot, which is a bar graph, shows that the posterior distribution accurately captures the number of different types of responses (discrete or continuous) in the data. For the continuous plot shown as a density plot with one line per posterior draw, the model does a very good job at capturing the distribution.</span>
<span id="cb41-1566"><a href="#cb41-1566" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1567"><a href="#cb41-1567" aria-hidden="true" tabindex="-1"></a>Overall, it is clear from the posterior distribution plot that the ordered beta model fits the data well. To fully understand model fit, both of these plots need to be inspected as they are conceptually distinct.</span>
<span id="cb41-1568"><a href="#cb41-1568" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1569"><a href="#cb41-1569" aria-hidden="true" tabindex="-1"></a><span class="fu">### Model Visualization</span></span>
<span id="cb41-1570"><a href="#cb41-1570" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1571"><a href="#cb41-1571" aria-hidden="true" tabindex="-1"></a>{ordbetareg} provides a useful visualization function called <span class="in">`plot_heiss()`</span> <span class="co">[</span><span class="ot">@yeEnforcingBoundariesChinas2023</span><span class="co">]</span> that can represent dispersion in the entire outcome as a function of discrete covariates. This function produces a plot of predicted proportions across the range of our <span class="in">`Fluency`</span> factor. In @fig-heiss we get predicted proportions for Fluency across the bounded scale. Looking at the figure we can see there is much overlap between instructors in the middle portion ($\mu$) . However, we do see some small differences at the zero bounds.</span>
<span id="cb41-1572"><a href="#cb41-1572" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1575"><a href="#cb41-1575" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1576"><a href="#cb41-1576" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-heiss</span></span>
<span id="cb41-1577"><a href="#cb41-1577" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: "Heiss plot of predicted probabilities across the scale (0-100)"</span></span>
<span id="cb41-1578"><a href="#cb41-1578" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1579"><a href="#cb41-1579" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_heiss</span>(</span>
<span id="cb41-1580"><a href="#cb41-1580" aria-hidden="true" tabindex="-1"></a>  ord_fit_brms,</span>
<span id="cb41-1581"><a href="#cb41-1581" aria-hidden="true" tabindex="-1"></a>  <span class="at">grouping_fac =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb41-1582"><a href="#cb41-1582" aria-hidden="true" tabindex="-1"></a>  <span class="at">recode_group_labels =</span> <span class="fu">c</span>(<span class="st">"Disfluency"</span>, <span class="st">"Fluency"</span>),</span>
<span id="cb41-1583"><a href="#cb41-1583" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-1584"><a href="#cb41-1584" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1585"><a href="#cb41-1585" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1586"><a href="#cb41-1586" aria-hidden="true" tabindex="-1"></a><span class="fu">### Ordered Beta Scale</span></span>
<span id="cb41-1587"><a href="#cb41-1587" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1588"><a href="#cb41-1588" aria-hidden="true" tabindex="-1"></a>In the {ordbetareg} function there is a <span class="in">`true_bound`</span> argument. In cases where your data is not bounded between 0-1, this argument can be used to specify the bounds of the argument to fit the ordered beta regression. For example, the response data might be bounded between 1 and 7. If so, {ordbetareg} can model it within the <span class="sc">\[</span>0,1<span class="sc">\]</span> interval and {ordbetareg} will convert the model predictions back to the true bounds after estimation.</span>
<span id="cb41-1589"><a href="#cb41-1589" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1592"><a href="#cb41-1592" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1593"><a href="#cb41-1593" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: model_fits</span></span>
<span id="cb41-1594"><a href="#cb41-1594" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1595"><a href="#cb41-1595" aria-hidden="true" tabindex="-1"></a><span class="co"># fit models from ran models so tables and figures are in correct place.</span></span>
<span id="cb41-1596"><a href="#cb41-1596" aria-hidden="true" tabindex="-1"></a>model1 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"model_reg_bayes.rds"</span>))</span>
<span id="cb41-1597"><a href="#cb41-1597" aria-hidden="true" tabindex="-1"></a>model2 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"model_beta_bayes_dis_run01.rds"</span>))</span>
<span id="cb41-1598"><a href="#cb41-1598" aria-hidden="true" tabindex="-1"></a>model3 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_zib_model0not1.rds"</span>))</span>
<span id="cb41-1599"><a href="#cb41-1599" aria-hidden="true" tabindex="-1"></a>model4 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_zoib_model.rds"</span>))</span>
<span id="cb41-1600"><a href="#cb41-1600" aria-hidden="true" tabindex="-1"></a>model5 <span class="ot">&lt;-</span> <span class="fu">read_rds</span>(<span class="fu">here</span>(<span class="st">"manuscript"</span>, <span class="st">"models"</span>, <span class="st">"bayes_ordbeta_phi_model.rds"</span>))</span>
<span id="cb41-1601"><a href="#cb41-1601" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1602"><a href="#cb41-1602" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1605"><a href="#cb41-1605" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1606"><a href="#cb41-1606" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: tbl-all</span></span>
<span id="cb41-1607"><a href="#cb41-1607" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-cap: "Bayesian regression summaries for each model"</span></span>
<span id="cb41-1608"><a href="#cb41-1608" aria-hidden="true" tabindex="-1"></a><span class="co">#| tbl-align: "center"</span></span>
<span id="cb41-1609"><a href="#cb41-1609" aria-hidden="true" tabindex="-1"></a><span class="co">#| after-caption-space: 0pt</span></span>
<span id="cb41-1610"><a href="#cb41-1610" aria-hidden="true" tabindex="-1"></a><span class="co">#| apa-note: "Link functions: b_mean = logit; b_phi = log; b_zoi (zero-one inflation) = logit; b_coi (conditional one-inflation) = logit. Asterisks reflect approximate two-sided p-values derived from the posterior pd. pd ≥ 0.975 (p ≤ .05) = `*`; pd ≥ 0.990 (p ≤ .01) = `**`; pd ≥ 0.998 (p ≤ .001) = `***`."</span></span>
<span id="cb41-1611"><a href="#cb41-1611" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1612"><a href="#cb41-1612" aria-hidden="true" tabindex="-1"></a><span class="co"># Define your models</span></span>
<span id="cb41-1613"><a href="#cb41-1613" aria-hidden="true" tabindex="-1"></a>model_list <span class="ot">&lt;-</span> <span class="fu">list</span>(</span>
<span id="cb41-1614"><a href="#cb41-1614" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Bayesian LM"</span> <span class="ot">=</span> model1,</span>
<span id="cb41-1615"><a href="#cb41-1615" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Beta Regression"</span> <span class="ot">=</span> model2,</span>
<span id="cb41-1616"><a href="#cb41-1616" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ZIB"</span> <span class="ot">=</span> model3,</span>
<span id="cb41-1617"><a href="#cb41-1617" aria-hidden="true" tabindex="-1"></a>  <span class="st">"ZOIB"</span> <span class="ot">=</span> model4,</span>
<span id="cb41-1618"><a href="#cb41-1618" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Ordered Beta"</span> <span class="ot">=</span> model5</span>
<span id="cb41-1619"><a href="#cb41-1619" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb41-1620"><a href="#cb41-1620" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1621"><a href="#cb41-1621" aria-hidden="true" tabindex="-1"></a><span class="co"># Process each model</span></span>
<span id="cb41-1622"><a href="#cb41-1622" aria-hidden="true" tabindex="-1"></a>cleaned_tables <span class="ot">&lt;-</span> <span class="fu">imap</span>(model_list, <span class="cf">function</span>(mod, name) {</span>
<span id="cb41-1623"><a href="#cb41-1623" aria-hidden="true" tabindex="-1"></a>  <span class="fu">model_parameters</span>(mod, <span class="at">centrality =</span> <span class="st">"mean"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1624"><a href="#cb41-1624" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb41-1625"><a href="#cb41-1625" aria-hidden="true" tabindex="-1"></a>      <span class="at">Mean =</span> <span class="fu">sprintf</span>(<span class="st">"%.3f"</span>, Mean),</span>
<span id="cb41-1626"><a href="#cb41-1626" aria-hidden="true" tabindex="-1"></a>      <span class="at">Cr.I =</span> <span class="fu">str_glue</span>(<span class="st">"[{round(CI_low, 3)}, {round(CI_high, 3)}]"</span>),</span>
<span id="cb41-1627"><a href="#cb41-1627" aria-hidden="true" tabindex="-1"></a>      <span class="at">pd_star =</span> <span class="fu">case_when</span>(</span>
<span id="cb41-1628"><a href="#cb41-1628" aria-hidden="true" tabindex="-1"></a>        pd <span class="sc">&gt;=</span> <span class="fl">0.995</span> <span class="sc">~</span> <span class="fu">str_glue</span>(<span class="st">"{sprintf('%.3f', pd)}***"</span>),</span>
<span id="cb41-1629"><a href="#cb41-1629" aria-hidden="true" tabindex="-1"></a>        pd <span class="sc">&gt;=</span> <span class="fl">0.985</span> <span class="sc">~</span> <span class="fu">str_glue</span>(<span class="st">"{sprintf('%.3f', pd)}**"</span>),</span>
<span id="cb41-1630"><a href="#cb41-1630" aria-hidden="true" tabindex="-1"></a>        pd <span class="sc">&gt;=</span> <span class="fl">0.975</span> <span class="sc">~</span> <span class="fu">str_glue</span>(<span class="st">"{sprintf('%.3f', pd)}*"</span>),</span>
<span id="cb41-1631"><a href="#cb41-1631" aria-hidden="true" tabindex="-1"></a>        <span class="cn">TRUE</span> <span class="sc">~</span> <span class="fu">sprintf</span>(<span class="st">"%.3f"</span>, pd)</span>
<span id="cb41-1632"><a href="#cb41-1632" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb41-1633"><a href="#cb41-1633" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span></span>
<span id="cb41-1634"><a href="#cb41-1634" aria-hidden="true" tabindex="-1"></a>    <span class="fu">pivot_longer</span>(</span>
<span id="cb41-1635"><a href="#cb41-1635" aria-hidden="true" tabindex="-1"></a>      <span class="at">cols =</span> <span class="fu">c</span>(Mean, Cr.I, pd_star),</span>
<span id="cb41-1636"><a href="#cb41-1636" aria-hidden="true" tabindex="-1"></a>      <span class="at">names_to =</span> <span class="st">"Stat"</span>,</span>
<span id="cb41-1637"><a href="#cb41-1637" aria-hidden="true" tabindex="-1"></a>      <span class="at">values_to =</span> <span class="st">"value"</span></span>
<span id="cb41-1638"><a href="#cb41-1638" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span></span>
<span id="cb41-1639"><a href="#cb41-1639" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(</span>
<span id="cb41-1640"><a href="#cb41-1640" aria-hidden="true" tabindex="-1"></a>      <span class="at">Stat =</span> <span class="fu">recode</span>(Stat, <span class="at">pd_star =</span> <span class="st">"pd"</span>),</span>
<span id="cb41-1641"><a href="#cb41-1641" aria-hidden="true" tabindex="-1"></a>      <span class="at">Model =</span> name</span>
<span id="cb41-1642"><a href="#cb41-1642" aria-hidden="true" tabindex="-1"></a>    ) <span class="sc">|&gt;</span></span>
<span id="cb41-1643"><a href="#cb41-1643" aria-hidden="true" tabindex="-1"></a>    <span class="fu">select</span>(Parameter, Stat, Model, value)</span>
<span id="cb41-1644"><a href="#cb41-1644" aria-hidden="true" tabindex="-1"></a>})</span>
<span id="cb41-1645"><a href="#cb41-1645" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1646"><a href="#cb41-1646" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine all tables</span></span>
<span id="cb41-1647"><a href="#cb41-1647" aria-hidden="true" tabindex="-1"></a>pretty_table <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(cleaned_tables) <span class="sc">|&gt;</span></span>
<span id="cb41-1648"><a href="#cb41-1648" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> Model, <span class="at">values_from =</span> value) <span class="sc">|&gt;</span></span>
<span id="cb41-1649"><a href="#cb41-1649" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">where</span>(is.character), <span class="sc">~</span> <span class="fu">replace_na</span>(.x, <span class="st">"-"</span>))) <span class="sc">|&gt;</span></span>
<span id="cb41-1650"><a href="#cb41-1650" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1651"><a href="#cb41-1651" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Simplify parameter names</span></span>
<span id="cb41-1652"><a href="#cb41-1652" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Parameter =</span> <span class="fu">str_replace_all</span>(Parameter, <span class="st">"FluencyFluent"</span>, <span class="st">"Fluency"</span>)) <span class="sc">|&gt;</span></span>
<span id="cb41-1653"><a href="#cb41-1653" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1654"><a href="#cb41-1654" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Set factor levels</span></span>
<span id="cb41-1655"><a href="#cb41-1655" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(</span>
<span id="cb41-1656"><a href="#cb41-1656" aria-hidden="true" tabindex="-1"></a>    <span class="at">Stat =</span> <span class="fu">factor</span>(Stat, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"Mean"</span>, <span class="st">"Cr.I"</span>, <span class="st">"pd"</span>)),</span>
<span id="cb41-1657"><a href="#cb41-1657" aria-hidden="true" tabindex="-1"></a>    <span class="at">Parameter =</span> <span class="fu">factor</span>(</span>
<span id="cb41-1658"><a href="#cb41-1658" aria-hidden="true" tabindex="-1"></a>      Parameter,</span>
<span id="cb41-1659"><a href="#cb41-1659" aria-hidden="true" tabindex="-1"></a>      <span class="at">levels =</span> <span class="fu">c</span>(</span>
<span id="cb41-1660"><a href="#cb41-1660" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_Intercept"</span>,</span>
<span id="cb41-1661"><a href="#cb41-1661" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_Fluency"</span>,</span>
<span id="cb41-1662"><a href="#cb41-1662" aria-hidden="true" tabindex="-1"></a>        <span class="st">"sigma"</span>,</span>
<span id="cb41-1663"><a href="#cb41-1663" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_phi_Intercept"</span>,</span>
<span id="cb41-1664"><a href="#cb41-1664" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_phi_Fluency"</span>,</span>
<span id="cb41-1665"><a href="#cb41-1665" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_zi_Intercept"</span>,</span>
<span id="cb41-1666"><a href="#cb41-1666" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_zi_Fluency"</span>,</span>
<span id="cb41-1667"><a href="#cb41-1667" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_zoi_Intercept"</span>,</span>
<span id="cb41-1668"><a href="#cb41-1668" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_zoi_Fluency"</span>,</span>
<span id="cb41-1669"><a href="#cb41-1669" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_coi_Intercept"</span>,</span>
<span id="cb41-1670"><a href="#cb41-1670" aria-hidden="true" tabindex="-1"></a>        <span class="st">"b_coi_Fluency"</span></span>
<span id="cb41-1671"><a href="#cb41-1671" aria-hidden="true" tabindex="-1"></a>      )</span>
<span id="cb41-1672"><a href="#cb41-1672" aria-hidden="true" tabindex="-1"></a>    )</span>
<span id="cb41-1673"><a href="#cb41-1673" aria-hidden="true" tabindex="-1"></a>  ) <span class="sc">|&gt;</span></span>
<span id="cb41-1674"><a href="#cb41-1674" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1675"><a href="#cb41-1675" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Parameter, Stat) <span class="sc">|&gt;</span></span>
<span id="cb41-1676"><a href="#cb41-1676" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Parameter) <span class="sc">|&gt;</span></span>
<span id="cb41-1677"><a href="#cb41-1677" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Parameter =</span> <span class="fu">if_else</span>(<span class="fu">row_number</span>() <span class="sc">==</span> <span class="dv">1</span>, <span class="fu">as.character</span>(Parameter), <span class="st">""</span>)) <span class="sc">|&gt;</span></span>
<span id="cb41-1678"><a href="#cb41-1678" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span>
<span id="cb41-1679"><a href="#cb41-1679" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1680"><a href="#cb41-1680" aria-hidden="true" tabindex="-1"></a><span class="co"># Display as styled table</span></span>
<span id="cb41-1681"><a href="#cb41-1681" aria-hidden="true" tabindex="-1"></a>pretty_table <span class="sc">|&gt;</span></span>
<span id="cb41-1682"><a href="#cb41-1682" aria-hidden="true" tabindex="-1"></a>  <span class="fu">tt</span>(<span class="at">width =</span> <span class="fl">0.95</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1683"><a href="#cb41-1683" aria-hidden="true" tabindex="-1"></a>  <span class="fu">format_tt</span>(<span class="at">escape =</span> <span class="cn">TRUE</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1684"><a href="#cb41-1684" aria-hidden="true" tabindex="-1"></a>  <span class="fu">style_tt</span>(<span class="at">fontsize =</span> .<span class="dv">9</span>, <span class="at">align =</span> <span class="st">"c"</span>) <span class="sc">|&gt;</span></span>
<span id="cb41-1685"><a href="#cb41-1685" aria-hidden="true" tabindex="-1"></a>  <span class="fu">style_tt</span>(</span>
<span id="cb41-1686"><a href="#cb41-1686" aria-hidden="true" tabindex="-1"></a>    <span class="at">i =</span> <span class="fu">c</span>(</span>
<span id="cb41-1687"><a href="#cb41-1687" aria-hidden="true" tabindex="-1"></a>      <span class="dv">2</span><span class="sc">:</span><span class="dv">3</span>,</span>
<span id="cb41-1688"><a href="#cb41-1688" aria-hidden="true" tabindex="-1"></a>      <span class="dv">5</span><span class="sc">:</span><span class="dv">6</span>,</span>
<span id="cb41-1689"><a href="#cb41-1689" aria-hidden="true" tabindex="-1"></a>      <span class="dv">8</span><span class="sc">:</span><span class="dv">9</span>,</span>
<span id="cb41-1690"><a href="#cb41-1690" aria-hidden="true" tabindex="-1"></a>      <span class="dv">11</span><span class="sc">:</span><span class="dv">12</span>,</span>
<span id="cb41-1691"><a href="#cb41-1691" aria-hidden="true" tabindex="-1"></a>      <span class="dv">14</span><span class="sc">:</span><span class="dv">15</span>,</span>
<span id="cb41-1692"><a href="#cb41-1692" aria-hidden="true" tabindex="-1"></a>      <span class="dv">17</span><span class="sc">:</span><span class="dv">18</span>,</span>
<span id="cb41-1693"><a href="#cb41-1693" aria-hidden="true" tabindex="-1"></a>      <span class="dv">20</span><span class="sc">:</span><span class="dv">21</span>,</span>
<span id="cb41-1694"><a href="#cb41-1694" aria-hidden="true" tabindex="-1"></a>      <span class="dv">23</span><span class="sc">:</span><span class="dv">24</span>,</span>
<span id="cb41-1695"><a href="#cb41-1695" aria-hidden="true" tabindex="-1"></a>      <span class="dv">26</span><span class="sc">:</span><span class="dv">27</span>,</span>
<span id="cb41-1696"><a href="#cb41-1696" aria-hidden="true" tabindex="-1"></a>      <span class="dv">29</span><span class="sc">:</span><span class="dv">30</span>,</span>
<span id="cb41-1697"><a href="#cb41-1697" aria-hidden="true" tabindex="-1"></a>      <span class="dv">32</span><span class="sc">:</span><span class="dv">33</span></span>
<span id="cb41-1698"><a href="#cb41-1698" aria-hidden="true" tabindex="-1"></a>    ),</span>
<span id="cb41-1699"><a href="#cb41-1699" aria-hidden="true" tabindex="-1"></a>    <span class="at">italic =</span> <span class="cn">TRUE</span>,</span>
<span id="cb41-1700"><a href="#cb41-1700" aria-hidden="true" tabindex="-1"></a>    <span class="at">fontsize =</span> .<span class="dv">6</span></span>
<span id="cb41-1701"><a href="#cb41-1701" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-1702"><a href="#cb41-1702" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1703"><a href="#cb41-1703" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1706"><a href="#cb41-1706" aria-hidden="true" tabindex="-1"></a><span class="in">```{r}</span></span>
<span id="cb41-1707"><a href="#cb41-1707" aria-hidden="true" tabindex="-1"></a><span class="co">#| label: fig-ppcheckall</span></span>
<span id="cb41-1708"><a href="#cb41-1708" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-cap: The plots show 100 posterior predicted distributions with the label yrep (light blue), the distribution of accuracy as function of fluency in dark blue for regular regression (A) , beta regression (B), ZIB (C) and ZOIB (D).</span></span>
<span id="cb41-1709"><a href="#cb41-1709" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-height: 20</span></span>
<span id="cb41-1710"><a href="#cb41-1710" aria-hidden="true" tabindex="-1"></a><span class="co">#| fig-width: 8</span></span>
<span id="cb41-1711"><a href="#cb41-1711" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1712"><a href="#cb41-1712" aria-hidden="true" tabindex="-1"></a>pp_check1 <span class="ot">&lt;-</span> <span class="fu">pp_check</span>(</span>
<span id="cb41-1713"><a href="#cb41-1713" aria-hidden="true" tabindex="-1"></a>  model1,</span>
<span id="cb41-1714"><a href="#cb41-1714" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">"dens_overlay_grouped"</span>,</span>
<span id="cb41-1715"><a href="#cb41-1715" aria-hidden="true" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb41-1716"><a href="#cb41-1716" aria-hidden="true" tabindex="-1"></a>  <span class="at">ndraws =</span> <span class="dv">100</span></span>
<span id="cb41-1717"><a href="#cb41-1717" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb41-1718"><a href="#cb41-1718" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>()</span>
<span id="cb41-1719"><a href="#cb41-1719" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1720"><a href="#cb41-1720" aria-hidden="true" tabindex="-1"></a>pp_check2 <span class="ot">&lt;-</span> <span class="fu">pp_check</span>(</span>
<span id="cb41-1721"><a href="#cb41-1721" aria-hidden="true" tabindex="-1"></a>  model2,</span>
<span id="cb41-1722"><a href="#cb41-1722" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">"dens_overlay_grouped"</span>,</span>
<span id="cb41-1723"><a href="#cb41-1723" aria-hidden="true" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb41-1724"><a href="#cb41-1724" aria-hidden="true" tabindex="-1"></a>  <span class="at">ndraws =</span> <span class="dv">100</span></span>
<span id="cb41-1725"><a href="#cb41-1725" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb41-1726"><a href="#cb41-1726" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb41-1727"><a href="#cb41-1727" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb41-1728"><a href="#cb41-1728" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">6</span>)) <span class="co"># example limit – adjust as needed</span></span>
<span id="cb41-1729"><a href="#cb41-1729" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1730"><a href="#cb41-1730" aria-hidden="true" tabindex="-1"></a>pp_check3 <span class="ot">&lt;-</span> <span class="fu">pp_check</span>(</span>
<span id="cb41-1731"><a href="#cb41-1731" aria-hidden="true" tabindex="-1"></a>  model3,</span>
<span id="cb41-1732"><a href="#cb41-1732" aria-hidden="true" tabindex="-1"></a>  <span class="at">type =</span> <span class="st">"dens_overlay_grouped"</span>,</span>
<span id="cb41-1733"><a href="#cb41-1733" aria-hidden="true" tabindex="-1"></a>  <span class="at">group =</span> <span class="st">"Fluency"</span>,</span>
<span id="cb41-1734"><a href="#cb41-1734" aria-hidden="true" tabindex="-1"></a>  <span class="at">ndraws =</span> <span class="dv">100</span></span>
<span id="cb41-1735"><a href="#cb41-1735" aria-hidden="true" tabindex="-1"></a>) <span class="sc">+</span></span>
<span id="cb41-1736"><a href="#cb41-1736" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_publication</span>() <span class="sc">+</span></span>
<span id="cb41-1737"><a href="#cb41-1737" aria-hidden="true" tabindex="-1"></a>  <span class="fu">xlim</span>(<span class="dv">0</span>, <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb41-1738"><a href="#cb41-1738" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_cartesian</span>(<span class="at">ylim =</span> <span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">6</span>))</span>
<span id="cb41-1739"><a href="#cb41-1739" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1740"><a href="#cb41-1740" aria-hidden="true" tabindex="-1"></a><span class="co"># Use `plot_spacer()` for the empty 4th spot</span></span>
<span id="cb41-1741"><a href="#cb41-1741" aria-hidden="true" tabindex="-1"></a>combined_plot <span class="ot">&lt;-</span> (pp_check1 <span class="sc">/</span> pp_check2 <span class="sc">/</span> pp_check3) <span class="sc">+</span></span>
<span id="cb41-1742"><a href="#cb41-1742" aria-hidden="true" tabindex="-1"></a>  <span class="fu">plot_annotation</span>(</span>
<span id="cb41-1743"><a href="#cb41-1743" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag_levels =</span> <span class="st">'A'</span>,</span>
<span id="cb41-1744"><a href="#cb41-1744" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag_prefix =</span> <span class="st">"("</span>,</span>
<span id="cb41-1745"><a href="#cb41-1745" aria-hidden="true" tabindex="-1"></a>    <span class="at">tag_suffix =</span> <span class="st">")"</span></span>
<span id="cb41-1746"><a href="#cb41-1746" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb41-1747"><a href="#cb41-1747" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1748"><a href="#cb41-1748" aria-hidden="true" tabindex="-1"></a>combined_plot</span>
<span id="cb41-1749"><a href="#cb41-1749" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb41-1750"><a href="#cb41-1750" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1751"><a href="#cb41-1751" aria-hidden="true" tabindex="-1"></a><span class="fu"># Discussion</span></span>
<span id="cb41-1752"><a href="#cb41-1752" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1753"><a href="#cb41-1753" aria-hidden="true" tabindex="-1"></a>The use of beta regression in psychology, and the social sciences in general, is rare. With this tutorial, we hope to turn the tides. Beta regression models are an attractive alternative to models that impose unrealistic assumptions like normality, linearity, homoscedasticity, and unbounded data. Beyond these models, there are a diverse array of different models that can be used depending on your outcome of interest.</span>
<span id="cb41-1754"><a href="#cb41-1754" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1755"><a href="#cb41-1755" aria-hidden="true" tabindex="-1"></a>Throughout this tutorial our main aim was to help guide researchers in running analyses with proportional or percentage outcomes using beta regression and some of its alternatives. In the current example, we used real data from @wilford2020 and discussed how to fit these models in R, interpret model parameters, extract predicted probabilities and marginal effects, and visualize the results using the {marginaleffects} package.</span>
<span id="cb41-1756"><a href="#cb41-1756" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1757"><a href="#cb41-1757" aria-hidden="true" tabindex="-1"></a>Comparing our analysis with that of @wilford2020, we demonstrated that using a traditional approache (e.g., *t*-test) to analyze mean accuracy data can lead to biased inferences. Although we successfully reproduced one of their key findings, our use of beta regression and its extensions revealed important nuances in the results. With a traditional beta regression model–which accounts for both the mean and the precision (dispersion)–we observed similar effects of instructor fluency on performance. However, the standard beta model does not accommodate boundary values (i.e., 0s and 1s).</span>
<span id="cb41-1758"><a href="#cb41-1758" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1759"><a href="#cb41-1759" aria-hidden="true" tabindex="-1"></a>When we applied a ZIB model, which explicitly accounts for structural 0s, we found no effect of fluency on the mean ($\mu$) part of the model. Instead, the effect of fluency emerged in the structural zero (inflated zero; $\alpha$) component. This pattern was consistent when using a zero-one-inflated beta (ZOIB) model. Furthermore, we fit an ordered beta regression model <span class="co">[</span><span class="ot">@kubinec2022</span><span class="co">]</span>, which appropriately models the full range of values, including 0s and 1s. Here, we did not observe a reliable effect of fluency on the mean once we accounted for dispersion.</span>
<span id="cb41-1760"><a href="#cb41-1760" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1761"><a href="#cb41-1761" aria-hidden="true" tabindex="-1"></a>These analyses emphasize the importance of fitting a model that aligns with the nature of the data. The simplest and recommended approach when dealing with data that contains 0s and/or 1s is to fit an ordered beta model, assuming the process is truly continuous. However, if you believe the process is distinct in nature, a ZIB or ZOIB model might be a better choice. Ultimately, this decision should be guided by theory.</span>
<span id="cb41-1762"><a href="#cb41-1762" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1763"><a href="#cb41-1763" aria-hidden="true" tabindex="-1"></a>For instance, if we believe fluency influences the boundaries (0 and 1), we might want to model this process separately using a ZIB or ZOIB. With the current dataset, fluency might affect specific aspects of performance (such as the likelihood of complete failure) rather than general performance levels. This effect could be due to participant disengagement during the disfluent lecture. If students fail to pay attention because of features of disfluency, they may miss relevant information, leading to a floor effect at the test. Follwoing from this, disfluency would be expected to influence the boundary (O) and not the continuous part of the model. If this is the case, we would want to model this appropriately. However, if we believe fluency effects general performance levels (the continuous part), a model that takes in to account the entire process accounting for the 0s and 1s might be appropriate.</span>
<span id="cb41-1764"><a href="#cb41-1764" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1765"><a href="#cb41-1765" aria-hidden="true" tabindex="-1"></a>In the discussion section of @wilford2020, they were unable to offer a tenable explanation for performance differences based on instructor fluency. A model that accounts for the excess 0s in the dataset provides one testable explanation: watching a disfluent lecture may lead to lapses in attention, resulting in poorer performance in that group. These lapses, in turn, contribute to the observed differences in the fluent condition. This modeling approach opens a promising avenue for future research--one that would have remained inaccessible otherwise.</span>
<span id="cb41-1766"><a href="#cb41-1766" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1767"><a href="#cb41-1767" aria-hidden="true" tabindex="-1"></a>Not everyone will be eager to implement the techniques discussed herein. In such cases, the key question becomes: What is the least problematic approach to handling proportional data? One reasonable option is to fit multiple models tailored to the specific characteristics of your data. For example, if your data contain 0s, you might fit two models: a traditional linear model excluding the 0s, and a logistic model to account for the zero versus non-zero distinction. If your data contain both 0s and 1s, you could fit separate models for the 0s and 1s in addition to the OLS model. There are many defensible strategies to choose from depending on the context. However, we do not recommend transforming the values of your data (e.g., 0s to .01 and 1s to .99) or ignoring the properties of your data simply to fit traditional statistical models.</span>
<span id="cb41-1768"><a href="#cb41-1768" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1769"><a href="#cb41-1769" aria-hidden="true" tabindex="-1"></a>In this tutorial, we demonstrated how to analyze these models from a Bayesian perspective. While we recognize that not everyone identifies as a Bayesian, implementing these models using a Bayesian framework is relatively straightforward--it requires only a single package, lowering the barrier to entry. For those who prefer frequentist analyses, several R packages are available. For example, the {betareg} package <span class="co">[</span><span class="ot">@betareg</span><span class="co">]</span> {glmmTMB} <span class="co">[</span><span class="ot">@glmmTMB</span><span class="co">]</span> and {gamlss} <span class="co">[</span><span class="ot">@gamlss</span><span class="co">]</span> are nice options. To this end, I have included supplemental materials demonstrating how to use frequentist packages to analyze the data presented herein.</span>
<span id="cb41-1770"><a href="#cb41-1770" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1771"><a href="#cb41-1771" aria-hidden="true" tabindex="-1"></a><span class="fu">## Conclusion</span></span>
<span id="cb41-1772"><a href="#cb41-1772" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1773"><a href="#cb41-1773" aria-hidden="true" tabindex="-1"></a>Overall, this tutorial emphasizes the importance of modeling the data you have. Although the example provided is relatively simple (a one-factor model with two levels), we hope it demonstrates that even with a basic dataset, there is much nuance in interpretation and inference. Properly modeling your data can lead to deeper insights, far beyond what traditional measures might offer. With the tools introduced in this tutorial, researchers now have the means to analyze their data effectively, uncover patterns, make accurate predictions, and support their findings with robust statistical evidence. By applying these modeling techniques, researchers can improve the validity and reliability of their studies, ultimately leading to more informed decisions and advancements in their respective fields.</span>
<span id="cb41-1774"><a href="#cb41-1774" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1775"><a href="#cb41-1775" aria-hidden="true" tabindex="-1"></a><span class="fu"># References</span></span>
<span id="cb41-1776"><a href="#cb41-1776" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb41-1777"><a href="#cb41-1777" aria-hidden="true" tabindex="-1"></a>::: {#refs}</span>
<span id="cb41-1778"><a href="#cb41-1778" aria-hidden="true" tabindex="-1"></a>:::</span>
</code><button title="Copy to Clipboard" class="code-copy-button" data-in-quarto-modal=""><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->




</body></html>